<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.53">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<meta name="author" content="Harmon Bhasin">
<meta name="dcterms.date" content="2024-07-08">
<title>Workflow of Cebria-Mendoza et al.&nbsp;(2021) – Harmon's Public Notebook</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>

<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light"><script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><style>

      .quarto-title-block .quarto-title-banner {
        background: #de2d26;
      }
</style>
<script src="../site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script><script src="../site_libs/d3-4.5.0/d3.min.js"></script><script src="../site_libs/sankey-1/sankey.js"></script><script src="../site_libs/sankeyNetwork-binding-0.4/sankeyNetwork.js"></script><link href="../site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet">
<script src="../site_libs/pagedtable-1.1/js/pagedtable.js"></script><script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script><script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script><script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script><link rel="stylesheet" href="../styles.css">
</head>
<body class="nav-fixed fullcontent">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top quarto-banner"><nav class="navbar navbar-expand-lg " data-bs-theme="dark"><div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">Harmon’s Public Notebook</span>
    </a>
  </div>
        <div class="quarto-navbar-tools tools-end">
</div>
          <div id="quarto-search" class="" title="Search"></div>
      </div> <!-- /container-fluid -->
    </nav></header><!-- content --><header id="title-block-header" class="quarto-title-block default page-columns page-full"><div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <div class="quarto-title-block"><div><h1 class="title">Workflow of Cebria-Mendoza et al.&nbsp;(2021)</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
            <p class="subtitle lead">Pooled plasma from Spain (DNA + RNA)</p>
                      </div>
  </div>
    
  
  <div class="quarto-title-meta">

      <div>
      <div class="quarto-title-meta-heading">Author</div>
      <div class="quarto-title-meta-contents">
               <p>Harmon Bhasin </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">July 8, 2024</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content"><p>As a part of my time here, I’m exploring blood-based surveillance for novel viral pathogen detection. We’ve decided to make three blog posts. The first blog post will cover the physical and biological characteristics of blood, the second blog post will cover the blood sampling strategies, and the last blog post will look at the metagenomic profile of blood (which is what we’re doing here, and will do for the next few posts).</p>
<p>This is the first of many studies that I’ll be analyzing. however I don’t expect all of them to be used in the last blog post. In this post, I analyze <a href="https://doi.org/10.3390/v13112322">Cebria-Mendoza 2021</a>, a dataset with 60 samples where each sample is a pool of 8-13 unique healthy donors, with a total of ~600 donors from Spain.</p>
<section id="the-raw-data" class="level1"><h1>The raw data</h1>
<section id="about" class="level2"><h2 class="anchored" data-anchor-id="about">About</h2>
<p>This dataset is composed of 60 samples which come from plasma pools of anywhere from 8-13 people from Spain. In total, there were 567 healthy individuals that have contributed to these pools. The demographic information that I have for each pool is the gender and age of the donor as well as the number that has contributed to each pool. For each pooled sample, they performed a combined DNA and RNA library preparation, resulting in a single sequencing output that captures both nucleic acid types. This approach provides comprehensive genetic information but precludes separate analysis of DNA and RNA from individual samples.</p>
<section id="sample-library-preparation" class="level3"><h3 class="anchored" data-anchor-id="sample-library-preparation">Sample + library preparation</h3>
<p>A total of 587 plasma samples from healthy donors were collected from the Centro de Transfusión de la Comunidad Valenciana (Valencia, Spain) from 15 September 2018 to 30 March 2019 and stored at −80 °C until use.</p>
<p>Each of the 60 pools (SP1-SP60) analyzed in this study was obtained by mixing 1 mL of plasma from a variable number of donors (between 8- and 13-mL total). To assess viral recovery, each pool was spiked with 103 PFU of ϕX174 and 104 PFU of vesicular stomatitis virus (VSV). The purification protocol has been previously described in detail [18]. Briefly, plasma pools were processed with 1.0 µM filters to remove cells and other non-viral particles and the filtered fractions were subject to high-speed centrifugation (87,000 g, 2 h, 4 °C), washed with PBS 1X (87,000 g, 1 h, 4 °C), and resuspended in 245 µL 1X digestion buffer (Turbo DNA Free kit, Ambion, Carlsbad, CA, USA). Then, 5 µL of Turbo DNase, 2 µL of Benzonase (Sigma, Darmstadt, Germany) and 2 µL of micrococcal nuclease (NEB) were added to the sample to remove unprotected nucleic acids. After incubation (1 h, 37 °C), 20 µL of stop reagent was added, following the manufacturer’s instructions. Then, 240 µL supernatant was transferred to a new tube and split into two fractions: 200 µL fraction was used for RNA extraction using TRIzol LS reagent (Invitrogen, Carlsbad, USA), followed by purification with the QIAamp Viral RNA Mini kit (Qiagen, Hilden, Germany) and amplification with the QuantiTect Whole Transcriptome kit (Qiagen), and 40 µL fraction was used for DNA extraction with the QIAamp Viral RNA Mini kit and amplification with the TruePrime WGA kit (Sygnis, Heidelberg, Germany).</p>
<p>For each pool, DNA and RNA amplification products were mixed in equimolar concentration before library preparation, which was carried out using Nextera XT DNA library preparation kit with 15 amplification cycles (Illumina, San Diego, USA), and subject to pair-end sequencing in a NextSeq device.</p>
<p><em>More details can be found in their <a href="https://doi.org/10.1038%2Fs41598-021-86427-4">original paper</a> which outlined this protocol.</em></p>
</section></section><section id="quality-control" class="level2"><h2 class="anchored" data-anchor-id="quality-control">Quality control</h2>
<p>In total, these 60 samples contained 230M read pairs. The samples had 2.3M - 4.8M (mean 3.8M) read pairs each.</p>
<p>The number of read pairs and total base pairs looks good, however the duplication rate is quite high (this can be attributed to their sample preparation).</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb1"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Prepare data</span></span>
<span><span class="va">basic_stats_raw_metrics</span> <span class="op">&lt;-</span> <span class="va">basic_stats_raw</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">library</span>,</span>
<span>         <span class="va">pool_size</span>,</span>
<span>         `# Read pairs` <span class="op">=</span> <span class="va">n_read_pairs</span>,</span>
<span>         `Total base pairs\n(approx)` <span class="op">=</span> <span class="va">n_bases_approx</span>,</span>
<span>         `% Duplicates\n(FASTQC)` <span class="op">=</span> <span class="va">percent_duplicates</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_longer</span><span class="op">(</span><span class="op">-</span><span class="op">(</span><span class="va">library</span><span class="op">:</span><span class="va">pool_size</span><span class="op">)</span>, names_to <span class="op">=</span> <span class="st">"metric"</span>, values_to <span class="op">=</span> <span class="st">"value"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>metric <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">metric</span><span class="op">)</span><span class="op">)</span> </span>
<span></span>
<span><span class="co"># Set up plot templates</span></span>
<span></span>
<span><span class="va">g_basic</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">basic_stats_raw_metrics</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">library</span>, y <span class="op">=</span> <span class="va">value</span>, fill <span class="op">=</span> <span class="va">pool_size</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>position <span class="op">=</span> <span class="st">"dodge"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_discrete</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">expand_limits</span><span class="op">(</span>y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">100</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Accent"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">metric</span> <span class="op">~</span> <span class="va">.</span>, scales <span class="op">=</span> <span class="st">"free"</span>, space <span class="op">=</span> <span class="st">"free_x"</span>, switch <span class="op">=</span> <span class="st">"y"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span></span>
<span>    axis.title.y <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    strip.text.y <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>face <span class="op">=</span> <span class="st">"plain"</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span><span class="va">g_basic</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-08_cebria-mendoza_files/figure-html/plot-basic-stats-1.png" class="img-fluid figure-img" width="864"></p>
</figure>
</div>
</div>
</div>
<p>Adapter content is low (the upper limit on the plot is 1%). Read quality seems good both over all the positions as well as over the number of sequences.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb2"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Set up plotting templates</span></span>
<span><span class="va">g_qual_raw</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span>mapping<span class="op">=</span><span class="fu">aes</span><span class="op">(</span>linetype<span class="op">=</span><span class="va">read_pair</span>, group<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/interaction.html">interaction</a></span><span class="op">(</span><span class="va">sample</span>,<span class="va">read_pair</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">scale_linetype_discrete</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Read Pair"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>color<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">2</span>,byrow<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>         linetype <span class="op">=</span> <span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">2</span>,byrow<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span></span>
<span></span>
<span><span class="co"># Visualize adapters</span></span>
<span><span class="va">g_adapters_raw</span> <span class="op">&lt;-</span> <span class="va">g_qual_raw</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">position</span>, y<span class="op">=</span><span class="va">pc_adapters</span><span class="op">)</span>, data<span class="op">=</span><span class="va">adapter_stats_raw</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% Adapters"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Position"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span>,</span>
<span>                     breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">500</span>,<span class="fl">20</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">.</span><span class="op">~</span><span class="va">adapter</span><span class="op">)</span></span>
<span><span class="va">g_adapters_raw</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-08_cebria-mendoza_files/figure-html/plot-raw-quality-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb3"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Visualize quality</span></span>
<span><span class="va">g_quality_base_raw</span> <span class="op">&lt;-</span> <span class="va">g_qual_raw</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept<span class="op">=</span><span class="fl">25</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept<span class="op">=</span><span class="fl">30</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">position</span>, y<span class="op">=</span><span class="va">mean_phred_score</span><span class="op">)</span>, data<span class="op">=</span><span class="va">quality_base_stats_raw</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Mean Phred score"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">10</span>,<span class="fl">45</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Position"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span>,</span>
<span>                     breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">500</span>,<span class="fl">20</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_quality_base_raw</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-08_cebria-mendoza_files/figure-html/plot-raw-quality-2.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb4"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_quality_seq_raw</span> <span class="op">&lt;-</span> <span class="va">g_qual_raw</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">25</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">30</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">mean_phred_score</span>, y<span class="op">=</span><span class="va">n_sequences</span><span class="op">)</span>, data<span class="op">=</span><span class="va">quality_seq_stats_raw</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Mean Phred score"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"# Sequences"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_quality_seq_raw</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-08_cebria-mendoza_files/figure-html/plot-raw-quality-3.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</section></section><section id="preprocessing" class="level1"><h1>Preprocessing</h1>
<section id="high-level-metrics" class="level2"><h2 class="anchored" data-anchor-id="high-level-metrics">High-level metrics</h2>
<p>The average fraction of reads at each stage in the preprocessing pipeline is shown in the following table. On average, cleaning &amp; deduplication removed about 57% of total read pairs, primarily during duplication which makes sense given that the raw metrics show that the samples are quite high in duplication. Ribodepletion removed about 6-8% during each round.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb5"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># TODO: Group by pool size as well</span></span>
<span><span class="co"># Count read losses</span></span>
<span><span class="va">n_reads_rel</span> <span class="op">&lt;-</span> <span class="va">basic_stats</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">pool_size</span>, <span class="va">stage</span>, <span class="va">percent_duplicates</span>, <span class="va">n_read_pairs</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">pool_size</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">sample</span>, <span class="va">stage</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_retained <span class="op">=</span> <span class="va">n_read_pairs</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/stats/lag.html">lag</a></span><span class="op">(</span><span class="va">n_read_pairs</span><span class="op">)</span>,</span>
<span>         p_reads_lost <span class="op">=</span> <span class="fl">1</span> <span class="op">-</span> <span class="va">p_reads_retained</span>,</span>
<span>         p_reads_retained_abs <span class="op">=</span> <span class="va">n_read_pairs</span> <span class="op">/</span> <span class="va">n_read_pairs</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,</span>
<span>         p_reads_lost_abs <span class="op">=</span> <span class="fl">1</span><span class="op">-</span><span class="va">p_reads_retained_abs</span>,</span>
<span>         p_reads_lost_abs_marginal <span class="op">=</span> <span class="va">p_reads_lost_abs</span> <span class="op">-</span> <span class="fu"><a href="https://rdrr.io/r/stats/lag.html">lag</a></span><span class="op">(</span><span class="va">p_reads_lost_abs</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">n_reads_rel_display</span> <span class="op">&lt;-</span> <span class="va">n_reads_rel</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">rename</span><span class="op">(</span>Stage<span class="op">=</span><span class="va">stage</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">Stage</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>`% Total Reads Lost (Cumulative)` <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">p_reads_lost_abs</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span>,<span class="fl">1</span><span class="op">)</span>, <span class="st">"-"</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_lost_abs</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span>,<span class="fl">1</span><span class="op">)</span>, <span class="st">" (mean "</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">p_reads_lost_abs</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span>,<span class="fl">1</span><span class="op">)</span>, <span class="st">")"</span><span class="op">)</span>,</span>
<span>            `% Total Reads Lost (Marginal)` <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">p_reads_lost_abs_marginal</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span>,<span class="fl">1</span><span class="op">)</span>, <span class="st">"-"</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_lost_abs_marginal</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span>,<span class="fl">1</span><span class="op">)</span>, <span class="st">" (mean "</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">p_reads_lost_abs_marginal</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span>,<span class="fl">1</span><span class="op">)</span>, <span class="st">")"</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">Stage</span> <span class="op">!=</span> <span class="st">"raw_concat"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>Stage <span class="op">=</span> <span class="va">Stage</span> <span class="op">%&gt;%</span> <span class="va">as.numeric</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span>labels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Trimming &amp; filtering"</span>, <span class="st">"Deduplication"</span>, <span class="st">"Initial ribodepletion"</span>, <span class="st">"Secondary ribodepletion"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">n_reads_rel_display</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["Stage"],"name":[1],"type":["fct"],"align":["left"]},{"label":["% Total Reads Lost (Cumulative)"],"name":[2],"type":["chr"],"align":["left"]},{"label":["% Total Reads Lost (Marginal)"],"name":[3],"type":["chr"],"align":["left"]}],"data":[{"1":"Trimming & filtering","2":"0-5 (mean 1.4)","3":"0-5 (mean 1.4)"},{"1":"Deduplication","2":"44.2-70.6 (mean 55.9)","3":"44.2-67.8 (mean 54.5)"},{"1":"Initial ribodepletion","2":"53.9-77.2 (mean 63.8)","3":"3.2-15.6 (mean 7.9)"},{"1":"Secondary ribodepletion","2":"57.7-85.7 (mean 70.2)","3":"2.2-12.6 (mean 6.4)"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<p>Adapter content staays low. Read quality stays pretty similar over the positions, but improves over the number of sequences (predominantly from cleaning) throughout the pipeline.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb6"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_qual</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span>mapping<span class="op">=</span><span class="fu">aes</span><span class="op">(</span>linetype<span class="op">=</span><span class="va">read_pair</span>, group<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/interaction.html">interaction</a></span><span class="op">(</span><span class="va">sample</span>,<span class="va">read_pair</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">scale_linetype_discrete</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Read Pair"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>color<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">2</span>,byrow<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>         linetype <span class="op">=</span> <span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">2</span>,byrow<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span></span>
<span></span>
<span><span class="co"># Visualize adapters</span></span>
<span><span class="va">g_adapters</span> <span class="op">&lt;-</span> <span class="va">g_qual</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">position</span>, y<span class="op">=</span><span class="va">pc_adapters</span><span class="op">)</span>, data<span class="op">=</span><span class="va">adapter_stats</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% Adapters"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span><span class="op">)</span>,</span>
<span>                     , expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Position"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span>,</span>
<span>                     breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">140</span>,<span class="fl">20</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">stage</span><span class="op">~</span><span class="va">adapter</span><span class="op">)</span></span>
<span><span class="va">g_adapters</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-08_cebria-mendoza_files/figure-html/plot-quality-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb7"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Visualize quality</span></span>
<span><span class="va">g_quality_base</span> <span class="op">&lt;-</span> <span class="va">g_qual</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept<span class="op">=</span><span class="fl">25</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept<span class="op">=</span><span class="fl">30</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">position</span>, y<span class="op">=</span><span class="va">mean_phred_score</span><span class="op">)</span>, data<span class="op">=</span><span class="va">quality_base_stats</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Mean Phred score"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">10</span>,<span class="fl">45</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Position"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span>,</span>
<span>                     breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">140</span>,<span class="fl">20</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">stage</span><span class="op">~</span><span class="va">.</span><span class="op">)</span></span>
<span><span class="va">g_quality_base</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-08_cebria-mendoza_files/figure-html/plot-quality-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb8"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_quality_seq</span> <span class="op">&lt;-</span> <span class="va">g_qual</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">25</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">30</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">mean_phred_score</span>, y<span class="op">=</span><span class="va">n_sequences</span><span class="op">)</span>, data<span class="op">=</span><span class="va">quality_seq_stats</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Mean Phred score"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"# Sequences"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">stage</span><span class="op">~</span><span class="va">.</span>, scales <span class="op">=</span> <span class="st">"free_y"</span><span class="op">)</span></span>
<span><span class="va">g_quality_seq</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-08_cebria-mendoza_files/figure-html/plot-quality-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb9"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_stage_trace</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">basic_stats</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">stage</span>, group<span class="op">=</span><span class="va">sample</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span></span>
<span></span>
<span><span class="co"># Plot reads over preprocessing</span></span>
<span><span class="va">g_reads_stages</span> <span class="op">&lt;-</span> <span class="va">g_stage_trace</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">n_read_pairs</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span><span class="st">"# Read pairs"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_reads_stages</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-08_cebria-mendoza_files/figure-html/preproc-figures-1.png" class="img-fluid figure-img" width="576"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb10"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Plot relative read losses during preprocessing</span></span>
<span><span class="va">g_reads_rel</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">n_reads_rel</span>, </span>
<span>                      <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">stage</span>, group<span class="op">=</span><span class="va">sample</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">p_reads_lost_abs_marginal</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span><span class="st">"% Total Reads Lost"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, </span>
<span>                     labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">x</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span></span>
<span><span class="va">g_reads_rel</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-08_cebria-mendoza_files/figure-html/preproc-figures-2.png" class="img-fluid figure-img" width="576"></p>
</figure>
</div>
</div>
</div>
<p>Duplication rates are still quite high which is a bit concerning. Mean read length stays close to expected 150.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb11"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">stage_dup</span> <span class="op">&lt;-</span> <span class="va">basic_stats</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">stage</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>dmin <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">percent_duplicates</span><span class="op">)</span>, dmax<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">percent_duplicates</span><span class="op">)</span>,</span>
<span>            dmean<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">percent_duplicates</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">g_dup_stages</span> <span class="op">&lt;-</span> <span class="va">g_stage_trace</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">percent_duplicates</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span><span class="st">"% Duplicates"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_dup_stages</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-08_cebria-mendoza_files/figure-html/preproc-dedup-1.png" class="img-fluid figure-img" width="576"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb12"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_readlen_stages</span> <span class="op">&lt;-</span> <span class="va">g_stage_trace</span> <span class="op">+</span> <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">mean_seq_len</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span><span class="st">"Mean read length (nt)"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_readlen_stages</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-08_cebria-mendoza_files/figure-html/preproc-dedup-2.png" class="img-fluid figure-img" width="576"></p>
</figure>
</div>
</div>
</div>
<p>Some samples had anywhere as low as 10% all the way up to 60% ribosomal reads.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb13"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Calculate reads lost during ribodepletion (approximation for % ribosomal reads)</span></span>
<span><span class="va">reads_ribo</span> <span class="op">&lt;-</span> <span class="va">n_reads_rel</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">stage</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"dedup"</span>, <span class="st">"ribo_secondary"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>p_reads_ribo<span class="op">=</span><span class="fl">1</span><span class="op">-</span><span class="va">n_read_pairs</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">/</span><span class="va">n_read_pairs</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by <span class="op">=</span> <span class="st">'sample'</span><span class="op">)</span></span>
<span><span class="va">reads_ribo_summ</span> <span class="op">&lt;-</span> <span class="va">reads_ribo</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>min<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">p_reads_ribo</span><span class="op">)</span>, max<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_ribo</span><span class="op">)</span>,</span>
<span>            mean<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">p_reads_ribo</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by <span class="op">=</span> <span class="st">'sample'</span><span class="op">)</span></span>
<span><span class="va">g_reads_ribo</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">reads_ribo</span>, </span>
<span>                       <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">library</span>, y<span class="op">=</span><span class="va">p_reads_ribo</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Approx % ribosomal reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span><span class="op">)</span>,</span>
<span>                     breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="va">theme_kit</span></span>
<span><span class="va">g_reads_ribo</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-08_cebria-mendoza_files/figure-html/ribo-frac-1.png" class="img-fluid figure-img" width="576"></p>
</figure>
</div>
</div>
</div>
</section></section><section id="taxonomic-composition" class="level1"><h1>Taxonomic composition</h1>
<section id="high-level-composition" class="level2"><h2 class="anchored" data-anchor-id="high-level-composition">High-level composition</h2>
<p>To assess the high-level composition of the reads, we ran them through Kraken2 and summarized the results with Bracken.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb14"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Prepare plotting templates</span></span>
<span><span class="va">g_comp_base</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span>mapping<span class="op">=</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">library</span>, y<span class="op">=</span><span class="va">p_reads</span>, fill<span class="op">=</span><span class="va">classification</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_discrete</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Plasma pool"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span> <span class="op">+</span> </span>
<span>  <span class="fu">theme</span><span class="op">(</span>plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>hjust<span class="op">=</span><span class="fl">0</span>, face<span class="op">=</span><span class="st">"plain"</span>, size<span class="op">=</span><span class="fu">rel</span><span class="op">(</span><span class="fl">1.5</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">scale_y_pc_reads</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/partial.html">partial</a></span><span class="op">(</span><span class="va">scale_y_continuous</span>, name <span class="op">=</span> <span class="st">"% Reads"</span>,</span>
<span>                                   expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span></span>
<span><span class="va">geom_comp</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/partial.html">partial</a></span><span class="op">(</span><span class="va">geom_col</span>, position <span class="op">=</span> <span class="st">"stack"</span>, width <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot overall composition</span></span>
<span><span class="va">g_comp</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> <span class="fu">geom_comp</span><span class="op">(</span>data <span class="op">=</span> <span class="va">comp</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_pc_reads</span><span class="op">(</span>limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set1"</span>, name <span class="op">=</span> <span class="st">"Classification"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Read composition (all reads, all groups)"</span><span class="op">)</span></span>
<span><span class="va">g_comp</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-08_cebria-mendoza_files/figure-html/plot-composition-all-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb15"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Repeat for classified reads only</span></span>
<span><span class="va">palette_assigned</span> <span class="op">&lt;-</span> <span class="fu">brewer.pal</span><span class="op">(</span><span class="fl">9</span>, <span class="st">"Set1"</span><span class="op">)</span><span class="op">[</span><span class="fl">5</span><span class="op">:</span><span class="fl">9</span><span class="op">]</span></span>
<span><span class="va">g_comp_assigned</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_comp</span><span class="op">(</span>data <span class="op">=</span> <span class="va">comp_assigned</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_pc_reads</span><span class="op">(</span>limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values<span class="op">=</span><span class="va">palette_assigned</span>, name <span class="op">=</span> <span class="st">"Classification"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Read composition (assigned reads, all groups)"</span><span class="op">)</span></span>
<span><span class="va">g_comp_assigned</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-08_cebria-mendoza_files/figure-html/plot-composition-all-2.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb16"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Plot composition of minor components</span></span>
<span><span class="va">palette_minor</span> <span class="op">&lt;-</span> <span class="fu">brewer.pal</span><span class="op">(</span><span class="fl">9</span>, <span class="st">"Set1"</span><span class="op">)</span><span class="op">[</span><span class="fl">6</span><span class="op">:</span><span class="fl">9</span><span class="op">]</span></span>
<span><span class="va">g_comp_minor</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_comp</span><span class="op">(</span>data<span class="op">=</span><span class="va">comp_minor</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_pc_reads</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values<span class="op">=</span><span class="va">palette_minor</span>, name <span class="op">=</span> <span class="st">"Classification"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Read composition (all reads, minor groups)"</span><span class="op">)</span></span>
<span><span class="va">g_comp_minor</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-08_cebria-mendoza_files/figure-html/plot-composition-all-3.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb17"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_comp_assigned_minor</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_comp</span><span class="op">(</span>data<span class="op">=</span><span class="va">comp_assigned_minor</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_pc_reads</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values<span class="op">=</span><span class="va">palette_minor</span>, name <span class="op">=</span> <span class="st">"Classification"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Read composition (assigned reads, minor groups)"</span><span class="op">)</span></span>
<span><span class="va">g_comp_assigned_minor</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-08_cebria-mendoza_files/figure-html/plot-composition-all-4.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</section><section id="total-viral-content" class="level2"><h2 class="anchored" data-anchor-id="total-viral-content">Total viral content</h2>
<p>Total viral fraction average <span class="math inline">\(1.81 \times 10^{-2}\)</span> across samples. As a fraction of assigned (rather than total) reads, this jumped to <span class="math inline">\(1.60 \times {-1}\)</span>:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb18"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">p_reads_viral_all</span> <span class="op">&lt;-</span> <span class="va">comp</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">classification</span> <span class="op">==</span> <span class="st">"Viral"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>read_group <span class="op">=</span> <span class="st">"All reads"</span><span class="op">)</span></span>
<span><span class="va">p_reads_viral_assigned</span> <span class="op">&lt;-</span> <span class="va">comp_assigned</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">classification</span> <span class="op">==</span> <span class="st">"Viral"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>read_group <span class="op">=</span> <span class="st">"Classified reads"</span><span class="op">)</span></span>
<span><span class="va">p_reads_viral</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">p_reads_viral_all</span>, <span class="va">p_reads_viral_assigned</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_viral</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">p_reads_viral</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">library</span>, y<span class="op">=</span><span class="va">p_reads</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_discrete</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Plasma pool"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_log10</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Viral read fraction"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">.</span><span class="op">~</span><span class="va">read_group</span>, scales <span class="op">=</span> <span class="st">"free"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>color<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">2</span><span class="op">)</span>, shape<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">2</span><span class="op">)</span>,</span>
<span>         linetype<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">2</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span></span>
<span><span class="va">g_viral</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-08_cebria-mendoza_files/figure-html/p-viral-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section><section id="taxonomic-composition-of-viruses" class="level2"><h2 class="anchored" data-anchor-id="taxonomic-composition-of-viruses">Taxonomic composition of viruses</h2>
<p>The two dominant viruses we see are Anellovirdae and Rhabdovirdae (spike-in). Followed by these two viral families is Flavivirdae, and lastly, also by a much smaller percent, Microviridae (spike-in). The threshold for the label “other” are the set of families that make up less than 1% composition in all samples.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb19"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get viral taxonomy</span></span>
<span><span class="va">viral_taxa_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"total-virus-db.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">viral_taxa</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">viral_taxa_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get Kraken reports</span></span>
<span><span class="va">reports_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">tax_final_dir</span>, <span class="st">"kraken_reports.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">reports</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">reports_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Filter to viral taxa</span></span>
<span><span class="va">kraken_reports_viral</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">reports</span>, <span class="va">taxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_viral <span class="op">=</span> <span class="va">n_reads_clade</span><span class="op">/</span><span class="va">n_reads_clade</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">kraken_reports_viral_cleaned</span> <span class="op">&lt;-</span> <span class="va">kraken_reports_viral</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">pc_reads_total</span>, <span class="op">-</span><span class="va">n_reads_direct</span>, <span class="op">-</span><span class="fu">contains</span><span class="op">(</span><span class="st">"minimizers"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span>, <span class="va">p_reads_viral</span>, <span class="va">n_reads_clade</span>, <span class="fu">everything</span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">ungroup</span></span>
<span></span>
<span><span class="va">viral_classes</span> <span class="op">&lt;-</span> <span class="va">kraken_reports_viral_cleaned</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">rank</span> <span class="op">==</span> <span class="st">"C"</span><span class="op">)</span></span>
<span><span class="va">viral_families</span> <span class="op">&lt;-</span> <span class="va">kraken_reports_viral_cleaned</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">rank</span> <span class="op">==</span> <span class="st">"F"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb20"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">major_threshold</span> <span class="op">&lt;-</span> <span class="fl">0.01</span></span>
<span></span>
<span><span class="co"># Identify major viral families</span></span>
<span><span class="va">viral_families_major_tab</span> <span class="op">&lt;-</span> <span class="va">viral_families</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>p_reads_viral_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_viral</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_viral_max</span> <span class="op">&gt;=</span> <span class="va">major_threshold</span><span class="op">)</span></span>
<span><span class="va">viral_families_major_list</span> <span class="op">&lt;-</span> <span class="va">viral_families_major_tab</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">name</span><span class="op">)</span></span>
<span><span class="va">viral_families_major</span> <span class="op">&lt;-</span> <span class="va">viral_families</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">viral_families_major_list</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span>, <span class="va">sample</span>, <span class="va">p_reads_viral</span><span class="op">)</span></span>
<span><span class="va">viral_families_minor</span> <span class="op">&lt;-</span> <span class="va">viral_families_major</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>p_reads_viral_major <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads_viral</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Other"</span>, taxid<span class="op">=</span><span class="cn">NA</span>, p_reads_viral <span class="op">=</span> <span class="fl">1</span><span class="op">-</span><span class="va">p_reads_viral_major</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span>, <span class="va">sample</span>, <span class="va">p_reads_viral</span><span class="op">)</span></span>
<span><span class="va">viral_families_display</span> <span class="op">&lt;-</span> <span class="va">viral_families_major</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">bind_rows</span><span class="op">(</span><span class="va">viral_families_minor</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p_reads_viral</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">name</span>, levels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">viral_families_major_list</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">p_reads_viral</span>, classification<span class="op">=</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by<span class="op">=</span><span class="st">'sample'</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_families</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_comp</span><span class="op">(</span>data<span class="op">=</span><span class="va">viral_families_display</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% Viral Reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, </span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">'Accent'</span><span class="op">)</span></span>
<span><span class="va">g_families</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-08_cebria-mendoza_files/figure-html/viral-family-composition-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
<p>Excluding <em>Anellovirdae</em> and <em>Rhabdovirdae</em>, remaining viral sequences are distributed across a wide variety:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb21"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">major_threshold_adj</span> <span class="op">&lt;-</span> <span class="fl">0.05</span></span>
<span></span>
<span><span class="co"># Adjust viral family counts</span></span>
<span><span class="va">viral_families_adj</span> <span class="op">&lt;-</span> <span class="va">viral_families</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span><span class="op">(</span><span class="va">name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Rhabdoviridae"</span>,<span class="st">"Anelloviridae"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_viral <span class="op">=</span> <span class="va">p_reads_viral</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads_viral</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Identify major viral families</span></span>
<span><span class="va">viral_families_major_tab</span> <span class="op">&lt;-</span> <span class="va">viral_families_adj</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>p_reads_viral_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_viral</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_viral_max</span> <span class="op">&gt;=</span> <span class="va">major_threshold</span><span class="op">)</span></span>
<span><span class="va">viral_families_major_list</span> <span class="op">&lt;-</span> <span class="va">viral_families_major_tab</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">name</span><span class="op">)</span></span>
<span><span class="va">viral_families_major</span> <span class="op">&lt;-</span> <span class="va">viral_families_adj</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">viral_families_major_list</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span>, <span class="va">sample</span>, <span class="va">p_reads_viral</span><span class="op">)</span></span>
<span><span class="va">viral_families_minor</span> <span class="op">&lt;-</span> <span class="va">viral_families_major</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>p_reads_viral_major <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads_viral</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Other"</span>, taxid<span class="op">=</span><span class="cn">NA</span>, p_reads_viral <span class="op">=</span> <span class="fl">1</span><span class="op">-</span><span class="va">p_reads_viral_major</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span>, <span class="va">sample</span>, <span class="va">p_reads_viral</span><span class="op">)</span></span>
<span><span class="va">viral_families_display</span> <span class="op">&lt;-</span> <span class="va">viral_families_major</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">bind_rows</span><span class="op">(</span><span class="va">viral_families_minor</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p_reads_viral</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">name</span>, levels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">viral_families_major_list</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">p_reads_viral</span>, classification<span class="op">=</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by <span class="op">=</span> <span class="st">'sample'</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">palette_viral</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu">brewer.pal</span><span class="op">(</span><span class="fl">12</span>, <span class="st">"Set3"</span><span class="op">)</span>, <span class="fu">brewer.pal</span><span class="op">(</span><span class="fl">8</span>, <span class="st">"Dark2"</span><span class="op">)</span>, <span class="fu">brewer.pal</span><span class="op">(</span><span class="fl">9</span>, <span class="st">"Set1"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_families_adj</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_comp</span><span class="op">(</span>data<span class="op">=</span><span class="va">viral_families_display</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% Viral Reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, </span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values<span class="op">=</span><span class="va">palette_viral</span>, name <span class="op">=</span> <span class="st">"Viral class"</span><span class="op">)</span></span>
<span><span class="va">g_families_adj</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-08_cebria-mendoza_files/figure-html/viral-family-composition-exclusion-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</section></section><section id="human-infecting-virus-reads" class="level1"><h1>Human-infecting virus reads</h1>
<section id="overall-relative-abundance" class="level2"><h2 class="anchored" data-anchor-id="overall-relative-abundance">Overall relative abundance</h2>
<p>I calculated the relative abundance of human-infecting viruses in two ways:</p>
<ul>
<li><p>First, as the total number of deduplicated human-virus reads in each sample, divided by the number of raw reads (“All reads”).</p></li>
<li><p>Second, as a fraction of preprocessed (cleaned, deduplicated, computationally ribodepleted) reads (“Preprocessed reads”).</p></li>
</ul>
<p>[Analyze]</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb22"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get raw read counts</span></span>
<span><span class="va">read_counts_raw</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">basic_stats_raw</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, n_reads_raw <span class="op">=</span> <span class="va">n_read_pairs</span><span class="op">)</span></span>
<span><span class="va">read_counts_preproc</span> <span class="op">&lt;-</span> <span class="va">basic_stats</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">stage</span> <span class="op">==</span> <span class="st">"ribo_initial"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, n_reads_preproc <span class="op">=</span> <span class="va">n_read_pairs</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get HV read counts</span></span>
<span><span class="va">read_counts_hv</span> <span class="op">&lt;-</span> <span class="va">mrg_hv</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">hv_status</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">count</span><span class="op">(</span>name<span class="op">=</span><span class="st">"n_reads_hv"</span><span class="op">)</span></span>
<span><span class="va">read_counts</span> <span class="op">&lt;-</span> <span class="va">read_counts_raw</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">read_counts_hv</span>, by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"sample"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>n_reads_hv <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">n_reads_hv</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">read_counts_preproc</span>, by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"sample"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"sample"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">n_reads_raw</span>, <span class="va">n_reads_preproc</span>, <span class="va">n_reads_hv</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>n_samples <span class="op">=</span> <span class="fl">1</span>,</span>
<span>         p_reads_total <span class="op">=</span> <span class="va">n_reads_hv</span><span class="op">/</span><span class="va">n_reads_raw</span>,</span>
<span>         p_reads_preproc <span class="op">=</span> <span class="va">n_reads_hv</span><span class="op">/</span><span class="va">n_reads_preproc</span><span class="op">)</span></span>
<span><span class="va">read_counts_long</span> <span class="op">&lt;-</span> <span class="va">read_counts</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_longer</span><span class="op">(</span><span class="fu">starts_with</span><span class="op">(</span><span class="st">"p_reads"</span><span class="op">)</span>, names_to<span class="op">=</span><span class="st">"read_group"</span>, values_to<span class="op">=</span><span class="st">"p_reads"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>read_group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">read_group</span> <span class="op">==</span> <span class="st">"p_reads_total"</span>, <span class="st">"All reads"</span>, <span class="st">"Preprocessed reads"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Combine for display</span></span>
<span><span class="va">read_counts_agg</span> <span class="op">&lt;-</span> <span class="va">read_counts</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_total <span class="op">=</span> <span class="va">n_reads_hv</span><span class="op">/</span><span class="va">n_reads_raw</span>,</span>
<span>         p_reads_preproc <span class="op">=</span> <span class="va">n_reads_hv</span><span class="op">/</span><span class="va">n_reads_preproc</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"sample"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">read_counts_agg_long</span> <span class="op">&lt;-</span> <span class="va">read_counts_agg</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_longer</span><span class="op">(</span><span class="fu">starts_with</span><span class="op">(</span><span class="st">"p_reads"</span><span class="op">)</span>, names_to<span class="op">=</span><span class="st">"read_group"</span>, values_to<span class="op">=</span><span class="st">"p_reads"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>read_group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">read_group</span> <span class="op">==</span> <span class="st">"p_reads_total"</span>, <span class="st">"All reads"</span>, <span class="st">"Preprocessed reads"</span><span class="op">)</span><span class="op">)</span> </span>
<span></span>
<span><span class="co"># Visualize</span></span>
<span><span class="va">g_read_counts</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">read_counts_agg_long</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">library</span>, y<span class="op">=</span><span class="va">p_reads</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_log10</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Unique human-viral read fraction"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">.</span><span class="op">~</span><span class="va">read_group</span>, scales <span class="op">=</span> <span class="st">"free"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span></span>
<span><span class="va">g_read_counts</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-08_cebria-mendoza_files/figure-html/count-hv-reads-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</section><section id="overall-taxonomy-and-composition" class="level2"><h2 class="anchored" data-anchor-id="overall-taxonomy-and-composition">Overall taxonomy and composition</h2>
<p>Composition of HV reads was not greatly changed from when looking at all viral reads. The two dominant viruses we see are Anellovirdae and Rhabdovirdae. Followed by these two viral families is Flavivirdae, and lastly, also by a much smaller percent, Microviridae. The threshold for the label “other” are the set of families that make up less than 5% composition in all samples.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb23"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">threshold_major_family</span> <span class="op">&lt;-</span> <span class="fl">0.05</span></span>
<span></span>
<span><span class="co"># Count reads for each human-viral family</span></span>
<span><span class="va">hv_family_counts</span> <span class="op">&lt;-</span> <span class="va">hv_reads_family</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">count</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"n_reads_hv"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_hv <span class="op">=</span> <span class="va">n_reads_hv</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Identify high-ranking families and group others</span></span>
<span><span class="va">hv_family_major_tab</span> <span class="op">&lt;-</span> <span class="va">hv_family_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_hv</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_hv</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">row_number</span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p_reads_hv</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_hv</span> <span class="op">&gt;</span> <span class="va">threshold_major_family</span><span class="op">)</span></span>
<span><span class="va">hv_family_counts_major</span> <span class="op">&lt;-</span> <span class="va">hv_family_counts</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">hv_family_major_tab</span><span class="op">$</span><span class="va">name</span>, <span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">name_display</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_hv <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>, p_reads_hv <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads_hv</span><span class="op">)</span>, </span>
<span>            .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">name_display</span>, </span>
<span>                               levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">hv_family_major_tab</span><span class="op">$</span><span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">hv_family_counts_display</span> <span class="op">&lt;-</span> <span class="va">hv_family_counts_major</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">p_reads_hv</span>, classification <span class="op">=</span> <span class="va">name_display</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by <span class="op">=</span> <span class="st">'sample'</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_hv_family</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>data<span class="op">=</span><span class="va">hv_family_counts_display</span>, position <span class="op">=</span> <span class="st">"stack"</span>, width<span class="op">=</span><span class="fl">1</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% HV Reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, </span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">'Accent'</span>, name <span class="op">=</span> <span class="st">"Viral family"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title<span class="op">=</span><span class="st">"Family composition of human-viral reads"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>fill<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>ncol<span class="op">=</span><span class="fl">4</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fu">rel</span><span class="op">(</span><span class="fl">1.4</span><span class="op">)</span>, hjust<span class="op">=</span><span class="fl">0</span>, face<span class="op">=</span><span class="st">"plain"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_hv_family</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-08_cebria-mendoza_files/figure-html/hv-family-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb24"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get most prominent families for text</span></span>
<span><span class="va">hv_family_collate</span> <span class="op">&lt;-</span> <span class="va">hv_family_counts</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_tot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>,</span>
<span>            p_reads_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_hv</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">n_reads_tot</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#hv_family_collate</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb25"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">threshold_major_family</span> <span class="op">&lt;-</span> <span class="fl">0.05</span></span>
<span></span>
<span><span class="co"># Count reads for each human-viral family</span></span>
<span><span class="va">hv_family_counts</span> <span class="op">&lt;-</span> <span class="va">hv_reads_family</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span><span class="op">(</span><span class="va">name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Anelloviridae"</span>, <span class="st">"Rhabdoviridae"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">count</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"n_reads_hv"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_hv <span class="op">=</span> <span class="va">n_reads_hv</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Identify high-ranking families and group others</span></span>
<span><span class="va">hv_family_major_tab</span> <span class="op">&lt;-</span> <span class="va">hv_family_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_hv</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_hv</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">row_number</span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p_reads_hv</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_hv</span> <span class="op">&gt;</span> <span class="va">threshold_major_family</span><span class="op">)</span></span>
<span><span class="va">hv_family_counts_major</span> <span class="op">&lt;-</span> <span class="va">hv_family_counts</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">hv_family_major_tab</span><span class="op">$</span><span class="va">name</span>, <span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">name_display</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_hv <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>, p_reads_hv <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads_hv</span><span class="op">)</span>, </span>
<span>            .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">name_display</span>, </span>
<span>                               levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">hv_family_major_tab</span><span class="op">$</span><span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">hv_family_counts_display</span> <span class="op">&lt;-</span> <span class="va">hv_family_counts_major</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">p_reads_hv</span>, classification <span class="op">=</span> <span class="va">name_display</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by <span class="op">=</span> <span class="st">'sample'</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_hv_family</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>data<span class="op">=</span><span class="va">hv_family_counts_display</span>, position <span class="op">=</span> <span class="st">"stack"</span>, width<span class="op">=</span><span class="fl">1</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% HV Reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, </span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">'Accent'</span>, name <span class="op">=</span> <span class="st">"Viral family"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title<span class="op">=</span><span class="st">"Family composition of human-viral reads"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>fill<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>ncol<span class="op">=</span><span class="fl">4</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fu">rel</span><span class="op">(</span><span class="fl">1.4</span><span class="op">)</span>, hjust<span class="op">=</span><span class="fl">0</span>, face<span class="op">=</span><span class="st">"plain"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_hv_family</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-08_cebria-mendoza_files/figure-html/hv-filter-family-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb26"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get most prominent families for text</span></span>
<span><span class="va">hv_family_collate</span> <span class="op">&lt;-</span> <span class="va">hv_family_counts</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_tot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>,</span>
<span>            p_reads_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_hv</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">n_reads_tot</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">hv_family_collate</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["name"],"name":[1],"type":["chr"],"align":["left"]},{"label":["taxid"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["n_reads_tot"],"name":[3],"type":["int"],"align":["right"]},{"label":["p_reads_max"],"name":[4],"type":["dbl"],"align":["right"]}],"data":[{"1":"Flaviviridae","2":"11050","3":"94741","4":"1.0000000"},{"1":"Microviridae","2":"10841","3":"81660","4":"1.0000000"},{"1":"Hepeviridae","2":"3079366","3":"1","4":"0.1666667"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
</section><section id="species-analysis" class="level2"><h2 class="anchored" data-anchor-id="species-analysis">Species analysis</h2>
<p>To get a good overview of families, genera, and species, we can look at a Sankey plot where the magnitude of relative abundance, averaged over all samples, is shown in parentheses.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb27"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Function to create links</span></span>
<span><span class="va">create_links</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">data</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">family_to_genus</span> <span class="op">&lt;-</span> <span class="va">data</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">genus</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">group_by</span><span class="op">(</span><span class="va">family</span>, <span class="va">genus</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">summarise</span><span class="op">(</span>value <span class="op">=</span> <span class="fu">n</span><span class="op">(</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>source <span class="op">=</span> <span class="va">family</span>, target <span class="op">=</span> <span class="va">genus</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="va">genus_to_species</span> <span class="op">&lt;-</span> <span class="va">data</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">group_by</span><span class="op">(</span><span class="va">genus</span>, <span class="va">species</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">summarise</span><span class="op">(</span>value <span class="op">=</span> <span class="fu">n</span><span class="op">(</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>source <span class="op">=</span> <span class="va">genus</span>, target <span class="op">=</span> <span class="va">species</span><span class="op">)</span></span>
<span></span>
<span>  <span class="va">family_to_species</span> <span class="op">&lt;-</span> <span class="va">data</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">genus</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">group_by</span><span class="op">(</span><span class="va">family</span>, <span class="va">species</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">summarise</span><span class="op">(</span>value <span class="op">=</span> <span class="fu">n</span><span class="op">(</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>source <span class="op">=</span> <span class="va">family</span>, target <span class="op">=</span> <span class="va">species</span><span class="op">)</span></span>
<span></span>
<span>  <span class="fu">bind_rows</span><span class="op">(</span><span class="va">family_to_genus</span>, <span class="va">genus_to_species</span>, <span class="va">family_to_species</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">source</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Function to create nodes</span></span>
<span><span class="va">create_nodes</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">links</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span></span>
<span>    name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">links</span><span class="op">$</span><span class="va">source</span>, <span class="va">links</span><span class="op">$</span><span class="va">target</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Function to prepare data for Sankey diagram</span></span>
<span><span class="va">prepare_sankey_data</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">links</span>, <span class="va">nodes</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">links</span><span class="op">$</span><span class="va">IDsource</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/match.html">match</a></span><span class="op">(</span><span class="va">links</span><span class="op">$</span><span class="va">source</span>, <span class="va">nodes</span><span class="op">$</span><span class="va">name</span><span class="op">)</span> <span class="op">-</span> <span class="fl">1</span></span>
<span>  <span class="va">links</span><span class="op">$</span><span class="va">IDtarget</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/match.html">match</a></span><span class="op">(</span><span class="va">links</span><span class="op">$</span><span class="va">target</span>, <span class="va">nodes</span><span class="op">$</span><span class="va">name</span><span class="op">)</span> <span class="op">-</span> <span class="fl">1</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>links <span class="op">=</span> <span class="va">links</span>, nodes <span class="op">=</span> <span class="va">nodes</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Function to create Sankey plot</span></span>
<span><span class="va">create_sankey_plot</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">sankey_data</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu">sankeyNetwork</span><span class="op">(</span></span>
<span>    Links <span class="op">=</span> <span class="va">sankey_data</span><span class="op">$</span><span class="va">links</span>, </span>
<span>    Nodes <span class="op">=</span> <span class="va">sankey_data</span><span class="op">$</span><span class="va">nodes</span>,</span>
<span>    Source <span class="op">=</span> <span class="st">"IDsource"</span>, </span>
<span>    Target <span class="op">=</span> <span class="st">"IDtarget"</span>,</span>
<span>    Value <span class="op">=</span> <span class="st">"value"</span>, </span>
<span>    NodeID <span class="op">=</span> <span class="st">"name"</span>,</span>
<span>    sinksRight <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    nodeWidth <span class="op">=</span> <span class="fl">50</span>,</span>
<span>    fontSize <span class="op">=</span> <span class="fl">14</span>,</span>
<span>    height <span class="op">=</span> <span class="fl">800</span>,</span>
<span>    width <span class="op">=</span> <span class="fl">1000</span></span>
<span>  <span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="va">save_sankey_as_png</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">sankey_plot</span>, <span class="va">width</span> <span class="op">=</span> <span class="fl">1000</span>, <span class="va">height</span> <span class="op">=</span> <span class="fl">800</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="co"># Save the plot as an HTML file</span></span>
<span>  <span class="fu">saveWidget</span><span class="op">(</span><span class="va">sankey_plot</span>, <span class="fu"><a href="https://rdrr.io/r/base/sprintf.html">sprintf</a></span><span class="op">(</span><span class="st">'%s/sankey.html'</span>,<span class="va">image_dir</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="va">format_scientific</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">digits</span><span class="op">=</span><span class="fl">2</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">x</span>, <span class="kw">function</span><span class="op">(</span><span class="va">val</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">val</span><span class="op">)</span> <span class="op">||</span> <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span><span class="op">(</span><span class="va">val</span><span class="op">)</span> <span class="op">&lt;</span> <span class="fl">1e-15</span><span class="op">)</span> <span class="op">{</span></span>
<span>      <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="st">"0"</span><span class="op">)</span></span>
<span>    <span class="op">}</span> <span class="kw">else</span> <span class="op">{</span></span>
<span>      <span class="va">exponent</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">floor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span><span class="op">(</span><span class="va">val</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span>      <span class="va">coef</span> <span class="op">&lt;-</span> <span class="va">val</span> <span class="op">/</span> <span class="fl">10</span><span class="op">^</span><span class="va">exponent</span></span>
<span>      <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sprintf.html">sprintf</a></span><span class="op">(</span><span class="st">"%.1f × 10^%d"</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="va">coef</span>, <span class="va">digits</span><span class="op">)</span>, <span class="va">exponent</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="op">}</span></span>
<span>  <span class="op">}</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="va">result</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span><span class="fu">across</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">genus_n_reads_tot</span>, <span class="va">genus_ra_reads_tot</span><span class="op">)</span>, <span class="op">~</span><span class="fu">replace_na</span><span class="op">(</span><span class="va">.</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span>,</span>
<span>         genus <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">genus</span><span class="op">)</span>, <span class="st">"Unknown Genus"</span>, <span class="va">genus</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>  species <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">species</span>, <span class="fu"><a href="https://rdrr.io/r/base/sprintf.html">sprintf</a></span><span class="op">(</span><span class="st">' (%s)'</span>, <span class="fu">format_scientific</span><span class="op">(</span><span class="va">species_ra_reads_tot</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,</span>
<span>  genus <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">genus</span>, <span class="fu"><a href="https://rdrr.io/r/base/sprintf.html">sprintf</a></span><span class="op">(</span><span class="st">' (%s)'</span>, <span class="fu">format_scientific</span><span class="op">(</span><span class="va">genus_ra_reads_tot</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,</span>
<span>  family <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">family</span>, <span class="fu"><a href="https://rdrr.io/r/base/sprintf.html">sprintf</a></span><span class="op">(</span><span class="st">' (%s)'</span>, <span class="fu">format_scientific</span><span class="op">(</span><span class="va">family_ra_reads_tot</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="va">links</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="fu">create_links</span><span class="op">(</span><span class="va">data</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">nodes</span> <span class="op">&lt;-</span> <span class="fu">create_nodes</span><span class="op">(</span><span class="va">links</span><span class="op">)</span></span>
<span><span class="va">sankey_data</span> <span class="op">&lt;-</span> <span class="fu">prepare_sankey_data</span><span class="op">(</span><span class="va">links</span>, <span class="va">nodes</span><span class="op">)</span></span>
<span><span class="va">sankey</span> <span class="op">&lt;-</span> <span class="fu">create_sankey_plot</span><span class="op">(</span><span class="va">sankey_data</span><span class="op">)</span></span>
<span></span>
<span><span class="va">sankey</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div id="htmlwidget-4dff9645c61c0a3dc275" style="width:100%;height:520px;" class="sankeyNetwork html-widget"></div>
<script type="application/json" data-for="htmlwidget-4dff9645c61c0a3dc275">{"x":{"links":{"source":[0,0,0,0,1,2,3,4,4,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,7,7,7,7,7,7,7,7,7,7,7,7,7,7,8,9,10,11,11,11,11,11,11,12,12,12],"target":[5,6,7,11,10,9,11,8,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77],"value":[23,16,14,4,1,1,2,1,3,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1]},"nodes":{"name":["Anelloviridae (3.6 <U+00D7> 10^-2)","Flaviviridae (1.5 <U+00D7> 10^-3)","Hepeviridae (3.4 <U+00D7> 10^-7)","Microviridae (1.1 <U+00D7> 10^-3)","Rhabdoviridae (4.1 <U+00D7> 10^-3)","Alphatorquevirus (1.9 <U+00D7> 10^-4)","Betatorquevirus (1.9 <U+00D7> 10^-3)","Gammatorquevirus (2.5 <U+00D7> 10^-5)","Ledantevirus (3.0 <U+00D7> 10^-7)","Paslahepevirus (3.4 <U+00D7> 10^-7)","Pegivirus (1.5 <U+00D7> 10^-3)","Unknown Genus (0)","Vesiculovirus (4.1 <U+00D7> 10^-3)","Alphatorquevirus homin1 (5.8 <U+00D7> 10^-5)","Alphatorquevirus homin10 (8.6 <U+00D7> 10^-6)","Alphatorquevirus homin13 (4.9 <U+00D7> 10^-6)","Alphatorquevirus homin15 (2.0 <U+00D7> 10^-4)","Alphatorquevirus homin17 (1.1 <U+00D7> 10^-6)","Alphatorquevirus homin18 (4.8 <U+00D7> 10^-6)","Alphatorquevirus homin19 (3.4 <U+00D7> 10^-6)","Alphatorquevirus homin20 (1.2 <U+00D7> 10^-6)","Alphatorquevirus homin21 (9.6 <U+00D7> 10^-6)","Alphatorquevirus homin24 (8.0 <U+00D7> 10^-5)","Alphatorquevirus homin29 (2.9 <U+00D7> 10^-5)","Alphatorquevirus homin3 (6.0 <U+00D7> 10^-6)","Alphatorquevirus homin5 (6.6 <U+00D7> 10^-6)","Alphatorquevirus homin6 (3.3 <U+00D7> 10^-5)","Alphatorquevirus homin7 (1.3 <U+00D7> 10^-6)","Alphatorquevirus homin9 (1.7 <U+00D7> 10^-6)","Torque teno virus 11 (5.8 <U+00D7> 10^-5)","Torque teno virus 12 (1.7 <U+00D7> 10^-6)","Torque teno virus 16 (2.0 <U+00D7> 10^-5)","Torque teno virus 22 (8.9 <U+00D7> 10^-5)","Torque teno virus 27 (9.1 <U+00D7> 10^-7)","Torque teno virus 28 (3.1 <U+00D7> 10^-5)","Torque teno virus 8 (6.4 <U+00D7> 10^-6)","Betatorquevirus homini1 (5.5 <U+00D7> 10^-4)","Betatorquevirus homini10 (3.6 <U+00D7> 10^-5)","Betatorquevirus homini11 (7.1 <U+00D7> 10^-4)","Betatorquevirus homini12 (1.2 <U+00D7> 10^-3)","Betatorquevirus homini16 (3.4 <U+00D7> 10^-5)","Betatorquevirus homini18 (1.1 <U+00D7> 10^-4)","Betatorquevirus homini2 (2.7 <U+00D7> 10^-5)","Betatorquevirus homini26 (7.7 <U+00D7> 10^-5)","Betatorquevirus homini27 (6.4 <U+00D7> 10^-5)","Betatorquevirus homini3 (4.3 <U+00D7> 10^-5)","Betatorquevirus homini5 (5.7 <U+00D7> 10^-5)","Betatorquevirus homini6 (5.7 <U+00D7> 10^-6)","Betatorquevirus homini7 (3.4 <U+00D7> 10^-4)","Betatorquevirus homini8 (1.5 <U+00D7> 10^-5)","Betatorquevirus homini9 (1.3 <U+00D7> 10^-4)","TTV-like mini virus (4.8 <U+00D7> 10^-4)","Gammatorquevirus homidi1 (1.5 <U+00D7> 10^-6)","Gammatorquevirus homidi10 (4.0 <U+00D7> 10^-6)","Gammatorquevirus homidi11 (1.0 <U+00D7> 10^-6)","Gammatorquevirus homidi12 (1.3 <U+00D7> 10^-6)","Gammatorquevirus homidi13 (1.8 <U+00D7> 10^-6)","Gammatorquevirus homidi14 (2.3 <U+00D7> 10^-6)","Gammatorquevirus homidi2 (2.6 <U+00D7> 10^-6)","Gammatorquevirus homidi3 (4.7 <U+00D7> 10^-6)","Gammatorquevirus homidi4 (3.1 <U+00D7> 10^-6)","Gammatorquevirus homidi5 (3.7 <U+00D7> 10^-6)","Gammatorquevirus homidi6 (2.2 <U+00D7> 10^-6)","Gammatorquevirus homidi7 (1.5 <U+00D7> 10^-5)","Gammatorquevirus homidi8 (1.2 <U+00D7> 10^-5)","Gammatorquevirus homidi9 (7.1 <U+00D7> 10^-7)","Ledantevirus ledantec (3.0 <U+00D7> 10^-7)","Paslahepevirus balayani (3.4 <U+00D7> 10^-7)","Pegivirus hominis (1.5 <U+00D7> 10^-3)","Anelloviridae sp. (3.5 <U+00D7> 10^-2)","Micro Torque teno virus (1.2 <U+00D7> 10^-4)","Microviridae sp. (1.2 <U+00D7> 10^-3)","Microvirus sp. (2.7 <U+00D7> 10^-5)","Small anellovirus (9.9 <U+00D7> 10^-6)","Torque teno virus (8.5 <U+00D7> 10^-5)","Vesiculovirus indiana (4.1 <U+00D7> 10^-3)","Vesiculovirus isfahan (2.9 <U+00D7> 10^-7)","Vesiculovirus maraba (7.7 <U+00D7> 10^-7)"],"group":["Anelloviridae (3.6 <U+00D7> 10^-2)","Flaviviridae (1.5 <U+00D7> 10^-3)","Hepeviridae (3.4 <U+00D7> 10^-7)","Microviridae (1.1 <U+00D7> 10^-3)","Rhabdoviridae (4.1 <U+00D7> 10^-3)","Alphatorquevirus (1.9 <U+00D7> 10^-4)","Betatorquevirus (1.9 <U+00D7> 10^-3)","Gammatorquevirus (2.5 <U+00D7> 10^-5)","Ledantevirus (3.0 <U+00D7> 10^-7)","Paslahepevirus (3.4 <U+00D7> 10^-7)","Pegivirus (1.5 <U+00D7> 10^-3)","Unknown Genus (0)","Vesiculovirus (4.1 <U+00D7> 10^-3)","Alphatorquevirus homin1 (5.8 <U+00D7> 10^-5)","Alphatorquevirus homin10 (8.6 <U+00D7> 10^-6)","Alphatorquevirus homin13 (4.9 <U+00D7> 10^-6)","Alphatorquevirus homin15 (2.0 <U+00D7> 10^-4)","Alphatorquevirus homin17 (1.1 <U+00D7> 10^-6)","Alphatorquevirus homin18 (4.8 <U+00D7> 10^-6)","Alphatorquevirus homin19 (3.4 <U+00D7> 10^-6)","Alphatorquevirus homin20 (1.2 <U+00D7> 10^-6)","Alphatorquevirus homin21 (9.6 <U+00D7> 10^-6)","Alphatorquevirus homin24 (8.0 <U+00D7> 10^-5)","Alphatorquevirus homin29 (2.9 <U+00D7> 10^-5)","Alphatorquevirus homin3 (6.0 <U+00D7> 10^-6)","Alphatorquevirus homin5 (6.6 <U+00D7> 10^-6)","Alphatorquevirus homin6 (3.3 <U+00D7> 10^-5)","Alphatorquevirus homin7 (1.3 <U+00D7> 10^-6)","Alphatorquevirus homin9 (1.7 <U+00D7> 10^-6)","Torque teno virus 11 (5.8 <U+00D7> 10^-5)","Torque teno virus 12 (1.7 <U+00D7> 10^-6)","Torque teno virus 16 (2.0 <U+00D7> 10^-5)","Torque teno virus 22 (8.9 <U+00D7> 10^-5)","Torque teno virus 27 (9.1 <U+00D7> 10^-7)","Torque teno virus 28 (3.1 <U+00D7> 10^-5)","Torque teno virus 8 (6.4 <U+00D7> 10^-6)","Betatorquevirus homini1 (5.5 <U+00D7> 10^-4)","Betatorquevirus homini10 (3.6 <U+00D7> 10^-5)","Betatorquevirus homini11 (7.1 <U+00D7> 10^-4)","Betatorquevirus homini12 (1.2 <U+00D7> 10^-3)","Betatorquevirus homini16 (3.4 <U+00D7> 10^-5)","Betatorquevirus homini18 (1.1 <U+00D7> 10^-4)","Betatorquevirus homini2 (2.7 <U+00D7> 10^-5)","Betatorquevirus homini26 (7.7 <U+00D7> 10^-5)","Betatorquevirus homini27 (6.4 <U+00D7> 10^-5)","Betatorquevirus homini3 (4.3 <U+00D7> 10^-5)","Betatorquevirus homini5 (5.7 <U+00D7> 10^-5)","Betatorquevirus homini6 (5.7 <U+00D7> 10^-6)","Betatorquevirus homini7 (3.4 <U+00D7> 10^-4)","Betatorquevirus homini8 (1.5 <U+00D7> 10^-5)","Betatorquevirus homini9 (1.3 <U+00D7> 10^-4)","TTV-like mini virus (4.8 <U+00D7> 10^-4)","Gammatorquevirus homidi1 (1.5 <U+00D7> 10^-6)","Gammatorquevirus homidi10 (4.0 <U+00D7> 10^-6)","Gammatorquevirus homidi11 (1.0 <U+00D7> 10^-6)","Gammatorquevirus homidi12 (1.3 <U+00D7> 10^-6)","Gammatorquevirus homidi13 (1.8 <U+00D7> 10^-6)","Gammatorquevirus homidi14 (2.3 <U+00D7> 10^-6)","Gammatorquevirus homidi2 (2.6 <U+00D7> 10^-6)","Gammatorquevirus homidi3 (4.7 <U+00D7> 10^-6)","Gammatorquevirus homidi4 (3.1 <U+00D7> 10^-6)","Gammatorquevirus homidi5 (3.7 <U+00D7> 10^-6)","Gammatorquevirus homidi6 (2.2 <U+00D7> 10^-6)","Gammatorquevirus homidi7 (1.5 <U+00D7> 10^-5)","Gammatorquevirus homidi8 (1.2 <U+00D7> 10^-5)","Gammatorquevirus homidi9 (7.1 <U+00D7> 10^-7)","Ledantevirus ledantec (3.0 <U+00D7> 10^-7)","Paslahepevirus balayani (3.4 <U+00D7> 10^-7)","Pegivirus hominis (1.5 <U+00D7> 10^-3)","Anelloviridae sp. (3.5 <U+00D7> 10^-2)","Micro Torque teno virus (1.2 <U+00D7> 10^-4)","Microviridae sp. (1.2 <U+00D7> 10^-3)","Microvirus sp. (2.7 <U+00D7> 10^-5)","Small anellovirus (9.9 <U+00D7> 10^-6)","Torque teno virus (8.5 <U+00D7> 10^-5)","Vesiculovirus indiana (4.1 <U+00D7> 10^-3)","Vesiculovirus isfahan (2.9 <U+00D7> 10^-7)","Vesiculovirus maraba (7.7 <U+00D7> 10^-7)"]},"options":{"NodeID":"name","NodeGroup":"name","LinkGroup":null,"colourScale":"d3.scaleOrdinal(d3.schemeCategory20);","fontSize":14,"fontFamily":null,"nodeWidth":50,"nodePadding":10,"units":"","margin":{"top":null,"right":null,"bottom":null,"left":null},"iterations":32,"sinksRight":true}},"evals":[],"jsHooks":[]}</script>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb28"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co">#save_sankey_as_png(sankey)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>To get a better idea of the relative abundance of species, we can create a dot plot where each dot represents the relative abundance of a particular species in a sample.</p>
<div class="cell" width="20" height="10">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb29"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">species_family</span> <span class="op">&lt;-</span> <span class="va">result</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">species</span>, <span class="va">family</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">rename</span><span class="op">(</span><span class="st">'name'</span> <span class="op">=</span> <span class="st">'species'</span><span class="op">)</span></span>
<span></span>
<span><span class="va">play</span> <span class="op">&lt;-</span> <span class="va">hv_species_counts</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">ungroup</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by <span class="op">=</span> <span class="st">'sample'</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">species_family</span>, by <span class="op">=</span> <span class="st">'name'</span><span class="op">)</span></span>
<span></span>
<span><span class="va">play</span><span class="op">$</span><span class="va">family</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">play</span><span class="op">$</span><span class="va">family</span>, levels<span class="op">=</span><span class="va">hv_family_collate</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">family</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">play</span> <span class="op">&lt;-</span> <span class="va">play</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">family</span>, <span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">name</span>, levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">name</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#name_order &lt;- play %&gt;% arrange(family) %&gt;% pull(name)</span></span>
<span><span class="co">#play$name &lt;- factor(play$name, levels = name_order)</span></span>
<span></span>
<span><span class="va">ra_dot</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">play</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">ra_reads_hv</span>, y<span class="op">=</span><span class="va">name</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_quasirandom</span><span class="op">(</span>orientation <span class="op">=</span> <span class="st">'y'</span>, <span class="fu">aes</span><span class="op">(</span>color <span class="op">=</span> <span class="va">family</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>   <span class="fu">scale_color_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Accent"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">scale_x_log10</span><span class="op">(</span></span>
<span>    <span class="st">"Relative abundance human virus reads"</span>,</span>
<span>    labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/parse.html">parse</a></span><span class="op">(</span>text <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"10^"</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1e-7</span>, <span class="fl">1</span><span class="op">)</span>,</span>
<span>    n.breaks <span class="op">=</span> <span class="fl">8</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>y <span class="op">=</span> <span class="st">'Human virus'</span>,</span>
<span>       color <span class="op">=</span> <span class="st">'Viral family'</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">theme_light</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu">theme</span><span class="op">(</span></span>
<span>    axis.text.y <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">8</span><span class="op">)</span>,</span>
<span>    axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">14</span><span class="op">)</span>,</span>
<span>    axis.ticks.y <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,axis.line <span class="op">=</span> <span class="fu">element_line</span><span class="op">(</span>colour <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span>,</span>
<span>    axis.title.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">15</span><span class="op">)</span>,    </span>
<span>    axis.title.y <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">15</span><span class="op">)</span>,  </span>
<span>    legend.text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">13</span><span class="op">)</span>,</span>
<span>    legend.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">16</span><span class="op">)</span>,</span>
<span>    panel.grid.minor <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    panel.border <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    panel.background <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">ra_dot</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-08_cebria-mendoza_files/figure-html/hv-species-dot-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb30"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co">#ggsave(sprintf('%s/ra_viral_dot.jpg',image_dir), ra_dot, height = 8 ,width = 12)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>I’d like to BLAST some of these reads against the NCBI nt database to see if we can get some more info on them.</p>
</section></section><section id="conclusion" class="level1"><h1>Conclusion</h1>
<p>Overall, we found a lot of anelloviridae, which makes sense given that this paper is about discovering new species of anelloviridae in blood.</p>


<!-- -->

</section></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb31" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "Workflow of Cebria-Mendoza et al. (2021)"</span></span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a><span class="an">subtitle:</span><span class="co"> "Pooled plasma from Spain (DNA + RNA)"</span></span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> "Harmon Bhasin"</span></span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> 2024-07-08</span></span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span></span>
<span id="cb31-7"><a href="#cb31-7" aria-hidden="true" tabindex="-1"></a><span class="co">  html:</span></span>
<span id="cb31-8"><a href="#cb31-8" aria-hidden="true" tabindex="-1"></a><span class="co">    code-fold: true</span></span>
<span id="cb31-9"><a href="#cb31-9" aria-hidden="true" tabindex="-1"></a><span class="co">    code-tools: true</span></span>
<span id="cb31-10"><a href="#cb31-10" aria-hidden="true" tabindex="-1"></a><span class="co">    code-link: true</span></span>
<span id="cb31-11"><a href="#cb31-11" aria-hidden="true" tabindex="-1"></a><span class="co">    df-print: paged</span></span>
<span id="cb31-12"><a href="#cb31-12" aria-hidden="true" tabindex="-1"></a><span class="an">execute:</span><span class="co"> </span></span>
<span id="cb31-13"><a href="#cb31-13" aria-hidden="true" tabindex="-1"></a><span class="co">  freeze: auto</span></span>
<span id="cb31-14"><a href="#cb31-14" aria-hidden="true" tabindex="-1"></a><span class="an">editor:</span><span class="co"> visual</span></span>
<span id="cb31-15"><a href="#cb31-15" aria-hidden="true" tabindex="-1"></a><span class="an">title-block-banner:</span><span class="co"> "#de2d26"</span></span>
<span id="cb31-16"><a href="#cb31-16" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb31-17"><a href="#cb31-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-20"><a href="#cb31-20" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb31-21"><a href="#cb31-21" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: load-packages</span></span>
<span id="cb31-22"><a href="#cb31-22" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb31-23"><a href="#cb31-23" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb31-24"><a href="#cb31-24" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cowplot)</span>
<span id="cb31-25"><a href="#cb31-25" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb31-26"><a href="#cb31-26" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(fastqcr)</span>
<span id="cb31-27"><a href="#cb31-27" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)</span>
<span id="cb31-28"><a href="#cb31-28" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(networkD3)</span>
<span id="cb31-29"><a href="#cb31-29" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb31-30"><a href="#cb31-30" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(htmlwidgets)</span>
<span id="cb31-31"><a href="#cb31-31" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggbeeswarm)</span>
<span id="cb31-32"><a href="#cb31-32" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(latex2exp)</span>
<span id="cb31-33"><a href="#cb31-33" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"/Users/harmonbhasin/work/securebio/sampling-strategies/scripts/aux_plot-theme.R"</span>)</span>
<span id="cb31-34"><a href="#cb31-34" aria-hidden="true" tabindex="-1"></a>theme_base <span class="ot">&lt;-</span> theme_base <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio =</span> <span class="cn">NULL</span>)</span>
<span id="cb31-35"><a href="#cb31-35" aria-hidden="true" tabindex="-1"></a>theme_kit <span class="ot">&lt;-</span> theme_base <span class="sc">+</span> <span class="fu">theme</span>(</span>
<span id="cb31-36"><a href="#cb31-36" aria-hidden="true" tabindex="-1"></a>  <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">angle =</span> <span class="dv">45</span>),</span>
<span id="cb31-37"><a href="#cb31-37" aria-hidden="true" tabindex="-1"></a>  <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb31-38"><a href="#cb31-38" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb31-39"><a href="#cb31-39" aria-hidden="true" tabindex="-1"></a>tnl <span class="ot">&lt;-</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb31-40"><a href="#cb31-40" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb31-41"><a href="#cb31-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-42"><a href="#cb31-42" aria-hidden="true" tabindex="-1"></a>As a part of my time here, I'm exploring blood-based surveillance for novel viral pathogen detection. We've decided to make three blog posts. The first blog post will cover the physical and biological characteristics of blood, the second blog post will cover the blood sampling strategies, and the last blog post will look at the metagenomic profile of blood (which is what we're doing here, and will do for the next few posts).</span>
<span id="cb31-43"><a href="#cb31-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-44"><a href="#cb31-44" aria-hidden="true" tabindex="-1"></a>This is the first of many studies that I'll be analyzing. however I don't expect all of them to be used in the last blog post. In this post, I analyze <span class="co">[</span><span class="ot">Cebria-Mendoza 2021</span><span class="co">](https://doi.org/10.3390/v13112322)</span>, a dataset with 60 samples where each sample is a pool of 8-13 unique healthy donors, with a total of \~600 donors from Spain.</span>
<span id="cb31-45"><a href="#cb31-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-46"><a href="#cb31-46" aria-hidden="true" tabindex="-1"></a><span class="fu"># The raw data</span></span>
<span id="cb31-47"><a href="#cb31-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-48"><a href="#cb31-48" aria-hidden="true" tabindex="-1"></a><span class="fu">## About</span></span>
<span id="cb31-49"><a href="#cb31-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-50"><a href="#cb31-50" aria-hidden="true" tabindex="-1"></a>This dataset is composed of 60 samples which come from plasma pools of anywhere from 8-13 people from Spain. In total, there were 567 healthy individuals that have contributed to these pools. The demographic information that I have for each pool is the gender and age of the donor as well as the number that has contributed to each pool. For each pooled sample, they performed a combined DNA and RNA library preparation, resulting in a single sequencing output that captures both nucleic acid types. This approach provides comprehensive genetic information but precludes separate analysis of DNA and RNA from individual samples.</span>
<span id="cb31-51"><a href="#cb31-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-52"><a href="#cb31-52" aria-hidden="true" tabindex="-1"></a><span class="fu">### Sample + library preparation </span></span>
<span id="cb31-53"><a href="#cb31-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-54"><a href="#cb31-54" aria-hidden="true" tabindex="-1"></a>A total of 587 plasma samples from healthy donors were collected from the Centro de Transfusión de la Comunidad Valenciana (Valencia, Spain) from 15 September 2018 to 30 March 2019 and stored at −80 °C until use.</span>
<span id="cb31-55"><a href="#cb31-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-56"><a href="#cb31-56" aria-hidden="true" tabindex="-1"></a>Each of the 60 pools (SP1-SP60) analyzed in this study was obtained by mixing 1 mL of plasma from a variable number of donors (between 8- and 13-mL total). To assess viral recovery, each pool was spiked with 103 PFU of ϕX174 and 104 PFU of vesicular stomatitis virus (VSV). The purification protocol has been previously described in detail <span class="co">[</span><span class="ot">18</span><span class="co">]</span>. Briefly, plasma pools were processed with 1.0 µM filters to remove cells and other non-viral particles and the filtered fractions were subject to high-speed centrifugation (87,000 g, 2 h, 4 °C), washed with PBS 1X (87,000 g, 1 h, 4 °C), and resuspended in 245 µL 1X digestion buffer (Turbo DNA Free kit, Ambion, Carlsbad, CA, USA). Then, 5 µL of Turbo DNase, 2 µL of Benzonase (Sigma, Darmstadt, Germany) and 2 µL of micrococcal nuclease (NEB) were added to the sample to remove unprotected nucleic acids. After incubation (1 h, 37 °C), 20 µL of stop reagent was added, following the manufacturer’s instructions. Then, 240 µL supernatant was transferred to a new tube and split into two fractions: 200 µL fraction was used for RNA extraction using TRIzol LS reagent (Invitrogen, Carlsbad, USA), followed by purification with the QIAamp Viral RNA Mini kit (Qiagen, Hilden, Germany) and amplification with the QuantiTect Whole Transcriptome kit (Qiagen), and 40 µL fraction was used for DNA extraction with the QIAamp Viral RNA Mini kit and amplification with the TruePrime WGA kit (Sygnis, Heidelberg, Germany).</span>
<span id="cb31-57"><a href="#cb31-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-58"><a href="#cb31-58" aria-hidden="true" tabindex="-1"></a>For each pool, DNA and RNA amplification products were mixed in equimolar concentration before library preparation, which was carried out using Nextera XT DNA library preparation kit with 15 amplification cycles (Illumina, San Diego, USA), and subject to pair-end sequencing in a NextSeq device.</span>
<span id="cb31-59"><a href="#cb31-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-60"><a href="#cb31-60" aria-hidden="true" tabindex="-1"></a>*More details can be found in their [original paper](https://doi.org/10.1038%2Fs41598-021-86427-4) which outlined this protocol.*</span>
<span id="cb31-61"><a href="#cb31-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-62"><a href="#cb31-62" aria-hidden="true" tabindex="-1"></a><span class="fu">## Quality control</span></span>
<span id="cb31-63"><a href="#cb31-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-64"><a href="#cb31-64" aria-hidden="true" tabindex="-1"></a>In total, these 60 samples contained 230M read pairs. The samples had 2.3M - 4.8M (mean 3.8M) read pairs each.</span>
<span id="cb31-65"><a href="#cb31-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-66"><a href="#cb31-66" aria-hidden="true" tabindex="-1"></a>The number of read pairs and total base pairs looks good, however the duplication rate is quite high (this can be attributed to their sample preparation).</span>
<span id="cb31-69"><a href="#cb31-69" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb31-70"><a href="#cb31-70" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb31-71"><a href="#cb31-71" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: read-qc-data</span></span>
<span id="cb31-72"><a href="#cb31-72" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb31-73"><a href="#cb31-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-74"><a href="#cb31-74" aria-hidden="true" tabindex="-1"></a><span class="co"># Data input paths</span></span>
<span id="cb31-75"><a href="#cb31-75" aria-hidden="true" tabindex="-1"></a>data_dir <span class="ot">&lt;-</span> <span class="st">"/Users/harmonbhasin/work/securebio/nao-harmon/cebriamendoza2021/analysis/"</span></span>
<span id="cb31-76"><a href="#cb31-76" aria-hidden="true" tabindex="-1"></a>input_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"input"</span>)</span>
<span id="cb31-77"><a href="#cb31-77" aria-hidden="true" tabindex="-1"></a>results_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"results"</span>)</span>
<span id="cb31-78"><a href="#cb31-78" aria-hidden="true" tabindex="-1"></a>qc_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(results_dir, <span class="st">"qc"</span>)</span>
<span id="cb31-79"><a href="#cb31-79" aria-hidden="true" tabindex="-1"></a>hv_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(results_dir, <span class="st">"hv"</span>)</span>
<span id="cb31-80"><a href="#cb31-80" aria-hidden="true" tabindex="-1"></a>libraries_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(input_dir, <span class="st">"libraries.csv"</span>)</span>
<span id="cb31-81"><a href="#cb31-81" aria-hidden="true" tabindex="-1"></a>basic_stats_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(qc_dir, <span class="st">"qc_basic_stats.tsv.gz"</span>)</span>
<span id="cb31-82"><a href="#cb31-82" aria-hidden="true" tabindex="-1"></a>adapter_stats_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(qc_dir, <span class="st">"qc_adapter_stats.tsv.gz"</span>)</span>
<span id="cb31-83"><a href="#cb31-83" aria-hidden="true" tabindex="-1"></a>quality_base_stats_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(qc_dir, <span class="st">"qc_quality_base_stats.tsv.gz"</span>)</span>
<span id="cb31-84"><a href="#cb31-84" aria-hidden="true" tabindex="-1"></a>quality_seq_stats_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(qc_dir, <span class="st">"qc_quality_sequence_stats.tsv.gz"</span>)</span>
<span id="cb31-85"><a href="#cb31-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-86"><a href="#cb31-86" aria-hidden="true" tabindex="-1"></a><span class="co"># Import libraries and extract metadata from sample names</span></span>
<span id="cb31-87"><a href="#cb31-87" aria-hidden="true" tabindex="-1"></a>libraries_raw <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(libraries_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb31-88"><a href="#cb31-88" aria-hidden="true" tabindex="-1"></a>meta_data <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="fu">sprintf</span>(<span class="st">'%s/SraRunTable.txt'</span>, data_dir)) <span class="sc">%&gt;%</span></span>
<span id="cb31-89"><a href="#cb31-89" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">library=</span>Run)</span>
<span id="cb31-90"><a href="#cb31-90" aria-hidden="true" tabindex="-1"></a>libraries <span class="ot">&lt;-</span> <span class="fu">left_join</span>(libraries_raw, meta_data) <span class="sc">%&gt;%</span></span>
<span id="cb31-91"><a href="#cb31-91" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(</span>
<span id="cb31-92"><a href="#cb31-92" aria-hidden="true" tabindex="-1"></a>    library,</span>
<span id="cb31-93"><a href="#cb31-93" aria-hidden="true" tabindex="-1"></a>    AvgSpotLen,</span>
<span id="cb31-94"><a href="#cb31-94" aria-hidden="true" tabindex="-1"></a>    Collection_Date,</span>
<span id="cb31-95"><a href="#cb31-95" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">Library Name</span><span class="st">`</span></span>
<span id="cb31-96"><a href="#cb31-96" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb31-97"><a href="#cb31-97" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">sample=</span>library,</span>
<span id="cb31-98"><a href="#cb31-98" aria-hidden="true" tabindex="-1"></a>    <span class="at">library=</span><span class="st">`</span><span class="at">Library Name</span><span class="st">`</span>)</span>
<span id="cb31-99"><a href="#cb31-99" aria-hidden="true" tabindex="-1"></a>supplementary_data <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(<span class="fu">sprintf</span>(<span class="st">'%s/Supplementary_Table_S1.xlsx'</span>, data_dir), <span class="at">col_names =</span> <span class="fu">c</span>(<span class="st">'library'</span>, <span class="st">'sex'</span>, <span class="st">'age'</span>))<span class="sc">%&gt;%</span> <span class="fu">slice</span>(<span class="sc">-</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb31-100"><a href="#cb31-100" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(library) <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(<span class="at">pool_size =</span> <span class="fu">n</span>())</span>
<span id="cb31-101"><a href="#cb31-101" aria-hidden="true" tabindex="-1"></a>supplementary_data<span class="sc">$</span>pool_size <span class="ot">&lt;-</span> <span class="fu">factor</span>(supplementary_data<span class="sc">$</span>pool_size)</span>
<span id="cb31-102"><a href="#cb31-102" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-103"><a href="#cb31-103" aria-hidden="true" tabindex="-1"></a><span class="co">#sp_list &lt;- paste0("SP", 1:60)</span></span>
<span id="cb31-104"><a href="#cb31-104" aria-hidden="true" tabindex="-1"></a><span class="co">#libraries$library &lt;- factor(libraries$library, levels=sp_list)</span></span>
<span id="cb31-105"><a href="#cb31-105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-106"><a href="#cb31-106" aria-hidden="true" tabindex="-1"></a>most_to_least_reads_sample_list <span class="ot">&lt;-</span> supplementary_data <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(pool_size)) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(library)</span>
<span id="cb31-107"><a href="#cb31-107" aria-hidden="true" tabindex="-1"></a>libraries <span class="ot">&lt;-</span> <span class="fu">left_join</span>(libraries, supplementary_data, <span class="at">by =</span> <span class="st">'library'</span>)</span>
<span id="cb31-108"><a href="#cb31-108" aria-hidden="true" tabindex="-1"></a>libraries<span class="sc">$</span>library <span class="ot">&lt;-</span> <span class="fu">factor</span>(libraries<span class="sc">$</span>library, <span class="at">levels=</span>most_to_least_reads_sample_list, <span class="at">labels=</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">60</span>))</span>
<span id="cb31-109"><a href="#cb31-109" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-110"><a href="#cb31-110" aria-hidden="true" tabindex="-1"></a><span class="co"># Import QC data</span></span>
<span id="cb31-111"><a href="#cb31-111" aria-hidden="true" tabindex="-1"></a>stages <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"raw_concat"</span>, <span class="st">"cleaned"</span>, <span class="st">"dedup"</span>, <span class="st">"ribo_initial"</span>, <span class="st">"ribo_secondary"</span>)</span>
<span id="cb31-112"><a href="#cb31-112" aria-hidden="true" tabindex="-1"></a>basic_stats <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(basic_stats_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb31-113"><a href="#cb31-113" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb31-114"><a href="#cb31-114" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">factor</span>(stage, <span class="at">levels =</span> stages),</span>
<span id="cb31-115"><a href="#cb31-115" aria-hidden="true" tabindex="-1"></a>         <span class="at">sample =</span> <span class="fu">fct_inorder</span>(sample))</span>
<span id="cb31-116"><a href="#cb31-116" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-117"><a href="#cb31-117" aria-hidden="true" tabindex="-1"></a>adapter_stats <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(adapter_stats_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb31-118"><a href="#cb31-118" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb31-119"><a href="#cb31-119" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">factor</span>(stage, <span class="at">levels =</span> stages),</span>
<span id="cb31-120"><a href="#cb31-120" aria-hidden="true" tabindex="-1"></a>         <span class="at">read_pair =</span> <span class="fu">fct_inorder</span>(<span class="fu">as.character</span>(read_pair)))</span>
<span id="cb31-121"><a href="#cb31-121" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-122"><a href="#cb31-122" aria-hidden="true" tabindex="-1"></a>quality_base_stats <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(quality_base_stats_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb31-123"><a href="#cb31-123" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb31-124"><a href="#cb31-124" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">factor</span>(stage, <span class="at">levels =</span> stages),</span>
<span id="cb31-125"><a href="#cb31-125" aria-hidden="true" tabindex="-1"></a>         <span class="at">read_pair =</span> <span class="fu">fct_inorder</span>(<span class="fu">as.character</span>(read_pair)))</span>
<span id="cb31-126"><a href="#cb31-126" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb31-127"><a href="#cb31-127" aria-hidden="true" tabindex="-1"></a>quality_seq_stats <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(quality_seq_stats_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb31-128"><a href="#cb31-128" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb31-129"><a href="#cb31-129" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">factor</span>(stage, <span class="at">levels =</span> stages),</span>
<span id="cb31-130"><a href="#cb31-130" aria-hidden="true" tabindex="-1"></a>         <span class="at">read_pair =</span> <span class="fu">fct_inorder</span>(<span class="fu">as.character</span>(read_pair)))</span>
<span id="cb31-131"><a href="#cb31-131" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb31-132"><a href="#cb31-132" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb31-133"><a href="#cb31-133" aria-hidden="true" tabindex="-1"></a> <span class="co"># Filter to raw data</span></span>
<span id="cb31-134"><a href="#cb31-134" aria-hidden="true" tabindex="-1"></a> basic_stats_raw <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"raw_concat"</span>)</span>
<span id="cb31-135"><a href="#cb31-135" aria-hidden="true" tabindex="-1"></a> adapter_stats_raw <span class="ot">&lt;-</span> adapter_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"raw_concat"</span>)</span>
<span id="cb31-136"><a href="#cb31-136" aria-hidden="true" tabindex="-1"></a> quality_base_stats_raw <span class="ot">&lt;-</span> quality_base_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"raw_concat"</span>)</span>
<span id="cb31-137"><a href="#cb31-137" aria-hidden="true" tabindex="-1"></a> quality_seq_stats_raw <span class="ot">&lt;-</span> quality_seq_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"raw_concat"</span>)</span>
<span id="cb31-138"><a href="#cb31-138" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb31-139"><a href="#cb31-139" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb31-140"><a href="#cb31-140" aria-hidden="true" tabindex="-1"></a> <span class="co"># Get key values for readout</span></span>
<span id="cb31-141"><a href="#cb31-141" aria-hidden="true" tabindex="-1"></a>raw_read_counts <span class="ot">&lt;-</span> basic_stats_raw <span class="sc">%&gt;%</span> </span>
<span id="cb31-142"><a href="#cb31-142" aria-hidden="true" tabindex="-1"></a>   <span class="fu">summarize</span>(<span class="at">rmin =</span> <span class="fu">min</span>(n_read_pairs), <span class="at">rmax=</span><span class="fu">max</span>(n_read_pairs),</span>
<span id="cb31-143"><a href="#cb31-143" aria-hidden="true" tabindex="-1"></a>             <span class="at">rmean=</span><span class="fu">mean</span>(n_read_pairs), </span>
<span id="cb31-144"><a href="#cb31-144" aria-hidden="true" tabindex="-1"></a>             <span class="at">rtot =</span> <span class="fu">sum</span>(n_read_pairs),</span>
<span id="cb31-145"><a href="#cb31-145" aria-hidden="true" tabindex="-1"></a>             <span class="at">btot =</span> <span class="fu">sum</span>(n_bases_approx),</span>
<span id="cb31-146"><a href="#cb31-146" aria-hidden="true" tabindex="-1"></a>             <span class="at">dmin =</span> <span class="fu">min</span>(percent_duplicates), <span class="at">dmax=</span><span class="fu">max</span>(percent_duplicates),</span>
<span id="cb31-147"><a href="#cb31-147" aria-hidden="true" tabindex="-1"></a>             <span class="at">dmean=</span><span class="fu">mean</span>(percent_duplicates), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb31-148"><a href="#cb31-148" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb31-149"><a href="#cb31-149" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-152"><a href="#cb31-152" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb31-153"><a href="#cb31-153" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 9</span></span>
<span id="cb31-154"><a href="#cb31-154" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb31-155"><a href="#cb31-155" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-basic-stats</span></span>
<span id="cb31-156"><a href="#cb31-156" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-157"><a href="#cb31-157" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare data</span></span>
<span id="cb31-158"><a href="#cb31-158" aria-hidden="true" tabindex="-1"></a>basic_stats_raw_metrics <span class="ot">&lt;-</span> basic_stats_raw <span class="sc">%&gt;%</span></span>
<span id="cb31-159"><a href="#cb31-159" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(library,</span>
<span id="cb31-160"><a href="#cb31-160" aria-hidden="true" tabindex="-1"></a>         pool_size,</span>
<span id="cb31-161"><a href="#cb31-161" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at"># Read pairs</span><span class="st">`</span> <span class="ot">=</span> n_read_pairs,</span>
<span id="cb31-162"><a href="#cb31-162" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Total base pairs</span><span class="sc">\n</span><span class="at">(approx)</span><span class="st">`</span> <span class="ot">=</span> n_bases_approx,</span>
<span id="cb31-163"><a href="#cb31-163" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">% Duplicates</span><span class="sc">\n</span><span class="at">(FASTQC)</span><span class="st">`</span> <span class="ot">=</span> percent_duplicates) <span class="sc">%&gt;%</span></span>
<span id="cb31-164"><a href="#cb31-164" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>(library<span class="sc">:</span>pool_size), <span class="at">names_to =</span> <span class="st">"metric"</span>, <span class="at">values_to =</span> <span class="st">"value"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb31-165"><a href="#cb31-165" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">metric =</span> <span class="fu">fct_inorder</span>(metric)) </span>
<span id="cb31-166"><a href="#cb31-166" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-167"><a href="#cb31-167" aria-hidden="true" tabindex="-1"></a><span class="co"># Set up plot templates</span></span>
<span id="cb31-168"><a href="#cb31-168" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-169"><a href="#cb31-169" aria-hidden="true" tabindex="-1"></a>g_basic <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(basic_stats_raw_metrics, <span class="fu">aes</span>(<span class="at">x =</span> library, <span class="at">y =</span> value, <span class="at">fill =</span> pool_size)) <span class="sc">+</span></span>
<span id="cb31-170"><a href="#cb31-170" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position =</span> <span class="st">"dodge"</span>) <span class="sc">+</span></span>
<span id="cb31-171"><a href="#cb31-171" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>() <span class="sc">+</span></span>
<span id="cb31-172"><a href="#cb31-172" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb31-173"><a href="#cb31-173" aria-hidden="true" tabindex="-1"></a>  <span class="fu">expand_limits</span>(<span class="at">y =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">100</span>)) <span class="sc">+</span></span>
<span id="cb31-174"><a href="#cb31-174" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Accent"</span>) <span class="sc">+</span> </span>
<span id="cb31-175"><a href="#cb31-175" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(metric <span class="sc">~</span> ., <span class="at">scales =</span> <span class="st">"free"</span>, <span class="at">space =</span> <span class="st">"free_x"</span>, <span class="at">switch =</span> <span class="st">"y"</span>) <span class="sc">+</span></span>
<span id="cb31-176"><a href="#cb31-176" aria-hidden="true" tabindex="-1"></a>  theme_kit <span class="sc">+</span> <span class="fu">theme</span>(</span>
<span id="cb31-177"><a href="#cb31-177" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb31-178"><a href="#cb31-178" aria-hidden="true" tabindex="-1"></a>    <span class="at">strip.text.y =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"plain"</span>)</span>
<span id="cb31-179"><a href="#cb31-179" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb31-180"><a href="#cb31-180" aria-hidden="true" tabindex="-1"></a>g_basic</span>
<span id="cb31-181"><a href="#cb31-181" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb31-182"><a href="#cb31-182" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-183"><a href="#cb31-183" aria-hidden="true" tabindex="-1"></a>Adapter content is low (the upper limit on the plot is 1%). Read quality seems good both over all the positions as well as over the number of sequences.</span>
<span id="cb31-184"><a href="#cb31-184" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-187"><a href="#cb31-187" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb31-188"><a href="#cb31-188" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-raw-quality</span></span>
<span id="cb31-189"><a href="#cb31-189" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 8</span></span>
<span id="cb31-190"><a href="#cb31-190" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-191"><a href="#cb31-191" aria-hidden="true" tabindex="-1"></a><span class="co"># Set up plotting templates</span></span>
<span id="cb31-192"><a href="#cb31-192" aria-hidden="true" tabindex="-1"></a>g_qual_raw <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">linetype=</span>read_pair, <span class="at">group=</span><span class="fu">interaction</span>(sample,read_pair))) <span class="sc">+</span> </span>
<span id="cb31-193"><a href="#cb31-193" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_linetype_discrete</span>(<span class="at">name =</span> <span class="st">"Read Pair"</span>) <span class="sc">+</span></span>
<span id="cb31-194"><a href="#cb31-194" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color=</span><span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>),</span>
<span id="cb31-195"><a href="#cb31-195" aria-hidden="true" tabindex="-1"></a>         <span class="at">linetype =</span> <span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>)) <span class="sc">+</span></span>
<span id="cb31-196"><a href="#cb31-196" aria-hidden="true" tabindex="-1"></a>  theme_base</span>
<span id="cb31-197"><a href="#cb31-197" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-198"><a href="#cb31-198" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize adapters</span></span>
<span id="cb31-199"><a href="#cb31-199" aria-hidden="true" tabindex="-1"></a>g_adapters_raw <span class="ot">&lt;-</span> g_qual_raw <span class="sc">+</span> </span>
<span id="cb31-200"><a href="#cb31-200" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>pc_adapters), <span class="at">data=</span>adapter_stats_raw) <span class="sc">+</span></span>
<span id="cb31-201"><a href="#cb31-201" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Adapters"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>),</span>
<span id="cb31-202"><a href="#cb31-202" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb31-203"><a href="#cb31-203" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Position"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb31-204"><a href="#cb31-204" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">500</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb31-205"><a href="#cb31-205" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(.<span class="sc">~</span>adapter)</span>
<span id="cb31-206"><a href="#cb31-206" aria-hidden="true" tabindex="-1"></a>g_adapters_raw</span>
<span id="cb31-207"><a href="#cb31-207" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-208"><a href="#cb31-208" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize quality</span></span>
<span id="cb31-209"><a href="#cb31-209" aria-hidden="true" tabindex="-1"></a>g_quality_base_raw <span class="ot">&lt;-</span> g_qual_raw <span class="sc">+</span></span>
<span id="cb31-210"><a href="#cb31-210" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb31-211"><a href="#cb31-211" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb31-212"><a href="#cb31-212" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>mean_phred_score), <span class="at">data=</span>quality_base_stats_raw) <span class="sc">+</span></span>
<span id="cb31-213"><a href="#cb31-213" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"Mean Phred score"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">10</span>,<span class="dv">45</span>)) <span class="sc">+</span></span>
<span id="cb31-214"><a href="#cb31-214" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Position"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb31-215"><a href="#cb31-215" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">500</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb31-216"><a href="#cb31-216" aria-hidden="true" tabindex="-1"></a>g_quality_base_raw</span>
<span id="cb31-217"><a href="#cb31-217" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-218"><a href="#cb31-218" aria-hidden="true" tabindex="-1"></a>g_quality_seq_raw <span class="ot">&lt;-</span> g_qual_raw <span class="sc">+</span></span>
<span id="cb31-219"><a href="#cb31-219" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb31-220"><a href="#cb31-220" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb31-221"><a href="#cb31-221" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>mean_phred_score, <span class="at">y=</span>n_sequences), <span class="at">data=</span>quality_seq_stats_raw) <span class="sc">+</span></span>
<span id="cb31-222"><a href="#cb31-222" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Mean Phred score"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb31-223"><a href="#cb31-223" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"# Sequences"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb31-224"><a href="#cb31-224" aria-hidden="true" tabindex="-1"></a>g_quality_seq_raw</span>
<span id="cb31-225"><a href="#cb31-225" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb31-226"><a href="#cb31-226" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-227"><a href="#cb31-227" aria-hidden="true" tabindex="-1"></a><span class="fu"># Preprocessing</span></span>
<span id="cb31-228"><a href="#cb31-228" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-229"><a href="#cb31-229" aria-hidden="true" tabindex="-1"></a><span class="fu">## High-level metrics</span></span>
<span id="cb31-230"><a href="#cb31-230" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-231"><a href="#cb31-231" aria-hidden="true" tabindex="-1"></a>The average fraction of reads at each stage in the preprocessing pipeline is shown in the following table. On average, cleaning &amp; deduplication removed about 57% of total read pairs, primarily during duplication which makes sense given that the raw metrics show that the samples are quite high in duplication. Ribodepletion removed about 6-8% during each round.</span>
<span id="cb31-232"><a href="#cb31-232" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-235"><a href="#cb31-235" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb31-236"><a href="#cb31-236" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: preproc-table</span></span>
<span id="cb31-237"><a href="#cb31-237" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-238"><a href="#cb31-238" aria-hidden="true" tabindex="-1"></a><span class="co"># </span><span class="al">TODO</span><span class="co">: Group by pool size as well</span></span>
<span id="cb31-239"><a href="#cb31-239" aria-hidden="true" tabindex="-1"></a><span class="co"># Count read losses</span></span>
<span id="cb31-240"><a href="#cb31-240" aria-hidden="true" tabindex="-1"></a>n_reads_rel <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> </span>
<span id="cb31-241"><a href="#cb31-241" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, pool_size, stage, percent_duplicates, n_read_pairs) <span class="sc">%&gt;%</span></span>
<span id="cb31-242"><a href="#cb31-242" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, pool_size) <span class="sc">%&gt;%</span> </span>
<span id="cb31-243"><a href="#cb31-243" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(sample, stage) <span class="sc">%&gt;%</span></span>
<span id="cb31-244"><a href="#cb31-244" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_retained =</span> n_read_pairs <span class="sc">/</span> <span class="fu">lag</span>(n_read_pairs),</span>
<span id="cb31-245"><a href="#cb31-245" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_lost =</span> <span class="dv">1</span> <span class="sc">-</span> p_reads_retained,</span>
<span id="cb31-246"><a href="#cb31-246" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_retained_abs =</span> n_read_pairs <span class="sc">/</span> n_read_pairs[<span class="dv">1</span>],</span>
<span id="cb31-247"><a href="#cb31-247" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_lost_abs =</span> <span class="dv">1</span><span class="sc">-</span>p_reads_retained_abs,</span>
<span id="cb31-248"><a href="#cb31-248" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_lost_abs_marginal =</span> p_reads_lost_abs <span class="sc">-</span> <span class="fu">lag</span>(p_reads_lost_abs))</span>
<span id="cb31-249"><a href="#cb31-249" aria-hidden="true" tabindex="-1"></a>n_reads_rel_display <span class="ot">&lt;-</span> n_reads_rel <span class="sc">%&gt;%</span> </span>
<span id="cb31-250"><a href="#cb31-250" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">Stage=</span>stage) <span class="sc">%&gt;%</span> </span>
<span id="cb31-251"><a href="#cb31-251" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Stage) <span class="sc">%&gt;%</span> </span>
<span id="cb31-252"><a href="#cb31-252" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="st">`</span><span class="at">% Total Reads Lost (Cumulative)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">round</span>(<span class="fu">min</span>(p_reads_lost_abs<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">"-"</span>, <span class="fu">round</span>(<span class="fu">max</span>(p_reads_lost_abs<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">" (mean "</span>, <span class="fu">round</span>(<span class="fu">mean</span>(p_reads_lost_abs<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">")"</span>),</span>
<span id="cb31-253"><a href="#cb31-253" aria-hidden="true" tabindex="-1"></a>            <span class="st">`</span><span class="at">% Total Reads Lost (Marginal)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">round</span>(<span class="fu">min</span>(p_reads_lost_abs_marginal<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">"-"</span>, <span class="fu">round</span>(<span class="fu">max</span>(p_reads_lost_abs_marginal<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">" (mean "</span>, <span class="fu">round</span>(<span class="fu">mean</span>(p_reads_lost_abs_marginal<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">")"</span>), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb31-254"><a href="#cb31-254" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Stage <span class="sc">!=</span> <span class="st">"raw_concat"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb31-255"><a href="#cb31-255" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Stage =</span> Stage <span class="sc">%&gt;%</span> as.numeric <span class="sc">%&gt;%</span> <span class="fu">factor</span>(<span class="at">labels=</span><span class="fu">c</span>(<span class="st">"Trimming &amp; filtering"</span>, <span class="st">"Deduplication"</span>, <span class="st">"Initial ribodepletion"</span>, <span class="st">"Secondary ribodepletion"</span>)))</span>
<span id="cb31-256"><a href="#cb31-256" aria-hidden="true" tabindex="-1"></a>n_reads_rel_display</span>
<span id="cb31-257"><a href="#cb31-257" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb31-258"><a href="#cb31-258" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-259"><a href="#cb31-259" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-260"><a href="#cb31-260" aria-hidden="true" tabindex="-1"></a>Adapter content staays low. Read quality stays pretty similar over the positions, but improves over the number of sequences (predominantly from cleaning) throughout the pipeline.</span>
<span id="cb31-261"><a href="#cb31-261" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-264"><a href="#cb31-264" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb31-265"><a href="#cb31-265" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb31-266"><a href="#cb31-266" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-quality</span></span>
<span id="cb31-267"><a href="#cb31-267" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 8</span></span>
<span id="cb31-268"><a href="#cb31-268" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-269"><a href="#cb31-269" aria-hidden="true" tabindex="-1"></a>g_qual <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">linetype=</span>read_pair, <span class="at">group=</span><span class="fu">interaction</span>(sample,read_pair))) <span class="sc">+</span> </span>
<span id="cb31-270"><a href="#cb31-270" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_linetype_discrete</span>(<span class="at">name =</span> <span class="st">"Read Pair"</span>) <span class="sc">+</span></span>
<span id="cb31-271"><a href="#cb31-271" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color=</span><span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>),</span>
<span id="cb31-272"><a href="#cb31-272" aria-hidden="true" tabindex="-1"></a>         <span class="at">linetype =</span> <span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>)) <span class="sc">+</span></span>
<span id="cb31-273"><a href="#cb31-273" aria-hidden="true" tabindex="-1"></a>  theme_base</span>
<span id="cb31-274"><a href="#cb31-274" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-275"><a href="#cb31-275" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize adapters</span></span>
<span id="cb31-276"><a href="#cb31-276" aria-hidden="true" tabindex="-1"></a>g_adapters <span class="ot">&lt;-</span> g_qual <span class="sc">+</span> </span>
<span id="cb31-277"><a href="#cb31-277" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>pc_adapters), <span class="at">data=</span>adapter_stats) <span class="sc">+</span></span>
<span id="cb31-278"><a href="#cb31-278" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Adapters"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>),</span>
<span id="cb31-279"><a href="#cb31-279" aria-hidden="true" tabindex="-1"></a>                     , <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb31-280"><a href="#cb31-280" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Position"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb31-281"><a href="#cb31-281" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">140</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb31-282"><a href="#cb31-282" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(stage<span class="sc">~</span>adapter)</span>
<span id="cb31-283"><a href="#cb31-283" aria-hidden="true" tabindex="-1"></a>g_adapters</span>
<span id="cb31-284"><a href="#cb31-284" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-285"><a href="#cb31-285" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize quality</span></span>
<span id="cb31-286"><a href="#cb31-286" aria-hidden="true" tabindex="-1"></a>g_quality_base <span class="ot">&lt;-</span> g_qual <span class="sc">+</span></span>
<span id="cb31-287"><a href="#cb31-287" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb31-288"><a href="#cb31-288" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb31-289"><a href="#cb31-289" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>mean_phred_score), <span class="at">data=</span>quality_base_stats) <span class="sc">+</span></span>
<span id="cb31-290"><a href="#cb31-290" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"Mean Phred score"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">10</span>,<span class="dv">45</span>)) <span class="sc">+</span></span>
<span id="cb31-291"><a href="#cb31-291" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Position"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb31-292"><a href="#cb31-292" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">140</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb31-293"><a href="#cb31-293" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(stage<span class="sc">~</span>.)</span>
<span id="cb31-294"><a href="#cb31-294" aria-hidden="true" tabindex="-1"></a>g_quality_base</span>
<span id="cb31-295"><a href="#cb31-295" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-296"><a href="#cb31-296" aria-hidden="true" tabindex="-1"></a>g_quality_seq <span class="ot">&lt;-</span> g_qual <span class="sc">+</span></span>
<span id="cb31-297"><a href="#cb31-297" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb31-298"><a href="#cb31-298" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb31-299"><a href="#cb31-299" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>mean_phred_score, <span class="at">y=</span>n_sequences), <span class="at">data=</span>quality_seq_stats) <span class="sc">+</span></span>
<span id="cb31-300"><a href="#cb31-300" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Mean Phred score"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb31-301"><a href="#cb31-301" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"# Sequences"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb31-302"><a href="#cb31-302" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(stage<span class="sc">~</span>., <span class="at">scales =</span> <span class="st">"free_y"</span>)</span>
<span id="cb31-303"><a href="#cb31-303" aria-hidden="true" tabindex="-1"></a>g_quality_seq</span>
<span id="cb31-304"><a href="#cb31-304" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb31-305"><a href="#cb31-305" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-308"><a href="#cb31-308" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb31-309"><a href="#cb31-309" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: preproc-figures</span></span>
<span id="cb31-310"><a href="#cb31-310" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb31-311"><a href="#cb31-311" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 3</span></span>
<span id="cb31-312"><a href="#cb31-312" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 6</span></span>
<span id="cb31-313"><a href="#cb31-313" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-314"><a href="#cb31-314" aria-hidden="true" tabindex="-1"></a>g_stage_trace <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(basic_stats, <span class="fu">aes</span>(<span class="at">x=</span>stage, <span class="at">group=</span>sample)) <span class="sc">+</span></span>
<span id="cb31-315"><a href="#cb31-315" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb31-316"><a href="#cb31-316" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-317"><a href="#cb31-317" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot reads over preprocessing</span></span>
<span id="cb31-318"><a href="#cb31-318" aria-hidden="true" tabindex="-1"></a>g_reads_stages <span class="ot">&lt;-</span> g_stage_trace <span class="sc">+</span></span>
<span id="cb31-319"><a href="#cb31-319" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>n_read_pairs)) <span class="sc">+</span></span>
<span id="cb31-320"><a href="#cb31-320" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"# Read pairs"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>))</span>
<span id="cb31-321"><a href="#cb31-321" aria-hidden="true" tabindex="-1"></a>g_reads_stages</span>
<span id="cb31-322"><a href="#cb31-322" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-323"><a href="#cb31-323" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot relative read losses during preprocessing</span></span>
<span id="cb31-324"><a href="#cb31-324" aria-hidden="true" tabindex="-1"></a>g_reads_rel <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(n_reads_rel, </span>
<span id="cb31-325"><a href="#cb31-325" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">aes</span>(<span class="at">x=</span>stage, <span class="at">group=</span>sample)) <span class="sc">+</span></span>
<span id="cb31-326"><a href="#cb31-326" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>p_reads_lost_abs_marginal)) <span class="sc">+</span></span>
<span id="cb31-327"><a href="#cb31-327" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"% Total Reads Lost"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), </span>
<span id="cb31-328"><a href="#cb31-328" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="cf">function</span>(x) x<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb31-329"><a href="#cb31-329" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb31-330"><a href="#cb31-330" aria-hidden="true" tabindex="-1"></a>g_reads_rel</span>
<span id="cb31-331"><a href="#cb31-331" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb31-332"><a href="#cb31-332" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-333"><a href="#cb31-333" aria-hidden="true" tabindex="-1"></a>Duplication rates are still quite high which is a bit concerning. Mean read length stays close to expected 150.</span>
<span id="cb31-334"><a href="#cb31-334" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-337"><a href="#cb31-337" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb31-338"><a href="#cb31-338" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: preproc-dedup</span></span>
<span id="cb31-339"><a href="#cb31-339" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 4</span></span>
<span id="cb31-340"><a href="#cb31-340" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 6</span></span>
<span id="cb31-341"><a href="#cb31-341" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-342"><a href="#cb31-342" aria-hidden="true" tabindex="-1"></a>stage_dup <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(stage) <span class="sc">%&gt;%</span> </span>
<span id="cb31-343"><a href="#cb31-343" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">dmin =</span> <span class="fu">min</span>(percent_duplicates), <span class="at">dmax=</span><span class="fu">max</span>(percent_duplicates),</span>
<span id="cb31-344"><a href="#cb31-344" aria-hidden="true" tabindex="-1"></a>            <span class="at">dmean=</span><span class="fu">mean</span>(percent_duplicates), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb31-345"><a href="#cb31-345" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-346"><a href="#cb31-346" aria-hidden="true" tabindex="-1"></a>g_dup_stages <span class="ot">&lt;-</span> g_stage_trace <span class="sc">+</span></span>
<span id="cb31-347"><a href="#cb31-347" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>percent_duplicates)) <span class="sc">+</span></span>
<span id="cb31-348"><a href="#cb31-348" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"% Duplicates"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb31-349"><a href="#cb31-349" aria-hidden="true" tabindex="-1"></a>g_dup_stages</span>
<span id="cb31-350"><a href="#cb31-350" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-351"><a href="#cb31-351" aria-hidden="true" tabindex="-1"></a>g_readlen_stages <span class="ot">&lt;-</span> g_stage_trace <span class="sc">+</span> <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>mean_seq_len)) <span class="sc">+</span></span>
<span id="cb31-352"><a href="#cb31-352" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"Mean read length (nt)"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>))</span>
<span id="cb31-353"><a href="#cb31-353" aria-hidden="true" tabindex="-1"></a>g_readlen_stages</span>
<span id="cb31-354"><a href="#cb31-354" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb31-355"><a href="#cb31-355" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-356"><a href="#cb31-356" aria-hidden="true" tabindex="-1"></a>Some samples had anywhere as low as 10% all the way up to 60% ribosomal reads.</span>
<span id="cb31-357"><a href="#cb31-357" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-360"><a href="#cb31-360" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb31-361"><a href="#cb31-361" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: ribo-frac</span></span>
<span id="cb31-362"><a href="#cb31-362" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 4</span></span>
<span id="cb31-363"><a href="#cb31-363" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 6</span></span>
<span id="cb31-364"><a href="#cb31-364" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate reads lost during ribodepletion (approximation for % ribosomal reads)</span></span>
<span id="cb31-365"><a href="#cb31-365" aria-hidden="true" tabindex="-1"></a>reads_ribo <span class="ot">&lt;-</span> n_reads_rel <span class="sc">%&gt;%</span> </span>
<span id="cb31-366"><a href="#cb31-366" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(stage <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"dedup"</span>, <span class="st">"ribo_secondary"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb31-367"><a href="#cb31-367" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb31-368"><a href="#cb31-368" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads_ribo=</span><span class="dv">1</span><span class="sc">-</span>n_read_pairs[<span class="dv">2</span>]<span class="sc">/</span>n_read_pairs[<span class="dv">1</span>], <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb31-369"><a href="#cb31-369" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by =</span> <span class="st">'sample'</span>)</span>
<span id="cb31-370"><a href="#cb31-370" aria-hidden="true" tabindex="-1"></a>reads_ribo_summ <span class="ot">&lt;-</span> reads_ribo <span class="sc">%&gt;%</span></span>
<span id="cb31-371"><a href="#cb31-371" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb31-372"><a href="#cb31-372" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">min=</span><span class="fu">min</span>(p_reads_ribo), <span class="at">max=</span><span class="fu">max</span>(p_reads_ribo),</span>
<span id="cb31-373"><a href="#cb31-373" aria-hidden="true" tabindex="-1"></a>            <span class="at">mean=</span><span class="fu">mean</span>(p_reads_ribo), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb31-374"><a href="#cb31-374" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by =</span> <span class="st">'sample'</span>)</span>
<span id="cb31-375"><a href="#cb31-375" aria-hidden="true" tabindex="-1"></a>g_reads_ribo <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(reads_ribo, </span>
<span id="cb31-376"><a href="#cb31-376" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">aes</span>(<span class="at">x=</span>library, <span class="at">y=</span>p_reads_ribo)) <span class="sc">+</span></span>
<span id="cb31-377"><a href="#cb31-377" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span> </span>
<span id="cb31-378"><a href="#cb31-378" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"Approx % ribosomal reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>),</span>
<span id="cb31-379"><a href="#cb31-379" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>)<span class="sc">+</span></span>
<span id="cb31-380"><a href="#cb31-380" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb31-381"><a href="#cb31-381" aria-hidden="true" tabindex="-1"></a>g_reads_ribo</span>
<span id="cb31-382"><a href="#cb31-382" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb31-383"><a href="#cb31-383" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-384"><a href="#cb31-384" aria-hidden="true" tabindex="-1"></a><span class="fu"># Taxonomic composition</span></span>
<span id="cb31-385"><a href="#cb31-385" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-386"><a href="#cb31-386" aria-hidden="true" tabindex="-1"></a><span class="fu">## High-level composition</span></span>
<span id="cb31-387"><a href="#cb31-387" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-388"><a href="#cb31-388" aria-hidden="true" tabindex="-1"></a>To assess the high-level composition of the reads, we ran them through Kraken2 and summarized the results with Bracken.</span>
<span id="cb31-389"><a href="#cb31-389" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-392"><a href="#cb31-392" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb31-393"><a href="#cb31-393" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: prepare-composition</span></span>
<span id="cb31-394"><a href="#cb31-394" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb31-395"><a href="#cb31-395" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-396"><a href="#cb31-396" aria-hidden="true" tabindex="-1"></a>classifications <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Filtered"</span>, <span class="st">"Duplicate"</span>, <span class="st">"Ribosomal"</span>, <span class="st">"Unassigned"</span>,</span>
<span id="cb31-397"><a href="#cb31-397" aria-hidden="true" tabindex="-1"></a>                     <span class="st">"Bacterial"</span>, <span class="st">"Archaeal"</span>, <span class="st">"Viral"</span>, <span class="st">"Human"</span>)</span>
<span id="cb31-398"><a href="#cb31-398" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-399"><a href="#cb31-399" aria-hidden="true" tabindex="-1"></a><span class="co"># Import composition data</span></span>
<span id="cb31-400"><a href="#cb31-400" aria-hidden="true" tabindex="-1"></a>tax_final_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(results_dir, <span class="st">"taxonomy_final"</span>)</span>
<span id="cb31-401"><a href="#cb31-401" aria-hidden="true" tabindex="-1"></a>comp_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(tax_final_dir, <span class="st">"taxonomic_composition.tsv.gz"</span>)</span>
<span id="cb31-402"><a href="#cb31-402" aria-hidden="true" tabindex="-1"></a>comp <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(comp_path) <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(libraries) </span>
<span id="cb31-403"><a href="#cb31-403" aria-hidden="true" tabindex="-1"></a>comp_minor <span class="ot">&lt;-</span> comp <span class="sc">%&gt;%</span> </span>
<span id="cb31-404"><a href="#cb31-404" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(classification <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Archaeal"</span>, <span class="st">"Viral"</span>, <span class="st">"Human"</span>, <span class="st">"Other"</span>))</span>
<span id="cb31-405"><a href="#cb31-405" aria-hidden="true" tabindex="-1"></a>comp_assigned <span class="ot">&lt;-</span> comp <span class="sc">%&gt;%</span></span>
<span id="cb31-406"><a href="#cb31-406" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span> classification <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Filtered"</span>, <span class="st">"Duplicate"</span>, </span>
<span id="cb31-407"><a href="#cb31-407" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">"Ribosomal"</span>, <span class="st">"Unassigned"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb31-408"><a href="#cb31-408" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb31-409"><a href="#cb31-409" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads =</span> n_reads<span class="sc">/</span><span class="fu">sum</span>(n_reads))</span>
<span id="cb31-410"><a href="#cb31-410" aria-hidden="true" tabindex="-1"></a>comp_assigned_minor <span class="ot">&lt;-</span> comp_assigned <span class="sc">%&gt;%</span> </span>
<span id="cb31-411"><a href="#cb31-411" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(classification <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Archaeal"</span>, <span class="st">"Viral"</span>, <span class="st">"Human"</span>, <span class="st">"Other"</span>))</span>
<span id="cb31-412"><a href="#cb31-412" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-413"><a href="#cb31-413" aria-hidden="true" tabindex="-1"></a><span class="co"># Summarize composition</span></span>
<span id="cb31-414"><a href="#cb31-414" aria-hidden="true" tabindex="-1"></a>read_comp_summ <span class="ot">&lt;-</span> comp <span class="sc">%&gt;%</span> </span>
<span id="cb31-415"><a href="#cb31-415" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(classification) <span class="sc">%&gt;%</span></span>
<span id="cb31-416"><a href="#cb31-416" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads =</span> <span class="fu">sum</span>(n_reads), <span class="at">.groups =</span> <span class="st">"drop_last"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb31-417"><a href="#cb31-417" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">n_reads =</span> <span class="fu">replace_na</span>(n_reads,<span class="dv">0</span>),</span>
<span id="cb31-418"><a href="#cb31-418" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads =</span> n_reads<span class="sc">/</span><span class="fu">sum</span>(n_reads),</span>
<span id="cb31-419"><a href="#cb31-419" aria-hidden="true" tabindex="-1"></a>         <span class="at">pc_reads =</span> p_reads<span class="sc">*</span><span class="dv">100</span>)</span>
<span id="cb31-420"><a href="#cb31-420" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-421"><a href="#cb31-421" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb31-422"><a href="#cb31-422" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-423"><a href="#cb31-423" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-426"><a href="#cb31-426" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb31-427"><a href="#cb31-427" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-composition-all</span></span>
<span id="cb31-428"><a href="#cb31-428" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 7</span></span>
<span id="cb31-429"><a href="#cb31-429" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 8</span></span>
<span id="cb31-430"><a href="#cb31-430" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-431"><a href="#cb31-431" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare plotting templates</span></span>
<span id="cb31-432"><a href="#cb31-432" aria-hidden="true" tabindex="-1"></a>g_comp_base <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">x=</span>library, <span class="at">y=</span>p_reads, <span class="at">fill=</span>classification)) <span class="sc">+</span></span>
<span id="cb31-433"><a href="#cb31-433" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">name=</span><span class="st">"Plasma pool"</span>) <span class="sc">+</span></span>
<span id="cb31-434"><a href="#cb31-434" aria-hidden="true" tabindex="-1"></a>  theme_kit <span class="sc">+</span> </span>
<span id="cb31-435"><a href="#cb31-435" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust=</span><span class="dv">0</span>, <span class="at">face=</span><span class="st">"plain"</span>, <span class="at">size=</span><span class="fu">rel</span>(<span class="fl">1.5</span>)))</span>
<span id="cb31-436"><a href="#cb31-436" aria-hidden="true" tabindex="-1"></a>scale_y_pc_reads <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">partial</span>(scale_y_continuous, <span class="at">name =</span> <span class="st">"% Reads"</span>,</span>
<span id="cb31-437"><a href="#cb31-437" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>)</span>
<span id="cb31-438"><a href="#cb31-438" aria-hidden="true" tabindex="-1"></a>geom_comp <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">partial</span>(geom_col, <span class="at">position =</span> <span class="st">"stack"</span>, <span class="at">width =</span> <span class="dv">1</span>)</span>
<span id="cb31-439"><a href="#cb31-439" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-440"><a href="#cb31-440" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot overall composition</span></span>
<span id="cb31-441"><a href="#cb31-441" aria-hidden="true" tabindex="-1"></a>g_comp <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> <span class="fu">geom_comp</span>(<span class="at">data =</span> comp) <span class="sc">+</span></span>
<span id="cb31-442"><a href="#cb31-442" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_pc_reads</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>)) <span class="sc">+</span></span>
<span id="cb31-443"><a href="#cb31-443" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>, <span class="at">name =</span> <span class="st">"Classification"</span>) <span class="sc">+</span></span>
<span id="cb31-444"><a href="#cb31-444" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Read composition (all reads, all groups)"</span>)</span>
<span id="cb31-445"><a href="#cb31-445" aria-hidden="true" tabindex="-1"></a>g_comp</span>
<span id="cb31-446"><a href="#cb31-446" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-447"><a href="#cb31-447" aria-hidden="true" tabindex="-1"></a><span class="co"># Repeat for classified reads only</span></span>
<span id="cb31-448"><a href="#cb31-448" aria-hidden="true" tabindex="-1"></a>palette_assigned <span class="ot">&lt;-</span> <span class="fu">brewer.pal</span>(<span class="dv">9</span>, <span class="st">"Set1"</span>)[<span class="dv">5</span><span class="sc">:</span><span class="dv">9</span>]</span>
<span id="cb31-449"><a href="#cb31-449" aria-hidden="true" tabindex="-1"></a>g_comp_assigned <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb31-450"><a href="#cb31-450" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_comp</span>(<span class="at">data =</span> comp_assigned) <span class="sc">+</span></span>
<span id="cb31-451"><a href="#cb31-451" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_pc_reads</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>)) <span class="sc">+</span></span>
<span id="cb31-452"><a href="#cb31-452" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>palette_assigned, <span class="at">name =</span> <span class="st">"Classification"</span>) <span class="sc">+</span></span>
<span id="cb31-453"><a href="#cb31-453" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Read composition (assigned reads, all groups)"</span>)</span>
<span id="cb31-454"><a href="#cb31-454" aria-hidden="true" tabindex="-1"></a>g_comp_assigned</span>
<span id="cb31-455"><a href="#cb31-455" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-456"><a href="#cb31-456" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot composition of minor components</span></span>
<span id="cb31-457"><a href="#cb31-457" aria-hidden="true" tabindex="-1"></a>palette_minor <span class="ot">&lt;-</span> <span class="fu">brewer.pal</span>(<span class="dv">9</span>, <span class="st">"Set1"</span>)[<span class="dv">6</span><span class="sc">:</span><span class="dv">9</span>]</span>
<span id="cb31-458"><a href="#cb31-458" aria-hidden="true" tabindex="-1"></a>g_comp_minor <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb31-459"><a href="#cb31-459" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_comp</span>(<span class="at">data=</span>comp_minor) <span class="sc">+</span></span>
<span id="cb31-460"><a href="#cb31-460" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_pc_reads</span>() <span class="sc">+</span></span>
<span id="cb31-461"><a href="#cb31-461" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>palette_minor, <span class="at">name =</span> <span class="st">"Classification"</span>) <span class="sc">+</span></span>
<span id="cb31-462"><a href="#cb31-462" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Read composition (all reads, minor groups)"</span>)</span>
<span id="cb31-463"><a href="#cb31-463" aria-hidden="true" tabindex="-1"></a>g_comp_minor</span>
<span id="cb31-464"><a href="#cb31-464" aria-hidden="true" tabindex="-1"></a>g_comp_assigned_minor <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb31-465"><a href="#cb31-465" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_comp</span>(<span class="at">data=</span>comp_assigned_minor) <span class="sc">+</span></span>
<span id="cb31-466"><a href="#cb31-466" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_pc_reads</span>() <span class="sc">+</span></span>
<span id="cb31-467"><a href="#cb31-467" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>palette_minor, <span class="at">name =</span> <span class="st">"Classification"</span>) <span class="sc">+</span></span>
<span id="cb31-468"><a href="#cb31-468" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Read composition (assigned reads, minor groups)"</span>)</span>
<span id="cb31-469"><a href="#cb31-469" aria-hidden="true" tabindex="-1"></a>g_comp_assigned_minor</span>
<span id="cb31-470"><a href="#cb31-470" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb31-471"><a href="#cb31-471" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-472"><a href="#cb31-472" aria-hidden="true" tabindex="-1"></a><span class="fu">## Total viral content</span></span>
<span id="cb31-473"><a href="#cb31-473" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-474"><a href="#cb31-474" aria-hidden="true" tabindex="-1"></a>Total viral fraction average $1.81 \times 10^{-2}$ across samples. As a fraction of assigned (rather than total) reads, this jumped to $1.60 \times {-1}$:</span>
<span id="cb31-475"><a href="#cb31-475" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-478"><a href="#cb31-478" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb31-479"><a href="#cb31-479" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: p-viral</span></span>
<span id="cb31-480"><a href="#cb31-480" aria-hidden="true" tabindex="-1"></a>p_reads_viral_all <span class="ot">&lt;-</span> comp <span class="sc">%&gt;%</span> <span class="fu">filter</span>(classification <span class="sc">==</span> <span class="st">"Viral"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb31-481"><a href="#cb31-481" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">read_group =</span> <span class="st">"All reads"</span>)</span>
<span id="cb31-482"><a href="#cb31-482" aria-hidden="true" tabindex="-1"></a>p_reads_viral_assigned <span class="ot">&lt;-</span> comp_assigned <span class="sc">%&gt;%</span> <span class="fu">filter</span>(classification <span class="sc">==</span> <span class="st">"Viral"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb31-483"><a href="#cb31-483" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">read_group =</span> <span class="st">"Classified reads"</span>)</span>
<span id="cb31-484"><a href="#cb31-484" aria-hidden="true" tabindex="-1"></a>p_reads_viral <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(p_reads_viral_all, p_reads_viral_assigned)</span>
<span id="cb31-485"><a href="#cb31-485" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-486"><a href="#cb31-486" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb31-487"><a href="#cb31-487" aria-hidden="true" tabindex="-1"></a>g_viral <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(p_reads_viral, <span class="fu">aes</span>(<span class="at">x=</span>library, <span class="at">y=</span>p_reads)) <span class="sc">+</span></span>
<span id="cb31-488"><a href="#cb31-488" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb31-489"><a href="#cb31-489" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">name=</span><span class="st">"Plasma pool"</span>) <span class="sc">+</span></span>
<span id="cb31-490"><a href="#cb31-490" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>(<span class="at">name=</span><span class="st">"Viral read fraction"</span>) <span class="sc">+</span></span>
<span id="cb31-491"><a href="#cb31-491" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(.<span class="sc">~</span>read_group, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb31-492"><a href="#cb31-492" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color=</span><span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>), <span class="at">shape=</span><span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>),</span>
<span id="cb31-493"><a href="#cb31-493" aria-hidden="true" tabindex="-1"></a>         <span class="at">linetype=</span><span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>)) <span class="sc">+</span></span>
<span id="cb31-494"><a href="#cb31-494" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb31-495"><a href="#cb31-495" aria-hidden="true" tabindex="-1"></a>g_viral</span>
<span id="cb31-496"><a href="#cb31-496" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb31-497"><a href="#cb31-497" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-498"><a href="#cb31-498" aria-hidden="true" tabindex="-1"></a><span class="fu">## Taxonomic composition of viruses</span></span>
<span id="cb31-499"><a href="#cb31-499" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-500"><a href="#cb31-500" aria-hidden="true" tabindex="-1"></a>The two dominant viruses we see are Anellovirdae and Rhabdovirdae (spike-in). Followed by these two viral families is Flavivirdae, and lastly, also by a much smaller percent, Microviridae (spike-in). The threshold for the label "other" are the set of families that make up less than 1% composition in all samples.</span>
<span id="cb31-501"><a href="#cb31-501" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-504"><a href="#cb31-504" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb31-505"><a href="#cb31-505" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: extract-viral-taxa-second</span></span>
<span id="cb31-506"><a href="#cb31-506" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-507"><a href="#cb31-507" aria-hidden="true" tabindex="-1"></a><span class="co"># Get viral taxonomy</span></span>
<span id="cb31-508"><a href="#cb31-508" aria-hidden="true" tabindex="-1"></a>viral_taxa_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"total-virus-db.tsv.gz"</span>)</span>
<span id="cb31-509"><a href="#cb31-509" aria-hidden="true" tabindex="-1"></a>viral_taxa <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(viral_taxa_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb31-510"><a href="#cb31-510" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-511"><a href="#cb31-511" aria-hidden="true" tabindex="-1"></a><span class="co"># Get Kraken reports</span></span>
<span id="cb31-512"><a href="#cb31-512" aria-hidden="true" tabindex="-1"></a>reports_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(tax_final_dir, <span class="st">"kraken_reports.tsv.gz"</span>)</span>
<span id="cb31-513"><a href="#cb31-513" aria-hidden="true" tabindex="-1"></a>reports <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(reports_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb31-514"><a href="#cb31-514" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(sample)</span>
<span id="cb31-515"><a href="#cb31-515" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-516"><a href="#cb31-516" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter to viral taxa</span></span>
<span id="cb31-517"><a href="#cb31-517" aria-hidden="true" tabindex="-1"></a>kraken_reports_viral <span class="ot">&lt;-</span> <span class="fu">filter</span>(reports, taxid <span class="sc">%in%</span> viral_taxa<span class="sc">$</span>taxid) <span class="sc">%&gt;%</span></span>
<span id="cb31-518"><a href="#cb31-518" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb31-519"><a href="#cb31-519" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_viral =</span> n_reads_clade<span class="sc">/</span>n_reads_clade[<span class="dv">1</span>])</span>
<span id="cb31-520"><a href="#cb31-520" aria-hidden="true" tabindex="-1"></a>kraken_reports_viral_cleaned <span class="ot">&lt;-</span> kraken_reports_viral <span class="sc">%&gt;%</span></span>
<span id="cb31-521"><a href="#cb31-521" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>pc_reads_total, <span class="sc">-</span>n_reads_direct, <span class="sc">-</span><span class="fu">contains</span>(<span class="st">"minimizers"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb31-522"><a href="#cb31-522" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, taxid, p_reads_viral, n_reads_clade, <span class="fu">everything</span>()) <span class="sc">%&gt;%</span> ungroup</span>
<span id="cb31-523"><a href="#cb31-523" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-524"><a href="#cb31-524" aria-hidden="true" tabindex="-1"></a>viral_classes <span class="ot">&lt;-</span> kraken_reports_viral_cleaned <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">==</span> <span class="st">"C"</span>)</span>
<span id="cb31-525"><a href="#cb31-525" aria-hidden="true" tabindex="-1"></a>viral_families <span class="ot">&lt;-</span> kraken_reports_viral_cleaned <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">==</span> <span class="st">"F"</span>)</span>
<span id="cb31-526"><a href="#cb31-526" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-527"><a href="#cb31-527" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb31-528"><a href="#cb31-528" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-531"><a href="#cb31-531" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb31-532"><a href="#cb31-532" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: viral-family-composition</span></span>
<span id="cb31-533"><a href="#cb31-533" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 4</span></span>
<span id="cb31-534"><a href="#cb31-534" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 8</span></span>
<span id="cb31-535"><a href="#cb31-535" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-536"><a href="#cb31-536" aria-hidden="true" tabindex="-1"></a>major_threshold <span class="ot">&lt;-</span> <span class="fl">0.01</span></span>
<span id="cb31-537"><a href="#cb31-537" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-538"><a href="#cb31-538" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify major viral families</span></span>
<span id="cb31-539"><a href="#cb31-539" aria-hidden="true" tabindex="-1"></a>viral_families_major_tab <span class="ot">&lt;-</span> viral_families <span class="sc">%&gt;%</span> </span>
<span id="cb31-540"><a href="#cb31-540" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb31-541"><a href="#cb31-541" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads_viral_max =</span> <span class="fu">max</span>(p_reads_viral), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb31-542"><a href="#cb31-542" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_viral_max <span class="sc">&gt;=</span> major_threshold)</span>
<span id="cb31-543"><a href="#cb31-543" aria-hidden="true" tabindex="-1"></a>viral_families_major_list <span class="ot">&lt;-</span> viral_families_major_tab <span class="sc">%&gt;%</span> <span class="fu">pull</span>(name)</span>
<span id="cb31-544"><a href="#cb31-544" aria-hidden="true" tabindex="-1"></a>viral_families_major <span class="ot">&lt;-</span> viral_families <span class="sc">%&gt;%</span> </span>
<span id="cb31-545"><a href="#cb31-545" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(name <span class="sc">%in%</span> viral_families_major_list) <span class="sc">%&gt;%</span></span>
<span id="cb31-546"><a href="#cb31-546" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, taxid, sample, p_reads_viral)</span>
<span id="cb31-547"><a href="#cb31-547" aria-hidden="true" tabindex="-1"></a>viral_families_minor <span class="ot">&lt;-</span> viral_families_major <span class="sc">%&gt;%</span> </span>
<span id="cb31-548"><a href="#cb31-548" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb31-549"><a href="#cb31-549" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads_viral_major =</span> <span class="fu">sum</span>(p_reads_viral), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb31-550"><a href="#cb31-550" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="st">"Other"</span>, <span class="at">taxid=</span><span class="cn">NA</span>, <span class="at">p_reads_viral =</span> <span class="dv">1</span><span class="sc">-</span>p_reads_viral_major) <span class="sc">%&gt;%</span></span>
<span id="cb31-551"><a href="#cb31-551" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, taxid, sample, p_reads_viral)</span>
<span id="cb31-552"><a href="#cb31-552" aria-hidden="true" tabindex="-1"></a>viral_families_display <span class="ot">&lt;-</span> viral_families_major <span class="sc">%&gt;%</span> </span>
<span id="cb31-553"><a href="#cb31-553" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_rows</span>(viral_families_minor) <span class="sc">%&gt;%</span></span>
<span id="cb31-554"><a href="#cb31-554" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_viral)) <span class="sc">%&gt;%</span> </span>
<span id="cb31-555"><a href="#cb31-555" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">factor</span>(name, <span class="at">levels=</span><span class="fu">c</span>(viral_families_major_list, <span class="st">"Other"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb31-556"><a href="#cb31-556" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_viral, <span class="at">classification=</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb31-557"><a href="#cb31-557" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">'sample'</span>)</span>
<span id="cb31-558"><a href="#cb31-558" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-559"><a href="#cb31-559" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb31-560"><a href="#cb31-560" aria-hidden="true" tabindex="-1"></a>g_families <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb31-561"><a href="#cb31-561" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_comp</span>(<span class="at">data=</span>viral_families_display) <span class="sc">+</span></span>
<span id="cb31-562"><a href="#cb31-562" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Viral Reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb31-563"><a href="#cb31-563" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb31-564"><a href="#cb31-564" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb31-565"><a href="#cb31-565" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">'Accent'</span>)</span>
<span id="cb31-566"><a href="#cb31-566" aria-hidden="true" tabindex="-1"></a>g_families</span>
<span id="cb31-567"><a href="#cb31-567" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb31-568"><a href="#cb31-568" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-569"><a href="#cb31-569" aria-hidden="true" tabindex="-1"></a>Excluding *Anellovirdae* and *Rhabdovirdae*, remaining viral sequences are distributed across a wide variety:</span>
<span id="cb31-570"><a href="#cb31-570" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-573"><a href="#cb31-573" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb31-574"><a href="#cb31-574" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: viral-family-composition-exclusion</span></span>
<span id="cb31-575"><a href="#cb31-575" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 4</span></span>
<span id="cb31-576"><a href="#cb31-576" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 8</span></span>
<span id="cb31-577"><a href="#cb31-577" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-578"><a href="#cb31-578" aria-hidden="true" tabindex="-1"></a>major_threshold_adj <span class="ot">&lt;-</span> <span class="fl">0.05</span></span>
<span id="cb31-579"><a href="#cb31-579" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-580"><a href="#cb31-580" aria-hidden="true" tabindex="-1"></a><span class="co"># Adjust viral family counts</span></span>
<span id="cb31-581"><a href="#cb31-581" aria-hidden="true" tabindex="-1"></a>viral_families_adj <span class="ot">&lt;-</span> viral_families <span class="sc">%&gt;%</span></span>
<span id="cb31-582"><a href="#cb31-582" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span>(name <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Rhabdoviridae"</span>,<span class="st">"Anelloviridae"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb31-583"><a href="#cb31-583" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb31-584"><a href="#cb31-584" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_viral =</span> p_reads_viral<span class="sc">/</span><span class="fu">sum</span>(p_reads_viral))</span>
<span id="cb31-585"><a href="#cb31-585" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-586"><a href="#cb31-586" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify major viral families</span></span>
<span id="cb31-587"><a href="#cb31-587" aria-hidden="true" tabindex="-1"></a>viral_families_major_tab <span class="ot">&lt;-</span> viral_families_adj <span class="sc">%&gt;%</span> </span>
<span id="cb31-588"><a href="#cb31-588" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb31-589"><a href="#cb31-589" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads_viral_max =</span> <span class="fu">max</span>(p_reads_viral), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb31-590"><a href="#cb31-590" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_viral_max <span class="sc">&gt;=</span> major_threshold)</span>
<span id="cb31-591"><a href="#cb31-591" aria-hidden="true" tabindex="-1"></a>viral_families_major_list <span class="ot">&lt;-</span> viral_families_major_tab <span class="sc">%&gt;%</span> <span class="fu">pull</span>(name)</span>
<span id="cb31-592"><a href="#cb31-592" aria-hidden="true" tabindex="-1"></a>viral_families_major <span class="ot">&lt;-</span> viral_families_adj <span class="sc">%&gt;%</span> </span>
<span id="cb31-593"><a href="#cb31-593" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(name <span class="sc">%in%</span> viral_families_major_list) <span class="sc">%&gt;%</span></span>
<span id="cb31-594"><a href="#cb31-594" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, taxid, sample, p_reads_viral)</span>
<span id="cb31-595"><a href="#cb31-595" aria-hidden="true" tabindex="-1"></a>viral_families_minor <span class="ot">&lt;-</span> viral_families_major <span class="sc">%&gt;%</span> </span>
<span id="cb31-596"><a href="#cb31-596" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb31-597"><a href="#cb31-597" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads_viral_major =</span> <span class="fu">sum</span>(p_reads_viral), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb31-598"><a href="#cb31-598" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="st">"Other"</span>, <span class="at">taxid=</span><span class="cn">NA</span>, <span class="at">p_reads_viral =</span> <span class="dv">1</span><span class="sc">-</span>p_reads_viral_major) <span class="sc">%&gt;%</span></span>
<span id="cb31-599"><a href="#cb31-599" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, taxid, sample, p_reads_viral)</span>
<span id="cb31-600"><a href="#cb31-600" aria-hidden="true" tabindex="-1"></a>viral_families_display <span class="ot">&lt;-</span> viral_families_major <span class="sc">%&gt;%</span> </span>
<span id="cb31-601"><a href="#cb31-601" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_rows</span>(viral_families_minor) <span class="sc">%&gt;%</span></span>
<span id="cb31-602"><a href="#cb31-602" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_viral)) <span class="sc">%&gt;%</span> </span>
<span id="cb31-603"><a href="#cb31-603" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">factor</span>(name, <span class="at">levels=</span><span class="fu">c</span>(viral_families_major_list, <span class="st">"Other"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb31-604"><a href="#cb31-604" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_viral, <span class="at">classification=</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb31-605"><a href="#cb31-605" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by =</span> <span class="st">'sample'</span>)</span>
<span id="cb31-606"><a href="#cb31-606" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-607"><a href="#cb31-607" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb31-608"><a href="#cb31-608" aria-hidden="true" tabindex="-1"></a>palette_viral <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">brewer.pal</span>(<span class="dv">12</span>, <span class="st">"Set3"</span>), <span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">"Dark2"</span>), <span class="fu">brewer.pal</span>(<span class="dv">9</span>, <span class="st">"Set1"</span>))</span>
<span id="cb31-609"><a href="#cb31-609" aria-hidden="true" tabindex="-1"></a>g_families_adj <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb31-610"><a href="#cb31-610" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_comp</span>(<span class="at">data=</span>viral_families_display) <span class="sc">+</span></span>
<span id="cb31-611"><a href="#cb31-611" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Viral Reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb31-612"><a href="#cb31-612" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb31-613"><a href="#cb31-613" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb31-614"><a href="#cb31-614" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>palette_viral, <span class="at">name =</span> <span class="st">"Viral class"</span>)</span>
<span id="cb31-615"><a href="#cb31-615" aria-hidden="true" tabindex="-1"></a>g_families_adj</span>
<span id="cb31-616"><a href="#cb31-616" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb31-617"><a href="#cb31-617" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-618"><a href="#cb31-618" aria-hidden="true" tabindex="-1"></a><span class="fu"># Human-infecting virus reads</span></span>
<span id="cb31-619"><a href="#cb31-619" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-620"><a href="#cb31-620" aria-hidden="true" tabindex="-1"></a><span class="fu">## Overall relative abundance</span></span>
<span id="cb31-621"><a href="#cb31-621" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-622"><a href="#cb31-622" aria-hidden="true" tabindex="-1"></a>I calculated the relative abundance of human-infecting viruses in two ways:</span>
<span id="cb31-623"><a href="#cb31-623" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-624"><a href="#cb31-624" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>First, as the total number of deduplicated human-virus reads in each sample, divided by the number of raw reads ("All reads").</span>
<span id="cb31-625"><a href="#cb31-625" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-626"><a href="#cb31-626" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Second, as a fraction of preprocessed (cleaned, deduplicated, computationally ribodepleted) reads ("Preprocessed reads").</span>
<span id="cb31-627"><a href="#cb31-627" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-628"><a href="#cb31-628" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-629"><a href="#cb31-629" aria-hidden="true" tabindex="-1"></a><span class="co">[</span><span class="ot">Analyze</span><span class="co">]</span></span>
<span id="cb31-630"><a href="#cb31-630" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-633"><a href="#cb31-633" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb31-634"><a href="#cb31-634" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: prepare-hv</span></span>
<span id="cb31-635"><a href="#cb31-635" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb31-636"><a href="#cb31-636" aria-hidden="true" tabindex="-1"></a><span class="co">#| cache.lazy: false</span></span>
<span id="cb31-637"><a href="#cb31-637" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-638"><a href="#cb31-638" aria-hidden="true" tabindex="-1"></a><span class="co"># Import and format reads</span></span>
<span id="cb31-639"><a href="#cb31-639" aria-hidden="true" tabindex="-1"></a>hv_reads_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(hv_dir, <span class="st">"hv_hits_putative_collapsed.tsv.gz"</span>)</span>
<span id="cb31-640"><a href="#cb31-640" aria-hidden="true" tabindex="-1"></a>mrg_hv <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(hv_reads_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb31-641"><a href="#cb31-641" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb31-642"><a href="#cb31-642" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">kraken_label =</span> <span class="fu">ifelse</span>(assigned_hv, <span class="st">"Kraken2 HV assignment"</span>,</span>
<span id="cb31-643"><a href="#cb31-643" aria-hidden="true" tabindex="-1"></a>                               <span class="st">"No Kraken2 assignment"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb31-644"><a href="#cb31-644" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">adj_score_max =</span> <span class="fu">pmax</span>(adj_score_fwd, adj_score_rev),</span>
<span id="cb31-645"><a href="#cb31-645" aria-hidden="true" tabindex="-1"></a>         <span class="at">highscore =</span> adj_score_max <span class="sc">&gt;=</span> <span class="dv">20</span>,</span>
<span id="cb31-646"><a href="#cb31-646" aria-hidden="true" tabindex="-1"></a>         <span class="at">hv_status =</span> assigned_hv <span class="sc">|</span> highscore) <span class="sc">%&gt;%</span></span>
<span id="cb31-647"><a href="#cb31-647" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">taxid_all =</span> taxid, <span class="at">taxid =</span> taxid_best)</span>
<span id="cb31-648"><a href="#cb31-648" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb31-649"><a href="#cb31-649" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-652"><a href="#cb31-652" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb31-653"><a href="#cb31-653" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: count-hv-reads</span></span>
<span id="cb31-654"><a href="#cb31-654" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 8</span></span>
<span id="cb31-655"><a href="#cb31-655" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb31-656"><a href="#cb31-656" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb31-657"><a href="#cb31-657" aria-hidden="true" tabindex="-1"></a><span class="co"># Get raw read counts</span></span>
<span id="cb31-658"><a href="#cb31-658" aria-hidden="true" tabindex="-1"></a>read_counts_raw <span class="ot">&lt;-</span> <span class="fu">filter</span>(basic_stats_raw) <span class="sc">%&gt;%</span></span>
<span id="cb31-659"><a href="#cb31-659" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, <span class="at">n_reads_raw =</span> n_read_pairs)</span>
<span id="cb31-660"><a href="#cb31-660" aria-hidden="true" tabindex="-1"></a>read_counts_preproc <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"ribo_initial"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb31-661"><a href="#cb31-661" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, <span class="at">n_reads_preproc =</span> n_read_pairs)</span>
<span id="cb31-662"><a href="#cb31-662" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-663"><a href="#cb31-663" aria-hidden="true" tabindex="-1"></a><span class="co"># Get HV read counts</span></span>
<span id="cb31-664"><a href="#cb31-664" aria-hidden="true" tabindex="-1"></a>read_counts_hv <span class="ot">&lt;-</span> mrg_hv <span class="sc">%&gt;%</span> <span class="fu">filter</span>(hv_status) <span class="sc">%&gt;%</span> </span>
<span id="cb31-665"><a href="#cb31-665" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb31-666"><a href="#cb31-666" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name=</span><span class="st">"n_reads_hv"</span>)</span>
<span id="cb31-667"><a href="#cb31-667" aria-hidden="true" tabindex="-1"></a>read_counts <span class="ot">&lt;-</span> read_counts_raw <span class="sc">%&gt;%</span></span>
<span id="cb31-668"><a href="#cb31-668" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(read_counts_hv, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"sample"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb31-669"><a href="#cb31-669" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">n_reads_hv =</span> <span class="fu">replace_na</span>(n_reads_hv, <span class="dv">0</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb31-670"><a href="#cb31-670" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(read_counts_preproc, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"sample"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb31-671"><a href="#cb31-671" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"sample"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb31-672"><a href="#cb31-672" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, n_reads_raw, n_reads_preproc, n_reads_hv) <span class="sc">%&gt;%</span></span>
<span id="cb31-673"><a href="#cb31-673" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">n_samples =</span> <span class="dv">1</span>,</span>
<span id="cb31-674"><a href="#cb31-674" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_total =</span> n_reads_hv<span class="sc">/</span>n_reads_raw,</span>
<span id="cb31-675"><a href="#cb31-675" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_preproc =</span> n_reads_hv<span class="sc">/</span>n_reads_preproc)</span>
<span id="cb31-676"><a href="#cb31-676" aria-hidden="true" tabindex="-1"></a>read_counts_long <span class="ot">&lt;-</span> read_counts <span class="sc">%&gt;%</span></span>
<span id="cb31-677"><a href="#cb31-677" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="fu">starts_with</span>(<span class="st">"p_reads"</span>), <span class="at">names_to=</span><span class="st">"read_group"</span>, <span class="at">values_to=</span><span class="st">"p_reads"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb31-678"><a href="#cb31-678" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">read_group =</span> <span class="fu">ifelse</span>(read_group <span class="sc">==</span> <span class="st">"p_reads_total"</span>, <span class="st">"All reads"</span>, <span class="st">"Preprocessed reads"</span>))</span>
<span id="cb31-679"><a href="#cb31-679" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-680"><a href="#cb31-680" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine for display</span></span>
<span id="cb31-681"><a href="#cb31-681" aria-hidden="true" tabindex="-1"></a>read_counts_agg <span class="ot">&lt;-</span> read_counts <span class="sc">%&gt;%</span></span>
<span id="cb31-682"><a href="#cb31-682" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_total =</span> n_reads_hv<span class="sc">/</span>n_reads_raw,</span>
<span id="cb31-683"><a href="#cb31-683" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_preproc =</span> n_reads_hv<span class="sc">/</span>n_reads_preproc) <span class="sc">%&gt;%</span></span>
<span id="cb31-684"><a href="#cb31-684" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"sample"</span>))</span>
<span id="cb31-685"><a href="#cb31-685" aria-hidden="true" tabindex="-1"></a>read_counts_agg_long <span class="ot">&lt;-</span> read_counts_agg <span class="sc">%&gt;%</span></span>
<span id="cb31-686"><a href="#cb31-686" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="fu">starts_with</span>(<span class="st">"p_reads"</span>), <span class="at">names_to=</span><span class="st">"read_group"</span>, <span class="at">values_to=</span><span class="st">"p_reads"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb31-687"><a href="#cb31-687" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">read_group =</span> <span class="fu">ifelse</span>(read_group <span class="sc">==</span> <span class="st">"p_reads_total"</span>, <span class="st">"All reads"</span>, <span class="st">"Preprocessed reads"</span>)) </span>
<span id="cb31-688"><a href="#cb31-688" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-689"><a href="#cb31-689" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize</span></span>
<span id="cb31-690"><a href="#cb31-690" aria-hidden="true" tabindex="-1"></a>g_read_counts <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(read_counts_agg_long, <span class="fu">aes</span>(<span class="at">x=</span>library, <span class="at">y=</span>p_reads)) <span class="sc">+</span></span>
<span id="cb31-691"><a href="#cb31-691" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb31-692"><a href="#cb31-692" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>(<span class="at">name =</span> <span class="st">"Unique human-viral read fraction"</span>) <span class="sc">+</span></span>
<span id="cb31-693"><a href="#cb31-693" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(.<span class="sc">~</span>read_group, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb31-694"><a href="#cb31-694" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb31-695"><a href="#cb31-695" aria-hidden="true" tabindex="-1"></a>g_read_counts</span>
<span id="cb31-696"><a href="#cb31-696" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb31-697"><a href="#cb31-697" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-698"><a href="#cb31-698" aria-hidden="true" tabindex="-1"></a><span class="fu">## Overall taxonomy and composition</span></span>
<span id="cb31-699"><a href="#cb31-699" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-700"><a href="#cb31-700" aria-hidden="true" tabindex="-1"></a>Composition of HV reads was not greatly changed from when looking at all viral reads. The two dominant viruses we see are Anellovirdae and Rhabdovirdae. Followed by these two viral families is Flavivirdae, and lastly, also by a much smaller percent, Microviridae. The threshold for the label "other" are the set of families that make up less than 5% composition in all samples.</span>
<span id="cb31-701"><a href="#cb31-701" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-704"><a href="#cb31-704" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb31-705"><a href="#cb31-705" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: raise-hv-taxa</span></span>
<span id="cb31-706"><a href="#cb31-706" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb31-707"><a href="#cb31-707" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb31-708"><a href="#cb31-708" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter samples and add viral taxa information</span></span>
<span id="cb31-709"><a href="#cb31-709" aria-hidden="true" tabindex="-1"></a>samples_keep <span class="ot">&lt;-</span> read_counts <span class="sc">%&gt;%</span> <span class="fu">filter</span>(n_reads_hv <span class="sc">&gt;</span> <span class="dv">5</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(sample)</span>
<span id="cb31-710"><a href="#cb31-710" aria-hidden="true" tabindex="-1"></a>mrg_hv_named <span class="ot">&lt;-</span> mrg_hv <span class="sc">%&gt;%</span> <span class="fu">filter</span>(sample <span class="sc">%in%</span> samples_keep, hv_status) <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(viral_taxa, <span class="at">by=</span><span class="st">"taxid"</span>) </span>
<span id="cb31-711"><a href="#cb31-711" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-712"><a href="#cb31-712" aria-hidden="true" tabindex="-1"></a><span class="co"># Discover viral species &amp; genera for HV reads</span></span>
<span id="cb31-713"><a href="#cb31-713" aria-hidden="true" tabindex="-1"></a>raise_rank <span class="ot">&lt;-</span> <span class="cf">function</span>(read_db, taxid_db, <span class="at">out_rank =</span> <span class="st">"species"</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>){</span>
<span id="cb31-714"><a href="#cb31-714" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get higher ranks than search rank</span></span>
<span id="cb31-715"><a href="#cb31-715" aria-hidden="true" tabindex="-1"></a>  ranks <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"subspecies"</span>, <span class="st">"species"</span>, <span class="st">"subgenus"</span>, <span class="st">"genus"</span>, <span class="st">"subfamily"</span>, <span class="st">"family"</span>, <span class="st">"suborder"</span>, <span class="st">"order"</span>, <span class="st">"class"</span>, <span class="st">"subphylum"</span>, <span class="st">"phylum"</span>, <span class="st">"kingdom"</span>, <span class="st">"superkingdom"</span>)</span>
<span id="cb31-716"><a href="#cb31-716" aria-hidden="true" tabindex="-1"></a>  rank_match <span class="ot">&lt;-</span> <span class="fu">which.max</span>(ranks <span class="sc">==</span> out_rank)</span>
<span id="cb31-717"><a href="#cb31-717" aria-hidden="true" tabindex="-1"></a>  high_ranks <span class="ot">&lt;-</span> ranks[rank_match<span class="sc">:</span><span class="fu">length</span>(ranks)]</span>
<span id="cb31-718"><a href="#cb31-718" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Merge read DB and taxid DB</span></span>
<span id="cb31-719"><a href="#cb31-719" aria-hidden="true" tabindex="-1"></a>  reads <span class="ot">&lt;-</span> read_db <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>parent_taxid, <span class="sc">-</span>rank, <span class="sc">-</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb31-720"><a href="#cb31-720" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(taxid_db, <span class="at">by=</span><span class="st">"taxid"</span>)</span>
<span id="cb31-721"><a href="#cb31-721" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract sequences that are already at appropriate rank</span></span>
<span id="cb31-722"><a href="#cb31-722" aria-hidden="true" tabindex="-1"></a>  reads_rank <span class="ot">&lt;-</span> <span class="fu">filter</span>(reads, rank <span class="sc">==</span> out_rank)</span>
<span id="cb31-723"><a href="#cb31-723" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Drop sequences at a higher rank and return unclassified sequences</span></span>
<span id="cb31-724"><a href="#cb31-724" aria-hidden="true" tabindex="-1"></a>  reads_norank <span class="ot">&lt;-</span> reads <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">!=</span> out_rank, <span class="sc">!</span>rank <span class="sc">%in%</span> high_ranks, <span class="sc">!</span><span class="fu">is.na</span>(taxid))</span>
<span id="cb31-725"><a href="#cb31-725" aria-hidden="true" tabindex="-1"></a>  <span class="cf">while</span>(<span class="fu">nrow</span>(reads_norank) <span class="sc">&gt;</span> <span class="dv">0</span>){ <span class="co"># As long as there are unclassified sequences...</span></span>
<span id="cb31-726"><a href="#cb31-726" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Promote read taxids and re-merge with taxid DB, then re-classify and filter</span></span>
<span id="cb31-727"><a href="#cb31-727" aria-hidden="true" tabindex="-1"></a>    reads_remaining <span class="ot">&lt;-</span> reads_norank <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">taxid =</span> parent_taxid) <span class="sc">%&gt;%</span></span>
<span id="cb31-728"><a href="#cb31-728" aria-hidden="true" tabindex="-1"></a>      <span class="fu">select</span>(<span class="sc">-</span>parent_taxid, <span class="sc">-</span>rank, <span class="sc">-</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb31-729"><a href="#cb31-729" aria-hidden="true" tabindex="-1"></a>      <span class="fu">left_join</span>(taxid_db, <span class="at">by=</span><span class="st">"taxid"</span>)</span>
<span id="cb31-730"><a href="#cb31-730" aria-hidden="true" tabindex="-1"></a>    reads_rank <span class="ot">&lt;-</span> reads_remaining <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">==</span> out_rank) <span class="sc">%&gt;%</span></span>
<span id="cb31-731"><a href="#cb31-731" aria-hidden="true" tabindex="-1"></a>      <span class="fu">bind_rows</span>(reads_rank)</span>
<span id="cb31-732"><a href="#cb31-732" aria-hidden="true" tabindex="-1"></a>    reads_norank <span class="ot">&lt;-</span> reads_remaining <span class="sc">%&gt;%</span></span>
<span id="cb31-733"><a href="#cb31-733" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(rank <span class="sc">!=</span> out_rank, <span class="sc">!</span>rank <span class="sc">%in%</span> high_ranks, <span class="sc">!</span><span class="fu">is.na</span>(taxid))</span>
<span id="cb31-734"><a href="#cb31-734" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb31-735"><a href="#cb31-735" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Finally, extract and append reads that were excluded during the process</span></span>
<span id="cb31-736"><a href="#cb31-736" aria-hidden="true" tabindex="-1"></a>  reads_dropped <span class="ot">&lt;-</span> reads <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span>seq_id <span class="sc">%in%</span> reads_rank<span class="sc">$</span>seq_id)</span>
<span id="cb31-737"><a href="#cb31-737" aria-hidden="true" tabindex="-1"></a>  reads_out <span class="ot">&lt;-</span> reads_rank <span class="sc">%&gt;%</span> <span class="fu">bind_rows</span>(reads_dropped) <span class="sc">%&gt;%</span></span>
<span id="cb31-738"><a href="#cb31-738" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(<span class="sc">-</span>parent_taxid, <span class="sc">-</span>rank, <span class="sc">-</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb31-739"><a href="#cb31-739" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(taxid_db, <span class="at">by=</span><span class="st">"taxid"</span>)</span>
<span id="cb31-740"><a href="#cb31-740" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(reads_out)</span>
<span id="cb31-741"><a href="#cb31-741" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb31-742"><a href="#cb31-742" aria-hidden="true" tabindex="-1"></a>hv_reads_species <span class="ot">&lt;-</span> <span class="fu">raise_rank</span>(mrg_hv_named, viral_taxa, <span class="st">"species"</span>)</span>
<span id="cb31-743"><a href="#cb31-743" aria-hidden="true" tabindex="-1"></a>hv_reads_genus <span class="ot">&lt;-</span> <span class="fu">raise_rank</span>(mrg_hv_named, viral_taxa, <span class="st">"genus"</span>)</span>
<span id="cb31-744"><a href="#cb31-744" aria-hidden="true" tabindex="-1"></a>hv_reads_family <span class="ot">&lt;-</span> <span class="fu">raise_rank</span>(mrg_hv_named, viral_taxa, <span class="st">"family"</span>)</span>
<span id="cb31-745"><a href="#cb31-745" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb31-746"><a href="#cb31-746" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-749"><a href="#cb31-749" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb31-750"><a href="#cb31-750" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-family</span></span>
<span id="cb31-751"><a href="#cb31-751" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 5</span></span>
<span id="cb31-752"><a href="#cb31-752" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 7</span></span>
<span id="cb31-753"><a href="#cb31-753" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb31-754"><a href="#cb31-754" aria-hidden="true" tabindex="-1"></a>threshold_major_family <span class="ot">&lt;-</span> <span class="fl">0.05</span></span>
<span id="cb31-755"><a href="#cb31-755" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-756"><a href="#cb31-756" aria-hidden="true" tabindex="-1"></a><span class="co"># Count reads for each human-viral family</span></span>
<span id="cb31-757"><a href="#cb31-757" aria-hidden="true" tabindex="-1"></a>hv_family_counts <span class="ot">&lt;-</span> hv_reads_family <span class="sc">%&gt;%</span> </span>
<span id="cb31-758"><a href="#cb31-758" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb31-759"><a href="#cb31-759" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb31-760"><a href="#cb31-760" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb31-761"><a href="#cb31-761" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_hv =</span> n_reads_hv<span class="sc">/</span><span class="fu">sum</span>(n_reads_hv))</span>
<span id="cb31-762"><a href="#cb31-762" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-763"><a href="#cb31-763" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify high-ranking families and group others</span></span>
<span id="cb31-764"><a href="#cb31-764" aria-hidden="true" tabindex="-1"></a>hv_family_major_tab <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span> </span>
<span id="cb31-765"><a href="#cb31-765" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_hv <span class="sc">==</span> <span class="fu">max</span>(p_reads_hv)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb31-766"><a href="#cb31-766" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_hv)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(p_reads_hv <span class="sc">&gt;</span> threshold_major_family)</span>
<span id="cb31-767"><a href="#cb31-767" aria-hidden="true" tabindex="-1"></a>hv_family_counts_major <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span></span>
<span id="cb31-768"><a href="#cb31-768" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">ifelse</span>(name <span class="sc">%in%</span> hv_family_major_tab<span class="sc">$</span>name, name, <span class="st">"Other"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb31-769"><a href="#cb31-769" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, name_display) <span class="sc">%&gt;%</span></span>
<span id="cb31-770"><a href="#cb31-770" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_hv =</span> <span class="fu">sum</span>(n_reads_hv), <span class="at">p_reads_hv =</span> <span class="fu">sum</span>(p_reads_hv), </span>
<span id="cb31-771"><a href="#cb31-771" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb31-772"><a href="#cb31-772" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">factor</span>(name_display, </span>
<span id="cb31-773"><a href="#cb31-773" aria-hidden="true" tabindex="-1"></a>                               <span class="at">levels =</span> <span class="fu">c</span>(hv_family_major_tab<span class="sc">$</span>name, <span class="st">"Other"</span>)))</span>
<span id="cb31-774"><a href="#cb31-774" aria-hidden="true" tabindex="-1"></a>hv_family_counts_display <span class="ot">&lt;-</span> hv_family_counts_major <span class="sc">%&gt;%</span></span>
<span id="cb31-775"><a href="#cb31-775" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_hv, <span class="at">classification =</span> name_display) <span class="sc">%&gt;%</span></span>
<span id="cb31-776"><a href="#cb31-776" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by =</span> <span class="st">'sample'</span>)</span>
<span id="cb31-777"><a href="#cb31-777" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-778"><a href="#cb31-778" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb31-779"><a href="#cb31-779" aria-hidden="true" tabindex="-1"></a>g_hv_family <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb31-780"><a href="#cb31-780" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">data=</span>hv_family_counts_display, <span class="at">position =</span> <span class="st">"stack"</span>, <span class="at">width=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb31-781"><a href="#cb31-781" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% HV Reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb31-782"><a href="#cb31-782" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb31-783"><a href="#cb31-783" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb31-784"><a href="#cb31-784" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">'Accent'</span>, <span class="at">name =</span> <span class="st">"Viral family"</span>) <span class="sc">+</span></span>
<span id="cb31-785"><a href="#cb31-785" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">"Family composition of human-viral reads"</span>) <span class="sc">+</span></span>
<span id="cb31-786"><a href="#cb31-786" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">4</span>)) <span class="sc">+</span></span>
<span id="cb31-787"><a href="#cb31-787" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">1.4</span>), <span class="at">hjust=</span><span class="dv">0</span>, <span class="at">face=</span><span class="st">"plain"</span>))</span>
<span id="cb31-788"><a href="#cb31-788" aria-hidden="true" tabindex="-1"></a>g_hv_family</span>
<span id="cb31-789"><a href="#cb31-789" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-790"><a href="#cb31-790" aria-hidden="true" tabindex="-1"></a><span class="co"># Get most prominent families for text</span></span>
<span id="cb31-791"><a href="#cb31-791" aria-hidden="true" tabindex="-1"></a>hv_family_collate <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span></span>
<span id="cb31-792"><a href="#cb31-792" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb31-793"><a href="#cb31-793" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_tot =</span> <span class="fu">sum</span>(n_reads_hv),</span>
<span id="cb31-794"><a href="#cb31-794" aria-hidden="true" tabindex="-1"></a>            <span class="at">p_reads_max =</span> <span class="fu">max</span>(p_reads_hv), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb31-795"><a href="#cb31-795" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n_reads_tot))</span>
<span id="cb31-796"><a href="#cb31-796" aria-hidden="true" tabindex="-1"></a><span class="co">#hv_family_collate</span></span>
<span id="cb31-797"><a href="#cb31-797" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb31-798"><a href="#cb31-798" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-801"><a href="#cb31-801" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb31-802"><a href="#cb31-802" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-filter-family</span></span>
<span id="cb31-803"><a href="#cb31-803" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 5</span></span>
<span id="cb31-804"><a href="#cb31-804" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 7</span></span>
<span id="cb31-805"><a href="#cb31-805" aria-hidden="true" tabindex="-1"></a>threshold_major_family <span class="ot">&lt;-</span> <span class="fl">0.05</span></span>
<span id="cb31-806"><a href="#cb31-806" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-807"><a href="#cb31-807" aria-hidden="true" tabindex="-1"></a><span class="co"># Count reads for each human-viral family</span></span>
<span id="cb31-808"><a href="#cb31-808" aria-hidden="true" tabindex="-1"></a>hv_family_counts <span class="ot">&lt;-</span> hv_reads_family <span class="sc">%&gt;%</span> </span>
<span id="cb31-809"><a href="#cb31-809" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span>(name <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Anelloviridae"</span>, <span class="st">"Rhabdoviridae"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb31-810"><a href="#cb31-810" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb31-811"><a href="#cb31-811" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb31-812"><a href="#cb31-812" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb31-813"><a href="#cb31-813" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_hv =</span> n_reads_hv<span class="sc">/</span><span class="fu">sum</span>(n_reads_hv))</span>
<span id="cb31-814"><a href="#cb31-814" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-815"><a href="#cb31-815" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify high-ranking families and group others</span></span>
<span id="cb31-816"><a href="#cb31-816" aria-hidden="true" tabindex="-1"></a>hv_family_major_tab <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span> </span>
<span id="cb31-817"><a href="#cb31-817" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_hv <span class="sc">==</span> <span class="fu">max</span>(p_reads_hv)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb31-818"><a href="#cb31-818" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_hv)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(p_reads_hv <span class="sc">&gt;</span> threshold_major_family)</span>
<span id="cb31-819"><a href="#cb31-819" aria-hidden="true" tabindex="-1"></a>hv_family_counts_major <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span></span>
<span id="cb31-820"><a href="#cb31-820" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">ifelse</span>(name <span class="sc">%in%</span> hv_family_major_tab<span class="sc">$</span>name, name, <span class="st">"Other"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb31-821"><a href="#cb31-821" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, name_display) <span class="sc">%&gt;%</span></span>
<span id="cb31-822"><a href="#cb31-822" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_hv =</span> <span class="fu">sum</span>(n_reads_hv), <span class="at">p_reads_hv =</span> <span class="fu">sum</span>(p_reads_hv), </span>
<span id="cb31-823"><a href="#cb31-823" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb31-824"><a href="#cb31-824" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">factor</span>(name_display, </span>
<span id="cb31-825"><a href="#cb31-825" aria-hidden="true" tabindex="-1"></a>                               <span class="at">levels =</span> <span class="fu">c</span>(hv_family_major_tab<span class="sc">$</span>name, <span class="st">"Other"</span>)))</span>
<span id="cb31-826"><a href="#cb31-826" aria-hidden="true" tabindex="-1"></a>hv_family_counts_display <span class="ot">&lt;-</span> hv_family_counts_major <span class="sc">%&gt;%</span></span>
<span id="cb31-827"><a href="#cb31-827" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_hv, <span class="at">classification =</span> name_display) <span class="sc">%&gt;%</span></span>
<span id="cb31-828"><a href="#cb31-828" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by =</span> <span class="st">'sample'</span>)</span>
<span id="cb31-829"><a href="#cb31-829" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-830"><a href="#cb31-830" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb31-831"><a href="#cb31-831" aria-hidden="true" tabindex="-1"></a>g_hv_family <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb31-832"><a href="#cb31-832" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">data=</span>hv_family_counts_display, <span class="at">position =</span> <span class="st">"stack"</span>, <span class="at">width=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb31-833"><a href="#cb31-833" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% HV Reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb31-834"><a href="#cb31-834" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb31-835"><a href="#cb31-835" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb31-836"><a href="#cb31-836" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">'Accent'</span>, <span class="at">name =</span> <span class="st">"Viral family"</span>) <span class="sc">+</span></span>
<span id="cb31-837"><a href="#cb31-837" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">"Family composition of human-viral reads"</span>) <span class="sc">+</span></span>
<span id="cb31-838"><a href="#cb31-838" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">4</span>)) <span class="sc">+</span></span>
<span id="cb31-839"><a href="#cb31-839" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">1.4</span>), <span class="at">hjust=</span><span class="dv">0</span>, <span class="at">face=</span><span class="st">"plain"</span>))</span>
<span id="cb31-840"><a href="#cb31-840" aria-hidden="true" tabindex="-1"></a>g_hv_family</span>
<span id="cb31-841"><a href="#cb31-841" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-842"><a href="#cb31-842" aria-hidden="true" tabindex="-1"></a><span class="co"># Get most prominent families for text</span></span>
<span id="cb31-843"><a href="#cb31-843" aria-hidden="true" tabindex="-1"></a>hv_family_collate <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span></span>
<span id="cb31-844"><a href="#cb31-844" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb31-845"><a href="#cb31-845" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_tot =</span> <span class="fu">sum</span>(n_reads_hv),</span>
<span id="cb31-846"><a href="#cb31-846" aria-hidden="true" tabindex="-1"></a>            <span class="at">p_reads_max =</span> <span class="fu">max</span>(p_reads_hv), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb31-847"><a href="#cb31-847" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n_reads_tot))</span>
<span id="cb31-848"><a href="#cb31-848" aria-hidden="true" tabindex="-1"></a>hv_family_collate</span>
<span id="cb31-849"><a href="#cb31-849" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb31-850"><a href="#cb31-850" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-851"><a href="#cb31-851" aria-hidden="true" tabindex="-1"></a><span class="fu">## Species analysis</span></span>
<span id="cb31-854"><a href="#cb31-854" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb31-855"><a href="#cb31-855" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-composition-exclusion</span></span>
<span id="cb31-856"><a href="#cb31-856" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb31-857"><a href="#cb31-857" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb31-858"><a href="#cb31-858" aria-hidden="true" tabindex="-1"></a>total <span class="ot">&lt;-</span> basic_stats_raw <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample, n_read_pairs)</span>
<span id="cb31-859"><a href="#cb31-859" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-860"><a href="#cb31-860" aria-hidden="true" tabindex="-1"></a>hv_family_counts <span class="ot">&lt;-</span> hv_reads_family <span class="sc">%&gt;%</span> </span>
<span id="cb31-861"><a href="#cb31-861" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb31-862"><a href="#cb31-862" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb31-863"><a href="#cb31-863" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(total, <span class="at">by=</span><span class="st">'sample'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb31-864"><a href="#cb31-864" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb31-865"><a href="#cb31-865" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">ra_reads_hv =</span> n_reads_hv<span class="sc">/</span>n_read_pairs)</span>
<span id="cb31-866"><a href="#cb31-866" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-867"><a href="#cb31-867" aria-hidden="true" tabindex="-1"></a>hv_family_collate <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span></span>
<span id="cb31-868"><a href="#cb31-868" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb31-869"><a href="#cb31-869" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_tot =</span> <span class="fu">sum</span>(n_reads_hv),</span>
<span id="cb31-870"><a href="#cb31-870" aria-hidden="true" tabindex="-1"></a>            <span class="at">ra_reads_tot =</span> <span class="fu">mean</span>(ra_reads_hv), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb31-871"><a href="#cb31-871" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n_reads_tot))</span>
<span id="cb31-872"><a href="#cb31-872" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-873"><a href="#cb31-873" aria-hidden="true" tabindex="-1"></a>hv_family_collate <span class="ot">&lt;-</span> hv_family_collate <span class="sc">%&gt;%</span> </span>
<span id="cb31-874"><a href="#cb31-874" aria-hidden="true" tabindex="-1"></a><span class="fu">rename</span>(</span>
<span id="cb31-875"><a href="#cb31-875" aria-hidden="true" tabindex="-1"></a><span class="st">'family'</span> <span class="ot">=</span> <span class="st">'name'</span>,</span>
<span id="cb31-876"><a href="#cb31-876" aria-hidden="true" tabindex="-1"></a><span class="st">'family_n_reads_tot'</span> <span class="ot">=</span> <span class="st">'n_reads_tot'</span>,</span>
<span id="cb31-877"><a href="#cb31-877" aria-hidden="true" tabindex="-1"></a><span class="st">'family_ra_reads_tot'</span> <span class="ot">=</span> <span class="st">'ra_reads_tot'</span>,</span>
<span id="cb31-878"><a href="#cb31-878" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb31-879"><a href="#cb31-879" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-880"><a href="#cb31-880" aria-hidden="true" tabindex="-1"></a>hv_genus_counts <span class="ot">&lt;-</span> hv_reads_genus <span class="sc">%&gt;%</span> </span>
<span id="cb31-881"><a href="#cb31-881" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb31-882"><a href="#cb31-882" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb31-883"><a href="#cb31-883" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(total, <span class="at">by=</span><span class="st">'sample'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb31-884"><a href="#cb31-884" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb31-885"><a href="#cb31-885" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">ra_reads_hv =</span> n_reads_hv<span class="sc">/</span>n_read_pairs)</span>
<span id="cb31-886"><a href="#cb31-886" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-887"><a href="#cb31-887" aria-hidden="true" tabindex="-1"></a>hv_genus_collate <span class="ot">&lt;-</span> hv_genus_counts <span class="sc">%&gt;%</span></span>
<span id="cb31-888"><a href="#cb31-888" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb31-889"><a href="#cb31-889" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_tot =</span> <span class="fu">sum</span>(n_reads_hv),</span>
<span id="cb31-890"><a href="#cb31-890" aria-hidden="true" tabindex="-1"></a>            <span class="at">ra_reads_tot =</span> <span class="fu">mean</span>(ra_reads_hv), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb31-891"><a href="#cb31-891" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n_reads_tot))</span>
<span id="cb31-892"><a href="#cb31-892" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-893"><a href="#cb31-893" aria-hidden="true" tabindex="-1"></a>hv_genus_collate <span class="ot">&lt;-</span> hv_genus_collate <span class="sc">%&gt;%</span> </span>
<span id="cb31-894"><a href="#cb31-894" aria-hidden="true" tabindex="-1"></a><span class="fu">rename</span>(</span>
<span id="cb31-895"><a href="#cb31-895" aria-hidden="true" tabindex="-1"></a><span class="st">'genus'</span> <span class="ot">=</span> <span class="st">'name'</span>,</span>
<span id="cb31-896"><a href="#cb31-896" aria-hidden="true" tabindex="-1"></a><span class="st">'genus_n_reads_tot'</span> <span class="ot">=</span> <span class="st">'n_reads_tot'</span>,</span>
<span id="cb31-897"><a href="#cb31-897" aria-hidden="true" tabindex="-1"></a><span class="st">'genus_ra_reads_tot'</span> <span class="ot">=</span> <span class="st">'ra_reads_tot'</span>,</span>
<span id="cb31-898"><a href="#cb31-898" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb31-899"><a href="#cb31-899" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-900"><a href="#cb31-900" aria-hidden="true" tabindex="-1"></a>hv_species_counts <span class="ot">&lt;-</span> hv_reads_species <span class="sc">%&gt;%</span> </span>
<span id="cb31-901"><a href="#cb31-901" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb31-902"><a href="#cb31-902" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb31-903"><a href="#cb31-903" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(total, <span class="at">by=</span><span class="st">'sample'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb31-904"><a href="#cb31-904" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb31-905"><a href="#cb31-905" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">ra_reads_hv =</span> n_reads_hv<span class="sc">/</span>n_read_pairs)</span>
<span id="cb31-906"><a href="#cb31-906" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-907"><a href="#cb31-907" aria-hidden="true" tabindex="-1"></a>hv_species_collate <span class="ot">&lt;-</span> hv_species_counts <span class="sc">%&gt;%</span></span>
<span id="cb31-908"><a href="#cb31-908" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb31-909"><a href="#cb31-909" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_tot =</span> <span class="fu">sum</span>(n_reads_hv),</span>
<span id="cb31-910"><a href="#cb31-910" aria-hidden="true" tabindex="-1"></a>            <span class="at">ra_reads_tot =</span> <span class="fu">mean</span>(ra_reads_hv), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb31-911"><a href="#cb31-911" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n_reads_tot))</span>
<span id="cb31-912"><a href="#cb31-912" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-913"><a href="#cb31-913" aria-hidden="true" tabindex="-1"></a>hv_species_collate <span class="ot">&lt;-</span> hv_species_collate <span class="sc">%&gt;%</span> </span>
<span id="cb31-914"><a href="#cb31-914" aria-hidden="true" tabindex="-1"></a><span class="fu">rename</span>(</span>
<span id="cb31-915"><a href="#cb31-915" aria-hidden="true" tabindex="-1"></a><span class="st">'species'</span> <span class="ot">=</span> <span class="st">'name'</span>,</span>
<span id="cb31-916"><a href="#cb31-916" aria-hidden="true" tabindex="-1"></a><span class="st">'species_n_reads_tot'</span> <span class="ot">=</span> <span class="st">'n_reads_tot'</span>,</span>
<span id="cb31-917"><a href="#cb31-917" aria-hidden="true" tabindex="-1"></a><span class="st">'species_ra_reads_tot'</span> <span class="ot">=</span> <span class="st">'ra_reads_tot'</span>,</span>
<span id="cb31-918"><a href="#cb31-918" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb31-919"><a href="#cb31-919" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-920"><a href="#cb31-920" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb31-921"><a href="#cb31-921" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-924"><a href="#cb31-924" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb31-925"><a href="#cb31-925" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-filter-exclusion</span></span>
<span id="cb31-926"><a href="#cb31-926" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb31-927"><a href="#cb31-927" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-928"><a href="#cb31-928" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to find parent taxa recursively</span></span>
<span id="cb31-929"><a href="#cb31-929" aria-hidden="true" tabindex="-1"></a>find_parent_taxa <span class="ot">&lt;-</span> <span class="cf">function</span>(taxid, target_ranks, taxa_data) {</span>
<span id="cb31-930"><a href="#cb31-930" aria-hidden="true" tabindex="-1"></a> current_taxid <span class="ot">&lt;-</span> taxid</span>
<span id="cb31-931"><a href="#cb31-931" aria-hidden="true" tabindex="-1"></a>  result <span class="ot">&lt;-</span> <span class="fu">setNames</span>(<span class="fu">vector</span>(<span class="st">"list"</span>, <span class="fu">length</span>(target_ranks)), target_ranks)</span>
<span id="cb31-932"><a href="#cb31-932" aria-hidden="true" tabindex="-1"></a>  <span class="cf">while</span> (<span class="cn">TRUE</span>) {</span>
<span id="cb31-933"><a href="#cb31-933" aria-hidden="true" tabindex="-1"></a>    row <span class="ot">&lt;-</span> taxa_data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(taxid <span class="sc">==</span> current_taxid) <span class="sc">%&gt;%</span> <span class="fu">slice</span>(<span class="dv">1</span>)</span>
<span id="cb31-934"><a href="#cb31-934" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">nrow</span>(row) <span class="sc">==</span> <span class="dv">0</span>) <span class="cf">break</span>  <span class="co"># If no matching taxid is found, break the loop</span></span>
<span id="cb31-935"><a href="#cb31-935" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (row<span class="sc">$</span>rank <span class="sc">%in%</span> target_ranks) {</span>
<span id="cb31-936"><a href="#cb31-936" aria-hidden="true" tabindex="-1"></a>      result[[row<span class="sc">$</span>rank]] <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">taxid =</span> row<span class="sc">$</span>taxid, <span class="at">name =</span> row<span class="sc">$</span>name)</span>
<span id="cb31-937"><a href="#cb31-937" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span> (<span class="fu">all</span>(<span class="sc">!</span><span class="fu">sapply</span>(result, is.null))) {</span>
<span id="cb31-938"><a href="#cb31-938" aria-hidden="true" tabindex="-1"></a>        <span class="cf">break</span>  <span class="co"># If all target ranks are found, break the loop</span></span>
<span id="cb31-939"><a href="#cb31-939" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb31-940"><a href="#cb31-940" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb31-941"><a href="#cb31-941" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (row<span class="sc">$</span>taxid <span class="sc">==</span> row<span class="sc">$</span>parent_taxid) {</span>
<span id="cb31-942"><a href="#cb31-942" aria-hidden="true" tabindex="-1"></a>      <span class="cf">break</span>  <span class="co"># Avoid infinite loop at root</span></span>
<span id="cb31-943"><a href="#cb31-943" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb31-944"><a href="#cb31-944" aria-hidden="true" tabindex="-1"></a>    current_taxid <span class="ot">&lt;-</span> row<span class="sc">$</span>parent_taxid</span>
<span id="cb31-945"><a href="#cb31-945" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb31-946"><a href="#cb31-946" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Replace NULL with NA for any unfound ranks</span></span>
<span id="cb31-947"><a href="#cb31-947" aria-hidden="true" tabindex="-1"></a>  result <span class="ot">&lt;-</span> <span class="fu">lapply</span>(result, <span class="cf">function</span>(x) <span class="cf">if</span> (<span class="fu">is.null</span>(x)) <span class="fu">list</span>(<span class="at">taxid =</span> <span class="cn">NA</span>, <span class="at">name =</span> <span class="cn">NA</span>) <span class="cf">else</span> x)</span>
<span id="cb31-948"><a href="#cb31-948" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(result)</span>
<span id="cb31-949"><a href="#cb31-949" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb31-950"><a href="#cb31-950" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-951"><a href="#cb31-951" aria-hidden="true" tabindex="-1"></a><span class="co"># Find all relevant parent taxa for each species</span></span>
<span id="cb31-952"><a href="#cb31-952" aria-hidden="true" tabindex="-1"></a>result <span class="ot">&lt;-</span> hv_species_collate <span class="sc">%&gt;%</span></span>
<span id="cb31-953"><a href="#cb31-953" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb31-954"><a href="#cb31-954" aria-hidden="true" tabindex="-1"></a>    <span class="at">parent_info =</span> <span class="fu">map</span>(taxid, <span class="sc">~</span><span class="fu">find_parent_taxa</span>(.x, <span class="fu">c</span>(<span class="st">"species"</span>,<span class="st">"subgenus"</span>, <span class="st">"genus"</span>, <span class="st">"subfamily"</span>, <span class="st">"family"</span>), viral_taxa)),</span>
<span id="cb31-955"><a href="#cb31-955" aria-hidden="true" tabindex="-1"></a>    <span class="at">genus_taxid =</span> <span class="fu">map_dbl</span>(parent_info, <span class="sc">~</span>.x<span class="sc">$</span>genus<span class="sc">$</span>taxid),</span>
<span id="cb31-956"><a href="#cb31-956" aria-hidden="true" tabindex="-1"></a>    <span class="at">genus =</span> <span class="fu">map_chr</span>(parent_info, <span class="sc">~</span>.x<span class="sc">$</span>genus<span class="sc">$</span>name),</span>
<span id="cb31-957"><a href="#cb31-957" aria-hidden="true" tabindex="-1"></a>    <span class="at">family_taxid =</span> <span class="fu">map_dbl</span>(parent_info, <span class="sc">~</span>.x<span class="sc">$</span>family<span class="sc">$</span>taxid),</span>
<span id="cb31-958"><a href="#cb31-958" aria-hidden="true" tabindex="-1"></a>    <span class="at">family =</span> <span class="fu">map_chr</span>(parent_info, <span class="sc">~</span>.x<span class="sc">$</span>family<span class="sc">$</span>name)</span>
<span id="cb31-959"><a href="#cb31-959" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb31-960"><a href="#cb31-960" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>parent_info)</span>
<span id="cb31-961"><a href="#cb31-961" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-962"><a href="#cb31-962" aria-hidden="true" tabindex="-1"></a><span class="co"># Join with genus counts</span></span>
<span id="cb31-963"><a href="#cb31-963" aria-hidden="true" tabindex="-1"></a>result <span class="ot">&lt;-</span> result <span class="sc">%&gt;%</span></span>
<span id="cb31-964"><a href="#cb31-964" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(</span>
<span id="cb31-965"><a href="#cb31-965" aria-hidden="true" tabindex="-1"></a>    hv_genus_collate <span class="sc">%&gt;%</span> <span class="fu">select</span>(taxid, genus_n_reads_tot, genus_ra_reads_tot),</span>
<span id="cb31-966"><a href="#cb31-966" aria-hidden="true" tabindex="-1"></a>    <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"genus_taxid"</span> <span class="ot">=</span> <span class="st">"taxid"</span>)</span>
<span id="cb31-967"><a href="#cb31-967" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb31-968"><a href="#cb31-968" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-969"><a href="#cb31-969" aria-hidden="true" tabindex="-1"></a><span class="co"># Join with family counts</span></span>
<span id="cb31-970"><a href="#cb31-970" aria-hidden="true" tabindex="-1"></a>result <span class="ot">&lt;-</span> result <span class="sc">%&gt;%</span></span>
<span id="cb31-971"><a href="#cb31-971" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(</span>
<span id="cb31-972"><a href="#cb31-972" aria-hidden="true" tabindex="-1"></a>    hv_family_collate <span class="sc">%&gt;%</span> <span class="fu">select</span>(taxid, family_n_reads_tot, family_ra_reads_tot),</span>
<span id="cb31-973"><a href="#cb31-973" aria-hidden="true" tabindex="-1"></a>    <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"family_taxid"</span> <span class="ot">=</span> <span class="st">"taxid"</span>)</span>
<span id="cb31-974"><a href="#cb31-974" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb31-975"><a href="#cb31-975" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-976"><a href="#cb31-976" aria-hidden="true" tabindex="-1"></a><span class="co"># Clean up</span></span>
<span id="cb31-977"><a href="#cb31-977" aria-hidden="true" tabindex="-1"></a>result <span class="ot">&lt;-</span> result <span class="sc">%&gt;%</span></span>
<span id="cb31-978"><a href="#cb31-978" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="fu">ends_with</span>(<span class="st">"_taxid"</span>))</span>
<span id="cb31-979"><a href="#cb31-979" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-980"><a href="#cb31-980" aria-hidden="true" tabindex="-1"></a><span class="co">#result</span></span>
<span id="cb31-981"><a href="#cb31-981" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb31-982"><a href="#cb31-982" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-983"><a href="#cb31-983" aria-hidden="true" tabindex="-1"></a>To get a good overview of families, genera, and species, we can look at a Sankey plot where the magnitude of relative abundance, averaged over all samples, is shown in parentheses.</span>
<span id="cb31-986"><a href="#cb31-986" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb31-987"><a href="#cb31-987" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-sankey</span></span>
<span id="cb31-988"><a href="#cb31-988" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 10</span></span>
<span id="cb31-989"><a href="#cb31-989" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb31-990"><a href="#cb31-990" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-991"><a href="#cb31-991" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to create links</span></span>
<span id="cb31-992"><a href="#cb31-992" aria-hidden="true" tabindex="-1"></a>create_links <span class="ot">&lt;-</span> <span class="cf">function</span>(data) {</span>
<span id="cb31-993"><a href="#cb31-993" aria-hidden="true" tabindex="-1"></a>  family_to_genus <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb31-994"><a href="#cb31-994" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(genus)) <span class="sc">%&gt;%</span></span>
<span id="cb31-995"><a href="#cb31-995" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(family, genus) <span class="sc">%&gt;%</span></span>
<span id="cb31-996"><a href="#cb31-996" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">value =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb31-997"><a href="#cb31-997" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">source =</span> family, <span class="at">target =</span> genus)</span>
<span id="cb31-998"><a href="#cb31-998" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb31-999"><a href="#cb31-999" aria-hidden="true" tabindex="-1"></a>  genus_to_species <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb31-1000"><a href="#cb31-1000" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(genus, species) <span class="sc">%&gt;%</span></span>
<span id="cb31-1001"><a href="#cb31-1001" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">value =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb31-1002"><a href="#cb31-1002" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">source =</span> genus, <span class="at">target =</span> species)</span>
<span id="cb31-1003"><a href="#cb31-1003" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-1004"><a href="#cb31-1004" aria-hidden="true" tabindex="-1"></a>  family_to_species <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb31-1005"><a href="#cb31-1005" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="fu">is.na</span>(genus)) <span class="sc">%&gt;%</span></span>
<span id="cb31-1006"><a href="#cb31-1006" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(family, species) <span class="sc">%&gt;%</span></span>
<span id="cb31-1007"><a href="#cb31-1007" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">value =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb31-1008"><a href="#cb31-1008" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">source =</span> family, <span class="at">target =</span> species)</span>
<span id="cb31-1009"><a href="#cb31-1009" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-1010"><a href="#cb31-1010" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_rows</span>(family_to_genus, genus_to_species, family_to_species) <span class="sc">%&gt;%</span></span>
<span id="cb31-1011"><a href="#cb31-1011" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(source))</span>
<span id="cb31-1012"><a href="#cb31-1012" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb31-1013"><a href="#cb31-1013" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-1014"><a href="#cb31-1014" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to create nodes</span></span>
<span id="cb31-1015"><a href="#cb31-1015" aria-hidden="true" tabindex="-1"></a>create_nodes <span class="ot">&lt;-</span> <span class="cf">function</span>(links) {</span>
<span id="cb31-1016"><a href="#cb31-1016" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>(</span>
<span id="cb31-1017"><a href="#cb31-1017" aria-hidden="true" tabindex="-1"></a>    <span class="at">name =</span> <span class="fu">c</span>(links<span class="sc">$</span>source, links<span class="sc">$</span>target) <span class="sc">%&gt;%</span> <span class="fu">unique</span>()</span>
<span id="cb31-1018"><a href="#cb31-1018" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb31-1019"><a href="#cb31-1019" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb31-1020"><a href="#cb31-1020" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-1021"><a href="#cb31-1021" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to prepare data for Sankey diagram</span></span>
<span id="cb31-1022"><a href="#cb31-1022" aria-hidden="true" tabindex="-1"></a>prepare_sankey_data <span class="ot">&lt;-</span> <span class="cf">function</span>(links, nodes) {</span>
<span id="cb31-1023"><a href="#cb31-1023" aria-hidden="true" tabindex="-1"></a>  links<span class="sc">$</span>IDsource <span class="ot">&lt;-</span> <span class="fu">match</span>(links<span class="sc">$</span>source, nodes<span class="sc">$</span>name) <span class="sc">-</span> <span class="dv">1</span></span>
<span id="cb31-1024"><a href="#cb31-1024" aria-hidden="true" tabindex="-1"></a>  links<span class="sc">$</span>IDtarget <span class="ot">&lt;-</span> <span class="fu">match</span>(links<span class="sc">$</span>target, nodes<span class="sc">$</span>name) <span class="sc">-</span> <span class="dv">1</span></span>
<span id="cb31-1025"><a href="#cb31-1025" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list</span>(<span class="at">links =</span> links, <span class="at">nodes =</span> nodes)</span>
<span id="cb31-1026"><a href="#cb31-1026" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb31-1027"><a href="#cb31-1027" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-1028"><a href="#cb31-1028" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to create Sankey plot</span></span>
<span id="cb31-1029"><a href="#cb31-1029" aria-hidden="true" tabindex="-1"></a>create_sankey_plot <span class="ot">&lt;-</span> <span class="cf">function</span>(sankey_data) {</span>
<span id="cb31-1030"><a href="#cb31-1030" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sankeyNetwork</span>(</span>
<span id="cb31-1031"><a href="#cb31-1031" aria-hidden="true" tabindex="-1"></a>    <span class="at">Links =</span> sankey_data<span class="sc">$</span>links, </span>
<span id="cb31-1032"><a href="#cb31-1032" aria-hidden="true" tabindex="-1"></a>    <span class="at">Nodes =</span> sankey_data<span class="sc">$</span>nodes,</span>
<span id="cb31-1033"><a href="#cb31-1033" aria-hidden="true" tabindex="-1"></a>    <span class="at">Source =</span> <span class="st">"IDsource"</span>, </span>
<span id="cb31-1034"><a href="#cb31-1034" aria-hidden="true" tabindex="-1"></a>    <span class="at">Target =</span> <span class="st">"IDtarget"</span>,</span>
<span id="cb31-1035"><a href="#cb31-1035" aria-hidden="true" tabindex="-1"></a>    <span class="at">Value =</span> <span class="st">"value"</span>, </span>
<span id="cb31-1036"><a href="#cb31-1036" aria-hidden="true" tabindex="-1"></a>    <span class="at">NodeID =</span> <span class="st">"name"</span>,</span>
<span id="cb31-1037"><a href="#cb31-1037" aria-hidden="true" tabindex="-1"></a>    <span class="at">sinksRight =</span> <span class="cn">TRUE</span>,</span>
<span id="cb31-1038"><a href="#cb31-1038" aria-hidden="true" tabindex="-1"></a>    <span class="at">nodeWidth =</span> <span class="dv">50</span>,</span>
<span id="cb31-1039"><a href="#cb31-1039" aria-hidden="true" tabindex="-1"></a>    <span class="at">fontSize =</span> <span class="dv">14</span>,</span>
<span id="cb31-1040"><a href="#cb31-1040" aria-hidden="true" tabindex="-1"></a>    <span class="at">height =</span> <span class="dv">800</span>,</span>
<span id="cb31-1041"><a href="#cb31-1041" aria-hidden="true" tabindex="-1"></a>    <span class="at">width =</span> <span class="dv">1000</span></span>
<span id="cb31-1042"><a href="#cb31-1042" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb31-1043"><a href="#cb31-1043" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb31-1044"><a href="#cb31-1044" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-1045"><a href="#cb31-1045" aria-hidden="true" tabindex="-1"></a>save_sankey_as_png <span class="ot">&lt;-</span> <span class="cf">function</span>(sankey_plot, <span class="at">width =</span> <span class="dv">1000</span>, <span class="at">height =</span> <span class="dv">800</span>) {</span>
<span id="cb31-1046"><a href="#cb31-1046" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Save the plot as an HTML file</span></span>
<span id="cb31-1047"><a href="#cb31-1047" aria-hidden="true" tabindex="-1"></a>  <span class="fu">saveWidget</span>(sankey_plot, <span class="fu">sprintf</span>(<span class="st">'%s/sankey.html'</span>,image_dir))</span>
<span id="cb31-1048"><a href="#cb31-1048" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb31-1049"><a href="#cb31-1049" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-1050"><a href="#cb31-1050" aria-hidden="true" tabindex="-1"></a>format_scientific <span class="ot">&lt;-</span> <span class="cf">function</span>(x, <span class="at">digits=</span><span class="dv">2</span>) {</span>
<span id="cb31-1051"><a href="#cb31-1051" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sapply</span>(x, <span class="cf">function</span>(val) {</span>
<span id="cb31-1052"><a href="#cb31-1052" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">is.na</span>(val) <span class="sc">||</span> <span class="fu">abs</span>(val) <span class="sc">&lt;</span> <span class="fl">1e-15</span>) {</span>
<span id="cb31-1053"><a href="#cb31-1053" aria-hidden="true" tabindex="-1"></a>      <span class="fu">return</span>(<span class="st">"0"</span>)</span>
<span id="cb31-1054"><a href="#cb31-1054" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb31-1055"><a href="#cb31-1055" aria-hidden="true" tabindex="-1"></a>      exponent <span class="ot">&lt;-</span> <span class="fu">floor</span>(<span class="fu">log10</span>(<span class="fu">abs</span>(val)))</span>
<span id="cb31-1056"><a href="#cb31-1056" aria-hidden="true" tabindex="-1"></a>      coef <span class="ot">&lt;-</span> val <span class="sc">/</span> <span class="dv">10</span><span class="sc">^</span>exponent</span>
<span id="cb31-1057"><a href="#cb31-1057" aria-hidden="true" tabindex="-1"></a>      <span class="fu">return</span>(<span class="fu">sprintf</span>(<span class="st">"%.1f × 10^%d"</span>, <span class="fu">round</span>(coef, digits), exponent))</span>
<span id="cb31-1058"><a href="#cb31-1058" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb31-1059"><a href="#cb31-1059" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb31-1060"><a href="#cb31-1060" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb31-1061"><a href="#cb31-1061" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-1062"><a href="#cb31-1062" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> result <span class="sc">%&gt;%</span> </span>
<span id="cb31-1063"><a href="#cb31-1063" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">c</span>(genus_n_reads_tot, genus_ra_reads_tot), <span class="sc">~</span><span class="fu">replace_na</span>(., <span class="dv">0</span>)),</span>
<span id="cb31-1064"><a href="#cb31-1064" aria-hidden="true" tabindex="-1"></a>         <span class="at">genus =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(genus), <span class="st">"Unknown Genus"</span>, genus)) <span class="sc">%&gt;%</span></span>
<span id="cb31-1065"><a href="#cb31-1065" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb31-1066"><a href="#cb31-1066" aria-hidden="true" tabindex="-1"></a>  <span class="at">species =</span> <span class="fu">paste0</span>(species, <span class="fu">sprintf</span>(<span class="st">' (%s)'</span>, <span class="fu">format_scientific</span>(species_ra_reads_tot))),</span>
<span id="cb31-1067"><a href="#cb31-1067" aria-hidden="true" tabindex="-1"></a>  <span class="at">genus =</span> <span class="fu">paste0</span>(genus, <span class="fu">sprintf</span>(<span class="st">' (%s)'</span>, <span class="fu">format_scientific</span>(genus_ra_reads_tot))),</span>
<span id="cb31-1068"><a href="#cb31-1068" aria-hidden="true" tabindex="-1"></a>  <span class="at">family =</span> <span class="fu">paste0</span>(family, <span class="fu">sprintf</span>(<span class="st">' (%s)'</span>, <span class="fu">format_scientific</span>(family_ra_reads_tot)))</span>
<span id="cb31-1069"><a href="#cb31-1069" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb31-1070"><a href="#cb31-1070" aria-hidden="true" tabindex="-1"></a>links <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">create_links</span>(data))</span>
<span id="cb31-1071"><a href="#cb31-1071" aria-hidden="true" tabindex="-1"></a>nodes <span class="ot">&lt;-</span> <span class="fu">create_nodes</span>(links)</span>
<span id="cb31-1072"><a href="#cb31-1072" aria-hidden="true" tabindex="-1"></a>sankey_data <span class="ot">&lt;-</span> <span class="fu">prepare_sankey_data</span>(links, nodes)</span>
<span id="cb31-1073"><a href="#cb31-1073" aria-hidden="true" tabindex="-1"></a>sankey <span class="ot">&lt;-</span> <span class="fu">create_sankey_plot</span>(sankey_data)</span>
<span id="cb31-1074"><a href="#cb31-1074" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-1075"><a href="#cb31-1075" aria-hidden="true" tabindex="-1"></a>sankey</span>
<span id="cb31-1076"><a href="#cb31-1076" aria-hidden="true" tabindex="-1"></a><span class="co">#save_sankey_as_png(sankey)</span></span>
<span id="cb31-1077"><a href="#cb31-1077" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb31-1078"><a href="#cb31-1078" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-1079"><a href="#cb31-1079" aria-hidden="true" tabindex="-1"></a>To get a better idea of the relative abundance of species, we can create a dot plot where each dot represents the relative abundance of a particular species in a sample.</span>
<span id="cb31-1082"><a href="#cb31-1082" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb31-1083"><a href="#cb31-1083" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-species-dot</span></span>
<span id="cb31-1084"><a href="#cb31-1084" aria-hidden="true" tabindex="-1"></a><span class="co">#| width: 20</span></span>
<span id="cb31-1085"><a href="#cb31-1085" aria-hidden="true" tabindex="-1"></a><span class="co">#| height: 10</span></span>
<span id="cb31-1086"><a href="#cb31-1086" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-1087"><a href="#cb31-1087" aria-hidden="true" tabindex="-1"></a>species_family <span class="ot">&lt;-</span> result <span class="sc">%&gt;%</span> <span class="fu">select</span>(species, family) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="st">'name'</span> <span class="ot">=</span> <span class="st">'species'</span>)</span>
<span id="cb31-1088"><a href="#cb31-1088" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-1089"><a href="#cb31-1089" aria-hidden="true" tabindex="-1"></a>play <span class="ot">&lt;-</span> hv_species_counts <span class="sc">%&gt;%</span> </span>
<span id="cb31-1090"><a href="#cb31-1090" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb31-1091"><a href="#cb31-1091" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by =</span> <span class="st">'sample'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb31-1092"><a href="#cb31-1092" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(species_family, <span class="at">by =</span> <span class="st">'name'</span>)</span>
<span id="cb31-1093"><a href="#cb31-1093" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-1094"><a href="#cb31-1094" aria-hidden="true" tabindex="-1"></a>play<span class="sc">$</span>family <span class="ot">&lt;-</span> <span class="fu">factor</span>(play<span class="sc">$</span>family, <span class="at">levels=</span>hv_family_collate <span class="sc">%&gt;%</span> <span class="fu">pull</span>(family))</span>
<span id="cb31-1095"><a href="#cb31-1095" aria-hidden="true" tabindex="-1"></a>play <span class="ot">&lt;-</span> play <span class="sc">%&gt;%</span></span>
<span id="cb31-1096"><a href="#cb31-1096" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(family, name) <span class="sc">%&gt;%</span></span>
<span id="cb31-1097"><a href="#cb31-1097" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">factor</span>(name, <span class="at">levels =</span> <span class="fu">unique</span>(name)))</span>
<span id="cb31-1098"><a href="#cb31-1098" aria-hidden="true" tabindex="-1"></a><span class="co">#name_order &lt;- play %&gt;% arrange(family) %&gt;% pull(name)</span></span>
<span id="cb31-1099"><a href="#cb31-1099" aria-hidden="true" tabindex="-1"></a><span class="co">#play$name &lt;- factor(play$name, levels = name_order)</span></span>
<span id="cb31-1100"><a href="#cb31-1100" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-1101"><a href="#cb31-1101" aria-hidden="true" tabindex="-1"></a>ra_dot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(play, <span class="fu">aes</span>(<span class="at">x =</span> ra_reads_hv, <span class="at">y=</span>name)) <span class="sc">+</span></span>
<span id="cb31-1102"><a href="#cb31-1102" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_quasirandom</span>(<span class="at">orientation =</span> <span class="st">'y'</span>, <span class="fu">aes</span>(<span class="at">color =</span> family)) <span class="sc">+</span></span>
<span id="cb31-1103"><a href="#cb31-1103" aria-hidden="true" tabindex="-1"></a>   <span class="fu">scale_color_brewer</span>(<span class="at">palette =</span> <span class="st">"Accent"</span>) <span class="sc">+</span></span>
<span id="cb31-1104"><a href="#cb31-1104" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_log10</span>(</span>
<span id="cb31-1105"><a href="#cb31-1105" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Relative abundance human virus reads"</span>,</span>
<span id="cb31-1106"><a href="#cb31-1106" aria-hidden="true" tabindex="-1"></a>    <span class="at">labels =</span> <span class="cf">function</span>(x) <span class="fu">parse</span>(<span class="at">text =</span> <span class="fu">paste0</span>(<span class="st">"10^"</span>, <span class="fu">round</span>(<span class="fu">log10</span>(x)))),</span>
<span id="cb31-1107"><a href="#cb31-1107" aria-hidden="true" tabindex="-1"></a>    <span class="at">limits =</span> <span class="fu">c</span>(<span class="fl">1e-7</span>, <span class="dv">1</span>),</span>
<span id="cb31-1108"><a href="#cb31-1108" aria-hidden="true" tabindex="-1"></a>    <span class="at">n.breaks =</span> <span class="dv">8</span></span>
<span id="cb31-1109"><a href="#cb31-1109" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb31-1110"><a href="#cb31-1110" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">'Human virus'</span>,</span>
<span id="cb31-1111"><a href="#cb31-1111" aria-hidden="true" tabindex="-1"></a>       <span class="at">color =</span> <span class="st">'Viral family'</span>) <span class="sc">+</span> </span>
<span id="cb31-1112"><a href="#cb31-1112" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_light</span>() <span class="sc">+</span> </span>
<span id="cb31-1113"><a href="#cb31-1113" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(</span>
<span id="cb31-1114"><a href="#cb31-1114" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>),</span>
<span id="cb31-1115"><a href="#cb31-1115" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">14</span>),</span>
<span id="cb31-1116"><a href="#cb31-1116" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.ticks.y =</span> <span class="fu">element_blank</span>(),<span class="at">axis.line =</span> <span class="fu">element_line</span>(<span class="at">colour =</span> <span class="st">"black"</span>),</span>
<span id="cb31-1117"><a href="#cb31-1117" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">15</span>),    </span>
<span id="cb31-1118"><a href="#cb31-1118" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">15</span>),  </span>
<span id="cb31-1119"><a href="#cb31-1119" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">13</span>),</span>
<span id="cb31-1120"><a href="#cb31-1120" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">16</span>),</span>
<span id="cb31-1121"><a href="#cb31-1121" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb31-1122"><a href="#cb31-1122" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.border =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb31-1123"><a href="#cb31-1123" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.background =</span> <span class="fu">element_blank</span>())</span>
<span id="cb31-1124"><a href="#cb31-1124" aria-hidden="true" tabindex="-1"></a>ra_dot</span>
<span id="cb31-1125"><a href="#cb31-1125" aria-hidden="true" tabindex="-1"></a><span class="co">#ggsave(sprintf('%s/ra_viral_dot.jpg',image_dir), ra_dot, height = 8 ,width = 12)</span></span>
<span id="cb31-1126"><a href="#cb31-1126" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-1127"><a href="#cb31-1127" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb31-1128"><a href="#cb31-1128" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-1131"><a href="#cb31-1131" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb31-1132"><a href="#cb31-1132" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv</span></span>
<span id="cb31-1133"><a href="#cb31-1133" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb31-1134"><a href="#cb31-1134" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-1135"><a href="#cb31-1135" aria-hidden="true" tabindex="-1"></a>all <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span> </span>
<span id="cb31-1136"><a href="#cb31-1136" aria-hidden="true" tabindex="-1"></a><span class="co">#  inner_join(basic_stats_raw, by = c('sample')) %&gt;%</span></span>
<span id="cb31-1137"><a href="#cb31-1137" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb31-1138"><a href="#cb31-1138" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">sum_reads=</span><span class="fu">sum</span>(n_reads_hv), <span class="at">n_read_pairs=</span><span class="fu">first</span>(n_read_pairs)) <span class="sc">%&gt;%</span></span>
<span id="cb31-1139"><a href="#cb31-1139" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">ra =</span> sum_reads<span class="sc">/</span>n_read_pairs) <span class="sc">%&gt;%</span> </span>
<span id="cb31-1140"><a href="#cb31-1140" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(ra) <span class="sc">%&gt;%</span> <span class="fu">mean</span>()</span>
<span id="cb31-1141"><a href="#cb31-1141" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-1142"><a href="#cb31-1142" aria-hidden="true" tabindex="-1"></a>prop_non <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span>  </span>
<span id="cb31-1143"><a href="#cb31-1143" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(name <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">'Rhabdoviridae'</span>,<span class="st">'Microviridae'</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb31-1144"><a href="#cb31-1144" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb31-1145"><a href="#cb31-1145" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">sum_reads_spiked =</span> <span class="fu">sum</span>(n_reads_hv)) <span class="sc">%&gt;%</span></span>
<span id="cb31-1146"><a href="#cb31-1146" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample,sum_reads_spiked)</span>
<span id="cb31-1147"><a href="#cb31-1147" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-1148"><a href="#cb31-1148" aria-hidden="true" tabindex="-1"></a>non_spiked <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span> </span>
<span id="cb31-1149"><a href="#cb31-1149" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb31-1150"><a href="#cb31-1150" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">sum_reads=</span><span class="fu">sum</span>(n_reads_hv), <span class="at">n_read_pairs=</span><span class="fu">first</span>(n_read_pairs)) <span class="sc">%&gt;%</span> </span>
<span id="cb31-1151"><a href="#cb31-1151" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(prop_non, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">'sample'</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb31-1152"><a href="#cb31-1152" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">ra_non_spiked =</span> (sum_reads<span class="sc">-</span>sum_reads_spiked)<span class="sc">/</span>(n_read_pairs<span class="sc">-</span>sum_reads_spiked)) <span class="sc">%&gt;%</span></span>
<span id="cb31-1153"><a href="#cb31-1153" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(ra_non_spiked) <span class="sc">%&gt;%</span> <span class="fu">mean</span>()</span>
<span id="cb31-1154"><a href="#cb31-1154" aria-hidden="true" tabindex="-1"></a>all</span>
<span id="cb31-1155"><a href="#cb31-1155" aria-hidden="true" tabindex="-1"></a>non_spiked</span>
<span id="cb31-1156"><a href="#cb31-1156" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-1157"><a href="#cb31-1157" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb31-1158"><a href="#cb31-1158" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-1159"><a href="#cb31-1159" aria-hidden="true" tabindex="-1"></a>I'd like to BLAST some of these reads against the NCBI nt database to see if we can get some more info on them.</span>
<span id="cb31-1160"><a href="#cb31-1160" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-1161"><a href="#cb31-1161" aria-hidden="true" tabindex="-1"></a><span class="fu"># Conclusion</span></span>
<span id="cb31-1162"><a href="#cb31-1162" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-1163"><a href="#cb31-1163" aria-hidden="true" tabindex="-1"></a>Overall, we found a lot of anelloviridae, which makes sense given that this paper is about discovering new species of anelloviridae in blood.</span>
</code><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>