<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.53">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Harmon Bhasin">
<meta name="dcterms.date" content="2024-09-12">

<title>Workflow of Thompson et al.&nbsp;(2023) – Harmon's Public Notebook</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script async="" src="https://hypothes.is/embed.js"></script>
<script>
  window.document.addEventListener("DOMContentLoaded", function (_event) {
    document.body.classList.add('hypothesis-enabled');
  });
</script>
<style>

      .quarto-title-block .quarto-title-banner {
        background: #de2d26;
      }
</style>
<script src="../site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<script src="../site_libs/d3-4.5.0/d3.min.js"></script>
<script src="../site_libs/sankey-1/sankey.js"></script>
<script src="../site_libs/sankeyNetwork-binding-0.4/sankeyNetwork.js"></script>
<link href="../site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet">
<script src="../site_libs/pagedtable-1.1/js/pagedtable.js"></script>

  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

<link rel="stylesheet" href="../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top quarto-banner">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">Harmon’s Public Notebook</span>
    </a>
  </div>
        <div class="quarto-navbar-tools tools-end">
</div>
          <div id="quarto-search" class="" title="Search"></div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-page-left">
      <div class="quarto-title-block"><div><h1 class="title">Workflow of Thompson et al.&nbsp;(2023)</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
            <p class="subtitle lead">Individual whole blood from US (RNA)</p>
                      </div>
  </div>
    
  
  <div class="quarto-title-meta column-page-left">

      <div>
      <div class="quarto-title-meta-heading">Author</div>
      <div class="quarto-title-meta-contents">
               <p>Harmon Bhasin </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">September 12, 2024</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#the-raw-data" id="toc-the-raw-data" class="nav-link active" data-scroll-target="#the-raw-data"><span class="header-section-number">1</span> The raw data</a>
  <ul class="collapse">
  <li><a href="#about" id="toc-about" class="nav-link" data-scroll-target="#about"><span class="header-section-number">1.1</span> About</a>
  <ul class="collapse">
  <li><a href="#sample-library-preparation" id="toc-sample-library-preparation" class="nav-link" data-scroll-target="#sample-library-preparation"><span class="header-section-number">1.1.1</span> Sample + library preparation</a></li>
  </ul></li>
  <li><a href="#quality-control-metrics" id="toc-quality-control-metrics" class="nav-link" data-scroll-target="#quality-control-metrics"><span class="header-section-number">1.2</span> Quality control metrics</a></li>
  </ul></li>
  <li><a href="#preprocessing" id="toc-preprocessing" class="nav-link" data-scroll-target="#preprocessing"><span class="header-section-number">2</span> Preprocessing</a>
  <ul class="collapse">
  <li><a href="#summary" id="toc-summary" class="nav-link" data-scroll-target="#summary"><span class="header-section-number">2.1</span> Summary</a></li>
  <li><a href="#quality-control-metrics-1" id="toc-quality-control-metrics-1" class="nav-link" data-scroll-target="#quality-control-metrics-1"><span class="header-section-number">2.2</span> Quality control metrics</a></li>
  </ul></li>
  <li><a href="#taxonomic-composition" id="toc-taxonomic-composition" class="nav-link" data-scroll-target="#taxonomic-composition"><span class="header-section-number">3</span> Taxonomic composition</a>
  <ul class="collapse">
  <li><a href="#high-level-composition" id="toc-high-level-composition" class="nav-link" data-scroll-target="#high-level-composition"><span class="header-section-number">3.1</span> High-level composition</a></li>
  <li><a href="#total-viral-content" id="toc-total-viral-content" class="nav-link" data-scroll-target="#total-viral-content"><span class="header-section-number">3.2</span> Total viral content</a></li>
  <li><a href="#taxonomic-composition-of-viruses" id="toc-taxonomic-composition-of-viruses" class="nav-link" data-scroll-target="#taxonomic-composition-of-viruses"><span class="header-section-number">3.3</span> Taxonomic composition of viruses</a></li>
  </ul></li>
  <li><a href="#human-infecting-virus-reads" id="toc-human-infecting-virus-reads" class="nav-link" data-scroll-target="#human-infecting-virus-reads"><span class="header-section-number">4</span> Human-infecting virus reads</a>
  <ul class="collapse">
  <li><a href="#overall-relative-abundance" id="toc-overall-relative-abundance" class="nav-link" data-scroll-target="#overall-relative-abundance"><span class="header-section-number">4.1</span> Overall relative abundance</a></li>
  <li><a href="#overall-taxonomy-and-composition" id="toc-overall-taxonomy-and-composition" class="nav-link" data-scroll-target="#overall-taxonomy-and-composition"><span class="header-section-number">4.2</span> Overall taxonomy and composition</a></li>
  <li><a href="#analyzing-specific-families" id="toc-analyzing-specific-families" class="nav-link" data-scroll-target="#analyzing-specific-families"><span class="header-section-number">4.3</span> Analyzing specific families</a>
  <ul class="collapse">
  <li><a href="#flaviviridae-number-of-reads-25997" id="toc-flaviviridae-number-of-reads-25997" class="nav-link" data-scroll-target="#flaviviridae-number-of-reads-25997"><span class="header-section-number">4.3.1</span> Flaviviridae (Number of reads: 25,997)</a></li>
  <li><a href="#anelloviridae-number-of-reads-10103" id="toc-anelloviridae-number-of-reads-10103" class="nav-link" data-scroll-target="#anelloviridae-number-of-reads-10103"><span class="header-section-number">4.3.2</span> Anelloviridae (Number of reads: 10,103)</a></li>
  <li><a href="#coronaviridae-number-of-reads-1104" id="toc-coronaviridae-number-of-reads-1104" class="nav-link" data-scroll-target="#coronaviridae-number-of-reads-1104"><span class="header-section-number">4.3.3</span> Coronaviridae (Number of reads: 1,104)</a></li>
  <li><a href="#orthoherpesviridae-number-of-reads-534" id="toc-orthoherpesviridae-number-of-reads-534" class="nav-link" data-scroll-target="#orthoherpesviridae-number-of-reads-534"><span class="header-section-number">4.3.4</span> Orthoherpesviridae (Number of reads: 534)</a></li>
  <li><a href="#parvoviridae-number-of-reads-335" id="toc-parvoviridae-number-of-reads-335" class="nav-link" data-scroll-target="#parvoviridae-number-of-reads-335"><span class="header-section-number">4.3.5</span> Parvoviridae (Number of reads: 335)</a></li>
  <li><a href="#retroviridae-number-of-reads-224" id="toc-retroviridae-number-of-reads-224" class="nav-link" data-scroll-target="#retroviridae-number-of-reads-224"><span class="header-section-number">4.3.6</span> Retroviridae (Number of reads: 224)</a></li>
  </ul></li>
  <li><a href="#relative-abundance-of-pathogenic-viruses-of-interest" id="toc-relative-abundance-of-pathogenic-viruses-of-interest" class="nav-link" data-scroll-target="#relative-abundance-of-pathogenic-viruses-of-interest"><span class="header-section-number">4.4</span> Relative abundance of pathogenic viruses of interest</a></li>
  <li><a href="#relative-abundance-assuming-perfect-human-read-removal" id="toc-relative-abundance-assuming-perfect-human-read-removal" class="nav-link" data-scroll-target="#relative-abundance-assuming-perfect-human-read-removal"><span class="header-section-number">4.5</span> Relative abundance assuming perfect human read removal</a></li>
  </ul></li>
  <li><a href="#conclusion" id="toc-conclusion" class="nav-link" data-scroll-target="#conclusion"><span class="header-section-number">5</span> Conclusion</a></li>
  <li><a href="#appendix" id="toc-appendix" class="nav-link" data-scroll-target="#appendix"><span class="header-section-number">6</span> Appendix</a>
  <ul class="collapse">
  <li><a href="#human-infecting-virus-families-genera-and-species" id="toc-human-infecting-virus-families-genera-and-species" class="nav-link" data-scroll-target="#human-infecting-virus-families-genera-and-species"><span class="header-section-number">6.1</span> Human-infecting virus families, genera, and species</a></li>
  </ul></li>
  </ul>
<div class="quarto-other-links"><h2>Other Links</h2><ul><li><a href="https://doi.org/10.1038/s41591-022-02107-4"><i class="bi bi-link-45deg"></i>Paper</a></li><li><a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA891054"><i class="bi bi-link-45deg"></i>Data</a></li></ul></div><div class="quarto-code-links"><h2>Code Links</h2><ul><li><a href="https://github.com/naobservatory/harmons-public-notebook/blob/main/notebooks/2024-09-12-thompson.qmd"><i class="bi bi-file-code"></i>Code for this post</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content quarto-banner-title-block column-page-left" id="quarto-document-content">





<p>This is another potential study of <a href="https://data.securebio.org/harmons-public-notebook/notebooks/2024-07-08_cebria-mendoza.html">this series</a>. In this post, I analyze <a href="https://doi.org/10.1038/s41591-022-02107-4">Thompson 2023</a>, a dataset with 417 samples of whole blood from healthy and SARS-CoV-2 infected individuals in the US.</p>
<p><em>I’d like to thank Lenni for giving me feedback on the notebook and Will for providing me with a boilerplate rmarkdown file. This notebook uses the MGS Workflow v2.2.1 (note that this is old).</em></p>
<section id="the-raw-data" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> The raw data</h1>
<section id="about" class="level2" data-number="1.1">
<h2 data-number="1.1" class="anchored" data-anchor-id="about"><span class="header-section-number">1.1</span> About</h2>
<p><a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA891054">This dataset</a> is composed of 417 samples of whole blood from the US. In total, there were 353 SARS-CoV-2 infected individuals and 64 healthy individuals that contributed to this study. For each sample, they performed RNA-sequencing, with Illumina NovaSeq 6000, producing 2x100 bp reads.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>low_sample_number <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">'SRR21924256'</span>)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Import libraries and extract metadata from sample names</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>libraries <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(libraries_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span>sample <span class="sc">%in%</span> low_sample_number)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="co">#meta_data &lt;- read_csv('/Users/harmonbhasin/work/securebio/nao-harmon/thijssen2023/preprocessing/SraRunTable.txt') %&gt;%</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="co">#  select('Library Name', Run) %&gt;% rename(sample = Run, library = 'Library Name')</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="co">#libraries &lt;- libraries %&gt;%</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="co">#  left_join(meta_data, by = 'sample') %&gt;%</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="co">#  select(sample, library=library.y) %&gt;% </span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="co">#  filter(library != 'PCL21') %&gt;%</span></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="co">#  mutate(library = factor(library, levels = c(paste0('PCL', 1:20))))</span></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="co">#libraries</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="sample-library-preparation" class="level3" data-number="1.1.1">
<h3 data-number="1.1.1" class="anchored" data-anchor-id="sample-library-preparation"><span class="header-section-number">1.1.1</span> Sample + library preparation</h3>
<p>This is paraphrased from the paper:</p>
<blockquote class="blockquote">
<p>Patients were seen between April and June 2020 for COVID-19. The whole blood used for RNA-seq was collected in Tempus RNA Blood Tubes (Thermo Fisher Scientific, 4342792). As soon as possible after blood collection, tubes were shaken and stored at −80 °C.</p>
<p>RNA extraction, library preparation and sequencing were performed as described previously <a href="https://www.nature.com/articles/s41467-021-24981-1">28</a>. In brief, frozen blood samples were thawed, and total RNA was extracted using a modification of the MagMax protocol for Stabilized Blood Tubes RNA Isolation Kit. Samples yielding sufficient RNA (&gt;50 ng) were barcoded and prepared for pooled whole transcriptome sequencing using the TruSeq Stranded Total RNA Library Prep Gold, which is designed to remove ribosomal, globin and mitochondrial RNA. Libraries were amplified with 15 cycles of PCR, pooled and sequenced on a NovaSeq 6000 (Illumina) using Sprime flow cells with 100-bp paired-end reads, targeting a mean of 50 million read pairs per sample.</p>
</blockquote>
</section>
</section>
<section id="quality-control-metrics" class="level2" data-number="1.2">
<h2 data-number="1.2" class="anchored" data-anchor-id="quality-control-metrics"><span class="header-section-number">1.2</span> Quality control metrics</h2>
<p>In total, these 417 samples contained 27.7B read pairs. The samples had 30.7M - 133.9M (mean ~66.4M) read pairs each. The number of reads looks pretty good, although a few samples have a much lower number of reads. The total number of base pairs also looks reasonable, matching the trends of the number of reads. The duplication rate is very high. Adapter content is variable, but most samples have a low adapter content (below 5%). As we’d expect, we see higher quality Q scores near the beginning of the read and a gradual decline towards the end of the read, however all positions seem to have a Q score of 35 which means that our reads are ~99.97% accurate. When looking at the Q score over all sequences, we can see a sharp peak around 35, which corresponds to the previous plot, indicating high read quality.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare data</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>basic_stats_raw_metrics <span class="ot">&lt;-</span> basic_stats_raw <span class="sc">%&gt;%</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(library,</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at"># Read pairs</span><span class="st">`</span> <span class="ot">=</span> n_read_pairs,</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Total base pairs</span><span class="sc">\n</span><span class="at">(approx)</span><span class="st">`</span> <span class="ot">=</span> n_bases_approx,</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">% Duplicates</span><span class="sc">\n</span><span class="at">(FASTQC)</span><span class="st">`</span> <span class="ot">=</span> percent_duplicates) <span class="sc">%&gt;%</span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>library, <span class="at">names_to =</span> <span class="st">"metric"</span>, <span class="at">values_to =</span> <span class="st">"value"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">metric =</span> <span class="fu">fct_inorder</span>(metric)) </span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Set up plot templates</span></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>g_basic <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(basic_stats_raw_metrics, <span class="fu">aes</span>(<span class="at">x =</span> library, <span class="at">y =</span> value)) <span class="sc">+</span></span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position =</span> <span class="st">"dodge"</span>) <span class="sc">+</span></span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>() <span class="sc">+</span></span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">expand_limits</span>(<span class="at">y =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">100</span>)) <span class="sc">+</span></span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(metric <span class="sc">~</span> ., <span class="at">scales =</span> <span class="st">"free"</span>, <span class="at">space =</span> <span class="st">"free_x"</span>, <span class="at">switch =</span> <span class="st">"y"</span>) <span class="sc">+</span></span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a>  theme_kit <span class="sc">+</span> <span class="fu">theme</span>(</span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">strip.text.y =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"plain"</span>)</span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a>g_basic</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="2024-09-12-thompson_files/figure-html/plot-basic-stats-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Set up plotting templates</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>g_qual_raw <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">linetype=</span>read_pair, <span class="at">group=</span><span class="fu">interaction</span>(sample,read_pair))) <span class="sc">+</span> </span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_linetype_discrete</span>(<span class="at">name =</span> <span class="st">"Read Pair"</span>) <span class="sc">+</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color=</span><span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>),</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">linetype =</span> <span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>)) <span class="sc">+</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>  theme_base</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize adapters</span></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>g_adapters_raw <span class="ot">&lt;-</span> g_qual_raw <span class="sc">+</span> </span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>pc_adapters), <span class="at">data=</span>adapter_stats_raw) <span class="sc">+</span></span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Adapters"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">10</span>,<span class="dv">1</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Position"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">500</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(.<span class="sc">~</span>adapter)</span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a>g_adapters_raw</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="2024-09-12-thompson_files/figure-html/plot-raw-quality-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize quality</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>g_quality_base_raw <span class="ot">&lt;-</span> g_qual_raw <span class="sc">+</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>mean_phred_score), <span class="at">data=</span>quality_base_stats_raw) <span class="sc">+</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"Mean Phred score"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">10</span>,<span class="dv">45</span>)) <span class="sc">+</span></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Position"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">500</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>g_quality_base_raw</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="2024-09-12-thompson_files/figure-html/plot-raw-quality-2.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>g_quality_seq_raw <span class="ot">&lt;-</span> g_qual_raw <span class="sc">+</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>mean_phred_score, <span class="at">y=</span>n_sequences), <span class="at">data=</span>quality_seq_stats_raw) <span class="sc">+</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Mean Phred score"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"# Sequences"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>g_quality_seq_raw</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="2024-09-12-thompson_files/figure-html/plot-raw-quality-3.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="preprocessing" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Preprocessing</h1>
<section id="summary" class="level2" data-number="2.1">
<h2 data-number="2.1" class="anchored" data-anchor-id="summary"><span class="header-section-number">2.1</span> Summary</h2>
<p>The average fraction of reads at each stage in the preprocessing pipeline is shown in the following table. Adapter trimming &amp; filtering doesn’t lose too many reads. Deduplication loses us about 55% of reads on average, which is good considering the high duplication content seen earlier. Ribodepletion only removes about 2% of reads in total, which makes sense as they did ribosomal depletion during sample preparation.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># </span><span class="al">TODO</span><span class="co">: Group by pool size as well</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Count read losses</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>n_reads_rel <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> </span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, stage, percent_duplicates, n_read_pairs) <span class="sc">%&gt;%</span></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(sample, stage) <span class="sc">%&gt;%</span></span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_retained =</span> n_read_pairs <span class="sc">/</span> <span class="fu">lag</span>(n_read_pairs),</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_lost =</span> <span class="dv">1</span> <span class="sc">-</span> p_reads_retained,</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_retained_abs =</span> n_read_pairs <span class="sc">/</span> n_read_pairs[<span class="dv">1</span>],</span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_lost_abs =</span> <span class="dv">1</span><span class="sc">-</span>p_reads_retained_abs,</span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_lost_abs_marginal =</span> p_reads_lost_abs <span class="sc">-</span> <span class="fu">lag</span>(p_reads_lost_abs))</span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>n_reads_rel_display <span class="ot">&lt;-</span> n_reads_rel <span class="sc">%&gt;%</span> </span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">Stage=</span>stage) <span class="sc">%&gt;%</span> </span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Stage) <span class="sc">%&gt;%</span> </span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="st">`</span><span class="at">% Total Reads Lost (Cumulative)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">round</span>(<span class="fu">min</span>(p_reads_lost_abs<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">"-"</span>, <span class="fu">round</span>(<span class="fu">max</span>(p_reads_lost_abs<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">" (mean "</span>, <span class="fu">round</span>(<span class="fu">mean</span>(p_reads_lost_abs<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">")"</span>),</span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a>            <span class="st">`</span><span class="at">% Total Reads Lost (Marginal)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">round</span>(<span class="fu">min</span>(p_reads_lost_abs_marginal<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">"-"</span>, <span class="fu">round</span>(<span class="fu">max</span>(p_reads_lost_abs_marginal<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">" (mean "</span>, <span class="fu">round</span>(<span class="fu">mean</span>(p_reads_lost_abs_marginal<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">")"</span>), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Stage <span class="sc">!=</span> <span class="st">"raw_concat"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Stage =</span> Stage <span class="sc">%&gt;%</span> as.numeric <span class="sc">%&gt;%</span> <span class="fu">factor</span>(<span class="at">labels=</span><span class="fu">c</span>(<span class="st">"Trimming &amp; filtering"</span>, <span class="st">"Deduplication"</span>, <span class="st">"Initial ribodepletion"</span>, <span class="st">"Secondary ribodepletion"</span>)))</span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a>n_reads_rel_display</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["Stage"],"name":[1],"type":["fct"],"align":["left"]},{"label":["% Total Reads Lost (Cumulative)"],"name":[2],"type":["chr"],"align":["left"]},{"label":["% Total Reads Lost (Marginal)"],"name":[3],"type":["chr"],"align":["left"]}],"data":[{"1":"Trimming & filtering","2":"1-33.3 (mean 3.4)","3":"1-33.3 (mean 3.4)"},{"1":"Deduplication","2":"23-93.9 (mean 58.4)","3":"0-88.6 (mean 55)"},{"1":"Initial ribodepletion","2":"24.4-98.5 (mean 60.3)","3":"0.1-33.3 (mean 1.9)"},{"1":"Secondary ribodepletion","2":"24.5-99.1 (mean 60.4)","3":"0-1.3 (mean 0.1)"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
</section>
<section id="quality-control-metrics-1" class="level2" data-number="2.2">
<h2 data-number="2.2" class="anchored" data-anchor-id="quality-control-metrics-1"><span class="header-section-number">2.2</span> Quality control metrics</h2>
<p>Cleaning seems to get rid of the library contamination and the polyg contamination. Polya contamination still is present, but I don’t think we’ve found a solution to remove it as yet. Q score remain the same during read cleaning when looking at the positions, with the end of the read actually improving in score. Q scores across all sequences look pretty much the same throughout cleaning.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>g_qual <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">linetype=</span>read_pair, <span class="at">group=</span><span class="fu">interaction</span>(sample,read_pair))) <span class="sc">+</span> </span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_linetype_discrete</span>(<span class="at">name =</span> <span class="st">"Read Pair"</span>) <span class="sc">+</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color=</span><span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>),</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">linetype =</span> <span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>)) <span class="sc">+</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>  theme_base</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize adapters</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>g_adapters <span class="ot">&lt;-</span> g_qual <span class="sc">+</span> </span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>pc_adapters), <span class="at">data=</span>adapter_stats) <span class="sc">+</span></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Adapters"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">20</span>),</span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">50</span>,<span class="dv">10</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Position"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">140</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(stage<span class="sc">~</span>adapter)</span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a>g_adapters</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="2024-09-12-thompson_files/figure-html/plot-quality-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize quality</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>g_quality_base <span class="ot">&lt;-</span> g_qual <span class="sc">+</span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>mean_phred_score), <span class="at">data=</span>quality_base_stats) <span class="sc">+</span></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"Mean Phred score"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">10</span>,<span class="dv">45</span>)) <span class="sc">+</span></span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Position"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">140</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(stage<span class="sc">~</span>.)</span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>g_quality_base</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="2024-09-12-thompson_files/figure-html/plot-quality-2.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>g_quality_seq <span class="ot">&lt;-</span> g_qual <span class="sc">+</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>mean_phred_score, <span class="at">y=</span>n_sequences), <span class="at">data=</span>quality_seq_stats) <span class="sc">+</span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Mean Phred score"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"# Sequences"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(stage<span class="sc">~</span>., <span class="at">scales =</span> <span class="st">"free_y"</span>)</span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>g_quality_seq</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="2024-09-12-thompson_files/figure-html/plot-quality-3.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<p>These plots below show the trends from above in each sample. Deduplication seems to drop by a lot post deduplication,which is a good thing given it was so high earlier. Mean read length remains relatively constant throuhgout. Some samples still seem to have high ribosomal content, however I’m not too worried about this.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>g_stage_trace <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(basic_stats, <span class="fu">aes</span>(<span class="at">x=</span>stage, <span class="at">group=</span>sample)) <span class="sc">+</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot reads over preprocessing</span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>g_reads_stages <span class="ot">&lt;-</span> g_stage_trace <span class="sc">+</span></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>n_read_pairs)) <span class="sc">+</span></span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"# Read pairs"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>))</span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>g_reads_stages</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="2024-09-12-thompson_files/figure-html/preproc-figures-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot relative read losses during preprocessing</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>g_reads_rel <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(n_reads_rel, </span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">aes</span>(<span class="at">x=</span>stage, <span class="at">group=</span>sample)) <span class="sc">+</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>p_reads_lost_abs_marginal)) <span class="sc">+</span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"% Total Reads Lost"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), </span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="cf">function</span>(x) x<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>g_reads_rel</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="2024-09-12-thompson_files/figure-html/preproc-figures-2.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>stage_dup <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(stage) <span class="sc">%&gt;%</span> </span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">dmin =</span> <span class="fu">min</span>(percent_duplicates), <span class="at">dmax=</span><span class="fu">max</span>(percent_duplicates),</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>            <span class="at">dmean=</span><span class="fu">mean</span>(percent_duplicates), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>g_dup_stages <span class="ot">&lt;-</span> g_stage_trace <span class="sc">+</span></span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>percent_duplicates)) <span class="sc">+</span></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"% Duplicates"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>g_dup_stages</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="2024-09-12-thompson_files/figure-html/preproc-dedup-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>g_readlen_stages <span class="ot">&lt;-</span> g_stage_trace <span class="sc">+</span> <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>mean_seq_len)) <span class="sc">+</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"Mean read length (nt)"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>))</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>g_readlen_stages</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="2024-09-12-thompson_files/figure-html/preproc-dedup-2.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate reads lost during ribodepletion (approximation for % ribosomal reads)</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>reads_ribo <span class="ot">&lt;-</span> n_reads_rel <span class="sc">%&gt;%</span> </span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(stage <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"dedup"</span>, <span class="st">"ribo_secondary"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads_ribo=</span><span class="dv">1</span><span class="sc">-</span>n_read_pairs[<span class="dv">2</span>]<span class="sc">/</span>n_read_pairs[<span class="dv">1</span>], <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by =</span> <span class="st">'sample'</span>)</span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>reads_ribo_summ <span class="ot">&lt;-</span> reads_ribo <span class="sc">%&gt;%</span></span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">min=</span><span class="fu">min</span>(p_reads_ribo), <span class="at">max=</span><span class="fu">max</span>(p_reads_ribo),</span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>            <span class="at">mean=</span><span class="fu">mean</span>(p_reads_ribo), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by =</span> <span class="st">'sample'</span>)</span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a>g_reads_ribo <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(reads_ribo, </span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">aes</span>(<span class="at">x=</span>library, <span class="at">y=</span>p_reads_ribo)) <span class="sc">+</span></span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span> </span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"Approx % ribosomal reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>),</span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>)<span class="sc">+</span></span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb14-18"><a href="#cb14-18" aria-hidden="true" tabindex="-1"></a>g_reads_ribo</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="2024-09-12-thompson_files/figure-html/ribo-frac-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="taxonomic-composition" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> Taxonomic composition</h1>
<section id="high-level-composition" class="level2" data-number="3.1">
<h2 data-number="3.1" class="anchored" data-anchor-id="high-level-composition"><span class="header-section-number">3.1</span> High-level composition</h2>
<p>To assess the high-level composition of the reads, I ran the ribodepleted files through Kraken2 and summarized the results with Bracken.</p>
<p>The groups listed below were created by Will:</p>
<ul>
<li>Filtered (removed during cleaning)</li>
<li>Duplicate (removed during deduplication)</li>
<li>Ribosomal (removed during ribodepletion)</li>
<li>Unassigned (non-ribosomal reads that were not assigned to any taxon by Kraken/Bracken)</li>
<li>Bacterial (non-ribosomal reads assigned to the Bacteria domain by Kraken/Bracken)</li>
<li>Archaeal (non-ribosomal reads assigned to the Archaea domain by Kraken/Bracken)</li>
<li>Viral (non-ribosomal reads assigned to the Viruses domain by Kraken/Bracken)</li>
<li>Human (non-ribosomal reads assigned to the Eukarya domain by Kraken/Bracken)</li>
</ul>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare plotting templates</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>g_comp_base <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">x=</span>library, <span class="at">y=</span>p_reads, <span class="at">fill=</span>classification)) <span class="sc">+</span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">name=</span><span class="st">"Plasma pool"</span>) <span class="sc">+</span></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>  theme_kit <span class="sc">+</span> </span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust=</span><span class="dv">0</span>, <span class="at">face=</span><span class="st">"plain"</span>, <span class="at">size=</span><span class="fu">rel</span>(<span class="fl">1.5</span>)))</span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>scale_y_pc_reads <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">partial</span>(scale_y_continuous, <span class="at">name =</span> <span class="st">"% Reads"</span>,</span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>)</span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a>geom_comp <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">partial</span>(geom_col, <span class="at">position =</span> <span class="st">"stack"</span>, <span class="at">width =</span> <span class="dv">1</span>)</span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Define a color palette for the classification</span></span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a>classification_colors <span class="ot">&lt;-</span> <span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">"Accent"</span>)</span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(classification_colors) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Filtered"</span>, <span class="st">"Duplicate"</span>, <span class="st">"Ribosomal"</span>, <span class="st">"Unassigned"</span>, <span class="st">"Bacterial"</span>, <span class="st">"Archaeal"</span>, <span class="st">"Viral"</span>, <span class="st">"Human"</span>)</span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a>scale_fill_classification <span class="ot">&lt;-</span> <span class="cf">function</span>() {</span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> classification_colors, <span class="at">name =</span> <span class="st">"Classification"</span>)</span>
<span id="cb15-15"><a href="#cb15-15" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb15-16"><a href="#cb15-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-17"><a href="#cb15-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot overall composition</span></span>
<span id="cb15-18"><a href="#cb15-18" aria-hidden="true" tabindex="-1"></a>g_comp <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> <span class="fu">geom_comp</span>(<span class="at">data =</span> comp) <span class="sc">+</span></span>
<span id="cb15-19"><a href="#cb15-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_pc_reads</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>)) <span class="sc">+</span></span>
<span id="cb15-20"><a href="#cb15-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_classification</span>() <span class="sc">+</span> </span>
<span id="cb15-21"><a href="#cb15-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Read composition (all reads, all groups)"</span>)</span>
<span id="cb15-22"><a href="#cb15-22" aria-hidden="true" tabindex="-1"></a>g_comp</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="2024-09-12-thompson_files/figure-html/plot-composition-all-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>g_comp_assigned <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_comp</span>(<span class="at">data =</span> comp_assigned) <span class="sc">+</span></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_pc_reads</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>)) <span class="sc">+</span></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_classification</span>() <span class="sc">+</span> </span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Read composition (assigned reads, all groups)"</span>)</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>g_comp_assigned</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="2024-09-12-thompson_files/figure-html/plot-composition-all-2.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="total-viral-content" class="level2" data-number="3.2">
<h2 data-number="3.2" class="anchored" data-anchor-id="total-viral-content"><span class="header-section-number">3.2</span> Total viral content</h2>
<p>Total viral fraction average <span class="math inline">\(1.12 \times 10^{-6}\)</span> across samples. As a fraction of assigned (rather than total) reads, this jumped to <span class="math inline">\(2.96 \times 10^{-6}\)</span>:</p>
</section>
<section id="taxonomic-composition-of-viruses" class="level2" data-number="3.3">
<h2 data-number="3.3" class="anchored" data-anchor-id="taxonomic-composition-of-viruses"><span class="header-section-number">3.3</span> Taxonomic composition of viruses</h2>
<p>There seems to be no dominant virus in the samples. Specifically, this seems to be the dataset with the most viral diversity (when compared to the previous plasma datasets). The threshold for the label “other” are the set of families that make up less than 30% composition in all samples.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>major_threshold <span class="ot">&lt;-</span> <span class="fl">0.3</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify major viral families</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>viral_families_major_tab <span class="ot">&lt;-</span> viral_families <span class="sc">%&gt;%</span> </span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads_viral_max =</span> <span class="fu">max</span>(p_reads_viral), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_viral_max <span class="sc">&gt;=</span> major_threshold)</span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>viral_families_major_list <span class="ot">&lt;-</span> viral_families_major_tab <span class="sc">%&gt;%</span> <span class="fu">pull</span>(name)</span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>viral_families_major <span class="ot">&lt;-</span> viral_families <span class="sc">%&gt;%</span> </span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(name <span class="sc">%in%</span> viral_families_major_list) <span class="sc">%&gt;%</span></span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, taxid, sample, p_reads_viral)</span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a>viral_families_minor <span class="ot">&lt;-</span> viral_families_major <span class="sc">%&gt;%</span> </span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads_viral_major =</span> <span class="fu">sum</span>(p_reads_viral), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="st">"Other"</span>, <span class="at">taxid=</span><span class="cn">NA</span>, <span class="at">p_reads_viral =</span> <span class="dv">1</span><span class="sc">-</span>p_reads_viral_major) <span class="sc">%&gt;%</span></span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, taxid, sample, p_reads_viral)</span>
<span id="cb17-17"><a href="#cb17-17" aria-hidden="true" tabindex="-1"></a>viral_families_display <span class="ot">&lt;-</span> viral_families_major <span class="sc">%&gt;%</span> </span>
<span id="cb17-18"><a href="#cb17-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_rows</span>(viral_families_minor) <span class="sc">%&gt;%</span></span>
<span id="cb17-19"><a href="#cb17-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_viral)) <span class="sc">%&gt;%</span> </span>
<span id="cb17-20"><a href="#cb17-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">factor</span>(name, <span class="at">levels=</span><span class="fu">c</span>(viral_families_major_list, <span class="st">"Other"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb17-21"><a href="#cb17-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_viral, <span class="at">classification=</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb17-22"><a href="#cb17-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">'sample'</span>)</span>
<span id="cb17-23"><a href="#cb17-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-24"><a href="#cb17-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-25"><a href="#cb17-25" aria-hidden="true" tabindex="-1"></a>pal <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">'Dark2'</span>),<span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">'Accent'</span>), <span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">'Set1'</span>), <span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">'Pastel1'</span>), <span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">'Pastel2'</span>))</span>
<span id="cb17-26"><a href="#cb17-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-27"><a href="#cb17-27" aria-hidden="true" tabindex="-1"></a>n_classifications <span class="ot">&lt;-</span> <span class="fu">length</span>(<span class="fu">unique</span>(viral_families_display<span class="sc">$</span>classification)) <span class="sc">-</span> <span class="dv">1</span></span>
<span id="cb17-28"><a href="#cb17-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-29"><a href="#cb17-29" aria-hidden="true" tabindex="-1"></a>custom_palette_with_black <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb17-30"><a href="#cb17-30" aria-hidden="true" tabindex="-1"></a>  pal[<span class="dv">1</span><span class="sc">:</span>n_classifications],</span>
<span id="cb17-31"><a href="#cb17-31" aria-hidden="true" tabindex="-1"></a>  <span class="st">"black"</span></span>
<span id="cb17-32"><a href="#cb17-32" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb17-33"><a href="#cb17-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-34"><a href="#cb17-34" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb17-35"><a href="#cb17-35" aria-hidden="true" tabindex="-1"></a>g_families <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb17-36"><a href="#cb17-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_comp</span>(<span class="at">data=</span>viral_families_display) <span class="sc">+</span></span>
<span id="cb17-37"><a href="#cb17-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Viral Reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb17-38"><a href="#cb17-38" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb17-39"><a href="#cb17-39" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb17-40"><a href="#cb17-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> custom_palette_with_black)</span>
<span id="cb17-41"><a href="#cb17-41" aria-hidden="true" tabindex="-1"></a>g_families</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="2024-09-12-thompson_files/figure-html/viral-family-composition-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="human-infecting-virus-reads" class="level1" data-number="4">
<h1 data-number="4"><span class="header-section-number">4</span> Human-infecting virus reads</h1>
<section id="overall-relative-abundance" class="level2" data-number="4.1">
<h2 data-number="4.1" class="anchored" data-anchor-id="overall-relative-abundance"><span class="header-section-number">4.1</span> Overall relative abundance</h2>
<p>I calculated the relative abundance of human-infecting viruses in two ways:</p>
<ul>
<li><p>First, as the total number of deduplicated human-virus reads in each sample, divided by the number of raw reads (“All reads”). This results in a viral fraction of <span class="math inline">\(1.5 \times 10^{-6}\)</span> across all samples</p></li>
<li><p>Second, as a fraction of preprocessed (cleaned, deduplicated, computationally ribodepleted) reads (“Preprocessed reads”). This results in a viral fraction of <span class="math inline">\(3.2 \times 10^{-6}\)</span> across all samples.</p></li>
</ul>
</section>
<section id="overall-taxonomy-and-composition" class="level2" data-number="4.2">
<h2 data-number="4.2" class="anchored" data-anchor-id="overall-taxonomy-and-composition"><span class="header-section-number">4.2</span> Overall taxonomy and composition</h2>
<p>Composition of HV reads was changed from when looking at all viral reads. First, only 15% of the individuals in this dataset had human infecting viruses detectable in their blood. Second, individuals seem to be dominated by a single viral family. The threshold for the label “other” are the set of families that make up less than 5% composition in all samples.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>threshold_major_family <span class="ot">&lt;-</span> <span class="fl">0.05</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Count reads for each human-viral family</span></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>hv_family_counts <span class="ot">&lt;-</span> hv_reads_family <span class="sc">%&gt;%</span> </span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_hv =</span> n_reads_hv<span class="sc">/</span><span class="fu">sum</span>(n_reads_hv))</span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify high-ranking families and group others</span></span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>hv_family_major_tab <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span> </span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_hv <span class="sc">==</span> <span class="fu">max</span>(p_reads_hv)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_hv)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(p_reads_hv <span class="sc">&gt;</span> threshold_major_family)</span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a>hv_family_counts_major <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span></span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">ifelse</span>(name <span class="sc">%in%</span> hv_family_major_tab<span class="sc">$</span>name, name, <span class="st">"Other"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb18-16"><a href="#cb18-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, name_display) <span class="sc">%&gt;%</span></span>
<span id="cb18-17"><a href="#cb18-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_hv =</span> <span class="fu">sum</span>(n_reads_hv), <span class="at">p_reads_hv =</span> <span class="fu">sum</span>(p_reads_hv), </span>
<span id="cb18-18"><a href="#cb18-18" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb18-19"><a href="#cb18-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">factor</span>(name_display, </span>
<span id="cb18-20"><a href="#cb18-20" aria-hidden="true" tabindex="-1"></a>                               <span class="at">levels =</span> <span class="fu">c</span>(hv_family_major_tab<span class="sc">$</span>name, <span class="st">"Other"</span>)))</span>
<span id="cb18-21"><a href="#cb18-21" aria-hidden="true" tabindex="-1"></a>hv_family_counts_display <span class="ot">&lt;-</span> hv_family_counts_major <span class="sc">%&gt;%</span></span>
<span id="cb18-22"><a href="#cb18-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_hv, <span class="at">classification =</span> name_display) <span class="sc">%&gt;%</span></span>
<span id="cb18-23"><a href="#cb18-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by =</span> <span class="st">'sample'</span>)</span>
<span id="cb18-24"><a href="#cb18-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-25"><a href="#cb18-25" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb18-26"><a href="#cb18-26" aria-hidden="true" tabindex="-1"></a>g_hv_family <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb18-27"><a href="#cb18-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">data=</span>hv_family_counts_display, <span class="at">position =</span> <span class="st">"stack"</span>, <span class="at">width=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb18-28"><a href="#cb18-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% HV Reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb18-29"><a href="#cb18-29" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb18-30"><a href="#cb18-30" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb18-31"><a href="#cb18-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">'Accent'</span>, <span class="at">name =</span> <span class="st">"Viral family"</span>) <span class="sc">+</span></span>
<span id="cb18-32"><a href="#cb18-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">"Family composition of human-viral reads"</span>) <span class="sc">+</span></span>
<span id="cb18-33"><a href="#cb18-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">4</span>)) <span class="sc">+</span></span>
<span id="cb18-34"><a href="#cb18-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">1.4</span>), <span class="at">hjust=</span><span class="dv">0</span>, <span class="at">face=</span><span class="st">"plain"</span>))</span>
<span id="cb18-35"><a href="#cb18-35" aria-hidden="true" tabindex="-1"></a>g_hv_family</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="2024-09-12-thompson_files/figure-html/hv-family-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get most prominent families for text</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>hv_family_collate <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_tot =</span> <span class="fu">sum</span>(n_reads_hv),</span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>            <span class="at">p_reads_max =</span> <span class="fu">max</span>(p_reads_hv), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n_reads_tot))</span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>hv_family_collate</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["name"],"name":[1],"type":["chr"],"align":["left"]},{"label":["taxid"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["n_reads_tot"],"name":[3],"type":["int"],"align":["right"]},{"label":["p_reads_max"],"name":[4],"type":["dbl"],"align":["right"]}],"data":[{"1":"Flaviviridae","2":"11050","3":"25997","4":"1.0000000"},{"1":"Anelloviridae","2":"687329","3":"10103","4":"1.0000000"},{"1":"Coronaviridae","2":"11118","3":"1104","4":"1.0000000"},{"1":"Orthoherpesviridae","2":"3044472","3":"534","4":"1.0000000"},{"1":"Parvoviridae","2":"10780","3":"335","4":"0.8333333"},{"1":"Retroviridae","2":"11632","3":"224","4":"1.0000000"},{"1":"Papillomaviridae","2":"151340","3":"7","4":"0.1666667"},{"1":"Polyomaviridae","2":"151341","3":"1","4":"0.1666667"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
</section>
<section id="analyzing-specific-families" class="level2" data-number="4.3">
<h2 data-number="4.3" class="anchored" data-anchor-id="analyzing-specific-families"><span class="header-section-number">4.3</span> Analyzing specific families</h2>
<p>We now investigate the composition of specific families that had more than 5 viral reads. In investigating individual viral families, to avoid distortions from a few rare reads, I restricted myself to samples where that family made up at least 1% of human-viral reads:</p>
<section id="flaviviridae-number-of-reads-25997" class="level3" data-number="4.3.1">
<h3 data-number="4.3.1" class="anchored" data-anchor-id="flaviviridae-number-of-reads-25997"><span class="header-section-number">4.3.1</span> Flaviviridae (Number of reads: 25,997)</h3>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Flaviviridae</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_histogram</span>(<span class="at">taxid_chosen=</span><span class="dv">11050</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="2024-09-12-thompson_files/figure-html/plot-flaviviridae-histogram-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Flaviviridae</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_composition</span>(<span class="at">taxid_chosen=</span><span class="dv">11050</span>, <span class="at">threshold_major_species =</span> <span class="fl">0.01</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="2024-09-12-thompson_files/figure-html/plot-flaviviridae-composition-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="anelloviridae-number-of-reads-10103" class="level3" data-number="4.3.2">
<h3 data-number="4.3.2" class="anchored" data-anchor-id="anelloviridae-number-of-reads-10103"><span class="header-section-number">4.3.2</span> Anelloviridae (Number of reads: 10,103)</h3>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Anelloviridae</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_histogram</span>(<span class="at">taxid_chosen=</span><span class="dv">687329</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="2024-09-12-thompson_files/figure-html/plot-anelloviridae-histogram-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Anelloviridae</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_composition</span>(<span class="at">taxid_chosen=</span><span class="dv">687329</span>, <span class="at">threshold_major_species =</span> <span class="fl">0.01</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="2024-09-12-thompson_files/figure-html/plot-anelloviridae-composition-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="coronaviridae-number-of-reads-1104" class="level3" data-number="4.3.3">
<h3 data-number="4.3.3" class="anchored" data-anchor-id="coronaviridae-number-of-reads-1104"><span class="header-section-number">4.3.3</span> Coronaviridae (Number of reads: 1,104)</h3>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Coronaviridae</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_histogram</span>(<span class="at">taxid_chosen=</span><span class="dv">11118</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="2024-09-12-thompson_files/figure-html/plot-coronaviridae-histogram-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Coronaviridae</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_composition</span>(<span class="at">taxid_chosen=</span><span class="dv">11118</span>, <span class="at">threshold_major_species =</span> <span class="fl">0.01</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="2024-09-12-thompson_files/figure-html/plot-coronaviridae-composition-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="orthoherpesviridae-number-of-reads-534" class="level3" data-number="4.3.4">
<h3 data-number="4.3.4" class="anchored" data-anchor-id="orthoherpesviridae-number-of-reads-534"><span class="header-section-number">4.3.4</span> Orthoherpesviridae (Number of reads: 534)</h3>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Orthoherpesviridae</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_histogram</span>(<span class="at">taxid_chosen=</span><span class="dv">3044472</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="2024-09-12-thompson_files/figure-html/plot-orthoherpesviridae-histogram-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Orthoherpesviridae</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_composition</span>(<span class="at">taxid_chosen=</span><span class="dv">3044472</span>, <span class="at">threshold_major_species =</span> <span class="fl">0.01</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="2024-09-12-thompson_files/figure-html/plot-orthoherpesviridae-composition-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="parvoviridae-number-of-reads-335" class="level3" data-number="4.3.5">
<h3 data-number="4.3.5" class="anchored" data-anchor-id="parvoviridae-number-of-reads-335"><span class="header-section-number">4.3.5</span> Parvoviridae (Number of reads: 335)</h3>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Parvoviridae</span></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_histogram</span>(<span class="at">taxid_chosen=</span><span class="dv">10780</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="2024-09-12-thompson_files/figure-html/plot-parvoviridae-histogram-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Parvoviridae</span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_composition</span>(<span class="at">taxid_chosen=</span><span class="dv">10780</span>, <span class="at">threshold_major_species =</span> <span class="fl">0.01</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="2024-09-12-thompson_files/figure-html/plot-parvoviridae-composition-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="retroviridae-number-of-reads-224" class="level3" data-number="4.3.6">
<h3 data-number="4.3.6" class="anchored" data-anchor-id="retroviridae-number-of-reads-224"><span class="header-section-number">4.3.6</span> Retroviridae (Number of reads: 224)</h3>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Retroviridae</span></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_histogram</span>(<span class="at">taxid_chosen=</span><span class="dv">11632</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="2024-09-12-thompson_files/figure-html/plot-retroviridae-histogram-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Retroviridae</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_composition</span>(<span class="at">taxid_chosen=</span><span class="dv">11632</span>, <span class="at">threshold_major_species =</span> <span class="fl">0.01</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="2024-09-12-thompson_files/figure-html/plot-retroviridae-composition-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="relative-abundance-of-pathogenic-viruses-of-interest" class="level2" data-number="4.4">
<h2 data-number="4.4" class="anchored" data-anchor-id="relative-abundance-of-pathogenic-viruses-of-interest"><span class="header-section-number">4.4</span> Relative abundance of pathogenic viruses of interest</h2>
<p>Each dot represents a sample, colored by viral family. The x-axis shows the relative abundance of human-infecting viruses, and the y-axis shows the species.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>species_family <span class="ot">&lt;-</span> result <span class="sc">%&gt;%</span> <span class="fu">select</span>(species, family) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="st">'name'</span> <span class="ot">=</span> <span class="st">'species'</span>)</span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>play <span class="ot">&lt;-</span> hv_species_counts <span class="sc">%&gt;%</span> </span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by =</span> <span class="st">'sample'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(species_family, <span class="at">by =</span> <span class="st">'name'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">ifelse</span>(name <span class="sc">==</span> <span class="st">"Severe acute respiratory syndrome-related coronavirus"</span>, <span class="st">"SARS-related coronavirus"</span>, name))</span>
<span id="cb32-8"><a href="#cb32-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-9"><a href="#cb32-9" aria-hidden="true" tabindex="-1"></a>adjusted_play <span class="ot">&lt;-</span> play <span class="sc">%&gt;%</span> </span>
<span id="cb32-10"><a href="#cb32-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span></span>
<span id="cb32-11"><a href="#cb32-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">virus_prevalence_num =</span> <span class="fu">n_distinct</span>(sample)<span class="sc">/</span><span class="fu">n_distinct</span>(libraries),</span>
<span id="cb32-12"><a href="#cb32-12" aria-hidden="true" tabindex="-1"></a>         <span class="at">total_reads_hv =</span> <span class="fu">sum</span>(n_reads_hv)) <span class="sc">%&gt;%</span></span>
<span id="cb32-13"><a href="#cb32-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb32-14"><a href="#cb32-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">fct_reorder</span>(name, virus_prevalence_num, <span class="at">.desc=</span><span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb32-15"><a href="#cb32-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, ra_reads_hv, family, virus_prevalence_num, total_reads_hv, n_reads_hv)</span>
<span id="cb32-16"><a href="#cb32-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-17"><a href="#cb32-17" aria-hidden="true" tabindex="-1"></a>pal <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">'Dark2'</span>),<span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">'Accent'</span>))</span>
<span id="cb32-18"><a href="#cb32-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-19"><a href="#cb32-19" aria-hidden="true" tabindex="-1"></a><span class="co">#, labels = label_log(digits=2)</span></span>
<span id="cb32-20"><a href="#cb32-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-21"><a href="#cb32-21" aria-hidden="true" tabindex="-1"></a>ra_dot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(adjusted_play, <span class="fu">aes</span>(<span class="at">x =</span> ra_reads_hv, <span class="at">y=</span>name)) <span class="sc">+</span></span>
<span id="cb32-22"><a href="#cb32-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_quasirandom</span>(<span class="at">orientation =</span> <span class="st">'y'</span>, <span class="fu">aes</span>(<span class="at">color =</span> family)) <span class="sc">+</span></span>
<span id="cb32-23"><a href="#cb32-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> pal) <span class="sc">+</span> </span>
<span id="cb32-24"><a href="#cb32-24" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_log10</span>(</span>
<span id="cb32-25"><a href="#cb32-25" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Relative abundance human virus reads"</span>,</span>
<span id="cb32-26"><a href="#cb32-26" aria-hidden="true" tabindex="-1"></a>    <span class="at">labels=</span><span class="fu">label_log</span>(<span class="at">digits=</span><span class="dv">3</span>)</span>
<span id="cb32-27"><a href="#cb32-27" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb32-28"><a href="#cb32-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span>  <span class="st">""</span>,</span>
<span id="cb32-29"><a href="#cb32-29" aria-hidden="true" tabindex="-1"></a>       <span class="at">color =</span> <span class="st">'Viral family'</span>) <span class="sc">+</span> </span>
<span id="cb32-30"><a href="#cb32-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color =</span> <span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">2</span>)) <span class="sc">+</span></span>
<span id="cb32-31"><a href="#cb32-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_light</span>() <span class="sc">+</span> </span>
<span id="cb32-32"><a href="#cb32-32" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(</span>
<span id="cb32-33"><a href="#cb32-33" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb32-34"><a href="#cb32-34" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>),</span>
<span id="cb32-35"><a href="#cb32-35" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.ticks.y =</span> <span class="fu">element_blank</span>(),<span class="at">axis.line =</span> <span class="fu">element_line</span>(<span class="at">colour =</span> <span class="st">"black"</span>),</span>
<span id="cb32-36"><a href="#cb32-36" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">14</span>),    </span>
<span id="cb32-37"><a href="#cb32-37" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb32-38"><a href="#cb32-38" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>, <span class="at">face=</span><span class="st">"bold"</span>),</span>
<span id="cb32-39"><a href="#cb32-39" aria-hidden="true" tabindex="-1"></a>    <span class="co">#legend.position = 'bottom',</span></span>
<span id="cb32-40"><a href="#cb32-40" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>),  <span class="co"># Move legend to top right</span></span>
<span id="cb32-41"><a href="#cb32-41" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.justification =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>),  <span class="co"># Align legend to top right</span></span>
<span id="cb32-42"><a href="#cb32-42" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb32-43"><a href="#cb32-43" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.border =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb32-44"><a href="#cb32-44" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.background =</span> <span class="fu">element_blank</span>())</span>
<span id="cb32-45"><a href="#cb32-45" aria-hidden="true" tabindex="-1"></a>ra_dot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="2024-09-12-thompson_files/figure-html/plot-pathogenic-viruses-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<p>I can then take all of the viruses that we found and look up what they’re responsible for and whether they’re dangerous.</p>
<table class="caption-top table">
<colgroup>
<col style="width: 25%">
<col style="width: 27%">
<col style="width: 46%">
</colgroup>
<thead>
<tr class="header">
<th>Virus Name</th>
<th>Common Name</th>
<th>Pathogenic Potential</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Human immunodeficiency virus 1</td>
<td>HIV-1</td>
<td>Very high, causes AIDS if left untreated, leading to severe immunodeficiency</td>
</tr>
<tr class="even">
<td>SARS-related coronavirus</td>
<td>SARS-CoV</td>
<td>High, causes severe acute respiratory syndrome with potentially fatal outcomes</td>
</tr>
<tr class="odd">
<td>Hepacivirus hominis</td>
<td>Hepatitis C Virus</td>
<td>High, causes chronic liver disease and can lead to cirrhosis and liver cancer</td>
</tr>
<tr class="even">
<td>Cytomegalovirus humanbeta5</td>
<td>Human Cytomegalovirus</td>
<td>Moderate to high, can cause severe disease in immunocompromised individuals and congenital infections</td>
</tr>
<tr class="odd">
<td>Lymphocryptovirus humangamma4</td>
<td>Epstein-Barr Virus</td>
<td>Moderate, causes infectious mononucleosis and associated with certain cancers</td>
</tr>
<tr class="even">
<td>Erythroparvovirus primate1</td>
<td>Human Parvovirus B19</td>
<td>Moderate, causes fifth disease and can be serious in certain populations (pregnant women, immunocompromised)</td>
</tr>
<tr class="odd">
<td>Roseolovirus humanbeta6a</td>
<td>Human Herpesvirus 6A</td>
<td>Low to moderate, can cause roseola in children and occasionally more severe complications</td>
</tr>
<tr class="even">
<td>Roseolovirus humanbeta6b</td>
<td>Human Herpesvirus 6B</td>
<td>Low to moderate, can cause roseola in children and occasionally more severe complications</td>
</tr>
<tr class="odd">
<td>Alphapapillomavirus 2</td>
<td>Human Papillomavirus 2</td>
<td>Low to moderate, can cause common warts and rarely progress to cancer</td>
</tr>
<tr class="even">
<td>Betapapillomavirus 5</td>
<td>Human Papillomavirus 5</td>
<td>Low to moderate, associated with skin lesions in immunosuppressed individuals</td>
</tr>
<tr class="odd">
<td>Betapapillomavirus 2</td>
<td>Human Papillomavirus 9</td>
<td>Low to moderate, associated with cutaneous lesions and rarely with non-melanoma skin cancer</td>
</tr>
<tr class="even">
<td>Primate T-lymphotropic virus 2</td>
<td>Human T-cell Lymphotropic Virus 2</td>
<td>Low, rarely associated with neurological disorders</td>
</tr>
<tr class="odd">
<td>Nupapillomavirus 1</td>
<td>Human Papillomavirus 41</td>
<td>Low, can cause benign cutaneous lesions</td>
</tr>
<tr class="even">
<td>Alphapolyomavirus quintihominis</td>
<td>Human Polyomavirus 5</td>
<td>Low, not known to cause significant disease in healthy individuals</td>
</tr>
<tr class="odd">
<td>Pegivirus hominis</td>
<td>Human Pegivirus</td>
<td>Very low, not known to cause disease in humans</td>
</tr>
<tr class="even">
<td>Pegivirus columbiaense</td>
<td>Pegivirus C</td>
<td>Very low, not known to cause disease in humans</td>
</tr>
</tbody>
</table>
<p>This was interesting to see. This is the first study where we’ve seen signficant numbers of human-infecting viruses in whole blood.</p>
</section>
<section id="relative-abundance-assuming-perfect-human-read-removal" class="level2" data-number="4.5">
<h2 data-number="4.5" class="anchored" data-anchor-id="relative-abundance-assuming-perfect-human-read-removal"><span class="header-section-number">4.5</span> Relative abundance assuming perfect human read removal</h2>
<p>Assuming we’re able to perfectly remove all human reads, the average relative abundance of known human infecting virus is <span class="math inline">\(1.91 \times 10^{-5}\)</span>.</p>
</section>
</section>
<section id="conclusion" class="level1" data-number="5">
<h1 data-number="5"><span class="header-section-number">5</span> Conclusion</h1>
<p>There were some interesting takeways from this analysis:</p>
<ol type="1">
<li>Whole blood metagenomics was able to detect HIV-1, SARS-CoV-2, HCV, CMV, EBV, HTLV-2, and other herpesviruses. This is important as some of these are latent viruses.</li>
<li>Far fewer individuals tend to have detectable levels of human infecting viruses, when compared to the plasma datasets.</li>
<li>Important viruses can be detected from RNA-seq data, even if the study wasn’t explictly designed to look at them.</li>
</ol>
</section>
<section id="appendix" class="level1" data-number="6">
<h1 data-number="6"><span class="header-section-number">6</span> Appendix</h1>
<section id="human-infecting-virus-families-genera-and-species" class="level2" data-number="6.1">
<h2 data-number="6.1" class="anchored" data-anchor-id="human-infecting-virus-families-genera-and-species"><span class="header-section-number">6.1</span> Human-infecting virus families, genera, and species</h2>
<p>To get a good overview of families, genera, and species, we can look at a Sankey plot where the magnitude of relative abundance, averaged over all samples, is shown in parentheses.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to create links</span></span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>create_links <span class="ot">&lt;-</span> <span class="cf">function</span>(data) {</span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>  family_to_genus <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(genus)) <span class="sc">%&gt;%</span></span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(family, genus) <span class="sc">%&gt;%</span></span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">value =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb33-7"><a href="#cb33-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">source =</span> family, <span class="at">target =</span> genus)</span>
<span id="cb33-8"><a href="#cb33-8" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb33-9"><a href="#cb33-9" aria-hidden="true" tabindex="-1"></a>  genus_to_species <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb33-10"><a href="#cb33-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(genus, species) <span class="sc">%&gt;%</span></span>
<span id="cb33-11"><a href="#cb33-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">value =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb33-12"><a href="#cb33-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">source =</span> genus, <span class="at">target =</span> species)</span>
<span id="cb33-13"><a href="#cb33-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-14"><a href="#cb33-14" aria-hidden="true" tabindex="-1"></a>  family_to_species <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb33-15"><a href="#cb33-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="fu">is.na</span>(genus)) <span class="sc">%&gt;%</span></span>
<span id="cb33-16"><a href="#cb33-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(family, species) <span class="sc">%&gt;%</span></span>
<span id="cb33-17"><a href="#cb33-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">value =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb33-18"><a href="#cb33-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">source =</span> family, <span class="at">target =</span> species)</span>
<span id="cb33-19"><a href="#cb33-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-20"><a href="#cb33-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_rows</span>(family_to_genus, genus_to_species, family_to_species) <span class="sc">%&gt;%</span></span>
<span id="cb33-21"><a href="#cb33-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(source))</span>
<span id="cb33-22"><a href="#cb33-22" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb33-23"><a href="#cb33-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-24"><a href="#cb33-24" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to create nodes</span></span>
<span id="cb33-25"><a href="#cb33-25" aria-hidden="true" tabindex="-1"></a>create_nodes <span class="ot">&lt;-</span> <span class="cf">function</span>(links) {</span>
<span id="cb33-26"><a href="#cb33-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>(</span>
<span id="cb33-27"><a href="#cb33-27" aria-hidden="true" tabindex="-1"></a>    <span class="at">name =</span> <span class="fu">c</span>(links<span class="sc">$</span>source, links<span class="sc">$</span>target) <span class="sc">%&gt;%</span> <span class="fu">unique</span>()</span>
<span id="cb33-28"><a href="#cb33-28" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb33-29"><a href="#cb33-29" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb33-30"><a href="#cb33-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-31"><a href="#cb33-31" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to prepare data for Sankey diagram</span></span>
<span id="cb33-32"><a href="#cb33-32" aria-hidden="true" tabindex="-1"></a>prepare_sankey_data <span class="ot">&lt;-</span> <span class="cf">function</span>(links, nodes) {</span>
<span id="cb33-33"><a href="#cb33-33" aria-hidden="true" tabindex="-1"></a>  links<span class="sc">$</span>IDsource <span class="ot">&lt;-</span> <span class="fu">match</span>(links<span class="sc">$</span>source, nodes<span class="sc">$</span>name) <span class="sc">-</span> <span class="dv">1</span></span>
<span id="cb33-34"><a href="#cb33-34" aria-hidden="true" tabindex="-1"></a>  links<span class="sc">$</span>IDtarget <span class="ot">&lt;-</span> <span class="fu">match</span>(links<span class="sc">$</span>target, nodes<span class="sc">$</span>name) <span class="sc">-</span> <span class="dv">1</span></span>
<span id="cb33-35"><a href="#cb33-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list</span>(<span class="at">links =</span> links, <span class="at">nodes =</span> nodes)</span>
<span id="cb33-36"><a href="#cb33-36" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb33-37"><a href="#cb33-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-38"><a href="#cb33-38" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to create Sankey plot</span></span>
<span id="cb33-39"><a href="#cb33-39" aria-hidden="true" tabindex="-1"></a>create_sankey_plot <span class="ot">&lt;-</span> <span class="cf">function</span>(sankey_data) {</span>
<span id="cb33-40"><a href="#cb33-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sankeyNetwork</span>(</span>
<span id="cb33-41"><a href="#cb33-41" aria-hidden="true" tabindex="-1"></a>    <span class="at">Links =</span> sankey_data<span class="sc">$</span>links, </span>
<span id="cb33-42"><a href="#cb33-42" aria-hidden="true" tabindex="-1"></a>    <span class="at">Nodes =</span> sankey_data<span class="sc">$</span>nodes,</span>
<span id="cb33-43"><a href="#cb33-43" aria-hidden="true" tabindex="-1"></a>    <span class="at">Source =</span> <span class="st">"IDsource"</span>, </span>
<span id="cb33-44"><a href="#cb33-44" aria-hidden="true" tabindex="-1"></a>    <span class="at">Target =</span> <span class="st">"IDtarget"</span>,</span>
<span id="cb33-45"><a href="#cb33-45" aria-hidden="true" tabindex="-1"></a>    <span class="at">Value =</span> <span class="st">"value"</span>, </span>
<span id="cb33-46"><a href="#cb33-46" aria-hidden="true" tabindex="-1"></a>    <span class="at">NodeID =</span> <span class="st">"name"</span>,</span>
<span id="cb33-47"><a href="#cb33-47" aria-hidden="true" tabindex="-1"></a>    <span class="at">sinksRight =</span> <span class="cn">TRUE</span>,</span>
<span id="cb33-48"><a href="#cb33-48" aria-hidden="true" tabindex="-1"></a>    <span class="at">nodeWidth =</span> <span class="dv">25</span>,</span>
<span id="cb33-49"><a href="#cb33-49" aria-hidden="true" tabindex="-1"></a>    <span class="at">fontSize =</span> <span class="dv">14</span>,</span>
<span id="cb33-50"><a href="#cb33-50" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb33-51"><a href="#cb33-51" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb33-52"><a href="#cb33-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-53"><a href="#cb33-53" aria-hidden="true" tabindex="-1"></a>save_sankey_as_png <span class="ot">&lt;-</span> <span class="cf">function</span>(sankey_plot, <span class="at">width =</span> <span class="dv">1000</span>, <span class="at">height =</span> <span class="dv">800</span>) {</span>
<span id="cb33-54"><a href="#cb33-54" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Save the plot as an HTML file</span></span>
<span id="cb33-55"><a href="#cb33-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">saveWidget</span>(sankey_plot, <span class="fu">sprintf</span>(<span class="st">'%s/sankey.html'</span>,data_dir))</span>
<span id="cb33-56"><a href="#cb33-56" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb33-57"><a href="#cb33-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-58"><a href="#cb33-58" aria-hidden="true" tabindex="-1"></a>format_scientific <span class="ot">&lt;-</span> <span class="cf">function</span>(x, <span class="at">digits=</span><span class="dv">2</span>) {</span>
<span id="cb33-59"><a href="#cb33-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sapply</span>(x, <span class="cf">function</span>(val) {</span>
<span id="cb33-60"><a href="#cb33-60" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">is.na</span>(val) <span class="sc">||</span> <span class="fu">abs</span>(val) <span class="sc">&lt;</span> <span class="fl">1e-15</span>) {</span>
<span id="cb33-61"><a href="#cb33-61" aria-hidden="true" tabindex="-1"></a>      <span class="fu">return</span>(<span class="st">"0"</span>)</span>
<span id="cb33-62"><a href="#cb33-62" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb33-63"><a href="#cb33-63" aria-hidden="true" tabindex="-1"></a>      exponent <span class="ot">&lt;-</span> <span class="fu">floor</span>(<span class="fu">log10</span>(<span class="fu">abs</span>(val)))</span>
<span id="cb33-64"><a href="#cb33-64" aria-hidden="true" tabindex="-1"></a>      coef <span class="ot">&lt;-</span> val <span class="sc">/</span> <span class="dv">10</span><span class="sc">^</span>exponent</span>
<span id="cb33-65"><a href="#cb33-65" aria-hidden="true" tabindex="-1"></a>      <span class="co">#return(sprintf("%.1f × 10^%d", round(coef, digits), exponent))</span></span>
<span id="cb33-66"><a href="#cb33-66" aria-hidden="true" tabindex="-1"></a>      <span class="co"># May or may not be smart, just keeping magnitude</span></span>
<span id="cb33-67"><a href="#cb33-67" aria-hidden="true" tabindex="-1"></a>      <span class="fu">return</span>(<span class="fu">sprintf</span>(<span class="st">"10^%d"</span>, exponent))</span>
<span id="cb33-68"><a href="#cb33-68" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb33-69"><a href="#cb33-69" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb33-70"><a href="#cb33-70" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb33-71"><a href="#cb33-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-72"><a href="#cb33-72" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> result <span class="sc">%&gt;%</span> </span>
<span id="cb33-73"><a href="#cb33-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">c</span>(genus_n_reads_tot, genus_ra_reads_tot), <span class="sc">~</span><span class="fu">replace_na</span>(., <span class="dv">0</span>)),</span>
<span id="cb33-74"><a href="#cb33-74" aria-hidden="true" tabindex="-1"></a>         <span class="at">genus =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(genus), <span class="st">"Unknown Genus"</span>, genus)) <span class="sc">%&gt;%</span></span>
<span id="cb33-75"><a href="#cb33-75" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb33-76"><a href="#cb33-76" aria-hidden="true" tabindex="-1"></a>  <span class="at">species =</span> <span class="fu">paste0</span>(species, <span class="fu">sprintf</span>(<span class="st">' (%s)'</span>, <span class="fu">format_scientific</span>(species_ra_reads_tot))),</span>
<span id="cb33-77"><a href="#cb33-77" aria-hidden="true" tabindex="-1"></a>  <span class="at">genus =</span> <span class="fu">paste0</span>(genus, <span class="fu">sprintf</span>(<span class="st">' (%s)'</span>, <span class="fu">format_scientific</span>(genus_ra_reads_tot))),</span>
<span id="cb33-78"><a href="#cb33-78" aria-hidden="true" tabindex="-1"></a>  <span class="at">family =</span> <span class="fu">paste0</span>(family, <span class="fu">sprintf</span>(<span class="st">' (%s)'</span>, <span class="fu">format_scientific</span>(family_ra_reads_tot)))</span>
<span id="cb33-79"><a href="#cb33-79" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb33-80"><a href="#cb33-80" aria-hidden="true" tabindex="-1"></a>links <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">create_links</span>(data))</span>
<span id="cb33-81"><a href="#cb33-81" aria-hidden="true" tabindex="-1"></a>nodes <span class="ot">&lt;-</span> <span class="fu">create_nodes</span>(links)</span>
<span id="cb33-82"><a href="#cb33-82" aria-hidden="true" tabindex="-1"></a>sankey_data <span class="ot">&lt;-</span> <span class="fu">prepare_sankey_data</span>(links, nodes)</span>
<span id="cb33-83"><a href="#cb33-83" aria-hidden="true" tabindex="-1"></a>sankey <span class="ot">&lt;-</span> <span class="fu">create_sankey_plot</span>(sankey_data)</span>
<span id="cb33-84"><a href="#cb33-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-85"><a href="#cb33-85" aria-hidden="true" tabindex="-1"></a>sankey</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="htmlwidget-d3865a9d338ef0382718" style="width:100%;height:1462px;" class="sankeyNetwork html-widget"></div>
<script type="application/json" data-for="htmlwidget-d3865a9d338ef0382718">{"x":{"links":{"source":[0,0,0,0,1,2,2,3,3,3,4,4,4,5,6,7,7,8,9,10,10,10,10,10,10,10,10,10,10,10,10,10,10,11,12,12,13,13,13,14,15,16,17,18,19,20,21,22,22,23,23,24,24,24],"target":[10,13,17,24,11,18,22,14,20,23,8,12,21,16,9,15,19,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61],"value":[14,3,1,3,1,1,2,1,1,2,1,2,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1]},"nodes":{"name":["Anelloviridae (10^-6)","Coronaviridae (10^-7)","Flaviviridae (10^-5)","Orthoherpesviridae (10^-7)","Papillomaviridae (10^-8)","Parvoviridae (10^-6)","Polyomaviridae (10^-8)","Retroviridae (10^-7)","Alphapapillomavirus (10^-8)","Alphapolyomavirus (10^-8)","Alphatorquevirus (10^-6)","Betacoronavirus (10^-7)","Betapapillomavirus (10^-8)","Betatorquevirus (10^-8)","Cytomegalovirus (10^-7)","Deltaretrovirus (10^-8)","Erythroparvovirus (10^-6)","Gammatorquevirus (10^-8)","Hepacivirus (10^-5)","Lentivirus (10^-7)","Lymphocryptovirus (10^-7)","Nupapillomavirus (10^-8)","Pegivirus (10^-5)","Roseolovirus (10^-7)","Unknown Genus (0)","Alphapapillomavirus 2 (10^-8)","Alphapolyomavirus quintihominis (10^-8)","Alphatorquevirus homin1 (10^-8)","Alphatorquevirus homin13 (10^-8)","Alphatorquevirus homin15 (10^-8)","Alphatorquevirus homin17 (10^-7)","Alphatorquevirus homin18 (10^-7)","Alphatorquevirus homin24 (10^-6)","Alphatorquevirus homin29 (10^-7)","Alphatorquevirus homin3 (10^-6)","Alphatorquevirus homin5 (10^-8)","Alphatorquevirus homin6 (10^-7)","Torque teno virus 11 (10^-8)","Torque teno virus 16 (10^-8)","Torque teno virus 22 (10^-7)","Torque teno virus 28 (10^-8)","Severe acute respiratory syndrome-related coronavirus (10^-7)","Betapapillomavirus 2 (10^-8)","Betapapillomavirus 5 (10^-8)","Betatorquevirus homini1 (10^-8)","Betatorquevirus homini7 (10^-8)","TTV-like mini virus (10^-8)","Cytomegalovirus humanbeta5 (10^-7)","Primate T-lymphotropic virus 2 (10^-8)","Erythroparvovirus primate1 (10^-6)","Gammatorquevirus homidi8 (10^-8)","Hepacivirus hominis (10^-5)","Human immunodeficiency virus 1 (10^-7)","Lymphocryptovirus humangamma4 (10^-7)","Nupapillomavirus 1 (10^-8)","Pegivirus columbiaense (10^-7)","Pegivirus hominis (10^-5)","Roseolovirus humanbeta6a (10^-7)","Roseolovirus humanbeta6b (10^-7)","Anelloviridae sp. (10^-6)","Micro Torque teno virus (10^-8)","Torque teno virus (10^-7)"],"group":["Anelloviridae (10^-6)","Coronaviridae (10^-7)","Flaviviridae (10^-5)","Orthoherpesviridae (10^-7)","Papillomaviridae (10^-8)","Parvoviridae (10^-6)","Polyomaviridae (10^-8)","Retroviridae (10^-7)","Alphapapillomavirus (10^-8)","Alphapolyomavirus (10^-8)","Alphatorquevirus (10^-6)","Betacoronavirus (10^-7)","Betapapillomavirus (10^-8)","Betatorquevirus (10^-8)","Cytomegalovirus (10^-7)","Deltaretrovirus (10^-8)","Erythroparvovirus (10^-6)","Gammatorquevirus (10^-8)","Hepacivirus (10^-5)","Lentivirus (10^-7)","Lymphocryptovirus (10^-7)","Nupapillomavirus (10^-8)","Pegivirus (10^-5)","Roseolovirus (10^-7)","Unknown Genus (0)","Alphapapillomavirus 2 (10^-8)","Alphapolyomavirus quintihominis (10^-8)","Alphatorquevirus homin1 (10^-8)","Alphatorquevirus homin13 (10^-8)","Alphatorquevirus homin15 (10^-8)","Alphatorquevirus homin17 (10^-7)","Alphatorquevirus homin18 (10^-7)","Alphatorquevirus homin24 (10^-6)","Alphatorquevirus homin29 (10^-7)","Alphatorquevirus homin3 (10^-6)","Alphatorquevirus homin5 (10^-8)","Alphatorquevirus homin6 (10^-7)","Torque teno virus 11 (10^-8)","Torque teno virus 16 (10^-8)","Torque teno virus 22 (10^-7)","Torque teno virus 28 (10^-8)","Severe acute respiratory syndrome-related coronavirus (10^-7)","Betapapillomavirus 2 (10^-8)","Betapapillomavirus 5 (10^-8)","Betatorquevirus homini1 (10^-8)","Betatorquevirus homini7 (10^-8)","TTV-like mini virus (10^-8)","Cytomegalovirus humanbeta5 (10^-7)","Primate T-lymphotropic virus 2 (10^-8)","Erythroparvovirus primate1 (10^-6)","Gammatorquevirus homidi8 (10^-8)","Hepacivirus hominis (10^-5)","Human immunodeficiency virus 1 (10^-7)","Lymphocryptovirus humangamma4 (10^-7)","Nupapillomavirus 1 (10^-8)","Pegivirus columbiaense (10^-7)","Pegivirus hominis (10^-5)","Roseolovirus humanbeta6a (10^-7)","Roseolovirus humanbeta6b (10^-7)","Anelloviridae sp. (10^-6)","Micro Torque teno virus (10^-8)","Torque teno virus (10^-7)"]},"options":{"NodeID":"name","NodeGroup":"name","LinkGroup":null,"colourScale":"d3.scaleOrdinal(d3.schemeCategory20);","fontSize":14,"fontFamily":null,"nodeWidth":25,"nodePadding":10,"units":"","margin":{"top":null,"right":null,"bottom":null,"left":null},"iterations":32,"sinksRight":true}},"evals":[],"jsHooks":[]}</script>
</div>
</div>


<!-- -->

</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb34" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "Workflow of Thompson et al. (2023)"</span></span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a><span class="an">subtitle:</span><span class="co"> "Individual whole blood from US (RNA)"</span></span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> "Harmon Bhasin"</span></span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> 2024-09-12</span></span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span></span>
<span id="cb34-7"><a href="#cb34-7" aria-hidden="true" tabindex="-1"></a><span class="co">  html:</span></span>
<span id="cb34-8"><a href="#cb34-8" aria-hidden="true" tabindex="-1"></a><span class="co">    toc: true # table of contents</span></span>
<span id="cb34-9"><a href="#cb34-9" aria-hidden="true" tabindex="-1"></a><span class="co">    toc-title: "Table of contents" # table of contents title</span></span>
<span id="cb34-10"><a href="#cb34-10" aria-hidden="true" tabindex="-1"></a><span class="co">    number-sections: true # number sections</span></span>
<span id="cb34-11"><a href="#cb34-11" aria-hidden="true" tabindex="-1"></a><span class="co">    number-depth: 3 # number depth to show in table of contents</span></span>
<span id="cb34-12"><a href="#cb34-12" aria-hidden="true" tabindex="-1"></a><span class="co">    toc-location: right # table of contents location</span></span>
<span id="cb34-13"><a href="#cb34-13" aria-hidden="true" tabindex="-1"></a><span class="co">    page-layout: full # full page layout</span></span>
<span id="cb34-14"><a href="#cb34-14" aria-hidden="true" tabindex="-1"></a><span class="co">    code-fold: true # Keep option to fold code (i.e. show or hide it; default: hide)</span></span>
<span id="cb34-15"><a href="#cb34-15" aria-hidden="true" tabindex="-1"></a><span class="co">    code-tools: true # Code menu in the header of your document that provides various tools for readers to interact with the source code</span></span>
<span id="cb34-16"><a href="#cb34-16" aria-hidden="true" tabindex="-1"></a><span class="co">    code-link: true # Enables hyper-linking of functions within code blocks to their online documentation</span></span>
<span id="cb34-17"><a href="#cb34-17" aria-hidden="true" tabindex="-1"></a><span class="co">    df-print: paged # print data frame</span></span>
<span id="cb34-18"><a href="#cb34-18" aria-hidden="true" tabindex="-1"></a><span class="co">    fig-format: svg</span></span>
<span id="cb34-19"><a href="#cb34-19" aria-hidden="true" tabindex="-1"></a><span class="co">    other-links:</span></span>
<span id="cb34-20"><a href="#cb34-20" aria-hidden="true" tabindex="-1"></a><span class="co">      - text: Paper</span></span>
<span id="cb34-21"><a href="#cb34-21" aria-hidden="true" tabindex="-1"></a><span class="co">        href: https://doi.org/10.1038/s41591-022-02107-4</span></span>
<span id="cb34-22"><a href="#cb34-22" aria-hidden="true" tabindex="-1"></a><span class="co">      - text: Data</span></span>
<span id="cb34-23"><a href="#cb34-23" aria-hidden="true" tabindex="-1"></a><span class="co">        href: https://www.ncbi.nlm.nih.gov/bioproject/PRJNA891054</span></span>
<span id="cb34-24"><a href="#cb34-24" aria-hidden="true" tabindex="-1"></a><span class="co">    code-links:</span></span>
<span id="cb34-25"><a href="#cb34-25" aria-hidden="true" tabindex="-1"></a><span class="co">      - text: Code for this post</span></span>
<span id="cb34-26"><a href="#cb34-26" aria-hidden="true" tabindex="-1"></a><span class="co">        icon: file-code</span></span>
<span id="cb34-27"><a href="#cb34-27" aria-hidden="true" tabindex="-1"></a><span class="co">        href: https://github.com/naobservatory/harmons-public-notebook/blob/main/notebooks/2024-09-12-thompson.qmd</span></span>
<span id="cb34-28"><a href="#cb34-28" aria-hidden="true" tabindex="-1"></a><span class="an">editor:</span><span class="co"> </span></span>
<span id="cb34-29"><a href="#cb34-29" aria-hidden="true" tabindex="-1"></a><span class="co">  visual: true</span></span>
<span id="cb34-30"><a href="#cb34-30" aria-hidden="true" tabindex="-1"></a><span class="co">  render-on-save: true</span></span>
<span id="cb34-31"><a href="#cb34-31" aria-hidden="true" tabindex="-1"></a><span class="an">comments:</span></span>
<span id="cb34-32"><a href="#cb34-32" aria-hidden="true" tabindex="-1"></a><span class="co">  hypothesis: true # hypothesis</span></span>
<span id="cb34-33"><a href="#cb34-33" aria-hidden="true" tabindex="-1"></a><span class="an">execute:</span><span class="co"> </span></span>
<span id="cb34-34"><a href="#cb34-34" aria-hidden="true" tabindex="-1"></a><span class="co">  freeze: auto</span></span>
<span id="cb34-35"><a href="#cb34-35" aria-hidden="true" tabindex="-1"></a><span class="co">  cache: true</span></span>
<span id="cb34-36"><a href="#cb34-36" aria-hidden="true" tabindex="-1"></a><span class="an">title-block-banner:</span><span class="co"> "#de2d26"</span></span>
<span id="cb34-37"><a href="#cb34-37" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb34-38"><a href="#cb34-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-41"><a href="#cb34-41" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-42"><a href="#cb34-42" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: load-packages</span></span>
<span id="cb34-43"><a href="#cb34-43" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb34-44"><a href="#cb34-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-45"><a href="#cb34-45" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pacman)</span>
<span id="cb34-46"><a href="#cb34-46" aria-hidden="true" tabindex="-1"></a>pacman<span class="sc">::</span><span class="fu">p_load</span>(tidyverse, cowplot, patchwork, fastqcr, RColorBrewer, networkD3, readxl, htmlwidgets, ggbeeswarm, latex2exp, languageserver, scales, ggpubr)</span>
<span id="cb34-47"><a href="#cb34-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-48"><a href="#cb34-48" aria-hidden="true" tabindex="-1"></a>head_dir <span class="ot">&lt;-</span> <span class="st">"/Users/harmonbhasin/work/securebio/"</span></span>
<span id="cb34-49"><a href="#cb34-49" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="fu">sprintf</span>(<span class="st">"%s/sampling-strategies/scripts/aux_plot-theme.R"</span>, head_dir))</span>
<span id="cb34-50"><a href="#cb34-50" aria-hidden="true" tabindex="-1"></a>image_dir <span class="ot">&lt;-</span> <span class="fu">sprintf</span>(<span class="st">'%s/lab_meeting/thijssen'</span>, head_dir)</span>
<span id="cb34-51"><a href="#cb34-51" aria-hidden="true" tabindex="-1"></a>theme_base <span class="ot">&lt;-</span> theme_base <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio =</span> <span class="cn">NULL</span>)</span>
<span id="cb34-52"><a href="#cb34-52" aria-hidden="true" tabindex="-1"></a>theme_kit <span class="ot">&lt;-</span> theme_base <span class="sc">+</span> <span class="fu">theme</span>(</span>
<span id="cb34-53"><a href="#cb34-53" aria-hidden="true" tabindex="-1"></a>  <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">angle =</span> <span class="dv">45</span>),</span>
<span id="cb34-54"><a href="#cb34-54" aria-hidden="true" tabindex="-1"></a>  <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb34-55"><a href="#cb34-55" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb34-56"><a href="#cb34-56" aria-hidden="true" tabindex="-1"></a>tnl <span class="ot">&lt;-</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb34-57"><a href="#cb34-57" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-58"><a href="#cb34-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-61"><a href="#cb34-61" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-62"><a href="#cb34-62" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: set-up-paths</span></span>
<span id="cb34-63"><a href="#cb34-63" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb34-64"><a href="#cb34-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-65"><a href="#cb34-65" aria-hidden="true" tabindex="-1"></a><span class="co"># Data input paths</span></span>
<span id="cb34-66"><a href="#cb34-66" aria-hidden="true" tabindex="-1"></a>data_dir <span class="ot">&lt;-</span> <span class="st">"/Users/harmonbhasin/work/securebio/nao-harmon/thompson2023/analysis/"</span></span>
<span id="cb34-67"><a href="#cb34-67" aria-hidden="true" tabindex="-1"></a>input_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"data/input"</span>)</span>
<span id="cb34-68"><a href="#cb34-68" aria-hidden="true" tabindex="-1"></a>results_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"data/results"</span>)</span>
<span id="cb34-69"><a href="#cb34-69" aria-hidden="true" tabindex="-1"></a>qc_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(results_dir, <span class="st">"qc"</span>)</span>
<span id="cb34-70"><a href="#cb34-70" aria-hidden="true" tabindex="-1"></a>hv_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(results_dir, <span class="st">"hv"</span>)</span>
<span id="cb34-71"><a href="#cb34-71" aria-hidden="true" tabindex="-1"></a>libraries_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(input_dir, <span class="st">"libraries.csv"</span>)</span>
<span id="cb34-72"><a href="#cb34-72" aria-hidden="true" tabindex="-1"></a>basic_stats_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(qc_dir, <span class="st">"qc_basic_stats.tsv.gz"</span>)</span>
<span id="cb34-73"><a href="#cb34-73" aria-hidden="true" tabindex="-1"></a>adapter_stats_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(qc_dir, <span class="st">"qc_adapter_stats.tsv.gz"</span>)</span>
<span id="cb34-74"><a href="#cb34-74" aria-hidden="true" tabindex="-1"></a>quality_base_stats_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(qc_dir, <span class="st">"qc_quality_base_stats.tsv.gz"</span>)</span>
<span id="cb34-75"><a href="#cb34-75" aria-hidden="true" tabindex="-1"></a>quality_seq_stats_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(qc_dir, <span class="st">"qc_quality_sequence_stats.tsv.gz"</span>)</span>
<span id="cb34-76"><a href="#cb34-76" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-77"><a href="#cb34-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-78"><a href="#cb34-78" aria-hidden="true" tabindex="-1"></a>This is another potential study of <span class="co">[</span><span class="ot">this series</span><span class="co">](https://data.securebio.org/harmons-public-notebook/notebooks/2024-07-08_cebria-mendoza.html)</span>. In this post, I analyze <span class="co">[</span><span class="ot">Thompson 2023</span><span class="co">](https://doi.org/10.1038/s41591-022-02107-4)</span>, a dataset with 417 samples of whole blood from healthy and SARS-CoV-2 infected individuals in the US.</span>
<span id="cb34-79"><a href="#cb34-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-80"><a href="#cb34-80" aria-hidden="true" tabindex="-1"></a>*I'd like to thank Lenni for giving me feedback on the notebook and Will for providing me with a boilerplate rmarkdown file. This notebook uses the MGS Workflow v2.2.1 (note that this is old).*</span>
<span id="cb34-81"><a href="#cb34-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-82"><a href="#cb34-82" aria-hidden="true" tabindex="-1"></a><span class="fu"># The raw data</span></span>
<span id="cb34-83"><a href="#cb34-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-84"><a href="#cb34-84" aria-hidden="true" tabindex="-1"></a><span class="fu">## About</span></span>
<span id="cb34-85"><a href="#cb34-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-86"><a href="#cb34-86" aria-hidden="true" tabindex="-1"></a><span class="co">[</span><span class="ot">This dataset</span><span class="co">](https://www.ncbi.nlm.nih.gov/bioproject/PRJNA891054)</span> is composed of 417 samples of whole blood from the US. In total, there were 353 SARS-CoV-2 infected individuals and 64 healthy individuals that contributed to this study. For each sample, they performed RNA-sequencing, with Illumina NovaSeq 6000, producing 2x100 bp reads.</span>
<span id="cb34-87"><a href="#cb34-87" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-90"><a href="#cb34-90" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-91"><a href="#cb34-91" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb34-92"><a href="#cb34-92" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: load-libraries</span></span>
<span id="cb34-93"><a href="#cb34-93" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-94"><a href="#cb34-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-95"><a href="#cb34-95" aria-hidden="true" tabindex="-1"></a>low_sample_number <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">'SRR21924256'</span>)</span>
<span id="cb34-96"><a href="#cb34-96" aria-hidden="true" tabindex="-1"></a><span class="co"># Import libraries and extract metadata from sample names</span></span>
<span id="cb34-97"><a href="#cb34-97" aria-hidden="true" tabindex="-1"></a>libraries <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(libraries_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span>sample <span class="sc">%in%</span> low_sample_number)</span>
<span id="cb34-98"><a href="#cb34-98" aria-hidden="true" tabindex="-1"></a><span class="co">#meta_data &lt;- read_csv('/Users/harmonbhasin/work/securebio/nao-harmon/thijssen2023/preprocessing/SraRunTable.txt') %&gt;%</span></span>
<span id="cb34-99"><a href="#cb34-99" aria-hidden="true" tabindex="-1"></a><span class="co">#  select('Library Name', Run) %&gt;% rename(sample = Run, library = 'Library Name')</span></span>
<span id="cb34-100"><a href="#cb34-100" aria-hidden="true" tabindex="-1"></a><span class="co">#libraries &lt;- libraries %&gt;%</span></span>
<span id="cb34-101"><a href="#cb34-101" aria-hidden="true" tabindex="-1"></a><span class="co">#  left_join(meta_data, by = 'sample') %&gt;%</span></span>
<span id="cb34-102"><a href="#cb34-102" aria-hidden="true" tabindex="-1"></a><span class="co">#  select(sample, library=library.y) %&gt;% </span></span>
<span id="cb34-103"><a href="#cb34-103" aria-hidden="true" tabindex="-1"></a><span class="co">#  filter(library != 'PCL21') %&gt;%</span></span>
<span id="cb34-104"><a href="#cb34-104" aria-hidden="true" tabindex="-1"></a><span class="co">#  mutate(library = factor(library, levels = c(paste0('PCL', 1:20))))</span></span>
<span id="cb34-105"><a href="#cb34-105" aria-hidden="true" tabindex="-1"></a><span class="co">#libraries</span></span>
<span id="cb34-106"><a href="#cb34-106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-107"><a href="#cb34-107" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-108"><a href="#cb34-108" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-109"><a href="#cb34-109" aria-hidden="true" tabindex="-1"></a><span class="fu">### Sample + library preparation </span></span>
<span id="cb34-110"><a href="#cb34-110" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-111"><a href="#cb34-111" aria-hidden="true" tabindex="-1"></a>This is paraphrased from the paper:</span>
<span id="cb34-112"><a href="#cb34-112" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-113"><a href="#cb34-113" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; Patients were seen between April and June 2020 for COVID-19. The whole blood used for RNA-seq was collected in Tempus RNA Blood Tubes (Thermo Fisher Scientific, 4342792). As soon as possible after blood collection, tubes were shaken and stored at −80 °C. </span></span>
<span id="cb34-114"><a href="#cb34-114" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; </span></span>
<span id="cb34-115"><a href="#cb34-115" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; RNA extraction, library preparation and sequencing were performed as described previously </span><span class="co">[</span><span class="ot">28</span><span class="co">](https://www.nature.com/articles/s41467-021-24981-1)</span><span class="at">. In brief, frozen blood samples were thawed, and total RNA was extracted using a modification of the MagMax protocol for Stabilized Blood Tubes RNA Isolation Kit. Samples yielding sufficient RNA (&gt;50 ng) were barcoded and prepared for pooled whole transcriptome sequencing using the TruSeq Stranded Total RNA Library Prep Gold, which is designed to remove ribosomal, globin and mitochondrial RNA. Libraries were amplified with 15 cycles of PCR, pooled and sequenced on a NovaSeq 6000 (Illumina) using Sprime flow cells with 100-bp paired-end reads, targeting a mean of 50 million read pairs per sample. </span></span>
<span id="cb34-116"><a href="#cb34-116" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-117"><a href="#cb34-117" aria-hidden="true" tabindex="-1"></a><span class="fu">## Quality control metrics</span></span>
<span id="cb34-118"><a href="#cb34-118" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-121"><a href="#cb34-121" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-122"><a href="#cb34-122" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: read-qc-data</span></span>
<span id="cb34-123"><a href="#cb34-123" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb34-124"><a href="#cb34-124" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-125"><a href="#cb34-125" aria-hidden="true" tabindex="-1"></a><span class="co"># Import QC data</span></span>
<span id="cb34-126"><a href="#cb34-126" aria-hidden="true" tabindex="-1"></a>stages <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"raw_concat"</span>, <span class="st">"cleaned"</span>, <span class="st">"dedup"</span>, <span class="st">"ribo_initial"</span>, <span class="st">"ribo_secondary"</span>)</span>
<span id="cb34-127"><a href="#cb34-127" aria-hidden="true" tabindex="-1"></a>basic_stats <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(basic_stats_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-128"><a href="#cb34-128" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb34-129"><a href="#cb34-129" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">factor</span>(stage, <span class="at">levels =</span> stages),</span>
<span id="cb34-130"><a href="#cb34-130" aria-hidden="true" tabindex="-1"></a>         <span class="at">sample =</span> <span class="fu">fct_inorder</span>(sample)) <span class="sc">%&gt;%</span></span>
<span id="cb34-131"><a href="#cb34-131" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span>sample <span class="sc">%in%</span> low_sample_number)</span>
<span id="cb34-132"><a href="#cb34-132" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-133"><a href="#cb34-133" aria-hidden="true" tabindex="-1"></a>adapter_stats <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(adapter_stats_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-134"><a href="#cb34-134" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb34-135"><a href="#cb34-135" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">factor</span>(stage, <span class="at">levels =</span> stages),</span>
<span id="cb34-136"><a href="#cb34-136" aria-hidden="true" tabindex="-1"></a>         <span class="at">read_pair =</span> <span class="fu">fct_inorder</span>(<span class="fu">as.character</span>(read_pair))) <span class="sc">%&gt;%</span></span>
<span id="cb34-137"><a href="#cb34-137" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span>sample <span class="sc">%in%</span> low_sample_number)</span>
<span id="cb34-138"><a href="#cb34-138" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-139"><a href="#cb34-139" aria-hidden="true" tabindex="-1"></a>quality_base_stats <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(quality_base_stats_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-140"><a href="#cb34-140" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb34-141"><a href="#cb34-141" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">factor</span>(stage, <span class="at">levels =</span> stages),</span>
<span id="cb34-142"><a href="#cb34-142" aria-hidden="true" tabindex="-1"></a>         <span class="at">read_pair =</span> <span class="fu">fct_inorder</span>(<span class="fu">as.character</span>(read_pair))) <span class="sc">%&gt;%</span></span>
<span id="cb34-143"><a href="#cb34-143" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span>sample <span class="sc">%in%</span> low_sample_number)</span>
<span id="cb34-144"><a href="#cb34-144" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb34-145"><a href="#cb34-145" aria-hidden="true" tabindex="-1"></a>quality_seq_stats <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(quality_seq_stats_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-146"><a href="#cb34-146" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb34-147"><a href="#cb34-147" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">factor</span>(stage, <span class="at">levels =</span> stages),</span>
<span id="cb34-148"><a href="#cb34-148" aria-hidden="true" tabindex="-1"></a>         <span class="at">read_pair =</span> <span class="fu">fct_inorder</span>(<span class="fu">as.character</span>(read_pair))) <span class="sc">%&gt;%</span></span>
<span id="cb34-149"><a href="#cb34-149" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span>sample <span class="sc">%in%</span> low_sample_number)</span>
<span id="cb34-150"><a href="#cb34-150" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb34-151"><a href="#cb34-151" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb34-152"><a href="#cb34-152" aria-hidden="true" tabindex="-1"></a> <span class="co"># Filter to raw data</span></span>
<span id="cb34-153"><a href="#cb34-153" aria-hidden="true" tabindex="-1"></a> basic_stats_raw <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"raw_concat"</span>)</span>
<span id="cb34-154"><a href="#cb34-154" aria-hidden="true" tabindex="-1"></a> adapter_stats_raw <span class="ot">&lt;-</span> adapter_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"raw_concat"</span>)</span>
<span id="cb34-155"><a href="#cb34-155" aria-hidden="true" tabindex="-1"></a> quality_base_stats_raw <span class="ot">&lt;-</span> quality_base_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"raw_concat"</span>)</span>
<span id="cb34-156"><a href="#cb34-156" aria-hidden="true" tabindex="-1"></a> quality_seq_stats_raw <span class="ot">&lt;-</span> quality_seq_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"raw_concat"</span>)</span>
<span id="cb34-157"><a href="#cb34-157" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb34-158"><a href="#cb34-158" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb34-159"><a href="#cb34-159" aria-hidden="true" tabindex="-1"></a> <span class="co"># Get key values for readout</span></span>
<span id="cb34-160"><a href="#cb34-160" aria-hidden="true" tabindex="-1"></a>raw_read_counts <span class="ot">&lt;-</span> basic_stats_raw <span class="sc">%&gt;%</span> </span>
<span id="cb34-161"><a href="#cb34-161" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">rmin =</span> <span class="fu">min</span>(n_read_pairs), </span>
<span id="cb34-162"><a href="#cb34-162" aria-hidden="true" tabindex="-1"></a>            <span class="at">rmax=</span><span class="fu">max</span>(n_read_pairs),</span>
<span id="cb34-163"><a href="#cb34-163" aria-hidden="true" tabindex="-1"></a>            <span class="at">rmean=</span><span class="fu">mean</span>(n_read_pairs), </span>
<span id="cb34-164"><a href="#cb34-164" aria-hidden="true" tabindex="-1"></a>            <span class="at">rtot =</span> <span class="fu">sum</span>(n_read_pairs),</span>
<span id="cb34-165"><a href="#cb34-165" aria-hidden="true" tabindex="-1"></a>            <span class="at">btot =</span> <span class="fu">sum</span>(n_bases_approx),</span>
<span id="cb34-166"><a href="#cb34-166" aria-hidden="true" tabindex="-1"></a>            <span class="at">dmin =</span> <span class="fu">min</span>(percent_duplicates), </span>
<span id="cb34-167"><a href="#cb34-167" aria-hidden="true" tabindex="-1"></a>            <span class="at">dmax=</span><span class="fu">max</span>(percent_duplicates),</span>
<span id="cb34-168"><a href="#cb34-168" aria-hidden="true" tabindex="-1"></a>            <span class="at">dmean=</span><span class="fu">mean</span>(percent_duplicates), </span>
<span id="cb34-169"><a href="#cb34-169" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb34-170"><a href="#cb34-170" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-171"><a href="#cb34-171" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-172"><a href="#cb34-172" aria-hidden="true" tabindex="-1"></a>In total, these 417 samples contained 27.7B read pairs. The samples had 30.7M - 133.9M (mean ~66.4M) read pairs each. The number of reads looks pretty good, although a few samples have a much lower number of reads. The total number of base pairs also looks reasonable, matching the trends of the number of reads. The duplication rate is very high. Adapter content is variable, but most samples have a low adapter content (below 5%). As we'd expect, we see higher quality Q scores near the beginning of the read and a gradual decline towards the end of the read, however all positions seem to have a Q score of 35 which means that our reads are \~99.97% accurate. When looking at the Q score over all sequences, we can see a sharp peak around 35, which corresponds to the previous plot, indicating high read quality. </span>
<span id="cb34-173"><a href="#cb34-173" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-176"><a href="#cb34-176" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-177"><a href="#cb34-177" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 9</span></span>
<span id="cb34-178"><a href="#cb34-178" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb34-179"><a href="#cb34-179" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-basic-stats</span></span>
<span id="cb34-180"><a href="#cb34-180" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-181"><a href="#cb34-181" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare data</span></span>
<span id="cb34-182"><a href="#cb34-182" aria-hidden="true" tabindex="-1"></a>basic_stats_raw_metrics <span class="ot">&lt;-</span> basic_stats_raw <span class="sc">%&gt;%</span></span>
<span id="cb34-183"><a href="#cb34-183" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(library,</span>
<span id="cb34-184"><a href="#cb34-184" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at"># Read pairs</span><span class="st">`</span> <span class="ot">=</span> n_read_pairs,</span>
<span id="cb34-185"><a href="#cb34-185" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Total base pairs</span><span class="sc">\n</span><span class="at">(approx)</span><span class="st">`</span> <span class="ot">=</span> n_bases_approx,</span>
<span id="cb34-186"><a href="#cb34-186" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">% Duplicates</span><span class="sc">\n</span><span class="at">(FASTQC)</span><span class="st">`</span> <span class="ot">=</span> percent_duplicates) <span class="sc">%&gt;%</span></span>
<span id="cb34-187"><a href="#cb34-187" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>library, <span class="at">names_to =</span> <span class="st">"metric"</span>, <span class="at">values_to =</span> <span class="st">"value"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-188"><a href="#cb34-188" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">metric =</span> <span class="fu">fct_inorder</span>(metric)) </span>
<span id="cb34-189"><a href="#cb34-189" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-190"><a href="#cb34-190" aria-hidden="true" tabindex="-1"></a><span class="co"># Set up plot templates</span></span>
<span id="cb34-191"><a href="#cb34-191" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-192"><a href="#cb34-192" aria-hidden="true" tabindex="-1"></a>g_basic <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(basic_stats_raw_metrics, <span class="fu">aes</span>(<span class="at">x =</span> library, <span class="at">y =</span> value)) <span class="sc">+</span></span>
<span id="cb34-193"><a href="#cb34-193" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position =</span> <span class="st">"dodge"</span>) <span class="sc">+</span></span>
<span id="cb34-194"><a href="#cb34-194" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>() <span class="sc">+</span></span>
<span id="cb34-195"><a href="#cb34-195" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb34-196"><a href="#cb34-196" aria-hidden="true" tabindex="-1"></a>  <span class="fu">expand_limits</span>(<span class="at">y =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">100</span>)) <span class="sc">+</span></span>
<span id="cb34-197"><a href="#cb34-197" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(metric <span class="sc">~</span> ., <span class="at">scales =</span> <span class="st">"free"</span>, <span class="at">space =</span> <span class="st">"free_x"</span>, <span class="at">switch =</span> <span class="st">"y"</span>) <span class="sc">+</span></span>
<span id="cb34-198"><a href="#cb34-198" aria-hidden="true" tabindex="-1"></a>  theme_kit <span class="sc">+</span> <span class="fu">theme</span>(</span>
<span id="cb34-199"><a href="#cb34-199" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb34-200"><a href="#cb34-200" aria-hidden="true" tabindex="-1"></a>    <span class="at">strip.text.y =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"plain"</span>)</span>
<span id="cb34-201"><a href="#cb34-201" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb34-202"><a href="#cb34-202" aria-hidden="true" tabindex="-1"></a>g_basic</span>
<span id="cb34-203"><a href="#cb34-203" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-204"><a href="#cb34-204" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-207"><a href="#cb34-207" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-208"><a href="#cb34-208" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-raw-quality</span></span>
<span id="cb34-209"><a href="#cb34-209" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 8</span></span>
<span id="cb34-210"><a href="#cb34-210" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-211"><a href="#cb34-211" aria-hidden="true" tabindex="-1"></a><span class="co"># Set up plotting templates</span></span>
<span id="cb34-212"><a href="#cb34-212" aria-hidden="true" tabindex="-1"></a>g_qual_raw <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">linetype=</span>read_pair, <span class="at">group=</span><span class="fu">interaction</span>(sample,read_pair))) <span class="sc">+</span> </span>
<span id="cb34-213"><a href="#cb34-213" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_linetype_discrete</span>(<span class="at">name =</span> <span class="st">"Read Pair"</span>) <span class="sc">+</span></span>
<span id="cb34-214"><a href="#cb34-214" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color=</span><span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>),</span>
<span id="cb34-215"><a href="#cb34-215" aria-hidden="true" tabindex="-1"></a>         <span class="at">linetype =</span> <span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>)) <span class="sc">+</span></span>
<span id="cb34-216"><a href="#cb34-216" aria-hidden="true" tabindex="-1"></a>  theme_base</span>
<span id="cb34-217"><a href="#cb34-217" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-218"><a href="#cb34-218" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize adapters</span></span>
<span id="cb34-219"><a href="#cb34-219" aria-hidden="true" tabindex="-1"></a>g_adapters_raw <span class="ot">&lt;-</span> g_qual_raw <span class="sc">+</span> </span>
<span id="cb34-220"><a href="#cb34-220" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>pc_adapters), <span class="at">data=</span>adapter_stats_raw) <span class="sc">+</span></span>
<span id="cb34-221"><a href="#cb34-221" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Adapters"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb34-222"><a href="#cb34-222" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">10</span>,<span class="dv">1</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb34-223"><a href="#cb34-223" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Position"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb34-224"><a href="#cb34-224" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">500</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb34-225"><a href="#cb34-225" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(.<span class="sc">~</span>adapter)</span>
<span id="cb34-226"><a href="#cb34-226" aria-hidden="true" tabindex="-1"></a>g_adapters_raw</span>
<span id="cb34-227"><a href="#cb34-227" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-228"><a href="#cb34-228" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize quality</span></span>
<span id="cb34-229"><a href="#cb34-229" aria-hidden="true" tabindex="-1"></a>g_quality_base_raw <span class="ot">&lt;-</span> g_qual_raw <span class="sc">+</span></span>
<span id="cb34-230"><a href="#cb34-230" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb34-231"><a href="#cb34-231" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb34-232"><a href="#cb34-232" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>mean_phred_score), <span class="at">data=</span>quality_base_stats_raw) <span class="sc">+</span></span>
<span id="cb34-233"><a href="#cb34-233" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"Mean Phred score"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">10</span>,<span class="dv">45</span>)) <span class="sc">+</span></span>
<span id="cb34-234"><a href="#cb34-234" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Position"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb34-235"><a href="#cb34-235" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">500</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb34-236"><a href="#cb34-236" aria-hidden="true" tabindex="-1"></a>g_quality_base_raw</span>
<span id="cb34-237"><a href="#cb34-237" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-238"><a href="#cb34-238" aria-hidden="true" tabindex="-1"></a>g_quality_seq_raw <span class="ot">&lt;-</span> g_qual_raw <span class="sc">+</span></span>
<span id="cb34-239"><a href="#cb34-239" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb34-240"><a href="#cb34-240" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb34-241"><a href="#cb34-241" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>mean_phred_score, <span class="at">y=</span>n_sequences), <span class="at">data=</span>quality_seq_stats_raw) <span class="sc">+</span></span>
<span id="cb34-242"><a href="#cb34-242" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Mean Phred score"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb34-243"><a href="#cb34-243" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"# Sequences"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb34-244"><a href="#cb34-244" aria-hidden="true" tabindex="-1"></a>g_quality_seq_raw</span>
<span id="cb34-245"><a href="#cb34-245" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-246"><a href="#cb34-246" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-247"><a href="#cb34-247" aria-hidden="true" tabindex="-1"></a><span class="fu"># Preprocessing</span></span>
<span id="cb34-248"><a href="#cb34-248" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-249"><a href="#cb34-249" aria-hidden="true" tabindex="-1"></a><span class="fu">## Summary</span></span>
<span id="cb34-250"><a href="#cb34-250" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-251"><a href="#cb34-251" aria-hidden="true" tabindex="-1"></a>The average fraction of reads at each stage in the preprocessing pipeline is shown in the following table. Adapter trimming &amp; filtering doesn't lose too many reads. Deduplication loses us about 55% of reads on average, which is good considering the high duplication content seen earlier. Ribodepletion only removes about 2% of reads in total, which makes sense as they did ribosomal depletion during sample preparation.</span>
<span id="cb34-252"><a href="#cb34-252" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-255"><a href="#cb34-255" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-256"><a href="#cb34-256" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: preproc-table</span></span>
<span id="cb34-257"><a href="#cb34-257" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-258"><a href="#cb34-258" aria-hidden="true" tabindex="-1"></a><span class="co"># </span><span class="al">TODO</span><span class="co">: Group by pool size as well</span></span>
<span id="cb34-259"><a href="#cb34-259" aria-hidden="true" tabindex="-1"></a><span class="co"># Count read losses</span></span>
<span id="cb34-260"><a href="#cb34-260" aria-hidden="true" tabindex="-1"></a>n_reads_rel <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> </span>
<span id="cb34-261"><a href="#cb34-261" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, stage, percent_duplicates, n_read_pairs) <span class="sc">%&gt;%</span></span>
<span id="cb34-262"><a href="#cb34-262" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb34-263"><a href="#cb34-263" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(sample, stage) <span class="sc">%&gt;%</span></span>
<span id="cb34-264"><a href="#cb34-264" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_retained =</span> n_read_pairs <span class="sc">/</span> <span class="fu">lag</span>(n_read_pairs),</span>
<span id="cb34-265"><a href="#cb34-265" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_lost =</span> <span class="dv">1</span> <span class="sc">-</span> p_reads_retained,</span>
<span id="cb34-266"><a href="#cb34-266" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_retained_abs =</span> n_read_pairs <span class="sc">/</span> n_read_pairs[<span class="dv">1</span>],</span>
<span id="cb34-267"><a href="#cb34-267" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_lost_abs =</span> <span class="dv">1</span><span class="sc">-</span>p_reads_retained_abs,</span>
<span id="cb34-268"><a href="#cb34-268" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_lost_abs_marginal =</span> p_reads_lost_abs <span class="sc">-</span> <span class="fu">lag</span>(p_reads_lost_abs))</span>
<span id="cb34-269"><a href="#cb34-269" aria-hidden="true" tabindex="-1"></a>n_reads_rel_display <span class="ot">&lt;-</span> n_reads_rel <span class="sc">%&gt;%</span> </span>
<span id="cb34-270"><a href="#cb34-270" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">Stage=</span>stage) <span class="sc">%&gt;%</span> </span>
<span id="cb34-271"><a href="#cb34-271" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Stage) <span class="sc">%&gt;%</span> </span>
<span id="cb34-272"><a href="#cb34-272" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="st">`</span><span class="at">% Total Reads Lost (Cumulative)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">round</span>(<span class="fu">min</span>(p_reads_lost_abs<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">"-"</span>, <span class="fu">round</span>(<span class="fu">max</span>(p_reads_lost_abs<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">" (mean "</span>, <span class="fu">round</span>(<span class="fu">mean</span>(p_reads_lost_abs<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">")"</span>),</span>
<span id="cb34-273"><a href="#cb34-273" aria-hidden="true" tabindex="-1"></a>            <span class="st">`</span><span class="at">% Total Reads Lost (Marginal)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">round</span>(<span class="fu">min</span>(p_reads_lost_abs_marginal<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">"-"</span>, <span class="fu">round</span>(<span class="fu">max</span>(p_reads_lost_abs_marginal<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">" (mean "</span>, <span class="fu">round</span>(<span class="fu">mean</span>(p_reads_lost_abs_marginal<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">")"</span>), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb34-274"><a href="#cb34-274" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Stage <span class="sc">!=</span> <span class="st">"raw_concat"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-275"><a href="#cb34-275" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Stage =</span> Stage <span class="sc">%&gt;%</span> as.numeric <span class="sc">%&gt;%</span> <span class="fu">factor</span>(<span class="at">labels=</span><span class="fu">c</span>(<span class="st">"Trimming &amp; filtering"</span>, <span class="st">"Deduplication"</span>, <span class="st">"Initial ribodepletion"</span>, <span class="st">"Secondary ribodepletion"</span>)))</span>
<span id="cb34-276"><a href="#cb34-276" aria-hidden="true" tabindex="-1"></a>n_reads_rel_display</span>
<span id="cb34-277"><a href="#cb34-277" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-278"><a href="#cb34-278" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-279"><a href="#cb34-279" aria-hidden="true" tabindex="-1"></a><span class="fu">## Quality control metrics</span></span>
<span id="cb34-280"><a href="#cb34-280" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-281"><a href="#cb34-281" aria-hidden="true" tabindex="-1"></a>Cleaning seems to get rid of the library contamination and the polyg contamination. Polya contamination still is present, but I don't think we've found a solution to remove it as yet. Q score remain the same during read cleaning when looking at the positions, with the end of the read actually improving in score. Q scores across all sequences look pretty much the same throughout cleaning.</span>
<span id="cb34-282"><a href="#cb34-282" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-285"><a href="#cb34-285" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-286"><a href="#cb34-286" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb34-287"><a href="#cb34-287" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-quality</span></span>
<span id="cb34-288"><a href="#cb34-288" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 8</span></span>
<span id="cb34-289"><a href="#cb34-289" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb34-290"><a href="#cb34-290" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-291"><a href="#cb34-291" aria-hidden="true" tabindex="-1"></a>g_qual <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">linetype=</span>read_pair, <span class="at">group=</span><span class="fu">interaction</span>(sample,read_pair))) <span class="sc">+</span> </span>
<span id="cb34-292"><a href="#cb34-292" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_linetype_discrete</span>(<span class="at">name =</span> <span class="st">"Read Pair"</span>) <span class="sc">+</span></span>
<span id="cb34-293"><a href="#cb34-293" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color=</span><span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>),</span>
<span id="cb34-294"><a href="#cb34-294" aria-hidden="true" tabindex="-1"></a>         <span class="at">linetype =</span> <span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>)) <span class="sc">+</span></span>
<span id="cb34-295"><a href="#cb34-295" aria-hidden="true" tabindex="-1"></a>  theme_base</span>
<span id="cb34-296"><a href="#cb34-296" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-297"><a href="#cb34-297" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize adapters</span></span>
<span id="cb34-298"><a href="#cb34-298" aria-hidden="true" tabindex="-1"></a>g_adapters <span class="ot">&lt;-</span> g_qual <span class="sc">+</span> </span>
<span id="cb34-299"><a href="#cb34-299" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>pc_adapters), <span class="at">data=</span>adapter_stats) <span class="sc">+</span></span>
<span id="cb34-300"><a href="#cb34-300" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Adapters"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">20</span>),</span>
<span id="cb34-301"><a href="#cb34-301" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">50</span>,<span class="dv">10</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb34-302"><a href="#cb34-302" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Position"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb34-303"><a href="#cb34-303" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">140</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb34-304"><a href="#cb34-304" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(stage<span class="sc">~</span>adapter)</span>
<span id="cb34-305"><a href="#cb34-305" aria-hidden="true" tabindex="-1"></a>g_adapters</span>
<span id="cb34-306"><a href="#cb34-306" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-307"><a href="#cb34-307" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize quality</span></span>
<span id="cb34-308"><a href="#cb34-308" aria-hidden="true" tabindex="-1"></a>g_quality_base <span class="ot">&lt;-</span> g_qual <span class="sc">+</span></span>
<span id="cb34-309"><a href="#cb34-309" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb34-310"><a href="#cb34-310" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb34-311"><a href="#cb34-311" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>mean_phred_score), <span class="at">data=</span>quality_base_stats) <span class="sc">+</span></span>
<span id="cb34-312"><a href="#cb34-312" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"Mean Phred score"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">10</span>,<span class="dv">45</span>)) <span class="sc">+</span></span>
<span id="cb34-313"><a href="#cb34-313" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Position"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb34-314"><a href="#cb34-314" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">140</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb34-315"><a href="#cb34-315" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(stage<span class="sc">~</span>.)</span>
<span id="cb34-316"><a href="#cb34-316" aria-hidden="true" tabindex="-1"></a>g_quality_base</span>
<span id="cb34-317"><a href="#cb34-317" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-318"><a href="#cb34-318" aria-hidden="true" tabindex="-1"></a>g_quality_seq <span class="ot">&lt;-</span> g_qual <span class="sc">+</span></span>
<span id="cb34-319"><a href="#cb34-319" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb34-320"><a href="#cb34-320" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb34-321"><a href="#cb34-321" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>mean_phred_score, <span class="at">y=</span>n_sequences), <span class="at">data=</span>quality_seq_stats) <span class="sc">+</span></span>
<span id="cb34-322"><a href="#cb34-322" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Mean Phred score"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb34-323"><a href="#cb34-323" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"# Sequences"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb34-324"><a href="#cb34-324" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(stage<span class="sc">~</span>., <span class="at">scales =</span> <span class="st">"free_y"</span>)</span>
<span id="cb34-325"><a href="#cb34-325" aria-hidden="true" tabindex="-1"></a>g_quality_seq</span>
<span id="cb34-326"><a href="#cb34-326" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-327"><a href="#cb34-327" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-328"><a href="#cb34-328" aria-hidden="true" tabindex="-1"></a>These plots below show the trends from above in each sample. Deduplication seems to drop by a lot post deduplication,which is a good thing given it was so high earlier. Mean read length remains relatively constant throuhgout. Some samples still seem to have high ribosomal content, however I'm not too worried about this.</span>
<span id="cb34-329"><a href="#cb34-329" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-332"><a href="#cb34-332" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-333"><a href="#cb34-333" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: preproc-figures</span></span>
<span id="cb34-334"><a href="#cb34-334" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb34-335"><a href="#cb34-335" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 3</span></span>
<span id="cb34-336"><a href="#cb34-336" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 6</span></span>
<span id="cb34-337"><a href="#cb34-337" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-338"><a href="#cb34-338" aria-hidden="true" tabindex="-1"></a>g_stage_trace <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(basic_stats, <span class="fu">aes</span>(<span class="at">x=</span>stage, <span class="at">group=</span>sample)) <span class="sc">+</span></span>
<span id="cb34-339"><a href="#cb34-339" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb34-340"><a href="#cb34-340" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-341"><a href="#cb34-341" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot reads over preprocessing</span></span>
<span id="cb34-342"><a href="#cb34-342" aria-hidden="true" tabindex="-1"></a>g_reads_stages <span class="ot">&lt;-</span> g_stage_trace <span class="sc">+</span></span>
<span id="cb34-343"><a href="#cb34-343" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>n_read_pairs)) <span class="sc">+</span></span>
<span id="cb34-344"><a href="#cb34-344" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"# Read pairs"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>))</span>
<span id="cb34-345"><a href="#cb34-345" aria-hidden="true" tabindex="-1"></a>g_reads_stages</span>
<span id="cb34-346"><a href="#cb34-346" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-347"><a href="#cb34-347" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot relative read losses during preprocessing</span></span>
<span id="cb34-348"><a href="#cb34-348" aria-hidden="true" tabindex="-1"></a>g_reads_rel <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(n_reads_rel, </span>
<span id="cb34-349"><a href="#cb34-349" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">aes</span>(<span class="at">x=</span>stage, <span class="at">group=</span>sample)) <span class="sc">+</span></span>
<span id="cb34-350"><a href="#cb34-350" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>p_reads_lost_abs_marginal)) <span class="sc">+</span></span>
<span id="cb34-351"><a href="#cb34-351" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"% Total Reads Lost"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), </span>
<span id="cb34-352"><a href="#cb34-352" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="cf">function</span>(x) x<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb34-353"><a href="#cb34-353" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb34-354"><a href="#cb34-354" aria-hidden="true" tabindex="-1"></a>g_reads_rel</span>
<span id="cb34-355"><a href="#cb34-355" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-356"><a href="#cb34-356" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-357"><a href="#cb34-357" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-360"><a href="#cb34-360" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-361"><a href="#cb34-361" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: preproc-dedup</span></span>
<span id="cb34-362"><a href="#cb34-362" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb34-363"><a href="#cb34-363" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-364"><a href="#cb34-364" aria-hidden="true" tabindex="-1"></a>stage_dup <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(stage) <span class="sc">%&gt;%</span> </span>
<span id="cb34-365"><a href="#cb34-365" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">dmin =</span> <span class="fu">min</span>(percent_duplicates), <span class="at">dmax=</span><span class="fu">max</span>(percent_duplicates),</span>
<span id="cb34-366"><a href="#cb34-366" aria-hidden="true" tabindex="-1"></a>            <span class="at">dmean=</span><span class="fu">mean</span>(percent_duplicates), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb34-367"><a href="#cb34-367" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-368"><a href="#cb34-368" aria-hidden="true" tabindex="-1"></a>g_dup_stages <span class="ot">&lt;-</span> g_stage_trace <span class="sc">+</span></span>
<span id="cb34-369"><a href="#cb34-369" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>percent_duplicates)) <span class="sc">+</span></span>
<span id="cb34-370"><a href="#cb34-370" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"% Duplicates"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb34-371"><a href="#cb34-371" aria-hidden="true" tabindex="-1"></a>g_dup_stages</span>
<span id="cb34-372"><a href="#cb34-372" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-373"><a href="#cb34-373" aria-hidden="true" tabindex="-1"></a>g_readlen_stages <span class="ot">&lt;-</span> g_stage_trace <span class="sc">+</span> <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>mean_seq_len)) <span class="sc">+</span></span>
<span id="cb34-374"><a href="#cb34-374" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"Mean read length (nt)"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>))</span>
<span id="cb34-375"><a href="#cb34-375" aria-hidden="true" tabindex="-1"></a>g_readlen_stages</span>
<span id="cb34-376"><a href="#cb34-376" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-377"><a href="#cb34-377" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-378"><a href="#cb34-378" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-379"><a href="#cb34-379" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-382"><a href="#cb34-382" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-383"><a href="#cb34-383" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: ribo-frac</span></span>
<span id="cb34-384"><a href="#cb34-384" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb34-385"><a href="#cb34-385" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate reads lost during ribodepletion (approximation for % ribosomal reads)</span></span>
<span id="cb34-386"><a href="#cb34-386" aria-hidden="true" tabindex="-1"></a>reads_ribo <span class="ot">&lt;-</span> n_reads_rel <span class="sc">%&gt;%</span> </span>
<span id="cb34-387"><a href="#cb34-387" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(stage <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"dedup"</span>, <span class="st">"ribo_secondary"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb34-388"><a href="#cb34-388" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb34-389"><a href="#cb34-389" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads_ribo=</span><span class="dv">1</span><span class="sc">-</span>n_read_pairs[<span class="dv">2</span>]<span class="sc">/</span>n_read_pairs[<span class="dv">1</span>], <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-390"><a href="#cb34-390" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by =</span> <span class="st">'sample'</span>)</span>
<span id="cb34-391"><a href="#cb34-391" aria-hidden="true" tabindex="-1"></a>reads_ribo_summ <span class="ot">&lt;-</span> reads_ribo <span class="sc">%&gt;%</span></span>
<span id="cb34-392"><a href="#cb34-392" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb34-393"><a href="#cb34-393" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">min=</span><span class="fu">min</span>(p_reads_ribo), <span class="at">max=</span><span class="fu">max</span>(p_reads_ribo),</span>
<span id="cb34-394"><a href="#cb34-394" aria-hidden="true" tabindex="-1"></a>            <span class="at">mean=</span><span class="fu">mean</span>(p_reads_ribo), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-395"><a href="#cb34-395" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by =</span> <span class="st">'sample'</span>)</span>
<span id="cb34-396"><a href="#cb34-396" aria-hidden="true" tabindex="-1"></a>g_reads_ribo <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(reads_ribo, </span>
<span id="cb34-397"><a href="#cb34-397" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">aes</span>(<span class="at">x=</span>library, <span class="at">y=</span>p_reads_ribo)) <span class="sc">+</span></span>
<span id="cb34-398"><a href="#cb34-398" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span> </span>
<span id="cb34-399"><a href="#cb34-399" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"Approx % ribosomal reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>),</span>
<span id="cb34-400"><a href="#cb34-400" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>)<span class="sc">+</span></span>
<span id="cb34-401"><a href="#cb34-401" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb34-402"><a href="#cb34-402" aria-hidden="true" tabindex="-1"></a>g_reads_ribo</span>
<span id="cb34-403"><a href="#cb34-403" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-404"><a href="#cb34-404" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-405"><a href="#cb34-405" aria-hidden="true" tabindex="-1"></a><span class="fu"># Taxonomic composition</span></span>
<span id="cb34-406"><a href="#cb34-406" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-407"><a href="#cb34-407" aria-hidden="true" tabindex="-1"></a><span class="fu">## High-level composition</span></span>
<span id="cb34-408"><a href="#cb34-408" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-409"><a href="#cb34-409" aria-hidden="true" tabindex="-1"></a>To assess the high-level composition of the reads, I ran the ribodepleted files through Kraken2 and summarized the results with Bracken.</span>
<span id="cb34-410"><a href="#cb34-410" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-411"><a href="#cb34-411" aria-hidden="true" tabindex="-1"></a>The groups listed below were created by Will:</span>
<span id="cb34-412"><a href="#cb34-412" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-413"><a href="#cb34-413" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Filtered (removed during cleaning)</span>
<span id="cb34-414"><a href="#cb34-414" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Duplicate (removed during deduplication)</span>
<span id="cb34-415"><a href="#cb34-415" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Ribosomal (removed during ribodepletion)</span>
<span id="cb34-416"><a href="#cb34-416" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Unassigned (non-ribosomal reads that were not assigned to any taxon by Kraken/Bracken)</span>
<span id="cb34-417"><a href="#cb34-417" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Bacterial (non-ribosomal reads assigned to the Bacteria domain by Kraken/Bracken)</span>
<span id="cb34-418"><a href="#cb34-418" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Archaeal (non-ribosomal reads assigned to the Archaea domain by Kraken/Bracken)</span>
<span id="cb34-419"><a href="#cb34-419" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Viral (non-ribosomal reads assigned to the Viruses domain by Kraken/Bracken)</span>
<span id="cb34-420"><a href="#cb34-420" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Human (non-ribosomal reads assigned to the Eukarya domain by Kraken/Bracken)</span>
<span id="cb34-421"><a href="#cb34-421" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-422"><a href="#cb34-422" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-425"><a href="#cb34-425" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-426"><a href="#cb34-426" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: prepare-composition</span></span>
<span id="cb34-427"><a href="#cb34-427" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb34-428"><a href="#cb34-428" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-429"><a href="#cb34-429" aria-hidden="true" tabindex="-1"></a>classifications <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Filtered"</span>, <span class="st">"Duplicate"</span>, <span class="st">"Ribosomal"</span>, <span class="st">"Unassigned"</span>,</span>
<span id="cb34-430"><a href="#cb34-430" aria-hidden="true" tabindex="-1"></a>                     <span class="st">"Bacterial"</span>, <span class="st">"Archaeal"</span>, <span class="st">"Viral"</span>, <span class="st">"Human"</span>)</span>
<span id="cb34-431"><a href="#cb34-431" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-432"><a href="#cb34-432" aria-hidden="true" tabindex="-1"></a><span class="co"># Import composition data</span></span>
<span id="cb34-433"><a href="#cb34-433" aria-hidden="true" tabindex="-1"></a>tax_final_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(results_dir, <span class="st">"taxonomy_final"</span>)</span>
<span id="cb34-434"><a href="#cb34-434" aria-hidden="true" tabindex="-1"></a>comp_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(tax_final_dir, <span class="st">"taxonomic_composition.tsv.gz"</span>)</span>
<span id="cb34-435"><a href="#cb34-435" aria-hidden="true" tabindex="-1"></a>comp <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(comp_path) <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(libraries) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(library <span class="sc">!=</span> <span class="st">"PCL21"</span>)</span>
<span id="cb34-436"><a href="#cb34-436" aria-hidden="true" tabindex="-1"></a>comp_minor <span class="ot">&lt;-</span> comp <span class="sc">%&gt;%</span> </span>
<span id="cb34-437"><a href="#cb34-437" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(classification <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Archaeal"</span>, <span class="st">"Viral"</span>, <span class="st">"Human"</span>, <span class="st">"Other"</span>))</span>
<span id="cb34-438"><a href="#cb34-438" aria-hidden="true" tabindex="-1"></a>comp_assigned <span class="ot">&lt;-</span> comp <span class="sc">%&gt;%</span></span>
<span id="cb34-439"><a href="#cb34-439" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span> classification <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Filtered"</span>, <span class="st">"Duplicate"</span>, </span>
<span id="cb34-440"><a href="#cb34-440" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">"Ribosomal"</span>, <span class="st">"Unassigned"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb34-441"><a href="#cb34-441" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb34-442"><a href="#cb34-442" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads =</span> n_reads<span class="sc">/</span><span class="fu">sum</span>(n_reads))</span>
<span id="cb34-443"><a href="#cb34-443" aria-hidden="true" tabindex="-1"></a>comp_assigned_minor <span class="ot">&lt;-</span> comp_assigned <span class="sc">%&gt;%</span> </span>
<span id="cb34-444"><a href="#cb34-444" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(classification <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Archaeal"</span>, <span class="st">"Viral"</span>, <span class="st">"Human"</span>, <span class="st">"Other"</span>))</span>
<span id="cb34-445"><a href="#cb34-445" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-446"><a href="#cb34-446" aria-hidden="true" tabindex="-1"></a><span class="co"># Summarize composition</span></span>
<span id="cb34-447"><a href="#cb34-447" aria-hidden="true" tabindex="-1"></a>read_comp_summ <span class="ot">&lt;-</span> comp <span class="sc">%&gt;%</span> </span>
<span id="cb34-448"><a href="#cb34-448" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(classification) <span class="sc">%&gt;%</span></span>
<span id="cb34-449"><a href="#cb34-449" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads =</span> <span class="fu">sum</span>(n_reads), <span class="at">.groups =</span> <span class="st">"drop_last"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-450"><a href="#cb34-450" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">n_reads =</span> <span class="fu">replace_na</span>(n_reads,<span class="dv">0</span>),</span>
<span id="cb34-451"><a href="#cb34-451" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads =</span> n_reads<span class="sc">/</span><span class="fu">sum</span>(n_reads),</span>
<span id="cb34-452"><a href="#cb34-452" aria-hidden="true" tabindex="-1"></a>         <span class="at">pc_reads =</span> p_reads<span class="sc">*</span><span class="dv">100</span>)</span>
<span id="cb34-453"><a href="#cb34-453" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-454"><a href="#cb34-454" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-455"><a href="#cb34-455" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-458"><a href="#cb34-458" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-459"><a href="#cb34-459" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-composition-all</span></span>
<span id="cb34-460"><a href="#cb34-460" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 7</span></span>
<span id="cb34-461"><a href="#cb34-461" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 8</span></span>
<span id="cb34-462"><a href="#cb34-462" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-463"><a href="#cb34-463" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare plotting templates</span></span>
<span id="cb34-464"><a href="#cb34-464" aria-hidden="true" tabindex="-1"></a>g_comp_base <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">x=</span>library, <span class="at">y=</span>p_reads, <span class="at">fill=</span>classification)) <span class="sc">+</span></span>
<span id="cb34-465"><a href="#cb34-465" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">name=</span><span class="st">"Plasma pool"</span>) <span class="sc">+</span></span>
<span id="cb34-466"><a href="#cb34-466" aria-hidden="true" tabindex="-1"></a>  theme_kit <span class="sc">+</span> </span>
<span id="cb34-467"><a href="#cb34-467" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust=</span><span class="dv">0</span>, <span class="at">face=</span><span class="st">"plain"</span>, <span class="at">size=</span><span class="fu">rel</span>(<span class="fl">1.5</span>)))</span>
<span id="cb34-468"><a href="#cb34-468" aria-hidden="true" tabindex="-1"></a>scale_y_pc_reads <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">partial</span>(scale_y_continuous, <span class="at">name =</span> <span class="st">"% Reads"</span>,</span>
<span id="cb34-469"><a href="#cb34-469" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>)</span>
<span id="cb34-470"><a href="#cb34-470" aria-hidden="true" tabindex="-1"></a>geom_comp <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">partial</span>(geom_col, <span class="at">position =</span> <span class="st">"stack"</span>, <span class="at">width =</span> <span class="dv">1</span>)</span>
<span id="cb34-471"><a href="#cb34-471" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-472"><a href="#cb34-472" aria-hidden="true" tabindex="-1"></a><span class="co"># Define a color palette for the classification</span></span>
<span id="cb34-473"><a href="#cb34-473" aria-hidden="true" tabindex="-1"></a>classification_colors <span class="ot">&lt;-</span> <span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">"Accent"</span>)</span>
<span id="cb34-474"><a href="#cb34-474" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(classification_colors) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Filtered"</span>, <span class="st">"Duplicate"</span>, <span class="st">"Ribosomal"</span>, <span class="st">"Unassigned"</span>, <span class="st">"Bacterial"</span>, <span class="st">"Archaeal"</span>, <span class="st">"Viral"</span>, <span class="st">"Human"</span>)</span>
<span id="cb34-475"><a href="#cb34-475" aria-hidden="true" tabindex="-1"></a>scale_fill_classification <span class="ot">&lt;-</span> <span class="cf">function</span>() {</span>
<span id="cb34-476"><a href="#cb34-476" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> classification_colors, <span class="at">name =</span> <span class="st">"Classification"</span>)</span>
<span id="cb34-477"><a href="#cb34-477" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb34-478"><a href="#cb34-478" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-479"><a href="#cb34-479" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot overall composition</span></span>
<span id="cb34-480"><a href="#cb34-480" aria-hidden="true" tabindex="-1"></a>g_comp <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> <span class="fu">geom_comp</span>(<span class="at">data =</span> comp) <span class="sc">+</span></span>
<span id="cb34-481"><a href="#cb34-481" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_pc_reads</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>)) <span class="sc">+</span></span>
<span id="cb34-482"><a href="#cb34-482" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_classification</span>() <span class="sc">+</span> </span>
<span id="cb34-483"><a href="#cb34-483" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Read composition (all reads, all groups)"</span>)</span>
<span id="cb34-484"><a href="#cb34-484" aria-hidden="true" tabindex="-1"></a>g_comp</span>
<span id="cb34-485"><a href="#cb34-485" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-486"><a href="#cb34-486" aria-hidden="true" tabindex="-1"></a>g_comp_assigned <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb34-487"><a href="#cb34-487" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_comp</span>(<span class="at">data =</span> comp_assigned) <span class="sc">+</span></span>
<span id="cb34-488"><a href="#cb34-488" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_pc_reads</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>)) <span class="sc">+</span></span>
<span id="cb34-489"><a href="#cb34-489" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_classification</span>() <span class="sc">+</span> </span>
<span id="cb34-490"><a href="#cb34-490" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Read composition (assigned reads, all groups)"</span>)</span>
<span id="cb34-491"><a href="#cb34-491" aria-hidden="true" tabindex="-1"></a>g_comp_assigned</span>
<span id="cb34-492"><a href="#cb34-492" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-493"><a href="#cb34-493" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-494"><a href="#cb34-494" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-495"><a href="#cb34-495" aria-hidden="true" tabindex="-1"></a><span class="fu">## Total viral content</span></span>
<span id="cb34-496"><a href="#cb34-496" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-497"><a href="#cb34-497" aria-hidden="true" tabindex="-1"></a>Total viral fraction average $1.12 \times 10^{-6}$ across samples. As a fraction of assigned (rather than total) reads, this jumped to $2.96 \times 10^{-6}$:</span>
<span id="cb34-498"><a href="#cb34-498" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-501"><a href="#cb34-501" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-502"><a href="#cb34-502" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: p-viral</span></span>
<span id="cb34-503"><a href="#cb34-503" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false </span></span>
<span id="cb34-504"><a href="#cb34-504" aria-hidden="true" tabindex="-1"></a>p_reads_viral_all <span class="ot">&lt;-</span> comp <span class="sc">%&gt;%</span> <span class="fu">filter</span>(classification <span class="sc">==</span> <span class="st">"Viral"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-505"><a href="#cb34-505" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">read_group =</span> <span class="st">"All reads"</span>)</span>
<span id="cb34-506"><a href="#cb34-506" aria-hidden="true" tabindex="-1"></a>p_reads_viral_assigned <span class="ot">&lt;-</span> comp_assigned <span class="sc">%&gt;%</span> <span class="fu">filter</span>(classification <span class="sc">==</span> <span class="st">"Viral"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-507"><a href="#cb34-507" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">read_group =</span> <span class="st">"Classified reads"</span>)</span>
<span id="cb34-508"><a href="#cb34-508" aria-hidden="true" tabindex="-1"></a>p_reads_viral <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(p_reads_viral_all, p_reads_viral_assigned)</span>
<span id="cb34-509"><a href="#cb34-509" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-510"><a href="#cb34-510" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb34-511"><a href="#cb34-511" aria-hidden="true" tabindex="-1"></a>g_viral <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(p_reads_viral, <span class="fu">aes</span>(<span class="at">x=</span>library, <span class="at">y=</span>p_reads, <span class="at">color =</span> read_group, <span class="at">group =</span> library)) <span class="sc">+</span></span>
<span id="cb34-512"><a href="#cb34-512" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb34-513"><a href="#cb34-513" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">color =</span> <span class="st">'black'</span>) <span class="sc">+</span></span>
<span id="cb34-514"><a href="#cb34-514" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">name=</span><span class="st">"Plasma pool"</span>) <span class="sc">+</span></span>
<span id="cb34-515"><a href="#cb34-515" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>(<span class="at">name=</span><span class="st">"Viral read fraction"</span>, <span class="at">labels =</span> <span class="fu">label_log</span>(<span class="at">digits=</span><span class="dv">2</span>), <span class="at">limits =</span> <span class="fu">c</span>(<span class="fl">1e-6</span>, <span class="fl">1e-2</span>)) <span class="sc">+</span></span>
<span id="cb34-516"><a href="#cb34-516" aria-hidden="true" tabindex="-1"></a>  theme_kit <span class="sc">+</span> </span>
<span id="cb34-517"><a href="#cb34-517" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>()</span>
<span id="cb34-518"><a href="#cb34-518" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-519"><a href="#cb34-519" aria-hidden="true" tabindex="-1"></a>g_viral</span>
<span id="cb34-520"><a href="#cb34-520" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-521"><a href="#cb34-521" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-522"><a href="#cb34-522" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-523"><a href="#cb34-523" aria-hidden="true" tabindex="-1"></a><span class="fu">## Taxonomic composition of viruses</span></span>
<span id="cb34-524"><a href="#cb34-524" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-525"><a href="#cb34-525" aria-hidden="true" tabindex="-1"></a>There seems to be no dominant virus in the samples. Specifically, this seems to be the dataset with the most viral diversity (when compared to the previous plasma datasets). The threshold for the label "other" are the set of families that make up less than 30% composition in all samples.</span>
<span id="cb34-526"><a href="#cb34-526" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-529"><a href="#cb34-529" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-530"><a href="#cb34-530" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: extract-viral-taxa</span></span>
<span id="cb34-531"><a href="#cb34-531" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb34-532"><a href="#cb34-532" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-533"><a href="#cb34-533" aria-hidden="true" tabindex="-1"></a><span class="co"># Get viral taxonomy</span></span>
<span id="cb34-534"><a href="#cb34-534" aria-hidden="true" tabindex="-1"></a>viral_taxa_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"total-virus-db.tsv.gz"</span>)</span>
<span id="cb34-535"><a href="#cb34-535" aria-hidden="true" tabindex="-1"></a>viral_taxa <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(viral_taxa_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb34-536"><a href="#cb34-536" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-537"><a href="#cb34-537" aria-hidden="true" tabindex="-1"></a><span class="co"># Get Kraken reports</span></span>
<span id="cb34-538"><a href="#cb34-538" aria-hidden="true" tabindex="-1"></a>reports_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(tax_final_dir, <span class="st">"kraken_reports.tsv.gz"</span>)</span>
<span id="cb34-539"><a href="#cb34-539" aria-hidden="true" tabindex="-1"></a>reports <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(reports_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-540"><a href="#cb34-540" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(sample)</span>
<span id="cb34-541"><a href="#cb34-541" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-542"><a href="#cb34-542" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter to viral taxa</span></span>
<span id="cb34-543"><a href="#cb34-543" aria-hidden="true" tabindex="-1"></a>kraken_reports_viral <span class="ot">&lt;-</span> <span class="fu">filter</span>(reports, taxid <span class="sc">%in%</span> viral_taxa<span class="sc">$</span>taxid) <span class="sc">%&gt;%</span></span>
<span id="cb34-544"><a href="#cb34-544" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb34-545"><a href="#cb34-545" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_viral =</span> n_reads_clade<span class="sc">/</span>n_reads_clade[<span class="dv">1</span>])</span>
<span id="cb34-546"><a href="#cb34-546" aria-hidden="true" tabindex="-1"></a>kraken_reports_viral_cleaned <span class="ot">&lt;-</span> kraken_reports_viral <span class="sc">%&gt;%</span></span>
<span id="cb34-547"><a href="#cb34-547" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>pc_reads_total, <span class="sc">-</span>n_reads_direct, <span class="sc">-</span><span class="fu">contains</span>(<span class="st">"minimizers"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb34-548"><a href="#cb34-548" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, taxid, p_reads_viral, n_reads_clade, <span class="fu">everything</span>()) <span class="sc">%&gt;%</span> ungroup</span>
<span id="cb34-549"><a href="#cb34-549" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-550"><a href="#cb34-550" aria-hidden="true" tabindex="-1"></a>viral_classes <span class="ot">&lt;-</span> kraken_reports_viral_cleaned <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">==</span> <span class="st">"C"</span>)</span>
<span id="cb34-551"><a href="#cb34-551" aria-hidden="true" tabindex="-1"></a>viral_families <span class="ot">&lt;-</span> kraken_reports_viral_cleaned <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">==</span> <span class="st">"F"</span>)</span>
<span id="cb34-552"><a href="#cb34-552" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-553"><a href="#cb34-553" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-554"><a href="#cb34-554" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-557"><a href="#cb34-557" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-558"><a href="#cb34-558" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: viral-family-composition</span></span>
<span id="cb34-559"><a href="#cb34-559" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb34-560"><a href="#cb34-560" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-561"><a href="#cb34-561" aria-hidden="true" tabindex="-1"></a>major_threshold <span class="ot">&lt;-</span> <span class="fl">0.3</span></span>
<span id="cb34-562"><a href="#cb34-562" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-563"><a href="#cb34-563" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify major viral families</span></span>
<span id="cb34-564"><a href="#cb34-564" aria-hidden="true" tabindex="-1"></a>viral_families_major_tab <span class="ot">&lt;-</span> viral_families <span class="sc">%&gt;%</span> </span>
<span id="cb34-565"><a href="#cb34-565" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb34-566"><a href="#cb34-566" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads_viral_max =</span> <span class="fu">max</span>(p_reads_viral), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-567"><a href="#cb34-567" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_viral_max <span class="sc">&gt;=</span> major_threshold)</span>
<span id="cb34-568"><a href="#cb34-568" aria-hidden="true" tabindex="-1"></a>viral_families_major_list <span class="ot">&lt;-</span> viral_families_major_tab <span class="sc">%&gt;%</span> <span class="fu">pull</span>(name)</span>
<span id="cb34-569"><a href="#cb34-569" aria-hidden="true" tabindex="-1"></a>viral_families_major <span class="ot">&lt;-</span> viral_families <span class="sc">%&gt;%</span> </span>
<span id="cb34-570"><a href="#cb34-570" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(name <span class="sc">%in%</span> viral_families_major_list) <span class="sc">%&gt;%</span></span>
<span id="cb34-571"><a href="#cb34-571" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, taxid, sample, p_reads_viral)</span>
<span id="cb34-572"><a href="#cb34-572" aria-hidden="true" tabindex="-1"></a>viral_families_minor <span class="ot">&lt;-</span> viral_families_major <span class="sc">%&gt;%</span> </span>
<span id="cb34-573"><a href="#cb34-573" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb34-574"><a href="#cb34-574" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads_viral_major =</span> <span class="fu">sum</span>(p_reads_viral), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-575"><a href="#cb34-575" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="st">"Other"</span>, <span class="at">taxid=</span><span class="cn">NA</span>, <span class="at">p_reads_viral =</span> <span class="dv">1</span><span class="sc">-</span>p_reads_viral_major) <span class="sc">%&gt;%</span></span>
<span id="cb34-576"><a href="#cb34-576" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, taxid, sample, p_reads_viral)</span>
<span id="cb34-577"><a href="#cb34-577" aria-hidden="true" tabindex="-1"></a>viral_families_display <span class="ot">&lt;-</span> viral_families_major <span class="sc">%&gt;%</span> </span>
<span id="cb34-578"><a href="#cb34-578" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_rows</span>(viral_families_minor) <span class="sc">%&gt;%</span></span>
<span id="cb34-579"><a href="#cb34-579" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_viral)) <span class="sc">%&gt;%</span> </span>
<span id="cb34-580"><a href="#cb34-580" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">factor</span>(name, <span class="at">levels=</span><span class="fu">c</span>(viral_families_major_list, <span class="st">"Other"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb34-581"><a href="#cb34-581" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_viral, <span class="at">classification=</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb34-582"><a href="#cb34-582" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">'sample'</span>)</span>
<span id="cb34-583"><a href="#cb34-583" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-584"><a href="#cb34-584" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-585"><a href="#cb34-585" aria-hidden="true" tabindex="-1"></a>pal <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">'Dark2'</span>),<span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">'Accent'</span>), <span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">'Set1'</span>), <span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">'Pastel1'</span>), <span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">'Pastel2'</span>))</span>
<span id="cb34-586"><a href="#cb34-586" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-587"><a href="#cb34-587" aria-hidden="true" tabindex="-1"></a>n_classifications <span class="ot">&lt;-</span> <span class="fu">length</span>(<span class="fu">unique</span>(viral_families_display<span class="sc">$</span>classification)) <span class="sc">-</span> <span class="dv">1</span></span>
<span id="cb34-588"><a href="#cb34-588" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-589"><a href="#cb34-589" aria-hidden="true" tabindex="-1"></a>custom_palette_with_black <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb34-590"><a href="#cb34-590" aria-hidden="true" tabindex="-1"></a>  pal[<span class="dv">1</span><span class="sc">:</span>n_classifications],</span>
<span id="cb34-591"><a href="#cb34-591" aria-hidden="true" tabindex="-1"></a>  <span class="st">"black"</span></span>
<span id="cb34-592"><a href="#cb34-592" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb34-593"><a href="#cb34-593" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-594"><a href="#cb34-594" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb34-595"><a href="#cb34-595" aria-hidden="true" tabindex="-1"></a>g_families <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb34-596"><a href="#cb34-596" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_comp</span>(<span class="at">data=</span>viral_families_display) <span class="sc">+</span></span>
<span id="cb34-597"><a href="#cb34-597" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Viral Reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb34-598"><a href="#cb34-598" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb34-599"><a href="#cb34-599" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb34-600"><a href="#cb34-600" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> custom_palette_with_black)</span>
<span id="cb34-601"><a href="#cb34-601" aria-hidden="true" tabindex="-1"></a>g_families</span>
<span id="cb34-602"><a href="#cb34-602" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-603"><a href="#cb34-603" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-604"><a href="#cb34-604" aria-hidden="true" tabindex="-1"></a><span class="fu"># Human-infecting virus reads</span></span>
<span id="cb34-605"><a href="#cb34-605" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-606"><a href="#cb34-606" aria-hidden="true" tabindex="-1"></a><span class="fu">## Overall relative abundance</span></span>
<span id="cb34-607"><a href="#cb34-607" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-608"><a href="#cb34-608" aria-hidden="true" tabindex="-1"></a>I calculated the relative abundance of human-infecting viruses in two ways:</span>
<span id="cb34-609"><a href="#cb34-609" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-610"><a href="#cb34-610" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>First, as the total number of deduplicated human-virus reads in each sample, divided by the number of raw reads ("All reads"). This results in a viral fraction of $1.5 \times 10^{-6}$ across all samples</span>
<span id="cb34-611"><a href="#cb34-611" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-612"><a href="#cb34-612" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Second, as a fraction of preprocessed (cleaned, deduplicated, computationally ribodepleted) reads ("Preprocessed reads"). This results in a viral fraction of $3.2 \times 10^{-6}$ across all samples.</span>
<span id="cb34-613"><a href="#cb34-613" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-616"><a href="#cb34-616" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-617"><a href="#cb34-617" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: prepare-hv</span></span>
<span id="cb34-618"><a href="#cb34-618" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb34-619"><a href="#cb34-619" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-620"><a href="#cb34-620" aria-hidden="true" tabindex="-1"></a><span class="co"># Import and format reads</span></span>
<span id="cb34-621"><a href="#cb34-621" aria-hidden="true" tabindex="-1"></a>hv_reads_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(hv_dir, <span class="st">"hv_hits_putative_collapsed.tsv.gz"</span>)</span>
<span id="cb34-622"><a href="#cb34-622" aria-hidden="true" tabindex="-1"></a>mrg_hv <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(hv_reads_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-623"><a href="#cb34-623" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb34-624"><a href="#cb34-624" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">kraken_label =</span> <span class="fu">ifelse</span>(assigned_hv, <span class="st">"Kraken2 HV assignment"</span>,</span>
<span id="cb34-625"><a href="#cb34-625" aria-hidden="true" tabindex="-1"></a>                               <span class="st">"No Kraken2 assignment"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb34-626"><a href="#cb34-626" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">adj_score_max =</span> <span class="fu">pmax</span>(adj_score_fwd, adj_score_rev),</span>
<span id="cb34-627"><a href="#cb34-627" aria-hidden="true" tabindex="-1"></a>         <span class="at">highscore =</span> adj_score_max <span class="sc">&gt;=</span> <span class="dv">20</span>,</span>
<span id="cb34-628"><a href="#cb34-628" aria-hidden="true" tabindex="-1"></a>         <span class="at">hv_status =</span> assigned_hv <span class="sc">|</span> highscore) <span class="sc">%&gt;%</span></span>
<span id="cb34-629"><a href="#cb34-629" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">taxid_all =</span> taxid, <span class="at">taxid =</span> taxid_best)</span>
<span id="cb34-630"><a href="#cb34-630" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-631"><a href="#cb34-631" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-634"><a href="#cb34-634" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-635"><a href="#cb34-635" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: count-hv-reads</span></span>
<span id="cb34-636"><a href="#cb34-636" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb34-637"><a href="#cb34-637" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb34-638"><a href="#cb34-638" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false </span></span>
<span id="cb34-639"><a href="#cb34-639" aria-hidden="true" tabindex="-1"></a><span class="co"># Get raw read counts</span></span>
<span id="cb34-640"><a href="#cb34-640" aria-hidden="true" tabindex="-1"></a>read_counts_raw <span class="ot">&lt;-</span> <span class="fu">filter</span>(basic_stats_raw) <span class="sc">%&gt;%</span></span>
<span id="cb34-641"><a href="#cb34-641" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, <span class="at">n_reads_raw =</span> n_read_pairs)</span>
<span id="cb34-642"><a href="#cb34-642" aria-hidden="true" tabindex="-1"></a>read_counts_preproc <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"ribo_initial"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-643"><a href="#cb34-643" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, <span class="at">n_reads_preproc =</span> n_read_pairs)</span>
<span id="cb34-644"><a href="#cb34-644" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-645"><a href="#cb34-645" aria-hidden="true" tabindex="-1"></a><span class="co"># Get HV read counts</span></span>
<span id="cb34-646"><a href="#cb34-646" aria-hidden="true" tabindex="-1"></a>read_counts_hv <span class="ot">&lt;-</span> mrg_hv <span class="sc">%&gt;%</span> <span class="fu">filter</span>(hv_status) <span class="sc">%&gt;%</span> </span>
<span id="cb34-647"><a href="#cb34-647" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb34-648"><a href="#cb34-648" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name=</span><span class="st">"n_reads_hv"</span>)</span>
<span id="cb34-649"><a href="#cb34-649" aria-hidden="true" tabindex="-1"></a>read_counts <span class="ot">&lt;-</span> read_counts_raw <span class="sc">%&gt;%</span></span>
<span id="cb34-650"><a href="#cb34-650" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(read_counts_hv, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"sample"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb34-651"><a href="#cb34-651" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">n_reads_hv =</span> <span class="fu">replace_na</span>(n_reads_hv, <span class="dv">0</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb34-652"><a href="#cb34-652" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(read_counts_preproc, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"sample"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb34-653"><a href="#cb34-653" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"sample"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb34-654"><a href="#cb34-654" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, n_reads_raw, n_reads_preproc, n_reads_hv) <span class="sc">%&gt;%</span></span>
<span id="cb34-655"><a href="#cb34-655" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">n_samples =</span> <span class="dv">1</span>,</span>
<span id="cb34-656"><a href="#cb34-656" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_total =</span> n_reads_hv<span class="sc">/</span>n_reads_raw,</span>
<span id="cb34-657"><a href="#cb34-657" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_preproc =</span> n_reads_hv<span class="sc">/</span>n_reads_preproc)</span>
<span id="cb34-658"><a href="#cb34-658" aria-hidden="true" tabindex="-1"></a>read_counts_long <span class="ot">&lt;-</span> read_counts <span class="sc">%&gt;%</span></span>
<span id="cb34-659"><a href="#cb34-659" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="fu">starts_with</span>(<span class="st">"p_reads"</span>), <span class="at">names_to=</span><span class="st">"read_group"</span>, <span class="at">values_to=</span><span class="st">"p_reads"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-660"><a href="#cb34-660" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">read_group =</span> <span class="fu">ifelse</span>(read_group <span class="sc">==</span> <span class="st">"p_reads_total"</span>, <span class="st">"All reads"</span>, <span class="st">"Preprocessed reads"</span>))</span>
<span id="cb34-661"><a href="#cb34-661" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-662"><a href="#cb34-662" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine for display</span></span>
<span id="cb34-663"><a href="#cb34-663" aria-hidden="true" tabindex="-1"></a>read_counts_agg <span class="ot">&lt;-</span> read_counts <span class="sc">%&gt;%</span></span>
<span id="cb34-664"><a href="#cb34-664" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_total =</span> n_reads_hv<span class="sc">/</span>n_reads_raw,</span>
<span id="cb34-665"><a href="#cb34-665" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_preproc =</span> n_reads_hv<span class="sc">/</span>n_reads_preproc) <span class="sc">%&gt;%</span></span>
<span id="cb34-666"><a href="#cb34-666" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"sample"</span>))</span>
<span id="cb34-667"><a href="#cb34-667" aria-hidden="true" tabindex="-1"></a>read_counts_agg_long <span class="ot">&lt;-</span> read_counts_agg <span class="sc">%&gt;%</span></span>
<span id="cb34-668"><a href="#cb34-668" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="fu">starts_with</span>(<span class="st">"p_reads"</span>), <span class="at">names_to=</span><span class="st">"read_group"</span>, <span class="at">values_to=</span><span class="st">"p_reads"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-669"><a href="#cb34-669" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">read_group =</span> <span class="fu">ifelse</span>(read_group <span class="sc">==</span> <span class="st">"p_reads_total"</span>, <span class="st">"All reads (HV)"</span>, <span class="st">"Preprocessed reads (HV)"</span>)) </span>
<span id="cb34-670"><a href="#cb34-670" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-671"><a href="#cb34-671" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize</span></span>
<span id="cb34-672"><a href="#cb34-672" aria-hidden="true" tabindex="-1"></a>g_read_counts <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(read_counts_agg_long, <span class="fu">aes</span>(<span class="at">x=</span>library, <span class="at">y=</span>p_reads, <span class="at">color =</span> read_group, <span class="at">group =</span> library)) <span class="sc">+</span></span>
<span id="cb34-673"><a href="#cb34-673" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb34-674"><a href="#cb34-674" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">color =</span> <span class="st">'black'</span>) <span class="sc">+</span></span>
<span id="cb34-675"><a href="#cb34-675" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>(<span class="at">name =</span> <span class="st">"Unique human-viral read fraction"</span>, <span class="at">labels =</span> <span class="fu">label_log</span>(<span class="at">digits=</span><span class="dv">2</span>), <span class="at">limits =</span> <span class="fu">c</span>(<span class="fl">1e-6</span>, <span class="fl">1e-2</span>)) <span class="sc">+</span></span>
<span id="cb34-676"><a href="#cb34-676" aria-hidden="true" tabindex="-1"></a>  theme_kit <span class="sc">+</span> </span>
<span id="cb34-677"><a href="#cb34-677" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>()</span>
<span id="cb34-678"><a href="#cb34-678" aria-hidden="true" tabindex="-1"></a>g_read_counts</span>
<span id="cb34-679"><a href="#cb34-679" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-680"><a href="#cb34-680" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-681"><a href="#cb34-681" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-682"><a href="#cb34-682" aria-hidden="true" tabindex="-1"></a><span class="fu">## Overall taxonomy and composition</span></span>
<span id="cb34-683"><a href="#cb34-683" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-684"><a href="#cb34-684" aria-hidden="true" tabindex="-1"></a>Composition of HV reads was changed from when looking at all viral reads. First, only 15% of the individuals in this dataset had human infecting viruses detectable in their blood. Second, individuals seem to be dominated by a single viral family. The threshold for the label "other" are the set of families that make up less than 5% composition in all samples.</span>
<span id="cb34-685"><a href="#cb34-685" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-688"><a href="#cb34-688" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-689"><a href="#cb34-689" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: raise-hv-taxa</span></span>
<span id="cb34-690"><a href="#cb34-690" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb34-691"><a href="#cb34-691" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-692"><a href="#cb34-692" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter samples and add viral taxa information</span></span>
<span id="cb34-693"><a href="#cb34-693" aria-hidden="true" tabindex="-1"></a>samples_keep <span class="ot">&lt;-</span> read_counts <span class="sc">%&gt;%</span> <span class="fu">filter</span>(n_reads_hv <span class="sc">&gt;</span> <span class="dv">5</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(sample)</span>
<span id="cb34-694"><a href="#cb34-694" aria-hidden="true" tabindex="-1"></a>mrg_hv_named <span class="ot">&lt;-</span> mrg_hv <span class="sc">%&gt;%</span> <span class="fu">filter</span>(sample <span class="sc">%in%</span> samples_keep, hv_status) <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(viral_taxa, <span class="at">by=</span><span class="st">"taxid"</span>) </span>
<span id="cb34-695"><a href="#cb34-695" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-696"><a href="#cb34-696" aria-hidden="true" tabindex="-1"></a><span class="co"># Discover viral species &amp; genera for HV reads</span></span>
<span id="cb34-697"><a href="#cb34-697" aria-hidden="true" tabindex="-1"></a>raise_rank <span class="ot">&lt;-</span> <span class="cf">function</span>(read_db, taxid_db, <span class="at">out_rank =</span> <span class="st">"species"</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>){</span>
<span id="cb34-698"><a href="#cb34-698" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get higher ranks than search rank</span></span>
<span id="cb34-699"><a href="#cb34-699" aria-hidden="true" tabindex="-1"></a>  ranks <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"subspecies"</span>, <span class="st">"species"</span>, <span class="st">"subgenus"</span>, <span class="st">"genus"</span>, <span class="st">"subfamily"</span>, <span class="st">"family"</span>, <span class="st">"suborder"</span>, <span class="st">"order"</span>, <span class="st">"class"</span>, <span class="st">"subphylum"</span>, <span class="st">"phylum"</span>, <span class="st">"kingdom"</span>, <span class="st">"superkingdom"</span>)</span>
<span id="cb34-700"><a href="#cb34-700" aria-hidden="true" tabindex="-1"></a>  rank_match <span class="ot">&lt;-</span> <span class="fu">which.max</span>(ranks <span class="sc">==</span> out_rank)</span>
<span id="cb34-701"><a href="#cb34-701" aria-hidden="true" tabindex="-1"></a>  high_ranks <span class="ot">&lt;-</span> ranks[rank_match<span class="sc">:</span><span class="fu">length</span>(ranks)]</span>
<span id="cb34-702"><a href="#cb34-702" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Merge read DB and taxid DB</span></span>
<span id="cb34-703"><a href="#cb34-703" aria-hidden="true" tabindex="-1"></a>  reads <span class="ot">&lt;-</span> read_db <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>parent_taxid, <span class="sc">-</span>rank, <span class="sc">-</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb34-704"><a href="#cb34-704" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(taxid_db, <span class="at">by=</span><span class="st">"taxid"</span>)</span>
<span id="cb34-705"><a href="#cb34-705" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract sequences that are already at appropriate rank</span></span>
<span id="cb34-706"><a href="#cb34-706" aria-hidden="true" tabindex="-1"></a>  reads_rank <span class="ot">&lt;-</span> <span class="fu">filter</span>(reads, rank <span class="sc">==</span> out_rank)</span>
<span id="cb34-707"><a href="#cb34-707" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Drop sequences at a higher rank and return unclassified sequences</span></span>
<span id="cb34-708"><a href="#cb34-708" aria-hidden="true" tabindex="-1"></a>  reads_norank <span class="ot">&lt;-</span> reads <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">!=</span> out_rank, <span class="sc">!</span>rank <span class="sc">%in%</span> high_ranks, <span class="sc">!</span><span class="fu">is.na</span>(taxid))</span>
<span id="cb34-709"><a href="#cb34-709" aria-hidden="true" tabindex="-1"></a>  <span class="cf">while</span>(<span class="fu">nrow</span>(reads_norank) <span class="sc">&gt;</span> <span class="dv">0</span>){ <span class="co"># As long as there are unclassified sequences...</span></span>
<span id="cb34-710"><a href="#cb34-710" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Promote read taxids and re-merge with taxid DB, then re-classify and filter</span></span>
<span id="cb34-711"><a href="#cb34-711" aria-hidden="true" tabindex="-1"></a>    reads_remaining <span class="ot">&lt;-</span> reads_norank <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">taxid =</span> parent_taxid) <span class="sc">%&gt;%</span></span>
<span id="cb34-712"><a href="#cb34-712" aria-hidden="true" tabindex="-1"></a>      <span class="fu">select</span>(<span class="sc">-</span>parent_taxid, <span class="sc">-</span>rank, <span class="sc">-</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb34-713"><a href="#cb34-713" aria-hidden="true" tabindex="-1"></a>      <span class="fu">left_join</span>(taxid_db, <span class="at">by=</span><span class="st">"taxid"</span>)</span>
<span id="cb34-714"><a href="#cb34-714" aria-hidden="true" tabindex="-1"></a>    reads_rank <span class="ot">&lt;-</span> reads_remaining <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">==</span> out_rank) <span class="sc">%&gt;%</span></span>
<span id="cb34-715"><a href="#cb34-715" aria-hidden="true" tabindex="-1"></a>      <span class="fu">bind_rows</span>(reads_rank)</span>
<span id="cb34-716"><a href="#cb34-716" aria-hidden="true" tabindex="-1"></a>    reads_norank <span class="ot">&lt;-</span> reads_remaining <span class="sc">%&gt;%</span></span>
<span id="cb34-717"><a href="#cb34-717" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(rank <span class="sc">!=</span> out_rank, <span class="sc">!</span>rank <span class="sc">%in%</span> high_ranks, <span class="sc">!</span><span class="fu">is.na</span>(taxid))</span>
<span id="cb34-718"><a href="#cb34-718" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb34-719"><a href="#cb34-719" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Finally, extract and append reads that were excluded during the process</span></span>
<span id="cb34-720"><a href="#cb34-720" aria-hidden="true" tabindex="-1"></a>  reads_dropped <span class="ot">&lt;-</span> reads <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span>seq_id <span class="sc">%in%</span> reads_rank<span class="sc">$</span>seq_id)</span>
<span id="cb34-721"><a href="#cb34-721" aria-hidden="true" tabindex="-1"></a>  reads_out <span class="ot">&lt;-</span> reads_rank <span class="sc">%&gt;%</span> <span class="fu">bind_rows</span>(reads_dropped) <span class="sc">%&gt;%</span></span>
<span id="cb34-722"><a href="#cb34-722" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(<span class="sc">-</span>parent_taxid, <span class="sc">-</span>rank, <span class="sc">-</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb34-723"><a href="#cb34-723" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(taxid_db, <span class="at">by=</span><span class="st">"taxid"</span>)</span>
<span id="cb34-724"><a href="#cb34-724" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(reads_out)</span>
<span id="cb34-725"><a href="#cb34-725" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb34-726"><a href="#cb34-726" aria-hidden="true" tabindex="-1"></a>hv_reads_species <span class="ot">&lt;-</span> <span class="fu">raise_rank</span>(mrg_hv_named, viral_taxa, <span class="st">"species"</span>)</span>
<span id="cb34-727"><a href="#cb34-727" aria-hidden="true" tabindex="-1"></a>hv_reads_genus <span class="ot">&lt;-</span> <span class="fu">raise_rank</span>(mrg_hv_named, viral_taxa, <span class="st">"genus"</span>)</span>
<span id="cb34-728"><a href="#cb34-728" aria-hidden="true" tabindex="-1"></a>hv_reads_family <span class="ot">&lt;-</span> <span class="fu">raise_rank</span>(mrg_hv_named, viral_taxa, <span class="st">"family"</span>)</span>
<span id="cb34-729"><a href="#cb34-729" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-730"><a href="#cb34-730" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-733"><a href="#cb34-733" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-734"><a href="#cb34-734" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-family</span></span>
<span id="cb34-735"><a href="#cb34-735" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb34-736"><a href="#cb34-736" aria-hidden="true" tabindex="-1"></a>threshold_major_family <span class="ot">&lt;-</span> <span class="fl">0.05</span></span>
<span id="cb34-737"><a href="#cb34-737" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-738"><a href="#cb34-738" aria-hidden="true" tabindex="-1"></a><span class="co"># Count reads for each human-viral family</span></span>
<span id="cb34-739"><a href="#cb34-739" aria-hidden="true" tabindex="-1"></a>hv_family_counts <span class="ot">&lt;-</span> hv_reads_family <span class="sc">%&gt;%</span> </span>
<span id="cb34-740"><a href="#cb34-740" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb34-741"><a href="#cb34-741" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-742"><a href="#cb34-742" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb34-743"><a href="#cb34-743" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_hv =</span> n_reads_hv<span class="sc">/</span><span class="fu">sum</span>(n_reads_hv))</span>
<span id="cb34-744"><a href="#cb34-744" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-745"><a href="#cb34-745" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify high-ranking families and group others</span></span>
<span id="cb34-746"><a href="#cb34-746" aria-hidden="true" tabindex="-1"></a>hv_family_major_tab <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span> </span>
<span id="cb34-747"><a href="#cb34-747" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_hv <span class="sc">==</span> <span class="fu">max</span>(p_reads_hv)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-748"><a href="#cb34-748" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_hv)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(p_reads_hv <span class="sc">&gt;</span> threshold_major_family)</span>
<span id="cb34-749"><a href="#cb34-749" aria-hidden="true" tabindex="-1"></a>hv_family_counts_major <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span></span>
<span id="cb34-750"><a href="#cb34-750" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">ifelse</span>(name <span class="sc">%in%</span> hv_family_major_tab<span class="sc">$</span>name, name, <span class="st">"Other"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb34-751"><a href="#cb34-751" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, name_display) <span class="sc">%&gt;%</span></span>
<span id="cb34-752"><a href="#cb34-752" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_hv =</span> <span class="fu">sum</span>(n_reads_hv), <span class="at">p_reads_hv =</span> <span class="fu">sum</span>(p_reads_hv), </span>
<span id="cb34-753"><a href="#cb34-753" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-754"><a href="#cb34-754" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">factor</span>(name_display, </span>
<span id="cb34-755"><a href="#cb34-755" aria-hidden="true" tabindex="-1"></a>                               <span class="at">levels =</span> <span class="fu">c</span>(hv_family_major_tab<span class="sc">$</span>name, <span class="st">"Other"</span>)))</span>
<span id="cb34-756"><a href="#cb34-756" aria-hidden="true" tabindex="-1"></a>hv_family_counts_display <span class="ot">&lt;-</span> hv_family_counts_major <span class="sc">%&gt;%</span></span>
<span id="cb34-757"><a href="#cb34-757" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_hv, <span class="at">classification =</span> name_display) <span class="sc">%&gt;%</span></span>
<span id="cb34-758"><a href="#cb34-758" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by =</span> <span class="st">'sample'</span>)</span>
<span id="cb34-759"><a href="#cb34-759" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-760"><a href="#cb34-760" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb34-761"><a href="#cb34-761" aria-hidden="true" tabindex="-1"></a>g_hv_family <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb34-762"><a href="#cb34-762" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">data=</span>hv_family_counts_display, <span class="at">position =</span> <span class="st">"stack"</span>, <span class="at">width=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb34-763"><a href="#cb34-763" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% HV Reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb34-764"><a href="#cb34-764" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb34-765"><a href="#cb34-765" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb34-766"><a href="#cb34-766" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">'Accent'</span>, <span class="at">name =</span> <span class="st">"Viral family"</span>) <span class="sc">+</span></span>
<span id="cb34-767"><a href="#cb34-767" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">"Family composition of human-viral reads"</span>) <span class="sc">+</span></span>
<span id="cb34-768"><a href="#cb34-768" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">4</span>)) <span class="sc">+</span></span>
<span id="cb34-769"><a href="#cb34-769" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">1.4</span>), <span class="at">hjust=</span><span class="dv">0</span>, <span class="at">face=</span><span class="st">"plain"</span>))</span>
<span id="cb34-770"><a href="#cb34-770" aria-hidden="true" tabindex="-1"></a>g_hv_family</span>
<span id="cb34-771"><a href="#cb34-771" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-772"><a href="#cb34-772" aria-hidden="true" tabindex="-1"></a><span class="co"># Get most prominent families for text</span></span>
<span id="cb34-773"><a href="#cb34-773" aria-hidden="true" tabindex="-1"></a>hv_family_collate <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span></span>
<span id="cb34-774"><a href="#cb34-774" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb34-775"><a href="#cb34-775" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_tot =</span> <span class="fu">sum</span>(n_reads_hv),</span>
<span id="cb34-776"><a href="#cb34-776" aria-hidden="true" tabindex="-1"></a>            <span class="at">p_reads_max =</span> <span class="fu">max</span>(p_reads_hv), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb34-777"><a href="#cb34-777" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n_reads_tot))</span>
<span id="cb34-778"><a href="#cb34-778" aria-hidden="true" tabindex="-1"></a>hv_family_collate</span>
<span id="cb34-779"><a href="#cb34-779" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-780"><a href="#cb34-780" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-781"><a href="#cb34-781" aria-hidden="true" tabindex="-1"></a><span class="fu">## Analyzing specific families</span></span>
<span id="cb34-782"><a href="#cb34-782" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-783"><a href="#cb34-783" aria-hidden="true" tabindex="-1"></a>We now investigate the composition of specific families that had more than 5 viral reads. In investigating individual viral families, to avoid distortions from a few rare reads, I restricted myself to samples where that family made up at least 1% of human-viral reads:</span>
<span id="cb34-784"><a href="#cb34-784" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-787"><a href="#cb34-787" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-788"><a href="#cb34-788" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-viral-family-scripts</span></span>
<span id="cb34-789"><a href="#cb34-789" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb34-790"><a href="#cb34-790" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-791"><a href="#cb34-791" aria-hidden="true" tabindex="-1"></a>plot_viral_family_composition <span class="ot">&lt;-</span> <span class="cf">function</span>(taxid_chosen, <span class="at">threshold_major_species =</span> <span class="fl">0.1</span>) {</span>
<span id="cb34-792"><a href="#cb34-792" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get set of chosen family reads</span></span>
<span id="cb34-793"><a href="#cb34-793" aria-hidden="true" tabindex="-1"></a>  family_samples <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span> <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_chosen) <span class="sc">%&gt;%</span></span>
<span id="cb34-794"><a href="#cb34-794" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(p_reads_hv <span class="sc">&gt;=</span> <span class="fl">0.1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-795"><a href="#cb34-795" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pull</span>(sample)</span>
<span id="cb34-796"><a href="#cb34-796" aria-hidden="true" tabindex="-1"></a>  family_ids <span class="ot">&lt;-</span> hv_reads_family <span class="sc">%&gt;%</span> </span>
<span id="cb34-797"><a href="#cb34-797" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_chosen, sample <span class="sc">%in%</span> family_samples) <span class="sc">%&gt;%</span></span>
<span id="cb34-798"><a href="#cb34-798" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pull</span>(seq_id)</span>
<span id="cb34-799"><a href="#cb34-799" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-800"><a href="#cb34-800" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Count reads for each species in the chosen family</span></span>
<span id="cb34-801"><a href="#cb34-801" aria-hidden="true" tabindex="-1"></a>  species_counts <span class="ot">&lt;-</span> hv_reads_species <span class="sc">%&gt;%</span></span>
<span id="cb34-802"><a href="#cb34-802" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(seq_id <span class="sc">%in%</span> family_ids) <span class="sc">%&gt;%</span></span>
<span id="cb34-803"><a href="#cb34-803" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(sample, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb34-804"><a href="#cb34-804" aria-hidden="true" tabindex="-1"></a>    <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-805"><a href="#cb34-805" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb34-806"><a href="#cb34-806" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">p_reads_family =</span> n_reads_hv<span class="sc">/</span><span class="fu">sum</span>(n_reads_hv))</span>
<span id="cb34-807"><a href="#cb34-807" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-808"><a href="#cb34-808" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Identify high-ranking species and group others</span></span>
<span id="cb34-809"><a href="#cb34-809" aria-hidden="true" tabindex="-1"></a>  species_major_tab <span class="ot">&lt;-</span> species_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span> </span>
<span id="cb34-810"><a href="#cb34-810" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(p_reads_family <span class="sc">==</span> <span class="fu">max</span>(p_reads_family)) <span class="sc">%&gt;%</span> </span>
<span id="cb34-811"><a href="#cb34-811" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-812"><a href="#cb34-812" aria-hidden="true" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_family)) <span class="sc">%&gt;%</span> </span>
<span id="cb34-813"><a href="#cb34-813" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(p_reads_family <span class="sc">&gt;</span> threshold_major_species)</span>
<span id="cb34-814"><a href="#cb34-814" aria-hidden="true" tabindex="-1"></a>  species_counts_major <span class="ot">&lt;-</span> species_counts <span class="sc">%&gt;%</span></span>
<span id="cb34-815"><a href="#cb34-815" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">ifelse</span>(name <span class="sc">%in%</span> species_major_tab<span class="sc">$</span>name, </span>
<span id="cb34-816"><a href="#cb34-816" aria-hidden="true" tabindex="-1"></a>                                 name, <span class="st">"Other"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb34-817"><a href="#cb34-817" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(sample, name_display) <span class="sc">%&gt;%</span></span>
<span id="cb34-818"><a href="#cb34-818" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarize</span>(<span class="at">n_reads_family =</span> <span class="fu">sum</span>(n_reads_hv),</span>
<span id="cb34-819"><a href="#cb34-819" aria-hidden="true" tabindex="-1"></a>              <span class="at">p_reads_family =</span> <span class="fu">sum</span>(p_reads_family), </span>
<span id="cb34-820"><a href="#cb34-820" aria-hidden="true" tabindex="-1"></a>              <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-821"><a href="#cb34-821" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">factor</span>(name_display, </span>
<span id="cb34-822"><a href="#cb34-822" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">levels =</span> <span class="fu">c</span>(species_major_tab<span class="sc">$</span>name, <span class="st">"Other"</span>)))</span>
<span id="cb34-823"><a href="#cb34-823" aria-hidden="true" tabindex="-1"></a>  species_counts_display <span class="ot">&lt;-</span> species_counts_major <span class="sc">%&gt;%</span></span>
<span id="cb34-824"><a href="#cb34-824" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_family, <span class="at">classification =</span> name_display) <span class="sc">%&gt;%</span></span>
<span id="cb34-825"><a href="#cb34-825" aria-hidden="true" tabindex="-1"></a>    <span class="fu">right_join</span>(libraries, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">'sample'</span>))</span>
<span id="cb34-826"><a href="#cb34-826" aria-hidden="true" tabindex="-1"></a>    <span class="co">#mutate(p_reads = coalesce(p_reads, 0),</span></span>
<span id="cb34-827"><a href="#cb34-827" aria-hidden="true" tabindex="-1"></a>    <span class="co">#       classification = coalesce(classification, species_major_tab$name[1])) %&gt;%</span></span>
<span id="cb34-828"><a href="#cb34-828" aria-hidden="true" tabindex="-1"></a>    <span class="co">#group_by(location) %&gt;%</span></span>
<span id="cb34-829"><a href="#cb34-829" aria-hidden="true" tabindex="-1"></a>    <span class="co">#mutate(samples_with_reads = sum(p_reads &gt; 0),</span></span>
<span id="cb34-830"><a href="#cb34-830" aria-hidden="true" tabindex="-1"></a>    <span class="co">#  label = paste0(location, " (", sprintf("%.1f%%", 100 * samples_with_reads / total_samples), ")")) %&gt;%</span></span>
<span id="cb34-831"><a href="#cb34-831" aria-hidden="true" tabindex="-1"></a>    <span class="co">#ungroup()</span></span>
<span id="cb34-832"><a href="#cb34-832" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-833"><a href="#cb34-833" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get family name for the title</span></span>
<span id="cb34-834"><a href="#cb34-834" aria-hidden="true" tabindex="-1"></a>  family_name <span class="ot">&lt;-</span> viral_taxa <span class="sc">%&gt;%</span> </span>
<span id="cb34-835"><a href="#cb34-835" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_chosen) <span class="sc">%&gt;%</span> </span>
<span id="cb34-836"><a href="#cb34-836" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pull</span>(name)</span>
<span id="cb34-837"><a href="#cb34-837" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-838"><a href="#cb34-838" aria-hidden="true" tabindex="-1"></a>  species_count <span class="ot">&lt;-</span> species_counts_display <span class="sc">%&gt;%</span> <span class="fu">select</span>(classification) <span class="sc">%&gt;%</span> <span class="fu">distinct</span>() <span class="sc">%&gt;%</span> <span class="fu">pull</span>() <span class="sc">%&gt;%</span> <span class="fu">length</span>()</span>
<span id="cb34-839"><a href="#cb34-839" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (species_count <span class="sc">&lt;=</span> <span class="dv">8</span>) {</span>
<span id="cb34-840"><a href="#cb34-840" aria-hidden="true" tabindex="-1"></a>    pal <span class="ot">&lt;-</span> <span class="fu">brewer.pal</span>(<span class="at">name =</span> <span class="st">"Accent"</span>, <span class="at">n =</span> species_count)</span>
<span id="cb34-841"><a href="#cb34-841" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb34-842"><a href="#cb34-842" aria-hidden="true" tabindex="-1"></a>    pal <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">brewer.pal</span>(<span class="at">name =</span> <span class="st">"Paired"</span>, <span class="at">n =</span> <span class="dv">12</span>), <span class="fu">brewer.pal</span>(<span class="at">name =</span> <span class="st">"Set3"</span>, <span class="at">n =</span><span class="dv">12</span>), <span class="fu">brewer.pal</span>(<span class="at">name =</span> <span class="st">"Set1"</span>, <span class="at">n =</span> <span class="dv">9</span>))[<span class="dv">1</span><span class="sc">:</span>species_count]</span>
<span id="cb34-843"><a href="#cb34-843" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb34-844"><a href="#cb34-844" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-845"><a href="#cb34-845" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Plot</span></span>
<span id="cb34-846"><a href="#cb34-846" aria-hidden="true" tabindex="-1"></a>  g_species <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb34-847"><a href="#cb34-847" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_col</span>(<span class="at">data=</span>species_counts_display, <span class="at">position =</span> <span class="st">"stack"</span>) <span class="sc">+</span></span>
<span id="cb34-848"><a href="#cb34-848" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="fu">paste0</span>(<span class="st">"% "</span>, family_name, <span class="st">" reads"</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb34-849"><a href="#cb34-849" aria-hidden="true" tabindex="-1"></a>                       <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb34-850"><a href="#cb34-850" aria-hidden="true" tabindex="-1"></a>                       <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb34-851"><a href="#cb34-851" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> pal, <span class="at">name =</span> <span class="st">"Viral species"</span>) <span class="sc">+</span></span>
<span id="cb34-852"><a href="#cb34-852" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title=</span><span class="fu">paste0</span>(<span class="st">"Species composition of "</span>, family_name, <span class="st">" reads"</span>)) <span class="sc">+</span></span>
<span id="cb34-853"><a href="#cb34-853" aria-hidden="true" tabindex="-1"></a>    <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">3</span>)) <span class="sc">+</span></span>
<span id="cb34-854"><a href="#cb34-854" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">1.4</span>), <span class="at">hjust=</span><span class="dv">0</span>, <span class="at">face=</span><span class="st">"plain"</span>))</span>
<span id="cb34-855"><a href="#cb34-855" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(g_species)</span>
<span id="cb34-856"><a href="#cb34-856" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb34-857"><a href="#cb34-857" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-858"><a href="#cb34-858" aria-hidden="true" tabindex="-1"></a>plot_viral_family_histogram <span class="ot">&lt;-</span> <span class="cf">function</span>(taxid_chosen) {</span>
<span id="cb34-859"><a href="#cb34-859" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filter data for the specified virus family</span></span>
<span id="cb34-860"><a href="#cb34-860" aria-hidden="true" tabindex="-1"></a>  family_data <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span></span>
<span id="cb34-861"><a href="#cb34-861" aria-hidden="true" tabindex="-1"></a>    <span class="fu">inner_join</span>(libraries, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">'sample'</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb34-862"><a href="#cb34-862" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_chosen)</span>
<span id="cb34-863"><a href="#cb34-863" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb34-864"><a href="#cb34-864" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get family name for the title</span></span>
<span id="cb34-865"><a href="#cb34-865" aria-hidden="true" tabindex="-1"></a>  family_name <span class="ot">&lt;-</span> viral_taxa <span class="sc">%&gt;%</span> </span>
<span id="cb34-866"><a href="#cb34-866" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_chosen) <span class="sc">%&gt;%</span> </span>
<span id="cb34-867"><a href="#cb34-867" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pull</span>(name)</span>
<span id="cb34-868"><a href="#cb34-868" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb34-869"><a href="#cb34-869" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create the histogram</span></span>
<span id="cb34-870"><a href="#cb34-870" aria-hidden="true" tabindex="-1"></a>  g_histogram <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(family_data, <span class="fu">aes</span>(<span class="at">x =</span> library, <span class="at">y =</span> n_reads_hv)) <span class="sc">+</span></span>
<span id="cb34-871"><a href="#cb34-871" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb34-872"><a href="#cb34-872" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label =</span> scales<span class="sc">::</span><span class="fu">comma</span>(n_reads_hv)), <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb34-873"><a href="#cb34-873" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb34-874"><a href="#cb34-874" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb34-875"><a href="#cb34-875" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_discrete</span>(<span class="at">limits =</span> libraries <span class="sc">%&gt;%</span> <span class="fu">pull</span>(library)) <span class="sc">+</span></span>
<span id="cb34-876"><a href="#cb34-876" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(</span>
<span id="cb34-877"><a href="#cb34-877" aria-hidden="true" tabindex="-1"></a>      <span class="at">title =</span> <span class="fu">paste</span>(<span class="st">"Distribution of"</span>, family_name, <span class="st">"Reads by Location"</span>),</span>
<span id="cb34-878"><a href="#cb34-878" aria-hidden="true" tabindex="-1"></a>      <span class="at">x =</span> <span class="st">"Location"</span>,</span>
<span id="cb34-879"><a href="#cb34-879" aria-hidden="true" tabindex="-1"></a>      <span class="at">y =</span> <span class="st">"Number of reads"</span></span>
<span id="cb34-880"><a href="#cb34-880" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb34-881"><a href="#cb34-881" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_continuous</span>(<span class="at">labels =</span> scales<span class="sc">::</span>comma) <span class="sc">+</span></span>
<span id="cb34-882"><a href="#cb34-882" aria-hidden="true" tabindex="-1"></a>    <span class="fu">expand_limits</span>(<span class="at">y =</span> <span class="fu">max</span>(family_data<span class="sc">$</span>n_reads_hv) <span class="sc">*</span> <span class="fl">1.1</span>)<span class="sc">+</span></span>
<span id="cb34-883"><a href="#cb34-883" aria-hidden="true" tabindex="-1"></a>    theme_kit  <span class="co"># Expand y-axis to make room for labels</span></span>
<span id="cb34-884"><a href="#cb34-884" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(g_histogram)</span>
<span id="cb34-885"><a href="#cb34-885" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb34-886"><a href="#cb34-886" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-887"><a href="#cb34-887" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-888"><a href="#cb34-888" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-889"><a href="#cb34-889" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-890"><a href="#cb34-890" aria-hidden="true" tabindex="-1"></a><span class="fu">### Flaviviridae (Number of reads: 25,997)</span></span>
<span id="cb34-893"><a href="#cb34-893" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-894"><a href="#cb34-894" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-flaviviridae-histogram</span></span>
<span id="cb34-895"><a href="#cb34-895" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb34-896"><a href="#cb34-896" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 2.5</span></span>
<span id="cb34-897"><a href="#cb34-897" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-898"><a href="#cb34-898" aria-hidden="true" tabindex="-1"></a><span class="co"># Flaviviridae</span></span>
<span id="cb34-899"><a href="#cb34-899" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_histogram</span>(<span class="at">taxid_chosen=</span><span class="dv">11050</span>)</span>
<span id="cb34-900"><a href="#cb34-900" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-903"><a href="#cb34-903" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-904"><a href="#cb34-904" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-flaviviridae-composition</span></span>
<span id="cb34-905"><a href="#cb34-905" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb34-906"><a href="#cb34-906" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 7.5</span></span>
<span id="cb34-907"><a href="#cb34-907" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb34-908"><a href="#cb34-908" aria-hidden="true" tabindex="-1"></a><span class="co"># Flaviviridae</span></span>
<span id="cb34-909"><a href="#cb34-909" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_composition</span>(<span class="at">taxid_chosen=</span><span class="dv">11050</span>, <span class="at">threshold_major_species =</span> <span class="fl">0.01</span>)</span>
<span id="cb34-910"><a href="#cb34-910" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-911"><a href="#cb34-911" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-912"><a href="#cb34-912" aria-hidden="true" tabindex="-1"></a><span class="fu">### Anelloviridae (Number of reads: 10,103)</span></span>
<span id="cb34-915"><a href="#cb34-915" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-916"><a href="#cb34-916" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-anelloviridae-histogram</span></span>
<span id="cb34-917"><a href="#cb34-917" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb34-918"><a href="#cb34-918" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 2.5</span></span>
<span id="cb34-919"><a href="#cb34-919" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-920"><a href="#cb34-920" aria-hidden="true" tabindex="-1"></a><span class="co"># Anelloviridae</span></span>
<span id="cb34-921"><a href="#cb34-921" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_histogram</span>(<span class="at">taxid_chosen=</span><span class="dv">687329</span>)</span>
<span id="cb34-922"><a href="#cb34-922" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-925"><a href="#cb34-925" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-926"><a href="#cb34-926" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-anelloviridae-composition</span></span>
<span id="cb34-927"><a href="#cb34-927" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb34-928"><a href="#cb34-928" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 7.5</span></span>
<span id="cb34-929"><a href="#cb34-929" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb34-930"><a href="#cb34-930" aria-hidden="true" tabindex="-1"></a><span class="co"># Anelloviridae</span></span>
<span id="cb34-931"><a href="#cb34-931" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_composition</span>(<span class="at">taxid_chosen=</span><span class="dv">687329</span>, <span class="at">threshold_major_species =</span> <span class="fl">0.01</span>)</span>
<span id="cb34-932"><a href="#cb34-932" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-933"><a href="#cb34-933" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-934"><a href="#cb34-934" aria-hidden="true" tabindex="-1"></a><span class="fu">### Coronaviridae (Number of reads: 1,104)</span></span>
<span id="cb34-937"><a href="#cb34-937" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-938"><a href="#cb34-938" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-coronaviridae-histogram</span></span>
<span id="cb34-939"><a href="#cb34-939" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb34-940"><a href="#cb34-940" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 2.5</span></span>
<span id="cb34-941"><a href="#cb34-941" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-942"><a href="#cb34-942" aria-hidden="true" tabindex="-1"></a><span class="co"># Coronaviridae</span></span>
<span id="cb34-943"><a href="#cb34-943" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_histogram</span>(<span class="at">taxid_chosen=</span><span class="dv">11118</span>)</span>
<span id="cb34-944"><a href="#cb34-944" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-947"><a href="#cb34-947" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-948"><a href="#cb34-948" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-coronaviridae-composition</span></span>
<span id="cb34-949"><a href="#cb34-949" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb34-950"><a href="#cb34-950" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 7.5</span></span>
<span id="cb34-951"><a href="#cb34-951" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb34-952"><a href="#cb34-952" aria-hidden="true" tabindex="-1"></a><span class="co"># Coronaviridae</span></span>
<span id="cb34-953"><a href="#cb34-953" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_composition</span>(<span class="at">taxid_chosen=</span><span class="dv">11118</span>, <span class="at">threshold_major_species =</span> <span class="fl">0.01</span>)</span>
<span id="cb34-954"><a href="#cb34-954" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-955"><a href="#cb34-955" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-956"><a href="#cb34-956" aria-hidden="true" tabindex="-1"></a><span class="fu">### Orthoherpesviridae (Number of reads: 534)</span></span>
<span id="cb34-959"><a href="#cb34-959" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-960"><a href="#cb34-960" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-orthoherpesviridae-histogram</span></span>
<span id="cb34-961"><a href="#cb34-961" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb34-962"><a href="#cb34-962" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 2.5</span></span>
<span id="cb34-963"><a href="#cb34-963" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-964"><a href="#cb34-964" aria-hidden="true" tabindex="-1"></a><span class="co"># Orthoherpesviridae</span></span>
<span id="cb34-965"><a href="#cb34-965" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_histogram</span>(<span class="at">taxid_chosen=</span><span class="dv">3044472</span>)</span>
<span id="cb34-966"><a href="#cb34-966" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-969"><a href="#cb34-969" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-970"><a href="#cb34-970" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-orthoherpesviridae-composition</span></span>
<span id="cb34-971"><a href="#cb34-971" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb34-972"><a href="#cb34-972" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 7.5</span></span>
<span id="cb34-973"><a href="#cb34-973" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb34-974"><a href="#cb34-974" aria-hidden="true" tabindex="-1"></a><span class="co"># Orthoherpesviridae</span></span>
<span id="cb34-975"><a href="#cb34-975" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_composition</span>(<span class="at">taxid_chosen=</span><span class="dv">3044472</span>, <span class="at">threshold_major_species =</span> <span class="fl">0.01</span>)</span>
<span id="cb34-976"><a href="#cb34-976" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-977"><a href="#cb34-977" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-978"><a href="#cb34-978" aria-hidden="true" tabindex="-1"></a><span class="fu">### Parvoviridae (Number of reads: 335) </span></span>
<span id="cb34-981"><a href="#cb34-981" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-982"><a href="#cb34-982" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-parvoviridae-histogram</span></span>
<span id="cb34-983"><a href="#cb34-983" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb34-984"><a href="#cb34-984" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 2.5</span></span>
<span id="cb34-985"><a href="#cb34-985" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-986"><a href="#cb34-986" aria-hidden="true" tabindex="-1"></a><span class="co"># Parvoviridae</span></span>
<span id="cb34-987"><a href="#cb34-987" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_histogram</span>(<span class="at">taxid_chosen=</span><span class="dv">10780</span>)</span>
<span id="cb34-988"><a href="#cb34-988" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-989"><a href="#cb34-989" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-992"><a href="#cb34-992" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-993"><a href="#cb34-993" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-parvoviridae-composition</span></span>
<span id="cb34-994"><a href="#cb34-994" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb34-995"><a href="#cb34-995" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 7.5</span></span>
<span id="cb34-996"><a href="#cb34-996" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb34-997"><a href="#cb34-997" aria-hidden="true" tabindex="-1"></a><span class="co"># Parvoviridae</span></span>
<span id="cb34-998"><a href="#cb34-998" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_composition</span>(<span class="at">taxid_chosen=</span><span class="dv">10780</span>, <span class="at">threshold_major_species =</span> <span class="fl">0.01</span>)</span>
<span id="cb34-999"><a href="#cb34-999" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-1000"><a href="#cb34-1000" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1001"><a href="#cb34-1001" aria-hidden="true" tabindex="-1"></a><span class="fu">### Retroviridae (Number of reads: 224) </span></span>
<span id="cb34-1004"><a href="#cb34-1004" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-1005"><a href="#cb34-1005" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-retroviridae-histogram</span></span>
<span id="cb34-1006"><a href="#cb34-1006" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb34-1007"><a href="#cb34-1007" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 2.5</span></span>
<span id="cb34-1008"><a href="#cb34-1008" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1009"><a href="#cb34-1009" aria-hidden="true" tabindex="-1"></a><span class="co"># Retroviridae</span></span>
<span id="cb34-1010"><a href="#cb34-1010" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_histogram</span>(<span class="at">taxid_chosen=</span><span class="dv">11632</span>)</span>
<span id="cb34-1011"><a href="#cb34-1011" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-1012"><a href="#cb34-1012" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1015"><a href="#cb34-1015" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-1016"><a href="#cb34-1016" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-retroviridae-composition</span></span>
<span id="cb34-1017"><a href="#cb34-1017" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb34-1018"><a href="#cb34-1018" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 7.5</span></span>
<span id="cb34-1019"><a href="#cb34-1019" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb34-1020"><a href="#cb34-1020" aria-hidden="true" tabindex="-1"></a><span class="co"># Retroviridae</span></span>
<span id="cb34-1021"><a href="#cb34-1021" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_composition</span>(<span class="at">taxid_chosen=</span><span class="dv">11632</span>, <span class="at">threshold_major_species =</span> <span class="fl">0.01</span>)</span>
<span id="cb34-1022"><a href="#cb34-1022" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-1023"><a href="#cb34-1023" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1024"><a href="#cb34-1024" aria-hidden="true" tabindex="-1"></a><span class="fu">## Relative abundance of pathogenic viruses of interest</span></span>
<span id="cb34-1025"><a href="#cb34-1025" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1028"><a href="#cb34-1028" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-1029"><a href="#cb34-1029" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-composition-exclusion</span></span>
<span id="cb34-1030"><a href="#cb34-1030" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb34-1031"><a href="#cb34-1031" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1032"><a href="#cb34-1032" aria-hidden="true" tabindex="-1"></a>total <span class="ot">&lt;-</span> basic_stats_raw <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample, n_read_pairs)</span>
<span id="cb34-1033"><a href="#cb34-1033" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1034"><a href="#cb34-1034" aria-hidden="true" tabindex="-1"></a>hv_family_counts <span class="ot">&lt;-</span> hv_reads_family <span class="sc">%&gt;%</span> </span>
<span id="cb34-1035"><a href="#cb34-1035" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb34-1036"><a href="#cb34-1036" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-1037"><a href="#cb34-1037" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(total, <span class="at">by=</span><span class="st">'sample'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-1038"><a href="#cb34-1038" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb34-1039"><a href="#cb34-1039" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">ra_reads_hv =</span> n_reads_hv<span class="sc">/</span>n_read_pairs)</span>
<span id="cb34-1040"><a href="#cb34-1040" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1041"><a href="#cb34-1041" aria-hidden="true" tabindex="-1"></a>hv_family_collate <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span></span>
<span id="cb34-1042"><a href="#cb34-1042" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb34-1043"><a href="#cb34-1043" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_tot =</span> <span class="fu">sum</span>(n_reads_hv),</span>
<span id="cb34-1044"><a href="#cb34-1044" aria-hidden="true" tabindex="-1"></a>            <span class="at">ra_reads_tot =</span> <span class="fu">mean</span>(ra_reads_hv), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb34-1045"><a href="#cb34-1045" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n_reads_tot))</span>
<span id="cb34-1046"><a href="#cb34-1046" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1047"><a href="#cb34-1047" aria-hidden="true" tabindex="-1"></a>hv_family_collate <span class="ot">&lt;-</span> hv_family_collate <span class="sc">%&gt;%</span> </span>
<span id="cb34-1048"><a href="#cb34-1048" aria-hidden="true" tabindex="-1"></a><span class="fu">rename</span>(</span>
<span id="cb34-1049"><a href="#cb34-1049" aria-hidden="true" tabindex="-1"></a><span class="st">'family'</span> <span class="ot">=</span> <span class="st">'name'</span>,</span>
<span id="cb34-1050"><a href="#cb34-1050" aria-hidden="true" tabindex="-1"></a><span class="st">'family_n_reads_tot'</span> <span class="ot">=</span> <span class="st">'n_reads_tot'</span>,</span>
<span id="cb34-1051"><a href="#cb34-1051" aria-hidden="true" tabindex="-1"></a><span class="st">'family_ra_reads_tot'</span> <span class="ot">=</span> <span class="st">'ra_reads_tot'</span>,</span>
<span id="cb34-1052"><a href="#cb34-1052" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb34-1053"><a href="#cb34-1053" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1054"><a href="#cb34-1054" aria-hidden="true" tabindex="-1"></a>hv_genus_counts <span class="ot">&lt;-</span> hv_reads_genus <span class="sc">%&gt;%</span> </span>
<span id="cb34-1055"><a href="#cb34-1055" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb34-1056"><a href="#cb34-1056" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-1057"><a href="#cb34-1057" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(total, <span class="at">by=</span><span class="st">'sample'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-1058"><a href="#cb34-1058" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb34-1059"><a href="#cb34-1059" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">ra_reads_hv =</span> n_reads_hv<span class="sc">/</span>n_read_pairs)</span>
<span id="cb34-1060"><a href="#cb34-1060" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1061"><a href="#cb34-1061" aria-hidden="true" tabindex="-1"></a>hv_genus_collate <span class="ot">&lt;-</span> hv_genus_counts <span class="sc">%&gt;%</span></span>
<span id="cb34-1062"><a href="#cb34-1062" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb34-1063"><a href="#cb34-1063" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_tot =</span> <span class="fu">sum</span>(n_reads_hv),</span>
<span id="cb34-1064"><a href="#cb34-1064" aria-hidden="true" tabindex="-1"></a>            <span class="at">ra_reads_tot =</span> <span class="fu">mean</span>(ra_reads_hv), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb34-1065"><a href="#cb34-1065" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n_reads_tot))</span>
<span id="cb34-1066"><a href="#cb34-1066" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1067"><a href="#cb34-1067" aria-hidden="true" tabindex="-1"></a>hv_genus_collate <span class="ot">&lt;-</span> hv_genus_collate <span class="sc">%&gt;%</span> </span>
<span id="cb34-1068"><a href="#cb34-1068" aria-hidden="true" tabindex="-1"></a><span class="fu">rename</span>(</span>
<span id="cb34-1069"><a href="#cb34-1069" aria-hidden="true" tabindex="-1"></a><span class="st">'genus'</span> <span class="ot">=</span> <span class="st">'name'</span>,</span>
<span id="cb34-1070"><a href="#cb34-1070" aria-hidden="true" tabindex="-1"></a><span class="st">'genus_n_reads_tot'</span> <span class="ot">=</span> <span class="st">'n_reads_tot'</span>,</span>
<span id="cb34-1071"><a href="#cb34-1071" aria-hidden="true" tabindex="-1"></a><span class="st">'genus_ra_reads_tot'</span> <span class="ot">=</span> <span class="st">'ra_reads_tot'</span>,</span>
<span id="cb34-1072"><a href="#cb34-1072" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb34-1073"><a href="#cb34-1073" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1074"><a href="#cb34-1074" aria-hidden="true" tabindex="-1"></a>hv_species_counts <span class="ot">&lt;-</span> hv_reads_species <span class="sc">%&gt;%</span> </span>
<span id="cb34-1075"><a href="#cb34-1075" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb34-1076"><a href="#cb34-1076" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-1077"><a href="#cb34-1077" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(total, <span class="at">by=</span><span class="st">'sample'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-1078"><a href="#cb34-1078" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb34-1079"><a href="#cb34-1079" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">ra_reads_hv =</span> n_reads_hv<span class="sc">/</span>n_read_pairs)</span>
<span id="cb34-1080"><a href="#cb34-1080" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1081"><a href="#cb34-1081" aria-hidden="true" tabindex="-1"></a>hv_species_collate <span class="ot">&lt;-</span> hv_species_counts <span class="sc">%&gt;%</span></span>
<span id="cb34-1082"><a href="#cb34-1082" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb34-1083"><a href="#cb34-1083" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_tot =</span> <span class="fu">sum</span>(n_reads_hv),</span>
<span id="cb34-1084"><a href="#cb34-1084" aria-hidden="true" tabindex="-1"></a>            <span class="at">ra_reads_tot =</span> <span class="fu">mean</span>(ra_reads_hv), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb34-1085"><a href="#cb34-1085" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n_reads_tot))</span>
<span id="cb34-1086"><a href="#cb34-1086" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1087"><a href="#cb34-1087" aria-hidden="true" tabindex="-1"></a>hv_species_collate <span class="ot">&lt;-</span> hv_species_collate <span class="sc">%&gt;%</span> </span>
<span id="cb34-1088"><a href="#cb34-1088" aria-hidden="true" tabindex="-1"></a><span class="fu">rename</span>(</span>
<span id="cb34-1089"><a href="#cb34-1089" aria-hidden="true" tabindex="-1"></a><span class="st">'species'</span> <span class="ot">=</span> <span class="st">'name'</span>,</span>
<span id="cb34-1090"><a href="#cb34-1090" aria-hidden="true" tabindex="-1"></a><span class="st">'species_n_reads_tot'</span> <span class="ot">=</span> <span class="st">'n_reads_tot'</span>,</span>
<span id="cb34-1091"><a href="#cb34-1091" aria-hidden="true" tabindex="-1"></a><span class="st">'species_ra_reads_tot'</span> <span class="ot">=</span> <span class="st">'ra_reads_tot'</span>,</span>
<span id="cb34-1092"><a href="#cb34-1092" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb34-1093"><a href="#cb34-1093" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1094"><a href="#cb34-1094" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-1095"><a href="#cb34-1095" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1098"><a href="#cb34-1098" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-1099"><a href="#cb34-1099" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-filter-exclusion</span></span>
<span id="cb34-1100"><a href="#cb34-1100" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb34-1101"><a href="#cb34-1101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1102"><a href="#cb34-1102" aria-hidden="true" tabindex="-1"></a><span class="co"># Assuming the data is already loaded into tibbles:</span></span>
<span id="cb34-1103"><a href="#cb34-1103" aria-hidden="true" tabindex="-1"></a><span class="co">#hv_family_collate</span></span>
<span id="cb34-1104"><a href="#cb34-1104" aria-hidden="true" tabindex="-1"></a><span class="co">#hv_genus_collate</span></span>
<span id="cb34-1105"><a href="#cb34-1105" aria-hidden="true" tabindex="-1"></a><span class="co">#hv_species_collate</span></span>
<span id="cb34-1106"><a href="#cb34-1106" aria-hidden="true" tabindex="-1"></a><span class="co">#viral_taxa</span></span>
<span id="cb34-1107"><a href="#cb34-1107" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1108"><a href="#cb34-1108" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to find parent taxa recursively</span></span>
<span id="cb34-1109"><a href="#cb34-1109" aria-hidden="true" tabindex="-1"></a>find_parent_taxa <span class="ot">&lt;-</span> <span class="cf">function</span>(taxid, target_ranks, taxa_data) {</span>
<span id="cb34-1110"><a href="#cb34-1110" aria-hidden="true" tabindex="-1"></a> current_taxid <span class="ot">&lt;-</span> taxid</span>
<span id="cb34-1111"><a href="#cb34-1111" aria-hidden="true" tabindex="-1"></a>  result <span class="ot">&lt;-</span> <span class="fu">setNames</span>(<span class="fu">vector</span>(<span class="st">"list"</span>, <span class="fu">length</span>(target_ranks)), target_ranks)</span>
<span id="cb34-1112"><a href="#cb34-1112" aria-hidden="true" tabindex="-1"></a>  <span class="cf">while</span> (<span class="cn">TRUE</span>) {</span>
<span id="cb34-1113"><a href="#cb34-1113" aria-hidden="true" tabindex="-1"></a>    row <span class="ot">&lt;-</span> taxa_data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(taxid <span class="sc">==</span> current_taxid) <span class="sc">%&gt;%</span> <span class="fu">slice</span>(<span class="dv">1</span>)</span>
<span id="cb34-1114"><a href="#cb34-1114" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">nrow</span>(row) <span class="sc">==</span> <span class="dv">0</span>) <span class="cf">break</span>  <span class="co"># If no matching taxid is found, break the loop</span></span>
<span id="cb34-1115"><a href="#cb34-1115" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (row<span class="sc">$</span>rank <span class="sc">%in%</span> target_ranks) {</span>
<span id="cb34-1116"><a href="#cb34-1116" aria-hidden="true" tabindex="-1"></a>      result[[row<span class="sc">$</span>rank]] <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">taxid =</span> row<span class="sc">$</span>taxid, <span class="at">name =</span> row<span class="sc">$</span>name)</span>
<span id="cb34-1117"><a href="#cb34-1117" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span> (<span class="fu">all</span>(<span class="sc">!</span><span class="fu">sapply</span>(result, is.null))) {</span>
<span id="cb34-1118"><a href="#cb34-1118" aria-hidden="true" tabindex="-1"></a>        <span class="cf">break</span>  <span class="co"># If all target ranks are found, break the loop</span></span>
<span id="cb34-1119"><a href="#cb34-1119" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb34-1120"><a href="#cb34-1120" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb34-1121"><a href="#cb34-1121" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (row<span class="sc">$</span>taxid <span class="sc">==</span> row<span class="sc">$</span>parent_taxid) {</span>
<span id="cb34-1122"><a href="#cb34-1122" aria-hidden="true" tabindex="-1"></a>      <span class="cf">break</span>  <span class="co"># Avoid infinite loop at root</span></span>
<span id="cb34-1123"><a href="#cb34-1123" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb34-1124"><a href="#cb34-1124" aria-hidden="true" tabindex="-1"></a>    current_taxid <span class="ot">&lt;-</span> row<span class="sc">$</span>parent_taxid</span>
<span id="cb34-1125"><a href="#cb34-1125" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb34-1126"><a href="#cb34-1126" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Replace NULL with NA for any unfound ranks</span></span>
<span id="cb34-1127"><a href="#cb34-1127" aria-hidden="true" tabindex="-1"></a>  result <span class="ot">&lt;-</span> <span class="fu">lapply</span>(result, <span class="cf">function</span>(x) <span class="cf">if</span> (<span class="fu">is.null</span>(x)) <span class="fu">list</span>(<span class="at">taxid =</span> <span class="cn">NA</span>, <span class="at">name =</span> <span class="cn">NA</span>) <span class="cf">else</span> x)</span>
<span id="cb34-1128"><a href="#cb34-1128" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(result)</span>
<span id="cb34-1129"><a href="#cb34-1129" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb34-1130"><a href="#cb34-1130" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1131"><a href="#cb34-1131" aria-hidden="true" tabindex="-1"></a><span class="co"># Find all relevant parent taxa for each species</span></span>
<span id="cb34-1132"><a href="#cb34-1132" aria-hidden="true" tabindex="-1"></a>result <span class="ot">&lt;-</span> hv_species_collate <span class="sc">%&gt;%</span></span>
<span id="cb34-1133"><a href="#cb34-1133" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb34-1134"><a href="#cb34-1134" aria-hidden="true" tabindex="-1"></a>    <span class="at">parent_info =</span> <span class="fu">map</span>(taxid, <span class="sc">~</span><span class="fu">find_parent_taxa</span>(.x, <span class="fu">c</span>(<span class="st">"species"</span>,<span class="st">"subgenus"</span>, <span class="st">"genus"</span>, <span class="st">"subfamily"</span>, <span class="st">"family"</span>), viral_taxa)),</span>
<span id="cb34-1135"><a href="#cb34-1135" aria-hidden="true" tabindex="-1"></a>    <span class="at">genus_taxid =</span> <span class="fu">map_dbl</span>(parent_info, <span class="sc">~</span>.x<span class="sc">$</span>genus<span class="sc">$</span>taxid),</span>
<span id="cb34-1136"><a href="#cb34-1136" aria-hidden="true" tabindex="-1"></a>    <span class="at">genus =</span> <span class="fu">map_chr</span>(parent_info, <span class="sc">~</span>.x<span class="sc">$</span>genus<span class="sc">$</span>name),</span>
<span id="cb34-1137"><a href="#cb34-1137" aria-hidden="true" tabindex="-1"></a>    <span class="at">family_taxid =</span> <span class="fu">map_dbl</span>(parent_info, <span class="sc">~</span>.x<span class="sc">$</span>family<span class="sc">$</span>taxid),</span>
<span id="cb34-1138"><a href="#cb34-1138" aria-hidden="true" tabindex="-1"></a>    <span class="at">family =</span> <span class="fu">map_chr</span>(parent_info, <span class="sc">~</span>.x<span class="sc">$</span>family<span class="sc">$</span>name)</span>
<span id="cb34-1139"><a href="#cb34-1139" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb34-1140"><a href="#cb34-1140" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>parent_info)</span>
<span id="cb34-1141"><a href="#cb34-1141" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1142"><a href="#cb34-1142" aria-hidden="true" tabindex="-1"></a><span class="co"># Join with genus counts</span></span>
<span id="cb34-1143"><a href="#cb34-1143" aria-hidden="true" tabindex="-1"></a>result <span class="ot">&lt;-</span> result <span class="sc">%&gt;%</span></span>
<span id="cb34-1144"><a href="#cb34-1144" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(</span>
<span id="cb34-1145"><a href="#cb34-1145" aria-hidden="true" tabindex="-1"></a>    hv_genus_collate <span class="sc">%&gt;%</span> <span class="fu">select</span>(taxid, genus_n_reads_tot, genus_ra_reads_tot),</span>
<span id="cb34-1146"><a href="#cb34-1146" aria-hidden="true" tabindex="-1"></a>    <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"genus_taxid"</span> <span class="ot">=</span> <span class="st">"taxid"</span>)</span>
<span id="cb34-1147"><a href="#cb34-1147" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb34-1148"><a href="#cb34-1148" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1149"><a href="#cb34-1149" aria-hidden="true" tabindex="-1"></a><span class="co"># Join with family counts</span></span>
<span id="cb34-1150"><a href="#cb34-1150" aria-hidden="true" tabindex="-1"></a>result <span class="ot">&lt;-</span> result <span class="sc">%&gt;%</span></span>
<span id="cb34-1151"><a href="#cb34-1151" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(</span>
<span id="cb34-1152"><a href="#cb34-1152" aria-hidden="true" tabindex="-1"></a>    hv_family_collate <span class="sc">%&gt;%</span> <span class="fu">select</span>(taxid, family_n_reads_tot, family_ra_reads_tot),</span>
<span id="cb34-1153"><a href="#cb34-1153" aria-hidden="true" tabindex="-1"></a>    <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"family_taxid"</span> <span class="ot">=</span> <span class="st">"taxid"</span>)</span>
<span id="cb34-1154"><a href="#cb34-1154" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb34-1155"><a href="#cb34-1155" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1156"><a href="#cb34-1156" aria-hidden="true" tabindex="-1"></a><span class="co"># Clean up</span></span>
<span id="cb34-1157"><a href="#cb34-1157" aria-hidden="true" tabindex="-1"></a>result <span class="ot">&lt;-</span> result <span class="sc">%&gt;%</span></span>
<span id="cb34-1158"><a href="#cb34-1158" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="fu">ends_with</span>(<span class="st">"_taxid"</span>))</span>
<span id="cb34-1159"><a href="#cb34-1159" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1160"><a href="#cb34-1160" aria-hidden="true" tabindex="-1"></a><span class="co">#result</span></span>
<span id="cb34-1161"><a href="#cb34-1161" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-1162"><a href="#cb34-1162" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1163"><a href="#cb34-1163" aria-hidden="true" tabindex="-1"></a>Each dot represents a sample, colored by viral family. The x-axis shows the relative abundance of human-infecting viruses, and the y-axis shows the species.</span>
<span id="cb34-1166"><a href="#cb34-1166" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-1167"><a href="#cb34-1167" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-pathogenic-viruses</span></span>
<span id="cb34-1168"><a href="#cb34-1168" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb34-1169"><a href="#cb34-1169" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 10</span></span>
<span id="cb34-1170"><a href="#cb34-1170" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb34-1171"><a href="#cb34-1171" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1172"><a href="#cb34-1172" aria-hidden="true" tabindex="-1"></a>species_family <span class="ot">&lt;-</span> result <span class="sc">%&gt;%</span> <span class="fu">select</span>(species, family) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="st">'name'</span> <span class="ot">=</span> <span class="st">'species'</span>)</span>
<span id="cb34-1173"><a href="#cb34-1173" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1174"><a href="#cb34-1174" aria-hidden="true" tabindex="-1"></a>play <span class="ot">&lt;-</span> hv_species_counts <span class="sc">%&gt;%</span> </span>
<span id="cb34-1175"><a href="#cb34-1175" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb34-1176"><a href="#cb34-1176" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by =</span> <span class="st">'sample'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-1177"><a href="#cb34-1177" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(species_family, <span class="at">by =</span> <span class="st">'name'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-1178"><a href="#cb34-1178" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">ifelse</span>(name <span class="sc">==</span> <span class="st">"Severe acute respiratory syndrome-related coronavirus"</span>, <span class="st">"SARS-related coronavirus"</span>, name))</span>
<span id="cb34-1179"><a href="#cb34-1179" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1180"><a href="#cb34-1180" aria-hidden="true" tabindex="-1"></a>adjusted_play <span class="ot">&lt;-</span> play <span class="sc">%&gt;%</span> </span>
<span id="cb34-1181"><a href="#cb34-1181" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span></span>
<span id="cb34-1182"><a href="#cb34-1182" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">virus_prevalence_num =</span> <span class="fu">n_distinct</span>(sample)<span class="sc">/</span><span class="fu">n_distinct</span>(libraries),</span>
<span id="cb34-1183"><a href="#cb34-1183" aria-hidden="true" tabindex="-1"></a>         <span class="at">total_reads_hv =</span> <span class="fu">sum</span>(n_reads_hv)) <span class="sc">%&gt;%</span></span>
<span id="cb34-1184"><a href="#cb34-1184" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb34-1185"><a href="#cb34-1185" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">fct_reorder</span>(name, virus_prevalence_num, <span class="at">.desc=</span><span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb34-1186"><a href="#cb34-1186" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, ra_reads_hv, family, virus_prevalence_num, total_reads_hv, n_reads_hv)</span>
<span id="cb34-1187"><a href="#cb34-1187" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1188"><a href="#cb34-1188" aria-hidden="true" tabindex="-1"></a>pal <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">'Dark2'</span>),<span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">'Accent'</span>))</span>
<span id="cb34-1189"><a href="#cb34-1189" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1190"><a href="#cb34-1190" aria-hidden="true" tabindex="-1"></a><span class="co">#, labels = label_log(digits=2)</span></span>
<span id="cb34-1191"><a href="#cb34-1191" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1192"><a href="#cb34-1192" aria-hidden="true" tabindex="-1"></a>ra_dot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(adjusted_play, <span class="fu">aes</span>(<span class="at">x =</span> ra_reads_hv, <span class="at">y=</span>name)) <span class="sc">+</span></span>
<span id="cb34-1193"><a href="#cb34-1193" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_quasirandom</span>(<span class="at">orientation =</span> <span class="st">'y'</span>, <span class="fu">aes</span>(<span class="at">color =</span> family)) <span class="sc">+</span></span>
<span id="cb34-1194"><a href="#cb34-1194" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> pal) <span class="sc">+</span> </span>
<span id="cb34-1195"><a href="#cb34-1195" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_log10</span>(</span>
<span id="cb34-1196"><a href="#cb34-1196" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Relative abundance human virus reads"</span>,</span>
<span id="cb34-1197"><a href="#cb34-1197" aria-hidden="true" tabindex="-1"></a>    <span class="at">labels=</span><span class="fu">label_log</span>(<span class="at">digits=</span><span class="dv">3</span>)</span>
<span id="cb34-1198"><a href="#cb34-1198" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb34-1199"><a href="#cb34-1199" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span>  <span class="st">""</span>,</span>
<span id="cb34-1200"><a href="#cb34-1200" aria-hidden="true" tabindex="-1"></a>       <span class="at">color =</span> <span class="st">'Viral family'</span>) <span class="sc">+</span> </span>
<span id="cb34-1201"><a href="#cb34-1201" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color =</span> <span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">2</span>)) <span class="sc">+</span></span>
<span id="cb34-1202"><a href="#cb34-1202" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_light</span>() <span class="sc">+</span> </span>
<span id="cb34-1203"><a href="#cb34-1203" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(</span>
<span id="cb34-1204"><a href="#cb34-1204" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb34-1205"><a href="#cb34-1205" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>),</span>
<span id="cb34-1206"><a href="#cb34-1206" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.ticks.y =</span> <span class="fu">element_blank</span>(),<span class="at">axis.line =</span> <span class="fu">element_line</span>(<span class="at">colour =</span> <span class="st">"black"</span>),</span>
<span id="cb34-1207"><a href="#cb34-1207" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">14</span>),    </span>
<span id="cb34-1208"><a href="#cb34-1208" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb34-1209"><a href="#cb34-1209" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>, <span class="at">face=</span><span class="st">"bold"</span>),</span>
<span id="cb34-1210"><a href="#cb34-1210" aria-hidden="true" tabindex="-1"></a>    <span class="co">#legend.position = 'bottom',</span></span>
<span id="cb34-1211"><a href="#cb34-1211" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>),  <span class="co"># Move legend to top right</span></span>
<span id="cb34-1212"><a href="#cb34-1212" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.justification =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>),  <span class="co"># Align legend to top right</span></span>
<span id="cb34-1213"><a href="#cb34-1213" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb34-1214"><a href="#cb34-1214" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.border =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb34-1215"><a href="#cb34-1215" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.background =</span> <span class="fu">element_blank</span>())</span>
<span id="cb34-1216"><a href="#cb34-1216" aria-hidden="true" tabindex="-1"></a>ra_dot</span>
<span id="cb34-1217"><a href="#cb34-1217" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-1218"><a href="#cb34-1218" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1219"><a href="#cb34-1219" aria-hidden="true" tabindex="-1"></a>I can then take all of the viruses that we found and look up what they're responsible for and whether they're dangerous.</span>
<span id="cb34-1220"><a href="#cb34-1220" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1221"><a href="#cb34-1221" aria-hidden="true" tabindex="-1"></a>| Virus Name | Common Name | Pathogenic Potential |</span>
<span id="cb34-1222"><a href="#cb34-1222" aria-hidden="true" tabindex="-1"></a>|------------|-------------|----------------------|</span>
<span id="cb34-1223"><a href="#cb34-1223" aria-hidden="true" tabindex="-1"></a>| Human immunodeficiency virus 1 | HIV-1 | Very high, causes AIDS if left untreated, leading to severe immunodeficiency |</span>
<span id="cb34-1224"><a href="#cb34-1224" aria-hidden="true" tabindex="-1"></a>| SARS-related coronavirus | SARS-CoV | High, causes severe acute respiratory syndrome with potentially fatal outcomes |</span>
<span id="cb34-1225"><a href="#cb34-1225" aria-hidden="true" tabindex="-1"></a>| Hepacivirus hominis | Hepatitis C Virus | High, causes chronic liver disease and can lead to cirrhosis and liver cancer |</span>
<span id="cb34-1226"><a href="#cb34-1226" aria-hidden="true" tabindex="-1"></a>| Cytomegalovirus humanbeta5 | Human Cytomegalovirus | Moderate to high, can cause severe disease in immunocompromised individuals and congenital infections |</span>
<span id="cb34-1227"><a href="#cb34-1227" aria-hidden="true" tabindex="-1"></a>| Lymphocryptovirus humangamma4 | Epstein-Barr Virus | Moderate, causes infectious mononucleosis and associated with certain cancers |</span>
<span id="cb34-1228"><a href="#cb34-1228" aria-hidden="true" tabindex="-1"></a>| Erythroparvovirus primate1 | Human Parvovirus B19 | Moderate, causes fifth disease and can be serious in certain populations (pregnant women, immunocompromised) |</span>
<span id="cb34-1229"><a href="#cb34-1229" aria-hidden="true" tabindex="-1"></a>| Roseolovirus humanbeta6a | Human Herpesvirus 6A | Low to moderate, can cause roseola in children and occasionally more severe complications |</span>
<span id="cb34-1230"><a href="#cb34-1230" aria-hidden="true" tabindex="-1"></a>| Roseolovirus humanbeta6b | Human Herpesvirus 6B | Low to moderate, can cause roseola in children and occasionally more severe complications |</span>
<span id="cb34-1231"><a href="#cb34-1231" aria-hidden="true" tabindex="-1"></a>| Alphapapillomavirus 2 | Human Papillomavirus 2 | Low to moderate, can cause common warts and rarely progress to cancer |</span>
<span id="cb34-1232"><a href="#cb34-1232" aria-hidden="true" tabindex="-1"></a>| Betapapillomavirus 5 | Human Papillomavirus 5 | Low to moderate, associated with skin lesions in immunosuppressed individuals |</span>
<span id="cb34-1233"><a href="#cb34-1233" aria-hidden="true" tabindex="-1"></a>| Betapapillomavirus 2 | Human Papillomavirus 9 | Low to moderate, associated with cutaneous lesions and rarely with non-melanoma skin cancer |</span>
<span id="cb34-1234"><a href="#cb34-1234" aria-hidden="true" tabindex="-1"></a>| Primate T-lymphotropic virus 2 | Human T-cell Lymphotropic Virus 2 | Low, rarely associated with neurological disorders |</span>
<span id="cb34-1235"><a href="#cb34-1235" aria-hidden="true" tabindex="-1"></a>| Nupapillomavirus 1 | Human Papillomavirus 41 | Low, can cause benign cutaneous lesions |</span>
<span id="cb34-1236"><a href="#cb34-1236" aria-hidden="true" tabindex="-1"></a>| Alphapolyomavirus quintihominis | Human Polyomavirus 5 | Low, not known to cause significant disease in healthy individuals |</span>
<span id="cb34-1237"><a href="#cb34-1237" aria-hidden="true" tabindex="-1"></a>| Pegivirus hominis | Human Pegivirus | Very low, not known to cause disease in humans |</span>
<span id="cb34-1238"><a href="#cb34-1238" aria-hidden="true" tabindex="-1"></a>| Pegivirus columbiaense | Pegivirus C | Very low, not known to cause disease in humans |                                            |</span>
<span id="cb34-1239"><a href="#cb34-1239" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1240"><a href="#cb34-1240" aria-hidden="true" tabindex="-1"></a>This was interesting to see. This is the first study where we've seen signficant numbers of human-infecting viruses in whole blood.</span>
<span id="cb34-1241"><a href="#cb34-1241" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1242"><a href="#cb34-1242" aria-hidden="true" tabindex="-1"></a><span class="fu">## Relative abundance assuming perfect human read removal</span></span>
<span id="cb34-1245"><a href="#cb34-1245" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-1246"><a href="#cb34-1246" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: perfect-hv-removal</span></span>
<span id="cb34-1247"><a href="#cb34-1247" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb34-1248"><a href="#cb34-1248" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1249"><a href="#cb34-1249" aria-hidden="true" tabindex="-1"></a>human_reads <span class="ot">&lt;-</span> comp_assigned <span class="sc">%&gt;%</span> <span class="fu">filter</span>(classification <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Human"</span>)) <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample, <span class="at">human_reads=</span>n_reads)</span>
<span id="cb34-1250"><a href="#cb34-1250" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1251"><a href="#cb34-1251" aria-hidden="true" tabindex="-1"></a>all <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span> </span>
<span id="cb34-1252"><a href="#cb34-1252" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(human_reads, <span class="at">by=</span><span class="st">'sample'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-1253"><a href="#cb34-1253" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb34-1254"><a href="#cb34-1254" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">sum_reads=</span><span class="fu">sum</span>(n_reads_hv), <span class="at">n_read_pairs=</span><span class="fu">first</span>(n_read_pairs) <span class="sc">-</span> <span class="fu">first</span>(human_reads)) <span class="sc">%&gt;%</span></span>
<span id="cb34-1255"><a href="#cb34-1255" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">ra =</span> sum_reads<span class="sc">/</span>n_read_pairs) <span class="sc">%&gt;%</span> </span>
<span id="cb34-1256"><a href="#cb34-1256" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(ra) <span class="sc">%&gt;%</span> <span class="fu">mean</span>()</span>
<span id="cb34-1257"><a href="#cb34-1257" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1258"><a href="#cb34-1258" aria-hidden="true" tabindex="-1"></a>all</span>
<span id="cb34-1259"><a href="#cb34-1259" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-1260"><a href="#cb34-1260" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1261"><a href="#cb34-1261" aria-hidden="true" tabindex="-1"></a>Assuming we're able to perfectly remove all human reads, the average relative abundance of known human infecting virus is $1.91 \times 10^{-5}$.</span>
<span id="cb34-1262"><a href="#cb34-1262" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1263"><a href="#cb34-1263" aria-hidden="true" tabindex="-1"></a><span class="fu"># Conclusion</span></span>
<span id="cb34-1264"><a href="#cb34-1264" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1265"><a href="#cb34-1265" aria-hidden="true" tabindex="-1"></a>There were some interesting takeways from this analysis:</span>
<span id="cb34-1266"><a href="#cb34-1266" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1267"><a href="#cb34-1267" aria-hidden="true" tabindex="-1"></a><span class="ss">1. </span>Whole blood metagenomics was able to detect HIV-1, SARS-CoV-2, HCV, CMV, EBV, HTLV-2, and other herpesviruses. This is important as some of these are latent viruses.</span>
<span id="cb34-1268"><a href="#cb34-1268" aria-hidden="true" tabindex="-1"></a><span class="ss">2. </span>Far fewer individuals tend to have detectable levels of human infecting viruses, when compared to the plasma datasets.</span>
<span id="cb34-1269"><a href="#cb34-1269" aria-hidden="true" tabindex="-1"></a><span class="ss">3. </span>Important viruses can be detected from RNA-seq data, even if the study wasn't explictly designed to look at them.</span>
<span id="cb34-1270"><a href="#cb34-1270" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1271"><a href="#cb34-1271" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1272"><a href="#cb34-1272" aria-hidden="true" tabindex="-1"></a><span class="fu"># Appendix</span></span>
<span id="cb34-1273"><a href="#cb34-1273" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1274"><a href="#cb34-1274" aria-hidden="true" tabindex="-1"></a><span class="fu">## Human-infecting virus families, genera, and species</span></span>
<span id="cb34-1275"><a href="#cb34-1275" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1276"><a href="#cb34-1276" aria-hidden="true" tabindex="-1"></a>To get a good overview of families, genera, and species, we can look at a Sankey plot where the magnitude of relative abundance, averaged over all samples, is shown in parentheses.</span>
<span id="cb34-1277"><a href="#cb34-1277" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1280"><a href="#cb34-1280" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-1281"><a href="#cb34-1281" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-sankey</span></span>
<span id="cb34-1282"><a href="#cb34-1282" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 22.5</span></span>
<span id="cb34-1283"><a href="#cb34-1283" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb34-1284"><a href="#cb34-1284" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb34-1285"><a href="#cb34-1285" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1286"><a href="#cb34-1286" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to create links</span></span>
<span id="cb34-1287"><a href="#cb34-1287" aria-hidden="true" tabindex="-1"></a>create_links <span class="ot">&lt;-</span> <span class="cf">function</span>(data) {</span>
<span id="cb34-1288"><a href="#cb34-1288" aria-hidden="true" tabindex="-1"></a>  family_to_genus <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb34-1289"><a href="#cb34-1289" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(genus)) <span class="sc">%&gt;%</span></span>
<span id="cb34-1290"><a href="#cb34-1290" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(family, genus) <span class="sc">%&gt;%</span></span>
<span id="cb34-1291"><a href="#cb34-1291" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">value =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-1292"><a href="#cb34-1292" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">source =</span> family, <span class="at">target =</span> genus)</span>
<span id="cb34-1293"><a href="#cb34-1293" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb34-1294"><a href="#cb34-1294" aria-hidden="true" tabindex="-1"></a>  genus_to_species <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb34-1295"><a href="#cb34-1295" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(genus, species) <span class="sc">%&gt;%</span></span>
<span id="cb34-1296"><a href="#cb34-1296" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">value =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-1297"><a href="#cb34-1297" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">source =</span> genus, <span class="at">target =</span> species)</span>
<span id="cb34-1298"><a href="#cb34-1298" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1299"><a href="#cb34-1299" aria-hidden="true" tabindex="-1"></a>  family_to_species <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb34-1300"><a href="#cb34-1300" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="fu">is.na</span>(genus)) <span class="sc">%&gt;%</span></span>
<span id="cb34-1301"><a href="#cb34-1301" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(family, species) <span class="sc">%&gt;%</span></span>
<span id="cb34-1302"><a href="#cb34-1302" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">value =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-1303"><a href="#cb34-1303" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">source =</span> family, <span class="at">target =</span> species)</span>
<span id="cb34-1304"><a href="#cb34-1304" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1305"><a href="#cb34-1305" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_rows</span>(family_to_genus, genus_to_species, family_to_species) <span class="sc">%&gt;%</span></span>
<span id="cb34-1306"><a href="#cb34-1306" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(source))</span>
<span id="cb34-1307"><a href="#cb34-1307" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb34-1308"><a href="#cb34-1308" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1309"><a href="#cb34-1309" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to create nodes</span></span>
<span id="cb34-1310"><a href="#cb34-1310" aria-hidden="true" tabindex="-1"></a>create_nodes <span class="ot">&lt;-</span> <span class="cf">function</span>(links) {</span>
<span id="cb34-1311"><a href="#cb34-1311" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>(</span>
<span id="cb34-1312"><a href="#cb34-1312" aria-hidden="true" tabindex="-1"></a>    <span class="at">name =</span> <span class="fu">c</span>(links<span class="sc">$</span>source, links<span class="sc">$</span>target) <span class="sc">%&gt;%</span> <span class="fu">unique</span>()</span>
<span id="cb34-1313"><a href="#cb34-1313" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb34-1314"><a href="#cb34-1314" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb34-1315"><a href="#cb34-1315" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1316"><a href="#cb34-1316" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to prepare data for Sankey diagram</span></span>
<span id="cb34-1317"><a href="#cb34-1317" aria-hidden="true" tabindex="-1"></a>prepare_sankey_data <span class="ot">&lt;-</span> <span class="cf">function</span>(links, nodes) {</span>
<span id="cb34-1318"><a href="#cb34-1318" aria-hidden="true" tabindex="-1"></a>  links<span class="sc">$</span>IDsource <span class="ot">&lt;-</span> <span class="fu">match</span>(links<span class="sc">$</span>source, nodes<span class="sc">$</span>name) <span class="sc">-</span> <span class="dv">1</span></span>
<span id="cb34-1319"><a href="#cb34-1319" aria-hidden="true" tabindex="-1"></a>  links<span class="sc">$</span>IDtarget <span class="ot">&lt;-</span> <span class="fu">match</span>(links<span class="sc">$</span>target, nodes<span class="sc">$</span>name) <span class="sc">-</span> <span class="dv">1</span></span>
<span id="cb34-1320"><a href="#cb34-1320" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list</span>(<span class="at">links =</span> links, <span class="at">nodes =</span> nodes)</span>
<span id="cb34-1321"><a href="#cb34-1321" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb34-1322"><a href="#cb34-1322" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1323"><a href="#cb34-1323" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to create Sankey plot</span></span>
<span id="cb34-1324"><a href="#cb34-1324" aria-hidden="true" tabindex="-1"></a>create_sankey_plot <span class="ot">&lt;-</span> <span class="cf">function</span>(sankey_data) {</span>
<span id="cb34-1325"><a href="#cb34-1325" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sankeyNetwork</span>(</span>
<span id="cb34-1326"><a href="#cb34-1326" aria-hidden="true" tabindex="-1"></a>    <span class="at">Links =</span> sankey_data<span class="sc">$</span>links, </span>
<span id="cb34-1327"><a href="#cb34-1327" aria-hidden="true" tabindex="-1"></a>    <span class="at">Nodes =</span> sankey_data<span class="sc">$</span>nodes,</span>
<span id="cb34-1328"><a href="#cb34-1328" aria-hidden="true" tabindex="-1"></a>    <span class="at">Source =</span> <span class="st">"IDsource"</span>, </span>
<span id="cb34-1329"><a href="#cb34-1329" aria-hidden="true" tabindex="-1"></a>    <span class="at">Target =</span> <span class="st">"IDtarget"</span>,</span>
<span id="cb34-1330"><a href="#cb34-1330" aria-hidden="true" tabindex="-1"></a>    <span class="at">Value =</span> <span class="st">"value"</span>, </span>
<span id="cb34-1331"><a href="#cb34-1331" aria-hidden="true" tabindex="-1"></a>    <span class="at">NodeID =</span> <span class="st">"name"</span>,</span>
<span id="cb34-1332"><a href="#cb34-1332" aria-hidden="true" tabindex="-1"></a>    <span class="at">sinksRight =</span> <span class="cn">TRUE</span>,</span>
<span id="cb34-1333"><a href="#cb34-1333" aria-hidden="true" tabindex="-1"></a>    <span class="at">nodeWidth =</span> <span class="dv">25</span>,</span>
<span id="cb34-1334"><a href="#cb34-1334" aria-hidden="true" tabindex="-1"></a>    <span class="at">fontSize =</span> <span class="dv">14</span>,</span>
<span id="cb34-1335"><a href="#cb34-1335" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb34-1336"><a href="#cb34-1336" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb34-1337"><a href="#cb34-1337" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1338"><a href="#cb34-1338" aria-hidden="true" tabindex="-1"></a>save_sankey_as_png <span class="ot">&lt;-</span> <span class="cf">function</span>(sankey_plot, <span class="at">width =</span> <span class="dv">1000</span>, <span class="at">height =</span> <span class="dv">800</span>) {</span>
<span id="cb34-1339"><a href="#cb34-1339" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Save the plot as an HTML file</span></span>
<span id="cb34-1340"><a href="#cb34-1340" aria-hidden="true" tabindex="-1"></a>  <span class="fu">saveWidget</span>(sankey_plot, <span class="fu">sprintf</span>(<span class="st">'%s/sankey.html'</span>,data_dir))</span>
<span id="cb34-1341"><a href="#cb34-1341" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb34-1342"><a href="#cb34-1342" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1343"><a href="#cb34-1343" aria-hidden="true" tabindex="-1"></a>format_scientific <span class="ot">&lt;-</span> <span class="cf">function</span>(x, <span class="at">digits=</span><span class="dv">2</span>) {</span>
<span id="cb34-1344"><a href="#cb34-1344" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sapply</span>(x, <span class="cf">function</span>(val) {</span>
<span id="cb34-1345"><a href="#cb34-1345" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">is.na</span>(val) <span class="sc">||</span> <span class="fu">abs</span>(val) <span class="sc">&lt;</span> <span class="fl">1e-15</span>) {</span>
<span id="cb34-1346"><a href="#cb34-1346" aria-hidden="true" tabindex="-1"></a>      <span class="fu">return</span>(<span class="st">"0"</span>)</span>
<span id="cb34-1347"><a href="#cb34-1347" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb34-1348"><a href="#cb34-1348" aria-hidden="true" tabindex="-1"></a>      exponent <span class="ot">&lt;-</span> <span class="fu">floor</span>(<span class="fu">log10</span>(<span class="fu">abs</span>(val)))</span>
<span id="cb34-1349"><a href="#cb34-1349" aria-hidden="true" tabindex="-1"></a>      coef <span class="ot">&lt;-</span> val <span class="sc">/</span> <span class="dv">10</span><span class="sc">^</span>exponent</span>
<span id="cb34-1350"><a href="#cb34-1350" aria-hidden="true" tabindex="-1"></a>      <span class="co">#return(sprintf("%.1f × 10^%d", round(coef, digits), exponent))</span></span>
<span id="cb34-1351"><a href="#cb34-1351" aria-hidden="true" tabindex="-1"></a>      <span class="co"># May or may not be smart, just keeping magnitude</span></span>
<span id="cb34-1352"><a href="#cb34-1352" aria-hidden="true" tabindex="-1"></a>      <span class="fu">return</span>(<span class="fu">sprintf</span>(<span class="st">"10^%d"</span>, exponent))</span>
<span id="cb34-1353"><a href="#cb34-1353" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb34-1354"><a href="#cb34-1354" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb34-1355"><a href="#cb34-1355" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb34-1356"><a href="#cb34-1356" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1357"><a href="#cb34-1357" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> result <span class="sc">%&gt;%</span> </span>
<span id="cb34-1358"><a href="#cb34-1358" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">c</span>(genus_n_reads_tot, genus_ra_reads_tot), <span class="sc">~</span><span class="fu">replace_na</span>(., <span class="dv">0</span>)),</span>
<span id="cb34-1359"><a href="#cb34-1359" aria-hidden="true" tabindex="-1"></a>         <span class="at">genus =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(genus), <span class="st">"Unknown Genus"</span>, genus)) <span class="sc">%&gt;%</span></span>
<span id="cb34-1360"><a href="#cb34-1360" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb34-1361"><a href="#cb34-1361" aria-hidden="true" tabindex="-1"></a>  <span class="at">species =</span> <span class="fu">paste0</span>(species, <span class="fu">sprintf</span>(<span class="st">' (%s)'</span>, <span class="fu">format_scientific</span>(species_ra_reads_tot))),</span>
<span id="cb34-1362"><a href="#cb34-1362" aria-hidden="true" tabindex="-1"></a>  <span class="at">genus =</span> <span class="fu">paste0</span>(genus, <span class="fu">sprintf</span>(<span class="st">' (%s)'</span>, <span class="fu">format_scientific</span>(genus_ra_reads_tot))),</span>
<span id="cb34-1363"><a href="#cb34-1363" aria-hidden="true" tabindex="-1"></a>  <span class="at">family =</span> <span class="fu">paste0</span>(family, <span class="fu">sprintf</span>(<span class="st">' (%s)'</span>, <span class="fu">format_scientific</span>(family_ra_reads_tot)))</span>
<span id="cb34-1364"><a href="#cb34-1364" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb34-1365"><a href="#cb34-1365" aria-hidden="true" tabindex="-1"></a>links <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">create_links</span>(data))</span>
<span id="cb34-1366"><a href="#cb34-1366" aria-hidden="true" tabindex="-1"></a>nodes <span class="ot">&lt;-</span> <span class="fu">create_nodes</span>(links)</span>
<span id="cb34-1367"><a href="#cb34-1367" aria-hidden="true" tabindex="-1"></a>sankey_data <span class="ot">&lt;-</span> <span class="fu">prepare_sankey_data</span>(links, nodes)</span>
<span id="cb34-1368"><a href="#cb34-1368" aria-hidden="true" tabindex="-1"></a>sankey <span class="ot">&lt;-</span> <span class="fu">create_sankey_plot</span>(sankey_data)</span>
<span id="cb34-1369"><a href="#cb34-1369" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1370"><a href="#cb34-1370" aria-hidden="true" tabindex="-1"></a>sankey</span>
<span id="cb34-1371"><a href="#cb34-1371" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->




</body></html>