<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.53">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Harmon Bhasin">
<meta name="dcterms.date" content="2024-09-05">

<title>Workflow of O’Connell et al.&nbsp;(2023) – Harmon's Public Notebook</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script async="" src="https://hypothes.is/embed.js"></script>
<script>
  window.document.addEventListener("DOMContentLoaded", function (_event) {
    document.body.classList.add('hypothesis-enabled');
  });
</script>
<style>

      .quarto-title-block .quarto-title-banner {
        background: #de2d26;
      }
</style>
<script src="../site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<script src="../site_libs/d3-4.5.0/d3.min.js"></script>
<script src="../site_libs/sankey-1/sankey.js"></script>
<script src="../site_libs/sankeyNetwork-binding-0.4/sankeyNetwork.js"></script>
<link href="../site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet">
<script src="../site_libs/pagedtable-1.1/js/pagedtable.js"></script>

  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

<link rel="stylesheet" href="../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top quarto-banner">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">Harmon’s Public Notebook</span>
    </a>
  </div>
        <div class="quarto-navbar-tools tools-end">
</div>
          <div id="quarto-search" class="" title="Search"></div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-page-left">
      <div class="quarto-title-block"><div><h1 class="title">Workflow of O’Connell et al.&nbsp;(2023)</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
            <p class="subtitle lead">Whole blood from US (RNA)</p>
                      </div>
  </div>
    
  
  <div class="quarto-title-meta column-page-left">

      <div>
      <div class="quarto-title-meta-heading">Author</div>
      <div class="quarto-title-meta-contents">
               <p>Harmon Bhasin </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">September 5, 2024</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#the-raw-data" id="toc-the-raw-data" class="nav-link active" data-scroll-target="#the-raw-data"><span class="header-section-number">1</span> The raw data</a>
  <ul class="collapse">
  <li><a href="#about" id="toc-about" class="nav-link" data-scroll-target="#about"><span class="header-section-number">1.1</span> About</a>
  <ul class="collapse">
  <li><a href="#sample-library-preparation" id="toc-sample-library-preparation" class="nav-link" data-scroll-target="#sample-library-preparation"><span class="header-section-number">1.1.1</span> Sample + library preparation</a></li>
  </ul></li>
  <li><a href="#quality-control-metrics" id="toc-quality-control-metrics" class="nav-link" data-scroll-target="#quality-control-metrics"><span class="header-section-number">1.2</span> Quality control metrics</a></li>
  </ul></li>
  <li><a href="#preprocessing" id="toc-preprocessing" class="nav-link" data-scroll-target="#preprocessing"><span class="header-section-number">2</span> Preprocessing</a>
  <ul class="collapse">
  <li><a href="#summary" id="toc-summary" class="nav-link" data-scroll-target="#summary"><span class="header-section-number">2.1</span> Summary</a></li>
  <li><a href="#quality-control-metrics-1" id="toc-quality-control-metrics-1" class="nav-link" data-scroll-target="#quality-control-metrics-1"><span class="header-section-number">2.2</span> Quality control metrics</a></li>
  </ul></li>
  <li><a href="#taxonomic-composition" id="toc-taxonomic-composition" class="nav-link" data-scroll-target="#taxonomic-composition"><span class="header-section-number">3</span> Taxonomic composition</a>
  <ul class="collapse">
  <li><a href="#high-level-composition" id="toc-high-level-composition" class="nav-link" data-scroll-target="#high-level-composition"><span class="header-section-number">3.1</span> High-level composition</a></li>
  <li><a href="#total-viral-content" id="toc-total-viral-content" class="nav-link" data-scroll-target="#total-viral-content"><span class="header-section-number">3.2</span> Total viral content</a></li>
  <li><a href="#taxonomic-composition-of-viruses" id="toc-taxonomic-composition-of-viruses" class="nav-link" data-scroll-target="#taxonomic-composition-of-viruses"><span class="header-section-number">3.3</span> Taxonomic composition of viruses</a></li>
  </ul></li>
  <li><a href="#human-infecting-virus-reads" id="toc-human-infecting-virus-reads" class="nav-link" data-scroll-target="#human-infecting-virus-reads"><span class="header-section-number">4</span> Human-infecting virus reads</a>
  <ul class="collapse">
  <li><a href="#overall-relative-abundance" id="toc-overall-relative-abundance" class="nav-link" data-scroll-target="#overall-relative-abundance"><span class="header-section-number">4.1</span> Overall relative abundance</a></li>
  <li><a href="#overall-taxonomy-and-composition" id="toc-overall-taxonomy-and-composition" class="nav-link" data-scroll-target="#overall-taxonomy-and-composition"><span class="header-section-number">4.2</span> Overall taxonomy and composition</a></li>
  <li><a href="#relative-abundance-of-pathogenic-viruses-of-interest" id="toc-relative-abundance-of-pathogenic-viruses-of-interest" class="nav-link" data-scroll-target="#relative-abundance-of-pathogenic-viruses-of-interest"><span class="header-section-number">4.3</span> Relative abundance of pathogenic viruses of interest</a></li>
  <li><a href="#relative-abundance-assuming-perfect-human-read-removal" id="toc-relative-abundance-assuming-perfect-human-read-removal" class="nav-link" data-scroll-target="#relative-abundance-assuming-perfect-human-read-removal"><span class="header-section-number">4.4</span> Relative abundance assuming perfect human read removal</a></li>
  </ul></li>
  <li><a href="#conclusion" id="toc-conclusion" class="nav-link" data-scroll-target="#conclusion"><span class="header-section-number">5</span> Conclusion</a></li>
  <li><a href="#appendix" id="toc-appendix" class="nav-link" data-scroll-target="#appendix"><span class="header-section-number">6</span> Appendix</a>
  <ul class="collapse">
  <li><a href="#human-infecting-virus-families-genera-and-species" id="toc-human-infecting-virus-families-genera-and-species" class="nav-link" data-scroll-target="#human-infecting-virus-families-genera-and-species"><span class="header-section-number">6.1</span> Human-infecting virus families, genera, and species</a></li>
  </ul></li>
  </ul>
<div class="quarto-other-links"><h2>Other Links</h2><ul><li><a href="https://doi.org/10.1186/s12863-024-01223-z"><i class="bi bi-link-45deg"></i>Paper</a></li><li><a href="https://www.ncbi.nlm.nih.gov/sra/?term=SRP429744"><i class="bi bi-link-45deg"></i>Data</a></li></ul></div><div class="quarto-code-links"><h2>Code Links</h2><ul><li><a href="https://github.com/naobservatory/harmons-public-notebook/blob/main/notebooks/2024-09-05-oconnell.qmd"><i class="bi bi-file-code"></i>Code for this post</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content quarto-banner-title-block column-page-left" id="quarto-document-content">





<p>This is the first whole blood study of <a href="https://data.securebio.org/harmons-public-notebook/notebooks/2024-07-08_cebria-mendoza.html">this series</a>. In this post, I analyze <a href="https://doi.org/10.1186/s12863-024-01223-z">O’Connell 2023</a>, a dataset with 138 adult whole blood donors from Austin, TX.</p>
<p><em>This notebook uses the MGS Workflow v2.2.1 (note that this is old).</em></p>
<section id="the-raw-data" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> The raw data</h1>
<section id="about" class="level2" data-number="1.1">
<h2 data-number="1.1" class="anchored" data-anchor-id="about"><span class="header-section-number">1.1</span> About</h2>
<p><a href="https://www.ncbi.nlm.nih.gov/sra/?term=SRP429744">This dataset</a> is composed of 138 samples which come from 138 individuals in Austin, TX. They performed RNA sequencing on whole blood samples.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Import libraries and extract metadata from sample names</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>libraries <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(libraries_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="co">#meta_data &lt;- read_csv('/Users/harmonbhasin/work/securebio/nao-harmon/thijssen2023/preprocessing/SraRunTable.txt') %&gt;%</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="co">#  select('Library Name', Run) %&gt;% rename(sample = Run, library = 'Library Name')</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="co">#libraries &lt;- libraries %&gt;%</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="co">#  left_join(meta_data, by = 'sample') %&gt;%</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="co">#  select(sample, library=library.y) %&gt;% </span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="co">#  filter(library != 'PCL21') %&gt;%</span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="co">#  mutate(library = factor(library, levels = c(paste0('PCL', 1:20))))</span></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="co">#libraries</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="sample-library-preparation" class="level3" data-number="1.1.1">
<h3 data-number="1.1.1" class="anchored" data-anchor-id="sample-library-preparation"><span class="header-section-number">1.1.1</span> Sample + library preparation</h3>
<p>Paraphrased from the paper:</p>
<blockquote class="blockquote">
<p>Whole blood specimens were collected from 138 adult donors via PAXgene vacutainers at admission to the Emergency Department at Dell-Seton Medical Center (Austin, TX). Total RNA was isolated from archived PAXgene-stabilized whole blood using the PAXgene IVD Blood RNA Kit. Ribosomal RNA and globin mRNA-depleted cDNA libraries were prepared from 500 ng of total RNA using the Illumina TruSeq Stranded Total RNA Ribo-Zero Globin kit. Paired-end 150 bp sequencing was performed on the Illumina NovaSeq 6000 platform.</p>
</blockquote>
<p><em>More details can be found <a href="https://www.sciencedirect.com/science/article/pii/S0888754323001520?via%3Dihub#s0035">here</a>.</em></p>
</section>
</section>
<section id="quality-control-metrics" class="level2" data-number="1.2">
<h2 data-number="1.2" class="anchored" data-anchor-id="quality-control-metrics"><span class="header-section-number">1.2</span> Quality control metrics</h2>
<p>In total, these 138 samples contained ~3B read pairs. The samples had ~20.1M - 46.9M (mean ~22.5M) read pairs each. The number of reads looks pretty good, with a few samples having a much higher count. The total number of base pairs also looks reasonable, matching the trends of the number of reads. The duplication rate is high. Adapter content is quite low, although we can see library contamination (illumina_universal_adapter), and polya/polyg contamination. As we’d expect, we see higher quality Q scores near the beginning of the read and a gradual decline towards the end of the read, however all positions seem to have a Q score of around 35 which means that our reads are ~99.97% accurate. When looking at the Q score over all sequences, we can see a sharp peak around 35, which corresponds to the previous plot, indicating high read quality.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare data</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>basic_stats_raw_metrics <span class="ot">&lt;-</span> basic_stats_raw <span class="sc">%&gt;%</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(library,</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at"># Read pairs</span><span class="st">`</span> <span class="ot">=</span> n_read_pairs,</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Total base pairs</span><span class="sc">\n</span><span class="at">(approx)</span><span class="st">`</span> <span class="ot">=</span> n_bases_approx,</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">% Duplicates</span><span class="sc">\n</span><span class="at">(FASTQC)</span><span class="st">`</span> <span class="ot">=</span> percent_duplicates) <span class="sc">%&gt;%</span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>library, <span class="at">names_to =</span> <span class="st">"metric"</span>, <span class="at">values_to =</span> <span class="st">"value"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">metric =</span> <span class="fu">fct_inorder</span>(metric)) </span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Set up plot templates</span></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>g_basic <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(basic_stats_raw_metrics, <span class="fu">aes</span>(<span class="at">x =</span> library, <span class="at">y =</span> value)) <span class="sc">+</span></span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position =</span> <span class="st">"dodge"</span>) <span class="sc">+</span></span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>() <span class="sc">+</span></span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">expand_limits</span>(<span class="at">y =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">100</span>)) <span class="sc">+</span></span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(metric <span class="sc">~</span> ., <span class="at">scales =</span> <span class="st">"free"</span>, <span class="at">space =</span> <span class="st">"free_x"</span>, <span class="at">switch =</span> <span class="st">"y"</span>) <span class="sc">+</span></span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a>  theme_kit <span class="sc">+</span> <span class="fu">theme</span>(</span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">strip.text.y =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"plain"</span>),</span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>()</span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a>g_basic</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="2024-09-05-oconnell_files/figure-html/plot-basic-stats-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Set up plotting templates</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>g_qual_raw <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">linetype=</span>read_pair, <span class="at">group=</span><span class="fu">interaction</span>(sample,read_pair))) <span class="sc">+</span> </span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_linetype_discrete</span>(<span class="at">name =</span> <span class="st">"Read Pair"</span>) <span class="sc">+</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color=</span><span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>),</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">linetype =</span> <span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>)) <span class="sc">+</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>  theme_base</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize adapters</span></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>g_adapters_raw <span class="ot">&lt;-</span> g_qual_raw <span class="sc">+</span> </span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>pc_adapters), <span class="at">data=</span>adapter_stats_raw) <span class="sc">+</span></span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Adapters"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">10</span>,<span class="dv">1</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Position"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">500</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(.<span class="sc">~</span>adapter)</span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a>g_adapters_raw</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="2024-09-05-oconnell_files/figure-html/plot-raw-quality-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize quality</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>g_quality_base_raw <span class="ot">&lt;-</span> g_qual_raw <span class="sc">+</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>mean_phred_score), <span class="at">data=</span>quality_base_stats_raw) <span class="sc">+</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"Mean Phred score"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">10</span>,<span class="dv">45</span>)) <span class="sc">+</span></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Position"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">500</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>g_quality_base_raw</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="2024-09-05-oconnell_files/figure-html/plot-raw-quality-2.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>g_quality_seq_raw <span class="ot">&lt;-</span> g_qual_raw <span class="sc">+</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>mean_phred_score, <span class="at">y=</span>n_sequences), <span class="at">data=</span>quality_seq_stats_raw) <span class="sc">+</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Mean Phred score"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"# Sequences"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>g_quality_seq_raw</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="2024-09-05-oconnell_files/figure-html/plot-raw-quality-3.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="preprocessing" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Preprocessing</h1>
<section id="summary" class="level2" data-number="2.1">
<h2 data-number="2.1" class="anchored" data-anchor-id="summary"><span class="header-section-number">2.1</span> Summary</h2>
<p>The average fraction of reads at each stage in the preprocessing pipeline is shown in the following table. Adapter trimming &amp; filtering doesn’t lose too many reads. Deduplication loses us about 27% of reads on average, then ribodepletion only loses as about 1% on average.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># </span><span class="al">TODO</span><span class="co">: Group by pool size as well</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Count read losses</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>n_reads_rel <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> </span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, stage, percent_duplicates, n_read_pairs) <span class="sc">%&gt;%</span></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(sample, stage) <span class="sc">%&gt;%</span></span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_retained =</span> n_read_pairs <span class="sc">/</span> <span class="fu">lag</span>(n_read_pairs),</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_lost =</span> <span class="dv">1</span> <span class="sc">-</span> p_reads_retained,</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_retained_abs =</span> n_read_pairs <span class="sc">/</span> n_read_pairs[<span class="dv">1</span>],</span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_lost_abs =</span> <span class="dv">1</span><span class="sc">-</span>p_reads_retained_abs,</span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_lost_abs_marginal =</span> p_reads_lost_abs <span class="sc">-</span> <span class="fu">lag</span>(p_reads_lost_abs))</span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>n_reads_rel_display <span class="ot">&lt;-</span> n_reads_rel <span class="sc">%&gt;%</span> </span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">Stage=</span>stage) <span class="sc">%&gt;%</span> </span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Stage) <span class="sc">%&gt;%</span> </span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="st">`</span><span class="at">% Total Reads Lost (Cumulative)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">round</span>(<span class="fu">min</span>(p_reads_lost_abs<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">"-"</span>, <span class="fu">round</span>(<span class="fu">max</span>(p_reads_lost_abs<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">" (mean "</span>, <span class="fu">round</span>(<span class="fu">mean</span>(p_reads_lost_abs<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">")"</span>),</span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a>            <span class="st">`</span><span class="at">% Total Reads Lost (Marginal)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">round</span>(<span class="fu">min</span>(p_reads_lost_abs_marginal<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">"-"</span>, <span class="fu">round</span>(<span class="fu">max</span>(p_reads_lost_abs_marginal<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">" (mean "</span>, <span class="fu">round</span>(<span class="fu">mean</span>(p_reads_lost_abs_marginal<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">")"</span>), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Stage <span class="sc">!=</span> <span class="st">"raw_concat"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Stage =</span> Stage <span class="sc">%&gt;%</span> as.numeric <span class="sc">%&gt;%</span> <span class="fu">factor</span>(<span class="at">labels=</span><span class="fu">c</span>(<span class="st">"Trimming &amp; filtering"</span>, <span class="st">"Deduplication"</span>, <span class="st">"Initial ribodepletion"</span>, <span class="st">"Secondary ribodepletion"</span>)))</span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a>n_reads_rel_display</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["Stage"],"name":[1],"type":["fct"],"align":["left"]},{"label":["% Total Reads Lost (Cumulative)"],"name":[2],"type":["chr"],"align":["left"]},{"label":["% Total Reads Lost (Marginal)"],"name":[3],"type":["chr"],"align":["left"]}],"data":[{"1":"Trimming & filtering","2":"0.6-2.1 (mean 1)","3":"0.6-2.1 (mean 1)"},{"1":"Deduplication","2":"23.5-58.5 (mean 27.8)","3":"22.5-56.5 (mean 26.8)"},{"1":"Initial ribodepletion","2":"24.4-59.9 (mean 29)","3":"0.6-4.1 (mean 1.2)"},{"1":"Secondary ribodepletion","2":"24.5-60.1 (mean 29.1)","3":"0.1-0.6 (mean 0.1)"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
</section>
<section id="quality-control-metrics-1" class="level2" data-number="2.2">
<h2 data-number="2.2" class="anchored" data-anchor-id="quality-control-metrics-1"><span class="header-section-number">2.2</span> Quality control metrics</h2>
<p>Trimming + cleaning gets rid of the library contamination, however there is still some polya + polyg contamination, however given that it makes a small portion of adapter content, I think we’re fine ignoring it. Read quality slightly improves after preprocessing + cleaning, this is noticable near the end positions. When looking at the q score over all reads, we see that cleaning gives us most of our improvement.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>g_qual <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">linetype=</span>read_pair, <span class="at">group=</span><span class="fu">interaction</span>(sample,read_pair))) <span class="sc">+</span> </span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_linetype_discrete</span>(<span class="at">name =</span> <span class="st">"Read Pair"</span>) <span class="sc">+</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color=</span><span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>),</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">linetype =</span> <span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>)) <span class="sc">+</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>  theme_base</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize adapters</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>g_adapters <span class="ot">&lt;-</span> g_qual <span class="sc">+</span> </span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>pc_adapters), <span class="at">data=</span>adapter_stats) <span class="sc">+</span></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Adapters"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">20</span>),</span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">50</span>,<span class="dv">10</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Position"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">140</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(stage<span class="sc">~</span>adapter)</span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a>g_adapters</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="2024-09-05-oconnell_files/figure-html/plot-quality-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize quality</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>g_quality_base <span class="ot">&lt;-</span> g_qual <span class="sc">+</span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>mean_phred_score), <span class="at">data=</span>quality_base_stats) <span class="sc">+</span></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"Mean Phred score"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">10</span>,<span class="dv">45</span>)) <span class="sc">+</span></span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Position"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">140</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(stage<span class="sc">~</span>.)</span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>g_quality_base</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="2024-09-05-oconnell_files/figure-html/plot-quality-2.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>g_quality_seq <span class="ot">&lt;-</span> g_qual <span class="sc">+</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>mean_phred_score, <span class="at">y=</span>n_sequences), <span class="at">data=</span>quality_seq_stats) <span class="sc">+</span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Mean Phred score"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"# Sequences"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(stage<span class="sc">~</span>., <span class="at">scales =</span> <span class="st">"free_y"</span>)</span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>g_quality_seq</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="2024-09-05-oconnell_files/figure-html/plot-quality-3.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<p>These plots below show the trends from above in each sample. Everything looks pretty good.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>g_stage_trace <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(basic_stats, <span class="fu">aes</span>(<span class="at">x=</span>stage, <span class="at">group=</span>sample)) <span class="sc">+</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot reads over preprocessing</span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>g_reads_stages <span class="ot">&lt;-</span> g_stage_trace <span class="sc">+</span></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>n_read_pairs)) <span class="sc">+</span></span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"# Read pairs"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>))</span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>g_reads_stages</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="2024-09-05-oconnell_files/figure-html/preproc-figures-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot relative read losses during preprocessing</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>g_reads_rel <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(n_reads_rel, </span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">aes</span>(<span class="at">x=</span>stage, <span class="at">group=</span>sample)) <span class="sc">+</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>p_reads_lost_abs_marginal)) <span class="sc">+</span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"% Total Reads Lost"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), </span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="cf">function</span>(x) x<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>g_reads_rel</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="2024-09-05-oconnell_files/figure-html/preproc-figures-2.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>stage_dup <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(stage) <span class="sc">%&gt;%</span> </span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">dmin =</span> <span class="fu">min</span>(percent_duplicates), <span class="at">dmax=</span><span class="fu">max</span>(percent_duplicates),</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>            <span class="at">dmean=</span><span class="fu">mean</span>(percent_duplicates), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>g_dup_stages <span class="ot">&lt;-</span> g_stage_trace <span class="sc">+</span></span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>percent_duplicates)) <span class="sc">+</span></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"% Duplicates"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>g_dup_stages</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="2024-09-05-oconnell_files/figure-html/preproc-dedup-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>g_readlen_stages <span class="ot">&lt;-</span> g_stage_trace <span class="sc">+</span> <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>mean_seq_len)) <span class="sc">+</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"Mean read length (nt)"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>))</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>g_readlen_stages</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="2024-09-05-oconnell_files/figure-html/preproc-dedup-2.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate reads lost during ribodepletion (approximation for % ribosomal reads)</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>reads_ribo <span class="ot">&lt;-</span> n_reads_rel <span class="sc">%&gt;%</span> </span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(stage <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"dedup"</span>, <span class="st">"ribo_secondary"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads_ribo=</span><span class="dv">1</span><span class="sc">-</span>n_read_pairs[<span class="dv">2</span>]<span class="sc">/</span>n_read_pairs[<span class="dv">1</span>], <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by =</span> <span class="st">'sample'</span>)</span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>reads_ribo_summ <span class="ot">&lt;-</span> reads_ribo <span class="sc">%&gt;%</span></span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">min=</span><span class="fu">min</span>(p_reads_ribo), <span class="at">max=</span><span class="fu">max</span>(p_reads_ribo),</span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>            <span class="at">mean=</span><span class="fu">mean</span>(p_reads_ribo), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by =</span> <span class="st">'sample'</span>)</span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a>g_reads_ribo <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(reads_ribo, </span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">aes</span>(<span class="at">x=</span>library, <span class="at">y=</span>p_reads_ribo)) <span class="sc">+</span></span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span> </span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"Approx % ribosomal reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>),</span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>)<span class="sc">+</span></span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb14-18"><a href="#cb14-18" aria-hidden="true" tabindex="-1"></a>g_reads_ribo</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="2024-09-05-oconnell_files/figure-html/ribo-frac-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="taxonomic-composition" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> Taxonomic composition</h1>
<section id="high-level-composition" class="level2" data-number="3.1">
<h2 data-number="3.1" class="anchored" data-anchor-id="high-level-composition"><span class="header-section-number">3.1</span> High-level composition</h2>
<p>To assess the high-level composition of the reads, I ran the ribodepleted files through Kraken2 and summarized the results with Bracken.</p>
<p>The groups listed below were created by Will:</p>
<ul>
<li>Filtered (removed during cleaning)</li>
<li>Duplicate (removed during deduplication)</li>
<li>Ribosomal (removed during ribodepletion)</li>
<li>Unassigned (non-ribosomal reads that were not assigned to any taxon by Kraken/Bracken)</li>
<li>Bacterial (non-ribosomal reads assigned to the Bacteria domain by Kraken/Bracken)</li>
<li>Archaeal (non-ribosomal reads assigned to the Archaea domain by Kraken/Bracken)</li>
<li>Viral (non-ribosomal reads assigned to the Viruses domain by Kraken/Bracken)</li>
<li>Human (non-ribosomal reads assigned to the Eukarya domain by Kraken/Bracken)</li>
</ul>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>classifications <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Filtered"</span>, <span class="st">"Duplicate"</span>, <span class="st">"Ribosomal"</span>, <span class="st">"Unassigned"</span>,</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>                     <span class="st">"Bacterial"</span>, <span class="st">"Archaeal"</span>, <span class="st">"Viral"</span>, <span class="st">"Human"</span>)</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Import composition data</span></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>tax_final_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(results_dir, <span class="st">"taxonomy_final"</span>)</span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>comp_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(tax_final_dir, <span class="st">"taxonomic_composition.tsv.gz"</span>)</span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>comp <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(comp_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>)</span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a>comp_minor <span class="ot">&lt;-</span> comp <span class="sc">%&gt;%</span> </span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(classification <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Archaeal"</span>, <span class="st">"Viral"</span>, <span class="st">"Human"</span>, <span class="st">"Other"</span>))</span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a>comp_assigned <span class="ot">&lt;-</span> comp <span class="sc">%&gt;%</span></span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span> classification <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Filtered"</span>, <span class="st">"Duplicate"</span>, </span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">"Ribosomal"</span>, <span class="st">"Unassigned"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads =</span> n_reads<span class="sc">/</span><span class="fu">sum</span>(n_reads))</span>
<span id="cb15-15"><a href="#cb15-15" aria-hidden="true" tabindex="-1"></a>comp_assigned_minor <span class="ot">&lt;-</span> comp_assigned <span class="sc">%&gt;%</span> </span>
<span id="cb15-16"><a href="#cb15-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(classification <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Archaeal"</span>, <span class="st">"Viral"</span>, <span class="st">"Human"</span>, <span class="st">"Other"</span>))</span>
<span id="cb15-17"><a href="#cb15-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-18"><a href="#cb15-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Summarize composition</span></span>
<span id="cb15-19"><a href="#cb15-19" aria-hidden="true" tabindex="-1"></a>read_comp_summ <span class="ot">&lt;-</span> comp <span class="sc">%&gt;%</span> </span>
<span id="cb15-20"><a href="#cb15-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(classification) <span class="sc">%&gt;%</span></span>
<span id="cb15-21"><a href="#cb15-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads =</span> <span class="fu">sum</span>(n_reads), <span class="at">.groups =</span> <span class="st">"drop_last"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb15-22"><a href="#cb15-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">n_reads =</span> <span class="fu">replace_na</span>(n_reads,<span class="dv">0</span>),</span>
<span id="cb15-23"><a href="#cb15-23" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads =</span> n_reads<span class="sc">/</span><span class="fu">sum</span>(n_reads),</span>
<span id="cb15-24"><a href="#cb15-24" aria-hidden="true" tabindex="-1"></a>         <span class="at">pc_reads =</span> p_reads<span class="sc">*</span><span class="dv">100</span>) <span class="sc">%&gt;%</span></span>
<span id="cb15-25"><a href="#cb15-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">classification =</span> <span class="fu">factor</span>(classification, <span class="at">levels=</span>classifications)) <span class="sc">%&gt;%</span></span>
<span id="cb15-26"><a href="#cb15-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(classification, n_reads, pc_reads) <span class="sc">%&gt;%</span></span>
<span id="cb15-27"><a href="#cb15-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">`</span><span class="at"># of reads</span><span class="st">`</span> <span class="ot">=</span> n_reads, <span class="st">"% of reads"</span> <span class="ot">=</span> pc_reads) <span class="sc">%&gt;%</span></span>
<span id="cb15-28"><a href="#cb15-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">% of reads</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sprintf</span>(<span class="st">"%.2f"</span>, <span class="st">`</span><span class="at">% of reads</span><span class="st">`</span>))</span>
<span id="cb15-29"><a href="#cb15-29" aria-hidden="true" tabindex="-1"></a>read_comp_summ</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["classification"],"name":[1],"type":["fct"],"align":["left"]},{"label":["# of reads"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["% of reads"],"name":[3],"type":["chr"],"align":["left"]}],"data":[{"1":"Archaeal","2":"836","3":"0.00"},{"1":"Bacterial","2":"1108806","3":"0.04"},{"1":"Duplicate","2":"834295220","3":"26.89"},{"1":"Filtered","2":"31542742","3":"1.02"},{"1":"Human","2":"2192104363","3":"70.66"},{"1":"Ribosomal","2":"41206479","3":"1.33"},{"1":"Unassigned","2":"1913558","3":"0.06"},{"1":"Viral","2":"2392","3":"0.00"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare plotting templates</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>g_comp_base <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">x=</span>library, <span class="at">y=</span>p_reads, <span class="at">fill=</span>classification)) <span class="sc">+</span></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">name=</span><span class="st">"Plasma pool"</span>) <span class="sc">+</span></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>  theme_kit <span class="sc">+</span> </span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust=</span><span class="dv">0</span>, <span class="at">face=</span><span class="st">"plain"</span>, <span class="at">size=</span><span class="fu">rel</span>(<span class="fl">1.5</span>)))</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>scale_y_pc_reads <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">partial</span>(scale_y_continuous, <span class="at">name =</span> <span class="st">"% Reads"</span>,</span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>)</span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>geom_comp <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">partial</span>(geom_col, <span class="at">position =</span> <span class="st">"stack"</span>, <span class="at">width =</span> <span class="dv">1</span>)</span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Define a color palette for the classification</span></span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a>classification_colors <span class="ot">&lt;-</span> <span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">"Accent"</span>)</span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(classification_colors) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Filtered"</span>, <span class="st">"Duplicate"</span>, <span class="st">"Ribosomal"</span>, <span class="st">"Unassigned"</span>, <span class="st">"Bacterial"</span>, <span class="st">"Archaeal"</span>, <span class="st">"Viral"</span>, <span class="st">"Human"</span>)</span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a>scale_fill_classification <span class="ot">&lt;-</span> <span class="cf">function</span>() {</span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> classification_colors, <span class="at">name =</span> <span class="st">"Classification"</span>)</span>
<span id="cb16-15"><a href="#cb16-15" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb16-16"><a href="#cb16-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-17"><a href="#cb16-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot overall composition</span></span>
<span id="cb16-18"><a href="#cb16-18" aria-hidden="true" tabindex="-1"></a>g_comp <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> <span class="fu">geom_comp</span>(<span class="at">data =</span> comp) <span class="sc">+</span></span>
<span id="cb16-19"><a href="#cb16-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_pc_reads</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>)) <span class="sc">+</span></span>
<span id="cb16-20"><a href="#cb16-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_classification</span>() <span class="sc">+</span> </span>
<span id="cb16-21"><a href="#cb16-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Read composition (all reads, all groups)"</span>)</span>
<span id="cb16-22"><a href="#cb16-22" aria-hidden="true" tabindex="-1"></a>g_comp</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="2024-09-05-oconnell_files/figure-html/plot-composition-all-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>g_comp_assigned <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_comp</span>(<span class="at">data =</span> comp_assigned) <span class="sc">+</span></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_pc_reads</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>)) <span class="sc">+</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_classification</span>() <span class="sc">+</span> </span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Read composition (assigned reads, all groups)"</span>)</span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>g_comp_assigned</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="2024-09-05-oconnell_files/figure-html/plot-composition-all-2.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="total-viral-content" class="level2" data-number="3.2">
<h2 data-number="3.2" class="anchored" data-anchor-id="total-viral-content"><span class="header-section-number">3.2</span> Total viral content</h2>
<p>Total viral fraction average <span class="math inline">\(7.70 \times 10^{-7}\)</span> across samples. As a fraction of assigned (rather than total) reads, this jumped to <span class="math inline">\(1.18 \times 10^{-6}\)</span>:</p>
</section>
<section id="taxonomic-composition-of-viruses" class="level2" data-number="3.3">
<h2 data-number="3.3" class="anchored" data-anchor-id="taxonomic-composition-of-viruses"><span class="header-section-number">3.3</span> Taxonomic composition of viruses</h2>
<p>The two dominant viruses we see are Anellovirdae and Phycodnaviridae. The threshold for the label “other” are the set of families that make up less than 30% composition in all samples.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>major_threshold <span class="ot">&lt;-</span> <span class="fl">0.3</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify major viral families</span></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>viral_families_major_tab <span class="ot">&lt;-</span> viral_families <span class="sc">%&gt;%</span> </span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads_viral_max =</span> <span class="fu">max</span>(p_reads_viral), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_viral_max <span class="sc">&gt;=</span> major_threshold)</span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>viral_families_major_list <span class="ot">&lt;-</span> viral_families_major_tab <span class="sc">%&gt;%</span> <span class="fu">pull</span>(name)</span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>viral_families_major <span class="ot">&lt;-</span> viral_families <span class="sc">%&gt;%</span> </span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(name <span class="sc">%in%</span> viral_families_major_list) <span class="sc">%&gt;%</span></span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, taxid, sample, p_reads_viral)</span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a>viral_families_minor <span class="ot">&lt;-</span> viral_families_major <span class="sc">%&gt;%</span> </span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads_viral_major =</span> <span class="fu">sum</span>(p_reads_viral), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="st">"Other"</span>, <span class="at">taxid=</span><span class="cn">NA</span>, <span class="at">p_reads_viral =</span> <span class="dv">1</span><span class="sc">-</span>p_reads_viral_major) <span class="sc">%&gt;%</span></span>
<span id="cb18-16"><a href="#cb18-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, taxid, sample, p_reads_viral)</span>
<span id="cb18-17"><a href="#cb18-17" aria-hidden="true" tabindex="-1"></a>viral_families_display <span class="ot">&lt;-</span> viral_families_major <span class="sc">%&gt;%</span> </span>
<span id="cb18-18"><a href="#cb18-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_rows</span>(viral_families_minor) <span class="sc">%&gt;%</span></span>
<span id="cb18-19"><a href="#cb18-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_viral)) <span class="sc">%&gt;%</span> </span>
<span id="cb18-20"><a href="#cb18-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">factor</span>(name, <span class="at">levels=</span><span class="fu">c</span>(viral_families_major_list, <span class="st">"Other"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb18-21"><a href="#cb18-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_viral, <span class="at">classification=</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb18-22"><a href="#cb18-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">'sample'</span>)</span>
<span id="cb18-23"><a href="#cb18-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-24"><a href="#cb18-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-25"><a href="#cb18-25" aria-hidden="true" tabindex="-1"></a>pal <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">'Dark2'</span>),<span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">'Accent'</span>), <span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">'Set1'</span>), <span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">'Pastel1'</span>), <span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">'Pastel2'</span>))</span>
<span id="cb18-26"><a href="#cb18-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-27"><a href="#cb18-27" aria-hidden="true" tabindex="-1"></a>n_classifications <span class="ot">&lt;-</span> <span class="fu">length</span>(<span class="fu">unique</span>(viral_families_display<span class="sc">$</span>classification)) <span class="sc">-</span> <span class="dv">1</span></span>
<span id="cb18-28"><a href="#cb18-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-29"><a href="#cb18-29" aria-hidden="true" tabindex="-1"></a>custom_palette_with_black <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb18-30"><a href="#cb18-30" aria-hidden="true" tabindex="-1"></a>  pal[<span class="dv">1</span><span class="sc">:</span>n_classifications],</span>
<span id="cb18-31"><a href="#cb18-31" aria-hidden="true" tabindex="-1"></a>  <span class="st">"black"</span></span>
<span id="cb18-32"><a href="#cb18-32" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb18-33"><a href="#cb18-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-34"><a href="#cb18-34" aria-hidden="true" tabindex="-1"></a>g_families <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> viral_families_display, <span class="fu">aes</span>(<span class="at">x =</span> library, <span class="at">y =</span> p_reads, <span class="at">fill =</span> classification)) <span class="sc">+</span></span>
<span id="cb18-35"><a href="#cb18-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position =</span> <span class="st">"stack"</span>, <span class="at">width =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb18-36"><a href="#cb18-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="st">"% Viral Reads"</span>, </span>
<span id="cb18-37"><a href="#cb18-37" aria-hidden="true" tabindex="-1"></a>                     <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fl">1.01</span>), </span>
<span id="cb18-38"><a href="#cb18-38" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="fl">0.2</span>),</span>
<span id="cb18-39"><a href="#cb18-39" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>), </span>
<span id="cb18-40"><a href="#cb18-40" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="cf">function</span>(y) y <span class="sc">*</span> <span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb18-41"><a href="#cb18-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> custom_palette_with_black) <span class="sc">+</span></span>
<span id="cb18-42"><a href="#cb18-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">name =</span> <span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb18-43"><a href="#cb18-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb18-44"><a href="#cb18-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb18-45"><a href="#cb18-45" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>),</span>
<span id="cb18-46"><a href="#cb18-46" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid.major.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb18-47"><a href="#cb18-47" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid.minor.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb18-48"><a href="#cb18-48" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">"bottom"</span></span>
<span id="cb18-49"><a href="#cb18-49" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb18-50"><a href="#cb18-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Viral Family Composition"</span>,</span>
<span id="cb18-51"><a href="#cb18-51" aria-hidden="true" tabindex="-1"></a>       <span class="at">fill =</span> <span class="st">"Classification"</span>)</span>
<span id="cb18-52"><a href="#cb18-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-53"><a href="#cb18-53" aria-hidden="true" tabindex="-1"></a>g_families</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="2024-09-05-oconnell_files/figure-html/viral-family-composition-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="human-infecting-virus-reads" class="level1" data-number="4">
<h1 data-number="4"><span class="header-section-number">4</span> Human-infecting virus reads</h1>
<section id="overall-relative-abundance" class="level2" data-number="4.1">
<h2 data-number="4.1" class="anchored" data-anchor-id="overall-relative-abundance"><span class="header-section-number">4.1</span> Overall relative abundance</h2>
<p>I calculated the relative abundance of human-infecting viruses in two ways:</p>
<ul>
<li><p>First, as the total number of deduplicated human-virus reads in each sample, divided by the number of raw reads (“All reads”). This results in a viral fraction of <span class="math inline">\(2.62 \times 10^{-8}\)</span> across all samples</p></li>
<li><p>Second, as a fraction of preprocessed (cleaned, deduplicated, computationally ribodepleted) reads (“Preprocessed reads”). This results in a viral fraction of <span class="math inline">\(3.73 \times 10^{-8}\)</span> across all samples.</p></li>
</ul>
</section>
<section id="overall-taxonomy-and-composition" class="level2" data-number="4.2">
<h2 data-number="4.2" class="anchored" data-anchor-id="overall-taxonomy-and-composition"><span class="header-section-number">4.2</span> Overall taxonomy and composition</h2>
<p>Composition of HV reads was changed from when looking at all viral reads. Only 1 out of the 137 samples had reads for human infecting viruses, with read composition coming from well established viral families in whole blood.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>threshold_major_family <span class="ot">&lt;-</span> <span class="fl">0.01</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Count reads for each human-viral family</span></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>hv_family_counts <span class="ot">&lt;-</span> hv_reads_family <span class="sc">%&gt;%</span> </span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_hv =</span> n_reads_hv<span class="sc">/</span><span class="fu">sum</span>(n_reads_hv))</span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify high-ranking families and group others</span></span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a>hv_family_major_tab <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span> </span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_hv <span class="sc">==</span> <span class="fu">max</span>(p_reads_hv)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_hv)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(p_reads_hv <span class="sc">&gt;</span> threshold_major_family)</span>
<span id="cb19-14"><a href="#cb19-14" aria-hidden="true" tabindex="-1"></a>hv_family_counts_major <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span></span>
<span id="cb19-15"><a href="#cb19-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">ifelse</span>(name <span class="sc">%in%</span> hv_family_major_tab<span class="sc">$</span>name, name, <span class="st">"Other"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb19-16"><a href="#cb19-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, name_display) <span class="sc">%&gt;%</span></span>
<span id="cb19-17"><a href="#cb19-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_hv =</span> <span class="fu">sum</span>(n_reads_hv), <span class="at">p_reads_hv =</span> <span class="fu">sum</span>(p_reads_hv), </span>
<span id="cb19-18"><a href="#cb19-18" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-19"><a href="#cb19-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">factor</span>(name_display, </span>
<span id="cb19-20"><a href="#cb19-20" aria-hidden="true" tabindex="-1"></a>                               <span class="at">levels =</span> <span class="fu">c</span>(hv_family_major_tab<span class="sc">$</span>name, <span class="st">"Other"</span>)))</span>
<span id="cb19-21"><a href="#cb19-21" aria-hidden="true" tabindex="-1"></a>hv_family_counts_display <span class="ot">&lt;-</span> hv_family_counts_major <span class="sc">%&gt;%</span></span>
<span id="cb19-22"><a href="#cb19-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_hv, <span class="at">classification =</span> name_display) <span class="sc">%&gt;%</span></span>
<span id="cb19-23"><a href="#cb19-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by =</span> <span class="st">'sample'</span>)</span>
<span id="cb19-24"><a href="#cb19-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-25"><a href="#cb19-25" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb19-26"><a href="#cb19-26" aria-hidden="true" tabindex="-1"></a>g_hv_family <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb19-27"><a href="#cb19-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">data=</span>hv_family_counts_display, <span class="at">position =</span> <span class="st">"stack"</span>, <span class="at">width=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb19-28"><a href="#cb19-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% HV Reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb19-29"><a href="#cb19-29" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb19-30"><a href="#cb19-30" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb19-31"><a href="#cb19-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">'Accent'</span>, <span class="at">name =</span> <span class="st">"Viral family"</span>) <span class="sc">+</span></span>
<span id="cb19-32"><a href="#cb19-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">"Family composition of human-viral reads"</span>) <span class="sc">+</span></span>
<span id="cb19-33"><a href="#cb19-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">4</span>)) <span class="sc">+</span></span>
<span id="cb19-34"><a href="#cb19-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">1.4</span>), <span class="at">hjust=</span><span class="dv">0</span>, <span class="at">face=</span><span class="st">"plain"</span>))</span>
<span id="cb19-35"><a href="#cb19-35" aria-hidden="true" tabindex="-1"></a>g_hv_family</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="2024-09-05-oconnell_files/figure-html/hv-family-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get most prominent families for text</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>hv_family_collate <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span></span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_tot =</span> <span class="fu">sum</span>(n_reads_hv),</span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>            <span class="at">p_reads_max =</span> <span class="fu">max</span>(p_reads_hv), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n_reads_tot))</span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a>hv_family_collate</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["name"],"name":[1],"type":["chr"],"align":["left"]},{"label":["taxid"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["n_reads_tot"],"name":[3],"type":["int"],"align":["right"]},{"label":["p_reads_max"],"name":[4],"type":["dbl"],"align":["right"]}],"data":[{"1":"Anelloviridae","2":"687329","3":"3","4":"0.4285714"},{"1":"Orthoherpesviridae","2":"3044472","3":"3","4":"0.4285714"},{"1":"Poxviridae","2":"10240","3":"1","4":"0.1428571"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
</section>
<section id="relative-abundance-of-pathogenic-viruses-of-interest" class="level2" data-number="4.3">
<h2 data-number="4.3" class="anchored" data-anchor-id="relative-abundance-of-pathogenic-viruses-of-interest"><span class="header-section-number">4.3</span> Relative abundance of pathogenic viruses of interest</h2>
<p>Each dot represents a sample, colored by viral family. The x-axis shows the relative abundance of human-infecting viruses, and the y-axis shows the species.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>species_family <span class="ot">&lt;-</span> result <span class="sc">%&gt;%</span> <span class="fu">select</span>(species, family) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="st">'name'</span> <span class="ot">=</span> <span class="st">'species'</span>)</span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>play <span class="ot">&lt;-</span> hv_species_counts <span class="sc">%&gt;%</span> </span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by =</span> <span class="st">'sample'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(species_family, <span class="at">by =</span> <span class="st">'name'</span>)</span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a>adjusted_play <span class="ot">&lt;-</span> play <span class="sc">%&gt;%</span> </span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span></span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">virus_prevalence_num =</span> <span class="fu">n_distinct</span>(sample)<span class="sc">/</span><span class="fu">n_distinct</span>(libraries),</span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a>         <span class="at">total_reads_hv =</span> <span class="fu">sum</span>(n_reads_hv)) <span class="sc">%&gt;%</span></span>
<span id="cb21-12"><a href="#cb21-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb21-13"><a href="#cb21-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">fct_reorder</span>(name, virus_prevalence_num, <span class="at">.desc=</span><span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb21-14"><a href="#cb21-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, ra_reads_hv, family, virus_prevalence_num, total_reads_hv, n_reads_hv)</span>
<span id="cb21-15"><a href="#cb21-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-16"><a href="#cb21-16" aria-hidden="true" tabindex="-1"></a>pal <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">'Dark2'</span>),<span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">'Accent'</span>))</span>
<span id="cb21-17"><a href="#cb21-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-18"><a href="#cb21-18" aria-hidden="true" tabindex="-1"></a><span class="co">#, labels = label_log(digits=2)</span></span>
<span id="cb21-19"><a href="#cb21-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-20"><a href="#cb21-20" aria-hidden="true" tabindex="-1"></a>ra_dot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(adjusted_play, <span class="fu">aes</span>(<span class="at">x =</span> ra_reads_hv, <span class="at">y=</span>name)) <span class="sc">+</span></span>
<span id="cb21-21"><a href="#cb21-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_quasirandom</span>(<span class="at">orientation =</span> <span class="st">'y'</span>, <span class="fu">aes</span>(<span class="at">color =</span> family)) <span class="sc">+</span></span>
<span id="cb21-22"><a href="#cb21-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> pal) <span class="sc">+</span> </span>
<span id="cb21-23"><a href="#cb21-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_log10</span>(</span>
<span id="cb21-24"><a href="#cb21-24" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Relative abundance human virus reads"</span>,</span>
<span id="cb21-25"><a href="#cb21-25" aria-hidden="true" tabindex="-1"></a>    <span class="at">labels=</span><span class="fu">label_log</span>(<span class="at">digits=</span><span class="dv">3</span>)</span>
<span id="cb21-26"><a href="#cb21-26" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb21-27"><a href="#cb21-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span>  <span class="st">""</span>,</span>
<span id="cb21-28"><a href="#cb21-28" aria-hidden="true" tabindex="-1"></a>       <span class="at">color =</span> <span class="st">'Viral family'</span>) <span class="sc">+</span> </span>
<span id="cb21-29"><a href="#cb21-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color =</span> <span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">2</span>)) <span class="sc">+</span></span>
<span id="cb21-30"><a href="#cb21-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_light</span>() <span class="sc">+</span> </span>
<span id="cb21-31"><a href="#cb21-31" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(</span>
<span id="cb21-32"><a href="#cb21-32" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb21-33"><a href="#cb21-33" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>),</span>
<span id="cb21-34"><a href="#cb21-34" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.ticks.y =</span> <span class="fu">element_blank</span>(),<span class="at">axis.line =</span> <span class="fu">element_line</span>(<span class="at">colour =</span> <span class="st">"black"</span>),</span>
<span id="cb21-35"><a href="#cb21-35" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">14</span>),    </span>
<span id="cb21-36"><a href="#cb21-36" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb21-37"><a href="#cb21-37" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>, <span class="at">face=</span><span class="st">"bold"</span>),</span>
<span id="cb21-38"><a href="#cb21-38" aria-hidden="true" tabindex="-1"></a>    <span class="co">#legend.position = 'bottom',</span></span>
<span id="cb21-39"><a href="#cb21-39" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>),  <span class="co"># Move legend to top right</span></span>
<span id="cb21-40"><a href="#cb21-40" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.justification =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>),  <span class="co"># Align legend to top right</span></span>
<span id="cb21-41"><a href="#cb21-41" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb21-42"><a href="#cb21-42" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.border =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb21-43"><a href="#cb21-43" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.background =</span> <span class="fu">element_blank</span>())</span>
<span id="cb21-44"><a href="#cb21-44" aria-hidden="true" tabindex="-1"></a>ra_dot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="2024-09-05-oconnell_files/figure-html/plot-pathogenic-viruses-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="co">#  geom_text(aes(label = total_reads_hv, y = name), </span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="co">#            x = 1e-9, hjust = 0, vjust = 0.5, size = 3, </span></span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a><span class="co">#            check_overlap = TRUE)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>We see EBV, MPX (most likely contamination), and unclassified Anelloviridae, all of what are viruses we’d expect to see in whole blood.</p>
</section>
<section id="relative-abundance-assuming-perfect-human-read-removal" class="level2" data-number="4.4">
<h2 data-number="4.4" class="anchored" data-anchor-id="relative-abundance-assuming-perfect-human-read-removal"><span class="header-section-number">4.4</span> Relative abundance assuming perfect human read removal</h2>
<p>Assuming we’re able to perfectly remove all human reads, the average relative abundance of known human infecting virus is <span class="math inline">\(9.89 \times 10^{-7}\)</span>.</p>
</section>
</section>
<section id="conclusion" class="level1" data-number="5">
<h1 data-number="5"><span class="header-section-number">5</span> Conclusion</h1>
<p>There were some takeways from this analysis:</p>
<ol type="1">
<li>Within smaller datasets, we don’t expect to find human infecting viruses in samples.</li>
<li>Human reads tend to make up the majority of reads from whole blood.</li>
<li>Anelloviridae doesn’t seem to be as abundant in whole blood as it is in plasma.</li>
</ol>
</section>
<section id="appendix" class="level1" data-number="6">
<h1 data-number="6"><span class="header-section-number">6</span> Appendix</h1>
<section id="human-infecting-virus-families-genera-and-species" class="level2" data-number="6.1">
<h2 data-number="6.1" class="anchored" data-anchor-id="human-infecting-virus-families-genera-and-species"><span class="header-section-number">6.1</span> Human-infecting virus families, genera, and species</h2>
<p>To get a good overview of families, genera, and species, we can look at a Sankey plot where the magnitude of relative abundance, averaged over all samples, is shown in parentheses.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to create links</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>create_links <span class="ot">&lt;-</span> <span class="cf">function</span>(data) {</span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>  family_to_genus <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(genus)) <span class="sc">%&gt;%</span></span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(family, genus) <span class="sc">%&gt;%</span></span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">value =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">source =</span> family, <span class="at">target =</span> genus)</span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a>  genus_to_species <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb23-10"><a href="#cb23-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(genus, species) <span class="sc">%&gt;%</span></span>
<span id="cb23-11"><a href="#cb23-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">value =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb23-12"><a href="#cb23-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">source =</span> genus, <span class="at">target =</span> species)</span>
<span id="cb23-13"><a href="#cb23-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-14"><a href="#cb23-14" aria-hidden="true" tabindex="-1"></a>  family_to_species <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb23-15"><a href="#cb23-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="fu">is.na</span>(genus)) <span class="sc">%&gt;%</span></span>
<span id="cb23-16"><a href="#cb23-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(family, species) <span class="sc">%&gt;%</span></span>
<span id="cb23-17"><a href="#cb23-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">value =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb23-18"><a href="#cb23-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">source =</span> family, <span class="at">target =</span> species)</span>
<span id="cb23-19"><a href="#cb23-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-20"><a href="#cb23-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_rows</span>(family_to_genus, genus_to_species, family_to_species) <span class="sc">%&gt;%</span></span>
<span id="cb23-21"><a href="#cb23-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(source))</span>
<span id="cb23-22"><a href="#cb23-22" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb23-23"><a href="#cb23-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-24"><a href="#cb23-24" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to create nodes</span></span>
<span id="cb23-25"><a href="#cb23-25" aria-hidden="true" tabindex="-1"></a>create_nodes <span class="ot">&lt;-</span> <span class="cf">function</span>(links) {</span>
<span id="cb23-26"><a href="#cb23-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>(</span>
<span id="cb23-27"><a href="#cb23-27" aria-hidden="true" tabindex="-1"></a>    <span class="at">name =</span> <span class="fu">c</span>(links<span class="sc">$</span>source, links<span class="sc">$</span>target) <span class="sc">%&gt;%</span> <span class="fu">unique</span>()</span>
<span id="cb23-28"><a href="#cb23-28" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb23-29"><a href="#cb23-29" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb23-30"><a href="#cb23-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-31"><a href="#cb23-31" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to prepare data for Sankey diagram</span></span>
<span id="cb23-32"><a href="#cb23-32" aria-hidden="true" tabindex="-1"></a>prepare_sankey_data <span class="ot">&lt;-</span> <span class="cf">function</span>(links, nodes) {</span>
<span id="cb23-33"><a href="#cb23-33" aria-hidden="true" tabindex="-1"></a>  links<span class="sc">$</span>IDsource <span class="ot">&lt;-</span> <span class="fu">match</span>(links<span class="sc">$</span>source, nodes<span class="sc">$</span>name) <span class="sc">-</span> <span class="dv">1</span></span>
<span id="cb23-34"><a href="#cb23-34" aria-hidden="true" tabindex="-1"></a>  links<span class="sc">$</span>IDtarget <span class="ot">&lt;-</span> <span class="fu">match</span>(links<span class="sc">$</span>target, nodes<span class="sc">$</span>name) <span class="sc">-</span> <span class="dv">1</span></span>
<span id="cb23-35"><a href="#cb23-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list</span>(<span class="at">links =</span> links, <span class="at">nodes =</span> nodes)</span>
<span id="cb23-36"><a href="#cb23-36" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb23-37"><a href="#cb23-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-38"><a href="#cb23-38" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to create Sankey plot</span></span>
<span id="cb23-39"><a href="#cb23-39" aria-hidden="true" tabindex="-1"></a>create_sankey_plot <span class="ot">&lt;-</span> <span class="cf">function</span>(sankey_data) {</span>
<span id="cb23-40"><a href="#cb23-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sankeyNetwork</span>(</span>
<span id="cb23-41"><a href="#cb23-41" aria-hidden="true" tabindex="-1"></a>    <span class="at">Links =</span> sankey_data<span class="sc">$</span>links, </span>
<span id="cb23-42"><a href="#cb23-42" aria-hidden="true" tabindex="-1"></a>    <span class="at">Nodes =</span> sankey_data<span class="sc">$</span>nodes,</span>
<span id="cb23-43"><a href="#cb23-43" aria-hidden="true" tabindex="-1"></a>    <span class="at">Source =</span> <span class="st">"IDsource"</span>, </span>
<span id="cb23-44"><a href="#cb23-44" aria-hidden="true" tabindex="-1"></a>    <span class="at">Target =</span> <span class="st">"IDtarget"</span>,</span>
<span id="cb23-45"><a href="#cb23-45" aria-hidden="true" tabindex="-1"></a>    <span class="at">Value =</span> <span class="st">"value"</span>, </span>
<span id="cb23-46"><a href="#cb23-46" aria-hidden="true" tabindex="-1"></a>    <span class="at">NodeID =</span> <span class="st">"name"</span>,</span>
<span id="cb23-47"><a href="#cb23-47" aria-hidden="true" tabindex="-1"></a>    <span class="at">sinksRight =</span> <span class="cn">TRUE</span>,</span>
<span id="cb23-48"><a href="#cb23-48" aria-hidden="true" tabindex="-1"></a>    <span class="at">nodeWidth =</span> <span class="dv">25</span>,</span>
<span id="cb23-49"><a href="#cb23-49" aria-hidden="true" tabindex="-1"></a>    <span class="at">fontSize =</span> <span class="dv">14</span>,</span>
<span id="cb23-50"><a href="#cb23-50" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb23-51"><a href="#cb23-51" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb23-52"><a href="#cb23-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-53"><a href="#cb23-53" aria-hidden="true" tabindex="-1"></a>save_sankey_as_png <span class="ot">&lt;-</span> <span class="cf">function</span>(sankey_plot, <span class="at">width =</span> <span class="dv">1000</span>, <span class="at">height =</span> <span class="dv">800</span>) {</span>
<span id="cb23-54"><a href="#cb23-54" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Save the plot as an HTML file</span></span>
<span id="cb23-55"><a href="#cb23-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">saveWidget</span>(sankey_plot, <span class="fu">sprintf</span>(<span class="st">'%s/sankey.html'</span>,data_dir))</span>
<span id="cb23-56"><a href="#cb23-56" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb23-57"><a href="#cb23-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-58"><a href="#cb23-58" aria-hidden="true" tabindex="-1"></a>format_scientific <span class="ot">&lt;-</span> <span class="cf">function</span>(x, <span class="at">digits=</span><span class="dv">2</span>) {</span>
<span id="cb23-59"><a href="#cb23-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sapply</span>(x, <span class="cf">function</span>(val) {</span>
<span id="cb23-60"><a href="#cb23-60" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">is.na</span>(val) <span class="sc">||</span> <span class="fu">abs</span>(val) <span class="sc">&lt;</span> <span class="fl">1e-15</span>) {</span>
<span id="cb23-61"><a href="#cb23-61" aria-hidden="true" tabindex="-1"></a>      <span class="fu">return</span>(<span class="st">"0"</span>)</span>
<span id="cb23-62"><a href="#cb23-62" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb23-63"><a href="#cb23-63" aria-hidden="true" tabindex="-1"></a>      exponent <span class="ot">&lt;-</span> <span class="fu">floor</span>(<span class="fu">log10</span>(<span class="fu">abs</span>(val)))</span>
<span id="cb23-64"><a href="#cb23-64" aria-hidden="true" tabindex="-1"></a>      coef <span class="ot">&lt;-</span> val <span class="sc">/</span> <span class="dv">10</span><span class="sc">^</span>exponent</span>
<span id="cb23-65"><a href="#cb23-65" aria-hidden="true" tabindex="-1"></a>      <span class="co">#return(sprintf("%.1f × 10^%d", round(coef, digits), exponent))</span></span>
<span id="cb23-66"><a href="#cb23-66" aria-hidden="true" tabindex="-1"></a>      <span class="co"># May or may not be smart, just keeping magnitude</span></span>
<span id="cb23-67"><a href="#cb23-67" aria-hidden="true" tabindex="-1"></a>      <span class="fu">return</span>(<span class="fu">sprintf</span>(<span class="st">"10^%d"</span>, exponent))</span>
<span id="cb23-68"><a href="#cb23-68" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb23-69"><a href="#cb23-69" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb23-70"><a href="#cb23-70" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb23-71"><a href="#cb23-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-72"><a href="#cb23-72" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> result <span class="sc">%&gt;%</span> </span>
<span id="cb23-73"><a href="#cb23-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">c</span>(genus_n_reads_tot, genus_ra_reads_tot), <span class="sc">~</span><span class="fu">replace_na</span>(., <span class="dv">0</span>)),</span>
<span id="cb23-74"><a href="#cb23-74" aria-hidden="true" tabindex="-1"></a>         <span class="at">genus =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(genus), <span class="st">"Unknown Genus"</span>, genus)) <span class="sc">%&gt;%</span></span>
<span id="cb23-75"><a href="#cb23-75" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb23-76"><a href="#cb23-76" aria-hidden="true" tabindex="-1"></a>  <span class="at">species =</span> <span class="fu">paste0</span>(species, <span class="fu">sprintf</span>(<span class="st">' (%s)'</span>, <span class="fu">format_scientific</span>(species_ra_reads_tot))),</span>
<span id="cb23-77"><a href="#cb23-77" aria-hidden="true" tabindex="-1"></a>  <span class="at">genus =</span> <span class="fu">paste0</span>(genus, <span class="fu">sprintf</span>(<span class="st">' (%s)'</span>, <span class="fu">format_scientific</span>(genus_ra_reads_tot))),</span>
<span id="cb23-78"><a href="#cb23-78" aria-hidden="true" tabindex="-1"></a>  <span class="at">family =</span> <span class="fu">paste0</span>(family, <span class="fu">sprintf</span>(<span class="st">' (%s)'</span>, <span class="fu">format_scientific</span>(family_ra_reads_tot)))</span>
<span id="cb23-79"><a href="#cb23-79" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb23-80"><a href="#cb23-80" aria-hidden="true" tabindex="-1"></a>links <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">create_links</span>(data))</span>
<span id="cb23-81"><a href="#cb23-81" aria-hidden="true" tabindex="-1"></a>nodes <span class="ot">&lt;-</span> <span class="fu">create_nodes</span>(links)</span>
<span id="cb23-82"><a href="#cb23-82" aria-hidden="true" tabindex="-1"></a>sankey_data <span class="ot">&lt;-</span> <span class="fu">prepare_sankey_data</span>(links, nodes)</span>
<span id="cb23-83"><a href="#cb23-83" aria-hidden="true" tabindex="-1"></a>sankey <span class="ot">&lt;-</span> <span class="fu">create_sankey_plot</span>(sankey_data)</span>
<span id="cb23-84"><a href="#cb23-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-85"><a href="#cb23-85" aria-hidden="true" tabindex="-1"></a>sankey</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="htmlwidget-781e424b3f620b570581" style="width:100%;height:1462px;" class="sankeyNetwork html-widget"></div>
<script type="application/json" data-for="htmlwidget-781e424b3f620b570581">{"x":{"links":{"source":[0,0,1,2,3,4,5,6],"target":[3,6,4,5,7,8,9,10],"value":[1,1,1,1,1,1,1,1]},"nodes":{"name":["Anelloviridae (10^-7)","Orthoherpesviridae (10^-7)","Poxviridae (10^-8)","Alphatorquevirus (10^-8)","Lymphocryptovirus (10^-7)","Orthopoxvirus (10^-8)","Unknown Genus (0)","Alphatorquevirus homin24 (10^-8)","Lymphocryptovirus humangamma4 (10^-7)","Monkeypox virus (10^-8)","Anelloviridae sp. (10^-8)"],"group":["Anelloviridae (10^-7)","Orthoherpesviridae (10^-7)","Poxviridae (10^-8)","Alphatorquevirus (10^-8)","Lymphocryptovirus (10^-7)","Orthopoxvirus (10^-8)","Unknown Genus (0)","Alphatorquevirus homin24 (10^-8)","Lymphocryptovirus humangamma4 (10^-7)","Monkeypox virus (10^-8)","Anelloviridae sp. (10^-8)"]},"options":{"NodeID":"name","NodeGroup":"name","LinkGroup":null,"colourScale":"d3.scaleOrdinal(d3.schemeCategory20);","fontSize":14,"fontFamily":null,"nodeWidth":25,"nodePadding":10,"units":"","margin":{"top":null,"right":null,"bottom":null,"left":null},"iterations":32,"sinksRight":true}},"evals":[],"jsHooks":[]}</script>
</div>
</div>


<!-- -->

</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb24" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "Workflow of O'Connell et al. (2023)"</span></span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a><span class="an">subtitle:</span><span class="co"> "Whole blood from US (RNA)"</span></span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> "Harmon Bhasin"</span></span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> 2024-09-05</span></span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span></span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a><span class="co">  html:</span></span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a><span class="co">    toc: true # table of contents</span></span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a><span class="co">    toc-title: "Table of contents" # table of contents title</span></span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a><span class="co">    number-sections: true # number sections</span></span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a><span class="co">    number-depth: 3 # number depth to show in table of contents</span></span>
<span id="cb24-12"><a href="#cb24-12" aria-hidden="true" tabindex="-1"></a><span class="co">    toc-location: right # table of contents location</span></span>
<span id="cb24-13"><a href="#cb24-13" aria-hidden="true" tabindex="-1"></a><span class="co">    page-layout: full # full page layout</span></span>
<span id="cb24-14"><a href="#cb24-14" aria-hidden="true" tabindex="-1"></a><span class="co">    code-fold: true # Keep option to fold code (i.e. show or hide it; default: hide)</span></span>
<span id="cb24-15"><a href="#cb24-15" aria-hidden="true" tabindex="-1"></a><span class="co">    code-tools: true # Code menu in the header of your document that provides various tools for readers to interact with the source code</span></span>
<span id="cb24-16"><a href="#cb24-16" aria-hidden="true" tabindex="-1"></a><span class="co">    code-link: true # Enables hyper-linking of functions within code blocks to their online documentation</span></span>
<span id="cb24-17"><a href="#cb24-17" aria-hidden="true" tabindex="-1"></a><span class="co">    df-print: paged # print data frame</span></span>
<span id="cb24-18"><a href="#cb24-18" aria-hidden="true" tabindex="-1"></a><span class="co">    fig-format: svg</span></span>
<span id="cb24-19"><a href="#cb24-19" aria-hidden="true" tabindex="-1"></a><span class="co">    other-links:</span></span>
<span id="cb24-20"><a href="#cb24-20" aria-hidden="true" tabindex="-1"></a><span class="co">      - text: Paper</span></span>
<span id="cb24-21"><a href="#cb24-21" aria-hidden="true" tabindex="-1"></a><span class="co">        href: https://doi.org/10.1186/s12863-024-01223-z</span></span>
<span id="cb24-22"><a href="#cb24-22" aria-hidden="true" tabindex="-1"></a><span class="co">      - text: Data</span></span>
<span id="cb24-23"><a href="#cb24-23" aria-hidden="true" tabindex="-1"></a><span class="co">        href: https://www.ncbi.nlm.nih.gov/sra/?term=SRP429744</span></span>
<span id="cb24-24"><a href="#cb24-24" aria-hidden="true" tabindex="-1"></a><span class="co">    code-links:</span></span>
<span id="cb24-25"><a href="#cb24-25" aria-hidden="true" tabindex="-1"></a><span class="co">      - text: Code for this post</span></span>
<span id="cb24-26"><a href="#cb24-26" aria-hidden="true" tabindex="-1"></a><span class="co">        icon: file-code</span></span>
<span id="cb24-27"><a href="#cb24-27" aria-hidden="true" tabindex="-1"></a><span class="co">        href: https://github.com/naobservatory/harmons-public-notebook/blob/main/notebooks/2024-09-05-oconnell.qmd</span></span>
<span id="cb24-28"><a href="#cb24-28" aria-hidden="true" tabindex="-1"></a><span class="an">editor:</span><span class="co"> </span></span>
<span id="cb24-29"><a href="#cb24-29" aria-hidden="true" tabindex="-1"></a><span class="co">  visual: true</span></span>
<span id="cb24-30"><a href="#cb24-30" aria-hidden="true" tabindex="-1"></a><span class="co">  render-on-save: true</span></span>
<span id="cb24-31"><a href="#cb24-31" aria-hidden="true" tabindex="-1"></a><span class="an">comments:</span></span>
<span id="cb24-32"><a href="#cb24-32" aria-hidden="true" tabindex="-1"></a><span class="co">  hypothesis: true # hypothesis</span></span>
<span id="cb24-33"><a href="#cb24-33" aria-hidden="true" tabindex="-1"></a><span class="an">execute:</span><span class="co"> </span></span>
<span id="cb24-34"><a href="#cb24-34" aria-hidden="true" tabindex="-1"></a><span class="co">  freeze: auto</span></span>
<span id="cb24-35"><a href="#cb24-35" aria-hidden="true" tabindex="-1"></a><span class="co">  cache: true</span></span>
<span id="cb24-36"><a href="#cb24-36" aria-hidden="true" tabindex="-1"></a><span class="an">title-block-banner:</span><span class="co"> "#de2d26"</span></span>
<span id="cb24-37"><a href="#cb24-37" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb24-38"><a href="#cb24-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-41"><a href="#cb24-41" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb24-42"><a href="#cb24-42" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: load-packages</span></span>
<span id="cb24-43"><a href="#cb24-43" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb24-44"><a href="#cb24-44" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb24-45"><a href="#cb24-45" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pacman)</span>
<span id="cb24-46"><a href="#cb24-46" aria-hidden="true" tabindex="-1"></a>pacman<span class="sc">::</span><span class="fu">p_load</span>(tidyverse, cowplot, patchwork, fastqcr, RColorBrewer, networkD3, readxl, htmlwidgets, ggbeeswarm, latex2exp, languageserver, scales, ggpubr)</span>
<span id="cb24-47"><a href="#cb24-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-48"><a href="#cb24-48" aria-hidden="true" tabindex="-1"></a>head_dir <span class="ot">&lt;-</span> <span class="st">"/Users/harmonbhasin/work/securebio/"</span></span>
<span id="cb24-49"><a href="#cb24-49" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="fu">sprintf</span>(<span class="st">"%s/sampling-strategies/scripts/aux_plot-theme.R"</span>, head_dir))</span>
<span id="cb24-50"><a href="#cb24-50" aria-hidden="true" tabindex="-1"></a>image_dir <span class="ot">&lt;-</span> <span class="fu">sprintf</span>(<span class="st">'%s/lab_meeting/thijssen'</span>, head_dir)</span>
<span id="cb24-51"><a href="#cb24-51" aria-hidden="true" tabindex="-1"></a>theme_base <span class="ot">&lt;-</span> theme_base <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio =</span> <span class="cn">NULL</span>)</span>
<span id="cb24-52"><a href="#cb24-52" aria-hidden="true" tabindex="-1"></a>theme_kit <span class="ot">&lt;-</span> theme_base <span class="sc">+</span> <span class="fu">theme</span>(</span>
<span id="cb24-53"><a href="#cb24-53" aria-hidden="true" tabindex="-1"></a>  <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">angle =</span> <span class="dv">45</span>),</span>
<span id="cb24-54"><a href="#cb24-54" aria-hidden="true" tabindex="-1"></a>  <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb24-55"><a href="#cb24-55" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb24-56"><a href="#cb24-56" aria-hidden="true" tabindex="-1"></a>tnl <span class="ot">&lt;-</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb24-57"><a href="#cb24-57" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb24-58"><a href="#cb24-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-61"><a href="#cb24-61" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb24-62"><a href="#cb24-62" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: set-up-paths</span></span>
<span id="cb24-63"><a href="#cb24-63" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb24-64"><a href="#cb24-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-65"><a href="#cb24-65" aria-hidden="true" tabindex="-1"></a><span class="co"># Data input paths</span></span>
<span id="cb24-66"><a href="#cb24-66" aria-hidden="true" tabindex="-1"></a>data_dir <span class="ot">&lt;-</span> <span class="st">"/Users/harmonbhasin/work/securebio/nao-harmon/oconnell2023/analysis/"</span></span>
<span id="cb24-67"><a href="#cb24-67" aria-hidden="true" tabindex="-1"></a>input_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"data/input"</span>)</span>
<span id="cb24-68"><a href="#cb24-68" aria-hidden="true" tabindex="-1"></a>results_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"data/results"</span>)</span>
<span id="cb24-69"><a href="#cb24-69" aria-hidden="true" tabindex="-1"></a>qc_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(results_dir, <span class="st">"qc"</span>)</span>
<span id="cb24-70"><a href="#cb24-70" aria-hidden="true" tabindex="-1"></a>hv_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(results_dir, <span class="st">"hv"</span>)</span>
<span id="cb24-71"><a href="#cb24-71" aria-hidden="true" tabindex="-1"></a>libraries_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(input_dir, <span class="st">"libraries.csv"</span>)</span>
<span id="cb24-72"><a href="#cb24-72" aria-hidden="true" tabindex="-1"></a>basic_stats_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(qc_dir, <span class="st">"qc_basic_stats.tsv.gz"</span>)</span>
<span id="cb24-73"><a href="#cb24-73" aria-hidden="true" tabindex="-1"></a>adapter_stats_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(qc_dir, <span class="st">"qc_adapter_stats.tsv.gz"</span>)</span>
<span id="cb24-74"><a href="#cb24-74" aria-hidden="true" tabindex="-1"></a>quality_base_stats_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(qc_dir, <span class="st">"qc_quality_base_stats.tsv.gz"</span>)</span>
<span id="cb24-75"><a href="#cb24-75" aria-hidden="true" tabindex="-1"></a>quality_seq_stats_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(qc_dir, <span class="st">"qc_quality_sequence_stats.tsv.gz"</span>)</span>
<span id="cb24-76"><a href="#cb24-76" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb24-77"><a href="#cb24-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-78"><a href="#cb24-78" aria-hidden="true" tabindex="-1"></a>This is the first whole blood study of <span class="co">[</span><span class="ot">this series</span><span class="co">](https://data.securebio.org/harmons-public-notebook/notebooks/2024-07-08_cebria-mendoza.html)</span>. In this post, I analyze <span class="co">[</span><span class="ot">O'Connell 2023</span><span class="co">](https://doi.org/10.1186/s12863-024-01223-z)</span>, a dataset with 138 adult whole blood donors from Austin, TX.</span>
<span id="cb24-79"><a href="#cb24-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-80"><a href="#cb24-80" aria-hidden="true" tabindex="-1"></a>*This notebook uses the MGS Workflow v2.2.1 (note that this is old).*</span>
<span id="cb24-81"><a href="#cb24-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-82"><a href="#cb24-82" aria-hidden="true" tabindex="-1"></a><span class="fu"># The raw data</span></span>
<span id="cb24-83"><a href="#cb24-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-84"><a href="#cb24-84" aria-hidden="true" tabindex="-1"></a><span class="fu">## About</span></span>
<span id="cb24-85"><a href="#cb24-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-86"><a href="#cb24-86" aria-hidden="true" tabindex="-1"></a><span class="co">[</span><span class="ot">This dataset</span><span class="co">](https://www.ncbi.nlm.nih.gov/sra/?term=SRP429744)</span> is composed of 138 samples which come from 138 individuals in Austin, TX. They performed RNA sequencing on whole blood samples.</span>
<span id="cb24-87"><a href="#cb24-87" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-90"><a href="#cb24-90" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb24-91"><a href="#cb24-91" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb24-92"><a href="#cb24-92" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: load-libraries</span></span>
<span id="cb24-93"><a href="#cb24-93" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-94"><a href="#cb24-94" aria-hidden="true" tabindex="-1"></a><span class="co"># Import libraries and extract metadata from sample names</span></span>
<span id="cb24-95"><a href="#cb24-95" aria-hidden="true" tabindex="-1"></a>libraries <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(libraries_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb24-96"><a href="#cb24-96" aria-hidden="true" tabindex="-1"></a><span class="co">#meta_data &lt;- read_csv('/Users/harmonbhasin/work/securebio/nao-harmon/thijssen2023/preprocessing/SraRunTable.txt') %&gt;%</span></span>
<span id="cb24-97"><a href="#cb24-97" aria-hidden="true" tabindex="-1"></a><span class="co">#  select('Library Name', Run) %&gt;% rename(sample = Run, library = 'Library Name')</span></span>
<span id="cb24-98"><a href="#cb24-98" aria-hidden="true" tabindex="-1"></a><span class="co">#libraries &lt;- libraries %&gt;%</span></span>
<span id="cb24-99"><a href="#cb24-99" aria-hidden="true" tabindex="-1"></a><span class="co">#  left_join(meta_data, by = 'sample') %&gt;%</span></span>
<span id="cb24-100"><a href="#cb24-100" aria-hidden="true" tabindex="-1"></a><span class="co">#  select(sample, library=library.y) %&gt;% </span></span>
<span id="cb24-101"><a href="#cb24-101" aria-hidden="true" tabindex="-1"></a><span class="co">#  filter(library != 'PCL21') %&gt;%</span></span>
<span id="cb24-102"><a href="#cb24-102" aria-hidden="true" tabindex="-1"></a><span class="co">#  mutate(library = factor(library, levels = c(paste0('PCL', 1:20))))</span></span>
<span id="cb24-103"><a href="#cb24-103" aria-hidden="true" tabindex="-1"></a><span class="co">#libraries</span></span>
<span id="cb24-104"><a href="#cb24-104" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-105"><a href="#cb24-105" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb24-106"><a href="#cb24-106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-107"><a href="#cb24-107" aria-hidden="true" tabindex="-1"></a><span class="fu">### Sample + library preparation </span></span>
<span id="cb24-108"><a href="#cb24-108" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-109"><a href="#cb24-109" aria-hidden="true" tabindex="-1"></a>Paraphrased from the paper:</span>
<span id="cb24-110"><a href="#cb24-110" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-111"><a href="#cb24-111" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; Whole blood specimens were collected from 138 adult donors via PAXgene vacutainers at admission to the Emergency Department at Dell-Seton Medical Center (Austin, TX). Total RNA was isolated from archived PAXgene-stabilized whole blood using the PAXgene IVD Blood RNA Kit. Ribosomal RNA and globin mRNA-depleted cDNA libraries were prepared from 500 ng of total RNA using the Illumina TruSeq Stranded Total RNA Ribo-Zero Globin kit. Paired-end 150 bp sequencing was performed on the Illumina NovaSeq 6000 platform.</span></span>
<span id="cb24-112"><a href="#cb24-112" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-113"><a href="#cb24-113" aria-hidden="true" tabindex="-1"></a>*More details can be found [here](https://www.sciencedirect.com/science/article/pii/S0888754323001520?via%3Dihub#s0035).*</span>
<span id="cb24-114"><a href="#cb24-114" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-115"><a href="#cb24-115" aria-hidden="true" tabindex="-1"></a><span class="fu">## Quality control metrics</span></span>
<span id="cb24-116"><a href="#cb24-116" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-119"><a href="#cb24-119" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb24-120"><a href="#cb24-120" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: read-qc-data</span></span>
<span id="cb24-121"><a href="#cb24-121" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb24-122"><a href="#cb24-122" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-123"><a href="#cb24-123" aria-hidden="true" tabindex="-1"></a><span class="co"># Import QC data</span></span>
<span id="cb24-124"><a href="#cb24-124" aria-hidden="true" tabindex="-1"></a>stages <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"raw_concat"</span>, <span class="st">"cleaned"</span>, <span class="st">"dedup"</span>, <span class="st">"ribo_initial"</span>, <span class="st">"ribo_secondary"</span>)</span>
<span id="cb24-125"><a href="#cb24-125" aria-hidden="true" tabindex="-1"></a>basic_stats <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(basic_stats_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-126"><a href="#cb24-126" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb24-127"><a href="#cb24-127" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">factor</span>(stage, <span class="at">levels =</span> stages),</span>
<span id="cb24-128"><a href="#cb24-128" aria-hidden="true" tabindex="-1"></a>         <span class="at">sample =</span> <span class="fu">fct_inorder</span>(sample))</span>
<span id="cb24-129"><a href="#cb24-129" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-130"><a href="#cb24-130" aria-hidden="true" tabindex="-1"></a>adapter_stats <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(adapter_stats_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-131"><a href="#cb24-131" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb24-132"><a href="#cb24-132" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">factor</span>(stage, <span class="at">levels =</span> stages),</span>
<span id="cb24-133"><a href="#cb24-133" aria-hidden="true" tabindex="-1"></a>         <span class="at">read_pair =</span> <span class="fu">fct_inorder</span>(<span class="fu">as.character</span>(read_pair)))</span>
<span id="cb24-134"><a href="#cb24-134" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-135"><a href="#cb24-135" aria-hidden="true" tabindex="-1"></a>quality_base_stats <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(quality_base_stats_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-136"><a href="#cb24-136" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb24-137"><a href="#cb24-137" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">factor</span>(stage, <span class="at">levels =</span> stages),</span>
<span id="cb24-138"><a href="#cb24-138" aria-hidden="true" tabindex="-1"></a>         <span class="at">read_pair =</span> <span class="fu">fct_inorder</span>(<span class="fu">as.character</span>(read_pair)))</span>
<span id="cb24-139"><a href="#cb24-139" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb24-140"><a href="#cb24-140" aria-hidden="true" tabindex="-1"></a>quality_seq_stats <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(quality_seq_stats_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-141"><a href="#cb24-141" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb24-142"><a href="#cb24-142" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">factor</span>(stage, <span class="at">levels =</span> stages),</span>
<span id="cb24-143"><a href="#cb24-143" aria-hidden="true" tabindex="-1"></a>         <span class="at">read_pair =</span> <span class="fu">fct_inorder</span>(<span class="fu">as.character</span>(read_pair)))</span>
<span id="cb24-144"><a href="#cb24-144" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb24-145"><a href="#cb24-145" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb24-146"><a href="#cb24-146" aria-hidden="true" tabindex="-1"></a> <span class="co"># Filter to raw data</span></span>
<span id="cb24-147"><a href="#cb24-147" aria-hidden="true" tabindex="-1"></a> basic_stats_raw <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"raw_concat"</span>)</span>
<span id="cb24-148"><a href="#cb24-148" aria-hidden="true" tabindex="-1"></a> adapter_stats_raw <span class="ot">&lt;-</span> adapter_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"raw_concat"</span>)</span>
<span id="cb24-149"><a href="#cb24-149" aria-hidden="true" tabindex="-1"></a> quality_base_stats_raw <span class="ot">&lt;-</span> quality_base_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"raw_concat"</span>)</span>
<span id="cb24-150"><a href="#cb24-150" aria-hidden="true" tabindex="-1"></a> quality_seq_stats_raw <span class="ot">&lt;-</span> quality_seq_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"raw_concat"</span>)</span>
<span id="cb24-151"><a href="#cb24-151" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb24-152"><a href="#cb24-152" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb24-153"><a href="#cb24-153" aria-hidden="true" tabindex="-1"></a> <span class="co"># Get key values for readout</span></span>
<span id="cb24-154"><a href="#cb24-154" aria-hidden="true" tabindex="-1"></a>raw_read_counts <span class="ot">&lt;-</span> basic_stats_raw <span class="sc">%&gt;%</span> </span>
<span id="cb24-155"><a href="#cb24-155" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">rmin =</span> <span class="fu">min</span>(n_read_pairs), </span>
<span id="cb24-156"><a href="#cb24-156" aria-hidden="true" tabindex="-1"></a>            <span class="at">rmax=</span><span class="fu">max</span>(n_read_pairs),</span>
<span id="cb24-157"><a href="#cb24-157" aria-hidden="true" tabindex="-1"></a>            <span class="at">rmean=</span><span class="fu">mean</span>(n_read_pairs), </span>
<span id="cb24-158"><a href="#cb24-158" aria-hidden="true" tabindex="-1"></a>            <span class="at">rtot =</span> <span class="fu">sum</span>(n_read_pairs),</span>
<span id="cb24-159"><a href="#cb24-159" aria-hidden="true" tabindex="-1"></a>            <span class="at">btot =</span> <span class="fu">sum</span>(n_bases_approx),</span>
<span id="cb24-160"><a href="#cb24-160" aria-hidden="true" tabindex="-1"></a>            <span class="at">dmin =</span> <span class="fu">min</span>(percent_duplicates), </span>
<span id="cb24-161"><a href="#cb24-161" aria-hidden="true" tabindex="-1"></a>            <span class="at">dmax=</span><span class="fu">max</span>(percent_duplicates),</span>
<span id="cb24-162"><a href="#cb24-162" aria-hidden="true" tabindex="-1"></a>            <span class="at">dmean=</span><span class="fu">mean</span>(percent_duplicates), </span>
<span id="cb24-163"><a href="#cb24-163" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb24-164"><a href="#cb24-164" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb24-165"><a href="#cb24-165" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-166"><a href="#cb24-166" aria-hidden="true" tabindex="-1"></a>In total, these 138 samples contained ~3B read pairs. The samples had ~20.1M - 46.9M (mean ~22.5M) read pairs each. The number of reads looks pretty good, with a few samples having a much higher count. The total number of base pairs also looks reasonable, matching the trends of the number of reads. The duplication rate is high. Adapter content is quite low, although we can see library contamination (illumina_universal_adapter), and polya/polyg contamination.  As we'd expect, we see higher quality Q scores near the beginning of the read and a gradual decline towards the end of the read, however all positions seem to have a Q score of around 35 which means that our reads are \~99.97% accurate. When looking at the Q score over all sequences, we can see a sharp peak around 35, which corresponds to the previous plot, indicating high read quality. </span>
<span id="cb24-167"><a href="#cb24-167" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-170"><a href="#cb24-170" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb24-171"><a href="#cb24-171" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 9</span></span>
<span id="cb24-172"><a href="#cb24-172" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb24-173"><a href="#cb24-173" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-basic-stats</span></span>
<span id="cb24-174"><a href="#cb24-174" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-175"><a href="#cb24-175" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare data</span></span>
<span id="cb24-176"><a href="#cb24-176" aria-hidden="true" tabindex="-1"></a>basic_stats_raw_metrics <span class="ot">&lt;-</span> basic_stats_raw <span class="sc">%&gt;%</span></span>
<span id="cb24-177"><a href="#cb24-177" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(library,</span>
<span id="cb24-178"><a href="#cb24-178" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at"># Read pairs</span><span class="st">`</span> <span class="ot">=</span> n_read_pairs,</span>
<span id="cb24-179"><a href="#cb24-179" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Total base pairs</span><span class="sc">\n</span><span class="at">(approx)</span><span class="st">`</span> <span class="ot">=</span> n_bases_approx,</span>
<span id="cb24-180"><a href="#cb24-180" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">% Duplicates</span><span class="sc">\n</span><span class="at">(FASTQC)</span><span class="st">`</span> <span class="ot">=</span> percent_duplicates) <span class="sc">%&gt;%</span></span>
<span id="cb24-181"><a href="#cb24-181" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>library, <span class="at">names_to =</span> <span class="st">"metric"</span>, <span class="at">values_to =</span> <span class="st">"value"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-182"><a href="#cb24-182" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">metric =</span> <span class="fu">fct_inorder</span>(metric)) </span>
<span id="cb24-183"><a href="#cb24-183" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-184"><a href="#cb24-184" aria-hidden="true" tabindex="-1"></a><span class="co"># Set up plot templates</span></span>
<span id="cb24-185"><a href="#cb24-185" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-186"><a href="#cb24-186" aria-hidden="true" tabindex="-1"></a>g_basic <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(basic_stats_raw_metrics, <span class="fu">aes</span>(<span class="at">x =</span> library, <span class="at">y =</span> value)) <span class="sc">+</span></span>
<span id="cb24-187"><a href="#cb24-187" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position =</span> <span class="st">"dodge"</span>) <span class="sc">+</span></span>
<span id="cb24-188"><a href="#cb24-188" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>() <span class="sc">+</span></span>
<span id="cb24-189"><a href="#cb24-189" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb24-190"><a href="#cb24-190" aria-hidden="true" tabindex="-1"></a>  <span class="fu">expand_limits</span>(<span class="at">y =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">100</span>)) <span class="sc">+</span></span>
<span id="cb24-191"><a href="#cb24-191" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(metric <span class="sc">~</span> ., <span class="at">scales =</span> <span class="st">"free"</span>, <span class="at">space =</span> <span class="st">"free_x"</span>, <span class="at">switch =</span> <span class="st">"y"</span>) <span class="sc">+</span></span>
<span id="cb24-192"><a href="#cb24-192" aria-hidden="true" tabindex="-1"></a>  theme_kit <span class="sc">+</span> <span class="fu">theme</span>(</span>
<span id="cb24-193"><a href="#cb24-193" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb24-194"><a href="#cb24-194" aria-hidden="true" tabindex="-1"></a>    <span class="at">strip.text.y =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"plain"</span>),</span>
<span id="cb24-195"><a href="#cb24-195" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>()</span>
<span id="cb24-196"><a href="#cb24-196" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb24-197"><a href="#cb24-197" aria-hidden="true" tabindex="-1"></a>g_basic</span>
<span id="cb24-198"><a href="#cb24-198" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb24-199"><a href="#cb24-199" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-202"><a href="#cb24-202" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb24-203"><a href="#cb24-203" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-raw-quality</span></span>
<span id="cb24-204"><a href="#cb24-204" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 8</span></span>
<span id="cb24-205"><a href="#cb24-205" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-206"><a href="#cb24-206" aria-hidden="true" tabindex="-1"></a><span class="co"># Set up plotting templates</span></span>
<span id="cb24-207"><a href="#cb24-207" aria-hidden="true" tabindex="-1"></a>g_qual_raw <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">linetype=</span>read_pair, <span class="at">group=</span><span class="fu">interaction</span>(sample,read_pair))) <span class="sc">+</span> </span>
<span id="cb24-208"><a href="#cb24-208" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_linetype_discrete</span>(<span class="at">name =</span> <span class="st">"Read Pair"</span>) <span class="sc">+</span></span>
<span id="cb24-209"><a href="#cb24-209" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color=</span><span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>),</span>
<span id="cb24-210"><a href="#cb24-210" aria-hidden="true" tabindex="-1"></a>         <span class="at">linetype =</span> <span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>)) <span class="sc">+</span></span>
<span id="cb24-211"><a href="#cb24-211" aria-hidden="true" tabindex="-1"></a>  theme_base</span>
<span id="cb24-212"><a href="#cb24-212" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-213"><a href="#cb24-213" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize adapters</span></span>
<span id="cb24-214"><a href="#cb24-214" aria-hidden="true" tabindex="-1"></a>g_adapters_raw <span class="ot">&lt;-</span> g_qual_raw <span class="sc">+</span> </span>
<span id="cb24-215"><a href="#cb24-215" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>pc_adapters), <span class="at">data=</span>adapter_stats_raw) <span class="sc">+</span></span>
<span id="cb24-216"><a href="#cb24-216" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Adapters"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb24-217"><a href="#cb24-217" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">10</span>,<span class="dv">1</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb24-218"><a href="#cb24-218" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Position"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb24-219"><a href="#cb24-219" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">500</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb24-220"><a href="#cb24-220" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(.<span class="sc">~</span>adapter)</span>
<span id="cb24-221"><a href="#cb24-221" aria-hidden="true" tabindex="-1"></a>g_adapters_raw</span>
<span id="cb24-222"><a href="#cb24-222" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-223"><a href="#cb24-223" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize quality</span></span>
<span id="cb24-224"><a href="#cb24-224" aria-hidden="true" tabindex="-1"></a>g_quality_base_raw <span class="ot">&lt;-</span> g_qual_raw <span class="sc">+</span></span>
<span id="cb24-225"><a href="#cb24-225" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb24-226"><a href="#cb24-226" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb24-227"><a href="#cb24-227" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>mean_phred_score), <span class="at">data=</span>quality_base_stats_raw) <span class="sc">+</span></span>
<span id="cb24-228"><a href="#cb24-228" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"Mean Phred score"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">10</span>,<span class="dv">45</span>)) <span class="sc">+</span></span>
<span id="cb24-229"><a href="#cb24-229" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Position"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb24-230"><a href="#cb24-230" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">500</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb24-231"><a href="#cb24-231" aria-hidden="true" tabindex="-1"></a>g_quality_base_raw</span>
<span id="cb24-232"><a href="#cb24-232" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-233"><a href="#cb24-233" aria-hidden="true" tabindex="-1"></a>g_quality_seq_raw <span class="ot">&lt;-</span> g_qual_raw <span class="sc">+</span></span>
<span id="cb24-234"><a href="#cb24-234" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb24-235"><a href="#cb24-235" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb24-236"><a href="#cb24-236" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>mean_phred_score, <span class="at">y=</span>n_sequences), <span class="at">data=</span>quality_seq_stats_raw) <span class="sc">+</span></span>
<span id="cb24-237"><a href="#cb24-237" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Mean Phred score"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb24-238"><a href="#cb24-238" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"# Sequences"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb24-239"><a href="#cb24-239" aria-hidden="true" tabindex="-1"></a>g_quality_seq_raw</span>
<span id="cb24-240"><a href="#cb24-240" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb24-241"><a href="#cb24-241" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-242"><a href="#cb24-242" aria-hidden="true" tabindex="-1"></a><span class="fu"># Preprocessing</span></span>
<span id="cb24-243"><a href="#cb24-243" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-244"><a href="#cb24-244" aria-hidden="true" tabindex="-1"></a><span class="fu">## Summary</span></span>
<span id="cb24-245"><a href="#cb24-245" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-246"><a href="#cb24-246" aria-hidden="true" tabindex="-1"></a>The average fraction of reads at each stage in the preprocessing pipeline is shown in the following table. Adapter trimming &amp; filtering doesn't lose too many reads. Deduplication loses us about 27% of reads on average, then ribodepletion only loses as about 1% on average.</span>
<span id="cb24-247"><a href="#cb24-247" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-250"><a href="#cb24-250" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb24-251"><a href="#cb24-251" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: preproc-table</span></span>
<span id="cb24-252"><a href="#cb24-252" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-253"><a href="#cb24-253" aria-hidden="true" tabindex="-1"></a><span class="co"># </span><span class="al">TODO</span><span class="co">: Group by pool size as well</span></span>
<span id="cb24-254"><a href="#cb24-254" aria-hidden="true" tabindex="-1"></a><span class="co"># Count read losses</span></span>
<span id="cb24-255"><a href="#cb24-255" aria-hidden="true" tabindex="-1"></a>n_reads_rel <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> </span>
<span id="cb24-256"><a href="#cb24-256" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, stage, percent_duplicates, n_read_pairs) <span class="sc">%&gt;%</span></span>
<span id="cb24-257"><a href="#cb24-257" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb24-258"><a href="#cb24-258" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(sample, stage) <span class="sc">%&gt;%</span></span>
<span id="cb24-259"><a href="#cb24-259" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_retained =</span> n_read_pairs <span class="sc">/</span> <span class="fu">lag</span>(n_read_pairs),</span>
<span id="cb24-260"><a href="#cb24-260" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_lost =</span> <span class="dv">1</span> <span class="sc">-</span> p_reads_retained,</span>
<span id="cb24-261"><a href="#cb24-261" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_retained_abs =</span> n_read_pairs <span class="sc">/</span> n_read_pairs[<span class="dv">1</span>],</span>
<span id="cb24-262"><a href="#cb24-262" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_lost_abs =</span> <span class="dv">1</span><span class="sc">-</span>p_reads_retained_abs,</span>
<span id="cb24-263"><a href="#cb24-263" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_lost_abs_marginal =</span> p_reads_lost_abs <span class="sc">-</span> <span class="fu">lag</span>(p_reads_lost_abs))</span>
<span id="cb24-264"><a href="#cb24-264" aria-hidden="true" tabindex="-1"></a>n_reads_rel_display <span class="ot">&lt;-</span> n_reads_rel <span class="sc">%&gt;%</span> </span>
<span id="cb24-265"><a href="#cb24-265" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">Stage=</span>stage) <span class="sc">%&gt;%</span> </span>
<span id="cb24-266"><a href="#cb24-266" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Stage) <span class="sc">%&gt;%</span> </span>
<span id="cb24-267"><a href="#cb24-267" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="st">`</span><span class="at">% Total Reads Lost (Cumulative)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">round</span>(<span class="fu">min</span>(p_reads_lost_abs<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">"-"</span>, <span class="fu">round</span>(<span class="fu">max</span>(p_reads_lost_abs<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">" (mean "</span>, <span class="fu">round</span>(<span class="fu">mean</span>(p_reads_lost_abs<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">")"</span>),</span>
<span id="cb24-268"><a href="#cb24-268" aria-hidden="true" tabindex="-1"></a>            <span class="st">`</span><span class="at">% Total Reads Lost (Marginal)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">round</span>(<span class="fu">min</span>(p_reads_lost_abs_marginal<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">"-"</span>, <span class="fu">round</span>(<span class="fu">max</span>(p_reads_lost_abs_marginal<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">" (mean "</span>, <span class="fu">round</span>(<span class="fu">mean</span>(p_reads_lost_abs_marginal<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">")"</span>), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb24-269"><a href="#cb24-269" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Stage <span class="sc">!=</span> <span class="st">"raw_concat"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-270"><a href="#cb24-270" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Stage =</span> Stage <span class="sc">%&gt;%</span> as.numeric <span class="sc">%&gt;%</span> <span class="fu">factor</span>(<span class="at">labels=</span><span class="fu">c</span>(<span class="st">"Trimming &amp; filtering"</span>, <span class="st">"Deduplication"</span>, <span class="st">"Initial ribodepletion"</span>, <span class="st">"Secondary ribodepletion"</span>)))</span>
<span id="cb24-271"><a href="#cb24-271" aria-hidden="true" tabindex="-1"></a>n_reads_rel_display</span>
<span id="cb24-272"><a href="#cb24-272" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb24-273"><a href="#cb24-273" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-274"><a href="#cb24-274" aria-hidden="true" tabindex="-1"></a><span class="fu">## Quality control metrics</span></span>
<span id="cb24-275"><a href="#cb24-275" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-276"><a href="#cb24-276" aria-hidden="true" tabindex="-1"></a>Trimming + cleaning gets rid of the library contamination, however there is still some polya + polyg contamination, however given that it makes a small portion of adapter content, I think we're fine ignoring it. Read quality slightly improves after preprocessing + cleaning, this is noticable near the end positions. When looking at the q score over all reads, we see that cleaning gives us most of our improvement.</span>
<span id="cb24-277"><a href="#cb24-277" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-280"><a href="#cb24-280" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb24-281"><a href="#cb24-281" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb24-282"><a href="#cb24-282" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-quality</span></span>
<span id="cb24-283"><a href="#cb24-283" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 8</span></span>
<span id="cb24-284"><a href="#cb24-284" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb24-285"><a href="#cb24-285" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-286"><a href="#cb24-286" aria-hidden="true" tabindex="-1"></a>g_qual <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">linetype=</span>read_pair, <span class="at">group=</span><span class="fu">interaction</span>(sample,read_pair))) <span class="sc">+</span> </span>
<span id="cb24-287"><a href="#cb24-287" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_linetype_discrete</span>(<span class="at">name =</span> <span class="st">"Read Pair"</span>) <span class="sc">+</span></span>
<span id="cb24-288"><a href="#cb24-288" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color=</span><span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>),</span>
<span id="cb24-289"><a href="#cb24-289" aria-hidden="true" tabindex="-1"></a>         <span class="at">linetype =</span> <span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>)) <span class="sc">+</span></span>
<span id="cb24-290"><a href="#cb24-290" aria-hidden="true" tabindex="-1"></a>  theme_base</span>
<span id="cb24-291"><a href="#cb24-291" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-292"><a href="#cb24-292" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize adapters</span></span>
<span id="cb24-293"><a href="#cb24-293" aria-hidden="true" tabindex="-1"></a>g_adapters <span class="ot">&lt;-</span> g_qual <span class="sc">+</span> </span>
<span id="cb24-294"><a href="#cb24-294" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>pc_adapters), <span class="at">data=</span>adapter_stats) <span class="sc">+</span></span>
<span id="cb24-295"><a href="#cb24-295" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Adapters"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">20</span>),</span>
<span id="cb24-296"><a href="#cb24-296" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">50</span>,<span class="dv">10</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb24-297"><a href="#cb24-297" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Position"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb24-298"><a href="#cb24-298" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">140</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb24-299"><a href="#cb24-299" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(stage<span class="sc">~</span>adapter)</span>
<span id="cb24-300"><a href="#cb24-300" aria-hidden="true" tabindex="-1"></a>g_adapters</span>
<span id="cb24-301"><a href="#cb24-301" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-302"><a href="#cb24-302" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize quality</span></span>
<span id="cb24-303"><a href="#cb24-303" aria-hidden="true" tabindex="-1"></a>g_quality_base <span class="ot">&lt;-</span> g_qual <span class="sc">+</span></span>
<span id="cb24-304"><a href="#cb24-304" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb24-305"><a href="#cb24-305" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb24-306"><a href="#cb24-306" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>mean_phred_score), <span class="at">data=</span>quality_base_stats) <span class="sc">+</span></span>
<span id="cb24-307"><a href="#cb24-307" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"Mean Phred score"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">10</span>,<span class="dv">45</span>)) <span class="sc">+</span></span>
<span id="cb24-308"><a href="#cb24-308" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Position"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb24-309"><a href="#cb24-309" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">140</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb24-310"><a href="#cb24-310" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(stage<span class="sc">~</span>.)</span>
<span id="cb24-311"><a href="#cb24-311" aria-hidden="true" tabindex="-1"></a>g_quality_base</span>
<span id="cb24-312"><a href="#cb24-312" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-313"><a href="#cb24-313" aria-hidden="true" tabindex="-1"></a>g_quality_seq <span class="ot">&lt;-</span> g_qual <span class="sc">+</span></span>
<span id="cb24-314"><a href="#cb24-314" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb24-315"><a href="#cb24-315" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb24-316"><a href="#cb24-316" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>mean_phred_score, <span class="at">y=</span>n_sequences), <span class="at">data=</span>quality_seq_stats) <span class="sc">+</span></span>
<span id="cb24-317"><a href="#cb24-317" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Mean Phred score"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb24-318"><a href="#cb24-318" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"# Sequences"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb24-319"><a href="#cb24-319" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(stage<span class="sc">~</span>., <span class="at">scales =</span> <span class="st">"free_y"</span>)</span>
<span id="cb24-320"><a href="#cb24-320" aria-hidden="true" tabindex="-1"></a>g_quality_seq</span>
<span id="cb24-321"><a href="#cb24-321" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb24-322"><a href="#cb24-322" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-323"><a href="#cb24-323" aria-hidden="true" tabindex="-1"></a>These plots below show the trends from above in each sample. Everything looks pretty good.</span>
<span id="cb24-324"><a href="#cb24-324" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-327"><a href="#cb24-327" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb24-328"><a href="#cb24-328" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: preproc-figures</span></span>
<span id="cb24-329"><a href="#cb24-329" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb24-330"><a href="#cb24-330" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 3</span></span>
<span id="cb24-331"><a href="#cb24-331" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 6</span></span>
<span id="cb24-332"><a href="#cb24-332" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-333"><a href="#cb24-333" aria-hidden="true" tabindex="-1"></a>g_stage_trace <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(basic_stats, <span class="fu">aes</span>(<span class="at">x=</span>stage, <span class="at">group=</span>sample)) <span class="sc">+</span></span>
<span id="cb24-334"><a href="#cb24-334" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb24-335"><a href="#cb24-335" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-336"><a href="#cb24-336" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot reads over preprocessing</span></span>
<span id="cb24-337"><a href="#cb24-337" aria-hidden="true" tabindex="-1"></a>g_reads_stages <span class="ot">&lt;-</span> g_stage_trace <span class="sc">+</span></span>
<span id="cb24-338"><a href="#cb24-338" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>n_read_pairs)) <span class="sc">+</span></span>
<span id="cb24-339"><a href="#cb24-339" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"# Read pairs"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>))</span>
<span id="cb24-340"><a href="#cb24-340" aria-hidden="true" tabindex="-1"></a>g_reads_stages</span>
<span id="cb24-341"><a href="#cb24-341" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-342"><a href="#cb24-342" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot relative read losses during preprocessing</span></span>
<span id="cb24-343"><a href="#cb24-343" aria-hidden="true" tabindex="-1"></a>g_reads_rel <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(n_reads_rel, </span>
<span id="cb24-344"><a href="#cb24-344" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">aes</span>(<span class="at">x=</span>stage, <span class="at">group=</span>sample)) <span class="sc">+</span></span>
<span id="cb24-345"><a href="#cb24-345" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>p_reads_lost_abs_marginal)) <span class="sc">+</span></span>
<span id="cb24-346"><a href="#cb24-346" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"% Total Reads Lost"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), </span>
<span id="cb24-347"><a href="#cb24-347" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="cf">function</span>(x) x<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb24-348"><a href="#cb24-348" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb24-349"><a href="#cb24-349" aria-hidden="true" tabindex="-1"></a>g_reads_rel</span>
<span id="cb24-350"><a href="#cb24-350" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb24-351"><a href="#cb24-351" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-352"><a href="#cb24-352" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-355"><a href="#cb24-355" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb24-356"><a href="#cb24-356" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: preproc-dedup</span></span>
<span id="cb24-357"><a href="#cb24-357" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb24-358"><a href="#cb24-358" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-359"><a href="#cb24-359" aria-hidden="true" tabindex="-1"></a>stage_dup <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(stage) <span class="sc">%&gt;%</span> </span>
<span id="cb24-360"><a href="#cb24-360" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">dmin =</span> <span class="fu">min</span>(percent_duplicates), <span class="at">dmax=</span><span class="fu">max</span>(percent_duplicates),</span>
<span id="cb24-361"><a href="#cb24-361" aria-hidden="true" tabindex="-1"></a>            <span class="at">dmean=</span><span class="fu">mean</span>(percent_duplicates), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb24-362"><a href="#cb24-362" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-363"><a href="#cb24-363" aria-hidden="true" tabindex="-1"></a>g_dup_stages <span class="ot">&lt;-</span> g_stage_trace <span class="sc">+</span></span>
<span id="cb24-364"><a href="#cb24-364" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>percent_duplicates)) <span class="sc">+</span></span>
<span id="cb24-365"><a href="#cb24-365" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"% Duplicates"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb24-366"><a href="#cb24-366" aria-hidden="true" tabindex="-1"></a>g_dup_stages</span>
<span id="cb24-367"><a href="#cb24-367" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-368"><a href="#cb24-368" aria-hidden="true" tabindex="-1"></a>g_readlen_stages <span class="ot">&lt;-</span> g_stage_trace <span class="sc">+</span> <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>mean_seq_len)) <span class="sc">+</span></span>
<span id="cb24-369"><a href="#cb24-369" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"Mean read length (nt)"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>))</span>
<span id="cb24-370"><a href="#cb24-370" aria-hidden="true" tabindex="-1"></a>g_readlen_stages</span>
<span id="cb24-371"><a href="#cb24-371" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb24-372"><a href="#cb24-372" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-373"><a href="#cb24-373" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-374"><a href="#cb24-374" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-377"><a href="#cb24-377" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb24-378"><a href="#cb24-378" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: ribo-frac</span></span>
<span id="cb24-379"><a href="#cb24-379" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb24-380"><a href="#cb24-380" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate reads lost during ribodepletion (approximation for % ribosomal reads)</span></span>
<span id="cb24-381"><a href="#cb24-381" aria-hidden="true" tabindex="-1"></a>reads_ribo <span class="ot">&lt;-</span> n_reads_rel <span class="sc">%&gt;%</span> </span>
<span id="cb24-382"><a href="#cb24-382" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(stage <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"dedup"</span>, <span class="st">"ribo_secondary"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb24-383"><a href="#cb24-383" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb24-384"><a href="#cb24-384" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads_ribo=</span><span class="dv">1</span><span class="sc">-</span>n_read_pairs[<span class="dv">2</span>]<span class="sc">/</span>n_read_pairs[<span class="dv">1</span>], <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-385"><a href="#cb24-385" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by =</span> <span class="st">'sample'</span>)</span>
<span id="cb24-386"><a href="#cb24-386" aria-hidden="true" tabindex="-1"></a>reads_ribo_summ <span class="ot">&lt;-</span> reads_ribo <span class="sc">%&gt;%</span></span>
<span id="cb24-387"><a href="#cb24-387" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb24-388"><a href="#cb24-388" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">min=</span><span class="fu">min</span>(p_reads_ribo), <span class="at">max=</span><span class="fu">max</span>(p_reads_ribo),</span>
<span id="cb24-389"><a href="#cb24-389" aria-hidden="true" tabindex="-1"></a>            <span class="at">mean=</span><span class="fu">mean</span>(p_reads_ribo), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-390"><a href="#cb24-390" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by =</span> <span class="st">'sample'</span>)</span>
<span id="cb24-391"><a href="#cb24-391" aria-hidden="true" tabindex="-1"></a>g_reads_ribo <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(reads_ribo, </span>
<span id="cb24-392"><a href="#cb24-392" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">aes</span>(<span class="at">x=</span>library, <span class="at">y=</span>p_reads_ribo)) <span class="sc">+</span></span>
<span id="cb24-393"><a href="#cb24-393" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span> </span>
<span id="cb24-394"><a href="#cb24-394" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"Approx % ribosomal reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>),</span>
<span id="cb24-395"><a href="#cb24-395" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>)<span class="sc">+</span></span>
<span id="cb24-396"><a href="#cb24-396" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb24-397"><a href="#cb24-397" aria-hidden="true" tabindex="-1"></a>g_reads_ribo</span>
<span id="cb24-398"><a href="#cb24-398" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb24-399"><a href="#cb24-399" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-400"><a href="#cb24-400" aria-hidden="true" tabindex="-1"></a><span class="fu"># Taxonomic composition</span></span>
<span id="cb24-401"><a href="#cb24-401" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-402"><a href="#cb24-402" aria-hidden="true" tabindex="-1"></a><span class="fu">## High-level composition</span></span>
<span id="cb24-403"><a href="#cb24-403" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-404"><a href="#cb24-404" aria-hidden="true" tabindex="-1"></a>To assess the high-level composition of the reads, I ran the ribodepleted files through Kraken2 and summarized the results with Bracken.</span>
<span id="cb24-405"><a href="#cb24-405" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-406"><a href="#cb24-406" aria-hidden="true" tabindex="-1"></a>The groups listed below were created by Will:</span>
<span id="cb24-407"><a href="#cb24-407" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-408"><a href="#cb24-408" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Filtered (removed during cleaning)</span>
<span id="cb24-409"><a href="#cb24-409" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Duplicate (removed during deduplication)</span>
<span id="cb24-410"><a href="#cb24-410" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Ribosomal (removed during ribodepletion)</span>
<span id="cb24-411"><a href="#cb24-411" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Unassigned (non-ribosomal reads that were not assigned to any taxon by Kraken/Bracken)</span>
<span id="cb24-412"><a href="#cb24-412" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Bacterial (non-ribosomal reads assigned to the Bacteria domain by Kraken/Bracken)</span>
<span id="cb24-413"><a href="#cb24-413" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Archaeal (non-ribosomal reads assigned to the Archaea domain by Kraken/Bracken)</span>
<span id="cb24-414"><a href="#cb24-414" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Viral (non-ribosomal reads assigned to the Viruses domain by Kraken/Bracken)</span>
<span id="cb24-415"><a href="#cb24-415" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Human (non-ribosomal reads assigned to the Eukarya domain by Kraken/Bracken)</span>
<span id="cb24-416"><a href="#cb24-416" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-417"><a href="#cb24-417" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-420"><a href="#cb24-420" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb24-421"><a href="#cb24-421" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: prepare-composition</span></span>
<span id="cb24-422"><a href="#cb24-422" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-423"><a href="#cb24-423" aria-hidden="true" tabindex="-1"></a>classifications <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Filtered"</span>, <span class="st">"Duplicate"</span>, <span class="st">"Ribosomal"</span>, <span class="st">"Unassigned"</span>,</span>
<span id="cb24-424"><a href="#cb24-424" aria-hidden="true" tabindex="-1"></a>                     <span class="st">"Bacterial"</span>, <span class="st">"Archaeal"</span>, <span class="st">"Viral"</span>, <span class="st">"Human"</span>)</span>
<span id="cb24-425"><a href="#cb24-425" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-426"><a href="#cb24-426" aria-hidden="true" tabindex="-1"></a><span class="co"># Import composition data</span></span>
<span id="cb24-427"><a href="#cb24-427" aria-hidden="true" tabindex="-1"></a>tax_final_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(results_dir, <span class="st">"taxonomy_final"</span>)</span>
<span id="cb24-428"><a href="#cb24-428" aria-hidden="true" tabindex="-1"></a>comp_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(tax_final_dir, <span class="st">"taxonomic_composition.tsv.gz"</span>)</span>
<span id="cb24-429"><a href="#cb24-429" aria-hidden="true" tabindex="-1"></a>comp <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(comp_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>)</span>
<span id="cb24-430"><a href="#cb24-430" aria-hidden="true" tabindex="-1"></a>comp_minor <span class="ot">&lt;-</span> comp <span class="sc">%&gt;%</span> </span>
<span id="cb24-431"><a href="#cb24-431" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(classification <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Archaeal"</span>, <span class="st">"Viral"</span>, <span class="st">"Human"</span>, <span class="st">"Other"</span>))</span>
<span id="cb24-432"><a href="#cb24-432" aria-hidden="true" tabindex="-1"></a>comp_assigned <span class="ot">&lt;-</span> comp <span class="sc">%&gt;%</span></span>
<span id="cb24-433"><a href="#cb24-433" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span> classification <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Filtered"</span>, <span class="st">"Duplicate"</span>, </span>
<span id="cb24-434"><a href="#cb24-434" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">"Ribosomal"</span>, <span class="st">"Unassigned"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb24-435"><a href="#cb24-435" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb24-436"><a href="#cb24-436" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads =</span> n_reads<span class="sc">/</span><span class="fu">sum</span>(n_reads))</span>
<span id="cb24-437"><a href="#cb24-437" aria-hidden="true" tabindex="-1"></a>comp_assigned_minor <span class="ot">&lt;-</span> comp_assigned <span class="sc">%&gt;%</span> </span>
<span id="cb24-438"><a href="#cb24-438" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(classification <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Archaeal"</span>, <span class="st">"Viral"</span>, <span class="st">"Human"</span>, <span class="st">"Other"</span>))</span>
<span id="cb24-439"><a href="#cb24-439" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-440"><a href="#cb24-440" aria-hidden="true" tabindex="-1"></a><span class="co"># Summarize composition</span></span>
<span id="cb24-441"><a href="#cb24-441" aria-hidden="true" tabindex="-1"></a>read_comp_summ <span class="ot">&lt;-</span> comp <span class="sc">%&gt;%</span> </span>
<span id="cb24-442"><a href="#cb24-442" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(classification) <span class="sc">%&gt;%</span></span>
<span id="cb24-443"><a href="#cb24-443" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads =</span> <span class="fu">sum</span>(n_reads), <span class="at">.groups =</span> <span class="st">"drop_last"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-444"><a href="#cb24-444" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">n_reads =</span> <span class="fu">replace_na</span>(n_reads,<span class="dv">0</span>),</span>
<span id="cb24-445"><a href="#cb24-445" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads =</span> n_reads<span class="sc">/</span><span class="fu">sum</span>(n_reads),</span>
<span id="cb24-446"><a href="#cb24-446" aria-hidden="true" tabindex="-1"></a>         <span class="at">pc_reads =</span> p_reads<span class="sc">*</span><span class="dv">100</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-447"><a href="#cb24-447" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">classification =</span> <span class="fu">factor</span>(classification, <span class="at">levels=</span>classifications)) <span class="sc">%&gt;%</span></span>
<span id="cb24-448"><a href="#cb24-448" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(classification, n_reads, pc_reads) <span class="sc">%&gt;%</span></span>
<span id="cb24-449"><a href="#cb24-449" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">`</span><span class="at"># of reads</span><span class="st">`</span> <span class="ot">=</span> n_reads, <span class="st">"% of reads"</span> <span class="ot">=</span> pc_reads) <span class="sc">%&gt;%</span></span>
<span id="cb24-450"><a href="#cb24-450" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">% of reads</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sprintf</span>(<span class="st">"%.2f"</span>, <span class="st">`</span><span class="at">% of reads</span><span class="st">`</span>))</span>
<span id="cb24-451"><a href="#cb24-451" aria-hidden="true" tabindex="-1"></a>read_comp_summ</span>
<span id="cb24-452"><a href="#cb24-452" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb24-453"><a href="#cb24-453" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-456"><a href="#cb24-456" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb24-457"><a href="#cb24-457" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-composition-all</span></span>
<span id="cb24-458"><a href="#cb24-458" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 7</span></span>
<span id="cb24-459"><a href="#cb24-459" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 8</span></span>
<span id="cb24-460"><a href="#cb24-460" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-461"><a href="#cb24-461" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare plotting templates</span></span>
<span id="cb24-462"><a href="#cb24-462" aria-hidden="true" tabindex="-1"></a>g_comp_base <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">x=</span>library, <span class="at">y=</span>p_reads, <span class="at">fill=</span>classification)) <span class="sc">+</span></span>
<span id="cb24-463"><a href="#cb24-463" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">name=</span><span class="st">"Plasma pool"</span>) <span class="sc">+</span></span>
<span id="cb24-464"><a href="#cb24-464" aria-hidden="true" tabindex="-1"></a>  theme_kit <span class="sc">+</span> </span>
<span id="cb24-465"><a href="#cb24-465" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust=</span><span class="dv">0</span>, <span class="at">face=</span><span class="st">"plain"</span>, <span class="at">size=</span><span class="fu">rel</span>(<span class="fl">1.5</span>)))</span>
<span id="cb24-466"><a href="#cb24-466" aria-hidden="true" tabindex="-1"></a>scale_y_pc_reads <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">partial</span>(scale_y_continuous, <span class="at">name =</span> <span class="st">"% Reads"</span>,</span>
<span id="cb24-467"><a href="#cb24-467" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>)</span>
<span id="cb24-468"><a href="#cb24-468" aria-hidden="true" tabindex="-1"></a>geom_comp <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">partial</span>(geom_col, <span class="at">position =</span> <span class="st">"stack"</span>, <span class="at">width =</span> <span class="dv">1</span>)</span>
<span id="cb24-469"><a href="#cb24-469" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-470"><a href="#cb24-470" aria-hidden="true" tabindex="-1"></a><span class="co"># Define a color palette for the classification</span></span>
<span id="cb24-471"><a href="#cb24-471" aria-hidden="true" tabindex="-1"></a>classification_colors <span class="ot">&lt;-</span> <span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">"Accent"</span>)</span>
<span id="cb24-472"><a href="#cb24-472" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(classification_colors) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Filtered"</span>, <span class="st">"Duplicate"</span>, <span class="st">"Ribosomal"</span>, <span class="st">"Unassigned"</span>, <span class="st">"Bacterial"</span>, <span class="st">"Archaeal"</span>, <span class="st">"Viral"</span>, <span class="st">"Human"</span>)</span>
<span id="cb24-473"><a href="#cb24-473" aria-hidden="true" tabindex="-1"></a>scale_fill_classification <span class="ot">&lt;-</span> <span class="cf">function</span>() {</span>
<span id="cb24-474"><a href="#cb24-474" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> classification_colors, <span class="at">name =</span> <span class="st">"Classification"</span>)</span>
<span id="cb24-475"><a href="#cb24-475" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb24-476"><a href="#cb24-476" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-477"><a href="#cb24-477" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot overall composition</span></span>
<span id="cb24-478"><a href="#cb24-478" aria-hidden="true" tabindex="-1"></a>g_comp <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> <span class="fu">geom_comp</span>(<span class="at">data =</span> comp) <span class="sc">+</span></span>
<span id="cb24-479"><a href="#cb24-479" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_pc_reads</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>)) <span class="sc">+</span></span>
<span id="cb24-480"><a href="#cb24-480" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_classification</span>() <span class="sc">+</span> </span>
<span id="cb24-481"><a href="#cb24-481" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Read composition (all reads, all groups)"</span>)</span>
<span id="cb24-482"><a href="#cb24-482" aria-hidden="true" tabindex="-1"></a>g_comp</span>
<span id="cb24-483"><a href="#cb24-483" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-484"><a href="#cb24-484" aria-hidden="true" tabindex="-1"></a>g_comp_assigned <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb24-485"><a href="#cb24-485" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_comp</span>(<span class="at">data =</span> comp_assigned) <span class="sc">+</span></span>
<span id="cb24-486"><a href="#cb24-486" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_pc_reads</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>)) <span class="sc">+</span></span>
<span id="cb24-487"><a href="#cb24-487" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_classification</span>() <span class="sc">+</span> </span>
<span id="cb24-488"><a href="#cb24-488" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Read composition (assigned reads, all groups)"</span>)</span>
<span id="cb24-489"><a href="#cb24-489" aria-hidden="true" tabindex="-1"></a>g_comp_assigned</span>
<span id="cb24-490"><a href="#cb24-490" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-491"><a href="#cb24-491" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb24-492"><a href="#cb24-492" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-493"><a href="#cb24-493" aria-hidden="true" tabindex="-1"></a><span class="fu">## Total viral content</span></span>
<span id="cb24-494"><a href="#cb24-494" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-495"><a href="#cb24-495" aria-hidden="true" tabindex="-1"></a>Total viral fraction average $7.70 \times 10^{-7}$ across samples. As a fraction of assigned (rather than total) reads, this jumped to $1.18 \times 10^{-6}$:</span>
<span id="cb24-496"><a href="#cb24-496" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-499"><a href="#cb24-499" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb24-500"><a href="#cb24-500" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: p-viral</span></span>
<span id="cb24-501"><a href="#cb24-501" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb24-502"><a href="#cb24-502" aria-hidden="true" tabindex="-1"></a>p_reads_viral_all <span class="ot">&lt;-</span> comp <span class="sc">%&gt;%</span> <span class="fu">filter</span>(classification <span class="sc">==</span> <span class="st">"Viral"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-503"><a href="#cb24-503" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">read_group =</span> <span class="st">"All reads"</span>)</span>
<span id="cb24-504"><a href="#cb24-504" aria-hidden="true" tabindex="-1"></a>p_reads_viral_assigned <span class="ot">&lt;-</span> comp_assigned <span class="sc">%&gt;%</span> <span class="fu">filter</span>(classification <span class="sc">==</span> <span class="st">"Viral"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-505"><a href="#cb24-505" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">read_group =</span> <span class="st">"Classified reads"</span>)</span>
<span id="cb24-506"><a href="#cb24-506" aria-hidden="true" tabindex="-1"></a>p_reads_viral <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(p_reads_viral_all, p_reads_viral_assigned)</span>
<span id="cb24-507"><a href="#cb24-507" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-508"><a href="#cb24-508" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb24-509"><a href="#cb24-509" aria-hidden="true" tabindex="-1"></a>g_viral <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(p_reads_viral, <span class="fu">aes</span>(<span class="at">x=</span>library, <span class="at">y=</span>p_reads, <span class="at">color =</span> read_group, <span class="at">group =</span> library)) <span class="sc">+</span></span>
<span id="cb24-510"><a href="#cb24-510" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb24-511"><a href="#cb24-511" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">color =</span> <span class="st">'black'</span>) <span class="sc">+</span></span>
<span id="cb24-512"><a href="#cb24-512" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">name=</span><span class="st">"Plasma pool"</span>) <span class="sc">+</span></span>
<span id="cb24-513"><a href="#cb24-513" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>(<span class="at">name=</span><span class="st">"Viral read fraction"</span>, <span class="at">labels =</span> <span class="fu">label_log</span>(<span class="at">digits=</span><span class="dv">2</span>), <span class="at">limits =</span> <span class="fu">c</span>(<span class="fl">1e-6</span>, <span class="fl">1e-2</span>)) <span class="sc">+</span></span>
<span id="cb24-514"><a href="#cb24-514" aria-hidden="true" tabindex="-1"></a>  theme_kit <span class="sc">+</span> </span>
<span id="cb24-515"><a href="#cb24-515" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>()</span>
<span id="cb24-516"><a href="#cb24-516" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-517"><a href="#cb24-517" aria-hidden="true" tabindex="-1"></a>g_viral</span>
<span id="cb24-518"><a href="#cb24-518" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-519"><a href="#cb24-519" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb24-520"><a href="#cb24-520" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-521"><a href="#cb24-521" aria-hidden="true" tabindex="-1"></a><span class="fu">## Taxonomic composition of viruses</span></span>
<span id="cb24-522"><a href="#cb24-522" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-523"><a href="#cb24-523" aria-hidden="true" tabindex="-1"></a>The two dominant viruses we see are Anellovirdae and Phycodnaviridae. The threshold for the label "other" are the set of families that make up less than 30% composition in all samples.</span>
<span id="cb24-524"><a href="#cb24-524" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-527"><a href="#cb24-527" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb24-528"><a href="#cb24-528" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: extract-viral-taxa</span></span>
<span id="cb24-529"><a href="#cb24-529" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb24-530"><a href="#cb24-530" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-531"><a href="#cb24-531" aria-hidden="true" tabindex="-1"></a><span class="co"># Get viral taxonomy</span></span>
<span id="cb24-532"><a href="#cb24-532" aria-hidden="true" tabindex="-1"></a>viral_taxa_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"total-virus-db.tsv.gz"</span>)</span>
<span id="cb24-533"><a href="#cb24-533" aria-hidden="true" tabindex="-1"></a>viral_taxa <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(viral_taxa_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb24-534"><a href="#cb24-534" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-535"><a href="#cb24-535" aria-hidden="true" tabindex="-1"></a><span class="co"># Get Kraken reports</span></span>
<span id="cb24-536"><a href="#cb24-536" aria-hidden="true" tabindex="-1"></a>reports_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(tax_final_dir, <span class="st">"kraken_reports.tsv.gz"</span>)</span>
<span id="cb24-537"><a href="#cb24-537" aria-hidden="true" tabindex="-1"></a>reports <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(reports_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-538"><a href="#cb24-538" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(sample)</span>
<span id="cb24-539"><a href="#cb24-539" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-540"><a href="#cb24-540" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter to viral taxa</span></span>
<span id="cb24-541"><a href="#cb24-541" aria-hidden="true" tabindex="-1"></a>kraken_reports_viral <span class="ot">&lt;-</span> <span class="fu">filter</span>(reports, taxid <span class="sc">%in%</span> viral_taxa<span class="sc">$</span>taxid) <span class="sc">%&gt;%</span></span>
<span id="cb24-542"><a href="#cb24-542" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb24-543"><a href="#cb24-543" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_viral =</span> n_reads_clade<span class="sc">/</span>n_reads_clade[<span class="dv">1</span>])</span>
<span id="cb24-544"><a href="#cb24-544" aria-hidden="true" tabindex="-1"></a>kraken_reports_viral_cleaned <span class="ot">&lt;-</span> kraken_reports_viral <span class="sc">%&gt;%</span></span>
<span id="cb24-545"><a href="#cb24-545" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>pc_reads_total, <span class="sc">-</span>n_reads_direct, <span class="sc">-</span><span class="fu">contains</span>(<span class="st">"minimizers"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb24-546"><a href="#cb24-546" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, taxid, p_reads_viral, n_reads_clade, <span class="fu">everything</span>()) <span class="sc">%&gt;%</span> ungroup</span>
<span id="cb24-547"><a href="#cb24-547" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-548"><a href="#cb24-548" aria-hidden="true" tabindex="-1"></a>viral_classes <span class="ot">&lt;-</span> kraken_reports_viral_cleaned <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">==</span> <span class="st">"C"</span>)</span>
<span id="cb24-549"><a href="#cb24-549" aria-hidden="true" tabindex="-1"></a>viral_families <span class="ot">&lt;-</span> kraken_reports_viral_cleaned <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">==</span> <span class="st">"F"</span>)</span>
<span id="cb24-550"><a href="#cb24-550" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb24-551"><a href="#cb24-551" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-554"><a href="#cb24-554" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb24-555"><a href="#cb24-555" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: viral-family-composition</span></span>
<span id="cb24-556"><a href="#cb24-556" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb24-557"><a href="#cb24-557" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-558"><a href="#cb24-558" aria-hidden="true" tabindex="-1"></a>major_threshold <span class="ot">&lt;-</span> <span class="fl">0.3</span></span>
<span id="cb24-559"><a href="#cb24-559" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-560"><a href="#cb24-560" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify major viral families</span></span>
<span id="cb24-561"><a href="#cb24-561" aria-hidden="true" tabindex="-1"></a>viral_families_major_tab <span class="ot">&lt;-</span> viral_families <span class="sc">%&gt;%</span> </span>
<span id="cb24-562"><a href="#cb24-562" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb24-563"><a href="#cb24-563" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads_viral_max =</span> <span class="fu">max</span>(p_reads_viral), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-564"><a href="#cb24-564" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_viral_max <span class="sc">&gt;=</span> major_threshold)</span>
<span id="cb24-565"><a href="#cb24-565" aria-hidden="true" tabindex="-1"></a>viral_families_major_list <span class="ot">&lt;-</span> viral_families_major_tab <span class="sc">%&gt;%</span> <span class="fu">pull</span>(name)</span>
<span id="cb24-566"><a href="#cb24-566" aria-hidden="true" tabindex="-1"></a>viral_families_major <span class="ot">&lt;-</span> viral_families <span class="sc">%&gt;%</span> </span>
<span id="cb24-567"><a href="#cb24-567" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(name <span class="sc">%in%</span> viral_families_major_list) <span class="sc">%&gt;%</span></span>
<span id="cb24-568"><a href="#cb24-568" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, taxid, sample, p_reads_viral)</span>
<span id="cb24-569"><a href="#cb24-569" aria-hidden="true" tabindex="-1"></a>viral_families_minor <span class="ot">&lt;-</span> viral_families_major <span class="sc">%&gt;%</span> </span>
<span id="cb24-570"><a href="#cb24-570" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb24-571"><a href="#cb24-571" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads_viral_major =</span> <span class="fu">sum</span>(p_reads_viral), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-572"><a href="#cb24-572" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="st">"Other"</span>, <span class="at">taxid=</span><span class="cn">NA</span>, <span class="at">p_reads_viral =</span> <span class="dv">1</span><span class="sc">-</span>p_reads_viral_major) <span class="sc">%&gt;%</span></span>
<span id="cb24-573"><a href="#cb24-573" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, taxid, sample, p_reads_viral)</span>
<span id="cb24-574"><a href="#cb24-574" aria-hidden="true" tabindex="-1"></a>viral_families_display <span class="ot">&lt;-</span> viral_families_major <span class="sc">%&gt;%</span> </span>
<span id="cb24-575"><a href="#cb24-575" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_rows</span>(viral_families_minor) <span class="sc">%&gt;%</span></span>
<span id="cb24-576"><a href="#cb24-576" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_viral)) <span class="sc">%&gt;%</span> </span>
<span id="cb24-577"><a href="#cb24-577" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">factor</span>(name, <span class="at">levels=</span><span class="fu">c</span>(viral_families_major_list, <span class="st">"Other"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb24-578"><a href="#cb24-578" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_viral, <span class="at">classification=</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb24-579"><a href="#cb24-579" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">'sample'</span>)</span>
<span id="cb24-580"><a href="#cb24-580" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-581"><a href="#cb24-581" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-582"><a href="#cb24-582" aria-hidden="true" tabindex="-1"></a>pal <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">'Dark2'</span>),<span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">'Accent'</span>), <span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">'Set1'</span>), <span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">'Pastel1'</span>), <span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">'Pastel2'</span>))</span>
<span id="cb24-583"><a href="#cb24-583" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-584"><a href="#cb24-584" aria-hidden="true" tabindex="-1"></a>n_classifications <span class="ot">&lt;-</span> <span class="fu">length</span>(<span class="fu">unique</span>(viral_families_display<span class="sc">$</span>classification)) <span class="sc">-</span> <span class="dv">1</span></span>
<span id="cb24-585"><a href="#cb24-585" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-586"><a href="#cb24-586" aria-hidden="true" tabindex="-1"></a>custom_palette_with_black <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb24-587"><a href="#cb24-587" aria-hidden="true" tabindex="-1"></a>  pal[<span class="dv">1</span><span class="sc">:</span>n_classifications],</span>
<span id="cb24-588"><a href="#cb24-588" aria-hidden="true" tabindex="-1"></a>  <span class="st">"black"</span></span>
<span id="cb24-589"><a href="#cb24-589" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb24-590"><a href="#cb24-590" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-591"><a href="#cb24-591" aria-hidden="true" tabindex="-1"></a>g_families <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> viral_families_display, <span class="fu">aes</span>(<span class="at">x =</span> library, <span class="at">y =</span> p_reads, <span class="at">fill =</span> classification)) <span class="sc">+</span></span>
<span id="cb24-592"><a href="#cb24-592" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position =</span> <span class="st">"stack"</span>, <span class="at">width =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb24-593"><a href="#cb24-593" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="st">"% Viral Reads"</span>, </span>
<span id="cb24-594"><a href="#cb24-594" aria-hidden="true" tabindex="-1"></a>                     <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fl">1.01</span>), </span>
<span id="cb24-595"><a href="#cb24-595" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="fl">0.2</span>),</span>
<span id="cb24-596"><a href="#cb24-596" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>), </span>
<span id="cb24-597"><a href="#cb24-597" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="cf">function</span>(y) y <span class="sc">*</span> <span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb24-598"><a href="#cb24-598" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> custom_palette_with_black) <span class="sc">+</span></span>
<span id="cb24-599"><a href="#cb24-599" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">name =</span> <span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb24-600"><a href="#cb24-600" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb24-601"><a href="#cb24-601" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb24-602"><a href="#cb24-602" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>),</span>
<span id="cb24-603"><a href="#cb24-603" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid.major.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb24-604"><a href="#cb24-604" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid.minor.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb24-605"><a href="#cb24-605" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">"bottom"</span></span>
<span id="cb24-606"><a href="#cb24-606" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb24-607"><a href="#cb24-607" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Viral Family Composition"</span>,</span>
<span id="cb24-608"><a href="#cb24-608" aria-hidden="true" tabindex="-1"></a>       <span class="at">fill =</span> <span class="st">"Classification"</span>)</span>
<span id="cb24-609"><a href="#cb24-609" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-610"><a href="#cb24-610" aria-hidden="true" tabindex="-1"></a>g_families</span>
<span id="cb24-611"><a href="#cb24-611" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-612"><a href="#cb24-612" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb24-613"><a href="#cb24-613" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-614"><a href="#cb24-614" aria-hidden="true" tabindex="-1"></a><span class="fu"># Human-infecting virus reads</span></span>
<span id="cb24-615"><a href="#cb24-615" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-616"><a href="#cb24-616" aria-hidden="true" tabindex="-1"></a><span class="fu">## Overall relative abundance</span></span>
<span id="cb24-617"><a href="#cb24-617" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-618"><a href="#cb24-618" aria-hidden="true" tabindex="-1"></a>I calculated the relative abundance of human-infecting viruses in two ways:</span>
<span id="cb24-619"><a href="#cb24-619" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-620"><a href="#cb24-620" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>First, as the total number of deduplicated human-virus reads in each sample, divided by the number of raw reads ("All reads"). This results in a viral fraction of $2.62 \times 10^{-8}$ across all samples</span>
<span id="cb24-621"><a href="#cb24-621" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-622"><a href="#cb24-622" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Second, as a fraction of preprocessed (cleaned, deduplicated, computationally ribodepleted) reads ("Preprocessed reads"). This results in a viral fraction of $3.73 \times 10^{-8}$ across all samples.</span>
<span id="cb24-623"><a href="#cb24-623" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-626"><a href="#cb24-626" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb24-627"><a href="#cb24-627" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: prepare-hv</span></span>
<span id="cb24-628"><a href="#cb24-628" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb24-629"><a href="#cb24-629" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-630"><a href="#cb24-630" aria-hidden="true" tabindex="-1"></a><span class="co"># Import and format reads</span></span>
<span id="cb24-631"><a href="#cb24-631" aria-hidden="true" tabindex="-1"></a>hv_reads_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(hv_dir, <span class="st">"hv_hits_putative_collapsed.tsv.gz"</span>)</span>
<span id="cb24-632"><a href="#cb24-632" aria-hidden="true" tabindex="-1"></a>mrg_hv <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(hv_reads_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-633"><a href="#cb24-633" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb24-634"><a href="#cb24-634" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">kraken_label =</span> <span class="fu">ifelse</span>(assigned_hv, <span class="st">"Kraken2 HV assignment"</span>,</span>
<span id="cb24-635"><a href="#cb24-635" aria-hidden="true" tabindex="-1"></a>                               <span class="st">"No Kraken2 assignment"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb24-636"><a href="#cb24-636" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">adj_score_max =</span> <span class="fu">pmax</span>(adj_score_fwd, adj_score_rev),</span>
<span id="cb24-637"><a href="#cb24-637" aria-hidden="true" tabindex="-1"></a>         <span class="at">highscore =</span> adj_score_max <span class="sc">&gt;=</span> <span class="dv">20</span>,</span>
<span id="cb24-638"><a href="#cb24-638" aria-hidden="true" tabindex="-1"></a>         <span class="at">hv_status =</span> assigned_hv <span class="sc">|</span> highscore) <span class="sc">%&gt;%</span></span>
<span id="cb24-639"><a href="#cb24-639" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">taxid_all =</span> taxid, <span class="at">taxid =</span> taxid_best)</span>
<span id="cb24-640"><a href="#cb24-640" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb24-641"><a href="#cb24-641" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-644"><a href="#cb24-644" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb24-645"><a href="#cb24-645" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: count-hv-reads</span></span>
<span id="cb24-646"><a href="#cb24-646" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb24-647"><a href="#cb24-647" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb24-648"><a href="#cb24-648" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb24-649"><a href="#cb24-649" aria-hidden="true" tabindex="-1"></a><span class="co"># Get raw read counts</span></span>
<span id="cb24-650"><a href="#cb24-650" aria-hidden="true" tabindex="-1"></a>read_counts_raw <span class="ot">&lt;-</span> <span class="fu">filter</span>(basic_stats_raw) <span class="sc">%&gt;%</span></span>
<span id="cb24-651"><a href="#cb24-651" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, <span class="at">n_reads_raw =</span> n_read_pairs)</span>
<span id="cb24-652"><a href="#cb24-652" aria-hidden="true" tabindex="-1"></a>read_counts_preproc <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"ribo_initial"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-653"><a href="#cb24-653" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, <span class="at">n_reads_preproc =</span> n_read_pairs)</span>
<span id="cb24-654"><a href="#cb24-654" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-655"><a href="#cb24-655" aria-hidden="true" tabindex="-1"></a><span class="co"># Get HV read counts</span></span>
<span id="cb24-656"><a href="#cb24-656" aria-hidden="true" tabindex="-1"></a>read_counts_hv <span class="ot">&lt;-</span> mrg_hv <span class="sc">%&gt;%</span> <span class="fu">filter</span>(hv_status) <span class="sc">%&gt;%</span> </span>
<span id="cb24-657"><a href="#cb24-657" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb24-658"><a href="#cb24-658" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name=</span><span class="st">"n_reads_hv"</span>)</span>
<span id="cb24-659"><a href="#cb24-659" aria-hidden="true" tabindex="-1"></a>read_counts <span class="ot">&lt;-</span> read_counts_raw <span class="sc">%&gt;%</span></span>
<span id="cb24-660"><a href="#cb24-660" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(read_counts_hv, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"sample"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb24-661"><a href="#cb24-661" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">n_reads_hv =</span> <span class="fu">replace_na</span>(n_reads_hv, <span class="dv">0</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb24-662"><a href="#cb24-662" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(read_counts_preproc, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"sample"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb24-663"><a href="#cb24-663" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"sample"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb24-664"><a href="#cb24-664" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, n_reads_raw, n_reads_preproc, n_reads_hv) <span class="sc">%&gt;%</span></span>
<span id="cb24-665"><a href="#cb24-665" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">n_samples =</span> <span class="dv">1</span>,</span>
<span id="cb24-666"><a href="#cb24-666" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_total =</span> n_reads_hv<span class="sc">/</span>n_reads_raw,</span>
<span id="cb24-667"><a href="#cb24-667" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_preproc =</span> n_reads_hv<span class="sc">/</span>n_reads_preproc)</span>
<span id="cb24-668"><a href="#cb24-668" aria-hidden="true" tabindex="-1"></a>read_counts_long <span class="ot">&lt;-</span> read_counts <span class="sc">%&gt;%</span></span>
<span id="cb24-669"><a href="#cb24-669" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="fu">starts_with</span>(<span class="st">"p_reads"</span>), <span class="at">names_to=</span><span class="st">"read_group"</span>, <span class="at">values_to=</span><span class="st">"p_reads"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-670"><a href="#cb24-670" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">read_group =</span> <span class="fu">ifelse</span>(read_group <span class="sc">==</span> <span class="st">"p_reads_total"</span>, <span class="st">"All reads"</span>, <span class="st">"Preprocessed reads"</span>))</span>
<span id="cb24-671"><a href="#cb24-671" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-672"><a href="#cb24-672" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine for display</span></span>
<span id="cb24-673"><a href="#cb24-673" aria-hidden="true" tabindex="-1"></a>read_counts_agg <span class="ot">&lt;-</span> read_counts <span class="sc">%&gt;%</span></span>
<span id="cb24-674"><a href="#cb24-674" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_total =</span> n_reads_hv<span class="sc">/</span>n_reads_raw,</span>
<span id="cb24-675"><a href="#cb24-675" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_preproc =</span> n_reads_hv<span class="sc">/</span>n_reads_preproc) <span class="sc">%&gt;%</span></span>
<span id="cb24-676"><a href="#cb24-676" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"sample"</span>))</span>
<span id="cb24-677"><a href="#cb24-677" aria-hidden="true" tabindex="-1"></a>read_counts_agg_long <span class="ot">&lt;-</span> read_counts_agg <span class="sc">%&gt;%</span></span>
<span id="cb24-678"><a href="#cb24-678" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="fu">starts_with</span>(<span class="st">"p_reads"</span>), <span class="at">names_to=</span><span class="st">"read_group"</span>, <span class="at">values_to=</span><span class="st">"p_reads"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-679"><a href="#cb24-679" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">read_group =</span> <span class="fu">ifelse</span>(read_group <span class="sc">==</span> <span class="st">"p_reads_total"</span>, <span class="st">"All reads (HV)"</span>, <span class="st">"Preprocessed reads (HV)"</span>)) </span>
<span id="cb24-680"><a href="#cb24-680" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-681"><a href="#cb24-681" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize</span></span>
<span id="cb24-682"><a href="#cb24-682" aria-hidden="true" tabindex="-1"></a>g_read_counts <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(read_counts_agg_long, <span class="fu">aes</span>(<span class="at">x=</span>library, <span class="at">y=</span>p_reads, <span class="at">color =</span> read_group, <span class="at">group =</span> library)) <span class="sc">+</span></span>
<span id="cb24-683"><a href="#cb24-683" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb24-684"><a href="#cb24-684" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">color =</span> <span class="st">'black'</span>) <span class="sc">+</span></span>
<span id="cb24-685"><a href="#cb24-685" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>(<span class="at">name =</span> <span class="st">"Unique human-viral read fraction"</span>, <span class="at">labels =</span> <span class="fu">label_log</span>(<span class="at">digits=</span><span class="dv">2</span>), <span class="at">limits =</span> <span class="fu">c</span>(<span class="fl">1e-6</span>, <span class="fl">1e-2</span>)) <span class="sc">+</span></span>
<span id="cb24-686"><a href="#cb24-686" aria-hidden="true" tabindex="-1"></a>  theme_kit <span class="sc">+</span> </span>
<span id="cb24-687"><a href="#cb24-687" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>()</span>
<span id="cb24-688"><a href="#cb24-688" aria-hidden="true" tabindex="-1"></a>g_read_counts</span>
<span id="cb24-689"><a href="#cb24-689" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-690"><a href="#cb24-690" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb24-691"><a href="#cb24-691" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-694"><a href="#cb24-694" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb24-695"><a href="#cb24-695" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: compare-virus-to-hv</span></span>
<span id="cb24-696"><a href="#cb24-696" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb24-697"><a href="#cb24-697" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb24-698"><a href="#cb24-698" aria-hidden="true" tabindex="-1"></a>combine_viruses <span class="ot">&lt;-</span> <span class="fu">rbind</span>(p_reads_viral  <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(read_group) <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(<span class="at">mean=</span><span class="fu">mean</span>(p_reads)) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">type =</span> <span class="st">"Viral reads"</span>), read_counts_agg_long  <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(read_group) <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(<span class="at">mean=</span><span class="fu">mean</span>(p_reads))<span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">type =</span> <span class="st">"HV reads"</span>))</span>
<span id="cb24-699"><a href="#cb24-699" aria-hidden="true" tabindex="-1"></a><span class="co"># Format the mean values in scientific notation</span></span>
<span id="cb24-700"><a href="#cb24-700" aria-hidden="true" tabindex="-1"></a>combine_viruses <span class="ot">&lt;-</span> combine_viruses <span class="sc">%&gt;%</span></span>
<span id="cb24-701"><a href="#cb24-701" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mean_formatted =</span> <span class="fu">sprintf</span>(<span class="st">"%.2e"</span>, mean))</span>
<span id="cb24-702"><a href="#cb24-702" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-703"><a href="#cb24-703" aria-hidden="true" tabindex="-1"></a>combine_viruses <span class="sc">%&gt;%</span> <span class="fu">select</span>(type, read_group, <span class="at">mean=</span>mean_formatted)</span>
<span id="cb24-704"><a href="#cb24-704" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-705"><a href="#cb24-705" aria-hidden="true" tabindex="-1"></a>g_read_counts <span class="sc">+</span></span>
<span id="cb24-706"><a href="#cb24-706" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> p_reads_viral, <span class="fu">aes</span>(<span class="at">x =</span> library, <span class="at">y =</span> p_reads, <span class="at">color =</span> read_group), <span class="at">size =</span> <span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb24-707"><a href="#cb24-707" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">data =</span> p_reads_viral, <span class="fu">aes</span>(<span class="at">x =</span> library, <span class="at">y =</span> p_reads, <span class="at">group =</span> library), <span class="at">color =</span> <span class="st">'grey'</span>) <span class="sc">+</span></span>
<span id="cb24-708"><a href="#cb24-708" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_brewer</span>(<span class="at">palette =</span> <span class="st">'Accent'</span>)</span>
<span id="cb24-709"><a href="#cb24-709" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-710"><a href="#cb24-710" aria-hidden="true" tabindex="-1"></a><span class="co">#ggarrange(g_viral, g_read_counts, ncol=2)</span></span>
<span id="cb24-711"><a href="#cb24-711" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb24-712"><a href="#cb24-712" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-713"><a href="#cb24-713" aria-hidden="true" tabindex="-1"></a><span class="fu">## Overall taxonomy and composition</span></span>
<span id="cb24-714"><a href="#cb24-714" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-715"><a href="#cb24-715" aria-hidden="true" tabindex="-1"></a>Composition of HV reads was changed from when looking at all viral reads. Only 1 out of the 137 samples had reads for human infecting viruses, with read composition coming from well established viral families in whole blood.</span>
<span id="cb24-716"><a href="#cb24-716" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-719"><a href="#cb24-719" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb24-720"><a href="#cb24-720" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: raise-hv-taxa</span></span>
<span id="cb24-721"><a href="#cb24-721" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb24-722"><a href="#cb24-722" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-723"><a href="#cb24-723" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter samples and add viral taxa information</span></span>
<span id="cb24-724"><a href="#cb24-724" aria-hidden="true" tabindex="-1"></a>samples_keep <span class="ot">&lt;-</span> read_counts <span class="sc">%&gt;%</span> <span class="fu">filter</span>(n_reads_hv <span class="sc">&gt;</span> <span class="dv">5</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(sample)</span>
<span id="cb24-725"><a href="#cb24-725" aria-hidden="true" tabindex="-1"></a>mrg_hv_named <span class="ot">&lt;-</span> mrg_hv <span class="sc">%&gt;%</span> <span class="fu">filter</span>(sample <span class="sc">%in%</span> samples_keep, hv_status) <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(viral_taxa, <span class="at">by=</span><span class="st">"taxid"</span>) </span>
<span id="cb24-726"><a href="#cb24-726" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-727"><a href="#cb24-727" aria-hidden="true" tabindex="-1"></a><span class="co"># Discover viral species &amp; genera for HV reads</span></span>
<span id="cb24-728"><a href="#cb24-728" aria-hidden="true" tabindex="-1"></a>raise_rank <span class="ot">&lt;-</span> <span class="cf">function</span>(read_db, taxid_db, <span class="at">out_rank =</span> <span class="st">"species"</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>){</span>
<span id="cb24-729"><a href="#cb24-729" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get higher ranks than search rank</span></span>
<span id="cb24-730"><a href="#cb24-730" aria-hidden="true" tabindex="-1"></a>  ranks <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"subspecies"</span>, <span class="st">"species"</span>, <span class="st">"subgenus"</span>, <span class="st">"genus"</span>, <span class="st">"subfamily"</span>, <span class="st">"family"</span>, <span class="st">"suborder"</span>, <span class="st">"order"</span>, <span class="st">"class"</span>, <span class="st">"subphylum"</span>, <span class="st">"phylum"</span>, <span class="st">"kingdom"</span>, <span class="st">"superkingdom"</span>)</span>
<span id="cb24-731"><a href="#cb24-731" aria-hidden="true" tabindex="-1"></a>  rank_match <span class="ot">&lt;-</span> <span class="fu">which.max</span>(ranks <span class="sc">==</span> out_rank)</span>
<span id="cb24-732"><a href="#cb24-732" aria-hidden="true" tabindex="-1"></a>  high_ranks <span class="ot">&lt;-</span> ranks[rank_match<span class="sc">:</span><span class="fu">length</span>(ranks)]</span>
<span id="cb24-733"><a href="#cb24-733" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Merge read DB and taxid DB</span></span>
<span id="cb24-734"><a href="#cb24-734" aria-hidden="true" tabindex="-1"></a>  reads <span class="ot">&lt;-</span> read_db <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>parent_taxid, <span class="sc">-</span>rank, <span class="sc">-</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb24-735"><a href="#cb24-735" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(taxid_db, <span class="at">by=</span><span class="st">"taxid"</span>)</span>
<span id="cb24-736"><a href="#cb24-736" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract sequences that are already at appropriate rank</span></span>
<span id="cb24-737"><a href="#cb24-737" aria-hidden="true" tabindex="-1"></a>  reads_rank <span class="ot">&lt;-</span> <span class="fu">filter</span>(reads, rank <span class="sc">==</span> out_rank)</span>
<span id="cb24-738"><a href="#cb24-738" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Drop sequences at a higher rank and return unclassified sequences</span></span>
<span id="cb24-739"><a href="#cb24-739" aria-hidden="true" tabindex="-1"></a>  reads_norank <span class="ot">&lt;-</span> reads <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">!=</span> out_rank, <span class="sc">!</span>rank <span class="sc">%in%</span> high_ranks, <span class="sc">!</span><span class="fu">is.na</span>(taxid))</span>
<span id="cb24-740"><a href="#cb24-740" aria-hidden="true" tabindex="-1"></a>  <span class="cf">while</span>(<span class="fu">nrow</span>(reads_norank) <span class="sc">&gt;</span> <span class="dv">0</span>){ <span class="co"># As long as there are unclassified sequences...</span></span>
<span id="cb24-741"><a href="#cb24-741" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Promote read taxids and re-merge with taxid DB, then re-classify and filter</span></span>
<span id="cb24-742"><a href="#cb24-742" aria-hidden="true" tabindex="-1"></a>    reads_remaining <span class="ot">&lt;-</span> reads_norank <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">taxid =</span> parent_taxid) <span class="sc">%&gt;%</span></span>
<span id="cb24-743"><a href="#cb24-743" aria-hidden="true" tabindex="-1"></a>      <span class="fu">select</span>(<span class="sc">-</span>parent_taxid, <span class="sc">-</span>rank, <span class="sc">-</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb24-744"><a href="#cb24-744" aria-hidden="true" tabindex="-1"></a>      <span class="fu">left_join</span>(taxid_db, <span class="at">by=</span><span class="st">"taxid"</span>)</span>
<span id="cb24-745"><a href="#cb24-745" aria-hidden="true" tabindex="-1"></a>    reads_rank <span class="ot">&lt;-</span> reads_remaining <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">==</span> out_rank) <span class="sc">%&gt;%</span></span>
<span id="cb24-746"><a href="#cb24-746" aria-hidden="true" tabindex="-1"></a>      <span class="fu">bind_rows</span>(reads_rank)</span>
<span id="cb24-747"><a href="#cb24-747" aria-hidden="true" tabindex="-1"></a>    reads_norank <span class="ot">&lt;-</span> reads_remaining <span class="sc">%&gt;%</span></span>
<span id="cb24-748"><a href="#cb24-748" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(rank <span class="sc">!=</span> out_rank, <span class="sc">!</span>rank <span class="sc">%in%</span> high_ranks, <span class="sc">!</span><span class="fu">is.na</span>(taxid))</span>
<span id="cb24-749"><a href="#cb24-749" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb24-750"><a href="#cb24-750" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Finally, extract and append reads that were excluded during the process</span></span>
<span id="cb24-751"><a href="#cb24-751" aria-hidden="true" tabindex="-1"></a>  reads_dropped <span class="ot">&lt;-</span> reads <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span>seq_id <span class="sc">%in%</span> reads_rank<span class="sc">$</span>seq_id)</span>
<span id="cb24-752"><a href="#cb24-752" aria-hidden="true" tabindex="-1"></a>  reads_out <span class="ot">&lt;-</span> reads_rank <span class="sc">%&gt;%</span> <span class="fu">bind_rows</span>(reads_dropped) <span class="sc">%&gt;%</span></span>
<span id="cb24-753"><a href="#cb24-753" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(<span class="sc">-</span>parent_taxid, <span class="sc">-</span>rank, <span class="sc">-</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb24-754"><a href="#cb24-754" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(taxid_db, <span class="at">by=</span><span class="st">"taxid"</span>)</span>
<span id="cb24-755"><a href="#cb24-755" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(reads_out)</span>
<span id="cb24-756"><a href="#cb24-756" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb24-757"><a href="#cb24-757" aria-hidden="true" tabindex="-1"></a>hv_reads_species <span class="ot">&lt;-</span> <span class="fu">raise_rank</span>(mrg_hv_named, viral_taxa, <span class="st">"species"</span>)</span>
<span id="cb24-758"><a href="#cb24-758" aria-hidden="true" tabindex="-1"></a>hv_reads_genus <span class="ot">&lt;-</span> <span class="fu">raise_rank</span>(mrg_hv_named, viral_taxa, <span class="st">"genus"</span>)</span>
<span id="cb24-759"><a href="#cb24-759" aria-hidden="true" tabindex="-1"></a>hv_reads_family <span class="ot">&lt;-</span> <span class="fu">raise_rank</span>(mrg_hv_named, viral_taxa, <span class="st">"family"</span>)</span>
<span id="cb24-760"><a href="#cb24-760" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb24-761"><a href="#cb24-761" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-764"><a href="#cb24-764" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb24-765"><a href="#cb24-765" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-family</span></span>
<span id="cb24-766"><a href="#cb24-766" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb24-767"><a href="#cb24-767" aria-hidden="true" tabindex="-1"></a>threshold_major_family <span class="ot">&lt;-</span> <span class="fl">0.01</span></span>
<span id="cb24-768"><a href="#cb24-768" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-769"><a href="#cb24-769" aria-hidden="true" tabindex="-1"></a><span class="co"># Count reads for each human-viral family</span></span>
<span id="cb24-770"><a href="#cb24-770" aria-hidden="true" tabindex="-1"></a>hv_family_counts <span class="ot">&lt;-</span> hv_reads_family <span class="sc">%&gt;%</span> </span>
<span id="cb24-771"><a href="#cb24-771" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb24-772"><a href="#cb24-772" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-773"><a href="#cb24-773" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb24-774"><a href="#cb24-774" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_hv =</span> n_reads_hv<span class="sc">/</span><span class="fu">sum</span>(n_reads_hv))</span>
<span id="cb24-775"><a href="#cb24-775" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-776"><a href="#cb24-776" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify high-ranking families and group others</span></span>
<span id="cb24-777"><a href="#cb24-777" aria-hidden="true" tabindex="-1"></a>hv_family_major_tab <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span> </span>
<span id="cb24-778"><a href="#cb24-778" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_hv <span class="sc">==</span> <span class="fu">max</span>(p_reads_hv)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-779"><a href="#cb24-779" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_hv)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(p_reads_hv <span class="sc">&gt;</span> threshold_major_family)</span>
<span id="cb24-780"><a href="#cb24-780" aria-hidden="true" tabindex="-1"></a>hv_family_counts_major <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span></span>
<span id="cb24-781"><a href="#cb24-781" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">ifelse</span>(name <span class="sc">%in%</span> hv_family_major_tab<span class="sc">$</span>name, name, <span class="st">"Other"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb24-782"><a href="#cb24-782" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, name_display) <span class="sc">%&gt;%</span></span>
<span id="cb24-783"><a href="#cb24-783" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_hv =</span> <span class="fu">sum</span>(n_reads_hv), <span class="at">p_reads_hv =</span> <span class="fu">sum</span>(p_reads_hv), </span>
<span id="cb24-784"><a href="#cb24-784" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-785"><a href="#cb24-785" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">factor</span>(name_display, </span>
<span id="cb24-786"><a href="#cb24-786" aria-hidden="true" tabindex="-1"></a>                               <span class="at">levels =</span> <span class="fu">c</span>(hv_family_major_tab<span class="sc">$</span>name, <span class="st">"Other"</span>)))</span>
<span id="cb24-787"><a href="#cb24-787" aria-hidden="true" tabindex="-1"></a>hv_family_counts_display <span class="ot">&lt;-</span> hv_family_counts_major <span class="sc">%&gt;%</span></span>
<span id="cb24-788"><a href="#cb24-788" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_hv, <span class="at">classification =</span> name_display) <span class="sc">%&gt;%</span></span>
<span id="cb24-789"><a href="#cb24-789" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by =</span> <span class="st">'sample'</span>)</span>
<span id="cb24-790"><a href="#cb24-790" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-791"><a href="#cb24-791" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb24-792"><a href="#cb24-792" aria-hidden="true" tabindex="-1"></a>g_hv_family <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb24-793"><a href="#cb24-793" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">data=</span>hv_family_counts_display, <span class="at">position =</span> <span class="st">"stack"</span>, <span class="at">width=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb24-794"><a href="#cb24-794" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% HV Reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb24-795"><a href="#cb24-795" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb24-796"><a href="#cb24-796" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb24-797"><a href="#cb24-797" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">'Accent'</span>, <span class="at">name =</span> <span class="st">"Viral family"</span>) <span class="sc">+</span></span>
<span id="cb24-798"><a href="#cb24-798" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">"Family composition of human-viral reads"</span>) <span class="sc">+</span></span>
<span id="cb24-799"><a href="#cb24-799" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">4</span>)) <span class="sc">+</span></span>
<span id="cb24-800"><a href="#cb24-800" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">1.4</span>), <span class="at">hjust=</span><span class="dv">0</span>, <span class="at">face=</span><span class="st">"plain"</span>))</span>
<span id="cb24-801"><a href="#cb24-801" aria-hidden="true" tabindex="-1"></a>g_hv_family</span>
<span id="cb24-802"><a href="#cb24-802" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-803"><a href="#cb24-803" aria-hidden="true" tabindex="-1"></a><span class="co"># Get most prominent families for text</span></span>
<span id="cb24-804"><a href="#cb24-804" aria-hidden="true" tabindex="-1"></a>hv_family_collate <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span></span>
<span id="cb24-805"><a href="#cb24-805" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb24-806"><a href="#cb24-806" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_tot =</span> <span class="fu">sum</span>(n_reads_hv),</span>
<span id="cb24-807"><a href="#cb24-807" aria-hidden="true" tabindex="-1"></a>            <span class="at">p_reads_max =</span> <span class="fu">max</span>(p_reads_hv), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb24-808"><a href="#cb24-808" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n_reads_tot))</span>
<span id="cb24-809"><a href="#cb24-809" aria-hidden="true" tabindex="-1"></a>hv_family_collate</span>
<span id="cb24-810"><a href="#cb24-810" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb24-811"><a href="#cb24-811" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-812"><a href="#cb24-812" aria-hidden="true" tabindex="-1"></a><span class="fu">## Relative abundance of pathogenic viruses of interest</span></span>
<span id="cb24-815"><a href="#cb24-815" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb24-816"><a href="#cb24-816" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-composition-exclusion</span></span>
<span id="cb24-817"><a href="#cb24-817" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb24-818"><a href="#cb24-818" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-819"><a href="#cb24-819" aria-hidden="true" tabindex="-1"></a>total <span class="ot">&lt;-</span> basic_stats_raw <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample, n_read_pairs)</span>
<span id="cb24-820"><a href="#cb24-820" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-821"><a href="#cb24-821" aria-hidden="true" tabindex="-1"></a>hv_family_counts <span class="ot">&lt;-</span> hv_reads_family <span class="sc">%&gt;%</span> </span>
<span id="cb24-822"><a href="#cb24-822" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb24-823"><a href="#cb24-823" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-824"><a href="#cb24-824" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(total, <span class="at">by=</span><span class="st">'sample'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-825"><a href="#cb24-825" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb24-826"><a href="#cb24-826" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">ra_reads_hv =</span> n_reads_hv<span class="sc">/</span>n_read_pairs)</span>
<span id="cb24-827"><a href="#cb24-827" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-828"><a href="#cb24-828" aria-hidden="true" tabindex="-1"></a>hv_family_collate <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span></span>
<span id="cb24-829"><a href="#cb24-829" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb24-830"><a href="#cb24-830" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_tot =</span> <span class="fu">sum</span>(n_reads_hv),</span>
<span id="cb24-831"><a href="#cb24-831" aria-hidden="true" tabindex="-1"></a>            <span class="at">ra_reads_tot =</span> <span class="fu">mean</span>(ra_reads_hv), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb24-832"><a href="#cb24-832" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n_reads_tot))</span>
<span id="cb24-833"><a href="#cb24-833" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-834"><a href="#cb24-834" aria-hidden="true" tabindex="-1"></a>hv_family_collate <span class="ot">&lt;-</span> hv_family_collate <span class="sc">%&gt;%</span> </span>
<span id="cb24-835"><a href="#cb24-835" aria-hidden="true" tabindex="-1"></a><span class="fu">rename</span>(</span>
<span id="cb24-836"><a href="#cb24-836" aria-hidden="true" tabindex="-1"></a><span class="st">'family'</span> <span class="ot">=</span> <span class="st">'name'</span>,</span>
<span id="cb24-837"><a href="#cb24-837" aria-hidden="true" tabindex="-1"></a><span class="st">'family_n_reads_tot'</span> <span class="ot">=</span> <span class="st">'n_reads_tot'</span>,</span>
<span id="cb24-838"><a href="#cb24-838" aria-hidden="true" tabindex="-1"></a><span class="st">'family_ra_reads_tot'</span> <span class="ot">=</span> <span class="st">'ra_reads_tot'</span>,</span>
<span id="cb24-839"><a href="#cb24-839" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb24-840"><a href="#cb24-840" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-841"><a href="#cb24-841" aria-hidden="true" tabindex="-1"></a>hv_genus_counts <span class="ot">&lt;-</span> hv_reads_genus <span class="sc">%&gt;%</span> </span>
<span id="cb24-842"><a href="#cb24-842" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb24-843"><a href="#cb24-843" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-844"><a href="#cb24-844" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(total, <span class="at">by=</span><span class="st">'sample'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-845"><a href="#cb24-845" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb24-846"><a href="#cb24-846" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">ra_reads_hv =</span> n_reads_hv<span class="sc">/</span>n_read_pairs)</span>
<span id="cb24-847"><a href="#cb24-847" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-848"><a href="#cb24-848" aria-hidden="true" tabindex="-1"></a>hv_genus_collate <span class="ot">&lt;-</span> hv_genus_counts <span class="sc">%&gt;%</span></span>
<span id="cb24-849"><a href="#cb24-849" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb24-850"><a href="#cb24-850" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_tot =</span> <span class="fu">sum</span>(n_reads_hv),</span>
<span id="cb24-851"><a href="#cb24-851" aria-hidden="true" tabindex="-1"></a>            <span class="at">ra_reads_tot =</span> <span class="fu">mean</span>(ra_reads_hv), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb24-852"><a href="#cb24-852" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n_reads_tot))</span>
<span id="cb24-853"><a href="#cb24-853" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-854"><a href="#cb24-854" aria-hidden="true" tabindex="-1"></a>hv_genus_collate <span class="ot">&lt;-</span> hv_genus_collate <span class="sc">%&gt;%</span> </span>
<span id="cb24-855"><a href="#cb24-855" aria-hidden="true" tabindex="-1"></a><span class="fu">rename</span>(</span>
<span id="cb24-856"><a href="#cb24-856" aria-hidden="true" tabindex="-1"></a><span class="st">'genus'</span> <span class="ot">=</span> <span class="st">'name'</span>,</span>
<span id="cb24-857"><a href="#cb24-857" aria-hidden="true" tabindex="-1"></a><span class="st">'genus_n_reads_tot'</span> <span class="ot">=</span> <span class="st">'n_reads_tot'</span>,</span>
<span id="cb24-858"><a href="#cb24-858" aria-hidden="true" tabindex="-1"></a><span class="st">'genus_ra_reads_tot'</span> <span class="ot">=</span> <span class="st">'ra_reads_tot'</span>,</span>
<span id="cb24-859"><a href="#cb24-859" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb24-860"><a href="#cb24-860" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-861"><a href="#cb24-861" aria-hidden="true" tabindex="-1"></a>hv_species_counts <span class="ot">&lt;-</span> hv_reads_species <span class="sc">%&gt;%</span> </span>
<span id="cb24-862"><a href="#cb24-862" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb24-863"><a href="#cb24-863" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-864"><a href="#cb24-864" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(total, <span class="at">by=</span><span class="st">'sample'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-865"><a href="#cb24-865" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb24-866"><a href="#cb24-866" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">ra_reads_hv =</span> n_reads_hv<span class="sc">/</span>n_read_pairs)</span>
<span id="cb24-867"><a href="#cb24-867" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-868"><a href="#cb24-868" aria-hidden="true" tabindex="-1"></a>hv_species_collate <span class="ot">&lt;-</span> hv_species_counts <span class="sc">%&gt;%</span></span>
<span id="cb24-869"><a href="#cb24-869" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb24-870"><a href="#cb24-870" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_tot =</span> <span class="fu">sum</span>(n_reads_hv),</span>
<span id="cb24-871"><a href="#cb24-871" aria-hidden="true" tabindex="-1"></a>            <span class="at">ra_reads_tot =</span> <span class="fu">mean</span>(ra_reads_hv), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb24-872"><a href="#cb24-872" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n_reads_tot))</span>
<span id="cb24-873"><a href="#cb24-873" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-874"><a href="#cb24-874" aria-hidden="true" tabindex="-1"></a>hv_species_collate <span class="ot">&lt;-</span> hv_species_collate <span class="sc">%&gt;%</span> </span>
<span id="cb24-875"><a href="#cb24-875" aria-hidden="true" tabindex="-1"></a><span class="fu">rename</span>(</span>
<span id="cb24-876"><a href="#cb24-876" aria-hidden="true" tabindex="-1"></a><span class="st">'species'</span> <span class="ot">=</span> <span class="st">'name'</span>,</span>
<span id="cb24-877"><a href="#cb24-877" aria-hidden="true" tabindex="-1"></a><span class="st">'species_n_reads_tot'</span> <span class="ot">=</span> <span class="st">'n_reads_tot'</span>,</span>
<span id="cb24-878"><a href="#cb24-878" aria-hidden="true" tabindex="-1"></a><span class="st">'species_ra_reads_tot'</span> <span class="ot">=</span> <span class="st">'ra_reads_tot'</span>,</span>
<span id="cb24-879"><a href="#cb24-879" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb24-880"><a href="#cb24-880" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-881"><a href="#cb24-881" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb24-882"><a href="#cb24-882" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-885"><a href="#cb24-885" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb24-886"><a href="#cb24-886" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-filter-exclusion</span></span>
<span id="cb24-887"><a href="#cb24-887" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb24-888"><a href="#cb24-888" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-889"><a href="#cb24-889" aria-hidden="true" tabindex="-1"></a><span class="co"># Assuming the data is already loaded into tibbles:</span></span>
<span id="cb24-890"><a href="#cb24-890" aria-hidden="true" tabindex="-1"></a><span class="co">#hv_family_collate</span></span>
<span id="cb24-891"><a href="#cb24-891" aria-hidden="true" tabindex="-1"></a><span class="co">#hv_genus_collate</span></span>
<span id="cb24-892"><a href="#cb24-892" aria-hidden="true" tabindex="-1"></a><span class="co">#hv_species_collate</span></span>
<span id="cb24-893"><a href="#cb24-893" aria-hidden="true" tabindex="-1"></a><span class="co">#viral_taxa</span></span>
<span id="cb24-894"><a href="#cb24-894" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-895"><a href="#cb24-895" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to find parent taxa recursively</span></span>
<span id="cb24-896"><a href="#cb24-896" aria-hidden="true" tabindex="-1"></a>find_parent_taxa <span class="ot">&lt;-</span> <span class="cf">function</span>(taxid, target_ranks, taxa_data) {</span>
<span id="cb24-897"><a href="#cb24-897" aria-hidden="true" tabindex="-1"></a> current_taxid <span class="ot">&lt;-</span> taxid</span>
<span id="cb24-898"><a href="#cb24-898" aria-hidden="true" tabindex="-1"></a>  result <span class="ot">&lt;-</span> <span class="fu">setNames</span>(<span class="fu">vector</span>(<span class="st">"list"</span>, <span class="fu">length</span>(target_ranks)), target_ranks)</span>
<span id="cb24-899"><a href="#cb24-899" aria-hidden="true" tabindex="-1"></a>  <span class="cf">while</span> (<span class="cn">TRUE</span>) {</span>
<span id="cb24-900"><a href="#cb24-900" aria-hidden="true" tabindex="-1"></a>    row <span class="ot">&lt;-</span> taxa_data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(taxid <span class="sc">==</span> current_taxid) <span class="sc">%&gt;%</span> <span class="fu">slice</span>(<span class="dv">1</span>)</span>
<span id="cb24-901"><a href="#cb24-901" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">nrow</span>(row) <span class="sc">==</span> <span class="dv">0</span>) <span class="cf">break</span>  <span class="co"># If no matching taxid is found, break the loop</span></span>
<span id="cb24-902"><a href="#cb24-902" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (row<span class="sc">$</span>rank <span class="sc">%in%</span> target_ranks) {</span>
<span id="cb24-903"><a href="#cb24-903" aria-hidden="true" tabindex="-1"></a>      result[[row<span class="sc">$</span>rank]] <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">taxid =</span> row<span class="sc">$</span>taxid, <span class="at">name =</span> row<span class="sc">$</span>name)</span>
<span id="cb24-904"><a href="#cb24-904" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span> (<span class="fu">all</span>(<span class="sc">!</span><span class="fu">sapply</span>(result, is.null))) {</span>
<span id="cb24-905"><a href="#cb24-905" aria-hidden="true" tabindex="-1"></a>        <span class="cf">break</span>  <span class="co"># If all target ranks are found, break the loop</span></span>
<span id="cb24-906"><a href="#cb24-906" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb24-907"><a href="#cb24-907" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb24-908"><a href="#cb24-908" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (row<span class="sc">$</span>taxid <span class="sc">==</span> row<span class="sc">$</span>parent_taxid) {</span>
<span id="cb24-909"><a href="#cb24-909" aria-hidden="true" tabindex="-1"></a>      <span class="cf">break</span>  <span class="co"># Avoid infinite loop at root</span></span>
<span id="cb24-910"><a href="#cb24-910" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb24-911"><a href="#cb24-911" aria-hidden="true" tabindex="-1"></a>    current_taxid <span class="ot">&lt;-</span> row<span class="sc">$</span>parent_taxid</span>
<span id="cb24-912"><a href="#cb24-912" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb24-913"><a href="#cb24-913" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Replace NULL with NA for any unfound ranks</span></span>
<span id="cb24-914"><a href="#cb24-914" aria-hidden="true" tabindex="-1"></a>  result <span class="ot">&lt;-</span> <span class="fu">lapply</span>(result, <span class="cf">function</span>(x) <span class="cf">if</span> (<span class="fu">is.null</span>(x)) <span class="fu">list</span>(<span class="at">taxid =</span> <span class="cn">NA</span>, <span class="at">name =</span> <span class="cn">NA</span>) <span class="cf">else</span> x)</span>
<span id="cb24-915"><a href="#cb24-915" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(result)</span>
<span id="cb24-916"><a href="#cb24-916" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb24-917"><a href="#cb24-917" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-918"><a href="#cb24-918" aria-hidden="true" tabindex="-1"></a><span class="co"># Find all relevant parent taxa for each species</span></span>
<span id="cb24-919"><a href="#cb24-919" aria-hidden="true" tabindex="-1"></a>result <span class="ot">&lt;-</span> hv_species_collate <span class="sc">%&gt;%</span></span>
<span id="cb24-920"><a href="#cb24-920" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb24-921"><a href="#cb24-921" aria-hidden="true" tabindex="-1"></a>    <span class="at">parent_info =</span> <span class="fu">map</span>(taxid, <span class="sc">~</span><span class="fu">find_parent_taxa</span>(.x, <span class="fu">c</span>(<span class="st">"species"</span>,<span class="st">"subgenus"</span>, <span class="st">"genus"</span>, <span class="st">"subfamily"</span>, <span class="st">"family"</span>), viral_taxa)),</span>
<span id="cb24-922"><a href="#cb24-922" aria-hidden="true" tabindex="-1"></a>    <span class="at">genus_taxid =</span> <span class="fu">map_dbl</span>(parent_info, <span class="sc">~</span>.x<span class="sc">$</span>genus<span class="sc">$</span>taxid),</span>
<span id="cb24-923"><a href="#cb24-923" aria-hidden="true" tabindex="-1"></a>    <span class="at">genus =</span> <span class="fu">map_chr</span>(parent_info, <span class="sc">~</span>.x<span class="sc">$</span>genus<span class="sc">$</span>name),</span>
<span id="cb24-924"><a href="#cb24-924" aria-hidden="true" tabindex="-1"></a>    <span class="at">family_taxid =</span> <span class="fu">map_dbl</span>(parent_info, <span class="sc">~</span>.x<span class="sc">$</span>family<span class="sc">$</span>taxid),</span>
<span id="cb24-925"><a href="#cb24-925" aria-hidden="true" tabindex="-1"></a>    <span class="at">family =</span> <span class="fu">map_chr</span>(parent_info, <span class="sc">~</span>.x<span class="sc">$</span>family<span class="sc">$</span>name)</span>
<span id="cb24-926"><a href="#cb24-926" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb24-927"><a href="#cb24-927" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>parent_info)</span>
<span id="cb24-928"><a href="#cb24-928" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-929"><a href="#cb24-929" aria-hidden="true" tabindex="-1"></a><span class="co"># Join with genus counts</span></span>
<span id="cb24-930"><a href="#cb24-930" aria-hidden="true" tabindex="-1"></a>result <span class="ot">&lt;-</span> result <span class="sc">%&gt;%</span></span>
<span id="cb24-931"><a href="#cb24-931" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(</span>
<span id="cb24-932"><a href="#cb24-932" aria-hidden="true" tabindex="-1"></a>    hv_genus_collate <span class="sc">%&gt;%</span> <span class="fu">select</span>(taxid, genus_n_reads_tot, genus_ra_reads_tot),</span>
<span id="cb24-933"><a href="#cb24-933" aria-hidden="true" tabindex="-1"></a>    <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"genus_taxid"</span> <span class="ot">=</span> <span class="st">"taxid"</span>)</span>
<span id="cb24-934"><a href="#cb24-934" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb24-935"><a href="#cb24-935" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-936"><a href="#cb24-936" aria-hidden="true" tabindex="-1"></a><span class="co"># Join with family counts</span></span>
<span id="cb24-937"><a href="#cb24-937" aria-hidden="true" tabindex="-1"></a>result <span class="ot">&lt;-</span> result <span class="sc">%&gt;%</span></span>
<span id="cb24-938"><a href="#cb24-938" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(</span>
<span id="cb24-939"><a href="#cb24-939" aria-hidden="true" tabindex="-1"></a>    hv_family_collate <span class="sc">%&gt;%</span> <span class="fu">select</span>(taxid, family_n_reads_tot, family_ra_reads_tot),</span>
<span id="cb24-940"><a href="#cb24-940" aria-hidden="true" tabindex="-1"></a>    <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"family_taxid"</span> <span class="ot">=</span> <span class="st">"taxid"</span>)</span>
<span id="cb24-941"><a href="#cb24-941" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb24-942"><a href="#cb24-942" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-943"><a href="#cb24-943" aria-hidden="true" tabindex="-1"></a><span class="co"># Clean up</span></span>
<span id="cb24-944"><a href="#cb24-944" aria-hidden="true" tabindex="-1"></a>result <span class="ot">&lt;-</span> result <span class="sc">%&gt;%</span></span>
<span id="cb24-945"><a href="#cb24-945" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="fu">ends_with</span>(<span class="st">"_taxid"</span>))</span>
<span id="cb24-946"><a href="#cb24-946" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-947"><a href="#cb24-947" aria-hidden="true" tabindex="-1"></a><span class="co">#result</span></span>
<span id="cb24-948"><a href="#cb24-948" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb24-949"><a href="#cb24-949" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-950"><a href="#cb24-950" aria-hidden="true" tabindex="-1"></a>Each dot represents a sample, colored by viral family. The x-axis shows the relative abundance of human-infecting viruses, and the y-axis shows the species.</span>
<span id="cb24-953"><a href="#cb24-953" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb24-954"><a href="#cb24-954" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-pathogenic-viruses</span></span>
<span id="cb24-955"><a href="#cb24-955" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb24-956"><a href="#cb24-956" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 10</span></span>
<span id="cb24-957"><a href="#cb24-957" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb24-958"><a href="#cb24-958" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-959"><a href="#cb24-959" aria-hidden="true" tabindex="-1"></a>species_family <span class="ot">&lt;-</span> result <span class="sc">%&gt;%</span> <span class="fu">select</span>(species, family) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="st">'name'</span> <span class="ot">=</span> <span class="st">'species'</span>)</span>
<span id="cb24-960"><a href="#cb24-960" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-961"><a href="#cb24-961" aria-hidden="true" tabindex="-1"></a>play <span class="ot">&lt;-</span> hv_species_counts <span class="sc">%&gt;%</span> </span>
<span id="cb24-962"><a href="#cb24-962" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb24-963"><a href="#cb24-963" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by =</span> <span class="st">'sample'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-964"><a href="#cb24-964" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(species_family, <span class="at">by =</span> <span class="st">'name'</span>)</span>
<span id="cb24-965"><a href="#cb24-965" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-966"><a href="#cb24-966" aria-hidden="true" tabindex="-1"></a>adjusted_play <span class="ot">&lt;-</span> play <span class="sc">%&gt;%</span> </span>
<span id="cb24-967"><a href="#cb24-967" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span></span>
<span id="cb24-968"><a href="#cb24-968" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">virus_prevalence_num =</span> <span class="fu">n_distinct</span>(sample)<span class="sc">/</span><span class="fu">n_distinct</span>(libraries),</span>
<span id="cb24-969"><a href="#cb24-969" aria-hidden="true" tabindex="-1"></a>         <span class="at">total_reads_hv =</span> <span class="fu">sum</span>(n_reads_hv)) <span class="sc">%&gt;%</span></span>
<span id="cb24-970"><a href="#cb24-970" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb24-971"><a href="#cb24-971" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">fct_reorder</span>(name, virus_prevalence_num, <span class="at">.desc=</span><span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb24-972"><a href="#cb24-972" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, ra_reads_hv, family, virus_prevalence_num, total_reads_hv, n_reads_hv)</span>
<span id="cb24-973"><a href="#cb24-973" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-974"><a href="#cb24-974" aria-hidden="true" tabindex="-1"></a>pal <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">'Dark2'</span>),<span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">'Accent'</span>))</span>
<span id="cb24-975"><a href="#cb24-975" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-976"><a href="#cb24-976" aria-hidden="true" tabindex="-1"></a><span class="co">#, labels = label_log(digits=2)</span></span>
<span id="cb24-977"><a href="#cb24-977" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-978"><a href="#cb24-978" aria-hidden="true" tabindex="-1"></a>ra_dot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(adjusted_play, <span class="fu">aes</span>(<span class="at">x =</span> ra_reads_hv, <span class="at">y=</span>name)) <span class="sc">+</span></span>
<span id="cb24-979"><a href="#cb24-979" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_quasirandom</span>(<span class="at">orientation =</span> <span class="st">'y'</span>, <span class="fu">aes</span>(<span class="at">color =</span> family)) <span class="sc">+</span></span>
<span id="cb24-980"><a href="#cb24-980" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> pal) <span class="sc">+</span> </span>
<span id="cb24-981"><a href="#cb24-981" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_log10</span>(</span>
<span id="cb24-982"><a href="#cb24-982" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Relative abundance human virus reads"</span>,</span>
<span id="cb24-983"><a href="#cb24-983" aria-hidden="true" tabindex="-1"></a>    <span class="at">labels=</span><span class="fu">label_log</span>(<span class="at">digits=</span><span class="dv">3</span>)</span>
<span id="cb24-984"><a href="#cb24-984" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb24-985"><a href="#cb24-985" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span>  <span class="st">""</span>,</span>
<span id="cb24-986"><a href="#cb24-986" aria-hidden="true" tabindex="-1"></a>       <span class="at">color =</span> <span class="st">'Viral family'</span>) <span class="sc">+</span> </span>
<span id="cb24-987"><a href="#cb24-987" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color =</span> <span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">2</span>)) <span class="sc">+</span></span>
<span id="cb24-988"><a href="#cb24-988" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_light</span>() <span class="sc">+</span> </span>
<span id="cb24-989"><a href="#cb24-989" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(</span>
<span id="cb24-990"><a href="#cb24-990" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb24-991"><a href="#cb24-991" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>),</span>
<span id="cb24-992"><a href="#cb24-992" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.ticks.y =</span> <span class="fu">element_blank</span>(),<span class="at">axis.line =</span> <span class="fu">element_line</span>(<span class="at">colour =</span> <span class="st">"black"</span>),</span>
<span id="cb24-993"><a href="#cb24-993" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">14</span>),    </span>
<span id="cb24-994"><a href="#cb24-994" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb24-995"><a href="#cb24-995" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>, <span class="at">face=</span><span class="st">"bold"</span>),</span>
<span id="cb24-996"><a href="#cb24-996" aria-hidden="true" tabindex="-1"></a>    <span class="co">#legend.position = 'bottom',</span></span>
<span id="cb24-997"><a href="#cb24-997" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>),  <span class="co"># Move legend to top right</span></span>
<span id="cb24-998"><a href="#cb24-998" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.justification =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>),  <span class="co"># Align legend to top right</span></span>
<span id="cb24-999"><a href="#cb24-999" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb24-1000"><a href="#cb24-1000" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.border =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb24-1001"><a href="#cb24-1001" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.background =</span> <span class="fu">element_blank</span>())</span>
<span id="cb24-1002"><a href="#cb24-1002" aria-hidden="true" tabindex="-1"></a>ra_dot</span>
<span id="cb24-1003"><a href="#cb24-1003" aria-hidden="true" tabindex="-1"></a><span class="co">#  geom_text(aes(label = total_reads_hv, y = name), </span></span>
<span id="cb24-1004"><a href="#cb24-1004" aria-hidden="true" tabindex="-1"></a><span class="co">#            x = 1e-9, hjust = 0, vjust = 0.5, size = 3, </span></span>
<span id="cb24-1005"><a href="#cb24-1005" aria-hidden="true" tabindex="-1"></a><span class="co">#            check_overlap = TRUE)</span></span>
<span id="cb24-1006"><a href="#cb24-1006" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb24-1007"><a href="#cb24-1007" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-1008"><a href="#cb24-1008" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-1009"><a href="#cb24-1009" aria-hidden="true" tabindex="-1"></a>We see EBV, MPX (most likely contamination), and unclassified Anelloviridae, all of what are viruses we'd expect to see in whole blood.</span>
<span id="cb24-1010"><a href="#cb24-1010" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-1011"><a href="#cb24-1011" aria-hidden="true" tabindex="-1"></a><span class="fu">## Relative abundance assuming perfect human read removal</span></span>
<span id="cb24-1014"><a href="#cb24-1014" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb24-1015"><a href="#cb24-1015" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: perfect-hv-removal</span></span>
<span id="cb24-1016"><a href="#cb24-1016" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb24-1017"><a href="#cb24-1017" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-1018"><a href="#cb24-1018" aria-hidden="true" tabindex="-1"></a>human_reads <span class="ot">&lt;-</span> comp_assigned <span class="sc">%&gt;%</span> <span class="fu">filter</span>(classification <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Human"</span>)) <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample, <span class="at">human_reads=</span>n_reads)</span>
<span id="cb24-1019"><a href="#cb24-1019" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-1020"><a href="#cb24-1020" aria-hidden="true" tabindex="-1"></a>all <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span> </span>
<span id="cb24-1021"><a href="#cb24-1021" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(human_reads, <span class="at">by=</span><span class="st">'sample'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-1022"><a href="#cb24-1022" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb24-1023"><a href="#cb24-1023" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">sum_reads=</span><span class="fu">sum</span>(n_reads_hv), <span class="at">n_read_pairs=</span><span class="fu">first</span>(n_read_pairs) <span class="sc">-</span> <span class="fu">first</span>(human_reads)) <span class="sc">%&gt;%</span></span>
<span id="cb24-1024"><a href="#cb24-1024" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">ra =</span> sum_reads<span class="sc">/</span>n_read_pairs) <span class="sc">%&gt;%</span> </span>
<span id="cb24-1025"><a href="#cb24-1025" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(ra) <span class="sc">%&gt;%</span> <span class="fu">mean</span>()</span>
<span id="cb24-1026"><a href="#cb24-1026" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-1027"><a href="#cb24-1027" aria-hidden="true" tabindex="-1"></a>all</span>
<span id="cb24-1028"><a href="#cb24-1028" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb24-1029"><a href="#cb24-1029" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-1030"><a href="#cb24-1030" aria-hidden="true" tabindex="-1"></a>Assuming we're able to perfectly remove all human reads, the average relative abundance of known human infecting virus is $9.89 \times 10^{-7}$.</span>
<span id="cb24-1031"><a href="#cb24-1031" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-1032"><a href="#cb24-1032" aria-hidden="true" tabindex="-1"></a><span class="fu"># Conclusion</span></span>
<span id="cb24-1033"><a href="#cb24-1033" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-1034"><a href="#cb24-1034" aria-hidden="true" tabindex="-1"></a>There were some takeways from this analysis:</span>
<span id="cb24-1035"><a href="#cb24-1035" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-1036"><a href="#cb24-1036" aria-hidden="true" tabindex="-1"></a><span class="ss">1. </span>Within smaller datasets, we don't expect to find human infecting viruses in samples.</span>
<span id="cb24-1037"><a href="#cb24-1037" aria-hidden="true" tabindex="-1"></a><span class="ss">2. </span>Human reads tend to make up the majority of reads from whole blood.</span>
<span id="cb24-1038"><a href="#cb24-1038" aria-hidden="true" tabindex="-1"></a><span class="ss">3. </span>Anelloviridae doesn't seem to be as abundant in whole blood as it is in plasma.</span>
<span id="cb24-1039"><a href="#cb24-1039" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-1040"><a href="#cb24-1040" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-1041"><a href="#cb24-1041" aria-hidden="true" tabindex="-1"></a><span class="fu"># Appendix</span></span>
<span id="cb24-1042"><a href="#cb24-1042" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-1043"><a href="#cb24-1043" aria-hidden="true" tabindex="-1"></a><span class="fu">## Human-infecting virus families, genera, and species</span></span>
<span id="cb24-1044"><a href="#cb24-1044" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-1045"><a href="#cb24-1045" aria-hidden="true" tabindex="-1"></a>To get a good overview of families, genera, and species, we can look at a Sankey plot where the magnitude of relative abundance, averaged over all samples, is shown in parentheses.</span>
<span id="cb24-1046"><a href="#cb24-1046" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-1049"><a href="#cb24-1049" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb24-1050"><a href="#cb24-1050" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-sankey</span></span>
<span id="cb24-1051"><a href="#cb24-1051" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 22.5</span></span>
<span id="cb24-1052"><a href="#cb24-1052" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb24-1053"><a href="#cb24-1053" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb24-1054"><a href="#cb24-1054" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-1055"><a href="#cb24-1055" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to create links</span></span>
<span id="cb24-1056"><a href="#cb24-1056" aria-hidden="true" tabindex="-1"></a>create_links <span class="ot">&lt;-</span> <span class="cf">function</span>(data) {</span>
<span id="cb24-1057"><a href="#cb24-1057" aria-hidden="true" tabindex="-1"></a>  family_to_genus <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb24-1058"><a href="#cb24-1058" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(genus)) <span class="sc">%&gt;%</span></span>
<span id="cb24-1059"><a href="#cb24-1059" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(family, genus) <span class="sc">%&gt;%</span></span>
<span id="cb24-1060"><a href="#cb24-1060" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">value =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-1061"><a href="#cb24-1061" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">source =</span> family, <span class="at">target =</span> genus)</span>
<span id="cb24-1062"><a href="#cb24-1062" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb24-1063"><a href="#cb24-1063" aria-hidden="true" tabindex="-1"></a>  genus_to_species <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb24-1064"><a href="#cb24-1064" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(genus, species) <span class="sc">%&gt;%</span></span>
<span id="cb24-1065"><a href="#cb24-1065" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">value =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-1066"><a href="#cb24-1066" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">source =</span> genus, <span class="at">target =</span> species)</span>
<span id="cb24-1067"><a href="#cb24-1067" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-1068"><a href="#cb24-1068" aria-hidden="true" tabindex="-1"></a>  family_to_species <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb24-1069"><a href="#cb24-1069" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="fu">is.na</span>(genus)) <span class="sc">%&gt;%</span></span>
<span id="cb24-1070"><a href="#cb24-1070" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(family, species) <span class="sc">%&gt;%</span></span>
<span id="cb24-1071"><a href="#cb24-1071" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">value =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-1072"><a href="#cb24-1072" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">source =</span> family, <span class="at">target =</span> species)</span>
<span id="cb24-1073"><a href="#cb24-1073" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-1074"><a href="#cb24-1074" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_rows</span>(family_to_genus, genus_to_species, family_to_species) <span class="sc">%&gt;%</span></span>
<span id="cb24-1075"><a href="#cb24-1075" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(source))</span>
<span id="cb24-1076"><a href="#cb24-1076" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb24-1077"><a href="#cb24-1077" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-1078"><a href="#cb24-1078" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to create nodes</span></span>
<span id="cb24-1079"><a href="#cb24-1079" aria-hidden="true" tabindex="-1"></a>create_nodes <span class="ot">&lt;-</span> <span class="cf">function</span>(links) {</span>
<span id="cb24-1080"><a href="#cb24-1080" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>(</span>
<span id="cb24-1081"><a href="#cb24-1081" aria-hidden="true" tabindex="-1"></a>    <span class="at">name =</span> <span class="fu">c</span>(links<span class="sc">$</span>source, links<span class="sc">$</span>target) <span class="sc">%&gt;%</span> <span class="fu">unique</span>()</span>
<span id="cb24-1082"><a href="#cb24-1082" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb24-1083"><a href="#cb24-1083" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb24-1084"><a href="#cb24-1084" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-1085"><a href="#cb24-1085" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to prepare data for Sankey diagram</span></span>
<span id="cb24-1086"><a href="#cb24-1086" aria-hidden="true" tabindex="-1"></a>prepare_sankey_data <span class="ot">&lt;-</span> <span class="cf">function</span>(links, nodes) {</span>
<span id="cb24-1087"><a href="#cb24-1087" aria-hidden="true" tabindex="-1"></a>  links<span class="sc">$</span>IDsource <span class="ot">&lt;-</span> <span class="fu">match</span>(links<span class="sc">$</span>source, nodes<span class="sc">$</span>name) <span class="sc">-</span> <span class="dv">1</span></span>
<span id="cb24-1088"><a href="#cb24-1088" aria-hidden="true" tabindex="-1"></a>  links<span class="sc">$</span>IDtarget <span class="ot">&lt;-</span> <span class="fu">match</span>(links<span class="sc">$</span>target, nodes<span class="sc">$</span>name) <span class="sc">-</span> <span class="dv">1</span></span>
<span id="cb24-1089"><a href="#cb24-1089" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list</span>(<span class="at">links =</span> links, <span class="at">nodes =</span> nodes)</span>
<span id="cb24-1090"><a href="#cb24-1090" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb24-1091"><a href="#cb24-1091" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-1092"><a href="#cb24-1092" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to create Sankey plot</span></span>
<span id="cb24-1093"><a href="#cb24-1093" aria-hidden="true" tabindex="-1"></a>create_sankey_plot <span class="ot">&lt;-</span> <span class="cf">function</span>(sankey_data) {</span>
<span id="cb24-1094"><a href="#cb24-1094" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sankeyNetwork</span>(</span>
<span id="cb24-1095"><a href="#cb24-1095" aria-hidden="true" tabindex="-1"></a>    <span class="at">Links =</span> sankey_data<span class="sc">$</span>links, </span>
<span id="cb24-1096"><a href="#cb24-1096" aria-hidden="true" tabindex="-1"></a>    <span class="at">Nodes =</span> sankey_data<span class="sc">$</span>nodes,</span>
<span id="cb24-1097"><a href="#cb24-1097" aria-hidden="true" tabindex="-1"></a>    <span class="at">Source =</span> <span class="st">"IDsource"</span>, </span>
<span id="cb24-1098"><a href="#cb24-1098" aria-hidden="true" tabindex="-1"></a>    <span class="at">Target =</span> <span class="st">"IDtarget"</span>,</span>
<span id="cb24-1099"><a href="#cb24-1099" aria-hidden="true" tabindex="-1"></a>    <span class="at">Value =</span> <span class="st">"value"</span>, </span>
<span id="cb24-1100"><a href="#cb24-1100" aria-hidden="true" tabindex="-1"></a>    <span class="at">NodeID =</span> <span class="st">"name"</span>,</span>
<span id="cb24-1101"><a href="#cb24-1101" aria-hidden="true" tabindex="-1"></a>    <span class="at">sinksRight =</span> <span class="cn">TRUE</span>,</span>
<span id="cb24-1102"><a href="#cb24-1102" aria-hidden="true" tabindex="-1"></a>    <span class="at">nodeWidth =</span> <span class="dv">25</span>,</span>
<span id="cb24-1103"><a href="#cb24-1103" aria-hidden="true" tabindex="-1"></a>    <span class="at">fontSize =</span> <span class="dv">14</span>,</span>
<span id="cb24-1104"><a href="#cb24-1104" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb24-1105"><a href="#cb24-1105" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb24-1106"><a href="#cb24-1106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-1107"><a href="#cb24-1107" aria-hidden="true" tabindex="-1"></a>save_sankey_as_png <span class="ot">&lt;-</span> <span class="cf">function</span>(sankey_plot, <span class="at">width =</span> <span class="dv">1000</span>, <span class="at">height =</span> <span class="dv">800</span>) {</span>
<span id="cb24-1108"><a href="#cb24-1108" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Save the plot as an HTML file</span></span>
<span id="cb24-1109"><a href="#cb24-1109" aria-hidden="true" tabindex="-1"></a>  <span class="fu">saveWidget</span>(sankey_plot, <span class="fu">sprintf</span>(<span class="st">'%s/sankey.html'</span>,data_dir))</span>
<span id="cb24-1110"><a href="#cb24-1110" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb24-1111"><a href="#cb24-1111" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-1112"><a href="#cb24-1112" aria-hidden="true" tabindex="-1"></a>format_scientific <span class="ot">&lt;-</span> <span class="cf">function</span>(x, <span class="at">digits=</span><span class="dv">2</span>) {</span>
<span id="cb24-1113"><a href="#cb24-1113" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sapply</span>(x, <span class="cf">function</span>(val) {</span>
<span id="cb24-1114"><a href="#cb24-1114" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">is.na</span>(val) <span class="sc">||</span> <span class="fu">abs</span>(val) <span class="sc">&lt;</span> <span class="fl">1e-15</span>) {</span>
<span id="cb24-1115"><a href="#cb24-1115" aria-hidden="true" tabindex="-1"></a>      <span class="fu">return</span>(<span class="st">"0"</span>)</span>
<span id="cb24-1116"><a href="#cb24-1116" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb24-1117"><a href="#cb24-1117" aria-hidden="true" tabindex="-1"></a>      exponent <span class="ot">&lt;-</span> <span class="fu">floor</span>(<span class="fu">log10</span>(<span class="fu">abs</span>(val)))</span>
<span id="cb24-1118"><a href="#cb24-1118" aria-hidden="true" tabindex="-1"></a>      coef <span class="ot">&lt;-</span> val <span class="sc">/</span> <span class="dv">10</span><span class="sc">^</span>exponent</span>
<span id="cb24-1119"><a href="#cb24-1119" aria-hidden="true" tabindex="-1"></a>      <span class="co">#return(sprintf("%.1f × 10^%d", round(coef, digits), exponent))</span></span>
<span id="cb24-1120"><a href="#cb24-1120" aria-hidden="true" tabindex="-1"></a>      <span class="co"># May or may not be smart, just keeping magnitude</span></span>
<span id="cb24-1121"><a href="#cb24-1121" aria-hidden="true" tabindex="-1"></a>      <span class="fu">return</span>(<span class="fu">sprintf</span>(<span class="st">"10^%d"</span>, exponent))</span>
<span id="cb24-1122"><a href="#cb24-1122" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb24-1123"><a href="#cb24-1123" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb24-1124"><a href="#cb24-1124" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb24-1125"><a href="#cb24-1125" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-1126"><a href="#cb24-1126" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> result <span class="sc">%&gt;%</span> </span>
<span id="cb24-1127"><a href="#cb24-1127" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">c</span>(genus_n_reads_tot, genus_ra_reads_tot), <span class="sc">~</span><span class="fu">replace_na</span>(., <span class="dv">0</span>)),</span>
<span id="cb24-1128"><a href="#cb24-1128" aria-hidden="true" tabindex="-1"></a>         <span class="at">genus =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(genus), <span class="st">"Unknown Genus"</span>, genus)) <span class="sc">%&gt;%</span></span>
<span id="cb24-1129"><a href="#cb24-1129" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb24-1130"><a href="#cb24-1130" aria-hidden="true" tabindex="-1"></a>  <span class="at">species =</span> <span class="fu">paste0</span>(species, <span class="fu">sprintf</span>(<span class="st">' (%s)'</span>, <span class="fu">format_scientific</span>(species_ra_reads_tot))),</span>
<span id="cb24-1131"><a href="#cb24-1131" aria-hidden="true" tabindex="-1"></a>  <span class="at">genus =</span> <span class="fu">paste0</span>(genus, <span class="fu">sprintf</span>(<span class="st">' (%s)'</span>, <span class="fu">format_scientific</span>(genus_ra_reads_tot))),</span>
<span id="cb24-1132"><a href="#cb24-1132" aria-hidden="true" tabindex="-1"></a>  <span class="at">family =</span> <span class="fu">paste0</span>(family, <span class="fu">sprintf</span>(<span class="st">' (%s)'</span>, <span class="fu">format_scientific</span>(family_ra_reads_tot)))</span>
<span id="cb24-1133"><a href="#cb24-1133" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb24-1134"><a href="#cb24-1134" aria-hidden="true" tabindex="-1"></a>links <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">create_links</span>(data))</span>
<span id="cb24-1135"><a href="#cb24-1135" aria-hidden="true" tabindex="-1"></a>nodes <span class="ot">&lt;-</span> <span class="fu">create_nodes</span>(links)</span>
<span id="cb24-1136"><a href="#cb24-1136" aria-hidden="true" tabindex="-1"></a>sankey_data <span class="ot">&lt;-</span> <span class="fu">prepare_sankey_data</span>(links, nodes)</span>
<span id="cb24-1137"><a href="#cb24-1137" aria-hidden="true" tabindex="-1"></a>sankey <span class="ot">&lt;-</span> <span class="fu">create_sankey_plot</span>(sankey_data)</span>
<span id="cb24-1138"><a href="#cb24-1138" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-1139"><a href="#cb24-1139" aria-hidden="true" tabindex="-1"></a>sankey</span>
<span id="cb24-1140"><a href="#cb24-1140" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->




</body></html>