<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.53">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<meta name="author" content="Harmon Bhasin">
<meta name="dcterms.date" content="2024-07-08">
<title>Workflow of Cebria-Mendoza et al.&nbsp;(2021) – Harmon's Public Notebook</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>

<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light"><script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><script async="" src="https://hypothes.is/embed.js"></script><script>
  window.document.addEventListener("DOMContentLoaded", function (_event) {
    document.body.classList.add('hypothesis-enabled');
  });
</script><style>

      .quarto-title-block .quarto-title-banner {
        background: #de2d26;
      }
</style>
<script src="../site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script><script src="../site_libs/d3-4.5.0/d3.min.js"></script><script src="../site_libs/sankey-1/sankey.js"></script><script src="../site_libs/sankeyNetwork-binding-0.4/sankeyNetwork.js"></script><link href="../site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet">
<script src="../site_libs/pagedtable-1.1/js/pagedtable.js"></script><script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script><script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script><script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script><link rel="stylesheet" href="../styles.css">
</head>
<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top quarto-banner"><nav class="navbar navbar-expand-lg " data-bs-theme="dark"><div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">Harmon’s Public Notebook</span>
    </a>
  </div>
        <div class="quarto-navbar-tools tools-end">
</div>
          <div id="quarto-search" class="" title="Search"></div>
      </div> <!-- /container-fluid -->
    </nav></header><!-- content --><header id="title-block-header" class="quarto-title-block default page-columns page-full"><div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-page-left">
      <div class="quarto-title-block"><div><h1 class="title">Workflow of Cebria-Mendoza et al.&nbsp;(2021)</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
            <p class="subtitle lead">Pooled plasma from Spain (DNA + RNA)</p>
                      </div>
  </div>
    
  
  <div class="quarto-title-meta column-page-left">

      <div>
      <div class="quarto-title-meta-heading">Author</div>
      <div class="quarto-title-meta-contents">
               <p>Harmon Bhasin </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">July 8, 2024</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active"><h2 id="toc-title">Table of contents</h2>
   
  <ul>
<li>
<a href="#the-raw-data" id="toc-the-raw-data" class="nav-link active" data-scroll-target="#the-raw-data"><span class="header-section-number">1</span> The raw data</a>
  <ul class="collapse">
<li>
<a href="#about" id="toc-about" class="nav-link" data-scroll-target="#about"><span class="header-section-number">1.1</span> About</a>
  <ul class="collapse">
<li><a href="#sample-library-preparation" id="toc-sample-library-preparation" class="nav-link" data-scroll-target="#sample-library-preparation"><span class="header-section-number">1.1.1</span> Sample + library preparation</a></li>
  </ul>
</li>
  <li><a href="#quality-control" id="toc-quality-control" class="nav-link" data-scroll-target="#quality-control"><span class="header-section-number">1.2</span> Quality control</a></li>
  </ul>
</li>
  <li>
<a href="#preprocessing" id="toc-preprocessing" class="nav-link" data-scroll-target="#preprocessing"><span class="header-section-number">2</span> Preprocessing</a>
  <ul class="collapse">
<li><a href="#high-level-metrics" id="toc-high-level-metrics" class="nav-link" data-scroll-target="#high-level-metrics"><span class="header-section-number">2.1</span> High-level metrics</a></li>
  </ul>
</li>
  <li>
<a href="#taxonomic-composition" id="toc-taxonomic-composition" class="nav-link" data-scroll-target="#taxonomic-composition"><span class="header-section-number">3</span> Taxonomic composition</a>
  <ul class="collapse">
<li><a href="#high-level-composition" id="toc-high-level-composition" class="nav-link" data-scroll-target="#high-level-composition"><span class="header-section-number">3.1</span> High-level composition</a></li>
  <li><a href="#total-viral-content" id="toc-total-viral-content" class="nav-link" data-scroll-target="#total-viral-content"><span class="header-section-number">3.2</span> Total viral content</a></li>
  <li><a href="#taxonomic-composition-of-viruses" id="toc-taxonomic-composition-of-viruses" class="nav-link" data-scroll-target="#taxonomic-composition-of-viruses"><span class="header-section-number">3.3</span> Taxonomic composition of viruses</a></li>
  </ul>
</li>
  <li>
<a href="#human-infecting-virus-reads" id="toc-human-infecting-virus-reads" class="nav-link" data-scroll-target="#human-infecting-virus-reads"><span class="header-section-number">4</span> Human-infecting virus reads</a>
  <ul class="collapse">
<li><a href="#overall-relative-abundance" id="toc-overall-relative-abundance" class="nav-link" data-scroll-target="#overall-relative-abundance"><span class="header-section-number">4.1</span> Overall relative abundance</a></li>
  <li><a href="#overall-taxonomy-and-composition" id="toc-overall-taxonomy-and-composition" class="nav-link" data-scroll-target="#overall-taxonomy-and-composition"><span class="header-section-number">4.2</span> Overall taxonomy and composition</a></li>
  <li>
<a href="#analyzing-specific-families" id="toc-analyzing-specific-families" class="nav-link" data-scroll-target="#analyzing-specific-families"><span class="header-section-number">4.3</span> Analyzing specific families</a>
  <ul class="collapse">
<li><a href="#anelloviridae-number-of-reads-8323297" id="toc-anelloviridae-number-of-reads-8323297" class="nav-link" data-scroll-target="#anelloviridae-number-of-reads-8323297"><span class="header-section-number">4.3.1</span> Anelloviridae (Number of reads: 8,323,297)</a></li>
  <li><a href="#rhabdoviridae-number-of-reads-935461-spike-in" id="toc-rhabdoviridae-number-of-reads-935461-spike-in" class="nav-link" data-scroll-target="#rhabdoviridae-number-of-reads-935461-spike-in"><span class="header-section-number">4.3.2</span> Rhabdoviridae (Number of reads: 935,461) [SPIKE-IN]</a></li>
  <li><a href="#flaviviridae-number-of-reads-94741" id="toc-flaviviridae-number-of-reads-94741" class="nav-link" data-scroll-target="#flaviviridae-number-of-reads-94741"><span class="header-section-number">4.3.3</span> Flaviviridae (Number of reads: 94,741)</a></li>
  <li><a href="#microviridae-number-of-reads-10841-spike-in" id="toc-microviridae-number-of-reads-10841-spike-in" class="nav-link" data-scroll-target="#microviridae-number-of-reads-10841-spike-in"><span class="header-section-number">4.3.4</span> Microviridae (Number of reads: 10,841) [SPIKE-IN]</a></li>
  </ul>
</li>
  <li><a href="#relative-abundance-of-pathogenic-viruses-of-interest" id="toc-relative-abundance-of-pathogenic-viruses-of-interest" class="nav-link" data-scroll-target="#relative-abundance-of-pathogenic-viruses-of-interest"><span class="header-section-number">4.4</span> Relative abundance of pathogenic viruses of interest</a></li>
  <li><a href="#relative-abundance-assuming-perfect-human-read-removal" id="toc-relative-abundance-assuming-perfect-human-read-removal" class="nav-link" data-scroll-target="#relative-abundance-assuming-perfect-human-read-removal"><span class="header-section-number">4.5</span> Relative abundance assuming perfect human read removal</a></li>
  </ul>
</li>
  <li><a href="#conclusion" id="toc-conclusion" class="nav-link" data-scroll-target="#conclusion"><span class="header-section-number">5</span> Conclusion</a></li>
  <li>
<a href="#appendix" id="toc-appendix" class="nav-link" data-scroll-target="#appendix"><span class="header-section-number">6</span> Appendix</a>
  <ul class="collapse">
<li><a href="#human-infecting-virus-families-genera-and-species" id="toc-human-infecting-virus-families-genera-and-species" class="nav-link" data-scroll-target="#human-infecting-virus-families-genera-and-species"><span class="header-section-number">6.1</span> Human-infecting virus families, genera, and species</a></li>
  </ul>
</li>
  </ul><div class="quarto-other-links"><h2>Other Links</h2><ul><li><a href="https://doi.org/10.3390/v15071425"><i class="bi bi-link-45deg"></i>Paper</a></li><li><a href="https://www.ncbi.nlm.nih.gov/bioproject/?term=PRJNA983534"><i class="bi bi-link-45deg"></i>Data</a></li></ul></div><div class="quarto-code-links"><h2>Code Links</h2><ul><li><a href="https://github.com/naobservatory/harmons-public-notebook/blob/main/notebooks/2024-07-22-thijssen.qmd"><i class="bi bi-file-code"></i>Code for this post</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content quarto-banner-title-block column-page-left" id="quarto-document-content"><p>As a part of my time here, I’m exploring blood-based surveillance for novel viral pathogen detection. We’ve decided to make three blog posts. The first blog post will cover the physical and biological characteristics of blood, the second blog post will cover the blood sampling strategies, and the last blog post will look at the metagenomic profile of blood (which is what we’re doing here, and will do for the next few posts).</p>
<p>This is the first of many studies that I’ll be analyzing. however I don’t expect all of them to be used in the last blog post. In this post, I analyze <a href="https://doi.org/10.3390/v13112322">Cebria-Mendoza 2021</a>, a dataset with 60 samples where each sample is a pool of 8-13 unique healthy donors, with a total of ~600 donors from Spain.</p>
<p><em>This notebook uses the MGS Workflow v2.2.1 (note that this is old).</em></p>
<section id="the-raw-data" class="level1" data-number="1"><h1 data-number="1">
<span class="header-section-number">1</span> The raw data</h1>
<section id="about" class="level2" data-number="1.1"><h2 data-number="1.1" class="anchored" data-anchor-id="about">
<span class="header-section-number">1.1</span> About</h2>
<p><a href="https://www.ncbi.nlm.nih.gov/bioproject/?term=PRJNA983534">This dataset</a> comprises 60 samples from plasma pools of 8-13 people from Spain. In total, 567 healthy individuals contributed to these pools. For each pooled sample, a combined DNA and RNA library preparation was performed, resulting in a single sequencing output that captures both nucleic acid types. This approach provides comprehensive genetic information but precludes separate analysis of DNA and RNA from individual samples.</p>
<section id="sample-library-preparation" class="level3" data-number="1.1.1"><h3 data-number="1.1.1" class="anchored" data-anchor-id="sample-library-preparation">
<span class="header-section-number">1.1.1</span> Sample + library preparation</h3>
<p>Paraphrased from the <a href="https://doi.org/10.3390/v15071425">paper</a>:</p>
<blockquote class="blockquote">
<p>A total of 587 plasma samples from healthy donors were collected from Valencia, Spain from 15 September 2018 to 30 March 2019 and stored at −80 °C until use.</p>
<p>Each of the 60 pools (SP1-SP60) was obtained by mixing 1 mL of plasma from a variable number of donors (between 8- and 13-mL total). To assess viral recovery, each pool was spiked with 103 PFU of ϕX174 [Microviridae] and 104 PFU of vesicular stomatitis virus (VSV) [Rhabdoviridae]. Briefly, plasma pools were processed with 1.0 µM filters to remove cells and other non-viral particles and the filtered fractions were subject to high-speed centrifugation (87,000 g, 2 h, 4 °C), washed with PBS 1X (87,000 g, 1 h, 4 °C), and resuspended in 245 µL 1X digestion buffer. Then, 5 µL of Turbo DNase, 2 µL of Benzonase and 2 µL of micrococcal nuclease (NEB) were added to the sample to remove unprotected nucleic acids. After incubation (1 h, 37 °C), 20 µL of stop reagent was added, following the manufacturer’s instructions. Then, 240 µL supernatant was transferred to a new tube and split into two fractions: 200 µL fraction was used for RNA extraction using TRIzol LS reagent, followed by purification with the QIAamp Viral RNA Mini kit and amplification with the QuantiTect Whole Transcriptome kit (Qiagen), and 40 µL fraction was used for DNA extraction with the QIAamp Viral RNA Mini kit and amplification with the TruePrime WGA kit.</p>
<p>For each pool, DNA and RNA amplification products were mixed in equimolar concentration before library preparation, which was carried out using Nextera XT DNA library preparation kit with 15 amplification cycles (Illumina, San Diego, USA), and subject to pair-end sequencing in a NextSeq device.</p>
</blockquote>
<p><em>More details can be found in their <a href="https://doi.org/10.1038%2Fs41598-021-86427-4">original paper</a> which outlined this protocol.</em></p>
</section></section><section id="quality-control" class="level2" data-number="1.2"><h2 data-number="1.2" class="anchored" data-anchor-id="quality-control">
<span class="header-section-number">1.2</span> Quality control</h2>
<p>In total, these 60 samples contained 230M read pairs. The samples had 2.3M - 4.8M (mean 3.8M) read pairs each.</p>
<p>The number of read pairs and total base pairs looks good, however the duplication rate is quite high (this can be attributed to their sample preparation). Adapter content is low (the upper limit on the plot is 1%). Read quality seems good both over all the positions as well as over the number of sequences.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb1"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Prepare data</span></span>
<span><span class="va">basic_stats_raw_metrics</span> <span class="op">&lt;-</span> <span class="va">basic_stats_raw</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">library</span>,</span>
<span>         <span class="va">pool_size</span>,</span>
<span>         `# Read pairs` <span class="op">=</span> <span class="va">n_read_pairs</span>,</span>
<span>         `Total base pairs\n(approx)` <span class="op">=</span> <span class="va">n_bases_approx</span>,</span>
<span>         `% Duplicates\n(FASTQC)` <span class="op">=</span> <span class="va">percent_duplicates</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_longer</span><span class="op">(</span><span class="op">-</span><span class="op">(</span><span class="va">library</span><span class="op">:</span><span class="va">pool_size</span><span class="op">)</span>, names_to <span class="op">=</span> <span class="st">"metric"</span>, values_to <span class="op">=</span> <span class="st">"value"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>metric <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">metric</span><span class="op">)</span><span class="op">)</span> </span>
<span></span>
<span><span class="co"># Set up plot templates</span></span>
<span></span>
<span><span class="va">g_basic</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">basic_stats_raw_metrics</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">library</span>, y <span class="op">=</span> <span class="va">value</span>, fill <span class="op">=</span> <span class="va">pool_size</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>position <span class="op">=</span> <span class="st">"dodge"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_discrete</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">expand_limits</span><span class="op">(</span>y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">100</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Accent"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">metric</span> <span class="op">~</span> <span class="va">.</span>, scales <span class="op">=</span> <span class="st">"free"</span>, space <span class="op">=</span> <span class="st">"free_x"</span>, switch <span class="op">=</span> <span class="st">"y"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span></span>
<span>    axis.title.y <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    strip.text.y <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>face <span class="op">=</span> <span class="st">"plain"</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span><span class="va">g_basic</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-08_cebria-mendoza_files/figure-html/plot-basic-stats-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb2"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Set up plotting templates</span></span>
<span><span class="va">g_qual_raw</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span>mapping<span class="op">=</span><span class="fu">aes</span><span class="op">(</span>linetype<span class="op">=</span><span class="va">read_pair</span>, group<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/interaction.html">interaction</a></span><span class="op">(</span><span class="va">sample</span>,<span class="va">read_pair</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">scale_linetype_discrete</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Read Pair"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>color<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">2</span>,byrow<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>         linetype <span class="op">=</span> <span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">2</span>,byrow<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span></span>
<span></span>
<span><span class="co"># Visualize adapters</span></span>
<span><span class="va">g_adapters_raw</span> <span class="op">&lt;-</span> <span class="va">g_qual_raw</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">position</span>, y<span class="op">=</span><span class="va">pc_adapters</span><span class="op">)</span>, data<span class="op">=</span><span class="va">adapter_stats_raw</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% Adapters"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Position"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span>,</span>
<span>                     breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">500</span>,<span class="fl">20</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">.</span><span class="op">~</span><span class="va">adapter</span><span class="op">)</span></span>
<span><span class="va">g_adapters_raw</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-08_cebria-mendoza_files/figure-html/plot-raw-quality-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb3"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Visualize quality</span></span>
<span><span class="va">g_quality_base_raw</span> <span class="op">&lt;-</span> <span class="va">g_qual_raw</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept<span class="op">=</span><span class="fl">25</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept<span class="op">=</span><span class="fl">30</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">position</span>, y<span class="op">=</span><span class="va">mean_phred_score</span><span class="op">)</span>, data<span class="op">=</span><span class="va">quality_base_stats_raw</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Mean Phred score"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">10</span>,<span class="fl">45</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Position"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span>,</span>
<span>                     breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">500</span>,<span class="fl">20</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_quality_base_raw</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-08_cebria-mendoza_files/figure-html/plot-raw-quality-2.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb4"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_quality_seq_raw</span> <span class="op">&lt;-</span> <span class="va">g_qual_raw</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">25</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">30</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">mean_phred_score</span>, y<span class="op">=</span><span class="va">n_sequences</span><span class="op">)</span>, data<span class="op">=</span><span class="va">quality_seq_stats_raw</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Mean Phred score"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"# Sequences"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_quality_seq_raw</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-08_cebria-mendoza_files/figure-html/plot-raw-quality-3.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section></section><section id="preprocessing" class="level1" data-number="2"><h1 data-number="2">
<span class="header-section-number">2</span> Preprocessing</h1>
<section id="high-level-metrics" class="level2" data-number="2.1"><h2 data-number="2.1" class="anchored" data-anchor-id="high-level-metrics">
<span class="header-section-number">2.1</span> High-level metrics</h2>
<p>The average fraction of reads at each stage in the preprocessing pipeline is shown in the following table. On average, cleaning &amp; deduplication removed about 57% of total read pairs, primarily during duplication which makes sense given that the raw metrics show that the samples are quite high in duplication. Ribodepletion removed about 6-8% during each round.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb5"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># TODO: Group by pool size as well</span></span>
<span><span class="co"># Count read losses</span></span>
<span><span class="va">n_reads_rel</span> <span class="op">&lt;-</span> <span class="va">basic_stats</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">pool_size</span>, <span class="va">stage</span>, <span class="va">percent_duplicates</span>, <span class="va">n_read_pairs</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">pool_size</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">sample</span>, <span class="va">stage</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_retained <span class="op">=</span> <span class="va">n_read_pairs</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/stats/lag.html">lag</a></span><span class="op">(</span><span class="va">n_read_pairs</span><span class="op">)</span>,</span>
<span>         p_reads_lost <span class="op">=</span> <span class="fl">1</span> <span class="op">-</span> <span class="va">p_reads_retained</span>,</span>
<span>         p_reads_retained_abs <span class="op">=</span> <span class="va">n_read_pairs</span> <span class="op">/</span> <span class="va">n_read_pairs</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,</span>
<span>         p_reads_lost_abs <span class="op">=</span> <span class="fl">1</span><span class="op">-</span><span class="va">p_reads_retained_abs</span>,</span>
<span>         p_reads_lost_abs_marginal <span class="op">=</span> <span class="va">p_reads_lost_abs</span> <span class="op">-</span> <span class="fu"><a href="https://rdrr.io/r/stats/lag.html">lag</a></span><span class="op">(</span><span class="va">p_reads_lost_abs</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">n_reads_rel_display</span> <span class="op">&lt;-</span> <span class="va">n_reads_rel</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">rename</span><span class="op">(</span>Stage<span class="op">=</span><span class="va">stage</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">Stage</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>`% Total Reads Lost (Cumulative)` <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">p_reads_lost_abs</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span>,<span class="fl">1</span><span class="op">)</span>, <span class="st">"-"</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_lost_abs</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span>,<span class="fl">1</span><span class="op">)</span>, <span class="st">" (mean "</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">p_reads_lost_abs</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span>,<span class="fl">1</span><span class="op">)</span>, <span class="st">")"</span><span class="op">)</span>,</span>
<span>            `% Total Reads Lost (Marginal)` <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">p_reads_lost_abs_marginal</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span>,<span class="fl">1</span><span class="op">)</span>, <span class="st">"-"</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_lost_abs_marginal</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span>,<span class="fl">1</span><span class="op">)</span>, <span class="st">" (mean "</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">p_reads_lost_abs_marginal</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span>,<span class="fl">1</span><span class="op">)</span>, <span class="st">")"</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">Stage</span> <span class="op">!=</span> <span class="st">"raw_concat"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>Stage <span class="op">=</span> <span class="va">Stage</span> <span class="op">%&gt;%</span> <span class="va">as.numeric</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span>labels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Trimming &amp; filtering"</span>, <span class="st">"Deduplication"</span>, <span class="st">"Initial ribodepletion"</span>, <span class="st">"Secondary ribodepletion"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">n_reads_rel_display</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["Stage"],"name":[1],"type":["fct"],"align":["left"]},{"label":["% Total Reads Lost (Cumulative)"],"name":[2],"type":["chr"],"align":["left"]},{"label":["% Total Reads Lost (Marginal)"],"name":[3],"type":["chr"],"align":["left"]}],"data":[{"1":"Trimming & filtering","2":"0-5 (mean 1.4)","3":"0-5 (mean 1.4)"},{"1":"Deduplication","2":"44.2-70.6 (mean 55.9)","3":"44.2-67.8 (mean 54.5)"},{"1":"Initial ribodepletion","2":"53.9-77.2 (mean 63.8)","3":"3.2-15.6 (mean 7.9)"},{"1":"Secondary ribodepletion","2":"57.7-85.7 (mean 70.2)","3":"2.2-12.6 (mean 6.4)"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<p>Adapter content stays low (it’s quite strange that FASTQC shows there being library contamination post ribosomal depletion and not anytime before that, but considering how small it is, I’m inclined to ignore it). Read quality stays pretty similar over the positions, but improves over the number of sequences (predominantly from cleaning) throughout the pipeline. Duplication rates are still quite high which is a bit concerning. Mean read length stays close to expected 150. The % of ribosomal reads vary a good amount, some samples had anywhere as low as 10% ribosomal reads whereas others had up to 60% ribosomal reads. This is probably an indication that I need to do more ribosomal read removal, but I’m going to hold off on doing this.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb6"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_qual</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span>mapping<span class="op">=</span><span class="fu">aes</span><span class="op">(</span>linetype<span class="op">=</span><span class="va">read_pair</span>, group<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/interaction.html">interaction</a></span><span class="op">(</span><span class="va">sample</span>,<span class="va">read_pair</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">scale_linetype_discrete</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Read Pair"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>color<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">2</span>,byrow<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>         linetype <span class="op">=</span> <span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">2</span>,byrow<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span></span>
<span></span>
<span><span class="co"># Visualize adapters</span></span>
<span><span class="va">g_adapters</span> <span class="op">&lt;-</span> <span class="va">g_qual</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">position</span>, y<span class="op">=</span><span class="va">pc_adapters</span><span class="op">)</span>, data<span class="op">=</span><span class="va">adapter_stats</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% Adapters"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span><span class="op">)</span>,</span>
<span>                     , expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Position"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span>,</span>
<span>                     breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">140</span>,<span class="fl">20</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">stage</span><span class="op">~</span><span class="va">adapter</span><span class="op">)</span></span>
<span><span class="va">g_adapters</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-08_cebria-mendoza_files/figure-html/plot-quality-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb7"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Visualize quality</span></span>
<span><span class="va">g_quality_base</span> <span class="op">&lt;-</span> <span class="va">g_qual</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept<span class="op">=</span><span class="fl">25</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept<span class="op">=</span><span class="fl">30</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">position</span>, y<span class="op">=</span><span class="va">mean_phred_score</span><span class="op">)</span>, data<span class="op">=</span><span class="va">quality_base_stats</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Mean Phred score"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">10</span>,<span class="fl">45</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Position"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span>,</span>
<span>                     breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">140</span>,<span class="fl">20</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">stage</span><span class="op">~</span><span class="va">.</span><span class="op">)</span></span>
<span><span class="va">g_quality_base</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-08_cebria-mendoza_files/figure-html/plot-quality-2.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb8"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_quality_seq</span> <span class="op">&lt;-</span> <span class="va">g_qual</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">25</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">30</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">mean_phred_score</span>, y<span class="op">=</span><span class="va">n_sequences</span><span class="op">)</span>, data<span class="op">=</span><span class="va">quality_seq_stats</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Mean Phred score"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"# Sequences"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">stage</span><span class="op">~</span><span class="va">.</span>, scales <span class="op">=</span> <span class="st">"free_y"</span><span class="op">)</span></span>
<span><span class="va">g_quality_seq</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-08_cebria-mendoza_files/figure-html/plot-quality-3.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb9"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_stage_trace</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">basic_stats</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">stage</span>, group<span class="op">=</span><span class="va">sample</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span></span>
<span></span>
<span><span class="co"># Plot reads over preprocessing</span></span>
<span><span class="va">g_reads_stages</span> <span class="op">&lt;-</span> <span class="va">g_stage_trace</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">n_read_pairs</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span><span class="st">"# Read pairs"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_reads_stages</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-08_cebria-mendoza_files/figure-html/preproc-figures-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb10"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Plot relative read losses during preprocessing</span></span>
<span><span class="va">g_reads_rel</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">n_reads_rel</span>, </span>
<span>                      <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">stage</span>, group<span class="op">=</span><span class="va">sample</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">p_reads_lost_abs_marginal</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span><span class="st">"% Total Reads Lost"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, </span>
<span>                     labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">x</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span></span>
<span><span class="va">g_reads_rel</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-08_cebria-mendoza_files/figure-html/preproc-figures-2.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb11"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">stage_dup</span> <span class="op">&lt;-</span> <span class="va">basic_stats</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">stage</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>dmin <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">percent_duplicates</span><span class="op">)</span>, dmax<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">percent_duplicates</span><span class="op">)</span>,</span>
<span>            dmean<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">percent_duplicates</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">g_dup_stages</span> <span class="op">&lt;-</span> <span class="va">g_stage_trace</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">percent_duplicates</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span><span class="st">"% Duplicates"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_dup_stages</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-08_cebria-mendoza_files/figure-html/preproc-dedup-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb12"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_readlen_stages</span> <span class="op">&lt;-</span> <span class="va">g_stage_trace</span> <span class="op">+</span> <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">mean_seq_len</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span><span class="st">"Mean read length (nt)"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_readlen_stages</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-08_cebria-mendoza_files/figure-html/preproc-dedup-2.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb13"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Calculate reads lost during ribodepletion (approximation for % ribosomal reads)</span></span>
<span><span class="va">reads_ribo</span> <span class="op">&lt;-</span> <span class="va">n_reads_rel</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">stage</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"dedup"</span>, <span class="st">"ribo_secondary"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>p_reads_ribo<span class="op">=</span><span class="fl">1</span><span class="op">-</span><span class="va">n_read_pairs</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">/</span><span class="va">n_read_pairs</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by <span class="op">=</span> <span class="st">'sample'</span><span class="op">)</span></span>
<span><span class="va">reads_ribo_summ</span> <span class="op">&lt;-</span> <span class="va">reads_ribo</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>min<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">p_reads_ribo</span><span class="op">)</span>, max<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_ribo</span><span class="op">)</span>,</span>
<span>            mean<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">p_reads_ribo</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by <span class="op">=</span> <span class="st">'sample'</span><span class="op">)</span></span>
<span><span class="va">g_reads_ribo</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">reads_ribo</span>, </span>
<span>                       <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">library</span>, y<span class="op">=</span><span class="va">p_reads_ribo</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Approx % ribosomal reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span><span class="op">)</span>,</span>
<span>                     breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="va">theme_kit</span></span>
<span><span class="va">g_reads_ribo</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-08_cebria-mendoza_files/figure-html/ribo-frac-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section></section><section id="taxonomic-composition" class="level1" data-number="3"><h1 data-number="3">
<span class="header-section-number">3</span> Taxonomic composition</h1>
<section id="high-level-composition" class="level2" data-number="3.1"><h2 data-number="3.1" class="anchored" data-anchor-id="high-level-composition">
<span class="header-section-number">3.1</span> High-level composition</h2>
<p>To assess the high-level composition of the reads, I ran the ribodepleted files through Kraken2 and summarized the results with Bracken.</p>
<p>The groups listed below were created by Will:</p>
<ul>
<li>Filtered (removed during cleaning)</li>
<li>Duplicate (removed during deduplication)</li>
<li>Ribosomal (removed during ribodepletion)</li>
<li>Unassigned (non-ribosomal reads that were not assigned to any taxon by Kraken/Bracken)</li>
<li>Bacterial (non-ribosomal reads assigned to the Bacteria domain by Kraken/Bracken)</li>
<li>Archaeal (non-ribosomal reads assigned to the Archaea domain by Kraken/Bracken)</li>
<li>Viral (non-ribosomal reads assigned to the Viruses domain by Kraken/Bracken)</li>
<li>Human (non-ribosomal reads assigned to the Eukarya domain by Kraken/Bracken)</li>
</ul>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb14"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">classifications</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Filtered"</span>, <span class="st">"Duplicate"</span>, <span class="st">"Ribosomal"</span>, <span class="st">"Unassigned"</span>,</span>
<span>                     <span class="st">"Bacterial"</span>, <span class="st">"Archaeal"</span>, <span class="st">"Viral"</span>, <span class="st">"Human"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Import composition data</span></span>
<span><span class="va">tax_final_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">results_dir</span>, <span class="st">"taxonomy_final"</span><span class="op">)</span></span>
<span><span class="va">comp_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">tax_final_dir</span>, <span class="st">"taxonomic_composition.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">comp</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">comp_path</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">left_join</span><span class="op">(</span><span class="va">libraries</span>, by <span class="op">=</span> <span class="st">"sample"</span><span class="op">)</span> </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stderr">
<pre><code>Rows: 480 Columns: 4
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr (2): sample, classification
dbl (2): n_reads, p_reads

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb16"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">comp_minor</span> <span class="op">&lt;-</span> <span class="va">comp</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">classification</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Archaeal"</span>, <span class="st">"Viral"</span>, <span class="st">"Human"</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">comp_assigned</span> <span class="op">&lt;-</span> <span class="va">comp</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span> <span class="va">classification</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Filtered"</span>, <span class="st">"Duplicate"</span>, </span>
<span>                                 <span class="st">"Ribosomal"</span>, <span class="st">"Unassigned"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">n_reads</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">comp_assigned_minor</span> <span class="op">&lt;-</span> <span class="va">comp_assigned</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">classification</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Archaeal"</span>, <span class="st">"Viral"</span>, <span class="st">"Human"</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Summarize composition</span></span>
<span><span class="va">read_comp_summ</span> <span class="op">&lt;-</span> <span class="va">comp</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">classification</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop_last"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>n_reads <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">n_reads</span>,<span class="fl">0</span><span class="op">)</span>,</span>
<span>         p_reads <span class="op">=</span> <span class="va">n_reads</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads</span><span class="op">)</span>,</span>
<span>         pc_reads <span class="op">=</span> <span class="va">p_reads</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>classification <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">classification</span>, levels<span class="op">=</span><span class="va">classifications</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">classification</span>, <span class="va">n_reads</span>, <span class="va">pc_reads</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>`# of reads` <span class="op">=</span> <span class="va">n_reads</span>, <span class="st">"% of reads"</span> <span class="op">=</span> <span class="va">pc_reads</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>`% of reads` <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sprintf.html">sprintf</a></span><span class="op">(</span><span class="st">"%.2f"</span>, <span class="va">`% of reads`</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">read_comp_summ</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["classification"],"name":[1],"type":["fct"],"align":["left"]},{"label":["# of reads"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["% of reads"],"name":[3],"type":["chr"],"align":["left"]}],"data":[{"1":"Archaeal","2":"23236","3":"0.01"},{"1":"Bacterial","2":"35013621","3":"15.15"},{"1":"Duplicate","2":"125800100","3":"54.45"},{"1":"Filtered","2":"3178616","3":"1.38"},{"1":"Human","2":"1352337","3":"0.59"},{"1":"Ribosomal","2":"31881359","3":"13.80"},{"1":"Unassigned","2":"29589981","3":"12.81"},{"1":"Viral","2":"4199802","3":"1.82"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb17"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Prepare plotting templates</span></span>
<span><span class="va">g_comp_base</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span>mapping<span class="op">=</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">library</span>, y<span class="op">=</span><span class="va">p_reads</span>, fill<span class="op">=</span><span class="va">classification</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_discrete</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Plasma pool"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span> <span class="op">+</span> </span>
<span>  <span class="fu">theme</span><span class="op">(</span>plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>hjust<span class="op">=</span><span class="fl">0</span>, face<span class="op">=</span><span class="st">"plain"</span>, size<span class="op">=</span><span class="fu">rel</span><span class="op">(</span><span class="fl">1.5</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">scale_y_pc_reads</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/partial.html">partial</a></span><span class="op">(</span><span class="va">scale_y_continuous</span>, name <span class="op">=</span> <span class="st">"% Reads"</span>,</span>
<span>                                   expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span></span>
<span><span class="va">geom_comp</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/partial.html">partial</a></span><span class="op">(</span><span class="va">geom_col</span>, position <span class="op">=</span> <span class="st">"stack"</span>, width <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Define a color palette for the classification</span></span>
<span><span class="va">classification_colors</span> <span class="op">&lt;-</span> <span class="fu">brewer.pal</span><span class="op">(</span><span class="fl">8</span>, <span class="st">"Accent"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">classification_colors</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Filtered"</span>, <span class="st">"Duplicate"</span>, <span class="st">"Ribosomal"</span>, <span class="st">"Unassigned"</span>, <span class="st">"Bacterial"</span>, <span class="st">"Archaeal"</span>, <span class="st">"Viral"</span>, <span class="st">"Human"</span><span class="op">)</span></span>
<span><span class="va">scale_fill_classification</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values <span class="op">=</span> <span class="va">classification_colors</span>, name <span class="op">=</span> <span class="st">"Classification"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Plot overall composition</span></span>
<span><span class="va">g_comp</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> <span class="fu">geom_comp</span><span class="op">(</span>data <span class="op">=</span> <span class="va">comp</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_pc_reads</span><span class="op">(</span>limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_classification</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Read composition (all reads, all groups)"</span><span class="op">)</span></span>
<span><span class="va">g_comp</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-08_cebria-mendoza_files/figure-html/plot-composition-all-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb18"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_comp_assigned</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_comp</span><span class="op">(</span>data <span class="op">=</span> <span class="va">comp_assigned</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_pc_reads</span><span class="op">(</span>limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_classification</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Read composition (assigned reads, all groups)"</span><span class="op">)</span></span>
<span><span class="va">g_comp_assigned</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-08_cebria-mendoza_files/figure-html/plot-composition-all-2.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section><section id="total-viral-content" class="level2" data-number="3.2"><h2 data-number="3.2" class="anchored" data-anchor-id="total-viral-content">
<span class="header-section-number">3.2</span> Total viral content</h2>
<p>Total viral fraction average <span class="math inline">\(1.81 \times 10^{-2}\)</span> across samples. As a fraction of assigned (rather than total) reads, this jumped to <span class="math inline">\(1.60 \times 10^{-1}\)</span>.</p>
</section><section id="taxonomic-composition-of-viruses" class="level2" data-number="3.3"><h2 data-number="3.3" class="anchored" data-anchor-id="taxonomic-composition-of-viruses">
<span class="header-section-number">3.3</span> Taxonomic composition of viruses</h2>
<p>The two dominant viruses we see are Anellovirdae and Rhabdovirdae (spike-in). Followed by these two viral families is Flavivirdae, and lastly, also by a much smaller percent, Microviridae (spike-in). The threshold for the label “other” are the set of families that make up less than 1% composition in all samples.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb19"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get viral taxonomy</span></span>
<span><span class="va">viral_taxa_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"total-virus-db.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">viral_taxa</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">viral_taxa_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get Kraken reports</span></span>
<span><span class="va">reports_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">tax_final_dir</span>, <span class="st">"kraken_reports.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">reports</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">reports_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Filter to viral taxa</span></span>
<span><span class="va">kraken_reports_viral</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">reports</span>, <span class="va">taxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_viral <span class="op">=</span> <span class="va">n_reads_clade</span><span class="op">/</span><span class="va">n_reads_clade</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">kraken_reports_viral_cleaned</span> <span class="op">&lt;-</span> <span class="va">kraken_reports_viral</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">pc_reads_total</span>, <span class="op">-</span><span class="va">n_reads_direct</span>, <span class="op">-</span><span class="fu">contains</span><span class="op">(</span><span class="st">"minimizers"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span>, <span class="va">p_reads_viral</span>, <span class="va">n_reads_clade</span>, <span class="fu">everything</span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">ungroup</span></span>
<span></span>
<span><span class="va">viral_classes</span> <span class="op">&lt;-</span> <span class="va">kraken_reports_viral_cleaned</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">rank</span> <span class="op">==</span> <span class="st">"C"</span><span class="op">)</span></span>
<span><span class="va">viral_families</span> <span class="op">&lt;-</span> <span class="va">kraken_reports_viral_cleaned</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">rank</span> <span class="op">==</span> <span class="st">"F"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb20"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">major_threshold</span> <span class="op">&lt;-</span> <span class="fl">0.01</span></span>
<span></span>
<span><span class="co"># Identify major viral families</span></span>
<span><span class="va">viral_families_major_tab</span> <span class="op">&lt;-</span> <span class="va">viral_families</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>p_reads_viral_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_viral</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_viral_max</span> <span class="op">&gt;=</span> <span class="va">major_threshold</span><span class="op">)</span></span>
<span><span class="va">viral_families_major_list</span> <span class="op">&lt;-</span> <span class="va">viral_families_major_tab</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">name</span><span class="op">)</span></span>
<span><span class="va">viral_families_major</span> <span class="op">&lt;-</span> <span class="va">viral_families</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">viral_families_major_list</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span>, <span class="va">sample</span>, <span class="va">p_reads_viral</span><span class="op">)</span></span>
<span><span class="va">viral_families_minor</span> <span class="op">&lt;-</span> <span class="va">viral_families_major</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>p_reads_viral_major <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads_viral</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Other"</span>, taxid<span class="op">=</span><span class="cn">NA</span>, p_reads_viral <span class="op">=</span> <span class="fl">1</span><span class="op">-</span><span class="va">p_reads_viral_major</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span>, <span class="va">sample</span>, <span class="va">p_reads_viral</span><span class="op">)</span></span>
<span><span class="va">viral_families_display</span> <span class="op">&lt;-</span> <span class="va">viral_families_major</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">bind_rows</span><span class="op">(</span><span class="va">viral_families_minor</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p_reads_viral</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">name</span>, levels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">viral_families_major_list</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">p_reads_viral</span>, classification<span class="op">=</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by<span class="op">=</span><span class="st">'sample'</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_families</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_comp</span><span class="op">(</span>data<span class="op">=</span><span class="va">viral_families_display</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% Viral Reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, </span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">'Accent'</span><span class="op">)</span></span>
<span><span class="va">g_families</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-08_cebria-mendoza_files/figure-html/viral-family-composition-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<p>Excluding <em>Microviridae</em> and <em>Rhabdovirdae</em> (spike-ins), remaining viral sequences are distributed across a wide variety:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb21"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">major_threshold_adj</span> <span class="op">&lt;-</span> <span class="fl">0.05</span></span>
<span></span>
<span><span class="co"># Adjust viral family counts</span></span>
<span><span class="va">viral_families_adj</span> <span class="op">&lt;-</span> <span class="va">viral_families</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span><span class="op">(</span><span class="va">name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Rhabdoviridae"</span>,<span class="st">"Microviridae"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_viral <span class="op">=</span> <span class="va">p_reads_viral</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads_viral</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Identify major viral families</span></span>
<span><span class="va">viral_families_major_tab</span> <span class="op">&lt;-</span> <span class="va">viral_families_adj</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>p_reads_viral_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_viral</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_viral_max</span> <span class="op">&gt;=</span> <span class="va">major_threshold</span><span class="op">)</span></span>
<span><span class="va">viral_families_major_list</span> <span class="op">&lt;-</span> <span class="va">viral_families_major_tab</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">name</span><span class="op">)</span></span>
<span><span class="va">viral_families_major</span> <span class="op">&lt;-</span> <span class="va">viral_families_adj</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">viral_families_major_list</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span>, <span class="va">sample</span>, <span class="va">p_reads_viral</span><span class="op">)</span></span>
<span><span class="va">viral_families_minor</span> <span class="op">&lt;-</span> <span class="va">viral_families_major</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>p_reads_viral_major <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads_viral</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Other"</span>, taxid<span class="op">=</span><span class="cn">NA</span>, p_reads_viral <span class="op">=</span> <span class="fl">1</span><span class="op">-</span><span class="va">p_reads_viral_major</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span>, <span class="va">sample</span>, <span class="va">p_reads_viral</span><span class="op">)</span></span>
<span><span class="va">viral_families_display</span> <span class="op">&lt;-</span> <span class="va">viral_families_major</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">bind_rows</span><span class="op">(</span><span class="va">viral_families_minor</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p_reads_viral</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">name</span>, levels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">viral_families_major_list</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">p_reads_viral</span>, classification<span class="op">=</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by <span class="op">=</span> <span class="st">'sample'</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">palette_viral</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu">brewer.pal</span><span class="op">(</span><span class="fl">12</span>, <span class="st">"Set3"</span><span class="op">)</span>, <span class="fu">brewer.pal</span><span class="op">(</span><span class="fl">8</span>, <span class="st">"Dark2"</span><span class="op">)</span>, <span class="fu">brewer.pal</span><span class="op">(</span><span class="fl">9</span>, <span class="st">"Set1"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_families_adj</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_comp</span><span class="op">(</span>data<span class="op">=</span><span class="va">viral_families_display</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% Viral Reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, </span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values<span class="op">=</span><span class="va">palette_viral</span>, name <span class="op">=</span> <span class="st">"Viral class"</span><span class="op">)</span></span>
<span><span class="va">g_families_adj</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-08_cebria-mendoza_files/figure-html/viral-family-composition-exclusion-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section></section><section id="human-infecting-virus-reads" class="level1" data-number="4"><h1 data-number="4">
<span class="header-section-number">4</span> Human-infecting virus reads</h1>
<section id="overall-relative-abundance" class="level2" data-number="4.1"><h2 data-number="4.1" class="anchored" data-anchor-id="overall-relative-abundance">
<span class="header-section-number">4.1</span> Overall relative abundance</h2>
<p>I calculated the relative abundance of human-infecting viruses in two ways:</p>
<ul>
<li><p>First, as the total number of deduplicated human-virus reads in each sample, divided by the number of raw reads (“All reads”). This results in a viral fraction of <span class="math inline">\(4.06 \times 10^{-2}\)</span> across all samples</p></li>
<li><p>Second, as a fraction of preprocessed (cleaned, deduplicated, computationally ribodepleted) reads (“Preprocessed reads”). This results in a viral fraction of <span class="math inline">\(1.11 \times 10^{-1}\)</span> across all samples.</p></li>
</ul></section><section id="overall-taxonomy-and-composition" class="level2" data-number="4.2"><h2 data-number="4.2" class="anchored" data-anchor-id="overall-taxonomy-and-composition">
<span class="header-section-number">4.2</span> Overall taxonomy and composition</h2>
<p>Composition of HV reads was not greatly changed from when looking at all viral reads. The two dominant viruses we see are Anellovirdae and Rhabdovirdae. Followed by these two viral families is Flavivirdae, and lastly, also by a much smaller percent, Microviridae. The threshold for the label “other” are the set of families that make up less than 5% composition in all samples.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb22"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">threshold_major_family</span> <span class="op">&lt;-</span> <span class="fl">0.05</span></span>
<span></span>
<span><span class="co"># Count reads for each human-viral family</span></span>
<span><span class="va">hv_family_counts</span> <span class="op">&lt;-</span> <span class="va">hv_reads_family</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">count</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"n_reads_hv"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_hv <span class="op">=</span> <span class="va">n_reads_hv</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Identify high-ranking families and group others</span></span>
<span><span class="va">hv_family_major_tab</span> <span class="op">&lt;-</span> <span class="va">hv_family_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_hv</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_hv</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">row_number</span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p_reads_hv</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_hv</span> <span class="op">&gt;</span> <span class="va">threshold_major_family</span><span class="op">)</span></span>
<span><span class="va">hv_family_counts_major</span> <span class="op">&lt;-</span> <span class="va">hv_family_counts</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">hv_family_major_tab</span><span class="op">$</span><span class="va">name</span>, <span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">name_display</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_hv <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>, p_reads_hv <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads_hv</span><span class="op">)</span>, </span>
<span>            .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">name_display</span>, </span>
<span>                               levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">hv_family_major_tab</span><span class="op">$</span><span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">hv_family_counts_display</span> <span class="op">&lt;-</span> <span class="va">hv_family_counts_major</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">p_reads_hv</span>, classification <span class="op">=</span> <span class="va">name_display</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by <span class="op">=</span> <span class="st">'sample'</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_hv_family</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>data<span class="op">=</span><span class="va">hv_family_counts_display</span>, position <span class="op">=</span> <span class="st">"stack"</span>, width<span class="op">=</span><span class="fl">1</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% HV Reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, </span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">'Accent'</span>, name <span class="op">=</span> <span class="st">"Viral family"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title<span class="op">=</span><span class="st">"Family composition of human-viral reads"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>fill<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>ncol<span class="op">=</span><span class="fl">4</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fu">rel</span><span class="op">(</span><span class="fl">1.4</span><span class="op">)</span>, hjust<span class="op">=</span><span class="fl">0</span>, face<span class="op">=</span><span class="st">"plain"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_hv_family</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-08_cebria-mendoza_files/figure-html/hv-family-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb23"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get most prominent families for text</span></span>
<span><span class="va">hv_family_collate</span> <span class="op">&lt;-</span> <span class="va">hv_family_counts</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_tot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>,</span>
<span>            p_reads_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_hv</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">n_reads_tot</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">hv_family_collate</span> <span class="op">%&gt;%</span></span>
<span> <span class="fu">select</span><span class="op">(</span><span class="va">name</span>,<span class="va">taxid</span>, <span class="va">n_reads_tot</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span> <span class="fu">rename</span><span class="op">(</span></span>
<span>  <span class="st">'family'</span> <span class="op">=</span> <span class="st">'name'</span>,</span>
<span>  <span class="st">'# of total reads'</span> <span class="op">=</span> <span class="st">'n_reads_tot'</span>,</span>
<span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["family"],"name":[1],"type":["chr"],"align":["left"]},{"label":["taxid"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["# of total reads"],"name":[3],"type":["int"],"align":["right"]}],"data":[{"1":"Anelloviridae","2":"687329","3":"8323297"},{"1":"Rhabdoviridae","2":"11270","3":"935461"},{"1":"Flaviviridae","2":"11050","3":"94741"},{"1":"Microviridae","2":"10841","3":"81660"},{"1":"Hepeviridae","2":"3079366","3":"1"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
</section><section id="analyzing-specific-families" class="level2" data-number="4.3"><h2 data-number="4.3" class="anchored" data-anchor-id="analyzing-specific-families">
<span class="header-section-number">4.3</span> Analyzing specific families</h2>
<p>We now investigate the composition of specific families that had more than 5 viral reads. In investigating individual viral families, to avoid distortions from a few rare reads, I restricted myself to samples where that family made up at least 1% of human-viral reads:</p>
<section id="anelloviridae-number-of-reads-8323297" class="level3" data-number="4.3.1"><h3 data-number="4.3.1" class="anchored" data-anchor-id="anelloviridae-number-of-reads-8323297">
<span class="header-section-number">4.3.1</span> Anelloviridae (Number of reads: 8,323,297)</h3>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb24"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Anelloviridae</span></span>
<span><span class="fu">plot_viral_family_histogram</span><span class="op">(</span>taxid_chosen<span class="op">=</span><span class="fl">687329</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-08_cebria-mendoza_files/figure-html/plot-anelloviridae-histogram-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb25"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Anelloviridae</span></span>
<span><span class="fu">plot_viral_family_composition</span><span class="op">(</span>taxid_chosen<span class="op">=</span><span class="fl">687329</span>, threshold_major_species <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-08_cebria-mendoza_files/figure-html/plot-anelloviridae-composition-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section><section id="rhabdoviridae-number-of-reads-935461-spike-in" class="level3" data-number="4.3.2"><h3 data-number="4.3.2" class="anchored" data-anchor-id="rhabdoviridae-number-of-reads-935461-spike-in">
<span class="header-section-number">4.3.2</span> Rhabdoviridae (Number of reads: 935,461) [SPIKE-IN]</h3>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb26"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">plot_viral_family_histogram</span><span class="op">(</span>taxid_chosen<span class="op">=</span><span class="fl">11270</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-08_cebria-mendoza_files/figure-html/plot-rhabdoviridae-histogram-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb27"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">plot_viral_family_composition</span><span class="op">(</span>taxid_chosen<span class="op">=</span><span class="fl">11270</span>, threshold_major_species <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-08_cebria-mendoza_files/figure-html/plot-rhabdoviridae-composition-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section><section id="flaviviridae-number-of-reads-94741" class="level3" data-number="4.3.3"><h3 data-number="4.3.3" class="anchored" data-anchor-id="flaviviridae-number-of-reads-94741">
<span class="header-section-number">4.3.3</span> Flaviviridae (Number of reads: 94,741)</h3>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb28"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">plot_viral_family_histogram</span><span class="op">(</span>taxid_chosen<span class="op">=</span><span class="fl">11050</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-08_cebria-mendoza_files/figure-html/plot-flavi-histogram-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb29"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">plot_viral_family_composition</span><span class="op">(</span>taxid_chosen<span class="op">=</span><span class="fl">11050</span>, threshold_major_species <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-08_cebria-mendoza_files/figure-html/plot-flavi-composition-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section><section id="microviridae-number-of-reads-10841-spike-in" class="level3" data-number="4.3.4"><h3 data-number="4.3.4" class="anchored" data-anchor-id="microviridae-number-of-reads-10841-spike-in">
<span class="header-section-number">4.3.4</span> Microviridae (Number of reads: 10,841) [SPIKE-IN]</h3>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb30"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">plot_viral_family_histogram</span><span class="op">(</span>taxid_chosen<span class="op">=</span><span class="fl">10841</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-08_cebria-mendoza_files/figure-html/plot-microviridae-histogram-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb31"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">plot_viral_family_composition</span><span class="op">(</span>taxid_chosen<span class="op">=</span><span class="fl">10841</span>, threshold_major_species <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-08_cebria-mendoza_files/figure-html/plot-microviridae-composition-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section></section><section id="relative-abundance-of-pathogenic-viruses-of-interest" class="level2" data-number="4.4"><h2 data-number="4.4" class="anchored" data-anchor-id="relative-abundance-of-pathogenic-viruses-of-interest">
<span class="header-section-number">4.4</span> Relative abundance of pathogenic viruses of interest</h2>
<p>Each dot represents a sample, colored by viral family. The x-axis shows the relative abundance of human-infecting viruses, and the y-axis shows the species. I’ve removed the species relating to the spiked-in family (Microviridae and Rhabdoviridae).</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb32"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">species_family</span> <span class="op">&lt;-</span> <span class="va">result</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">species</span>, <span class="va">family</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">rename</span><span class="op">(</span><span class="st">'name'</span> <span class="op">=</span> <span class="st">'species'</span><span class="op">)</span></span>
<span></span>
<span><span class="va">play</span> <span class="op">&lt;-</span> <span class="va">hv_species_counts</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">ungroup</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by <span class="op">=</span> <span class="st">'sample'</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">species_family</span>, by <span class="op">=</span> <span class="st">'name'</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span><span class="op">(</span><span class="va">family</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Microviridae"</span>, <span class="st">"Rhabdoviridae"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> </span>
<span></span>
<span><span class="va">adjusted_play</span> <span class="op">&lt;-</span> <span class="va">play</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>virus_prevalence_num <span class="op">=</span> <span class="fu">n_distinct</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span><span class="op">/</span><span class="fu">n_distinct</span><span class="op">(</span><span class="va">libraries</span><span class="op">)</span>,</span>
<span>         total_reads_hv <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ungroup</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name <span class="op">=</span> <span class="fu">fct_reorder</span><span class="op">(</span><span class="va">name</span>, <span class="va">virus_prevalence_num</span>, .desc<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">name</span>, <span class="va">ra_reads_hv</span>, <span class="va">family</span>, <span class="va">virus_prevalence_num</span>, <span class="va">total_reads_hv</span>, <span class="va">n_reads_hv</span><span class="op">)</span></span>
<span></span>
<span><span class="va">pal</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu">brewer.pal</span><span class="op">(</span><span class="fl">8</span>, <span class="st">'Dark2'</span><span class="op">)</span>,<span class="fu">brewer.pal</span><span class="op">(</span><span class="fl">8</span>, <span class="st">'Accent'</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#, labels = label_log(digits=2)</span></span>
<span></span>
<span><span class="va">ra_dot</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">adjusted_play</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">ra_reads_hv</span>, y<span class="op">=</span><span class="va">name</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_quasirandom</span><span class="op">(</span>orientation <span class="op">=</span> <span class="st">'y'</span>, <span class="fu">aes</span><span class="op">(</span>color <span class="op">=</span> <span class="va">family</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_manual</span><span class="op">(</span>values <span class="op">=</span> <span class="va">pal</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu">scale_x_log10</span><span class="op">(</span></span>
<span>    <span class="st">"Relative abundance human virus reads"</span>,</span>
<span>    labels<span class="op">=</span><span class="fu">label_log</span><span class="op">(</span>digits<span class="op">=</span><span class="fl">3</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>y <span class="op">=</span>  <span class="st">""</span>,</span>
<span>       color <span class="op">=</span> <span class="st">'Viral family'</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">guides</span><span class="op">(</span>color <span class="op">=</span> <span class="fu">guide_legend</span><span class="op">(</span>ncol<span class="op">=</span><span class="fl">2</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme_light</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu">theme</span><span class="op">(</span></span>
<span>    axis.text.y <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,</span>
<span>    axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    axis.ticks.y <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,axis.line <span class="op">=</span> <span class="fu">element_line</span><span class="op">(</span>colour <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span>,</span>
<span>    axis.title.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">14</span><span class="op">)</span>,    </span>
<span>    legend.text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,</span>
<span>    legend.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span>, face<span class="op">=</span><span class="st">"bold"</span><span class="op">)</span>,</span>
<span>    <span class="co">#legend.position = 'bottom',</span></span>
<span>    legend.position <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span>,  <span class="co"># Move legend to top right</span></span>
<span>    legend.justification <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span>,  <span class="co"># Align legend to top right</span></span>
<span>    panel.grid.minor <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    panel.border <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    panel.background <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">ra_dot</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-08_cebria-mendoza_files/figure-html/plot-pathogenic-viruses-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<p>All of the viruses found above are known to be non-pathogenic in healthy humans.</p>
</section><section id="relative-abundance-assuming-perfect-human-read-removal" class="level2" data-number="4.5"><h2 data-number="4.5" class="anchored" data-anchor-id="relative-abundance-assuming-perfect-human-read-removal">
<span class="header-section-number">4.5</span> Relative abundance assuming perfect human read removal</h2>
<p>Assuming we’re able to perfectly remove all human reads, the average relative abundance of known human infecting virus is <span class="math inline">\(4.08 \times 10^{-2}\)</span> pre-viral spike removal and <span class="math inline">\(3.64 \times 10^{-2}\)</span> post-viral spike removal.</p>
</section></section><section id="conclusion" class="level1" data-number="5"><h1 data-number="5">
<span class="header-section-number">5</span> Conclusion</h1>
<p>Overall, we found a lot of anelloviridae, which makes sense given that this paper is about discovering new species of anelloviridae in blood.</p>
<p>There were some interesting takeways from this analysis:</p>
<ol type="1">
<li>Sample processing specifically enhancing for viruses can increase the viral relative abundance substantially. In this study, we saw viral RA as high as <span class="math inline">\(10^{-2}\)</span> to <span class="math inline">\(10^{-1}\)</span> whereas in prior studies <a href="https://data.securebio.org/harmons-public-notebook/notebooks/2024-07-23-mengyi.html">(1)</a> <a href="https://data.securebio.org/harmons-public-notebook/notebooks/2024-07-22-thijssen.html">(2)</a> we saw viral RA around the <span class="math inline">\(10^{-4}\)</span> to <span class="math inline">\(10^{-3}\)</span> range.</li>
<li>A lot of the Anelloviridae are largely uncharacterized (denoted by the family name followed by the abbreviation “sp.”).</li>
<li>In small sample populations, it’s unlikely for us to see <a href="https://virology.ws/2017/03/23/the-viruses-in-your-blood/">well-known blood viruses</a>.</li>
</ol></section><section id="appendix" class="level1" data-number="6"><h1 data-number="6">
<span class="header-section-number">6</span> Appendix</h1>
<section id="human-infecting-virus-families-genera-and-species" class="level2" data-number="6.1"><h2 data-number="6.1" class="anchored" data-anchor-id="human-infecting-virus-families-genera-and-species">
<span class="header-section-number">6.1</span> Human-infecting virus families, genera, and species</h2>
<p>To get a good overview of families, genera, and species, we can look at a Sankey plot where the magnitude of relative abundance, averaged over all samples, is shown in parentheses.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb33"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Function to create links</span></span>
<span><span class="va">create_links</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">data</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">family_to_genus</span> <span class="op">&lt;-</span> <span class="va">data</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">genus</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">group_by</span><span class="op">(</span><span class="va">family</span>, <span class="va">genus</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">summarise</span><span class="op">(</span>value <span class="op">=</span> <span class="fu">n</span><span class="op">(</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>source <span class="op">=</span> <span class="va">family</span>, target <span class="op">=</span> <span class="va">genus</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="va">genus_to_species</span> <span class="op">&lt;-</span> <span class="va">data</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">group_by</span><span class="op">(</span><span class="va">genus</span>, <span class="va">species</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">summarise</span><span class="op">(</span>value <span class="op">=</span> <span class="fu">n</span><span class="op">(</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>source <span class="op">=</span> <span class="va">genus</span>, target <span class="op">=</span> <span class="va">species</span><span class="op">)</span></span>
<span></span>
<span>  <span class="va">family_to_species</span> <span class="op">&lt;-</span> <span class="va">data</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">genus</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">group_by</span><span class="op">(</span><span class="va">family</span>, <span class="va">species</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">summarise</span><span class="op">(</span>value <span class="op">=</span> <span class="fu">n</span><span class="op">(</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>source <span class="op">=</span> <span class="va">family</span>, target <span class="op">=</span> <span class="va">species</span><span class="op">)</span></span>
<span></span>
<span>  <span class="fu">bind_rows</span><span class="op">(</span><span class="va">family_to_genus</span>, <span class="va">genus_to_species</span>, <span class="va">family_to_species</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">source</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Function to create nodes</span></span>
<span><span class="va">create_nodes</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">links</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span></span>
<span>    name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">links</span><span class="op">$</span><span class="va">source</span>, <span class="va">links</span><span class="op">$</span><span class="va">target</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Function to prepare data for Sankey diagram</span></span>
<span><span class="va">prepare_sankey_data</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">links</span>, <span class="va">nodes</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">links</span><span class="op">$</span><span class="va">IDsource</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/match.html">match</a></span><span class="op">(</span><span class="va">links</span><span class="op">$</span><span class="va">source</span>, <span class="va">nodes</span><span class="op">$</span><span class="va">name</span><span class="op">)</span> <span class="op">-</span> <span class="fl">1</span></span>
<span>  <span class="va">links</span><span class="op">$</span><span class="va">IDtarget</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/match.html">match</a></span><span class="op">(</span><span class="va">links</span><span class="op">$</span><span class="va">target</span>, <span class="va">nodes</span><span class="op">$</span><span class="va">name</span><span class="op">)</span> <span class="op">-</span> <span class="fl">1</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>links <span class="op">=</span> <span class="va">links</span>, nodes <span class="op">=</span> <span class="va">nodes</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Function to create Sankey plot</span></span>
<span><span class="va">create_sankey_plot</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">sankey_data</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu">sankeyNetwork</span><span class="op">(</span></span>
<span>    Links <span class="op">=</span> <span class="va">sankey_data</span><span class="op">$</span><span class="va">links</span>, </span>
<span>    Nodes <span class="op">=</span> <span class="va">sankey_data</span><span class="op">$</span><span class="va">nodes</span>,</span>
<span>    Source <span class="op">=</span> <span class="st">"IDsource"</span>, </span>
<span>    Target <span class="op">=</span> <span class="st">"IDtarget"</span>,</span>
<span>    Value <span class="op">=</span> <span class="st">"value"</span>, </span>
<span>    NodeID <span class="op">=</span> <span class="st">"name"</span>,</span>
<span>    sinksRight <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    nodeWidth <span class="op">=</span> <span class="fl">25</span>,</span>
<span>    fontSize <span class="op">=</span> <span class="fl">14</span>,</span>
<span>  <span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="va">save_sankey_as_png</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">sankey_plot</span>, <span class="va">width</span> <span class="op">=</span> <span class="fl">1000</span>, <span class="va">height</span> <span class="op">=</span> <span class="fl">800</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="co"># Save the plot as an HTML file</span></span>
<span>  <span class="fu">saveWidget</span><span class="op">(</span><span class="va">sankey_plot</span>, <span class="fu"><a href="https://rdrr.io/r/base/sprintf.html">sprintf</a></span><span class="op">(</span><span class="st">'%s/sankey.html'</span>,<span class="va">data_dir</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="va">format_scientific</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">digits</span><span class="op">=</span><span class="fl">2</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">x</span>, <span class="kw">function</span><span class="op">(</span><span class="va">val</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">val</span><span class="op">)</span> <span class="op">||</span> <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span><span class="op">(</span><span class="va">val</span><span class="op">)</span> <span class="op">&lt;</span> <span class="fl">1e-15</span><span class="op">)</span> <span class="op">{</span></span>
<span>      <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="st">"0"</span><span class="op">)</span></span>
<span>    <span class="op">}</span> <span class="kw">else</span> <span class="op">{</span></span>
<span>      <span class="va">exponent</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">floor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span><span class="op">(</span><span class="va">val</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span>      <span class="va">coef</span> <span class="op">&lt;-</span> <span class="va">val</span> <span class="op">/</span> <span class="fl">10</span><span class="op">^</span><span class="va">exponent</span></span>
<span>      <span class="co">#return(sprintf("%.1f × 10^%d", round(coef, digits), exponent))</span></span>
<span>      <span class="co"># May or may not be smart, just keeping magnitude</span></span>
<span>      <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sprintf.html">sprintf</a></span><span class="op">(</span><span class="st">"10^%d"</span>, <span class="va">exponent</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="op">}</span></span>
<span>  <span class="op">}</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="va">result</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span><span class="fu">across</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">genus_n_reads_tot</span>, <span class="va">genus_ra_reads_tot</span><span class="op">)</span>, <span class="op">~</span><span class="fu">replace_na</span><span class="op">(</span><span class="va">.</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span>,</span>
<span>         genus <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">genus</span><span class="op">)</span>, <span class="st">"Unknown Genus"</span>, <span class="va">genus</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>  species <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">species</span>, <span class="fu"><a href="https://rdrr.io/r/base/sprintf.html">sprintf</a></span><span class="op">(</span><span class="st">' (%s)'</span>, <span class="fu">format_scientific</span><span class="op">(</span><span class="va">species_ra_reads_tot</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,</span>
<span>  genus <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">genus</span>, <span class="fu"><a href="https://rdrr.io/r/base/sprintf.html">sprintf</a></span><span class="op">(</span><span class="st">' (%s)'</span>, <span class="fu">format_scientific</span><span class="op">(</span><span class="va">genus_ra_reads_tot</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,</span>
<span>  family <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">family</span>, <span class="fu"><a href="https://rdrr.io/r/base/sprintf.html">sprintf</a></span><span class="op">(</span><span class="st">' (%s)'</span>, <span class="fu">format_scientific</span><span class="op">(</span><span class="va">family_ra_reads_tot</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="va">links</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="fu">create_links</span><span class="op">(</span><span class="va">data</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">nodes</span> <span class="op">&lt;-</span> <span class="fu">create_nodes</span><span class="op">(</span><span class="va">links</span><span class="op">)</span></span>
<span><span class="va">sankey_data</span> <span class="op">&lt;-</span> <span class="fu">prepare_sankey_data</span><span class="op">(</span><span class="va">links</span>, <span class="va">nodes</span><span class="op">)</span></span>
<span><span class="va">sankey</span> <span class="op">&lt;-</span> <span class="fu">create_sankey_plot</span><span class="op">(</span><span class="va">sankey_data</span><span class="op">)</span></span>
<span></span>
<span><span class="va">sankey</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div id="htmlwidget-1ddacc645b381077ba05" style="width:100%;height:1462px;" class="sankeyNetwork html-widget"></div>
<script type="application/json" data-for="htmlwidget-1ddacc645b381077ba05">{"x":{"links":{"source":[0,0,0,0,1,2,3,4,4,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,7,7,7,7,7,7,7,7,7,7,7,7,7,7,8,9,10,11,11,11,11,11,11,12,12,12],"target":[5,6,7,11,10,9,11,8,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77],"value":[23,16,14,4,1,1,2,1,3,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1]},"nodes":{"name":["Anelloviridae (10^-2)","Flaviviridae (10^-3)","Hepeviridae (10^-7)","Microviridae (10^-3)","Rhabdoviridae (10^-3)","Alphatorquevirus (10^-4)","Betatorquevirus (10^-3)","Gammatorquevirus (10^-5)","Ledantevirus (10^-7)","Paslahepevirus (10^-7)","Pegivirus (10^-3)","Unknown Genus (0)","Vesiculovirus (10^-3)","Alphatorquevirus homin1 (10^-5)","Alphatorquevirus homin10 (10^-6)","Alphatorquevirus homin13 (10^-6)","Alphatorquevirus homin15 (10^-4)","Alphatorquevirus homin17 (10^-6)","Alphatorquevirus homin18 (10^-6)","Alphatorquevirus homin19 (10^-6)","Alphatorquevirus homin20 (10^-6)","Alphatorquevirus homin21 (10^-6)","Alphatorquevirus homin24 (10^-5)","Alphatorquevirus homin29 (10^-5)","Alphatorquevirus homin3 (10^-6)","Alphatorquevirus homin5 (10^-6)","Alphatorquevirus homin6 (10^-5)","Alphatorquevirus homin7 (10^-6)","Alphatorquevirus homin9 (10^-6)","Torque teno virus 11 (10^-5)","Torque teno virus 12 (10^-6)","Torque teno virus 16 (10^-5)","Torque teno virus 22 (10^-5)","Torque teno virus 27 (10^-7)","Torque teno virus 28 (10^-5)","Torque teno virus 8 (10^-6)","Betatorquevirus homini1 (10^-4)","Betatorquevirus homini10 (10^-5)","Betatorquevirus homini11 (10^-4)","Betatorquevirus homini12 (10^-3)","Betatorquevirus homini16 (10^-5)","Betatorquevirus homini18 (10^-4)","Betatorquevirus homini2 (10^-5)","Betatorquevirus homini26 (10^-5)","Betatorquevirus homini27 (10^-5)","Betatorquevirus homini3 (10^-5)","Betatorquevirus homini5 (10^-5)","Betatorquevirus homini6 (10^-6)","Betatorquevirus homini7 (10^-4)","Betatorquevirus homini8 (10^-5)","Betatorquevirus homini9 (10^-4)","TTV-like mini virus (10^-4)","Gammatorquevirus homidi1 (10^-6)","Gammatorquevirus homidi10 (10^-6)","Gammatorquevirus homidi11 (10^-6)","Gammatorquevirus homidi12 (10^-6)","Gammatorquevirus homidi13 (10^-6)","Gammatorquevirus homidi14 (10^-6)","Gammatorquevirus homidi2 (10^-6)","Gammatorquevirus homidi3 (10^-6)","Gammatorquevirus homidi4 (10^-6)","Gammatorquevirus homidi5 (10^-6)","Gammatorquevirus homidi6 (10^-6)","Gammatorquevirus homidi7 (10^-5)","Gammatorquevirus homidi8 (10^-5)","Gammatorquevirus homidi9 (10^-7)","Ledantevirus ledantec (10^-7)","Paslahepevirus balayani (10^-7)","Pegivirus hominis (10^-3)","Anelloviridae sp. (10^-2)","Micro Torque teno virus (10^-4)","Microviridae sp. (10^-3)","Microvirus sp. (10^-5)","Small anellovirus (10^-6)","Torque teno virus (10^-5)","Vesiculovirus indiana (10^-3)","Vesiculovirus isfahan (10^-7)","Vesiculovirus maraba (10^-7)"],"group":["Anelloviridae (10^-2)","Flaviviridae (10^-3)","Hepeviridae (10^-7)","Microviridae (10^-3)","Rhabdoviridae (10^-3)","Alphatorquevirus (10^-4)","Betatorquevirus (10^-3)","Gammatorquevirus (10^-5)","Ledantevirus (10^-7)","Paslahepevirus (10^-7)","Pegivirus (10^-3)","Unknown Genus (0)","Vesiculovirus (10^-3)","Alphatorquevirus homin1 (10^-5)","Alphatorquevirus homin10 (10^-6)","Alphatorquevirus homin13 (10^-6)","Alphatorquevirus homin15 (10^-4)","Alphatorquevirus homin17 (10^-6)","Alphatorquevirus homin18 (10^-6)","Alphatorquevirus homin19 (10^-6)","Alphatorquevirus homin20 (10^-6)","Alphatorquevirus homin21 (10^-6)","Alphatorquevirus homin24 (10^-5)","Alphatorquevirus homin29 (10^-5)","Alphatorquevirus homin3 (10^-6)","Alphatorquevirus homin5 (10^-6)","Alphatorquevirus homin6 (10^-5)","Alphatorquevirus homin7 (10^-6)","Alphatorquevirus homin9 (10^-6)","Torque teno virus 11 (10^-5)","Torque teno virus 12 (10^-6)","Torque teno virus 16 (10^-5)","Torque teno virus 22 (10^-5)","Torque teno virus 27 (10^-7)","Torque teno virus 28 (10^-5)","Torque teno virus 8 (10^-6)","Betatorquevirus homini1 (10^-4)","Betatorquevirus homini10 (10^-5)","Betatorquevirus homini11 (10^-4)","Betatorquevirus homini12 (10^-3)","Betatorquevirus homini16 (10^-5)","Betatorquevirus homini18 (10^-4)","Betatorquevirus homini2 (10^-5)","Betatorquevirus homini26 (10^-5)","Betatorquevirus homini27 (10^-5)","Betatorquevirus homini3 (10^-5)","Betatorquevirus homini5 (10^-5)","Betatorquevirus homini6 (10^-6)","Betatorquevirus homini7 (10^-4)","Betatorquevirus homini8 (10^-5)","Betatorquevirus homini9 (10^-4)","TTV-like mini virus (10^-4)","Gammatorquevirus homidi1 (10^-6)","Gammatorquevirus homidi10 (10^-6)","Gammatorquevirus homidi11 (10^-6)","Gammatorquevirus homidi12 (10^-6)","Gammatorquevirus homidi13 (10^-6)","Gammatorquevirus homidi14 (10^-6)","Gammatorquevirus homidi2 (10^-6)","Gammatorquevirus homidi3 (10^-6)","Gammatorquevirus homidi4 (10^-6)","Gammatorquevirus homidi5 (10^-6)","Gammatorquevirus homidi6 (10^-6)","Gammatorquevirus homidi7 (10^-5)","Gammatorquevirus homidi8 (10^-5)","Gammatorquevirus homidi9 (10^-7)","Ledantevirus ledantec (10^-7)","Paslahepevirus balayani (10^-7)","Pegivirus hominis (10^-3)","Anelloviridae sp. (10^-2)","Micro Torque teno virus (10^-4)","Microviridae sp. (10^-3)","Microvirus sp. (10^-5)","Small anellovirus (10^-6)","Torque teno virus (10^-5)","Vesiculovirus indiana (10^-3)","Vesiculovirus isfahan (10^-7)","Vesiculovirus maraba (10^-7)"]},"options":{"NodeID":"name","NodeGroup":"name","LinkGroup":null,"colourScale":"d3.scaleOrdinal(d3.schemeCategory20);","fontSize":14,"fontFamily":null,"nodeWidth":25,"nodePadding":10,"units":"","margin":{"top":null,"right":null,"bottom":null,"left":null},"iterations":32,"sinksRight":true}},"evals":[],"jsHooks":[]}</script>
</div>
</div>


<!-- -->

</section></section></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb34" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "Workflow of Cebria-Mendoza et al. (2021)"</span></span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a><span class="an">subtitle:</span><span class="co"> "Pooled plasma from Spain (DNA + RNA)"</span></span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> "Harmon Bhasin"</span></span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> 2024-07-08</span></span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span></span>
<span id="cb34-7"><a href="#cb34-7" aria-hidden="true" tabindex="-1"></a><span class="co">  html:</span></span>
<span id="cb34-8"><a href="#cb34-8" aria-hidden="true" tabindex="-1"></a><span class="co">    toc: true # table of contents</span></span>
<span id="cb34-9"><a href="#cb34-9" aria-hidden="true" tabindex="-1"></a><span class="co">    toc-title: "Table of contents" # table of contents title</span></span>
<span id="cb34-10"><a href="#cb34-10" aria-hidden="true" tabindex="-1"></a><span class="co">    number-sections: true # number sections</span></span>
<span id="cb34-11"><a href="#cb34-11" aria-hidden="true" tabindex="-1"></a><span class="co">    number-depth: 3 # number depth to show in table of contents</span></span>
<span id="cb34-12"><a href="#cb34-12" aria-hidden="true" tabindex="-1"></a><span class="co">    toc-location: right # table of contents location</span></span>
<span id="cb34-13"><a href="#cb34-13" aria-hidden="true" tabindex="-1"></a><span class="co">    page-layout: full # full page layout</span></span>
<span id="cb34-14"><a href="#cb34-14" aria-hidden="true" tabindex="-1"></a><span class="co">    code-fold: true # Keep option to fold code (i.e. show or hide it; default: hide)</span></span>
<span id="cb34-15"><a href="#cb34-15" aria-hidden="true" tabindex="-1"></a><span class="co">    code-tools: true # Code menu in the header of your document that provides various tools for readers to interact with the source code</span></span>
<span id="cb34-16"><a href="#cb34-16" aria-hidden="true" tabindex="-1"></a><span class="co">    code-link: true # Enables hyper-linking of functions within code blocks to their online documentation</span></span>
<span id="cb34-17"><a href="#cb34-17" aria-hidden="true" tabindex="-1"></a><span class="co">    df-print: paged # print data frame</span></span>
<span id="cb34-18"><a href="#cb34-18" aria-hidden="true" tabindex="-1"></a><span class="co">    fig-format: svg</span></span>
<span id="cb34-19"><a href="#cb34-19" aria-hidden="true" tabindex="-1"></a><span class="co">    other-links:</span></span>
<span id="cb34-20"><a href="#cb34-20" aria-hidden="true" tabindex="-1"></a><span class="co">      - text: Paper</span></span>
<span id="cb34-21"><a href="#cb34-21" aria-hidden="true" tabindex="-1"></a><span class="co">        href: https://doi.org/10.3390/v15071425</span></span>
<span id="cb34-22"><a href="#cb34-22" aria-hidden="true" tabindex="-1"></a><span class="co">      - text: Data</span></span>
<span id="cb34-23"><a href="#cb34-23" aria-hidden="true" tabindex="-1"></a><span class="co">        href: https://www.ncbi.nlm.nih.gov/bioproject/?term=PRJNA983534</span></span>
<span id="cb34-24"><a href="#cb34-24" aria-hidden="true" tabindex="-1"></a><span class="co">    code-links:</span></span>
<span id="cb34-25"><a href="#cb34-25" aria-hidden="true" tabindex="-1"></a><span class="co">      - text: Code for this post</span></span>
<span id="cb34-26"><a href="#cb34-26" aria-hidden="true" tabindex="-1"></a><span class="co">        icon: file-code</span></span>
<span id="cb34-27"><a href="#cb34-27" aria-hidden="true" tabindex="-1"></a><span class="co">        href: https://github.com/naobservatory/harmons-public-notebook/blob/main/notebooks/2024-07-22-thijssen.qmd</span></span>
<span id="cb34-28"><a href="#cb34-28" aria-hidden="true" tabindex="-1"></a><span class="an">editor:</span><span class="co"> </span></span>
<span id="cb34-29"><a href="#cb34-29" aria-hidden="true" tabindex="-1"></a><span class="co">  visual: true</span></span>
<span id="cb34-30"><a href="#cb34-30" aria-hidden="true" tabindex="-1"></a><span class="co">  render-on-save: true</span></span>
<span id="cb34-31"><a href="#cb34-31" aria-hidden="true" tabindex="-1"></a><span class="an">comments:</span></span>
<span id="cb34-32"><a href="#cb34-32" aria-hidden="true" tabindex="-1"></a><span class="co">  hypothesis: true # hypothesis</span></span>
<span id="cb34-33"><a href="#cb34-33" aria-hidden="true" tabindex="-1"></a><span class="an">execute:</span><span class="co"> </span></span>
<span id="cb34-34"><a href="#cb34-34" aria-hidden="true" tabindex="-1"></a><span class="co">  freeze: auto</span></span>
<span id="cb34-35"><a href="#cb34-35" aria-hidden="true" tabindex="-1"></a><span class="an">title-block-banner:</span><span class="co"> "#de2d26"</span></span>
<span id="cb34-36"><a href="#cb34-36" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb34-37"><a href="#cb34-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-40"><a href="#cb34-40" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-41"><a href="#cb34-41" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: load-packages</span></span>
<span id="cb34-42"><a href="#cb34-42" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb34-43"><a href="#cb34-43" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb34-44"><a href="#cb34-44" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cowplot)</span>
<span id="cb34-45"><a href="#cb34-45" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb34-46"><a href="#cb34-46" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(fastqcr)</span>
<span id="cb34-47"><a href="#cb34-47" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)</span>
<span id="cb34-48"><a href="#cb34-48" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(networkD3)</span>
<span id="cb34-49"><a href="#cb34-49" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb34-50"><a href="#cb34-50" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(htmlwidgets)</span>
<span id="cb34-51"><a href="#cb34-51" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggbeeswarm)</span>
<span id="cb34-52"><a href="#cb34-52" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(latex2exp)</span>
<span id="cb34-53"><a href="#cb34-53" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scales)</span>
<span id="cb34-54"><a href="#cb34-54" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb34-55"><a href="#cb34-55" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"/Users/harmonbhasin/work/securebio/sampling-strategies/scripts/aux_plot-theme.R"</span>)</span>
<span id="cb34-56"><a href="#cb34-56" aria-hidden="true" tabindex="-1"></a>theme_base <span class="ot">&lt;-</span> theme_base <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio =</span> <span class="cn">NULL</span>)</span>
<span id="cb34-57"><a href="#cb34-57" aria-hidden="true" tabindex="-1"></a>theme_kit <span class="ot">&lt;-</span> theme_base <span class="sc">+</span> <span class="fu">theme</span>(</span>
<span id="cb34-58"><a href="#cb34-58" aria-hidden="true" tabindex="-1"></a>  <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">angle =</span> <span class="dv">45</span>),</span>
<span id="cb34-59"><a href="#cb34-59" aria-hidden="true" tabindex="-1"></a>  <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb34-60"><a href="#cb34-60" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb34-61"><a href="#cb34-61" aria-hidden="true" tabindex="-1"></a>tnl <span class="ot">&lt;-</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb34-62"><a href="#cb34-62" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-63"><a href="#cb34-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-66"><a href="#cb34-66" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-67"><a href="#cb34-67" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: set-up-paths</span></span>
<span id="cb34-68"><a href="#cb34-68" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb34-69"><a href="#cb34-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-70"><a href="#cb34-70" aria-hidden="true" tabindex="-1"></a><span class="co"># Data input paths</span></span>
<span id="cb34-71"><a href="#cb34-71" aria-hidden="true" tabindex="-1"></a>data_dir <span class="ot">&lt;-</span> <span class="st">"/Users/harmonbhasin/work/securebio/nao-harmon/cebriamendoza2021/analysis/"</span></span>
<span id="cb34-72"><a href="#cb34-72" aria-hidden="true" tabindex="-1"></a>input_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"input"</span>)</span>
<span id="cb34-73"><a href="#cb34-73" aria-hidden="true" tabindex="-1"></a>results_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"results"</span>)</span>
<span id="cb34-74"><a href="#cb34-74" aria-hidden="true" tabindex="-1"></a>qc_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(results_dir, <span class="st">"qc"</span>)</span>
<span id="cb34-75"><a href="#cb34-75" aria-hidden="true" tabindex="-1"></a>hv_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(results_dir, <span class="st">"hv"</span>)</span>
<span id="cb34-76"><a href="#cb34-76" aria-hidden="true" tabindex="-1"></a>libraries_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(input_dir, <span class="st">"libraries.csv"</span>)</span>
<span id="cb34-77"><a href="#cb34-77" aria-hidden="true" tabindex="-1"></a>basic_stats_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(qc_dir, <span class="st">"qc_basic_stats.tsv.gz"</span>)</span>
<span id="cb34-78"><a href="#cb34-78" aria-hidden="true" tabindex="-1"></a>adapter_stats_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(qc_dir, <span class="st">"qc_adapter_stats.tsv.gz"</span>)</span>
<span id="cb34-79"><a href="#cb34-79" aria-hidden="true" tabindex="-1"></a>quality_base_stats_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(qc_dir, <span class="st">"qc_quality_base_stats.tsv.gz"</span>)</span>
<span id="cb34-80"><a href="#cb34-80" aria-hidden="true" tabindex="-1"></a>quality_seq_stats_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(qc_dir, <span class="st">"qc_quality_sequence_stats.tsv.gz"</span>)</span>
<span id="cb34-81"><a href="#cb34-81" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-82"><a href="#cb34-82" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-83"><a href="#cb34-83" aria-hidden="true" tabindex="-1"></a>As a part of my time here, I'm exploring blood-based surveillance for novel viral pathogen detection. We've decided to make three blog posts. The first blog post will cover the physical and biological characteristics of blood, the second blog post will cover the blood sampling strategies, and the last blog post will look at the metagenomic profile of blood (which is what we're doing here, and will do for the next few posts).</span>
<span id="cb34-84"><a href="#cb34-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-85"><a href="#cb34-85" aria-hidden="true" tabindex="-1"></a>This is the first of many studies that I'll be analyzing. however I don't expect all of them to be used in the last blog post. In this post, I analyze <span class="co">[</span><span class="ot">Cebria-Mendoza 2021</span><span class="co">](https://doi.org/10.3390/v13112322)</span>, a dataset with 60 samples where each sample is a pool of 8-13 unique healthy donors, with a total of \~600 donors from Spain.</span>
<span id="cb34-86"><a href="#cb34-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-87"><a href="#cb34-87" aria-hidden="true" tabindex="-1"></a>*This notebook uses the MGS Workflow v2.2.1 (note that this is old).*</span>
<span id="cb34-88"><a href="#cb34-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-89"><a href="#cb34-89" aria-hidden="true" tabindex="-1"></a><span class="fu"># The raw data</span></span>
<span id="cb34-90"><a href="#cb34-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-91"><a href="#cb34-91" aria-hidden="true" tabindex="-1"></a><span class="fu">## About</span></span>
<span id="cb34-92"><a href="#cb34-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-93"><a href="#cb34-93" aria-hidden="true" tabindex="-1"></a><span class="co">[</span><span class="ot">This dataset</span><span class="co">](https://www.ncbi.nlm.nih.gov/bioproject/?term=PRJNA983534)</span> comprises 60 samples from plasma pools of 8-13 people from Spain. In total, 567 healthy individuals contributed to these pools. For each pooled sample, a combined DNA and RNA library preparation was performed, resulting in a single sequencing output that captures both nucleic acid types. This approach provides comprehensive genetic information but precludes separate analysis of DNA and RNA from individual samples.</span>
<span id="cb34-94"><a href="#cb34-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-97"><a href="#cb34-97" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-98"><a href="#cb34-98" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: load-libraries</span></span>
<span id="cb34-99"><a href="#cb34-99" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb34-100"><a href="#cb34-100" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-101"><a href="#cb34-101" aria-hidden="true" tabindex="-1"></a><span class="co"># Import libraries and extract metadata from sample names</span></span>
<span id="cb34-102"><a href="#cb34-102" aria-hidden="true" tabindex="-1"></a>libraries_raw <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(libraries_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb34-103"><a href="#cb34-103" aria-hidden="true" tabindex="-1"></a>meta_data <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="fu">sprintf</span>(<span class="st">'%s/SraRunTable.txt'</span>, data_dir)) <span class="sc">%&gt;%</span></span>
<span id="cb34-104"><a href="#cb34-104" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">library=</span>Run)</span>
<span id="cb34-105"><a href="#cb34-105" aria-hidden="true" tabindex="-1"></a>libraries <span class="ot">&lt;-</span> <span class="fu">left_join</span>(libraries_raw, meta_data) <span class="sc">%&gt;%</span></span>
<span id="cb34-106"><a href="#cb34-106" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(</span>
<span id="cb34-107"><a href="#cb34-107" aria-hidden="true" tabindex="-1"></a>    library,</span>
<span id="cb34-108"><a href="#cb34-108" aria-hidden="true" tabindex="-1"></a>    AvgSpotLen,</span>
<span id="cb34-109"><a href="#cb34-109" aria-hidden="true" tabindex="-1"></a>    Collection_Date,</span>
<span id="cb34-110"><a href="#cb34-110" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">Library Name</span><span class="st">`</span></span>
<span id="cb34-111"><a href="#cb34-111" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb34-112"><a href="#cb34-112" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">sample=</span>library,</span>
<span id="cb34-113"><a href="#cb34-113" aria-hidden="true" tabindex="-1"></a>    <span class="at">library=</span><span class="st">`</span><span class="at">Library Name</span><span class="st">`</span>)</span>
<span id="cb34-114"><a href="#cb34-114" aria-hidden="true" tabindex="-1"></a>supplementary_data <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(<span class="fu">sprintf</span>(<span class="st">'%s/Supplementary_Table_S1.xlsx'</span>, data_dir), <span class="at">col_names =</span> <span class="fu">c</span>(<span class="st">'library'</span>, <span class="st">'sex'</span>, <span class="st">'age'</span>))<span class="sc">%&gt;%</span> <span class="fu">slice</span>(<span class="sc">-</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb34-115"><a href="#cb34-115" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(library) <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(<span class="at">pool_size =</span> <span class="fu">n</span>())</span>
<span id="cb34-116"><a href="#cb34-116" aria-hidden="true" tabindex="-1"></a>supplementary_data<span class="sc">$</span>pool_size <span class="ot">&lt;-</span> <span class="fu">factor</span>(supplementary_data<span class="sc">$</span>pool_size)</span>
<span id="cb34-117"><a href="#cb34-117" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-118"><a href="#cb34-118" aria-hidden="true" tabindex="-1"></a><span class="co">#sp_list &lt;- paste0("SP", 1:60)</span></span>
<span id="cb34-119"><a href="#cb34-119" aria-hidden="true" tabindex="-1"></a><span class="co">#libraries$library &lt;- factor(libraries$library, levels=sp_list)</span></span>
<span id="cb34-120"><a href="#cb34-120" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-121"><a href="#cb34-121" aria-hidden="true" tabindex="-1"></a>most_to_least_reads_sample_list <span class="ot">&lt;-</span> supplementary_data <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(pool_size)) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(library)</span>
<span id="cb34-122"><a href="#cb34-122" aria-hidden="true" tabindex="-1"></a>libraries <span class="ot">&lt;-</span> <span class="fu">left_join</span>(libraries, supplementary_data, <span class="at">by =</span> <span class="st">'library'</span>)</span>
<span id="cb34-123"><a href="#cb34-123" aria-hidden="true" tabindex="-1"></a>libraries<span class="sc">$</span>library <span class="ot">&lt;-</span> <span class="fu">factor</span>(libraries<span class="sc">$</span>library, <span class="at">levels=</span>most_to_least_reads_sample_list, <span class="at">labels=</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">60</span>))</span>
<span id="cb34-124"><a href="#cb34-124" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-125"><a href="#cb34-125" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-126"><a href="#cb34-126" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-127"><a href="#cb34-127" aria-hidden="true" tabindex="-1"></a><span class="fu">### Sample + library preparation </span></span>
<span id="cb34-128"><a href="#cb34-128" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-129"><a href="#cb34-129" aria-hidden="true" tabindex="-1"></a>Paraphrased from the <span class="co">[</span><span class="ot">paper</span><span class="co">](https://doi.org/10.3390/v15071425)</span>:</span>
<span id="cb34-130"><a href="#cb34-130" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-131"><a href="#cb34-131" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; A total of 587 plasma samples from healthy donors were collected from Valencia, Spain from 15 September 2018 to 30 March 2019 and stored at −80 °C until use.</span></span>
<span id="cb34-132"><a href="#cb34-132" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; </span></span>
<span id="cb34-133"><a href="#cb34-133" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; Each of the 60 pools (SP1-SP60) was obtained by mixing 1 mL of plasma from a variable number of donors (between 8- and 13-mL total). To assess viral recovery, each pool was spiked with 103 PFU of ϕX174 </span><span class="co">[</span><span class="ot">Microviridae</span><span class="co">]</span><span class="at"> and 104 PFU of vesicular stomatitis virus (VSV) </span><span class="co">[</span><span class="ot">Rhabdoviridae</span><span class="co">]</span><span class="at">. Briefly, plasma pools were processed with 1.0 µM filters to remove cells and other non-viral particles and the filtered fractions were subject to high-speed centrifugation (87,000 g, 2 h, 4 °C), washed with PBS 1X (87,000 g, 1 h, 4 °C), and resuspended in 245 µL 1X digestion buffer. Then, 5 µL of Turbo DNase, 2 µL of Benzonase and 2 µL of micrococcal nuclease (NEB) were added to the sample to remove unprotected nucleic acids. After incubation (1 h, 37 °C), 20 µL of stop reagent was added, following the manufacturer’s instructions. Then, 240 µL supernatant was transferred to a new tube and split into two fractions: 200 µL fraction was used for RNA extraction using TRIzol LS reagent, followed by purification with the QIAamp Viral RNA Mini kit and amplification with the QuantiTect Whole Transcriptome kit (Qiagen), and 40 µL fraction was used for DNA extraction with the QIAamp Viral RNA Mini kit and amplification with the TruePrime WGA kit.</span></span>
<span id="cb34-134"><a href="#cb34-134" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; </span></span>
<span id="cb34-135"><a href="#cb34-135" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; For each pool, DNA and RNA amplification products were mixed in equimolar concentration before library preparation, which was carried out using Nextera XT DNA library preparation kit with 15 amplification cycles (Illumina, San Diego, USA), and subject to pair-end sequencing in a NextSeq device.</span></span>
<span id="cb34-136"><a href="#cb34-136" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-137"><a href="#cb34-137" aria-hidden="true" tabindex="-1"></a>*More details can be found in their [original paper](https://doi.org/10.1038%2Fs41598-021-86427-4) which outlined this protocol.*</span>
<span id="cb34-138"><a href="#cb34-138" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-139"><a href="#cb34-139" aria-hidden="true" tabindex="-1"></a><span class="fu">## Quality control</span></span>
<span id="cb34-140"><a href="#cb34-140" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-143"><a href="#cb34-143" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-144"><a href="#cb34-144" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb34-145"><a href="#cb34-145" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: read-qc-data</span></span>
<span id="cb34-146"><a href="#cb34-146" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb34-147"><a href="#cb34-147" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-148"><a href="#cb34-148" aria-hidden="true" tabindex="-1"></a><span class="co"># Import QC data</span></span>
<span id="cb34-149"><a href="#cb34-149" aria-hidden="true" tabindex="-1"></a>stages <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"raw_concat"</span>, <span class="st">"cleaned"</span>, <span class="st">"dedup"</span>, <span class="st">"ribo_initial"</span>, <span class="st">"ribo_secondary"</span>)</span>
<span id="cb34-150"><a href="#cb34-150" aria-hidden="true" tabindex="-1"></a>basic_stats <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(basic_stats_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-151"><a href="#cb34-151" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb34-152"><a href="#cb34-152" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">factor</span>(stage, <span class="at">levels =</span> stages),</span>
<span id="cb34-153"><a href="#cb34-153" aria-hidden="true" tabindex="-1"></a>         <span class="at">sample =</span> <span class="fu">fct_inorder</span>(sample))</span>
<span id="cb34-154"><a href="#cb34-154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-155"><a href="#cb34-155" aria-hidden="true" tabindex="-1"></a>adapter_stats <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(adapter_stats_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-156"><a href="#cb34-156" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb34-157"><a href="#cb34-157" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">factor</span>(stage, <span class="at">levels =</span> stages),</span>
<span id="cb34-158"><a href="#cb34-158" aria-hidden="true" tabindex="-1"></a>         <span class="at">read_pair =</span> <span class="fu">fct_inorder</span>(<span class="fu">as.character</span>(read_pair)))</span>
<span id="cb34-159"><a href="#cb34-159" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-160"><a href="#cb34-160" aria-hidden="true" tabindex="-1"></a>quality_base_stats <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(quality_base_stats_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-161"><a href="#cb34-161" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb34-162"><a href="#cb34-162" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">factor</span>(stage, <span class="at">levels =</span> stages),</span>
<span id="cb34-163"><a href="#cb34-163" aria-hidden="true" tabindex="-1"></a>         <span class="at">read_pair =</span> <span class="fu">fct_inorder</span>(<span class="fu">as.character</span>(read_pair)))</span>
<span id="cb34-164"><a href="#cb34-164" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb34-165"><a href="#cb34-165" aria-hidden="true" tabindex="-1"></a>quality_seq_stats <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(quality_seq_stats_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-166"><a href="#cb34-166" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb34-167"><a href="#cb34-167" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">factor</span>(stage, <span class="at">levels =</span> stages),</span>
<span id="cb34-168"><a href="#cb34-168" aria-hidden="true" tabindex="-1"></a>         <span class="at">read_pair =</span> <span class="fu">fct_inorder</span>(<span class="fu">as.character</span>(read_pair)))</span>
<span id="cb34-169"><a href="#cb34-169" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb34-170"><a href="#cb34-170" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb34-171"><a href="#cb34-171" aria-hidden="true" tabindex="-1"></a> <span class="co"># Filter to raw data</span></span>
<span id="cb34-172"><a href="#cb34-172" aria-hidden="true" tabindex="-1"></a> basic_stats_raw <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"raw_concat"</span>)</span>
<span id="cb34-173"><a href="#cb34-173" aria-hidden="true" tabindex="-1"></a> adapter_stats_raw <span class="ot">&lt;-</span> adapter_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"raw_concat"</span>)</span>
<span id="cb34-174"><a href="#cb34-174" aria-hidden="true" tabindex="-1"></a> quality_base_stats_raw <span class="ot">&lt;-</span> quality_base_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"raw_concat"</span>)</span>
<span id="cb34-175"><a href="#cb34-175" aria-hidden="true" tabindex="-1"></a> quality_seq_stats_raw <span class="ot">&lt;-</span> quality_seq_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"raw_concat"</span>)</span>
<span id="cb34-176"><a href="#cb34-176" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb34-177"><a href="#cb34-177" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb34-178"><a href="#cb34-178" aria-hidden="true" tabindex="-1"></a> <span class="co"># Get key values for readout</span></span>
<span id="cb34-179"><a href="#cb34-179" aria-hidden="true" tabindex="-1"></a>raw_read_counts <span class="ot">&lt;-</span> basic_stats_raw <span class="sc">%&gt;%</span> </span>
<span id="cb34-180"><a href="#cb34-180" aria-hidden="true" tabindex="-1"></a>   <span class="fu">summarize</span>(<span class="at">rmin =</span> <span class="fu">min</span>(n_read_pairs), <span class="at">rmax=</span><span class="fu">max</span>(n_read_pairs),</span>
<span id="cb34-181"><a href="#cb34-181" aria-hidden="true" tabindex="-1"></a>             <span class="at">rmean=</span><span class="fu">mean</span>(n_read_pairs), </span>
<span id="cb34-182"><a href="#cb34-182" aria-hidden="true" tabindex="-1"></a>             <span class="at">rtot =</span> <span class="fu">sum</span>(n_read_pairs),</span>
<span id="cb34-183"><a href="#cb34-183" aria-hidden="true" tabindex="-1"></a>             <span class="at">btot =</span> <span class="fu">sum</span>(n_bases_approx),</span>
<span id="cb34-184"><a href="#cb34-184" aria-hidden="true" tabindex="-1"></a>             <span class="at">dmin =</span> <span class="fu">min</span>(percent_duplicates), <span class="at">dmax=</span><span class="fu">max</span>(percent_duplicates),</span>
<span id="cb34-185"><a href="#cb34-185" aria-hidden="true" tabindex="-1"></a>             <span class="at">dmean=</span><span class="fu">mean</span>(percent_duplicates), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb34-186"><a href="#cb34-186" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-187"><a href="#cb34-187" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-188"><a href="#cb34-188" aria-hidden="true" tabindex="-1"></a>In total, these 60 samples contained 230M read pairs. The samples had 2.3M - 4.8M (mean 3.8M) read pairs each. </span>
<span id="cb34-189"><a href="#cb34-189" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-190"><a href="#cb34-190" aria-hidden="true" tabindex="-1"></a>The number of read pairs and total base pairs looks good, however the duplication rate is quite high (this can be attributed to their sample preparation). Adapter content is low (the upper limit on the plot is 1%). Read quality seems good both over all the positions as well as over the number of sequences.</span>
<span id="cb34-191"><a href="#cb34-191" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-194"><a href="#cb34-194" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-195"><a href="#cb34-195" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 9</span></span>
<span id="cb34-196"><a href="#cb34-196" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb34-197"><a href="#cb34-197" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-basic-stats</span></span>
<span id="cb34-198"><a href="#cb34-198" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-199"><a href="#cb34-199" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare data</span></span>
<span id="cb34-200"><a href="#cb34-200" aria-hidden="true" tabindex="-1"></a>basic_stats_raw_metrics <span class="ot">&lt;-</span> basic_stats_raw <span class="sc">%&gt;%</span></span>
<span id="cb34-201"><a href="#cb34-201" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(library,</span>
<span id="cb34-202"><a href="#cb34-202" aria-hidden="true" tabindex="-1"></a>         pool_size,</span>
<span id="cb34-203"><a href="#cb34-203" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at"># Read pairs</span><span class="st">`</span> <span class="ot">=</span> n_read_pairs,</span>
<span id="cb34-204"><a href="#cb34-204" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Total base pairs</span><span class="sc">\n</span><span class="at">(approx)</span><span class="st">`</span> <span class="ot">=</span> n_bases_approx,</span>
<span id="cb34-205"><a href="#cb34-205" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">% Duplicates</span><span class="sc">\n</span><span class="at">(FASTQC)</span><span class="st">`</span> <span class="ot">=</span> percent_duplicates) <span class="sc">%&gt;%</span></span>
<span id="cb34-206"><a href="#cb34-206" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>(library<span class="sc">:</span>pool_size), <span class="at">names_to =</span> <span class="st">"metric"</span>, <span class="at">values_to =</span> <span class="st">"value"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-207"><a href="#cb34-207" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">metric =</span> <span class="fu">fct_inorder</span>(metric)) </span>
<span id="cb34-208"><a href="#cb34-208" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-209"><a href="#cb34-209" aria-hidden="true" tabindex="-1"></a><span class="co"># Set up plot templates</span></span>
<span id="cb34-210"><a href="#cb34-210" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-211"><a href="#cb34-211" aria-hidden="true" tabindex="-1"></a>g_basic <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(basic_stats_raw_metrics, <span class="fu">aes</span>(<span class="at">x =</span> library, <span class="at">y =</span> value, <span class="at">fill =</span> pool_size)) <span class="sc">+</span></span>
<span id="cb34-212"><a href="#cb34-212" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position =</span> <span class="st">"dodge"</span>) <span class="sc">+</span></span>
<span id="cb34-213"><a href="#cb34-213" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>() <span class="sc">+</span></span>
<span id="cb34-214"><a href="#cb34-214" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb34-215"><a href="#cb34-215" aria-hidden="true" tabindex="-1"></a>  <span class="fu">expand_limits</span>(<span class="at">y =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">100</span>)) <span class="sc">+</span></span>
<span id="cb34-216"><a href="#cb34-216" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Accent"</span>) <span class="sc">+</span> </span>
<span id="cb34-217"><a href="#cb34-217" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(metric <span class="sc">~</span> ., <span class="at">scales =</span> <span class="st">"free"</span>, <span class="at">space =</span> <span class="st">"free_x"</span>, <span class="at">switch =</span> <span class="st">"y"</span>) <span class="sc">+</span></span>
<span id="cb34-218"><a href="#cb34-218" aria-hidden="true" tabindex="-1"></a>  theme_kit <span class="sc">+</span> <span class="fu">theme</span>(</span>
<span id="cb34-219"><a href="#cb34-219" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb34-220"><a href="#cb34-220" aria-hidden="true" tabindex="-1"></a>    <span class="at">strip.text.y =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"plain"</span>)</span>
<span id="cb34-221"><a href="#cb34-221" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb34-222"><a href="#cb34-222" aria-hidden="true" tabindex="-1"></a>g_basic</span>
<span id="cb34-223"><a href="#cb34-223" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-224"><a href="#cb34-224" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-227"><a href="#cb34-227" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-228"><a href="#cb34-228" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-raw-quality</span></span>
<span id="cb34-229"><a href="#cb34-229" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 8</span></span>
<span id="cb34-230"><a href="#cb34-230" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-231"><a href="#cb34-231" aria-hidden="true" tabindex="-1"></a><span class="co"># Set up plotting templates</span></span>
<span id="cb34-232"><a href="#cb34-232" aria-hidden="true" tabindex="-1"></a>g_qual_raw <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">linetype=</span>read_pair, <span class="at">group=</span><span class="fu">interaction</span>(sample,read_pair))) <span class="sc">+</span> </span>
<span id="cb34-233"><a href="#cb34-233" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_linetype_discrete</span>(<span class="at">name =</span> <span class="st">"Read Pair"</span>) <span class="sc">+</span></span>
<span id="cb34-234"><a href="#cb34-234" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color=</span><span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>),</span>
<span id="cb34-235"><a href="#cb34-235" aria-hidden="true" tabindex="-1"></a>         <span class="at">linetype =</span> <span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>)) <span class="sc">+</span></span>
<span id="cb34-236"><a href="#cb34-236" aria-hidden="true" tabindex="-1"></a>  theme_base</span>
<span id="cb34-237"><a href="#cb34-237" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-238"><a href="#cb34-238" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize adapters</span></span>
<span id="cb34-239"><a href="#cb34-239" aria-hidden="true" tabindex="-1"></a>g_adapters_raw <span class="ot">&lt;-</span> g_qual_raw <span class="sc">+</span> </span>
<span id="cb34-240"><a href="#cb34-240" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>pc_adapters), <span class="at">data=</span>adapter_stats_raw) <span class="sc">+</span></span>
<span id="cb34-241"><a href="#cb34-241" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Adapters"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>),</span>
<span id="cb34-242"><a href="#cb34-242" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb34-243"><a href="#cb34-243" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Position"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb34-244"><a href="#cb34-244" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">500</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb34-245"><a href="#cb34-245" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(.<span class="sc">~</span>adapter)</span>
<span id="cb34-246"><a href="#cb34-246" aria-hidden="true" tabindex="-1"></a>g_adapters_raw</span>
<span id="cb34-247"><a href="#cb34-247" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-248"><a href="#cb34-248" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize quality</span></span>
<span id="cb34-249"><a href="#cb34-249" aria-hidden="true" tabindex="-1"></a>g_quality_base_raw <span class="ot">&lt;-</span> g_qual_raw <span class="sc">+</span></span>
<span id="cb34-250"><a href="#cb34-250" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb34-251"><a href="#cb34-251" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb34-252"><a href="#cb34-252" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>mean_phred_score), <span class="at">data=</span>quality_base_stats_raw) <span class="sc">+</span></span>
<span id="cb34-253"><a href="#cb34-253" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"Mean Phred score"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">10</span>,<span class="dv">45</span>)) <span class="sc">+</span></span>
<span id="cb34-254"><a href="#cb34-254" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Position"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb34-255"><a href="#cb34-255" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">500</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb34-256"><a href="#cb34-256" aria-hidden="true" tabindex="-1"></a>g_quality_base_raw</span>
<span id="cb34-257"><a href="#cb34-257" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-258"><a href="#cb34-258" aria-hidden="true" tabindex="-1"></a>g_quality_seq_raw <span class="ot">&lt;-</span> g_qual_raw <span class="sc">+</span></span>
<span id="cb34-259"><a href="#cb34-259" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb34-260"><a href="#cb34-260" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb34-261"><a href="#cb34-261" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>mean_phred_score, <span class="at">y=</span>n_sequences), <span class="at">data=</span>quality_seq_stats_raw) <span class="sc">+</span></span>
<span id="cb34-262"><a href="#cb34-262" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Mean Phred score"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb34-263"><a href="#cb34-263" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"# Sequences"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb34-264"><a href="#cb34-264" aria-hidden="true" tabindex="-1"></a>g_quality_seq_raw</span>
<span id="cb34-265"><a href="#cb34-265" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-266"><a href="#cb34-266" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-267"><a href="#cb34-267" aria-hidden="true" tabindex="-1"></a><span class="fu"># Preprocessing</span></span>
<span id="cb34-268"><a href="#cb34-268" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-269"><a href="#cb34-269" aria-hidden="true" tabindex="-1"></a><span class="fu">## High-level metrics</span></span>
<span id="cb34-270"><a href="#cb34-270" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-271"><a href="#cb34-271" aria-hidden="true" tabindex="-1"></a>The average fraction of reads at each stage in the preprocessing pipeline is shown in the following table. On average, cleaning &amp; deduplication removed about 57% of total read pairs, primarily during duplication which makes sense given that the raw metrics show that the samples are quite high in duplication. Ribodepletion removed about 6-8% during each round.</span>
<span id="cb34-272"><a href="#cb34-272" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-275"><a href="#cb34-275" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-276"><a href="#cb34-276" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: preproc-table</span></span>
<span id="cb34-277"><a href="#cb34-277" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-278"><a href="#cb34-278" aria-hidden="true" tabindex="-1"></a><span class="co"># </span><span class="al">TODO</span><span class="co">: Group by pool size as well</span></span>
<span id="cb34-279"><a href="#cb34-279" aria-hidden="true" tabindex="-1"></a><span class="co"># Count read losses</span></span>
<span id="cb34-280"><a href="#cb34-280" aria-hidden="true" tabindex="-1"></a>n_reads_rel <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> </span>
<span id="cb34-281"><a href="#cb34-281" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, pool_size, stage, percent_duplicates, n_read_pairs) <span class="sc">%&gt;%</span></span>
<span id="cb34-282"><a href="#cb34-282" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, pool_size) <span class="sc">%&gt;%</span> </span>
<span id="cb34-283"><a href="#cb34-283" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(sample, stage) <span class="sc">%&gt;%</span></span>
<span id="cb34-284"><a href="#cb34-284" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_retained =</span> n_read_pairs <span class="sc">/</span> <span class="fu">lag</span>(n_read_pairs),</span>
<span id="cb34-285"><a href="#cb34-285" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_lost =</span> <span class="dv">1</span> <span class="sc">-</span> p_reads_retained,</span>
<span id="cb34-286"><a href="#cb34-286" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_retained_abs =</span> n_read_pairs <span class="sc">/</span> n_read_pairs[<span class="dv">1</span>],</span>
<span id="cb34-287"><a href="#cb34-287" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_lost_abs =</span> <span class="dv">1</span><span class="sc">-</span>p_reads_retained_abs,</span>
<span id="cb34-288"><a href="#cb34-288" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_lost_abs_marginal =</span> p_reads_lost_abs <span class="sc">-</span> <span class="fu">lag</span>(p_reads_lost_abs))</span>
<span id="cb34-289"><a href="#cb34-289" aria-hidden="true" tabindex="-1"></a>n_reads_rel_display <span class="ot">&lt;-</span> n_reads_rel <span class="sc">%&gt;%</span> </span>
<span id="cb34-290"><a href="#cb34-290" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">Stage=</span>stage) <span class="sc">%&gt;%</span> </span>
<span id="cb34-291"><a href="#cb34-291" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Stage) <span class="sc">%&gt;%</span> </span>
<span id="cb34-292"><a href="#cb34-292" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="st">`</span><span class="at">% Total Reads Lost (Cumulative)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">round</span>(<span class="fu">min</span>(p_reads_lost_abs<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">"-"</span>, <span class="fu">round</span>(<span class="fu">max</span>(p_reads_lost_abs<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">" (mean "</span>, <span class="fu">round</span>(<span class="fu">mean</span>(p_reads_lost_abs<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">")"</span>),</span>
<span id="cb34-293"><a href="#cb34-293" aria-hidden="true" tabindex="-1"></a>            <span class="st">`</span><span class="at">% Total Reads Lost (Marginal)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">round</span>(<span class="fu">min</span>(p_reads_lost_abs_marginal<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">"-"</span>, <span class="fu">round</span>(<span class="fu">max</span>(p_reads_lost_abs_marginal<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">" (mean "</span>, <span class="fu">round</span>(<span class="fu">mean</span>(p_reads_lost_abs_marginal<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">")"</span>), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb34-294"><a href="#cb34-294" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Stage <span class="sc">!=</span> <span class="st">"raw_concat"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-295"><a href="#cb34-295" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Stage =</span> Stage <span class="sc">%&gt;%</span> as.numeric <span class="sc">%&gt;%</span> <span class="fu">factor</span>(<span class="at">labels=</span><span class="fu">c</span>(<span class="st">"Trimming &amp; filtering"</span>, <span class="st">"Deduplication"</span>, <span class="st">"Initial ribodepletion"</span>, <span class="st">"Secondary ribodepletion"</span>)))</span>
<span id="cb34-296"><a href="#cb34-296" aria-hidden="true" tabindex="-1"></a>n_reads_rel_display</span>
<span id="cb34-297"><a href="#cb34-297" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-298"><a href="#cb34-298" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-299"><a href="#cb34-299" aria-hidden="true" tabindex="-1"></a>Adapter content stays low (it's quite strange that FASTQC shows there being library contamination post ribosomal depletion and not anytime before that, but considering how small it is, I'm inclined to ignore it). Read quality stays pretty similar over the positions, but improves over the number of sequences (predominantly from cleaning) throughout the pipeline. Duplication rates are still quite high which is a bit concerning. Mean read length stays close to expected 150. The % of ribosomal reads vary a good amount, some samples had anywhere as low as 10% ribosomal reads whereas others had up to 60% ribosomal reads. This is probably an indication that I need to do more ribosomal read removal, but I'm going to hold off on doing this.</span>
<span id="cb34-300"><a href="#cb34-300" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-303"><a href="#cb34-303" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-304"><a href="#cb34-304" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb34-305"><a href="#cb34-305" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-quality</span></span>
<span id="cb34-306"><a href="#cb34-306" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 8</span></span>
<span id="cb34-307"><a href="#cb34-307" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-308"><a href="#cb34-308" aria-hidden="true" tabindex="-1"></a>g_qual <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">linetype=</span>read_pair, <span class="at">group=</span><span class="fu">interaction</span>(sample,read_pair))) <span class="sc">+</span> </span>
<span id="cb34-309"><a href="#cb34-309" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_linetype_discrete</span>(<span class="at">name =</span> <span class="st">"Read Pair"</span>) <span class="sc">+</span></span>
<span id="cb34-310"><a href="#cb34-310" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color=</span><span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>),</span>
<span id="cb34-311"><a href="#cb34-311" aria-hidden="true" tabindex="-1"></a>         <span class="at">linetype =</span> <span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>)) <span class="sc">+</span></span>
<span id="cb34-312"><a href="#cb34-312" aria-hidden="true" tabindex="-1"></a>  theme_base</span>
<span id="cb34-313"><a href="#cb34-313" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-314"><a href="#cb34-314" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize adapters</span></span>
<span id="cb34-315"><a href="#cb34-315" aria-hidden="true" tabindex="-1"></a>g_adapters <span class="ot">&lt;-</span> g_qual <span class="sc">+</span> </span>
<span id="cb34-316"><a href="#cb34-316" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>pc_adapters), <span class="at">data=</span>adapter_stats) <span class="sc">+</span></span>
<span id="cb34-317"><a href="#cb34-317" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Adapters"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>),</span>
<span id="cb34-318"><a href="#cb34-318" aria-hidden="true" tabindex="-1"></a>                     , <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb34-319"><a href="#cb34-319" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Position"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb34-320"><a href="#cb34-320" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">140</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb34-321"><a href="#cb34-321" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(stage<span class="sc">~</span>adapter)</span>
<span id="cb34-322"><a href="#cb34-322" aria-hidden="true" tabindex="-1"></a>g_adapters</span>
<span id="cb34-323"><a href="#cb34-323" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-324"><a href="#cb34-324" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize quality</span></span>
<span id="cb34-325"><a href="#cb34-325" aria-hidden="true" tabindex="-1"></a>g_quality_base <span class="ot">&lt;-</span> g_qual <span class="sc">+</span></span>
<span id="cb34-326"><a href="#cb34-326" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb34-327"><a href="#cb34-327" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb34-328"><a href="#cb34-328" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>mean_phred_score), <span class="at">data=</span>quality_base_stats) <span class="sc">+</span></span>
<span id="cb34-329"><a href="#cb34-329" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"Mean Phred score"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">10</span>,<span class="dv">45</span>)) <span class="sc">+</span></span>
<span id="cb34-330"><a href="#cb34-330" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Position"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb34-331"><a href="#cb34-331" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">140</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb34-332"><a href="#cb34-332" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(stage<span class="sc">~</span>.)</span>
<span id="cb34-333"><a href="#cb34-333" aria-hidden="true" tabindex="-1"></a>g_quality_base</span>
<span id="cb34-334"><a href="#cb34-334" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-335"><a href="#cb34-335" aria-hidden="true" tabindex="-1"></a>g_quality_seq <span class="ot">&lt;-</span> g_qual <span class="sc">+</span></span>
<span id="cb34-336"><a href="#cb34-336" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb34-337"><a href="#cb34-337" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb34-338"><a href="#cb34-338" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>mean_phred_score, <span class="at">y=</span>n_sequences), <span class="at">data=</span>quality_seq_stats) <span class="sc">+</span></span>
<span id="cb34-339"><a href="#cb34-339" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Mean Phred score"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb34-340"><a href="#cb34-340" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"# Sequences"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb34-341"><a href="#cb34-341" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(stage<span class="sc">~</span>., <span class="at">scales =</span> <span class="st">"free_y"</span>)</span>
<span id="cb34-342"><a href="#cb34-342" aria-hidden="true" tabindex="-1"></a>g_quality_seq</span>
<span id="cb34-343"><a href="#cb34-343" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-344"><a href="#cb34-344" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-345"><a href="#cb34-345" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-348"><a href="#cb34-348" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-349"><a href="#cb34-349" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: preproc-figures</span></span>
<span id="cb34-350"><a href="#cb34-350" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb34-351"><a href="#cb34-351" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 3</span></span>
<span id="cb34-352"><a href="#cb34-352" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 6</span></span>
<span id="cb34-353"><a href="#cb34-353" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-354"><a href="#cb34-354" aria-hidden="true" tabindex="-1"></a>g_stage_trace <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(basic_stats, <span class="fu">aes</span>(<span class="at">x=</span>stage, <span class="at">group=</span>sample)) <span class="sc">+</span></span>
<span id="cb34-355"><a href="#cb34-355" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb34-356"><a href="#cb34-356" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-357"><a href="#cb34-357" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot reads over preprocessing</span></span>
<span id="cb34-358"><a href="#cb34-358" aria-hidden="true" tabindex="-1"></a>g_reads_stages <span class="ot">&lt;-</span> g_stage_trace <span class="sc">+</span></span>
<span id="cb34-359"><a href="#cb34-359" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>n_read_pairs)) <span class="sc">+</span></span>
<span id="cb34-360"><a href="#cb34-360" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"# Read pairs"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>))</span>
<span id="cb34-361"><a href="#cb34-361" aria-hidden="true" tabindex="-1"></a>g_reads_stages</span>
<span id="cb34-362"><a href="#cb34-362" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-363"><a href="#cb34-363" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot relative read losses during preprocessing</span></span>
<span id="cb34-364"><a href="#cb34-364" aria-hidden="true" tabindex="-1"></a>g_reads_rel <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(n_reads_rel, </span>
<span id="cb34-365"><a href="#cb34-365" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">aes</span>(<span class="at">x=</span>stage, <span class="at">group=</span>sample)) <span class="sc">+</span></span>
<span id="cb34-366"><a href="#cb34-366" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>p_reads_lost_abs_marginal)) <span class="sc">+</span></span>
<span id="cb34-367"><a href="#cb34-367" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"% Total Reads Lost"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), </span>
<span id="cb34-368"><a href="#cb34-368" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="cf">function</span>(x) x<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb34-369"><a href="#cb34-369" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb34-370"><a href="#cb34-370" aria-hidden="true" tabindex="-1"></a>g_reads_rel</span>
<span id="cb34-371"><a href="#cb34-371" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-372"><a href="#cb34-372" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-375"><a href="#cb34-375" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-376"><a href="#cb34-376" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: preproc-dedup</span></span>
<span id="cb34-377"><a href="#cb34-377" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 4</span></span>
<span id="cb34-378"><a href="#cb34-378" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 6</span></span>
<span id="cb34-379"><a href="#cb34-379" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-380"><a href="#cb34-380" aria-hidden="true" tabindex="-1"></a>stage_dup <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(stage) <span class="sc">%&gt;%</span> </span>
<span id="cb34-381"><a href="#cb34-381" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">dmin =</span> <span class="fu">min</span>(percent_duplicates), <span class="at">dmax=</span><span class="fu">max</span>(percent_duplicates),</span>
<span id="cb34-382"><a href="#cb34-382" aria-hidden="true" tabindex="-1"></a>            <span class="at">dmean=</span><span class="fu">mean</span>(percent_duplicates), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb34-383"><a href="#cb34-383" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-384"><a href="#cb34-384" aria-hidden="true" tabindex="-1"></a>g_dup_stages <span class="ot">&lt;-</span> g_stage_trace <span class="sc">+</span></span>
<span id="cb34-385"><a href="#cb34-385" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>percent_duplicates)) <span class="sc">+</span></span>
<span id="cb34-386"><a href="#cb34-386" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"% Duplicates"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb34-387"><a href="#cb34-387" aria-hidden="true" tabindex="-1"></a>g_dup_stages</span>
<span id="cb34-388"><a href="#cb34-388" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-389"><a href="#cb34-389" aria-hidden="true" tabindex="-1"></a>g_readlen_stages <span class="ot">&lt;-</span> g_stage_trace <span class="sc">+</span> <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>mean_seq_len)) <span class="sc">+</span></span>
<span id="cb34-390"><a href="#cb34-390" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"Mean read length (nt)"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>))</span>
<span id="cb34-391"><a href="#cb34-391" aria-hidden="true" tabindex="-1"></a>g_readlen_stages</span>
<span id="cb34-392"><a href="#cb34-392" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-393"><a href="#cb34-393" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-394"><a href="#cb34-394" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-397"><a href="#cb34-397" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-398"><a href="#cb34-398" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: ribo-frac</span></span>
<span id="cb34-399"><a href="#cb34-399" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 4</span></span>
<span id="cb34-400"><a href="#cb34-400" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 6</span></span>
<span id="cb34-401"><a href="#cb34-401" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate reads lost during ribodepletion (approximation for % ribosomal reads)</span></span>
<span id="cb34-402"><a href="#cb34-402" aria-hidden="true" tabindex="-1"></a>reads_ribo <span class="ot">&lt;-</span> n_reads_rel <span class="sc">%&gt;%</span> </span>
<span id="cb34-403"><a href="#cb34-403" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(stage <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"dedup"</span>, <span class="st">"ribo_secondary"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb34-404"><a href="#cb34-404" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb34-405"><a href="#cb34-405" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads_ribo=</span><span class="dv">1</span><span class="sc">-</span>n_read_pairs[<span class="dv">2</span>]<span class="sc">/</span>n_read_pairs[<span class="dv">1</span>], <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-406"><a href="#cb34-406" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by =</span> <span class="st">'sample'</span>)</span>
<span id="cb34-407"><a href="#cb34-407" aria-hidden="true" tabindex="-1"></a>reads_ribo_summ <span class="ot">&lt;-</span> reads_ribo <span class="sc">%&gt;%</span></span>
<span id="cb34-408"><a href="#cb34-408" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb34-409"><a href="#cb34-409" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">min=</span><span class="fu">min</span>(p_reads_ribo), <span class="at">max=</span><span class="fu">max</span>(p_reads_ribo),</span>
<span id="cb34-410"><a href="#cb34-410" aria-hidden="true" tabindex="-1"></a>            <span class="at">mean=</span><span class="fu">mean</span>(p_reads_ribo), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-411"><a href="#cb34-411" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by =</span> <span class="st">'sample'</span>)</span>
<span id="cb34-412"><a href="#cb34-412" aria-hidden="true" tabindex="-1"></a>g_reads_ribo <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(reads_ribo, </span>
<span id="cb34-413"><a href="#cb34-413" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">aes</span>(<span class="at">x=</span>library, <span class="at">y=</span>p_reads_ribo)) <span class="sc">+</span></span>
<span id="cb34-414"><a href="#cb34-414" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span> </span>
<span id="cb34-415"><a href="#cb34-415" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"Approx % ribosomal reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>),</span>
<span id="cb34-416"><a href="#cb34-416" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>)<span class="sc">+</span></span>
<span id="cb34-417"><a href="#cb34-417" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb34-418"><a href="#cb34-418" aria-hidden="true" tabindex="-1"></a>g_reads_ribo</span>
<span id="cb34-419"><a href="#cb34-419" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-420"><a href="#cb34-420" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-421"><a href="#cb34-421" aria-hidden="true" tabindex="-1"></a><span class="fu"># Taxonomic composition</span></span>
<span id="cb34-422"><a href="#cb34-422" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-423"><a href="#cb34-423" aria-hidden="true" tabindex="-1"></a><span class="fu">## High-level composition</span></span>
<span id="cb34-424"><a href="#cb34-424" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-425"><a href="#cb34-425" aria-hidden="true" tabindex="-1"></a>To assess the high-level composition of the reads, I ran the ribodepleted files through Kraken2 and summarized the results with Bracken.</span>
<span id="cb34-426"><a href="#cb34-426" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-427"><a href="#cb34-427" aria-hidden="true" tabindex="-1"></a>The groups listed below were created by Will:</span>
<span id="cb34-428"><a href="#cb34-428" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-429"><a href="#cb34-429" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Filtered (removed during cleaning)</span>
<span id="cb34-430"><a href="#cb34-430" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Duplicate (removed during deduplication)</span>
<span id="cb34-431"><a href="#cb34-431" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Ribosomal (removed during ribodepletion)</span>
<span id="cb34-432"><a href="#cb34-432" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Unassigned (non-ribosomal reads that were not assigned to any taxon by Kraken/Bracken)</span>
<span id="cb34-433"><a href="#cb34-433" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Bacterial (non-ribosomal reads assigned to the Bacteria domain by Kraken/Bracken)</span>
<span id="cb34-434"><a href="#cb34-434" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Archaeal (non-ribosomal reads assigned to the Archaea domain by Kraken/Bracken)</span>
<span id="cb34-435"><a href="#cb34-435" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Viral (non-ribosomal reads assigned to the Viruses domain by Kraken/Bracken)</span>
<span id="cb34-436"><a href="#cb34-436" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Human (non-ribosomal reads assigned to the Eukarya domain by Kraken/Bracken)</span>
<span id="cb34-437"><a href="#cb34-437" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-440"><a href="#cb34-440" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-441"><a href="#cb34-441" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: prepare-composition</span></span>
<span id="cb34-442"><a href="#cb34-442" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-443"><a href="#cb34-443" aria-hidden="true" tabindex="-1"></a>classifications <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Filtered"</span>, <span class="st">"Duplicate"</span>, <span class="st">"Ribosomal"</span>, <span class="st">"Unassigned"</span>,</span>
<span id="cb34-444"><a href="#cb34-444" aria-hidden="true" tabindex="-1"></a>                     <span class="st">"Bacterial"</span>, <span class="st">"Archaeal"</span>, <span class="st">"Viral"</span>, <span class="st">"Human"</span>)</span>
<span id="cb34-445"><a href="#cb34-445" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-446"><a href="#cb34-446" aria-hidden="true" tabindex="-1"></a><span class="co"># Import composition data</span></span>
<span id="cb34-447"><a href="#cb34-447" aria-hidden="true" tabindex="-1"></a>tax_final_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(results_dir, <span class="st">"taxonomy_final"</span>)</span>
<span id="cb34-448"><a href="#cb34-448" aria-hidden="true" tabindex="-1"></a>comp_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(tax_final_dir, <span class="st">"taxonomic_composition.tsv.gz"</span>)</span>
<span id="cb34-449"><a href="#cb34-449" aria-hidden="true" tabindex="-1"></a>comp <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(comp_path) <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(libraries, <span class="at">by =</span> <span class="st">"sample"</span>) </span>
<span id="cb34-450"><a href="#cb34-450" aria-hidden="true" tabindex="-1"></a>comp_minor <span class="ot">&lt;-</span> comp <span class="sc">%&gt;%</span> </span>
<span id="cb34-451"><a href="#cb34-451" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(classification <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Archaeal"</span>, <span class="st">"Viral"</span>, <span class="st">"Human"</span>, <span class="st">"Other"</span>))</span>
<span id="cb34-452"><a href="#cb34-452" aria-hidden="true" tabindex="-1"></a>comp_assigned <span class="ot">&lt;-</span> comp <span class="sc">%&gt;%</span></span>
<span id="cb34-453"><a href="#cb34-453" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span> classification <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Filtered"</span>, <span class="st">"Duplicate"</span>, </span>
<span id="cb34-454"><a href="#cb34-454" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">"Ribosomal"</span>, <span class="st">"Unassigned"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb34-455"><a href="#cb34-455" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb34-456"><a href="#cb34-456" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads =</span> n_reads<span class="sc">/</span><span class="fu">sum</span>(n_reads))</span>
<span id="cb34-457"><a href="#cb34-457" aria-hidden="true" tabindex="-1"></a>comp_assigned_minor <span class="ot">&lt;-</span> comp_assigned <span class="sc">%&gt;%</span> </span>
<span id="cb34-458"><a href="#cb34-458" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(classification <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Archaeal"</span>, <span class="st">"Viral"</span>, <span class="st">"Human"</span>, <span class="st">"Other"</span>))</span>
<span id="cb34-459"><a href="#cb34-459" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-460"><a href="#cb34-460" aria-hidden="true" tabindex="-1"></a><span class="co"># Summarize composition</span></span>
<span id="cb34-461"><a href="#cb34-461" aria-hidden="true" tabindex="-1"></a>read_comp_summ <span class="ot">&lt;-</span> comp <span class="sc">%&gt;%</span> </span>
<span id="cb34-462"><a href="#cb34-462" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(classification) <span class="sc">%&gt;%</span></span>
<span id="cb34-463"><a href="#cb34-463" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads =</span> <span class="fu">sum</span>(n_reads), <span class="at">.groups =</span> <span class="st">"drop_last"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-464"><a href="#cb34-464" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">n_reads =</span> <span class="fu">replace_na</span>(n_reads,<span class="dv">0</span>),</span>
<span id="cb34-465"><a href="#cb34-465" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads =</span> n_reads<span class="sc">/</span><span class="fu">sum</span>(n_reads),</span>
<span id="cb34-466"><a href="#cb34-466" aria-hidden="true" tabindex="-1"></a>         <span class="at">pc_reads =</span> p_reads<span class="sc">*</span><span class="dv">100</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-467"><a href="#cb34-467" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">classification =</span> <span class="fu">factor</span>(classification, <span class="at">levels=</span>classifications)) <span class="sc">%&gt;%</span></span>
<span id="cb34-468"><a href="#cb34-468" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(classification, n_reads, pc_reads) <span class="sc">%&gt;%</span></span>
<span id="cb34-469"><a href="#cb34-469" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">`</span><span class="at"># of reads</span><span class="st">`</span> <span class="ot">=</span> n_reads, <span class="st">"% of reads"</span> <span class="ot">=</span> pc_reads) <span class="sc">%&gt;%</span></span>
<span id="cb34-470"><a href="#cb34-470" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">% of reads</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sprintf</span>(<span class="st">"%.2f"</span>, <span class="st">`</span><span class="at">% of reads</span><span class="st">`</span>))</span>
<span id="cb34-471"><a href="#cb34-471" aria-hidden="true" tabindex="-1"></a>read_comp_summ</span>
<span id="cb34-472"><a href="#cb34-472" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-473"><a href="#cb34-473" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-474"><a href="#cb34-474" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-477"><a href="#cb34-477" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-478"><a href="#cb34-478" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-composition-all</span></span>
<span id="cb34-479"><a href="#cb34-479" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 7</span></span>
<span id="cb34-480"><a href="#cb34-480" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 8</span></span>
<span id="cb34-481"><a href="#cb34-481" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-482"><a href="#cb34-482" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare plotting templates</span></span>
<span id="cb34-483"><a href="#cb34-483" aria-hidden="true" tabindex="-1"></a>g_comp_base <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">x=</span>library, <span class="at">y=</span>p_reads, <span class="at">fill=</span>classification)) <span class="sc">+</span></span>
<span id="cb34-484"><a href="#cb34-484" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">name=</span><span class="st">"Plasma pool"</span>) <span class="sc">+</span></span>
<span id="cb34-485"><a href="#cb34-485" aria-hidden="true" tabindex="-1"></a>  theme_kit <span class="sc">+</span> </span>
<span id="cb34-486"><a href="#cb34-486" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust=</span><span class="dv">0</span>, <span class="at">face=</span><span class="st">"plain"</span>, <span class="at">size=</span><span class="fu">rel</span>(<span class="fl">1.5</span>)))</span>
<span id="cb34-487"><a href="#cb34-487" aria-hidden="true" tabindex="-1"></a>scale_y_pc_reads <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">partial</span>(scale_y_continuous, <span class="at">name =</span> <span class="st">"% Reads"</span>,</span>
<span id="cb34-488"><a href="#cb34-488" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>)</span>
<span id="cb34-489"><a href="#cb34-489" aria-hidden="true" tabindex="-1"></a>geom_comp <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">partial</span>(geom_col, <span class="at">position =</span> <span class="st">"stack"</span>, <span class="at">width =</span> <span class="dv">1</span>)</span>
<span id="cb34-490"><a href="#cb34-490" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-491"><a href="#cb34-491" aria-hidden="true" tabindex="-1"></a><span class="co"># Define a color palette for the classification</span></span>
<span id="cb34-492"><a href="#cb34-492" aria-hidden="true" tabindex="-1"></a>classification_colors <span class="ot">&lt;-</span> <span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">"Accent"</span>)</span>
<span id="cb34-493"><a href="#cb34-493" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(classification_colors) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Filtered"</span>, <span class="st">"Duplicate"</span>, <span class="st">"Ribosomal"</span>, <span class="st">"Unassigned"</span>, <span class="st">"Bacterial"</span>, <span class="st">"Archaeal"</span>, <span class="st">"Viral"</span>, <span class="st">"Human"</span>)</span>
<span id="cb34-494"><a href="#cb34-494" aria-hidden="true" tabindex="-1"></a>scale_fill_classification <span class="ot">&lt;-</span> <span class="cf">function</span>() {</span>
<span id="cb34-495"><a href="#cb34-495" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> classification_colors, <span class="at">name =</span> <span class="st">"Classification"</span>)</span>
<span id="cb34-496"><a href="#cb34-496" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb34-497"><a href="#cb34-497" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-498"><a href="#cb34-498" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot overall composition</span></span>
<span id="cb34-499"><a href="#cb34-499" aria-hidden="true" tabindex="-1"></a>g_comp <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> <span class="fu">geom_comp</span>(<span class="at">data =</span> comp) <span class="sc">+</span></span>
<span id="cb34-500"><a href="#cb34-500" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_pc_reads</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>)) <span class="sc">+</span></span>
<span id="cb34-501"><a href="#cb34-501" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_classification</span>() <span class="sc">+</span> </span>
<span id="cb34-502"><a href="#cb34-502" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Read composition (all reads, all groups)"</span>)</span>
<span id="cb34-503"><a href="#cb34-503" aria-hidden="true" tabindex="-1"></a>g_comp</span>
<span id="cb34-504"><a href="#cb34-504" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-505"><a href="#cb34-505" aria-hidden="true" tabindex="-1"></a>g_comp_assigned <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb34-506"><a href="#cb34-506" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_comp</span>(<span class="at">data =</span> comp_assigned) <span class="sc">+</span></span>
<span id="cb34-507"><a href="#cb34-507" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_pc_reads</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>)) <span class="sc">+</span></span>
<span id="cb34-508"><a href="#cb34-508" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_classification</span>() <span class="sc">+</span> </span>
<span id="cb34-509"><a href="#cb34-509" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Read composition (assigned reads, all groups)"</span>)</span>
<span id="cb34-510"><a href="#cb34-510" aria-hidden="true" tabindex="-1"></a>g_comp_assigned</span>
<span id="cb34-511"><a href="#cb34-511" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-512"><a href="#cb34-512" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-513"><a href="#cb34-513" aria-hidden="true" tabindex="-1"></a><span class="fu">## Total viral content</span></span>
<span id="cb34-514"><a href="#cb34-514" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-515"><a href="#cb34-515" aria-hidden="true" tabindex="-1"></a>Total viral fraction average $1.81 \times 10^{-2}$ across samples. As a fraction of assigned (rather than total) reads, this jumped to $1.60 \times 10^{-1}$.</span>
<span id="cb34-516"><a href="#cb34-516" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-519"><a href="#cb34-519" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-520"><a href="#cb34-520" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: p-viral</span></span>
<span id="cb34-521"><a href="#cb34-521" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb34-522"><a href="#cb34-522" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 10 </span></span>
<span id="cb34-523"><a href="#cb34-523" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb34-524"><a href="#cb34-524" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-525"><a href="#cb34-525" aria-hidden="true" tabindex="-1"></a>p_reads_viral_all <span class="ot">&lt;-</span> comp <span class="sc">%&gt;%</span> <span class="fu">filter</span>(classification <span class="sc">==</span> <span class="st">"Viral"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-526"><a href="#cb34-526" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">read_group =</span> <span class="st">"All reads"</span>)</span>
<span id="cb34-527"><a href="#cb34-527" aria-hidden="true" tabindex="-1"></a>p_reads_viral_assigned <span class="ot">&lt;-</span> comp_assigned <span class="sc">%&gt;%</span> <span class="fu">filter</span>(classification <span class="sc">==</span> <span class="st">"Viral"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-528"><a href="#cb34-528" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">read_group =</span> <span class="st">"Classified reads"</span>)</span>
<span id="cb34-529"><a href="#cb34-529" aria-hidden="true" tabindex="-1"></a>p_reads_viral <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(p_reads_viral_all, p_reads_viral_assigned)</span>
<span id="cb34-530"><a href="#cb34-530" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-531"><a href="#cb34-531" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb34-532"><a href="#cb34-532" aria-hidden="true" tabindex="-1"></a>g_viral <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(p_reads_viral, <span class="fu">aes</span>(<span class="at">x=</span>library, <span class="at">y=</span>p_reads, <span class="at">color =</span> read_group, <span class="at">group =</span> library)) <span class="sc">+</span></span>
<span id="cb34-533"><a href="#cb34-533" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb34-534"><a href="#cb34-534" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">color =</span> <span class="st">'black'</span>) <span class="sc">+</span></span>
<span id="cb34-535"><a href="#cb34-535" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">name=</span><span class="st">"Plasma pool"</span>) <span class="sc">+</span></span>
<span id="cb34-536"><a href="#cb34-536" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>(<span class="at">name=</span><span class="st">"Viral read fraction"</span>, <span class="at">labels =</span> <span class="fu">label_log</span>(<span class="at">digits=</span><span class="dv">2</span>)) <span class="sc">+</span></span>
<span id="cb34-537"><a href="#cb34-537" aria-hidden="true" tabindex="-1"></a>  theme_kit <span class="sc">+</span> </span>
<span id="cb34-538"><a href="#cb34-538" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>()</span>
<span id="cb34-539"><a href="#cb34-539" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-540"><a href="#cb34-540" aria-hidden="true" tabindex="-1"></a>g_viral</span>
<span id="cb34-541"><a href="#cb34-541" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-542"><a href="#cb34-542" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-543"><a href="#cb34-543" aria-hidden="true" tabindex="-1"></a><span class="fu">## Taxonomic composition of viruses</span></span>
<span id="cb34-544"><a href="#cb34-544" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-545"><a href="#cb34-545" aria-hidden="true" tabindex="-1"></a>The two dominant viruses we see are Anellovirdae and Rhabdovirdae (spike-in). Followed by these two viral families is Flavivirdae, and lastly, also by a much smaller percent, Microviridae (spike-in). The threshold for the label "other" are the set of families that make up less than 1% composition in all samples.</span>
<span id="cb34-546"><a href="#cb34-546" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-549"><a href="#cb34-549" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-550"><a href="#cb34-550" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: extract-viral-taxa-second</span></span>
<span id="cb34-551"><a href="#cb34-551" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-552"><a href="#cb34-552" aria-hidden="true" tabindex="-1"></a><span class="co"># Get viral taxonomy</span></span>
<span id="cb34-553"><a href="#cb34-553" aria-hidden="true" tabindex="-1"></a>viral_taxa_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"total-virus-db.tsv.gz"</span>)</span>
<span id="cb34-554"><a href="#cb34-554" aria-hidden="true" tabindex="-1"></a>viral_taxa <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(viral_taxa_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb34-555"><a href="#cb34-555" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-556"><a href="#cb34-556" aria-hidden="true" tabindex="-1"></a><span class="co"># Get Kraken reports</span></span>
<span id="cb34-557"><a href="#cb34-557" aria-hidden="true" tabindex="-1"></a>reports_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(tax_final_dir, <span class="st">"kraken_reports.tsv.gz"</span>)</span>
<span id="cb34-558"><a href="#cb34-558" aria-hidden="true" tabindex="-1"></a>reports <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(reports_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-559"><a href="#cb34-559" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(sample)</span>
<span id="cb34-560"><a href="#cb34-560" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-561"><a href="#cb34-561" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter to viral taxa</span></span>
<span id="cb34-562"><a href="#cb34-562" aria-hidden="true" tabindex="-1"></a>kraken_reports_viral <span class="ot">&lt;-</span> <span class="fu">filter</span>(reports, taxid <span class="sc">%in%</span> viral_taxa<span class="sc">$</span>taxid) <span class="sc">%&gt;%</span></span>
<span id="cb34-563"><a href="#cb34-563" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb34-564"><a href="#cb34-564" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_viral =</span> n_reads_clade<span class="sc">/</span>n_reads_clade[<span class="dv">1</span>])</span>
<span id="cb34-565"><a href="#cb34-565" aria-hidden="true" tabindex="-1"></a>kraken_reports_viral_cleaned <span class="ot">&lt;-</span> kraken_reports_viral <span class="sc">%&gt;%</span></span>
<span id="cb34-566"><a href="#cb34-566" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>pc_reads_total, <span class="sc">-</span>n_reads_direct, <span class="sc">-</span><span class="fu">contains</span>(<span class="st">"minimizers"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb34-567"><a href="#cb34-567" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, taxid, p_reads_viral, n_reads_clade, <span class="fu">everything</span>()) <span class="sc">%&gt;%</span> ungroup</span>
<span id="cb34-568"><a href="#cb34-568" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-569"><a href="#cb34-569" aria-hidden="true" tabindex="-1"></a>viral_classes <span class="ot">&lt;-</span> kraken_reports_viral_cleaned <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">==</span> <span class="st">"C"</span>)</span>
<span id="cb34-570"><a href="#cb34-570" aria-hidden="true" tabindex="-1"></a>viral_families <span class="ot">&lt;-</span> kraken_reports_viral_cleaned <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">==</span> <span class="st">"F"</span>)</span>
<span id="cb34-571"><a href="#cb34-571" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-572"><a href="#cb34-572" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-573"><a href="#cb34-573" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-576"><a href="#cb34-576" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-577"><a href="#cb34-577" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: viral-family-composition</span></span>
<span id="cb34-578"><a href="#cb34-578" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 4</span></span>
<span id="cb34-579"><a href="#cb34-579" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 8</span></span>
<span id="cb34-580"><a href="#cb34-580" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-581"><a href="#cb34-581" aria-hidden="true" tabindex="-1"></a>major_threshold <span class="ot">&lt;-</span> <span class="fl">0.01</span></span>
<span id="cb34-582"><a href="#cb34-582" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-583"><a href="#cb34-583" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify major viral families</span></span>
<span id="cb34-584"><a href="#cb34-584" aria-hidden="true" tabindex="-1"></a>viral_families_major_tab <span class="ot">&lt;-</span> viral_families <span class="sc">%&gt;%</span> </span>
<span id="cb34-585"><a href="#cb34-585" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb34-586"><a href="#cb34-586" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads_viral_max =</span> <span class="fu">max</span>(p_reads_viral), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-587"><a href="#cb34-587" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_viral_max <span class="sc">&gt;=</span> major_threshold)</span>
<span id="cb34-588"><a href="#cb34-588" aria-hidden="true" tabindex="-1"></a>viral_families_major_list <span class="ot">&lt;-</span> viral_families_major_tab <span class="sc">%&gt;%</span> <span class="fu">pull</span>(name)</span>
<span id="cb34-589"><a href="#cb34-589" aria-hidden="true" tabindex="-1"></a>viral_families_major <span class="ot">&lt;-</span> viral_families <span class="sc">%&gt;%</span> </span>
<span id="cb34-590"><a href="#cb34-590" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(name <span class="sc">%in%</span> viral_families_major_list) <span class="sc">%&gt;%</span></span>
<span id="cb34-591"><a href="#cb34-591" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, taxid, sample, p_reads_viral)</span>
<span id="cb34-592"><a href="#cb34-592" aria-hidden="true" tabindex="-1"></a>viral_families_minor <span class="ot">&lt;-</span> viral_families_major <span class="sc">%&gt;%</span> </span>
<span id="cb34-593"><a href="#cb34-593" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb34-594"><a href="#cb34-594" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads_viral_major =</span> <span class="fu">sum</span>(p_reads_viral), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-595"><a href="#cb34-595" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="st">"Other"</span>, <span class="at">taxid=</span><span class="cn">NA</span>, <span class="at">p_reads_viral =</span> <span class="dv">1</span><span class="sc">-</span>p_reads_viral_major) <span class="sc">%&gt;%</span></span>
<span id="cb34-596"><a href="#cb34-596" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, taxid, sample, p_reads_viral)</span>
<span id="cb34-597"><a href="#cb34-597" aria-hidden="true" tabindex="-1"></a>viral_families_display <span class="ot">&lt;-</span> viral_families_major <span class="sc">%&gt;%</span> </span>
<span id="cb34-598"><a href="#cb34-598" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_rows</span>(viral_families_minor) <span class="sc">%&gt;%</span></span>
<span id="cb34-599"><a href="#cb34-599" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_viral)) <span class="sc">%&gt;%</span> </span>
<span id="cb34-600"><a href="#cb34-600" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">factor</span>(name, <span class="at">levels=</span><span class="fu">c</span>(viral_families_major_list, <span class="st">"Other"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb34-601"><a href="#cb34-601" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_viral, <span class="at">classification=</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb34-602"><a href="#cb34-602" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">'sample'</span>)</span>
<span id="cb34-603"><a href="#cb34-603" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-604"><a href="#cb34-604" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb34-605"><a href="#cb34-605" aria-hidden="true" tabindex="-1"></a>g_families <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb34-606"><a href="#cb34-606" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_comp</span>(<span class="at">data=</span>viral_families_display) <span class="sc">+</span></span>
<span id="cb34-607"><a href="#cb34-607" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Viral Reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb34-608"><a href="#cb34-608" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb34-609"><a href="#cb34-609" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb34-610"><a href="#cb34-610" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">'Accent'</span>)</span>
<span id="cb34-611"><a href="#cb34-611" aria-hidden="true" tabindex="-1"></a>g_families</span>
<span id="cb34-612"><a href="#cb34-612" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-613"><a href="#cb34-613" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-614"><a href="#cb34-614" aria-hidden="true" tabindex="-1"></a>Excluding *Microviridae* and *Rhabdovirdae* (spike-ins), remaining viral sequences are distributed across a wide variety:</span>
<span id="cb34-615"><a href="#cb34-615" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-618"><a href="#cb34-618" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-619"><a href="#cb34-619" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: viral-family-composition-exclusion</span></span>
<span id="cb34-620"><a href="#cb34-620" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 4</span></span>
<span id="cb34-621"><a href="#cb34-621" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 8</span></span>
<span id="cb34-622"><a href="#cb34-622" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-623"><a href="#cb34-623" aria-hidden="true" tabindex="-1"></a>major_threshold_adj <span class="ot">&lt;-</span> <span class="fl">0.05</span></span>
<span id="cb34-624"><a href="#cb34-624" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-625"><a href="#cb34-625" aria-hidden="true" tabindex="-1"></a><span class="co"># Adjust viral family counts</span></span>
<span id="cb34-626"><a href="#cb34-626" aria-hidden="true" tabindex="-1"></a>viral_families_adj <span class="ot">&lt;-</span> viral_families <span class="sc">%&gt;%</span></span>
<span id="cb34-627"><a href="#cb34-627" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span>(name <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Rhabdoviridae"</span>,<span class="st">"Microviridae"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb34-628"><a href="#cb34-628" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb34-629"><a href="#cb34-629" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_viral =</span> p_reads_viral<span class="sc">/</span><span class="fu">sum</span>(p_reads_viral))</span>
<span id="cb34-630"><a href="#cb34-630" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-631"><a href="#cb34-631" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify major viral families</span></span>
<span id="cb34-632"><a href="#cb34-632" aria-hidden="true" tabindex="-1"></a>viral_families_major_tab <span class="ot">&lt;-</span> viral_families_adj <span class="sc">%&gt;%</span> </span>
<span id="cb34-633"><a href="#cb34-633" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb34-634"><a href="#cb34-634" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads_viral_max =</span> <span class="fu">max</span>(p_reads_viral), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-635"><a href="#cb34-635" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_viral_max <span class="sc">&gt;=</span> major_threshold)</span>
<span id="cb34-636"><a href="#cb34-636" aria-hidden="true" tabindex="-1"></a>viral_families_major_list <span class="ot">&lt;-</span> viral_families_major_tab <span class="sc">%&gt;%</span> <span class="fu">pull</span>(name)</span>
<span id="cb34-637"><a href="#cb34-637" aria-hidden="true" tabindex="-1"></a>viral_families_major <span class="ot">&lt;-</span> viral_families_adj <span class="sc">%&gt;%</span> </span>
<span id="cb34-638"><a href="#cb34-638" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(name <span class="sc">%in%</span> viral_families_major_list) <span class="sc">%&gt;%</span></span>
<span id="cb34-639"><a href="#cb34-639" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, taxid, sample, p_reads_viral)</span>
<span id="cb34-640"><a href="#cb34-640" aria-hidden="true" tabindex="-1"></a>viral_families_minor <span class="ot">&lt;-</span> viral_families_major <span class="sc">%&gt;%</span> </span>
<span id="cb34-641"><a href="#cb34-641" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb34-642"><a href="#cb34-642" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads_viral_major =</span> <span class="fu">sum</span>(p_reads_viral), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-643"><a href="#cb34-643" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="st">"Other"</span>, <span class="at">taxid=</span><span class="cn">NA</span>, <span class="at">p_reads_viral =</span> <span class="dv">1</span><span class="sc">-</span>p_reads_viral_major) <span class="sc">%&gt;%</span></span>
<span id="cb34-644"><a href="#cb34-644" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, taxid, sample, p_reads_viral)</span>
<span id="cb34-645"><a href="#cb34-645" aria-hidden="true" tabindex="-1"></a>viral_families_display <span class="ot">&lt;-</span> viral_families_major <span class="sc">%&gt;%</span> </span>
<span id="cb34-646"><a href="#cb34-646" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_rows</span>(viral_families_minor) <span class="sc">%&gt;%</span></span>
<span id="cb34-647"><a href="#cb34-647" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_viral)) <span class="sc">%&gt;%</span> </span>
<span id="cb34-648"><a href="#cb34-648" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">factor</span>(name, <span class="at">levels=</span><span class="fu">c</span>(viral_families_major_list, <span class="st">"Other"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb34-649"><a href="#cb34-649" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_viral, <span class="at">classification=</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb34-650"><a href="#cb34-650" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by =</span> <span class="st">'sample'</span>)</span>
<span id="cb34-651"><a href="#cb34-651" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-652"><a href="#cb34-652" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb34-653"><a href="#cb34-653" aria-hidden="true" tabindex="-1"></a>palette_viral <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">brewer.pal</span>(<span class="dv">12</span>, <span class="st">"Set3"</span>), <span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">"Dark2"</span>), <span class="fu">brewer.pal</span>(<span class="dv">9</span>, <span class="st">"Set1"</span>))</span>
<span id="cb34-654"><a href="#cb34-654" aria-hidden="true" tabindex="-1"></a>g_families_adj <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb34-655"><a href="#cb34-655" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_comp</span>(<span class="at">data=</span>viral_families_display) <span class="sc">+</span></span>
<span id="cb34-656"><a href="#cb34-656" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Viral Reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb34-657"><a href="#cb34-657" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb34-658"><a href="#cb34-658" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb34-659"><a href="#cb34-659" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>palette_viral, <span class="at">name =</span> <span class="st">"Viral class"</span>)</span>
<span id="cb34-660"><a href="#cb34-660" aria-hidden="true" tabindex="-1"></a>g_families_adj</span>
<span id="cb34-661"><a href="#cb34-661" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-662"><a href="#cb34-662" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-663"><a href="#cb34-663" aria-hidden="true" tabindex="-1"></a><span class="fu"># Human-infecting virus reads</span></span>
<span id="cb34-664"><a href="#cb34-664" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-665"><a href="#cb34-665" aria-hidden="true" tabindex="-1"></a><span class="fu">## Overall relative abundance</span></span>
<span id="cb34-666"><a href="#cb34-666" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-667"><a href="#cb34-667" aria-hidden="true" tabindex="-1"></a>I calculated the relative abundance of human-infecting viruses in two ways:</span>
<span id="cb34-668"><a href="#cb34-668" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-669"><a href="#cb34-669" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>First, as the total number of deduplicated human-virus reads in each sample, divided by the number of raw reads ("All reads"). This results in a viral fraction of $4.06 \times 10^{-2}$ across all samples</span>
<span id="cb34-670"><a href="#cb34-670" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-671"><a href="#cb34-671" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Second, as a fraction of preprocessed (cleaned, deduplicated, computationally ribodepleted) reads ("Preprocessed reads"). This results in a viral fraction of $1.11 \times 10^{-1}$ across all samples.</span>
<span id="cb34-672"><a href="#cb34-672" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-675"><a href="#cb34-675" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-676"><a href="#cb34-676" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: prepare-hv</span></span>
<span id="cb34-677"><a href="#cb34-677" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb34-678"><a href="#cb34-678" aria-hidden="true" tabindex="-1"></a><span class="co">#| cache.lazy: false</span></span>
<span id="cb34-679"><a href="#cb34-679" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-680"><a href="#cb34-680" aria-hidden="true" tabindex="-1"></a><span class="co"># Import and format reads</span></span>
<span id="cb34-681"><a href="#cb34-681" aria-hidden="true" tabindex="-1"></a>hv_reads_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(hv_dir, <span class="st">"hv_hits_putative_collapsed.tsv.gz"</span>)</span>
<span id="cb34-682"><a href="#cb34-682" aria-hidden="true" tabindex="-1"></a>mrg_hv <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(hv_reads_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-683"><a href="#cb34-683" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb34-684"><a href="#cb34-684" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">kraken_label =</span> <span class="fu">ifelse</span>(assigned_hv, <span class="st">"Kraken2 HV assignment"</span>,</span>
<span id="cb34-685"><a href="#cb34-685" aria-hidden="true" tabindex="-1"></a>                               <span class="st">"No Kraken2 assignment"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb34-686"><a href="#cb34-686" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">adj_score_max =</span> <span class="fu">pmax</span>(adj_score_fwd, adj_score_rev),</span>
<span id="cb34-687"><a href="#cb34-687" aria-hidden="true" tabindex="-1"></a>         <span class="at">highscore =</span> adj_score_max <span class="sc">&gt;=</span> <span class="dv">20</span>,</span>
<span id="cb34-688"><a href="#cb34-688" aria-hidden="true" tabindex="-1"></a>         <span class="at">hv_status =</span> assigned_hv <span class="sc">|</span> highscore) <span class="sc">%&gt;%</span></span>
<span id="cb34-689"><a href="#cb34-689" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">taxid_all =</span> taxid, <span class="at">taxid =</span> taxid_best)</span>
<span id="cb34-690"><a href="#cb34-690" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-691"><a href="#cb34-691" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-694"><a href="#cb34-694" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-695"><a href="#cb34-695" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: count-hv-reads</span></span>
<span id="cb34-696"><a href="#cb34-696" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb34-697"><a href="#cb34-697" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 10 </span></span>
<span id="cb34-698"><a href="#cb34-698" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb34-699"><a href="#cb34-699" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb34-700"><a href="#cb34-700" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb34-701"><a href="#cb34-701" aria-hidden="true" tabindex="-1"></a><span class="co"># Get raw read counts</span></span>
<span id="cb34-702"><a href="#cb34-702" aria-hidden="true" tabindex="-1"></a>read_counts_raw <span class="ot">&lt;-</span> <span class="fu">filter</span>(basic_stats_raw) <span class="sc">%&gt;%</span></span>
<span id="cb34-703"><a href="#cb34-703" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, <span class="at">n_reads_raw =</span> n_read_pairs)</span>
<span id="cb34-704"><a href="#cb34-704" aria-hidden="true" tabindex="-1"></a>read_counts_preproc <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"ribo_initial"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-705"><a href="#cb34-705" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, <span class="at">n_reads_preproc =</span> n_read_pairs)</span>
<span id="cb34-706"><a href="#cb34-706" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-707"><a href="#cb34-707" aria-hidden="true" tabindex="-1"></a><span class="co"># Get HV read counts</span></span>
<span id="cb34-708"><a href="#cb34-708" aria-hidden="true" tabindex="-1"></a>read_counts_hv <span class="ot">&lt;-</span> mrg_hv <span class="sc">%&gt;%</span> <span class="fu">filter</span>(hv_status) <span class="sc">%&gt;%</span> </span>
<span id="cb34-709"><a href="#cb34-709" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb34-710"><a href="#cb34-710" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name=</span><span class="st">"n_reads_hv"</span>)</span>
<span id="cb34-711"><a href="#cb34-711" aria-hidden="true" tabindex="-1"></a>read_counts <span class="ot">&lt;-</span> read_counts_raw <span class="sc">%&gt;%</span></span>
<span id="cb34-712"><a href="#cb34-712" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(read_counts_hv, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"sample"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb34-713"><a href="#cb34-713" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">n_reads_hv =</span> <span class="fu">replace_na</span>(n_reads_hv, <span class="dv">0</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb34-714"><a href="#cb34-714" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(read_counts_preproc, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"sample"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb34-715"><a href="#cb34-715" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"sample"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb34-716"><a href="#cb34-716" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, n_reads_raw, n_reads_preproc, n_reads_hv) <span class="sc">%&gt;%</span></span>
<span id="cb34-717"><a href="#cb34-717" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">n_samples =</span> <span class="dv">1</span>,</span>
<span id="cb34-718"><a href="#cb34-718" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_total =</span> n_reads_hv<span class="sc">/</span>n_reads_raw,</span>
<span id="cb34-719"><a href="#cb34-719" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_preproc =</span> n_reads_hv<span class="sc">/</span>n_reads_preproc)</span>
<span id="cb34-720"><a href="#cb34-720" aria-hidden="true" tabindex="-1"></a>read_counts_long <span class="ot">&lt;-</span> read_counts <span class="sc">%&gt;%</span></span>
<span id="cb34-721"><a href="#cb34-721" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="fu">starts_with</span>(<span class="st">"p_reads"</span>), <span class="at">names_to=</span><span class="st">"read_group"</span>, <span class="at">values_to=</span><span class="st">"p_reads"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-722"><a href="#cb34-722" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">read_group =</span> <span class="fu">ifelse</span>(read_group <span class="sc">==</span> <span class="st">"p_reads_total"</span>, <span class="st">"All reads"</span>, <span class="st">"Preprocessed reads"</span>))</span>
<span id="cb34-723"><a href="#cb34-723" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-724"><a href="#cb34-724" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine for display</span></span>
<span id="cb34-725"><a href="#cb34-725" aria-hidden="true" tabindex="-1"></a>read_counts_agg <span class="ot">&lt;-</span> read_counts <span class="sc">%&gt;%</span></span>
<span id="cb34-726"><a href="#cb34-726" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_total =</span> n_reads_hv<span class="sc">/</span>n_reads_raw,</span>
<span id="cb34-727"><a href="#cb34-727" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_preproc =</span> n_reads_hv<span class="sc">/</span>n_reads_preproc) <span class="sc">%&gt;%</span></span>
<span id="cb34-728"><a href="#cb34-728" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"sample"</span>))</span>
<span id="cb34-729"><a href="#cb34-729" aria-hidden="true" tabindex="-1"></a>read_counts_agg_long <span class="ot">&lt;-</span> read_counts_agg <span class="sc">%&gt;%</span></span>
<span id="cb34-730"><a href="#cb34-730" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="fu">starts_with</span>(<span class="st">"p_reads"</span>), <span class="at">names_to=</span><span class="st">"read_group"</span>, <span class="at">values_to=</span><span class="st">"p_reads"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-731"><a href="#cb34-731" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">read_group =</span> <span class="fu">ifelse</span>(read_group <span class="sc">==</span> <span class="st">"p_reads_total"</span>, <span class="st">"All reads"</span>, <span class="st">"Preprocessed reads"</span>)) </span>
<span id="cb34-732"><a href="#cb34-732" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-733"><a href="#cb34-733" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize</span></span>
<span id="cb34-734"><a href="#cb34-734" aria-hidden="true" tabindex="-1"></a>g_read_counts <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(read_counts_agg_long, <span class="fu">aes</span>(<span class="at">x=</span>library, <span class="at">y=</span>p_reads, <span class="at">color =</span> read_group, <span class="at">group =</span> library)) <span class="sc">+</span></span>
<span id="cb34-735"><a href="#cb34-735" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb34-736"><a href="#cb34-736" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">color =</span> <span class="st">'black'</span>) <span class="sc">+</span></span>
<span id="cb34-737"><a href="#cb34-737" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>(<span class="at">name =</span> <span class="st">"Unique human-viral read fraction"</span>, <span class="at">labels =</span> <span class="fu">label_log</span>(<span class="at">digits=</span><span class="dv">2</span>)) <span class="sc">+</span></span>
<span id="cb34-738"><a href="#cb34-738" aria-hidden="true" tabindex="-1"></a>  theme_kit <span class="sc">+</span> </span>
<span id="cb34-739"><a href="#cb34-739" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>()</span>
<span id="cb34-740"><a href="#cb34-740" aria-hidden="true" tabindex="-1"></a>g_read_counts</span>
<span id="cb34-741"><a href="#cb34-741" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-742"><a href="#cb34-742" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-743"><a href="#cb34-743" aria-hidden="true" tabindex="-1"></a><span class="fu">## Overall taxonomy and composition</span></span>
<span id="cb34-744"><a href="#cb34-744" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-745"><a href="#cb34-745" aria-hidden="true" tabindex="-1"></a>Composition of HV reads was not greatly changed from when looking at all viral reads. The two dominant viruses we see are Anellovirdae and Rhabdovirdae. Followed by these two viral families is Flavivirdae, and lastly, also by a much smaller percent, Microviridae. The threshold for the label "other" are the set of families that make up less than 5% composition in all samples.</span>
<span id="cb34-746"><a href="#cb34-746" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-749"><a href="#cb34-749" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-750"><a href="#cb34-750" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: raise-hv-taxa</span></span>
<span id="cb34-751"><a href="#cb34-751" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb34-752"><a href="#cb34-752" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb34-753"><a href="#cb34-753" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter samples and add viral taxa information</span></span>
<span id="cb34-754"><a href="#cb34-754" aria-hidden="true" tabindex="-1"></a>samples_keep <span class="ot">&lt;-</span> read_counts <span class="sc">%&gt;%</span> <span class="fu">filter</span>(n_reads_hv <span class="sc">&gt;</span> <span class="dv">5</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(sample)</span>
<span id="cb34-755"><a href="#cb34-755" aria-hidden="true" tabindex="-1"></a>mrg_hv_named <span class="ot">&lt;-</span> mrg_hv <span class="sc">%&gt;%</span> <span class="fu">filter</span>(sample <span class="sc">%in%</span> samples_keep, hv_status) <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(viral_taxa, <span class="at">by=</span><span class="st">"taxid"</span>) </span>
<span id="cb34-756"><a href="#cb34-756" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-757"><a href="#cb34-757" aria-hidden="true" tabindex="-1"></a><span class="co"># Discover viral species &amp; genera for HV reads</span></span>
<span id="cb34-758"><a href="#cb34-758" aria-hidden="true" tabindex="-1"></a>raise_rank <span class="ot">&lt;-</span> <span class="cf">function</span>(read_db, taxid_db, <span class="at">out_rank =</span> <span class="st">"species"</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>){</span>
<span id="cb34-759"><a href="#cb34-759" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get higher ranks than search rank</span></span>
<span id="cb34-760"><a href="#cb34-760" aria-hidden="true" tabindex="-1"></a>  ranks <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"subspecies"</span>, <span class="st">"species"</span>, <span class="st">"subgenus"</span>, <span class="st">"genus"</span>, <span class="st">"subfamily"</span>, <span class="st">"family"</span>, <span class="st">"suborder"</span>, <span class="st">"order"</span>, <span class="st">"class"</span>, <span class="st">"subphylum"</span>, <span class="st">"phylum"</span>, <span class="st">"kingdom"</span>, <span class="st">"superkingdom"</span>)</span>
<span id="cb34-761"><a href="#cb34-761" aria-hidden="true" tabindex="-1"></a>  rank_match <span class="ot">&lt;-</span> <span class="fu">which.max</span>(ranks <span class="sc">==</span> out_rank)</span>
<span id="cb34-762"><a href="#cb34-762" aria-hidden="true" tabindex="-1"></a>  high_ranks <span class="ot">&lt;-</span> ranks[rank_match<span class="sc">:</span><span class="fu">length</span>(ranks)]</span>
<span id="cb34-763"><a href="#cb34-763" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Merge read DB and taxid DB</span></span>
<span id="cb34-764"><a href="#cb34-764" aria-hidden="true" tabindex="-1"></a>  reads <span class="ot">&lt;-</span> read_db <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>parent_taxid, <span class="sc">-</span>rank, <span class="sc">-</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb34-765"><a href="#cb34-765" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(taxid_db, <span class="at">by=</span><span class="st">"taxid"</span>)</span>
<span id="cb34-766"><a href="#cb34-766" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract sequences that are already at appropriate rank</span></span>
<span id="cb34-767"><a href="#cb34-767" aria-hidden="true" tabindex="-1"></a>  reads_rank <span class="ot">&lt;-</span> <span class="fu">filter</span>(reads, rank <span class="sc">==</span> out_rank)</span>
<span id="cb34-768"><a href="#cb34-768" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Drop sequences at a higher rank and return unclassified sequences</span></span>
<span id="cb34-769"><a href="#cb34-769" aria-hidden="true" tabindex="-1"></a>  reads_norank <span class="ot">&lt;-</span> reads <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">!=</span> out_rank, <span class="sc">!</span>rank <span class="sc">%in%</span> high_ranks, <span class="sc">!</span><span class="fu">is.na</span>(taxid))</span>
<span id="cb34-770"><a href="#cb34-770" aria-hidden="true" tabindex="-1"></a>  <span class="cf">while</span>(<span class="fu">nrow</span>(reads_norank) <span class="sc">&gt;</span> <span class="dv">0</span>){ <span class="co"># As long as there are unclassified sequences...</span></span>
<span id="cb34-771"><a href="#cb34-771" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Promote read taxids and re-merge with taxid DB, then re-classify and filter</span></span>
<span id="cb34-772"><a href="#cb34-772" aria-hidden="true" tabindex="-1"></a>    reads_remaining <span class="ot">&lt;-</span> reads_norank <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">taxid =</span> parent_taxid) <span class="sc">%&gt;%</span></span>
<span id="cb34-773"><a href="#cb34-773" aria-hidden="true" tabindex="-1"></a>      <span class="fu">select</span>(<span class="sc">-</span>parent_taxid, <span class="sc">-</span>rank, <span class="sc">-</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb34-774"><a href="#cb34-774" aria-hidden="true" tabindex="-1"></a>      <span class="fu">left_join</span>(taxid_db, <span class="at">by=</span><span class="st">"taxid"</span>)</span>
<span id="cb34-775"><a href="#cb34-775" aria-hidden="true" tabindex="-1"></a>    reads_rank <span class="ot">&lt;-</span> reads_remaining <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">==</span> out_rank) <span class="sc">%&gt;%</span></span>
<span id="cb34-776"><a href="#cb34-776" aria-hidden="true" tabindex="-1"></a>      <span class="fu">bind_rows</span>(reads_rank)</span>
<span id="cb34-777"><a href="#cb34-777" aria-hidden="true" tabindex="-1"></a>    reads_norank <span class="ot">&lt;-</span> reads_remaining <span class="sc">%&gt;%</span></span>
<span id="cb34-778"><a href="#cb34-778" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(rank <span class="sc">!=</span> out_rank, <span class="sc">!</span>rank <span class="sc">%in%</span> high_ranks, <span class="sc">!</span><span class="fu">is.na</span>(taxid))</span>
<span id="cb34-779"><a href="#cb34-779" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb34-780"><a href="#cb34-780" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Finally, extract and append reads that were excluded during the process</span></span>
<span id="cb34-781"><a href="#cb34-781" aria-hidden="true" tabindex="-1"></a>  reads_dropped <span class="ot">&lt;-</span> reads <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span>seq_id <span class="sc">%in%</span> reads_rank<span class="sc">$</span>seq_id)</span>
<span id="cb34-782"><a href="#cb34-782" aria-hidden="true" tabindex="-1"></a>  reads_out <span class="ot">&lt;-</span> reads_rank <span class="sc">%&gt;%</span> <span class="fu">bind_rows</span>(reads_dropped) <span class="sc">%&gt;%</span></span>
<span id="cb34-783"><a href="#cb34-783" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(<span class="sc">-</span>parent_taxid, <span class="sc">-</span>rank, <span class="sc">-</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb34-784"><a href="#cb34-784" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(taxid_db, <span class="at">by=</span><span class="st">"taxid"</span>)</span>
<span id="cb34-785"><a href="#cb34-785" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(reads_out)</span>
<span id="cb34-786"><a href="#cb34-786" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb34-787"><a href="#cb34-787" aria-hidden="true" tabindex="-1"></a>hv_reads_species <span class="ot">&lt;-</span> <span class="fu">raise_rank</span>(mrg_hv_named, viral_taxa, <span class="st">"species"</span>)</span>
<span id="cb34-788"><a href="#cb34-788" aria-hidden="true" tabindex="-1"></a>hv_reads_genus <span class="ot">&lt;-</span> <span class="fu">raise_rank</span>(mrg_hv_named, viral_taxa, <span class="st">"genus"</span>)</span>
<span id="cb34-789"><a href="#cb34-789" aria-hidden="true" tabindex="-1"></a>hv_reads_family <span class="ot">&lt;-</span> <span class="fu">raise_rank</span>(mrg_hv_named, viral_taxa, <span class="st">"family"</span>)</span>
<span id="cb34-790"><a href="#cb34-790" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-791"><a href="#cb34-791" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-794"><a href="#cb34-794" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-795"><a href="#cb34-795" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-family</span></span>
<span id="cb34-796"><a href="#cb34-796" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 5</span></span>
<span id="cb34-797"><a href="#cb34-797" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 7</span></span>
<span id="cb34-798"><a href="#cb34-798" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb34-799"><a href="#cb34-799" aria-hidden="true" tabindex="-1"></a>threshold_major_family <span class="ot">&lt;-</span> <span class="fl">0.05</span></span>
<span id="cb34-800"><a href="#cb34-800" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-801"><a href="#cb34-801" aria-hidden="true" tabindex="-1"></a><span class="co"># Count reads for each human-viral family</span></span>
<span id="cb34-802"><a href="#cb34-802" aria-hidden="true" tabindex="-1"></a>hv_family_counts <span class="ot">&lt;-</span> hv_reads_family <span class="sc">%&gt;%</span> </span>
<span id="cb34-803"><a href="#cb34-803" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb34-804"><a href="#cb34-804" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-805"><a href="#cb34-805" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb34-806"><a href="#cb34-806" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_hv =</span> n_reads_hv<span class="sc">/</span><span class="fu">sum</span>(n_reads_hv))</span>
<span id="cb34-807"><a href="#cb34-807" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-808"><a href="#cb34-808" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify high-ranking families and group others</span></span>
<span id="cb34-809"><a href="#cb34-809" aria-hidden="true" tabindex="-1"></a>hv_family_major_tab <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span> </span>
<span id="cb34-810"><a href="#cb34-810" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_hv <span class="sc">==</span> <span class="fu">max</span>(p_reads_hv)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-811"><a href="#cb34-811" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_hv)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(p_reads_hv <span class="sc">&gt;</span> threshold_major_family)</span>
<span id="cb34-812"><a href="#cb34-812" aria-hidden="true" tabindex="-1"></a>hv_family_counts_major <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span></span>
<span id="cb34-813"><a href="#cb34-813" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">ifelse</span>(name <span class="sc">%in%</span> hv_family_major_tab<span class="sc">$</span>name, name, <span class="st">"Other"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb34-814"><a href="#cb34-814" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, name_display) <span class="sc">%&gt;%</span></span>
<span id="cb34-815"><a href="#cb34-815" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_hv =</span> <span class="fu">sum</span>(n_reads_hv), <span class="at">p_reads_hv =</span> <span class="fu">sum</span>(p_reads_hv), </span>
<span id="cb34-816"><a href="#cb34-816" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-817"><a href="#cb34-817" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">factor</span>(name_display, </span>
<span id="cb34-818"><a href="#cb34-818" aria-hidden="true" tabindex="-1"></a>                               <span class="at">levels =</span> <span class="fu">c</span>(hv_family_major_tab<span class="sc">$</span>name, <span class="st">"Other"</span>)))</span>
<span id="cb34-819"><a href="#cb34-819" aria-hidden="true" tabindex="-1"></a>hv_family_counts_display <span class="ot">&lt;-</span> hv_family_counts_major <span class="sc">%&gt;%</span></span>
<span id="cb34-820"><a href="#cb34-820" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_hv, <span class="at">classification =</span> name_display) <span class="sc">%&gt;%</span></span>
<span id="cb34-821"><a href="#cb34-821" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by =</span> <span class="st">'sample'</span>)</span>
<span id="cb34-822"><a href="#cb34-822" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-823"><a href="#cb34-823" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb34-824"><a href="#cb34-824" aria-hidden="true" tabindex="-1"></a>g_hv_family <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb34-825"><a href="#cb34-825" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">data=</span>hv_family_counts_display, <span class="at">position =</span> <span class="st">"stack"</span>, <span class="at">width=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb34-826"><a href="#cb34-826" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% HV Reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb34-827"><a href="#cb34-827" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb34-828"><a href="#cb34-828" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb34-829"><a href="#cb34-829" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">'Accent'</span>, <span class="at">name =</span> <span class="st">"Viral family"</span>) <span class="sc">+</span></span>
<span id="cb34-830"><a href="#cb34-830" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">"Family composition of human-viral reads"</span>) <span class="sc">+</span></span>
<span id="cb34-831"><a href="#cb34-831" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">4</span>)) <span class="sc">+</span></span>
<span id="cb34-832"><a href="#cb34-832" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">1.4</span>), <span class="at">hjust=</span><span class="dv">0</span>, <span class="at">face=</span><span class="st">"plain"</span>))</span>
<span id="cb34-833"><a href="#cb34-833" aria-hidden="true" tabindex="-1"></a>g_hv_family</span>
<span id="cb34-834"><a href="#cb34-834" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-835"><a href="#cb34-835" aria-hidden="true" tabindex="-1"></a><span class="co"># Get most prominent families for text</span></span>
<span id="cb34-836"><a href="#cb34-836" aria-hidden="true" tabindex="-1"></a>hv_family_collate <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span></span>
<span id="cb34-837"><a href="#cb34-837" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb34-838"><a href="#cb34-838" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_tot =</span> <span class="fu">sum</span>(n_reads_hv),</span>
<span id="cb34-839"><a href="#cb34-839" aria-hidden="true" tabindex="-1"></a>            <span class="at">p_reads_max =</span> <span class="fu">max</span>(p_reads_hv), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb34-840"><a href="#cb34-840" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n_reads_tot))</span>
<span id="cb34-841"><a href="#cb34-841" aria-hidden="true" tabindex="-1"></a>hv_family_collate <span class="sc">%&gt;%</span></span>
<span id="cb34-842"><a href="#cb34-842" aria-hidden="true" tabindex="-1"></a> <span class="fu">select</span>(name,taxid, n_reads_tot) <span class="sc">%&gt;%</span></span>
<span id="cb34-843"><a href="#cb34-843" aria-hidden="true" tabindex="-1"></a> <span class="fu">rename</span>(</span>
<span id="cb34-844"><a href="#cb34-844" aria-hidden="true" tabindex="-1"></a>  <span class="st">'family'</span> <span class="ot">=</span> <span class="st">'name'</span>,</span>
<span id="cb34-845"><a href="#cb34-845" aria-hidden="true" tabindex="-1"></a>  <span class="st">'# of total reads'</span> <span class="ot">=</span> <span class="st">'n_reads_tot'</span>,</span>
<span id="cb34-846"><a href="#cb34-846" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb34-847"><a href="#cb34-847" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-848"><a href="#cb34-848" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-849"><a href="#cb34-849" aria-hidden="true" tabindex="-1"></a><span class="fu">## Analyzing specific families</span></span>
<span id="cb34-850"><a href="#cb34-850" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-851"><a href="#cb34-851" aria-hidden="true" tabindex="-1"></a>We now investigate the composition of specific families that had more than 5 viral reads. In investigating individual viral families, to avoid distortions from a few rare reads, I restricted myself to samples where that family made up at least 1% of human-viral reads:</span>
<span id="cb34-852"><a href="#cb34-852" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-855"><a href="#cb34-855" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-856"><a href="#cb34-856" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-viral-family-scripts</span></span>
<span id="cb34-857"><a href="#cb34-857" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb34-858"><a href="#cb34-858" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-859"><a href="#cb34-859" aria-hidden="true" tabindex="-1"></a>plot_viral_family_composition <span class="ot">&lt;-</span> <span class="cf">function</span>(taxid_chosen, <span class="at">threshold_major_species =</span> <span class="fl">0.1</span>) {</span>
<span id="cb34-860"><a href="#cb34-860" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get set of chosen family reads</span></span>
<span id="cb34-861"><a href="#cb34-861" aria-hidden="true" tabindex="-1"></a>  family_samples <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span> <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_chosen) <span class="sc">%&gt;%</span></span>
<span id="cb34-862"><a href="#cb34-862" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(p_reads_hv <span class="sc">&gt;=</span> <span class="fl">0.1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-863"><a href="#cb34-863" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pull</span>(sample)</span>
<span id="cb34-864"><a href="#cb34-864" aria-hidden="true" tabindex="-1"></a>  family_ids <span class="ot">&lt;-</span> hv_reads_family <span class="sc">%&gt;%</span> </span>
<span id="cb34-865"><a href="#cb34-865" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_chosen, sample <span class="sc">%in%</span> family_samples) <span class="sc">%&gt;%</span></span>
<span id="cb34-866"><a href="#cb34-866" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pull</span>(seq_id)</span>
<span id="cb34-867"><a href="#cb34-867" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-868"><a href="#cb34-868" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Count reads for each species in the chosen family</span></span>
<span id="cb34-869"><a href="#cb34-869" aria-hidden="true" tabindex="-1"></a>  species_counts <span class="ot">&lt;-</span> hv_reads_species <span class="sc">%&gt;%</span></span>
<span id="cb34-870"><a href="#cb34-870" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(seq_id <span class="sc">%in%</span> family_ids) <span class="sc">%&gt;%</span></span>
<span id="cb34-871"><a href="#cb34-871" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(sample, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb34-872"><a href="#cb34-872" aria-hidden="true" tabindex="-1"></a>    <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-873"><a href="#cb34-873" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb34-874"><a href="#cb34-874" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">p_reads_family =</span> n_reads_hv<span class="sc">/</span><span class="fu">sum</span>(n_reads_hv))</span>
<span id="cb34-875"><a href="#cb34-875" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-876"><a href="#cb34-876" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Identify high-ranking species and group others</span></span>
<span id="cb34-877"><a href="#cb34-877" aria-hidden="true" tabindex="-1"></a>  species_major_tab <span class="ot">&lt;-</span> species_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span> </span>
<span id="cb34-878"><a href="#cb34-878" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(p_reads_family <span class="sc">==</span> <span class="fu">max</span>(p_reads_family)) <span class="sc">%&gt;%</span> </span>
<span id="cb34-879"><a href="#cb34-879" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-880"><a href="#cb34-880" aria-hidden="true" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_family)) <span class="sc">%&gt;%</span> </span>
<span id="cb34-881"><a href="#cb34-881" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(p_reads_family <span class="sc">&gt;</span> threshold_major_species)</span>
<span id="cb34-882"><a href="#cb34-882" aria-hidden="true" tabindex="-1"></a>  species_counts_major <span class="ot">&lt;-</span> species_counts <span class="sc">%&gt;%</span></span>
<span id="cb34-883"><a href="#cb34-883" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">ifelse</span>(name <span class="sc">%in%</span> species_major_tab<span class="sc">$</span>name, </span>
<span id="cb34-884"><a href="#cb34-884" aria-hidden="true" tabindex="-1"></a>                                 name, <span class="st">"Other"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb34-885"><a href="#cb34-885" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(sample, name_display) <span class="sc">%&gt;%</span></span>
<span id="cb34-886"><a href="#cb34-886" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarize</span>(<span class="at">n_reads_family =</span> <span class="fu">sum</span>(n_reads_hv),</span>
<span id="cb34-887"><a href="#cb34-887" aria-hidden="true" tabindex="-1"></a>              <span class="at">p_reads_family =</span> <span class="fu">sum</span>(p_reads_family), </span>
<span id="cb34-888"><a href="#cb34-888" aria-hidden="true" tabindex="-1"></a>              <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-889"><a href="#cb34-889" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">factor</span>(name_display, </span>
<span id="cb34-890"><a href="#cb34-890" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">levels =</span> <span class="fu">c</span>(species_major_tab<span class="sc">$</span>name, <span class="st">"Other"</span>)))</span>
<span id="cb34-891"><a href="#cb34-891" aria-hidden="true" tabindex="-1"></a>  species_counts_display <span class="ot">&lt;-</span> species_counts_major <span class="sc">%&gt;%</span></span>
<span id="cb34-892"><a href="#cb34-892" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_family, <span class="at">classification =</span> name_display) <span class="sc">%&gt;%</span></span>
<span id="cb34-893"><a href="#cb34-893" aria-hidden="true" tabindex="-1"></a>    <span class="fu">right_join</span>(libraries, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">'sample'</span>))</span>
<span id="cb34-894"><a href="#cb34-894" aria-hidden="true" tabindex="-1"></a>    <span class="co">#mutate(p_reads = coalesce(p_reads, 0),</span></span>
<span id="cb34-895"><a href="#cb34-895" aria-hidden="true" tabindex="-1"></a>    <span class="co">#       classification = coalesce(classification, species_major_tab$name[1])) %&gt;%</span></span>
<span id="cb34-896"><a href="#cb34-896" aria-hidden="true" tabindex="-1"></a>    <span class="co">#group_by(location) %&gt;%</span></span>
<span id="cb34-897"><a href="#cb34-897" aria-hidden="true" tabindex="-1"></a>    <span class="co">#mutate(samples_with_reads = sum(p_reads &gt; 0),</span></span>
<span id="cb34-898"><a href="#cb34-898" aria-hidden="true" tabindex="-1"></a>    <span class="co">#  label = paste0(location, " (", sprintf("%.1f%%", 100 * samples_with_reads / total_samples), ")")) %&gt;%</span></span>
<span id="cb34-899"><a href="#cb34-899" aria-hidden="true" tabindex="-1"></a>    <span class="co">#ungroup()</span></span>
<span id="cb34-900"><a href="#cb34-900" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-901"><a href="#cb34-901" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get family name for the title</span></span>
<span id="cb34-902"><a href="#cb34-902" aria-hidden="true" tabindex="-1"></a>  family_name <span class="ot">&lt;-</span> viral_taxa <span class="sc">%&gt;%</span> </span>
<span id="cb34-903"><a href="#cb34-903" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_chosen) <span class="sc">%&gt;%</span> </span>
<span id="cb34-904"><a href="#cb34-904" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pull</span>(name)</span>
<span id="cb34-905"><a href="#cb34-905" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-906"><a href="#cb34-906" aria-hidden="true" tabindex="-1"></a>  species_count <span class="ot">&lt;-</span> species_counts_display <span class="sc">%&gt;%</span> <span class="fu">select</span>(classification) <span class="sc">%&gt;%</span> <span class="fu">distinct</span>() <span class="sc">%&gt;%</span> <span class="fu">pull</span>() <span class="sc">%&gt;%</span> <span class="fu">length</span>()</span>
<span id="cb34-907"><a href="#cb34-907" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (species_count <span class="sc">&lt;=</span> <span class="dv">8</span>) {</span>
<span id="cb34-908"><a href="#cb34-908" aria-hidden="true" tabindex="-1"></a>    pal <span class="ot">&lt;-</span> <span class="fu">brewer.pal</span>(<span class="at">name =</span> <span class="st">"Accent"</span>, <span class="at">n =</span> species_count)</span>
<span id="cb34-909"><a href="#cb34-909" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb34-910"><a href="#cb34-910" aria-hidden="true" tabindex="-1"></a>    pal <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">brewer.pal</span>(<span class="at">name =</span> <span class="st">"Paired"</span>, <span class="at">n =</span> <span class="dv">12</span>), <span class="fu">brewer.pal</span>(<span class="at">name =</span> <span class="st">"Set3"</span>, <span class="at">n =</span><span class="dv">12</span>), <span class="fu">brewer.pal</span>(<span class="at">name =</span> <span class="st">"Set1"</span>, <span class="at">n =</span> <span class="dv">9</span>))[<span class="dv">1</span><span class="sc">:</span>species_count]</span>
<span id="cb34-911"><a href="#cb34-911" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb34-912"><a href="#cb34-912" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-913"><a href="#cb34-913" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Plot</span></span>
<span id="cb34-914"><a href="#cb34-914" aria-hidden="true" tabindex="-1"></a>  g_species <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb34-915"><a href="#cb34-915" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_col</span>(<span class="at">data=</span>species_counts_display, <span class="at">position =</span> <span class="st">"stack"</span>) <span class="sc">+</span></span>
<span id="cb34-916"><a href="#cb34-916" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="fu">paste0</span>(<span class="st">"% "</span>, family_name, <span class="st">" reads"</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb34-917"><a href="#cb34-917" aria-hidden="true" tabindex="-1"></a>                       <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb34-918"><a href="#cb34-918" aria-hidden="true" tabindex="-1"></a>                       <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb34-919"><a href="#cb34-919" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> pal, <span class="at">name =</span> <span class="st">"Viral species"</span>) <span class="sc">+</span></span>
<span id="cb34-920"><a href="#cb34-920" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title=</span><span class="fu">paste0</span>(<span class="st">"Species composition of "</span>, family_name, <span class="st">" reads"</span>)) <span class="sc">+</span></span>
<span id="cb34-921"><a href="#cb34-921" aria-hidden="true" tabindex="-1"></a>    <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">3</span>)) <span class="sc">+</span></span>
<span id="cb34-922"><a href="#cb34-922" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">1.4</span>), <span class="at">hjust=</span><span class="dv">0</span>, <span class="at">face=</span><span class="st">"plain"</span>))</span>
<span id="cb34-923"><a href="#cb34-923" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(g_species)</span>
<span id="cb34-924"><a href="#cb34-924" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb34-925"><a href="#cb34-925" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-926"><a href="#cb34-926" aria-hidden="true" tabindex="-1"></a>plot_viral_family_histogram <span class="ot">&lt;-</span> <span class="cf">function</span>(taxid_chosen) {</span>
<span id="cb34-927"><a href="#cb34-927" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filter data for the specified virus family</span></span>
<span id="cb34-928"><a href="#cb34-928" aria-hidden="true" tabindex="-1"></a>  family_data <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span></span>
<span id="cb34-929"><a href="#cb34-929" aria-hidden="true" tabindex="-1"></a>    <span class="fu">inner_join</span>(libraries, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">'sample'</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb34-930"><a href="#cb34-930" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_chosen)</span>
<span id="cb34-931"><a href="#cb34-931" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb34-932"><a href="#cb34-932" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get family name for the title</span></span>
<span id="cb34-933"><a href="#cb34-933" aria-hidden="true" tabindex="-1"></a>  family_name <span class="ot">&lt;-</span> viral_taxa <span class="sc">%&gt;%</span> </span>
<span id="cb34-934"><a href="#cb34-934" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_chosen) <span class="sc">%&gt;%</span> </span>
<span id="cb34-935"><a href="#cb34-935" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pull</span>(name)</span>
<span id="cb34-936"><a href="#cb34-936" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb34-937"><a href="#cb34-937" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create the histogram</span></span>
<span id="cb34-938"><a href="#cb34-938" aria-hidden="true" tabindex="-1"></a>  g_histogram <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(family_data, <span class="fu">aes</span>(<span class="at">x =</span> library, <span class="at">y =</span> n_reads_hv)) <span class="sc">+</span></span>
<span id="cb34-939"><a href="#cb34-939" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb34-940"><a href="#cb34-940" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label =</span> scales<span class="sc">::</span><span class="fu">comma</span>(n_reads_hv)), <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb34-941"><a href="#cb34-941" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb34-942"><a href="#cb34-942" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb34-943"><a href="#cb34-943" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_discrete</span>(<span class="at">limits =</span> libraries <span class="sc">%&gt;%</span> <span class="fu">pull</span>(library)) <span class="sc">+</span></span>
<span id="cb34-944"><a href="#cb34-944" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(</span>
<span id="cb34-945"><a href="#cb34-945" aria-hidden="true" tabindex="-1"></a>      <span class="at">title =</span> <span class="fu">paste</span>(<span class="st">"Distribution of"</span>, family_name, <span class="st">"Reads by Location"</span>),</span>
<span id="cb34-946"><a href="#cb34-946" aria-hidden="true" tabindex="-1"></a>      <span class="at">x =</span> <span class="st">"Location"</span>,</span>
<span id="cb34-947"><a href="#cb34-947" aria-hidden="true" tabindex="-1"></a>      <span class="at">y =</span> <span class="st">"Number of reads"</span></span>
<span id="cb34-948"><a href="#cb34-948" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb34-949"><a href="#cb34-949" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_continuous</span>(<span class="at">labels =</span> scales<span class="sc">::</span>comma) <span class="sc">+</span></span>
<span id="cb34-950"><a href="#cb34-950" aria-hidden="true" tabindex="-1"></a>    <span class="fu">expand_limits</span>(<span class="at">y =</span> <span class="fu">max</span>(family_data<span class="sc">$</span>n_reads_hv) <span class="sc">*</span> <span class="fl">1.1</span>)<span class="sc">+</span></span>
<span id="cb34-951"><a href="#cb34-951" aria-hidden="true" tabindex="-1"></a>    theme_kit  <span class="co"># Expand y-axis to make room for labels</span></span>
<span id="cb34-952"><a href="#cb34-952" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(g_histogram)</span>
<span id="cb34-953"><a href="#cb34-953" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb34-954"><a href="#cb34-954" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-955"><a href="#cb34-955" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-956"><a href="#cb34-956" aria-hidden="true" tabindex="-1"></a><span class="fu">### Anelloviridae (Number of reads: 8,323,297)</span></span>
<span id="cb34-959"><a href="#cb34-959" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-960"><a href="#cb34-960" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-anelloviridae-histogram</span></span>
<span id="cb34-961"><a href="#cb34-961" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb34-962"><a href="#cb34-962" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 2.5</span></span>
<span id="cb34-963"><a href="#cb34-963" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-964"><a href="#cb34-964" aria-hidden="true" tabindex="-1"></a><span class="co"># Anelloviridae</span></span>
<span id="cb34-965"><a href="#cb34-965" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_histogram</span>(<span class="at">taxid_chosen=</span><span class="dv">687329</span>)</span>
<span id="cb34-966"><a href="#cb34-966" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-969"><a href="#cb34-969" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-970"><a href="#cb34-970" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-anelloviridae-composition</span></span>
<span id="cb34-971"><a href="#cb34-971" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb34-972"><a href="#cb34-972" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 7.5</span></span>
<span id="cb34-973"><a href="#cb34-973" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb34-974"><a href="#cb34-974" aria-hidden="true" tabindex="-1"></a><span class="co"># Anelloviridae</span></span>
<span id="cb34-975"><a href="#cb34-975" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_composition</span>(<span class="at">taxid_chosen=</span><span class="dv">687329</span>, <span class="at">threshold_major_species =</span> <span class="fl">0.1</span>)</span>
<span id="cb34-976"><a href="#cb34-976" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-977"><a href="#cb34-977" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-978"><a href="#cb34-978" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-979"><a href="#cb34-979" aria-hidden="true" tabindex="-1"></a><span class="fu">### Rhabdoviridae (Number of reads: 935,461) [SPIKE-IN]</span></span>
<span id="cb34-982"><a href="#cb34-982" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-983"><a href="#cb34-983" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-rhabdoviridae-histogram</span></span>
<span id="cb34-984"><a href="#cb34-984" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb34-985"><a href="#cb34-985" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 2.5</span></span>
<span id="cb34-986"><a href="#cb34-986" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-987"><a href="#cb34-987" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_histogram</span>(<span class="at">taxid_chosen=</span><span class="dv">11270</span>)</span>
<span id="cb34-988"><a href="#cb34-988" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-991"><a href="#cb34-991" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-992"><a href="#cb34-992" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-rhabdoviridae-composition</span></span>
<span id="cb34-993"><a href="#cb34-993" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb34-994"><a href="#cb34-994" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 7.5</span></span>
<span id="cb34-995"><a href="#cb34-995" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb34-996"><a href="#cb34-996" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb34-997"><a href="#cb34-997" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_composition</span>(<span class="at">taxid_chosen=</span><span class="dv">11270</span>, <span class="at">threshold_major_species =</span> <span class="fl">0.1</span>)</span>
<span id="cb34-998"><a href="#cb34-998" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-999"><a href="#cb34-999" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1000"><a href="#cb34-1000" aria-hidden="true" tabindex="-1"></a><span class="fu">### Flaviviridae (Number of reads: 94,741)</span></span>
<span id="cb34-1003"><a href="#cb34-1003" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-1004"><a href="#cb34-1004" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-flavi-histogram</span></span>
<span id="cb34-1005"><a href="#cb34-1005" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb34-1006"><a href="#cb34-1006" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 2.5</span></span>
<span id="cb34-1007"><a href="#cb34-1007" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1008"><a href="#cb34-1008" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_histogram</span>(<span class="at">taxid_chosen=</span><span class="dv">11050</span>)</span>
<span id="cb34-1009"><a href="#cb34-1009" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-1012"><a href="#cb34-1012" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-1013"><a href="#cb34-1013" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-flavi-composition</span></span>
<span id="cb34-1014"><a href="#cb34-1014" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb34-1015"><a href="#cb34-1015" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 7.5</span></span>
<span id="cb34-1016"><a href="#cb34-1016" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb34-1017"><a href="#cb34-1017" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb34-1018"><a href="#cb34-1018" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_composition</span>(<span class="at">taxid_chosen=</span><span class="dv">11050</span>, <span class="at">threshold_major_species =</span> <span class="fl">0.1</span>)</span>
<span id="cb34-1019"><a href="#cb34-1019" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-1020"><a href="#cb34-1020" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1021"><a href="#cb34-1021" aria-hidden="true" tabindex="-1"></a><span class="fu">### Microviridae (Number of reads: 10,841) [SPIKE-IN]</span></span>
<span id="cb34-1024"><a href="#cb34-1024" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-1025"><a href="#cb34-1025" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-microviridae-histogram</span></span>
<span id="cb34-1026"><a href="#cb34-1026" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb34-1027"><a href="#cb34-1027" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 2.5</span></span>
<span id="cb34-1028"><a href="#cb34-1028" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1029"><a href="#cb34-1029" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_histogram</span>(<span class="at">taxid_chosen=</span><span class="dv">10841</span>)</span>
<span id="cb34-1030"><a href="#cb34-1030" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-1033"><a href="#cb34-1033" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-1034"><a href="#cb34-1034" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-microviridae-composition</span></span>
<span id="cb34-1035"><a href="#cb34-1035" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb34-1036"><a href="#cb34-1036" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 7.5</span></span>
<span id="cb34-1037"><a href="#cb34-1037" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb34-1038"><a href="#cb34-1038" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb34-1039"><a href="#cb34-1039" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_composition</span>(<span class="at">taxid_chosen=</span><span class="dv">10841</span>, <span class="at">threshold_major_species =</span> <span class="fl">0.1</span>)</span>
<span id="cb34-1040"><a href="#cb34-1040" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-1041"><a href="#cb34-1041" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1042"><a href="#cb34-1042" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1043"><a href="#cb34-1043" aria-hidden="true" tabindex="-1"></a><span class="fu">## Relative abundance of pathogenic viruses of interest</span></span>
<span id="cb34-1044"><a href="#cb34-1044" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1047"><a href="#cb34-1047" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-1048"><a href="#cb34-1048" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-find-each</span></span>
<span id="cb34-1049"><a href="#cb34-1049" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb34-1050"><a href="#cb34-1050" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1051"><a href="#cb34-1051" aria-hidden="true" tabindex="-1"></a>total <span class="ot">&lt;-</span> basic_stats_raw <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample, n_read_pairs)</span>
<span id="cb34-1052"><a href="#cb34-1052" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1053"><a href="#cb34-1053" aria-hidden="true" tabindex="-1"></a>hv_family_counts <span class="ot">&lt;-</span> hv_reads_family <span class="sc">%&gt;%</span> </span>
<span id="cb34-1054"><a href="#cb34-1054" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb34-1055"><a href="#cb34-1055" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-1056"><a href="#cb34-1056" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(total, <span class="at">by=</span><span class="st">'sample'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-1057"><a href="#cb34-1057" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb34-1058"><a href="#cb34-1058" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">ra_reads_hv =</span> n_reads_hv<span class="sc">/</span>n_read_pairs)</span>
<span id="cb34-1059"><a href="#cb34-1059" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1060"><a href="#cb34-1060" aria-hidden="true" tabindex="-1"></a>hv_family_collate <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span></span>
<span id="cb34-1061"><a href="#cb34-1061" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb34-1062"><a href="#cb34-1062" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_tot =</span> <span class="fu">sum</span>(n_reads_hv),</span>
<span id="cb34-1063"><a href="#cb34-1063" aria-hidden="true" tabindex="-1"></a>            <span class="at">ra_reads_tot =</span> <span class="fu">mean</span>(ra_reads_hv), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb34-1064"><a href="#cb34-1064" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n_reads_tot))</span>
<span id="cb34-1065"><a href="#cb34-1065" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1066"><a href="#cb34-1066" aria-hidden="true" tabindex="-1"></a>hv_family_collate <span class="ot">&lt;-</span> hv_family_collate <span class="sc">%&gt;%</span> </span>
<span id="cb34-1067"><a href="#cb34-1067" aria-hidden="true" tabindex="-1"></a><span class="fu">rename</span>(</span>
<span id="cb34-1068"><a href="#cb34-1068" aria-hidden="true" tabindex="-1"></a><span class="st">'family'</span> <span class="ot">=</span> <span class="st">'name'</span>,</span>
<span id="cb34-1069"><a href="#cb34-1069" aria-hidden="true" tabindex="-1"></a><span class="st">'family_n_reads_tot'</span> <span class="ot">=</span> <span class="st">'n_reads_tot'</span>,</span>
<span id="cb34-1070"><a href="#cb34-1070" aria-hidden="true" tabindex="-1"></a><span class="st">'family_ra_reads_tot'</span> <span class="ot">=</span> <span class="st">'ra_reads_tot'</span>,</span>
<span id="cb34-1071"><a href="#cb34-1071" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb34-1072"><a href="#cb34-1072" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1073"><a href="#cb34-1073" aria-hidden="true" tabindex="-1"></a>hv_genus_counts <span class="ot">&lt;-</span> hv_reads_genus <span class="sc">%&gt;%</span> </span>
<span id="cb34-1074"><a href="#cb34-1074" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb34-1075"><a href="#cb34-1075" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-1076"><a href="#cb34-1076" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(total, <span class="at">by=</span><span class="st">'sample'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-1077"><a href="#cb34-1077" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb34-1078"><a href="#cb34-1078" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">ra_reads_hv =</span> n_reads_hv<span class="sc">/</span>n_read_pairs)</span>
<span id="cb34-1079"><a href="#cb34-1079" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1080"><a href="#cb34-1080" aria-hidden="true" tabindex="-1"></a>hv_genus_collate <span class="ot">&lt;-</span> hv_genus_counts <span class="sc">%&gt;%</span></span>
<span id="cb34-1081"><a href="#cb34-1081" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb34-1082"><a href="#cb34-1082" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_tot =</span> <span class="fu">sum</span>(n_reads_hv),</span>
<span id="cb34-1083"><a href="#cb34-1083" aria-hidden="true" tabindex="-1"></a>            <span class="at">ra_reads_tot =</span> <span class="fu">mean</span>(ra_reads_hv), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb34-1084"><a href="#cb34-1084" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n_reads_tot))</span>
<span id="cb34-1085"><a href="#cb34-1085" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1086"><a href="#cb34-1086" aria-hidden="true" tabindex="-1"></a>hv_genus_collate <span class="ot">&lt;-</span> hv_genus_collate <span class="sc">%&gt;%</span> </span>
<span id="cb34-1087"><a href="#cb34-1087" aria-hidden="true" tabindex="-1"></a><span class="fu">rename</span>(</span>
<span id="cb34-1088"><a href="#cb34-1088" aria-hidden="true" tabindex="-1"></a><span class="st">'genus'</span> <span class="ot">=</span> <span class="st">'name'</span>,</span>
<span id="cb34-1089"><a href="#cb34-1089" aria-hidden="true" tabindex="-1"></a><span class="st">'genus_n_reads_tot'</span> <span class="ot">=</span> <span class="st">'n_reads_tot'</span>,</span>
<span id="cb34-1090"><a href="#cb34-1090" aria-hidden="true" tabindex="-1"></a><span class="st">'genus_ra_reads_tot'</span> <span class="ot">=</span> <span class="st">'ra_reads_tot'</span>,</span>
<span id="cb34-1091"><a href="#cb34-1091" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb34-1092"><a href="#cb34-1092" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1093"><a href="#cb34-1093" aria-hidden="true" tabindex="-1"></a>hv_species_counts <span class="ot">&lt;-</span> hv_reads_species <span class="sc">%&gt;%</span> </span>
<span id="cb34-1094"><a href="#cb34-1094" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb34-1095"><a href="#cb34-1095" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-1096"><a href="#cb34-1096" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(total, <span class="at">by=</span><span class="st">'sample'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-1097"><a href="#cb34-1097" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb34-1098"><a href="#cb34-1098" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">ra_reads_hv =</span> n_reads_hv<span class="sc">/</span>n_read_pairs)</span>
<span id="cb34-1099"><a href="#cb34-1099" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1100"><a href="#cb34-1100" aria-hidden="true" tabindex="-1"></a>hv_species_collate <span class="ot">&lt;-</span> hv_species_counts <span class="sc">%&gt;%</span></span>
<span id="cb34-1101"><a href="#cb34-1101" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb34-1102"><a href="#cb34-1102" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_tot =</span> <span class="fu">sum</span>(n_reads_hv),</span>
<span id="cb34-1103"><a href="#cb34-1103" aria-hidden="true" tabindex="-1"></a>            <span class="at">ra_reads_tot =</span> <span class="fu">mean</span>(ra_reads_hv), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb34-1104"><a href="#cb34-1104" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n_reads_tot))</span>
<span id="cb34-1105"><a href="#cb34-1105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1106"><a href="#cb34-1106" aria-hidden="true" tabindex="-1"></a>hv_species_collate <span class="ot">&lt;-</span> hv_species_collate <span class="sc">%&gt;%</span> </span>
<span id="cb34-1107"><a href="#cb34-1107" aria-hidden="true" tabindex="-1"></a><span class="fu">rename</span>(</span>
<span id="cb34-1108"><a href="#cb34-1108" aria-hidden="true" tabindex="-1"></a><span class="st">'species'</span> <span class="ot">=</span> <span class="st">'name'</span>,</span>
<span id="cb34-1109"><a href="#cb34-1109" aria-hidden="true" tabindex="-1"></a><span class="st">'species_n_reads_tot'</span> <span class="ot">=</span> <span class="st">'n_reads_tot'</span>,</span>
<span id="cb34-1110"><a href="#cb34-1110" aria-hidden="true" tabindex="-1"></a><span class="st">'species_ra_reads_tot'</span> <span class="ot">=</span> <span class="st">'ra_reads_tot'</span>,</span>
<span id="cb34-1111"><a href="#cb34-1111" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb34-1112"><a href="#cb34-1112" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1113"><a href="#cb34-1113" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-1114"><a href="#cb34-1114" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1117"><a href="#cb34-1117" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-1118"><a href="#cb34-1118" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-filter-exclusion</span></span>
<span id="cb34-1119"><a href="#cb34-1119" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb34-1120"><a href="#cb34-1120" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1121"><a href="#cb34-1121" aria-hidden="true" tabindex="-1"></a><span class="co"># Assuming the data is already loaded into tibbles:</span></span>
<span id="cb34-1122"><a href="#cb34-1122" aria-hidden="true" tabindex="-1"></a><span class="co">#hv_family_collate</span></span>
<span id="cb34-1123"><a href="#cb34-1123" aria-hidden="true" tabindex="-1"></a><span class="co">#hv_genus_collate</span></span>
<span id="cb34-1124"><a href="#cb34-1124" aria-hidden="true" tabindex="-1"></a><span class="co">#hv_species_collate</span></span>
<span id="cb34-1125"><a href="#cb34-1125" aria-hidden="true" tabindex="-1"></a><span class="co">#viral_taxa</span></span>
<span id="cb34-1126"><a href="#cb34-1126" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1127"><a href="#cb34-1127" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to find parent taxa recursively</span></span>
<span id="cb34-1128"><a href="#cb34-1128" aria-hidden="true" tabindex="-1"></a>find_parent_taxa <span class="ot">&lt;-</span> <span class="cf">function</span>(taxid, target_ranks, taxa_data) {</span>
<span id="cb34-1129"><a href="#cb34-1129" aria-hidden="true" tabindex="-1"></a> current_taxid <span class="ot">&lt;-</span> taxid</span>
<span id="cb34-1130"><a href="#cb34-1130" aria-hidden="true" tabindex="-1"></a>  result <span class="ot">&lt;-</span> <span class="fu">setNames</span>(<span class="fu">vector</span>(<span class="st">"list"</span>, <span class="fu">length</span>(target_ranks)), target_ranks)</span>
<span id="cb34-1131"><a href="#cb34-1131" aria-hidden="true" tabindex="-1"></a>  <span class="cf">while</span> (<span class="cn">TRUE</span>) {</span>
<span id="cb34-1132"><a href="#cb34-1132" aria-hidden="true" tabindex="-1"></a>    row <span class="ot">&lt;-</span> taxa_data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(taxid <span class="sc">==</span> current_taxid) <span class="sc">%&gt;%</span> <span class="fu">slice</span>(<span class="dv">1</span>)</span>
<span id="cb34-1133"><a href="#cb34-1133" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">nrow</span>(row) <span class="sc">==</span> <span class="dv">0</span>) <span class="cf">break</span>  <span class="co"># If no matching taxid is found, break the loop</span></span>
<span id="cb34-1134"><a href="#cb34-1134" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (row<span class="sc">$</span>rank <span class="sc">%in%</span> target_ranks) {</span>
<span id="cb34-1135"><a href="#cb34-1135" aria-hidden="true" tabindex="-1"></a>      result[[row<span class="sc">$</span>rank]] <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">taxid =</span> row<span class="sc">$</span>taxid, <span class="at">name =</span> row<span class="sc">$</span>name)</span>
<span id="cb34-1136"><a href="#cb34-1136" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span> (<span class="fu">all</span>(<span class="sc">!</span><span class="fu">sapply</span>(result, is.null))) {</span>
<span id="cb34-1137"><a href="#cb34-1137" aria-hidden="true" tabindex="-1"></a>        <span class="cf">break</span>  <span class="co"># If all target ranks are found, break the loop</span></span>
<span id="cb34-1138"><a href="#cb34-1138" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb34-1139"><a href="#cb34-1139" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb34-1140"><a href="#cb34-1140" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (row<span class="sc">$</span>taxid <span class="sc">==</span> row<span class="sc">$</span>parent_taxid) {</span>
<span id="cb34-1141"><a href="#cb34-1141" aria-hidden="true" tabindex="-1"></a>      <span class="cf">break</span>  <span class="co"># Avoid infinite loop at root</span></span>
<span id="cb34-1142"><a href="#cb34-1142" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb34-1143"><a href="#cb34-1143" aria-hidden="true" tabindex="-1"></a>    current_taxid <span class="ot">&lt;-</span> row<span class="sc">$</span>parent_taxid</span>
<span id="cb34-1144"><a href="#cb34-1144" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb34-1145"><a href="#cb34-1145" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Replace NULL with NA for any unfound ranks</span></span>
<span id="cb34-1146"><a href="#cb34-1146" aria-hidden="true" tabindex="-1"></a>  result <span class="ot">&lt;-</span> <span class="fu">lapply</span>(result, <span class="cf">function</span>(x) <span class="cf">if</span> (<span class="fu">is.null</span>(x)) <span class="fu">list</span>(<span class="at">taxid =</span> <span class="cn">NA</span>, <span class="at">name =</span> <span class="cn">NA</span>) <span class="cf">else</span> x)</span>
<span id="cb34-1147"><a href="#cb34-1147" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(result)</span>
<span id="cb34-1148"><a href="#cb34-1148" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb34-1149"><a href="#cb34-1149" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1150"><a href="#cb34-1150" aria-hidden="true" tabindex="-1"></a><span class="co"># Find all relevant parent taxa for each species</span></span>
<span id="cb34-1151"><a href="#cb34-1151" aria-hidden="true" tabindex="-1"></a>result <span class="ot">&lt;-</span> hv_species_collate <span class="sc">%&gt;%</span></span>
<span id="cb34-1152"><a href="#cb34-1152" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb34-1153"><a href="#cb34-1153" aria-hidden="true" tabindex="-1"></a>    <span class="at">parent_info =</span> <span class="fu">map</span>(taxid, <span class="sc">~</span><span class="fu">find_parent_taxa</span>(.x, <span class="fu">c</span>(<span class="st">"species"</span>,<span class="st">"subgenus"</span>, <span class="st">"genus"</span>, <span class="st">"subfamily"</span>, <span class="st">"family"</span>), viral_taxa)),</span>
<span id="cb34-1154"><a href="#cb34-1154" aria-hidden="true" tabindex="-1"></a>    <span class="at">genus_taxid =</span> <span class="fu">map_dbl</span>(parent_info, <span class="sc">~</span>.x<span class="sc">$</span>genus<span class="sc">$</span>taxid),</span>
<span id="cb34-1155"><a href="#cb34-1155" aria-hidden="true" tabindex="-1"></a>    <span class="at">genus =</span> <span class="fu">map_chr</span>(parent_info, <span class="sc">~</span>.x<span class="sc">$</span>genus<span class="sc">$</span>name),</span>
<span id="cb34-1156"><a href="#cb34-1156" aria-hidden="true" tabindex="-1"></a>    <span class="at">family_taxid =</span> <span class="fu">map_dbl</span>(parent_info, <span class="sc">~</span>.x<span class="sc">$</span>family<span class="sc">$</span>taxid),</span>
<span id="cb34-1157"><a href="#cb34-1157" aria-hidden="true" tabindex="-1"></a>    <span class="at">family =</span> <span class="fu">map_chr</span>(parent_info, <span class="sc">~</span>.x<span class="sc">$</span>family<span class="sc">$</span>name)</span>
<span id="cb34-1158"><a href="#cb34-1158" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb34-1159"><a href="#cb34-1159" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>parent_info)</span>
<span id="cb34-1160"><a href="#cb34-1160" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1161"><a href="#cb34-1161" aria-hidden="true" tabindex="-1"></a><span class="co"># Join with genus counts</span></span>
<span id="cb34-1162"><a href="#cb34-1162" aria-hidden="true" tabindex="-1"></a>result <span class="ot">&lt;-</span> result <span class="sc">%&gt;%</span></span>
<span id="cb34-1163"><a href="#cb34-1163" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(</span>
<span id="cb34-1164"><a href="#cb34-1164" aria-hidden="true" tabindex="-1"></a>    hv_genus_collate <span class="sc">%&gt;%</span> <span class="fu">select</span>(taxid, genus_n_reads_tot, genus_ra_reads_tot),</span>
<span id="cb34-1165"><a href="#cb34-1165" aria-hidden="true" tabindex="-1"></a>    <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"genus_taxid"</span> <span class="ot">=</span> <span class="st">"taxid"</span>)</span>
<span id="cb34-1166"><a href="#cb34-1166" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb34-1167"><a href="#cb34-1167" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1168"><a href="#cb34-1168" aria-hidden="true" tabindex="-1"></a><span class="co"># Join with family counts</span></span>
<span id="cb34-1169"><a href="#cb34-1169" aria-hidden="true" tabindex="-1"></a>result <span class="ot">&lt;-</span> result <span class="sc">%&gt;%</span></span>
<span id="cb34-1170"><a href="#cb34-1170" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(</span>
<span id="cb34-1171"><a href="#cb34-1171" aria-hidden="true" tabindex="-1"></a>    hv_family_collate <span class="sc">%&gt;%</span> <span class="fu">select</span>(taxid, family_n_reads_tot, family_ra_reads_tot),</span>
<span id="cb34-1172"><a href="#cb34-1172" aria-hidden="true" tabindex="-1"></a>    <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"family_taxid"</span> <span class="ot">=</span> <span class="st">"taxid"</span>)</span>
<span id="cb34-1173"><a href="#cb34-1173" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb34-1174"><a href="#cb34-1174" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1175"><a href="#cb34-1175" aria-hidden="true" tabindex="-1"></a><span class="co"># Clean up</span></span>
<span id="cb34-1176"><a href="#cb34-1176" aria-hidden="true" tabindex="-1"></a>result <span class="ot">&lt;-</span> result <span class="sc">%&gt;%</span></span>
<span id="cb34-1177"><a href="#cb34-1177" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="fu">ends_with</span>(<span class="st">"_taxid"</span>))</span>
<span id="cb34-1178"><a href="#cb34-1178" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1179"><a href="#cb34-1179" aria-hidden="true" tabindex="-1"></a><span class="co">#result</span></span>
<span id="cb34-1180"><a href="#cb34-1180" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-1181"><a href="#cb34-1181" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1182"><a href="#cb34-1182" aria-hidden="true" tabindex="-1"></a>Each dot represents a sample, colored by viral family. The x-axis shows the relative abundance of human-infecting viruses, and the y-axis shows the species. I've removed the species relating to the spiked-in family (Microviridae and Rhabdoviridae).</span>
<span id="cb34-1185"><a href="#cb34-1185" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-1186"><a href="#cb34-1186" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-pathogenic-viruses</span></span>
<span id="cb34-1187"><a href="#cb34-1187" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb34-1188"><a href="#cb34-1188" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 10</span></span>
<span id="cb34-1189"><a href="#cb34-1189" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb34-1190"><a href="#cb34-1190" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1191"><a href="#cb34-1191" aria-hidden="true" tabindex="-1"></a>species_family <span class="ot">&lt;-</span> result <span class="sc">%&gt;%</span> <span class="fu">select</span>(species, family) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="st">'name'</span> <span class="ot">=</span> <span class="st">'species'</span>)</span>
<span id="cb34-1192"><a href="#cb34-1192" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1193"><a href="#cb34-1193" aria-hidden="true" tabindex="-1"></a>play <span class="ot">&lt;-</span> hv_species_counts <span class="sc">%&gt;%</span> </span>
<span id="cb34-1194"><a href="#cb34-1194" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb34-1195"><a href="#cb34-1195" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by =</span> <span class="st">'sample'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-1196"><a href="#cb34-1196" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(species_family, <span class="at">by =</span> <span class="st">'name'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-1197"><a href="#cb34-1197" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span>(family <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Microviridae"</span>, <span class="st">"Rhabdoviridae"</span>))) </span>
<span id="cb34-1198"><a href="#cb34-1198" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1199"><a href="#cb34-1199" aria-hidden="true" tabindex="-1"></a>adjusted_play <span class="ot">&lt;-</span> play <span class="sc">%&gt;%</span> </span>
<span id="cb34-1200"><a href="#cb34-1200" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span></span>
<span id="cb34-1201"><a href="#cb34-1201" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">virus_prevalence_num =</span> <span class="fu">n_distinct</span>(sample)<span class="sc">/</span><span class="fu">n_distinct</span>(libraries),</span>
<span id="cb34-1202"><a href="#cb34-1202" aria-hidden="true" tabindex="-1"></a>         <span class="at">total_reads_hv =</span> <span class="fu">sum</span>(n_reads_hv)) <span class="sc">%&gt;%</span></span>
<span id="cb34-1203"><a href="#cb34-1203" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb34-1204"><a href="#cb34-1204" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">fct_reorder</span>(name, virus_prevalence_num, <span class="at">.desc=</span><span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb34-1205"><a href="#cb34-1205" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, ra_reads_hv, family, virus_prevalence_num, total_reads_hv, n_reads_hv)</span>
<span id="cb34-1206"><a href="#cb34-1206" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1207"><a href="#cb34-1207" aria-hidden="true" tabindex="-1"></a>pal <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">'Dark2'</span>),<span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">'Accent'</span>))</span>
<span id="cb34-1208"><a href="#cb34-1208" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1209"><a href="#cb34-1209" aria-hidden="true" tabindex="-1"></a><span class="co">#, labels = label_log(digits=2)</span></span>
<span id="cb34-1210"><a href="#cb34-1210" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1211"><a href="#cb34-1211" aria-hidden="true" tabindex="-1"></a>ra_dot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(adjusted_play, <span class="fu">aes</span>(<span class="at">x =</span> ra_reads_hv, <span class="at">y=</span>name)) <span class="sc">+</span></span>
<span id="cb34-1212"><a href="#cb34-1212" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_quasirandom</span>(<span class="at">orientation =</span> <span class="st">'y'</span>, <span class="fu">aes</span>(<span class="at">color =</span> family)) <span class="sc">+</span></span>
<span id="cb34-1213"><a href="#cb34-1213" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> pal) <span class="sc">+</span> </span>
<span id="cb34-1214"><a href="#cb34-1214" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_log10</span>(</span>
<span id="cb34-1215"><a href="#cb34-1215" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Relative abundance human virus reads"</span>,</span>
<span id="cb34-1216"><a href="#cb34-1216" aria-hidden="true" tabindex="-1"></a>    <span class="at">labels=</span><span class="fu">label_log</span>(<span class="at">digits=</span><span class="dv">3</span>)</span>
<span id="cb34-1217"><a href="#cb34-1217" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb34-1218"><a href="#cb34-1218" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span>  <span class="st">""</span>,</span>
<span id="cb34-1219"><a href="#cb34-1219" aria-hidden="true" tabindex="-1"></a>       <span class="at">color =</span> <span class="st">'Viral family'</span>) <span class="sc">+</span> </span>
<span id="cb34-1220"><a href="#cb34-1220" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color =</span> <span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">2</span>)) <span class="sc">+</span></span>
<span id="cb34-1221"><a href="#cb34-1221" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_light</span>() <span class="sc">+</span> </span>
<span id="cb34-1222"><a href="#cb34-1222" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(</span>
<span id="cb34-1223"><a href="#cb34-1223" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb34-1224"><a href="#cb34-1224" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>),</span>
<span id="cb34-1225"><a href="#cb34-1225" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.ticks.y =</span> <span class="fu">element_blank</span>(),<span class="at">axis.line =</span> <span class="fu">element_line</span>(<span class="at">colour =</span> <span class="st">"black"</span>),</span>
<span id="cb34-1226"><a href="#cb34-1226" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">14</span>),    </span>
<span id="cb34-1227"><a href="#cb34-1227" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb34-1228"><a href="#cb34-1228" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>, <span class="at">face=</span><span class="st">"bold"</span>),</span>
<span id="cb34-1229"><a href="#cb34-1229" aria-hidden="true" tabindex="-1"></a>    <span class="co">#legend.position = 'bottom',</span></span>
<span id="cb34-1230"><a href="#cb34-1230" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>),  <span class="co"># Move legend to top right</span></span>
<span id="cb34-1231"><a href="#cb34-1231" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.justification =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>),  <span class="co"># Align legend to top right</span></span>
<span id="cb34-1232"><a href="#cb34-1232" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb34-1233"><a href="#cb34-1233" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.border =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb34-1234"><a href="#cb34-1234" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.background =</span> <span class="fu">element_blank</span>())</span>
<span id="cb34-1235"><a href="#cb34-1235" aria-hidden="true" tabindex="-1"></a>ra_dot</span>
<span id="cb34-1236"><a href="#cb34-1236" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-1237"><a href="#cb34-1237" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1238"><a href="#cb34-1238" aria-hidden="true" tabindex="-1"></a>All of the viruses found above are known to be non-pathogenic in healthy humans.</span>
<span id="cb34-1239"><a href="#cb34-1239" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1240"><a href="#cb34-1240" aria-hidden="true" tabindex="-1"></a><span class="fu">## Relative abundance assuming perfect human read removal</span></span>
<span id="cb34-1243"><a href="#cb34-1243" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-1244"><a href="#cb34-1244" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv</span></span>
<span id="cb34-1245"><a href="#cb34-1245" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb34-1246"><a href="#cb34-1246" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1247"><a href="#cb34-1247" aria-hidden="true" tabindex="-1"></a>human_reads <span class="ot">&lt;-</span> comp_assigned <span class="sc">%&gt;%</span> <span class="fu">filter</span>(classification <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Human"</span>)) <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample, <span class="at">human_reads=</span>n_reads)</span>
<span id="cb34-1248"><a href="#cb34-1248" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1249"><a href="#cb34-1249" aria-hidden="true" tabindex="-1"></a>all <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span> </span>
<span id="cb34-1250"><a href="#cb34-1250" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(human_reads, <span class="at">by=</span><span class="st">'sample'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-1251"><a href="#cb34-1251" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb34-1252"><a href="#cb34-1252" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">sum_reads=</span><span class="fu">sum</span>(n_reads_hv), <span class="at">n_read_pairs=</span><span class="fu">first</span>(n_read_pairs) <span class="sc">-</span> <span class="fu">first</span>(human_reads)) <span class="sc">%&gt;%</span></span>
<span id="cb34-1253"><a href="#cb34-1253" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">ra =</span> sum_reads<span class="sc">/</span>n_read_pairs) <span class="sc">%&gt;%</span> </span>
<span id="cb34-1254"><a href="#cb34-1254" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(ra) <span class="sc">%&gt;%</span> <span class="fu">mean</span>()</span>
<span id="cb34-1255"><a href="#cb34-1255" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1256"><a href="#cb34-1256" aria-hidden="true" tabindex="-1"></a>all</span>
<span id="cb34-1257"><a href="#cb34-1257" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1258"><a href="#cb34-1258" aria-hidden="true" tabindex="-1"></a>prop_non <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span>  </span>
<span id="cb34-1259"><a href="#cb34-1259" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(name <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">'Rhabdoviridae'</span>,<span class="st">'Microviridae'</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb34-1260"><a href="#cb34-1260" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb34-1261"><a href="#cb34-1261" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">sum_reads_spiked =</span> <span class="fu">sum</span>(n_reads_hv)) <span class="sc">%&gt;%</span></span>
<span id="cb34-1262"><a href="#cb34-1262" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample,sum_reads_spiked)</span>
<span id="cb34-1263"><a href="#cb34-1263" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1264"><a href="#cb34-1264" aria-hidden="true" tabindex="-1"></a>non_spiked <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span> </span>
<span id="cb34-1265"><a href="#cb34-1265" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(human_reads, <span class="at">by=</span><span class="st">'sample'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-1266"><a href="#cb34-1266" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb34-1267"><a href="#cb34-1267" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">sum_reads=</span><span class="fu">sum</span>(n_reads_hv), <span class="at">n_read_pairs=</span><span class="fu">first</span>(n_read_pairs) <span class="sc">-</span> <span class="fu">first</span>(human_reads)) <span class="sc">%&gt;%</span></span>
<span id="cb34-1268"><a href="#cb34-1268" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(prop_non, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">'sample'</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb34-1269"><a href="#cb34-1269" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">ra_non_spiked =</span> (sum_reads<span class="sc">-</span>sum_reads_spiked)<span class="sc">/</span>(n_read_pairs<span class="sc">-</span>sum_reads_spiked)) <span class="sc">%&gt;%</span></span>
<span id="cb34-1270"><a href="#cb34-1270" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(ra_non_spiked) <span class="sc">%&gt;%</span> <span class="fu">mean</span>()</span>
<span id="cb34-1271"><a href="#cb34-1271" aria-hidden="true" tabindex="-1"></a>all</span>
<span id="cb34-1272"><a href="#cb34-1272" aria-hidden="true" tabindex="-1"></a>non_spiked</span>
<span id="cb34-1273"><a href="#cb34-1273" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb34-1274"><a href="#cb34-1274" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1275"><a href="#cb34-1275" aria-hidden="true" tabindex="-1"></a>Assuming we're able to perfectly remove all human reads, the average relative abundance of known human infecting virus is $4.08 \times 10^{-2}$ pre-viral spike removal and $3.64 \times 10^{-2}$ post-viral spike removal.</span>
<span id="cb34-1276"><a href="#cb34-1276" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1277"><a href="#cb34-1277" aria-hidden="true" tabindex="-1"></a><span class="fu"># Conclusion</span></span>
<span id="cb34-1278"><a href="#cb34-1278" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1279"><a href="#cb34-1279" aria-hidden="true" tabindex="-1"></a>Overall, we found a lot of anelloviridae, which makes sense given that this paper is about discovering new species of anelloviridae in blood.</span>
<span id="cb34-1280"><a href="#cb34-1280" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1281"><a href="#cb34-1281" aria-hidden="true" tabindex="-1"></a>There were some interesting takeways from this analysis:</span>
<span id="cb34-1282"><a href="#cb34-1282" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1283"><a href="#cb34-1283" aria-hidden="true" tabindex="-1"></a><span class="ss">1. </span>Sample processing specifically enhancing for viruses can increase the viral relative abundance substantially. In this study, we saw viral RA as high as $10^{-2}$ to $10^{-1}$ whereas in prior studies <span class="co">[</span><span class="ot">(1)</span><span class="co">](https://data.securebio.org/harmons-public-notebook/notebooks/2024-07-23-mengyi.html)</span> <span class="co">[</span><span class="ot">(2)</span><span class="co">](https://data.securebio.org/harmons-public-notebook/notebooks/2024-07-22-thijssen.html)</span> we saw viral RA around the $10^{-4}$ to $10^{-3}$ range.</span>
<span id="cb34-1284"><a href="#cb34-1284" aria-hidden="true" tabindex="-1"></a><span class="ss">2. </span>A lot of the Anelloviridae are largely uncharacterized (denoted by the family name followed by the abbreviation "sp.").</span>
<span id="cb34-1285"><a href="#cb34-1285" aria-hidden="true" tabindex="-1"></a><span class="ss">3. </span>In small sample populations, it's unlikely for us to see <span class="co">[</span><span class="ot">well-known blood viruses</span><span class="co">](https://virology.ws/2017/03/23/the-viruses-in-your-blood/)</span>.</span>
<span id="cb34-1286"><a href="#cb34-1286" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1287"><a href="#cb34-1287" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1288"><a href="#cb34-1288" aria-hidden="true" tabindex="-1"></a><span class="fu"># Appendix</span></span>
<span id="cb34-1289"><a href="#cb34-1289" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1290"><a href="#cb34-1290" aria-hidden="true" tabindex="-1"></a><span class="fu">## Human-infecting virus families, genera, and species</span></span>
<span id="cb34-1291"><a href="#cb34-1291" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1292"><a href="#cb34-1292" aria-hidden="true" tabindex="-1"></a>To get a good overview of families, genera, and species, we can look at a Sankey plot where the magnitude of relative abundance, averaged over all samples, is shown in parentheses.</span>
<span id="cb34-1293"><a href="#cb34-1293" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1296"><a href="#cb34-1296" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb34-1297"><a href="#cb34-1297" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-sankey</span></span>
<span id="cb34-1298"><a href="#cb34-1298" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 22.5</span></span>
<span id="cb34-1299"><a href="#cb34-1299" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb34-1300"><a href="#cb34-1300" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb34-1301"><a href="#cb34-1301" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1302"><a href="#cb34-1302" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to create links</span></span>
<span id="cb34-1303"><a href="#cb34-1303" aria-hidden="true" tabindex="-1"></a>create_links <span class="ot">&lt;-</span> <span class="cf">function</span>(data) {</span>
<span id="cb34-1304"><a href="#cb34-1304" aria-hidden="true" tabindex="-1"></a>  family_to_genus <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb34-1305"><a href="#cb34-1305" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(genus)) <span class="sc">%&gt;%</span></span>
<span id="cb34-1306"><a href="#cb34-1306" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(family, genus) <span class="sc">%&gt;%</span></span>
<span id="cb34-1307"><a href="#cb34-1307" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">value =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-1308"><a href="#cb34-1308" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">source =</span> family, <span class="at">target =</span> genus)</span>
<span id="cb34-1309"><a href="#cb34-1309" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb34-1310"><a href="#cb34-1310" aria-hidden="true" tabindex="-1"></a>  genus_to_species <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb34-1311"><a href="#cb34-1311" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(genus, species) <span class="sc">%&gt;%</span></span>
<span id="cb34-1312"><a href="#cb34-1312" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">value =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-1313"><a href="#cb34-1313" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">source =</span> genus, <span class="at">target =</span> species)</span>
<span id="cb34-1314"><a href="#cb34-1314" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1315"><a href="#cb34-1315" aria-hidden="true" tabindex="-1"></a>  family_to_species <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb34-1316"><a href="#cb34-1316" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="fu">is.na</span>(genus)) <span class="sc">%&gt;%</span></span>
<span id="cb34-1317"><a href="#cb34-1317" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(family, species) <span class="sc">%&gt;%</span></span>
<span id="cb34-1318"><a href="#cb34-1318" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">value =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-1319"><a href="#cb34-1319" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">source =</span> family, <span class="at">target =</span> species)</span>
<span id="cb34-1320"><a href="#cb34-1320" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1321"><a href="#cb34-1321" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_rows</span>(family_to_genus, genus_to_species, family_to_species) <span class="sc">%&gt;%</span></span>
<span id="cb34-1322"><a href="#cb34-1322" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(source))</span>
<span id="cb34-1323"><a href="#cb34-1323" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb34-1324"><a href="#cb34-1324" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1325"><a href="#cb34-1325" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to create nodes</span></span>
<span id="cb34-1326"><a href="#cb34-1326" aria-hidden="true" tabindex="-1"></a>create_nodes <span class="ot">&lt;-</span> <span class="cf">function</span>(links) {</span>
<span id="cb34-1327"><a href="#cb34-1327" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>(</span>
<span id="cb34-1328"><a href="#cb34-1328" aria-hidden="true" tabindex="-1"></a>    <span class="at">name =</span> <span class="fu">c</span>(links<span class="sc">$</span>source, links<span class="sc">$</span>target) <span class="sc">%&gt;%</span> <span class="fu">unique</span>()</span>
<span id="cb34-1329"><a href="#cb34-1329" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb34-1330"><a href="#cb34-1330" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb34-1331"><a href="#cb34-1331" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1332"><a href="#cb34-1332" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to prepare data for Sankey diagram</span></span>
<span id="cb34-1333"><a href="#cb34-1333" aria-hidden="true" tabindex="-1"></a>prepare_sankey_data <span class="ot">&lt;-</span> <span class="cf">function</span>(links, nodes) {</span>
<span id="cb34-1334"><a href="#cb34-1334" aria-hidden="true" tabindex="-1"></a>  links<span class="sc">$</span>IDsource <span class="ot">&lt;-</span> <span class="fu">match</span>(links<span class="sc">$</span>source, nodes<span class="sc">$</span>name) <span class="sc">-</span> <span class="dv">1</span></span>
<span id="cb34-1335"><a href="#cb34-1335" aria-hidden="true" tabindex="-1"></a>  links<span class="sc">$</span>IDtarget <span class="ot">&lt;-</span> <span class="fu">match</span>(links<span class="sc">$</span>target, nodes<span class="sc">$</span>name) <span class="sc">-</span> <span class="dv">1</span></span>
<span id="cb34-1336"><a href="#cb34-1336" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list</span>(<span class="at">links =</span> links, <span class="at">nodes =</span> nodes)</span>
<span id="cb34-1337"><a href="#cb34-1337" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb34-1338"><a href="#cb34-1338" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1339"><a href="#cb34-1339" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to create Sankey plot</span></span>
<span id="cb34-1340"><a href="#cb34-1340" aria-hidden="true" tabindex="-1"></a>create_sankey_plot <span class="ot">&lt;-</span> <span class="cf">function</span>(sankey_data) {</span>
<span id="cb34-1341"><a href="#cb34-1341" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sankeyNetwork</span>(</span>
<span id="cb34-1342"><a href="#cb34-1342" aria-hidden="true" tabindex="-1"></a>    <span class="at">Links =</span> sankey_data<span class="sc">$</span>links, </span>
<span id="cb34-1343"><a href="#cb34-1343" aria-hidden="true" tabindex="-1"></a>    <span class="at">Nodes =</span> sankey_data<span class="sc">$</span>nodes,</span>
<span id="cb34-1344"><a href="#cb34-1344" aria-hidden="true" tabindex="-1"></a>    <span class="at">Source =</span> <span class="st">"IDsource"</span>, </span>
<span id="cb34-1345"><a href="#cb34-1345" aria-hidden="true" tabindex="-1"></a>    <span class="at">Target =</span> <span class="st">"IDtarget"</span>,</span>
<span id="cb34-1346"><a href="#cb34-1346" aria-hidden="true" tabindex="-1"></a>    <span class="at">Value =</span> <span class="st">"value"</span>, </span>
<span id="cb34-1347"><a href="#cb34-1347" aria-hidden="true" tabindex="-1"></a>    <span class="at">NodeID =</span> <span class="st">"name"</span>,</span>
<span id="cb34-1348"><a href="#cb34-1348" aria-hidden="true" tabindex="-1"></a>    <span class="at">sinksRight =</span> <span class="cn">TRUE</span>,</span>
<span id="cb34-1349"><a href="#cb34-1349" aria-hidden="true" tabindex="-1"></a>    <span class="at">nodeWidth =</span> <span class="dv">25</span>,</span>
<span id="cb34-1350"><a href="#cb34-1350" aria-hidden="true" tabindex="-1"></a>    <span class="at">fontSize =</span> <span class="dv">14</span>,</span>
<span id="cb34-1351"><a href="#cb34-1351" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb34-1352"><a href="#cb34-1352" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb34-1353"><a href="#cb34-1353" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1354"><a href="#cb34-1354" aria-hidden="true" tabindex="-1"></a>save_sankey_as_png <span class="ot">&lt;-</span> <span class="cf">function</span>(sankey_plot, <span class="at">width =</span> <span class="dv">1000</span>, <span class="at">height =</span> <span class="dv">800</span>) {</span>
<span id="cb34-1355"><a href="#cb34-1355" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Save the plot as an HTML file</span></span>
<span id="cb34-1356"><a href="#cb34-1356" aria-hidden="true" tabindex="-1"></a>  <span class="fu">saveWidget</span>(sankey_plot, <span class="fu">sprintf</span>(<span class="st">'%s/sankey.html'</span>,data_dir))</span>
<span id="cb34-1357"><a href="#cb34-1357" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb34-1358"><a href="#cb34-1358" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1359"><a href="#cb34-1359" aria-hidden="true" tabindex="-1"></a>format_scientific <span class="ot">&lt;-</span> <span class="cf">function</span>(x, <span class="at">digits=</span><span class="dv">2</span>) {</span>
<span id="cb34-1360"><a href="#cb34-1360" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sapply</span>(x, <span class="cf">function</span>(val) {</span>
<span id="cb34-1361"><a href="#cb34-1361" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">is.na</span>(val) <span class="sc">||</span> <span class="fu">abs</span>(val) <span class="sc">&lt;</span> <span class="fl">1e-15</span>) {</span>
<span id="cb34-1362"><a href="#cb34-1362" aria-hidden="true" tabindex="-1"></a>      <span class="fu">return</span>(<span class="st">"0"</span>)</span>
<span id="cb34-1363"><a href="#cb34-1363" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb34-1364"><a href="#cb34-1364" aria-hidden="true" tabindex="-1"></a>      exponent <span class="ot">&lt;-</span> <span class="fu">floor</span>(<span class="fu">log10</span>(<span class="fu">abs</span>(val)))</span>
<span id="cb34-1365"><a href="#cb34-1365" aria-hidden="true" tabindex="-1"></a>      coef <span class="ot">&lt;-</span> val <span class="sc">/</span> <span class="dv">10</span><span class="sc">^</span>exponent</span>
<span id="cb34-1366"><a href="#cb34-1366" aria-hidden="true" tabindex="-1"></a>      <span class="co">#return(sprintf("%.1f × 10^%d", round(coef, digits), exponent))</span></span>
<span id="cb34-1367"><a href="#cb34-1367" aria-hidden="true" tabindex="-1"></a>      <span class="co"># May or may not be smart, just keeping magnitude</span></span>
<span id="cb34-1368"><a href="#cb34-1368" aria-hidden="true" tabindex="-1"></a>      <span class="fu">return</span>(<span class="fu">sprintf</span>(<span class="st">"10^%d"</span>, exponent))</span>
<span id="cb34-1369"><a href="#cb34-1369" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb34-1370"><a href="#cb34-1370" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb34-1371"><a href="#cb34-1371" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb34-1372"><a href="#cb34-1372" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1373"><a href="#cb34-1373" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> result <span class="sc">%&gt;%</span> </span>
<span id="cb34-1374"><a href="#cb34-1374" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">c</span>(genus_n_reads_tot, genus_ra_reads_tot), <span class="sc">~</span><span class="fu">replace_na</span>(., <span class="dv">0</span>)),</span>
<span id="cb34-1375"><a href="#cb34-1375" aria-hidden="true" tabindex="-1"></a>         <span class="at">genus =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(genus), <span class="st">"Unknown Genus"</span>, genus)) <span class="sc">%&gt;%</span></span>
<span id="cb34-1376"><a href="#cb34-1376" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb34-1377"><a href="#cb34-1377" aria-hidden="true" tabindex="-1"></a>  <span class="at">species =</span> <span class="fu">paste0</span>(species, <span class="fu">sprintf</span>(<span class="st">' (%s)'</span>, <span class="fu">format_scientific</span>(species_ra_reads_tot))),</span>
<span id="cb34-1378"><a href="#cb34-1378" aria-hidden="true" tabindex="-1"></a>  <span class="at">genus =</span> <span class="fu">paste0</span>(genus, <span class="fu">sprintf</span>(<span class="st">' (%s)'</span>, <span class="fu">format_scientific</span>(genus_ra_reads_tot))),</span>
<span id="cb34-1379"><a href="#cb34-1379" aria-hidden="true" tabindex="-1"></a>  <span class="at">family =</span> <span class="fu">paste0</span>(family, <span class="fu">sprintf</span>(<span class="st">' (%s)'</span>, <span class="fu">format_scientific</span>(family_ra_reads_tot)))</span>
<span id="cb34-1380"><a href="#cb34-1380" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb34-1381"><a href="#cb34-1381" aria-hidden="true" tabindex="-1"></a>links <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">create_links</span>(data))</span>
<span id="cb34-1382"><a href="#cb34-1382" aria-hidden="true" tabindex="-1"></a>nodes <span class="ot">&lt;-</span> <span class="fu">create_nodes</span>(links)</span>
<span id="cb34-1383"><a href="#cb34-1383" aria-hidden="true" tabindex="-1"></a>sankey_data <span class="ot">&lt;-</span> <span class="fu">prepare_sankey_data</span>(links, nodes)</span>
<span id="cb34-1384"><a href="#cb34-1384" aria-hidden="true" tabindex="-1"></a>sankey <span class="ot">&lt;-</span> <span class="fu">create_sankey_plot</span>(sankey_data)</span>
<span id="cb34-1385"><a href="#cb34-1385" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-1386"><a href="#cb34-1386" aria-hidden="true" tabindex="-1"></a>sankey</span>
<span id="cb34-1387"><a href="#cb34-1387" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>