<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.53">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Harmon Bhasin">
<meta name="dcterms.date" content="2024-09-24">

<title>Blog 3: Pre-analysis – Harmon’s Public Notebook</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script async="" src="https://hypothes.is/embed.js"></script>
<script>
  window.document.addEventListener("DOMContentLoaded", function (_event) {
    document.body.classList.add('hypothesis-enabled');
  });
</script>
<style>

      .quarto-title-block .quarto-title-banner {
        background: #de2d26;
      }
</style>
<link href="../site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet">
<script src="../site_libs/pagedtable-1.1/js/pagedtable.js"></script>


<link rel="stylesheet" href="../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top quarto-banner">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">Harmon’s Public Notebook</span>
    </a>
  </div>
        <div class="quarto-navbar-tools tools-end">
</div>
          <div id="quarto-search" class="" title="Search"></div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-page-left">
      <h1 class="title">Blog 3: Pre-analysis</h1>
            <p class="subtitle lead">Combined whole blood and plasma analysis</p>
                      </div>
  </div>
    
  
  <div class="quarto-title-meta column-page-left">

      <div>
      <div class="quarto-title-meta-heading">Author</div>
      <div class="quarto-title-meta-contents">
               <p>Harmon Bhasin </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">September 24, 2024</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction"><span class="header-section-number">1</span> Introduction</a></li>
  <li><a href="#data" id="toc-data" class="nav-link" data-scroll-target="#data"><span class="header-section-number">2</span> Data</a>
  <ul class="collapse">
  <li><a href="#plasma" id="toc-plasma" class="nav-link" data-scroll-target="#plasma"><span class="header-section-number">2.1</span> Plasma</a>
  <ul class="collapse">
  <li><a href="#cebria-mendoza-et-al.-2021" id="toc-cebria-mendoza-et-al.-2021" class="nav-link" data-scroll-target="#cebria-mendoza-et-al.-2021"><span class="header-section-number">2.1.1</span> Cebria-Mendoza et al.&nbsp;(2021)</a></li>
  <li><a href="#thijssen-et-al.-2023" id="toc-thijssen-et-al.-2023" class="nav-link" data-scroll-target="#thijssen-et-al.-2023"><span class="header-section-number">2.1.2</span> Thijssen et al.&nbsp;(2023)</a></li>
  <li><a href="#mengyi-et-al.-2023" id="toc-mengyi-et-al.-2023" class="nav-link" data-scroll-target="#mengyi-et-al.-2023"><span class="header-section-number">2.1.3</span> Mengyi et al.&nbsp;(2023)</a></li>
  </ul></li>
  <li><a href="#whole-blood" id="toc-whole-blood" class="nav-link" data-scroll-target="#whole-blood"><span class="header-section-number">2.2</span> Whole blood</a>
  <ul class="collapse">
  <li><a href="#thompson-et-al.-2023" id="toc-thompson-et-al.-2023" class="nav-link" data-scroll-target="#thompson-et-al.-2023"><span class="header-section-number">2.2.1</span> Thompson et al.&nbsp;(2023)</a></li>
  <li><a href="#oconnell-et-al.-2023" id="toc-oconnell-et-al.-2023" class="nav-link" data-scroll-target="#oconnell-et-al.-2023"><span class="header-section-number">2.2.2</span> O’Connell et al.&nbsp;(2023)</a></li>
  <li><a href="#aydillo-et-al.-2022" id="toc-aydillo-et-al.-2022" class="nav-link" data-scroll-target="#aydillo-et-al.-2022"><span class="header-section-number">2.2.3</span> Aydillo et al.&nbsp;(2022)</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#methods" id="toc-methods" class="nav-link" data-scroll-target="#methods"><span class="header-section-number">3</span> Methods</a></li>
  <li><a href="#results" id="toc-results" class="nav-link" data-scroll-target="#results"><span class="header-section-number">4</span> Results</a>
  <ul class="collapse">
  <li><a href="#kingdom-composition" id="toc-kingdom-composition" class="nav-link" data-scroll-target="#kingdom-composition"><span class="header-section-number">4.1</span> Kingdom composition</a></li>
  <li><a href="#human-infecting-viruses" id="toc-human-infecting-viruses" class="nav-link" data-scroll-target="#human-infecting-viruses"><span class="header-section-number">4.2</span> Human-infecting viruses</a>
  <ul class="collapse">
  <li><a href="#overview" id="toc-overview" class="nav-link" data-scroll-target="#overview"><span class="header-section-number">4.2.1</span> Overview</a></li>
  <li><a href="#family-composition" id="toc-family-composition" class="nav-link" data-scroll-target="#family-composition"><span class="header-section-number">4.2.2</span> Family composition</a></li>
  <li><a href="#species-composition" id="toc-species-composition" class="nav-link" data-scroll-target="#species-composition"><span class="header-section-number">4.2.3</span> Species composition</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#discussion" id="toc-discussion" class="nav-link" data-scroll-target="#discussion"><span class="header-section-number">5</span> Discussion</a></li>
  <li><a href="#conclusion" id="toc-conclusion" class="nav-link" data-scroll-target="#conclusion"><span class="header-section-number">6</span> Conclusion</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content quarto-banner-title-block column-page-left" id="quarto-document-content">





<section id="introduction" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Introduction</h1>
<p>In our <a href="https://naobservatory.org/blog/exploring-blood-biosurveillance-part2">previous post</a>, we identfied four sampling strategies, each coming from either whole blood or plasma. To take this analysis a step further, we analyze the metagenomic sequencing data from various whole blood and plasma based studies. This analysis will help us understand the last piece missing from the puzzle: what viruses are present and detectable in whole blood and plasma using metagenomic sequencing and how do both sample types compare to each other.</p>
<p>We performed a literature search for large (&gt;100M read pairs), untargeted whole blood and plasma metagenomic studies. We identified 6 such datasets (Table 1), obtained the raw sequencing data, and processed it with a Kraken2-based computational pipeline to estimate the relative abundance of human-infecting viruses (Methods).</p>
</section>
<section id="data" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Data</h1>
<table class="table-striped table-hover caption-top table">
<caption><strong>Table 1</strong>: Studies included in the analysis.</caption>
<thead>
<tr class="header">
<th><strong>Study</strong></th>
<th><strong>Sample type</strong></th>
<th><strong>Human read removal (during sample prep)</strong></th>
<th><strong>Sequencing type</strong></th>
<th><strong>Country</strong></th>
<th><strong>Read pairs</strong></th>
<th><strong>Reference</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Cebria-Mendoza et al.&nbsp;(2021)</td>
<td>Plasma</td>
<td>Yes</td>
<td>DNA + RNA</td>
<td>Spain</td>
<td>230M</td>
<td><a href="https://doi.org/10.3390/v13112322">link</a></td>
</tr>
<tr class="even">
<td>Thijssen et al.&nbsp;(2023)</td>
<td>Plasma</td>
<td>No</td>
<td>DNA + RNA</td>
<td>Iran</td>
<td>116M</td>
<td><a href="https://doi.org/10.3390/v15071425">link</a></td>
</tr>
<tr class="odd">
<td>Mengyi et al.&nbsp;(2023)</td>
<td>Plasma</td>
<td>No</td>
<td>DNA</td>
<td>China</td>
<td>3.4B</td>
<td><a href="https://doi.org/10.1016%2Fj.onehlt.2023.100602">link</a></td>
</tr>
<tr class="even">
<td>Thompson et al.&nbsp;(2023)</td>
<td>Whole blood</td>
<td>No</td>
<td>RNA</td>
<td>USA</td>
<td>27.7B</td>
<td><a href="https://doi.org/10.1038/s41591-022-02107-4">link</a></td>
</tr>
<tr class="odd">
<td>O’Connell et al.&nbsp;(2023)</td>
<td>Whole blood</td>
<td>No</td>
<td>RNA</td>
<td>USA</td>
<td>3.1B</td>
<td><a href="https://doi.org/10.1186/s12863-024-01223-z">link</a></td>
</tr>
<tr class="even">
<td>Aydillo et al.&nbsp;(2022)</td>
<td>Whole blood</td>
<td>No</td>
<td>RNA</td>
<td>USA</td>
<td>1.7B</td>
<td><a href="https://doi.org/10.1038/s41541-022-00583-w">link</a></td>
</tr>
</tbody>
</table>
<section id="plasma" class="level2" data-number="2.1">
<h2 data-number="2.1" class="anchored" data-anchor-id="plasma"><span class="header-section-number">2.1</span> Plasma</h2>
<section id="cebria-mendoza-et-al.-2021" class="level3" data-number="2.1.1">
<h3 data-number="2.1.1" class="anchored" data-anchor-id="cebria-mendoza-et-al.-2021"><span class="header-section-number">2.1.1</span> Cebria-Mendoza et al.&nbsp;(2021)</h3>
<p>This dataset from Spain has 60 samples, each derived from plasma pools of 8-13 people from Spain. In total, 567 healthy individuals contributed to these pools. For each pooled sample, a combined DNA and RNA library preparation was performed, resulting in a single sequencing output that captures both nucleic acid types. This approach provides comprehensive genetic information but precludes separate analysis of DNA and RNA from individual samples.</p>
<p><em>Full analysis can be found <a href="https://data.securebio.org/harmons-public-notebook/notebooks/2024-07-08_cebria-mendoza.html">here</a></em></p>
</section>
<section id="thijssen-et-al.-2023" class="level3" data-number="2.1.2">
<h3 data-number="2.1.2" class="anchored" data-anchor-id="thijssen-et-al.-2023"><span class="header-section-number">2.1.2</span> Thijssen et al.&nbsp;(2023)</h3>
<p>This dataset from Iran has 21 samples, each derived from plasma pools of 5 people from Iran. In total, 100 healthy individuals contributed to these pools. For each pooled sample, a combined DNA and RNA library preparation was performed, resulting in a single sequencing output that captures both nucleic acid types, with Illumina NextSeq 500, producing 2x150 bp reads.</p>
<p><em>Full analysis can be found <a href="https://data.securebio.org/harmons-public-notebook/notebooks/2024-07-22-thijssen.html">here</a></em></p>
</section>
<section id="mengyi-et-al.-2023" class="level3" data-number="2.1.3">
<h3 data-number="2.1.3" class="anchored" data-anchor-id="mengyi-et-al.-2023"><span class="header-section-number">2.1.3</span> Mengyi et al.&nbsp;(2023)</h3>
<p>This dataset from China has 201 samples, each derived from plasma pools of 160 donations from 7 different locations between 2012-2018. In total, 10,720 donations contributed to these pools (we do not know the number of individuals). They did DNA-sequencing for each pool, with Illumina HiSeq 4500, producing 2x150 bp reads.</p>
<p><em>Full analysis can be found <a href="https://data.securebio.org/harmons-public-notebook/notebooks/2024-07-23-mengyi.html">here</a></em></p>
</section>
</section>
<section id="whole-blood" class="level2" data-number="2.2">
<h2 data-number="2.2" class="anchored" data-anchor-id="whole-blood"><span class="header-section-number">2.2</span> Whole blood</h2>
<section id="thompson-et-al.-2023" class="level3" data-number="2.2.1">
<h3 data-number="2.2.1" class="anchored" data-anchor-id="thompson-et-al.-2023"><span class="header-section-number">2.2.1</span> Thompson et al.&nbsp;(2023)</h3>
<p>This dataset from the USA has 417 samples, with 353 individuals positive for SARS-CoV-2 and 65 individuals negative, for a total of 418 individuals. They did RNA-sequencing for each sample, with Illumina NovaSeq 6000, producing 2x100 bp reads.</p>
<p><em>Full analysis is in progress</em></p>
</section>
<section id="oconnell-et-al.-2023" class="level3" data-number="2.2.2">
<h3 data-number="2.2.2" class="anchored" data-anchor-id="oconnell-et-al.-2023"><span class="header-section-number">2.2.2</span> O’Connell et al.&nbsp;(2023)</h3>
<p>This dataset from the USA has 138 samples, with 138 individuals presenting to the emergency department with symptoms or complications. They did RNA-sequencing for each sample using Illumina NovaSeq 6000, producing 2x150 bp reads.</p>
<p><em>Full analysis is in progress</em></p>
</section>
<section id="aydillo-et-al.-2022" class="level3" data-number="2.2.3">
<h3 data-number="2.2.3" class="anchored" data-anchor-id="aydillo-et-al.-2022"><span class="header-section-number">2.2.3</span> Aydillo et al.&nbsp;(2022)</h3>
<p>This dataset from the USA has 53 samples, with 53 healthy individuals. They did RNA-sequencing for each sample, with Illumina NovaSeq 6000, producing 2 x 95 bp reads.</p>
<p><em>Full analysis is in progress</em></p>
</section>
</section>
</section>
<section id="methods" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> Methods</h1>
<p>The pipeline for analysis was developed by William Bradshaw, and slightly modifyed by me. It can be found <a href="https://github.com/harmonbhasin/mgs-workflow">here</a>. Here is a brief description of the pipeline, taken directly from the README:</p>
<blockquote class="blockquote">
<p>The run workflow then consists of four phases:</p>
<ol type="1">
<li>A preprocessing phase, in which input files undergo adapter &amp; quality trimming, deduplication, and ribodepletion.</li>
<li>A taxonomic profiling phase, in which Kraken2 is used to assess the overall taxonomic composition of the input data and assess how it changes across different steps in the preprocessing phase.</li>
<li>A viral identification phase, in which a custom multi-step pipeline based around Bowtie2 and Kraken2 is used to sensitively and specifically identify human-infecting virus (HV) reads in the input data for downstream analysis.</li>
<li>A final QC and output phase, in which FASTQC, MultiQC and other tools are used to assess the quality of the data produced by the pipeline at various steps and produce summarized output data for downstream analysis and visualization.</li>
</ol>
</blockquote>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="2024-09-24-blog3-preanalysis_files/figure-html/simplified-diagram.png?" class="img-fluid figure-img"></p>
<figcaption>Diagram of the analysis pipeline (we do not use the BLAST Validation and only use FASTP for quality trimming and cleaning).</figcaption>
</figure>
</div>
</section>
<section id="results" class="level1" data-number="4">
<h1 data-number="4"><span class="header-section-number">4</span> Results</h1>
<section id="kingdom-composition" class="level2" data-number="4.1">
<h2 data-number="4.1" class="anchored" data-anchor-id="kingdom-composition"><span class="header-section-number">4.1</span> Kingdom composition</h2>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="2024-09-24-blog3-preanalysis_files/figure-html/kingdom-fraction-plot-1.svg?" class="img-fluid figure-img"></p>
<figcaption><strong>Figure 1</strong>: Read composition</figcaption>
</figure>
</div>
</div>
</div>
<p>Comparing the average kingdom fraction between sample types reveals that whole blood samples have higher human fractions than plasma, whereas plasma has higher unassigned, archaeal, bacterial, and viral fractions. However, if we instead only look at the plasma dataset that did viral enrichment, we see a stark difference in composition. Specifically, the plasma samples only have around 2% human reads, with around 6% viral reads.</p>
<div class="cell">
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<colgroup>
<col style="width: 21%">
<col style="width: 14%">
<col style="width: 12%">
<col style="width: 13%">
<col style="width: 13%">
<col style="width: 13%">
<col style="width: 12%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">dataset</th>
<th style="text-align: left;">sample_type</th>
<th style="text-align: left;">Archaeal</th>
<th style="text-align: left;">Bacterial</th>
<th style="text-align: left;">Human</th>
<th style="text-align: left;">Unassigned</th>
<th style="text-align: left;">Viral</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">cebriamendoza2021</td>
<td style="text-align: left;">plasma</td>
<td style="text-align: left;">0.03311 %</td>
<td style="text-align: left;">49.89189 %</td>
<td style="text-align: left;">1.92698 %</td>
<td style="text-align: left;">42.16360 %</td>
<td style="text-align: left;">5.98442 %</td>
</tr>
<tr class="even">
<td style="text-align: left;">thijssen2023</td>
<td style="text-align: left;">plasma</td>
<td style="text-align: left;">0.01280 %</td>
<td style="text-align: left;">4.51508 %</td>
<td style="text-align: left;">22.31281 %</td>
<td style="text-align: left;">73.12558 %</td>
<td style="text-align: left;">0.03373 %</td>
</tr>
<tr class="odd">
<td style="text-align: left;">mengyi2023</td>
<td style="text-align: left;">plasma</td>
<td style="text-align: left;">0.00096 %</td>
<td style="text-align: left;">0.26607 %</td>
<td style="text-align: left;">98.38344 %</td>
<td style="text-align: left;">1.33651 %</td>
<td style="text-align: left;">0.01302 %</td>
</tr>
<tr class="even">
<td style="text-align: left;">thompson2023</td>
<td style="text-align: left;">whole-blood</td>
<td style="text-align: left;">0.00007 %</td>
<td style="text-align: left;">0.01282 %</td>
<td style="text-align: left;">99.87494 %</td>
<td style="text-align: left;">0.11189 %</td>
<td style="text-align: left;">0.00027 %</td>
</tr>
<tr class="odd">
<td style="text-align: left;">oconnell2023</td>
<td style="text-align: left;">whole-blood</td>
<td style="text-align: left;">0.00004 %</td>
<td style="text-align: left;">0.05051 %</td>
<td style="text-align: left;">99.86217 %</td>
<td style="text-align: left;">0.08717 %</td>
<td style="text-align: left;">0.00011 %</td>
</tr>
<tr class="even">
<td style="text-align: left;">aydillo2022</td>
<td style="text-align: left;">whole-blood</td>
<td style="text-align: left;">0.00061 %</td>
<td style="text-align: left;">0.10959 %</td>
<td style="text-align: left;">99.54011 %</td>
<td style="text-align: left;">0.34940 %</td>
<td style="text-align: left;">0.00028 %</td>
</tr>
</tbody>
</table>
<p><strong>Table 2</strong>: Read composition</p>
</div>
</div>
<div class="cell">
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<colgroup>
<col style="width: 9%">
<col style="width: 6%">
<col style="width: 20%">
<col style="width: 21%">
<col style="width: 21%">
<col style="width: 19%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">dataset</th>
<th style="text-align: left;">sample_type</th>
<th style="text-align: left;">Archaeal (controlled for human reads)</th>
<th style="text-align: left;">Bacterial (controlled for human reads)</th>
<th style="text-align: left;">Unassigned (controlled for human reads)</th>
<th style="text-align: left;">Viral (controlled for human reads)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">cebriamendoza2021</td>
<td style="text-align: left;">plasma</td>
<td style="text-align: left;">0.03376 %</td>
<td style="text-align: left;">50.87219 %</td>
<td style="text-align: left;">42.99205 %</td>
<td style="text-align: left;">6.10200 %</td>
</tr>
<tr class="even">
<td style="text-align: left;">thijssen2023</td>
<td style="text-align: left;">plasma</td>
<td style="text-align: left;">0.01648 %</td>
<td style="text-align: left;">5.81187 %</td>
<td style="text-align: left;">94.12824 %</td>
<td style="text-align: left;">0.04341 %</td>
</tr>
<tr class="odd">
<td style="text-align: left;">mengyi2023</td>
<td style="text-align: left;">plasma</td>
<td style="text-align: left;">0.05935 %</td>
<td style="text-align: left;">16.45903 %</td>
<td style="text-align: left;">82.67637 %</td>
<td style="text-align: left;">0.80525 %</td>
</tr>
<tr class="even">
<td style="text-align: left;">thompson2023</td>
<td style="text-align: left;">whole-blood</td>
<td style="text-align: left;">0.05890 %</td>
<td style="text-align: left;">10.25520 %</td>
<td style="text-align: left;">89.47051 %</td>
<td style="text-align: left;">0.21539 %</td>
</tr>
<tr class="odd">
<td style="text-align: left;">oconnell2023</td>
<td style="text-align: left;">whole-blood</td>
<td style="text-align: left;">0.02763 %</td>
<td style="text-align: left;">36.64757 %</td>
<td style="text-align: left;">63.24574 %</td>
<td style="text-align: left;">0.07906 %</td>
</tr>
<tr class="even">
<td style="text-align: left;">aydillo2022</td>
<td style="text-align: left;">whole-blood</td>
<td style="text-align: left;">0.13290 %</td>
<td style="text-align: left;">23.83067 %</td>
<td style="text-align: left;">75.97604 %</td>
<td style="text-align: left;">0.06039 %</td>
</tr>
</tbody>
</table>
<p><strong>Table 3</strong>: Read composition without human reads</p>
</div>
</div>
<p>Another way we can look at this is by controlling for human reads across all datasets. When we do this over all whole blood and plasma datasets, we can see that they have similar read composition across the board. However, plasma has a higher viral read fraction by a magnitude of 10.</p>
</section>
<section id="human-infecting-viruses" class="level2" data-number="4.2">
<h2 data-number="4.2" class="anchored" data-anchor-id="human-infecting-viruses"><span class="header-section-number">4.2</span> Human-infecting viruses</h2>
<section id="overview" class="level3" data-number="4.2.1">
<h3 data-number="4.2.1" class="anchored" data-anchor-id="overview"><span class="header-section-number">4.2.1</span> Overview</h3>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="2024-09-24-blog3-preanalysis_files/figure-html/count-hv-reads-1.svg?" class="img-fluid figure-img"></p>
<figcaption><strong>Figure 2</strong>: Proportion of all reads in each dataset</figcaption>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<thead>
<tr class="header">
<th style="text-align: left;">dataset</th>
<th style="text-align: left;">RA</th>
<th style="text-align: left;">RA (control for human reads)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">cebriamendoza2021</td>
<td style="text-align: left;">4.06e-02</td>
<td style="text-align: left;">4.08e-02</td>
</tr>
<tr class="even">
<td style="text-align: left;">thijssen2023</td>
<td style="text-align: left;">1.60e-04</td>
<td style="text-align: left;">1.82e-04</td>
</tr>
<tr class="odd">
<td style="text-align: left;">mengyi2023</td>
<td style="text-align: left;">9.84e-05</td>
<td style="text-align: left;">5.11e-04</td>
</tr>
<tr class="even">
<td style="text-align: left;">thompson2023</td>
<td style="text-align: left;">1.51e-06</td>
<td style="text-align: left;">3.18e-06</td>
</tr>
<tr class="odd">
<td style="text-align: left;">oconnell2023</td>
<td style="text-align: left;">2.62e-08</td>
<td style="text-align: left;">8.87e-08</td>
</tr>
<tr class="even">
<td style="text-align: left;">aydillo2022</td>
<td style="text-align: left;">3.58e-07</td>
<td style="text-align: left;">7.49e-07</td>
</tr>
</tbody>
</table>
<p><strong>Table 4</strong>: Relative abundance of human-infecting viruses in each dataset, with and without human reads</p>
</div>
</div>
<p>When comparing the relative abundance of human-infecting viruses between sample types, we see that the plasma samples have a higher relative abundance of human-infecting viruses than the whole blood samples (samples with 0 reads are filtered out). Compare the mean relative abundance of human infecting viruses between plasma and whole blood and we see that plasma has a relative abundance 3 orders of magnitude higher. If we control for human reads, we see that the relationship remains the same.</p>
</section>
<section id="family-composition" class="level3" data-number="4.2.2">
<h3 data-number="4.2.2" class="anchored" data-anchor-id="family-composition"><span class="header-section-number">4.2.2</span> Family composition</h3>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="2024-09-24-blog3-preanalysis_files/figure-html/explore-hv-family-sample-type-1.svg?" class="img-fluid figure-img"></p>
<figcaption><strong>Figure 3</strong>: Family composition of human-viral reads</figcaption>
</figure>
</div>
</div>
</div>
<p>We can start off by looking at the family composition of human-infecting viruses in both sample types. We can see that Anelloviridae is the most abudant family in plasma, whereas Flavivirdae is the most abundant family in whole blood. To get a better understanding of this breakdown, we can look at the families in each study.</p>
<p>Comparing individual studies to each other, we see that there is a wide range of human-infecting virus depending on the study, even within the same sample type.</p>
</section>
<section id="species-composition" class="level3" data-number="4.2.3">
<h3 data-number="4.2.3" class="anchored" data-anchor-id="species-composition"><span class="header-section-number">4.2.3</span> Species composition</h3>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="2024-09-24-blog3-preanalysis_files/figure-html/hv-species-1.svg?" class="img-fluid figure-img"></p>
<figcaption><strong>Figure 4</strong>: Relative abundance of <em>selected</em> human-infecting viruses in each dataset</figcaption>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="2024-09-24-blog3-preanalysis_files/figure-html/hv-species-without-human-reads-1.svg?" class="img-fluid figure-img"></p>
<figcaption><strong>Figure 5</strong>: Relative abundance of <em>selected</em> human-infecting viruses in each dataset without human reads</figcaption>
</figure>
</div>
</div>
</div>
<p>To gain a more granular understanding of each sample type, we can look at the relative abundance of specific viruses in each sample type that may be of interest (totalling to 22): those that are linked to disease, characterized by latent infections (this is in accordance to the NAO’s interest to detecting “stealth” pandemics), and those that are blood-borne. We get a wide range of human infecting viruses including Hepatitis, HIV, Herpes, and HPV. Importantly, most viruses seem to be detected in both sample types. When comparing the coverage of all viruses in each sample type, we see that plasma contains 104 unique viruses, whereas whole blood only countains about 56. However, when looking at these closely, we see that the majority of viruses captured by plasma all come from Anellovirdae, a non-harmful family of viruses. When we only look at the viruses specified above, we find that plasma contains 18 out of the 22 viruses, whereas whole blood contains 17 out of the 22 viruses. Specifically, plasma uniquely contains HBV, whereas whole blood uniquely contains HTLV-1 and HTLV-2.</p>
</section>
</section>
</section>
<section id="discussion" class="level1" data-number="5">
<h1 data-number="5"><span class="header-section-number">5</span> Discussion</h1>
</section>
<section id="conclusion" class="level1" data-number="6">
<h1 data-number="6"><span class="header-section-number">6</span> Conclusion</h1>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>
