<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.53">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<meta name="author" content="Harmon Bhasin">
<meta name="dcterms.date" content="2024-07-23">
<title>Workflow of Mengyi et al.&nbsp;(2023) – Harmon's Public Notebook</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>

<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light"><script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><script async="" src="https://hypothes.is/embed.js"></script><script>
  window.document.addEventListener("DOMContentLoaded", function (_event) {
    document.body.classList.add('hypothesis-enabled');
  });
</script><style>

      .quarto-title-block .quarto-title-banner {
        background: #de2d26;
      }
</style>
<script src="../site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script><script src="../site_libs/d3-4.5.0/d3.min.js"></script><script src="../site_libs/sankey-1/sankey.js"></script><script src="../site_libs/sankeyNetwork-binding-0.4/sankeyNetwork.js"></script><link href="../site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet">
<script src="../site_libs/pagedtable-1.1/js/pagedtable.js"></script><script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script><script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script><script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script><link rel="stylesheet" href="../styles.css">
</head>
<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top quarto-banner"><nav class="navbar navbar-expand-lg " data-bs-theme="dark"><div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">Harmon’s Public Notebook</span>
    </a>
  </div>
        <div class="quarto-navbar-tools tools-end">
</div>
          <div id="quarto-search" class="" title="Search"></div>
      </div> <!-- /container-fluid -->
    </nav></header><!-- content --><header id="title-block-header" class="quarto-title-block default page-columns page-full"><div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-page-left">
      <div class="quarto-title-block"><div><h1 class="title">Workflow of Mengyi et al.&nbsp;(2023)</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
            <p class="subtitle lead">Pooled plasma from China (DNA)</p>
                      </div>
  </div>
    
  
  <div class="quarto-title-meta column-page-left">

      <div>
      <div class="quarto-title-meta-heading">Author</div>
      <div class="quarto-title-meta-contents">
               <p>Harmon Bhasin </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">July 23, 2024</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active"><h2 id="toc-title">Table of contents</h2>
   
  <ul>
<li>
<a href="#raw-data" id="toc-raw-data" class="nav-link active" data-scroll-target="#raw-data"><span class="header-section-number">1</span> Raw data</a>
  <ul class="collapse">
<li>
<a href="#about" id="toc-about" class="nav-link" data-scroll-target="#about"><span class="header-section-number">1.1</span> About</a>
  <ul class="collapse">
<li><a href="#sample-library-preparation" id="toc-sample-library-preparation" class="nav-link" data-scroll-target="#sample-library-preparation"><span class="header-section-number">1.1.1</span> Sample + library preparation</a></li>
  </ul>
</li>
  <li><a href="#quality-control-metrics" id="toc-quality-control-metrics" class="nav-link" data-scroll-target="#quality-control-metrics"><span class="header-section-number">1.2</span> Quality control metrics</a></li>
  </ul>
</li>
  <li>
<a href="#preprocessing" id="toc-preprocessing" class="nav-link" data-scroll-target="#preprocessing"><span class="header-section-number">2</span> Preprocessing</a>
  <ul class="collapse">
<li><a href="#summary" id="toc-summary" class="nav-link" data-scroll-target="#summary"><span class="header-section-number">2.1</span> Summary</a></li>
  <li><a href="#quality-control-metrics-1" id="toc-quality-control-metrics-1" class="nav-link" data-scroll-target="#quality-control-metrics-1"><span class="header-section-number">2.2</span> Quality control metrics</a></li>
  </ul>
</li>
  <li>
<a href="#taxonomic-composition" id="toc-taxonomic-composition" class="nav-link" data-scroll-target="#taxonomic-composition"><span class="header-section-number">3</span> Taxonomic composition</a>
  <ul class="collapse">
<li><a href="#high-level-composition" id="toc-high-level-composition" class="nav-link" data-scroll-target="#high-level-composition"><span class="header-section-number">3.1</span> High-level composition</a></li>
  <li><a href="#total-viral-content" id="toc-total-viral-content" class="nav-link" data-scroll-target="#total-viral-content"><span class="header-section-number">3.2</span> Total viral content</a></li>
  <li><a href="#taxonomic-composition-of-viruses" id="toc-taxonomic-composition-of-viruses" class="nav-link" data-scroll-target="#taxonomic-composition-of-viruses"><span class="header-section-number">3.3</span> Taxonomic composition of viruses</a></li>
  </ul>
</li>
  <li>
<a href="#human-infecting-virus-reads" id="toc-human-infecting-virus-reads" class="nav-link" data-scroll-target="#human-infecting-virus-reads"><span class="header-section-number">4</span> Human-infecting virus reads</a>
  <ul class="collapse">
<li><a href="#overall-relative-abundance" id="toc-overall-relative-abundance" class="nav-link" data-scroll-target="#overall-relative-abundance"><span class="header-section-number">4.1</span> Overall relative abundance</a></li>
  <li><a href="#overall-taxonomy-and-composition" id="toc-overall-taxonomy-and-composition" class="nav-link" data-scroll-target="#overall-taxonomy-and-composition"><span class="header-section-number">4.2</span> Overall taxonomy and composition</a></li>
  <li>
<a href="#analyzing-specific-families" id="toc-analyzing-specific-families" class="nav-link" data-scroll-target="#analyzing-specific-families"><span class="header-section-number">4.3</span> Analyzing specific families</a>
  <ul class="collapse">
<li><a href="#hepadnaviridae-number-of-reads-339427" id="toc-hepadnaviridae-number-of-reads-339427" class="nav-link" data-scroll-target="#hepadnaviridae-number-of-reads-339427"><span class="header-section-number">4.3.1</span> Hepadnaviridae (Number of reads: 339,427)</a></li>
  <li><a href="#parvoviridae-number-of-reads-17401" id="toc-parvoviridae-number-of-reads-17401" class="nav-link" data-scroll-target="#parvoviridae-number-of-reads-17401"><span class="header-section-number">4.3.2</span> Parvoviridae (Number of reads: 17,401)</a></li>
  <li><a href="#anelloviridae-number-of-reads-3001" id="toc-anelloviridae-number-of-reads-3001" class="nav-link" data-scroll-target="#anelloviridae-number-of-reads-3001"><span class="header-section-number">4.3.3</span> Anelloviridae (Number of reads: 3001)</a></li>
  <li><a href="#orthoherpesviridae-number-of-reads-298" id="toc-orthoherpesviridae-number-of-reads-298" class="nav-link" data-scroll-target="#orthoherpesviridae-number-of-reads-298"><span class="header-section-number">4.3.4</span> Orthoherpesviridae (Number of reads: 298)</a></li>
  <li><a href="#retroviridae-number-of-reads-115" id="toc-retroviridae-number-of-reads-115" class="nav-link" data-scroll-target="#retroviridae-number-of-reads-115"><span class="header-section-number">4.3.5</span> Retroviridae (Number of reads: 115)</a></li>
  <li><a href="#microviridae-number-of-reads-61" id="toc-microviridae-number-of-reads-61" class="nav-link" data-scroll-target="#microviridae-number-of-reads-61"><span class="header-section-number">4.3.6</span> Microviridae (Number of reads: 61)</a></li>
  <li><a href="#adenoviridae-number-of-reads-28" id="toc-adenoviridae-number-of-reads-28" class="nav-link" data-scroll-target="#adenoviridae-number-of-reads-28"><span class="header-section-number">4.3.7</span> Adenoviridae (Number of reads: 28)</a></li>
  </ul>
</li>
  <li><a href="#relative-abundance-of-pathogenic-viruses-of-interest" id="toc-relative-abundance-of-pathogenic-viruses-of-interest" class="nav-link" data-scroll-target="#relative-abundance-of-pathogenic-viruses-of-interest"><span class="header-section-number">4.4</span> Relative abundance of pathogenic viruses of interest</a></li>
  <li><a href="#relative-abundance-assuming-perfect-human-read-removal" id="toc-relative-abundance-assuming-perfect-human-read-removal" class="nav-link" data-scroll-target="#relative-abundance-assuming-perfect-human-read-removal"><span class="header-section-number">4.5</span> Relative abundance assuming perfect human read removal</a></li>
  </ul>
</li>
  <li><a href="#conclusion" id="toc-conclusion" class="nav-link" data-scroll-target="#conclusion"><span class="header-section-number">5</span> Conclusion</a></li>
  <li>
<a href="#appendix" id="toc-appendix" class="nav-link" data-scroll-target="#appendix"><span class="header-section-number">6</span> Appendix</a>
  <ul class="collapse">
<li><a href="#human-infecting-virus-families-genera-and-species" id="toc-human-infecting-virus-families-genera-and-species" class="nav-link" data-scroll-target="#human-infecting-virus-families-genera-and-species"><span class="header-section-number">6.1</span> Human-infecting virus families, genera, and species</a></li>
  </ul>
</li>
  </ul><div class="quarto-other-links"><h2>Other Links</h2><ul><li><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10372899/"><i class="bi bi-link-45deg"></i>Paper</a></li><li><a href="https://bigd.big.ac.cn/gsa/browse/CRA006191"><i class="bi bi-link-45deg"></i>Data</a></li></ul></div><div class="quarto-code-links"><h2>Code Links</h2><ul><li><a href="https://github.com/naobservatory/harmons-public-notebook/blob/main/notebooks/2024-07-23-mengyi.qmd"><i class="bi bi-file-code"></i>Code for this post</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content quarto-banner-title-block page-columns page-full column-page-left" id="quarto-document-content"><p>This one of the studies that we hope to discuss in <a href="https://naobservatory.org/blog/exploring-blood-biosurveillance-part1">our third blog post</a>, which will cover the metagenomic analysis of whole blood/plasma. In this notebook, I analyze <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10372899/">Mengyi 2023</a>, a dataset from China with 201 pools of <strong>plasma</strong>, where each pool contains 160 samples (we don’t know whether 1 sample = 1 donation) from 7 different locations, for a total of 10,720 samples.</p>
<p><em>I’d like to thank Lenni for giving me feedback on the notebook, Will for providing me with a boilerplate rmarkdown file, and Simon for giving me feedback on my figures. This notebook uses the MGS Workflow v2.2.1 (note that this is old).</em></p>
<section id="raw-data" class="level1" data-number="1"><h1 data-number="1">
<span class="header-section-number">1</span> Raw data</h1>
<section id="about" class="level2" data-number="1.1"><h2 data-number="1.1" class="anchored" data-anchor-id="about">
<span class="header-section-number">1.1</span> About</h2>
<p><a href="https://bigd.big.ac.cn/gsa/browse/CRA006191">This dataset</a> from China has 201 pools of plasma, where each pool contains 160 samples from 7 different locations between 2012-2018, for a total of 10,720 samples. This paper did not discuss the number of individuals that contributed to the samples, but we’re going to attempt to contact the authors to get this information (this shouldn’t hold up any further analysis, but would be good to have this information). They did DNA-sequencing for each pool, with Illumina HiSeq 4500, producing 2x150 bp reads.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb1"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Import libraries and extract metadata from sample names</span></span>
<span><span class="va">libraries</span> <span class="op">&lt;-</span> <span class="fu">read_csv</span><span class="op">(</span><span class="va">libraries_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">metadata</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sprintf.html">sprintf</a></span><span class="op">(</span><span class="st">'%s/metadata.tsv'</span>, <span class="va">data_dir</span><span class="op">)</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="st">'sample'</span><span class="op">=</span><span class="va">`Library`</span>, `library`<span class="op">=</span><span class="va">`Sample title`</span>, <span class="st">'Geographic location'</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span><span class="st">'location'</span> <span class="op">=</span> <span class="fu">str_split</span><span class="op">(</span><span class="va">`Geographic location`</span>, <span class="st">":"</span>, simplify <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span><span class="op">(</span><span class="fu">str_split</span><span class="op">(</span><span class="va">`Geographic location`</span>, <span class="st">":"</span>, simplify <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">libraries</span> <span class="op">&lt;-</span> <span class="va">libraries</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">metadata</span>, by <span class="op">=</span> <span class="st">'sample'</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, library<span class="op">=</span><span class="va">library.y</span>, <span class="va">location</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>library <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">library</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">libraries</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">location</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">summarize</span><span class="op">(</span>n_pools <span class="op">=</span> <span class="fu">n</span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["location"],"name":[1],"type":["chr"],"align":["left"]},{"label":["n_pools"],"name":[2],"type":["int"],"align":["right"]}],"data":[{"1":"Chongqing","2":"21"},{"1":"Dehong,Mangshi","2":"15"},{"1":"Guangxi,Liuzhou","2":"30"},{"1":"Heilongjiang,Mudanjiang","2":"39"},{"1":"Jiangsu,Nanjing","2":"36"},{"1":"Sichuan,Mianyang","2":"30"},{"1":"Xinjiang,Urumqi","2":"30"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<section id="sample-library-preparation" class="level3" data-number="1.1.1"><h3 data-number="1.1.1" class="anchored" data-anchor-id="sample-library-preparation">
<span class="header-section-number">1.1.1</span> Sample + library preparation</h3>
<p>The following excerpt from the paper describes the sample and library preparation process. It’s crucial to note that while blood samples were initially collected from volunteers, these samples were converted to plasma through ultracentrifugation prior to sequencing:</p>
<blockquote class="blockquote">
<p>From January 1, 2012, to December 31, 2018, a total of 10,720 blood samples of 10 ml each were randomly selected from voluntary blood donors in 7 regions. The blood samples taken from various places were mixed in units of 160 (each 100 μl) for ultracentrifugation (32,000 rpm, 120 min, maximum centrifugal radius of 91.9 mm). Afterward, we rinsed and resuspended the precipitate with 500 μl PBS.</p>
<p>The pooled suspensions were subjected to extraction of total DNA using QIAamp® DNA Blood mini Kit (QIAGEN Cat. NO.160019269, Frankfurt, Germany), DNA concentration was measured by Equalbit® 1 × dsDNA HS Assay Kit (Vazyme Cat. NO. 7E302K9, Nanjing, China).</p>
<p>The metagenomic library was constructed using KAPA HyperPlus Kit (KAPA Cat. NO. 0000097583, Boston, USA) with dual-indexed Adapters (KAPA Cat. NO. 0000093370, Boston, USA), the DNA was fragmented to 250 bp approximately by the enzyme at 37 °C for 20 min, after end repair and A-tailing, adapter ligation, post-ligation cleanup, library amplification, and post-amplification cleanup, the library was constructed.</p>
<p>Agilent 2100 Bioanalyzer (Agilent Technologies, Beijing, China) was used for library quality control, and qualified DNA library was sent to the Novogene company to sequence in HiSeq 4500.</p>
</blockquote>
</section></section><section id="quality-control-metrics" class="level2" data-number="1.2"><h2 data-number="1.2" class="anchored" data-anchor-id="quality-control-metrics">
<span class="header-section-number">1.2</span> Quality control metrics</h2>
<p>These 201 samples contained 3.4B read pairs. The samples had 8.2M - 30.1M (mean 17.1M) read pairs each. The number of read pairs and total bases look relatively evenly distributed across the locations. The duplication rate is also quite low, around ~10%. Adapter content is low. As we’d expect, we see higher quality Q scores near the beginning of the read and a gradual decline towards the end of the read, however all positions seem to have a Q score of 35 which means that our reads are ~99.97% accurate. When looking at the Q score over all sequences, we can see a sharp peak after 35, which closely follows the previous plot, indicating high read quality.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb2"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Prepare data</span></span>
<span><span class="va">basic_stats_raw_metrics</span> <span class="op">&lt;-</span> <span class="va">basic_stats_raw</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">library</span>,</span>
<span>         <span class="va">location</span>,</span>
<span>         `# Read pairs` <span class="op">=</span> <span class="va">n_read_pairs</span>,</span>
<span>         `Total base pairs\n(approx)` <span class="op">=</span> <span class="va">n_bases_approx</span>,</span>
<span>         `% Duplicates\n(FASTQC)` <span class="op">=</span> <span class="va">percent_duplicates</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_longer</span><span class="op">(</span><span class="op">-</span><span class="op">(</span><span class="va">library</span><span class="op">:</span><span class="va">location</span><span class="op">)</span>, names_to <span class="op">=</span> <span class="st">"metric"</span>, values_to <span class="op">=</span> <span class="st">"value"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>metric <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">metric</span><span class="op">)</span><span class="op">)</span> </span>
<span></span>
<span><span class="co"># Set up plot templates</span></span>
<span></span>
<span><span class="va">g_basic</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">basic_stats_raw_metrics</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">library</span>, y <span class="op">=</span> <span class="va">value</span>, fill<span class="op">=</span><span class="va">location</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>position <span class="op">=</span> <span class="st">"dodge"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_discrete</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Dark2"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">expand_limits</span><span class="op">(</span>y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">100</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">metric</span> <span class="op">~</span> <span class="va">.</span>, scales <span class="op">=</span> <span class="st">"free"</span>, space <span class="op">=</span> <span class="st">"free_x"</span>, switch <span class="op">=</span> <span class="st">"y"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span></span>
<span>    axis.title.y <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    strip.text.y <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>face <span class="op">=</span> <span class="st">"plain"</span><span class="op">)</span>,</span>
<span>    axis.text.x <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span><span class="va">g_basic</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-23-mengyi_files/figure-html/plot-basic-stats-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb3"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Set up plotting templates</span></span>
<span><span class="va">g_qual_raw</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span>mapping<span class="op">=</span><span class="fu">aes</span><span class="op">(</span>color <span class="op">=</span> <span class="va">location</span>, linetype<span class="op">=</span><span class="va">read_pair</span>, group<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/interaction.html">interaction</a></span><span class="op">(</span><span class="va">sample</span>,<span class="va">read_pair</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">scale_linetype_discrete</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Read Pair"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Dark2"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>color<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">2</span>,byrow<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>         linetype <span class="op">=</span> <span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">2</span>,byrow<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span></span>
<span></span>
<span><span class="co"># Visualize adapters</span></span>
<span><span class="va">g_adapters_raw</span> <span class="op">&lt;-</span> <span class="va">g_qual_raw</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">position</span>, y<span class="op">=</span><span class="va">pc_adapters</span><span class="op">)</span>, data<span class="op">=</span><span class="va">adapter_stats_raw</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% Adapters"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">5</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Position"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span>,</span>
<span>                     breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">500</span>,<span class="fl">20</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">.</span><span class="op">~</span><span class="va">adapter</span><span class="op">)</span></span>
<span><span class="va">g_adapters_raw</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-23-mengyi_files/figure-html/plot-raw-quality-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb4"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Visualize quality</span></span>
<span><span class="va">g_quality_base_raw</span> <span class="op">&lt;-</span> <span class="va">g_qual_raw</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept<span class="op">=</span><span class="fl">25</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept<span class="op">=</span><span class="fl">30</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">position</span>, y<span class="op">=</span><span class="va">mean_phred_score</span><span class="op">)</span>, data<span class="op">=</span><span class="va">quality_base_stats_raw</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Mean Phred score"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">10</span>,<span class="fl">45</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Position"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span>,</span>
<span>                     breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">500</span>,<span class="fl">20</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_quality_base_raw</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-23-mengyi_files/figure-html/plot-raw-quality-2.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb5"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_quality_seq_raw</span> <span class="op">&lt;-</span> <span class="va">g_qual_raw</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">25</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">30</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">mean_phred_score</span>, y<span class="op">=</span><span class="va">n_sequences</span><span class="op">)</span>, data<span class="op">=</span><span class="va">quality_seq_stats_raw</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Mean Phred score"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"# Sequences"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_quality_seq_raw</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-23-mengyi_files/figure-html/plot-raw-quality-3.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section></section><section id="preprocessing" class="level1 page-columns page-full" data-number="2"><h1 data-number="2">
<span class="header-section-number">2</span> Preprocessing</h1>
<section id="summary" class="level2" data-number="2.1"><h2 data-number="2.1" class="anchored" data-anchor-id="summary">
<span class="header-section-number">2.1</span> Summary</h2>
<p>The average fraction of reads at each stage in the preprocessing pipeline is shown in the following table. Reads lost during trimming and filtering approximately matches what we’d expect based on the raw adapter content. Deduplication loses us about 10% of reads which matched the amount of estimated duplicated reads by QC. Low ribodepletion is observed which makes sense because they only sequenced DNA.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb6"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># TODO: Group by pool size as well</span></span>
<span><span class="co"># Count read losses</span></span>
<span><span class="va">n_reads_rel</span> <span class="op">&lt;-</span> <span class="va">basic_stats</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>,<span class="va">location</span>, <span class="va">stage</span>, <span class="va">percent_duplicates</span>, <span class="va">n_read_pairs</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">sample</span>, <span class="va">stage</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_retained <span class="op">=</span> <span class="va">n_read_pairs</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/stats/lag.html">lag</a></span><span class="op">(</span><span class="va">n_read_pairs</span><span class="op">)</span>,</span>
<span>         p_reads_lost <span class="op">=</span> <span class="fl">1</span> <span class="op">-</span> <span class="va">p_reads_retained</span>,</span>
<span>         p_reads_retained_abs <span class="op">=</span> <span class="va">n_read_pairs</span> <span class="op">/</span> <span class="va">n_read_pairs</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,</span>
<span>         p_reads_lost_abs <span class="op">=</span> <span class="fl">1</span><span class="op">-</span><span class="va">p_reads_retained_abs</span>,</span>
<span>         p_reads_lost_abs_marginal <span class="op">=</span> <span class="va">p_reads_lost_abs</span> <span class="op">-</span> <span class="fu"><a href="https://rdrr.io/r/stats/lag.html">lag</a></span><span class="op">(</span><span class="va">p_reads_lost_abs</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">n_reads_rel_display</span> <span class="op">&lt;-</span> <span class="va">n_reads_rel</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">rename</span><span class="op">(</span>Stage<span class="op">=</span><span class="va">stage</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">Stage</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>`% Total Reads Lost (Cumulative)` <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">p_reads_lost_abs</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span>,<span class="fl">1</span><span class="op">)</span>, <span class="st">"-"</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_lost_abs</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span>,<span class="fl">1</span><span class="op">)</span>, <span class="st">" (mean "</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">p_reads_lost_abs</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span>,<span class="fl">1</span><span class="op">)</span>, <span class="st">")"</span><span class="op">)</span>,</span>
<span>            `% Total Reads Lost (Marginal)` <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">p_reads_lost_abs_marginal</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span>,<span class="fl">1</span><span class="op">)</span>, <span class="st">"-"</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_lost_abs_marginal</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span>,<span class="fl">1</span><span class="op">)</span>, <span class="st">" (mean "</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">p_reads_lost_abs_marginal</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span>,<span class="fl">1</span><span class="op">)</span>, <span class="st">")"</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">Stage</span> <span class="op">!=</span> <span class="st">"raw_concat"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>Stage <span class="op">=</span> <span class="va">Stage</span> <span class="op">%&gt;%</span> <span class="va">as.numeric</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span>labels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Trimming &amp; filtering"</span>, <span class="st">"Deduplication"</span>, <span class="st">"Initial ribodepletion"</span>, <span class="st">"Secondary ribodepletion"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">n_reads_rel_display</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["Stage"],"name":[1],"type":["fct"],"align":["left"]},{"label":["% Total Reads Lost (Cumulative)"],"name":[2],"type":["chr"],"align":["left"]},{"label":["% Total Reads Lost (Marginal)"],"name":[3],"type":["chr"],"align":["left"]}],"data":[{"1":"Trimming & filtering","2":"0.9-3.4 (mean 1.8)","3":"0.9-3.4 (mean 1.8)"},{"1":"Deduplication","2":"10.9-24.5 (mean 15.2)","3":"9.3-23.6 (mean 13.5)"},{"1":"Initial ribodepletion","2":"11-24.5 (mean 15.3)","3":"0-0.1 (mean 0.1)"},{"1":"Secondary ribodepletion","2":"11-24.5 (mean 15.3)","3":"0-0.1 (mean 0)"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
</section><section id="quality-control-metrics-1" class="level2 page-columns page-full" data-number="2.2"><h2 data-number="2.2" class="anchored" data-anchor-id="quality-control-metrics-1">
<span class="header-section-number">2.2</span> Quality control metrics</h2>
<p>Trimming and cleaning gets rid of the Illumina adapter as well as polyg, and we see a decrease in polya. Q score remain the same during read cleaning when looking at the positions, with the end of the read actually improving in score. Q scores across all sequences look pretty much the same throughout cleaning.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb7"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_qual</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span>mapping<span class="op">=</span><span class="fu">aes</span><span class="op">(</span>color <span class="op">=</span> <span class="va">location</span>, linetype<span class="op">=</span><span class="va">read_pair</span>, group<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/interaction.html">interaction</a></span><span class="op">(</span><span class="va">sample</span>,<span class="va">read_pair</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">scale_linetype_discrete</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Read Pair"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Dark2"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>color<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">2</span>,byrow<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>         linetype <span class="op">=</span> <span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">2</span>,byrow<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span></span>
<span></span>
<span><span class="co"># Visualize adapters</span></span>
<span><span class="va">g_adapters</span> <span class="op">&lt;-</span> <span class="va">g_qual</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">position</span>, y<span class="op">=</span><span class="va">pc_adapters</span><span class="op">)</span>, data<span class="op">=</span><span class="va">adapter_stats</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% Adapters"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">5</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Position"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span>,</span>
<span>                     breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">140</span>,<span class="fl">20</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">stage</span><span class="op">~</span><span class="va">adapter</span><span class="op">)</span></span>
<span><span class="va">g_adapters</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-23-mengyi_files/figure-html/plot-quality-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb8"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Visualize quality</span></span>
<span><span class="va">g_quality_base</span> <span class="op">&lt;-</span> <span class="va">g_qual</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept<span class="op">=</span><span class="fl">25</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept<span class="op">=</span><span class="fl">30</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">position</span>, y<span class="op">=</span><span class="va">mean_phred_score</span><span class="op">)</span>, data<span class="op">=</span><span class="va">quality_base_stats</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Mean Phred score"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">10</span>,<span class="fl">45</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Position"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span>,</span>
<span>                     breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">140</span>,<span class="fl">20</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">stage</span><span class="op">~</span><span class="va">.</span><span class="op">)</span></span>
<span><span class="va">g_quality_base</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-23-mengyi_files/figure-html/plot-quality-2.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb9"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_quality_seq</span> <span class="op">&lt;-</span> <span class="va">g_qual</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">25</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">30</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">mean_phred_score</span>, y<span class="op">=</span><span class="va">n_sequences</span><span class="op">)</span>, data<span class="op">=</span><span class="va">quality_seq_stats</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Mean Phred score"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"# Sequences"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">stage</span><span class="op">~</span><span class="va">.</span>, scales <span class="op">=</span> <span class="st">"free_y"</span><span class="op">)</span></span>
<span><span class="va">g_quality_seq</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-23-mengyi_files/figure-html/plot-quality-3.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<p>Number of read pairs look reasonable, same with a large portion being lost during deduplication. At the end of deduplication it seems that only 5% of reads are duplicates which doesn’t seem too bad. Mean read length stays around 150 bp which is pretty good.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb10"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_stage_trace</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">basic_stats</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">stage</span>, group<span class="op">=</span><span class="va">sample</span>, color <span class="op">=</span> <span class="va">location</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Dark2"</span><span class="op">)</span></span>
<span><span class="co"># Plot reads over preprocessing</span></span>
<span><span class="va">g_reads_stages</span> <span class="op">&lt;-</span> <span class="va">g_stage_trace</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">n_read_pairs</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span><span class="st">"# Read pairs"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_reads_stages</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-23-mengyi_files/figure-html/preproc-figures-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb11"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Plot relative read losses during preprocessing</span></span>
<span><span class="va">g_reads_rel</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">n_reads_rel</span>, </span>
<span>                      <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">stage</span>, group<span class="op">=</span><span class="va">sample</span>, color <span class="op">=</span> <span class="va">location</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">p_reads_lost_abs_marginal</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span><span class="st">"% Total Reads Lost"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, </span>
<span>                     labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">x</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Dark2"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span></span>
<span><span class="va">g_reads_rel</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-23-mengyi_files/figure-html/preproc-figures-2.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb12"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">stage_dup</span> <span class="op">&lt;-</span> <span class="va">basic_stats</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">stage</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>dmin <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">percent_duplicates</span><span class="op">)</span>, dmax<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">percent_duplicates</span><span class="op">)</span>,</span>
<span>            dmean<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">percent_duplicates</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">g_dup_stages</span> <span class="op">&lt;-</span> <span class="va">g_stage_trace</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">percent_duplicates</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span><span class="st">"% Duplicates"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_dup_stages</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-23-mengyi_files/figure-html/preproc-dedup-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<div class="cell page-columns page-full" data-cap-location="margin">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb13" data-cap-location="margin"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_readlen_stages</span> <span class="op">&lt;-</span> <span class="va">g_stage_trace</span> <span class="op">+</span> <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">mean_seq_len</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span><span class="st">"Mean read length (nt)"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_readlen_stages</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display page-columns page-full">
<div class="quarto-figure quarto-figure-center page-columns page-full">
<figure class="figure page-columns page-full"><p><img src="2024-07-23-mengyi_files/figure-html/preproc-mean-readlen-1.svg" class="img-fluid figure-img"></p>
<figcaption class="margin-caption">Reminder that this data is 2x150 bp.</figcaption></figure>
</div>
</div>
</div>
</section></section><section id="taxonomic-composition" class="level1" data-number="3"><h1 data-number="3">
<span class="header-section-number">3</span> Taxonomic composition</h1>
<section id="high-level-composition" class="level2" data-number="3.1"><h2 data-number="3.1" class="anchored" data-anchor-id="high-level-composition">
<span class="header-section-number">3.1</span> High-level composition</h2>
<p>To assess the high-level composition of the reads, I ran the files through Kraken2 and summarized the results with Bracken.</p>
<p>The groups listed below were created by Will:</p>
<ul>
<li>Filtered (removed during cleaning)</li>
<li>Duplicate (removed during deduplication)</li>
<li>Ribosomal (removed during ribodepletion)</li>
<li>Unassigned (non-ribosomal reads that were not assigned to any taxon by Kraken/Bracken)</li>
<li>Bacterial (non-ribosomal reads assigned to the Bacteria domain by Kraken/Bracken)</li>
<li>Archaeal (non-ribosomal reads assigned to the Archaea domain by Kraken/Bracken)</li>
<li>Viral (non-ribosomal reads assigned to the Viruses domain by Kraken/Bracken)</li>
<li>Human (non-ribosomal reads assigned to the Eukarya domain by Kraken/Bracken)</li>
</ul>
<p>Human reads make up a large proportion of all reads at 82% total composition, whereas viruses only account for 0.01% of all reads.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb14"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">classifications</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Filtered"</span>, <span class="st">"Duplicate"</span>, <span class="st">"Ribosomal"</span>, <span class="st">"Unassigned"</span>,</span>
<span>                     <span class="st">"Bacterial"</span>, <span class="st">"Archaeal"</span>, <span class="st">"Viral"</span>, <span class="st">"Human"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Import composition data</span></span>
<span><span class="va">tax_final_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">results_dir</span>, <span class="st">"taxonomy_final"</span><span class="op">)</span></span>
<span><span class="va">comp</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sprintf.html">sprintf</a></span><span class="op">(</span><span class="st">"%s/taxonomic_composition.tsv.gz"</span>, <span class="va">tax_final_dir</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by<span class="op">=</span><span class="st">'sample'</span><span class="op">)</span></span>
<span></span>
<span><span class="va">comp_minor</span> <span class="op">&lt;-</span> <span class="va">comp</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">classification</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Archaeal"</span>, <span class="st">"Viral"</span>, <span class="st">"Human"</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">comp_assigned</span> <span class="op">&lt;-</span> <span class="va">comp</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span> <span class="va">classification</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Filtered"</span>, <span class="st">"Duplicate"</span>, </span>
<span>                                 <span class="st">"Ribosomal"</span>, <span class="st">"Unassigned"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">n_reads</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">comp_assigned_minor</span> <span class="op">&lt;-</span> <span class="va">comp_assigned</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">classification</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Archaeal"</span>, <span class="st">"Viral"</span>, <span class="st">"Human"</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Summarize composition</span></span>
<span><span class="va">read_comp_summ</span> <span class="op">&lt;-</span> <span class="va">comp</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">classification</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop_last"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>n_reads <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">n_reads</span>,<span class="fl">0</span><span class="op">)</span>,</span>
<span>         p_reads <span class="op">=</span> <span class="va">n_reads</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads</span><span class="op">)</span>,</span>
<span>         pc_reads <span class="op">=</span> <span class="va">p_reads</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>classification <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">classification</span>, levels<span class="op">=</span><span class="va">classifications</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">classification</span>, <span class="va">n_reads</span>, <span class="va">pc_reads</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>`# of reads` <span class="op">=</span> <span class="va">n_reads</span>, <span class="st">"% of reads"</span> <span class="op">=</span> <span class="va">pc_reads</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>`% of reads` <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sprintf.html">sprintf</a></span><span class="op">(</span><span class="st">"%.2f"</span>, <span class="va">`% of reads`</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">read_comp_summ</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["classification"],"name":[1],"type":["fct"],"align":["left"]},{"label":["# of reads"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["% of reads"],"name":[3],"type":["chr"],"align":["left"]}],"data":[{"1":"Archaeal","2":"27877","3":"0.00"},{"1":"Bacterial","2":"7731190","3":"0.22"},{"1":"Duplicate","2":"468108063","3":"13.62"},{"1":"Filtered","2":"59464986","3":"1.73"},{"1":"Human","2":"2858732231","3":"83.19"},{"1":"Ribosomal","2":"3032786","3":"0.09"},{"1":"Unassigned","2":"38835019","3":"1.13"},{"1":"Viral","2":"378247","3":"0.01"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb15"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Prepare plotting templates</span></span>
<span><span class="va">g_comp_base</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span>mapping<span class="op">=</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">library</span>, y<span class="op">=</span><span class="va">p_reads</span>, fill<span class="op">=</span><span class="va">classification</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_discrete</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Plasma pool"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span> <span class="op">+</span> </span>
<span>  <span class="fu">theme</span><span class="op">(</span>plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>hjust<span class="op">=</span><span class="fl">0</span>, face<span class="op">=</span><span class="st">"plain"</span>, size<span class="op">=</span><span class="fu">rel</span><span class="op">(</span><span class="fl">1.5</span><span class="op">)</span><span class="op">)</span>,</span>
<span>        axis.text.x <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">scale_y_pc_reads</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/partial.html">partial</a></span><span class="op">(</span><span class="va">scale_y_continuous</span>, name <span class="op">=</span> <span class="st">"% Reads"</span>,</span>
<span>                                   expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span></span>
<span><span class="va">geom_comp</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/partial.html">partial</a></span><span class="op">(</span><span class="va">geom_col</span>, position <span class="op">=</span> <span class="st">"stack"</span>, width <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Define a color palette for the classification</span></span>
<span><span class="va">classification_colors</span> <span class="op">&lt;-</span> <span class="fu">brewer.pal</span><span class="op">(</span><span class="fl">8</span>, <span class="st">"Accent"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">classification_colors</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Filtered"</span>, <span class="st">"Duplicate"</span>, <span class="st">"Ribosomal"</span>, <span class="st">"Unassigned"</span>, <span class="st">"Bacterial"</span>, <span class="st">"Archaeal"</span>, <span class="st">"Viral"</span>, <span class="st">"Human"</span><span class="op">)</span></span>
<span><span class="va">scale_fill_classification</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values <span class="op">=</span> <span class="va">classification_colors</span>, name <span class="op">=</span> <span class="st">"Classification"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Plot overall composition</span></span>
<span><span class="va">g_comp</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> <span class="fu">geom_comp</span><span class="op">(</span>data <span class="op">=</span> <span class="va">comp</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_pc_reads</span><span class="op">(</span>limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_classification</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Read composition (all reads, all groups)"</span><span class="op">)</span></span>
<span><span class="va">g_comp</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-23-mengyi_files/figure-html/plot-composition-all-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb16"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Create a faceted plot of composition by location</span></span>
<span><span class="co">#g_comp_by_location &lt;- g_comp_base +</span></span>
<span><span class="co">#  geom_comp(data = comp) +</span></span>
<span><span class="co">#  scale_y_pc_reads(limits = c(0,1.01), breaks = seq(0,1,0.2)) +</span></span>
<span><span class="co">#  scale_fill_classification() +</span></span>
<span><span class="co">#  facet_wrap(~ location, scales = "free_x", ncol = 2) +</span></span>
<span><span class="co">#  theme(axis.text.x = element_blank(),  # Remove x-axis labels</span></span>
<span><span class="co">#        axis.ticks.x = element_blank()) +  # Remove x-axis ticks</span></span>
<span><span class="co">#  ggtitle("Read composition (all reads, all groups)")</span></span>
<span></span>
<span><span class="co"># Display the plot</span></span>
<span><span class="co">#g_comp_by_location</span></span>
<span></span>
<span><span class="co"># Repeat for classified reads only</span></span>
<span><span class="va">g_comp_assigned</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_comp</span><span class="op">(</span>data <span class="op">=</span> <span class="va">comp_assigned</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_pc_reads</span><span class="op">(</span>limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_classification</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Read composition (assigned reads, all groups)"</span><span class="op">)</span></span>
<span><span class="va">g_comp_assigned</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-23-mengyi_files/figure-html/plot-composition-all-2.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb17"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Plot composition of minor components</span></span>
<span><span class="co">#g_comp_minor &lt;- g_comp_base + </span></span>
<span><span class="co">#  geom_comp(data=comp_minor) +</span></span>
<span><span class="co">#  scale_y_pc_reads() +</span></span>
<span><span class="co">#  scale_fill_classification() + </span></span>
<span><span class="co">#  ggtitle("Read composition (all reads, minor groups)")</span></span>
<span><span class="co">#g_comp_minor</span></span>
<span><span class="co">#g_comp_assigned_minor &lt;- g_comp_base + </span></span>
<span><span class="co">#  geom_comp(data=comp_assigned_minor) +</span></span>
<span><span class="co">#  scale_y_pc_reads() +</span></span>
<span><span class="co">#  scale_fill_classification() + </span></span>
<span><span class="co">#  ggtitle("Read composition (assigned reads, minor groups)")</span></span>
<span><span class="co">#g_comp_assigned_minor</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section><section id="total-viral-content" class="level2" data-number="3.2"><h2 data-number="3.2" class="anchored" data-anchor-id="total-viral-content">
<span class="header-section-number">3.2</span> Total viral content</h2>
<p>Total viral fraction average <span class="math inline">\(1.03 \times 10^{-4}\)</span> across samples. As a fraction of assigned (rather than total) reads, this jumped to <span class="math inline">\(1.28 \times 10^{-4}\)</span>. The increase in viral fraction from assigned reads to total reads was constant across all samples, however, we can see that the viral fractions differed by location. Note that one of the pools from “Heilongjiang,Mudanjiang” had no viral reads.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb18"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co">#p_reads_viral_all &lt;- comp %&gt;% filter(classification == "Viral") %&gt;%</span></span>
<span><span class="co">#  mutate(read_group = "All reads")</span></span>
<span><span class="va">p_reads_viral_assigned</span> <span class="op">&lt;-</span> <span class="va">comp_assigned</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">classification</span> <span class="op">==</span> <span class="st">"Viral"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>read_group <span class="op">=</span> <span class="st">"Classified reads"</span><span class="op">)</span></span>
<span><span class="co">#p_reads_viral &lt;- bind_rows(p_reads_viral_all, p_reads_viral_assigned)</span></span>
<span></span>
<span><span class="va">location_viral</span> <span class="op">&lt;-</span> <span class="va">comp_assigned</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">classification</span> <span class="op">==</span> <span class="st">"Viral"</span> <span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">location</span><span class="op">)</span> </span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="co"># g_viral &lt;- ggplot(p_reads_viral, aes(x=p_reads, y=library, color=location, shape=read_group)) +</span></span>
<span><span class="co">#   geom_point() +</span></span>
<span><span class="co">#   scale_y_discrete(name="Plasma pool") +</span></span>
<span><span class="co">#   scale_x_log10(name="Viral read fraction") +</span></span>
<span><span class="co">#   scale_color_brewer(palette = "Dark2") +</span></span>
<span><span class="co">#   #facet_grid(.~read_group, scales = "free") +</span></span>
<span><span class="co">#   guides(color=guide_legend(nrow=2), shape=guide_legend(nrow=2),</span></span>
<span><span class="co">#          linetype=guide_legend(nrow=2)) +</span></span>
<span><span class="co">#   theme_kit</span></span>
<span></span>
<span><span class="co">#ggplot(location_viral, aes(x=p_reads, y=location, color = location)) +</span></span>
<span><span class="co">#  geom_quasirandom(size=2) +</span></span>
<span><span class="co">#  scale_x_log10(name="Viral read fraction", labels = label_log(digits=2)) +</span></span>
<span><span class="co">#  scale_y_discrete(name="Location") +</span></span>
<span><span class="co">#  scale_color_brewer(palette = "Dark2") +</span></span>
<span><span class="co">#  theme_light() + </span></span>
<span><span class="co">#    theme(</span></span>
<span><span class="co">#    axis.text.y = element_text(size = 8),</span></span>
<span><span class="co">#    axis.text.x = element_text(size = 14),</span></span>
<span><span class="co">#    axis.ticks.y = element_blank(),</span></span>
<span><span class="co">#    axis.line = element_line(colour = "black"),</span></span>
<span><span class="co">#    axis.title.x = element_text(size = 15),    </span></span>
<span><span class="co">#    axis.title.y = element_text(size = 15),  </span></span>
<span><span class="co">#    legend.text = element_text(size = 13),</span></span>
<span><span class="co">#    legend.title = element_text(size = 16),</span></span>
<span><span class="co">#    legend.position = c(1, 1),  # Move legend to top right</span></span>
<span><span class="co">##    legend.justification = c(1, 1),  # Align legend to top right</span></span>
<span><span class="co">#    panel.grid.minor = element_blank(),</span></span>
<span><span class="co">#    panel.border = element_blank(),</span></span>
<span><span class="co">#    panel.background = element_blank())</span></span>
<span></span>
<span><span class="co"># Calculate the number of samples per location</span></span>
<span><span class="va">location_counts</span> <span class="op">&lt;-</span> <span class="va">location_viral</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">location</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarise</span><span class="op">(</span>count <span class="op">=</span> <span class="fu">n</span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">location</span>, <span class="st">" (n="</span>, <span class="va">count</span>, <span class="st">")"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create a named vector for easy mapping</span></span>
<span><span class="va">location_labels</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/setNames.html">setNames</a></span><span class="op">(</span><span class="va">location_counts</span><span class="op">$</span><span class="va">label</span>, <span class="va">location_counts</span><span class="op">$</span><span class="va">location</span><span class="op">)</span></span>
<span></span>
<span><span class="va">g_viral</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">location_viral</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">location</span>, y<span class="op">=</span><span class="va">p_reads</span>, fill<span class="op">=</span><span class="va">location</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_violin</span><span class="op">(</span>trim<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_boxplot</span><span class="op">(</span>width<span class="op">=</span><span class="fl">0.1</span>, fill<span class="op">=</span><span class="st">"white"</span>, color<span class="op">=</span><span class="st">"black"</span>, outlier.shape<span class="op">=</span><span class="cn">NA</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_jitter</span><span class="op">(</span>width<span class="op">=</span><span class="fl">0.1</span>, size<span class="op">=</span><span class="fl">0.5</span>, alpha<span class="op">=</span><span class="fl">0.5</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_log10</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Viral read fraction"</span>, labels <span class="op">=</span> <span class="fu">label_log</span><span class="op">(</span>digits<span class="op">=</span><span class="fl">2</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_discrete</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Location"</span>, labels <span class="op">=</span> <span class="va">location_labels</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Dark2"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title<span class="op">=</span><span class="st">"Total viral fraction based on assigned reads over all locations"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme_light</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">theme</span><span class="op">(</span></span>
<span>    axis.text.y <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">12</span>, angle <span class="op">=</span> <span class="fl">45</span>, hjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>,</span>
<span>    axis.title.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">15</span><span class="op">)</span>,    </span>
<span>    axis.title.y <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">15</span><span class="op">)</span>,  </span>
<span>    legend.position <span class="op">=</span> <span class="st">"none"</span>,</span>
<span>    panel.grid.minor <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    panel.border <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    panel.background <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span> <span class="op">+</span></span>
<span>  <span class="fu">coord_flip</span><span class="op">(</span><span class="op">)</span>  <span class="co"># This flips the coordinates to make the plot horizontal</span></span>
<span></span>
<span><span class="va">g_viral</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-23-mengyi_files/figure-html/p-viral-scatter-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section><section id="taxonomic-composition-of-viruses" class="level2" data-number="3.3"><h2 data-number="3.3" class="anchored" data-anchor-id="taxonomic-composition-of-viruses">
<span class="header-section-number">3.3</span> Taxonomic composition of viruses</h2>
<p>The threshold for the label “other” are the set of families that make up less than 20% composition in all samples (the only reason I did this was because there were too many matches). We can see that other makes up a big portion of the viral composition. Outside of the “other” label, Hepadnaviridae makes up the largest portion of the viral composition.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb19"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">major_threshold</span> <span class="op">&lt;-</span> <span class="fl">0.20</span></span>
<span></span>
<span><span class="co"># Identify major viral families</span></span>
<span><span class="va">viral_families_major_tab</span> <span class="op">&lt;-</span> <span class="va">viral_families</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>p_reads_viral_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_viral</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_viral_max</span> <span class="op">&gt;=</span> <span class="va">major_threshold</span><span class="op">)</span></span>
<span><span class="va">viral_families_major_list</span> <span class="op">&lt;-</span> <span class="va">viral_families_major_tab</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">name</span><span class="op">)</span></span>
<span><span class="va">viral_families_major</span> <span class="op">&lt;-</span> <span class="va">viral_families</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">viral_families_major_list</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span>, <span class="va">sample</span>, <span class="va">p_reads_viral</span><span class="op">)</span></span>
<span><span class="va">viral_families_minor</span> <span class="op">&lt;-</span> <span class="va">viral_families_major</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>p_reads_viral_major <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads_viral</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Other"</span>, taxid<span class="op">=</span><span class="cn">NA</span>, p_reads_viral <span class="op">=</span> <span class="fl">1</span><span class="op">-</span><span class="va">p_reads_viral_major</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span>, <span class="va">sample</span>, <span class="va">p_reads_viral</span><span class="op">)</span></span>
<span><span class="va">viral_families_display</span> <span class="op">&lt;-</span> <span class="va">viral_families_major</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">bind_rows</span><span class="op">(</span><span class="va">viral_families_minor</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p_reads_viral</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">name</span>, levels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">viral_families_major_list</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">p_reads_viral</span>, classification<span class="op">=</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by<span class="op">=</span><span class="st">'sample'</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create a custom color palette with up to 20 colors</span></span>
<span><span class="va">custom_palette</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span></span>
<span>  <span class="fu">brewer.pal</span><span class="op">(</span><span class="fl">12</span>, <span class="st">"Paired"</span><span class="op">)</span>,</span>
<span>  <span class="fu">brewer.pal</span><span class="op">(</span><span class="fl">12</span>, <span class="st">"Set3"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">n_classifications</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">viral_families_display</span><span class="op">$</span><span class="va">classification</span><span class="op">)</span><span class="op">)</span> <span class="op">-</span> <span class="fl">1</span></span>
<span></span>
<span><span class="va">custom_palette_with_black</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span></span>
<span>  <span class="va">custom_palette</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="va">n_classifications</span><span class="op">]</span>,</span>
<span>  <span class="st">"black"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create a new color palette with black for "Other"</span></span>
<span><span class="va">custom_palette_with_black</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span></span>
<span>  <span class="va">custom_palette</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="va">n_classifications</span><span class="op">]</span>,</span>
<span>  <span class="st">"black"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_families</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_comp</span><span class="op">(</span>data<span class="op">=</span><span class="va">viral_families_display</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% Viral Reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, </span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values <span class="op">=</span> <span class="va">custom_palette_with_black</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span> <span class="va">location</span>, scales <span class="op">=</span> <span class="st">"free_x"</span>, ncol <span class="op">=</span> <span class="fl">2</span><span class="op">)</span> </span>
<span><span class="va">g_families</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-23-mengyi_files/figure-html/viral-family-composition-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section></section><section id="human-infecting-virus-reads" class="level1" data-number="4"><h1 data-number="4">
<span class="header-section-number">4</span> Human-infecting virus reads</h1>
<section id="overall-relative-abundance" class="level2" data-number="4.1"><h2 data-number="4.1" class="anchored" data-anchor-id="overall-relative-abundance">
<span class="header-section-number">4.1</span> Overall relative abundance</h2>
<p>I calculated the relative abundance of human-infecting viruses in two ways:</p>
<ul>
<li><p>First, as the total number of deduplicated human-virus reads in each sample, divided by the number of raw reads (“All reads”). This results in a viral fraction of <span class="math inline">\(9.84 \times 10^{-5}\)</span> across all samples.</p></li>
<li><p>Second, as a fraction of preprocessed (cleaned, deduplicated, computationally ribodepleted) reads (“Preprocessed reads”). This results in a viral fraction of <span class="math inline">\(1.18 \times 10^{-4}\)</span> across all samples.</p></li>
</ul>
<p>Note that 10 of the pools had no viral reads: 5 from “Heilongjiang, Mudanjiang”, 4 from “Jiangsu, Nanjing”, and 1 from “Xinjiang, Urmqi”.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb20"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get raw read counts</span></span>
<span><span class="va">read_counts_raw</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">basic_stats_raw</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, n_reads_raw <span class="op">=</span> <span class="va">n_read_pairs</span><span class="op">)</span></span>
<span><span class="va">read_counts_preproc</span> <span class="op">&lt;-</span> <span class="va">basic_stats</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">stage</span> <span class="op">==</span> <span class="st">"ribo_initial"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, n_reads_preproc <span class="op">=</span> <span class="va">n_read_pairs</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get HV read counts</span></span>
<span><span class="va">read_counts_hv</span> <span class="op">&lt;-</span> <span class="va">mrg_hv</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">hv_status</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">count</span><span class="op">(</span>name<span class="op">=</span><span class="st">"n_reads_hv"</span><span class="op">)</span></span>
<span><span class="va">read_counts</span> <span class="op">&lt;-</span> <span class="va">read_counts_raw</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">read_counts_hv</span>, by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"sample"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>n_reads_hv <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">n_reads_hv</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">read_counts_preproc</span>, by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"sample"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"sample"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">n_reads_raw</span>, <span class="va">n_reads_preproc</span>, <span class="va">n_reads_hv</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>n_samples <span class="op">=</span> <span class="fl">1</span>,</span>
<span>         p_reads_total <span class="op">=</span> <span class="va">n_reads_hv</span><span class="op">/</span><span class="va">n_reads_raw</span>,</span>
<span>         p_reads_preproc <span class="op">=</span> <span class="va">n_reads_hv</span><span class="op">/</span><span class="va">n_reads_preproc</span><span class="op">)</span></span>
<span><span class="va">read_counts_long</span> <span class="op">&lt;-</span> <span class="va">read_counts</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_longer</span><span class="op">(</span><span class="fu">starts_with</span><span class="op">(</span><span class="st">"p_reads"</span><span class="op">)</span>, names_to<span class="op">=</span><span class="st">"read_group"</span>, values_to<span class="op">=</span><span class="st">"p_reads"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>read_group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">read_group</span> <span class="op">==</span> <span class="st">"p_reads_total"</span>, <span class="st">"All reads"</span>, <span class="st">"Preprocessed reads"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Combine for display</span></span>
<span><span class="va">read_counts_agg</span> <span class="op">&lt;-</span> <span class="va">read_counts</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_total <span class="op">=</span> <span class="va">n_reads_hv</span><span class="op">/</span><span class="va">n_reads_raw</span>,</span>
<span>         p_reads_preproc <span class="op">=</span> <span class="va">n_reads_hv</span><span class="op">/</span><span class="va">n_reads_preproc</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"sample"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">read_counts_agg_long</span> <span class="op">&lt;-</span> <span class="va">read_counts_agg</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_longer</span><span class="op">(</span><span class="fu">starts_with</span><span class="op">(</span><span class="st">"p_reads"</span><span class="op">)</span>, names_to<span class="op">=</span><span class="st">"read_group"</span>, values_to<span class="op">=</span><span class="st">"p_reads"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>read_group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">read_group</span> <span class="op">==</span> <span class="st">"p_reads_total"</span>, <span class="st">"All reads"</span>, <span class="st">"Preprocessed reads"</span><span class="op">)</span><span class="op">)</span> </span>
<span></span>
<span><span class="co"># Visualize</span></span>
<span><span class="co">#g_read_counts &lt;- ggplot(read_counts_agg_long, aes(x=library, y=p_reads)) +</span></span>
<span><span class="co">#  geom_point() +</span></span>
<span><span class="co">#  scale_y_log10(name = "Unique human-viral read fraction") +</span></span>
<span><span class="co">#  facet_grid(.~read_group, scales = "free") +</span></span>
<span><span class="co">#  theme_kit</span></span>
<span><span class="co">#g_read_counts</span></span>
<span></span>
<span><span class="co">#g_viral &lt;- ggplot(read_counts_agg_long, aes(x=p_reads, y=library, color=location, shape=read_group)) +</span></span>
<span><span class="co">#  geom_point() +</span></span>
<span><span class="co">#  scale_y_discrete(name="Plasma pool") +</span></span>
<span><span class="co">#  scale_x_log10(name="Viral read fraction") +</span></span>
<span><span class="co">#  scale_color_brewer(palette = "Dark2") +</span></span>
<span><span class="co">#  facet_grid(.~read_group, scales = "free") +</span></span>
<span><span class="co">#  guides(color=guide_legend(nrow=2), shape=guide_legend(nrow=2),</span></span>
<span><span class="co">#         linetype=guide_legend(nrow=2)) +</span></span>
<span><span class="co">#  theme_kit</span></span>
<span><span class="co">#g_viral</span></span>
<span></span>
<span><span class="va">hv_location</span> <span class="op">&lt;-</span> <span class="va">read_counts_agg_long</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">read_group</span> <span class="op">==</span> <span class="st">"Preprocessed reads"</span> <span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">location</span><span class="op">)</span> </span>
<span></span>
<span><span class="va">hv_location_counts</span> <span class="op">&lt;-</span> <span class="va">hv_location</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">location</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarise</span><span class="op">(</span>count <span class="op">=</span> <span class="fu">n</span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">location</span>, <span class="st">" (n="</span>, <span class="va">count</span>, <span class="st">")"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span>  <span class="co"># Create a named vector for easy mapping</span></span>
<span><span class="va">location_labels</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/setNames.html">setNames</a></span><span class="op">(</span><span class="va">hv_location_counts</span><span class="op">$</span><span class="va">label</span>, <span class="va">hv_location_counts</span><span class="op">$</span><span class="va">location</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create a named vector for easy mapping</span></span>
<span><span class="co"># Plot</span></span>
<span><span class="co"># g_viral &lt;- ggplot(p_reads_viral, aes(x=p_reads, y=library, color=location, shape=read_group)) +</span></span>
<span><span class="co">#   geom_point() +</span></span>
<span><span class="co">#   scale_y_discrete(name="Plasma pool") +</span></span>
<span><span class="co">#   scale_x_log10(name="Viral read fraction") +</span></span>
<span><span class="co">#   scale_color_brewer(palette = "Dark2") +</span></span>
<span><span class="co">#   #facet_grid(.~read_group, scales = "free") +</span></span>
<span><span class="co">#   guides(color=guide_legend(nrow=2), shape=guide_legend(nrow=2),</span></span>
<span><span class="co">#          linetype=guide_legend(nrow=2)) +</span></span>
<span><span class="co">#   theme_kit</span></span>
<span></span>
<span><span class="co">#ggplot(location_viral, aes(x=p_reads, y=location, color = location)) +</span></span>
<span><span class="co">#  geom_quasirandom(size=2) +</span></span>
<span><span class="co">#  scale_x_log10(name="Viral read fraction", labels = label_log(digits=2)) +</span></span>
<span><span class="co">#  scale_y_discrete(name="Location") +</span></span>
<span><span class="co">#  scale_color_brewer(palette = "Dark2") +</span></span>
<span><span class="co">#  theme_light() + </span></span>
<span><span class="co">#    theme(</span></span>
<span><span class="co">#    axis.text.y = element_text(size = 8),</span></span>
<span><span class="co">#    axis.text.x = element_text(size = 14),</span></span>
<span><span class="co">#    axis.ticks.y = element_blank(),</span></span>
<span><span class="co">#    axis.line = element_line(colour = "black"),</span></span>
<span><span class="co">#    axis.title.x = element_text(size = 15),    </span></span>
<span><span class="co">#    axis.title.y = element_text(size = 15),  </span></span>
<span><span class="co">#    legend.text = element_text(size = 13),</span></span>
<span><span class="co">#    legend.title = element_text(size = 16),</span></span>
<span><span class="co">#    legend.position = c(1, 1),  # Move legend to top right</span></span>
<span><span class="co">##    legend.justification = c(1, 1),  # Align legend to top right</span></span>
<span><span class="co">#    panel.grid.minor = element_blank(),</span></span>
<span><span class="co">#    panel.border = element_blank(),</span></span>
<span><span class="co">#    panel.background = element_blank())</span></span>
<span></span>
<span><span class="va">g_viral</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">hv_location</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">location</span>, y<span class="op">=</span><span class="va">p_reads</span>, fill<span class="op">=</span><span class="va">location</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_violin</span><span class="op">(</span>trim<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_boxplot</span><span class="op">(</span>width<span class="op">=</span><span class="fl">0.1</span>, fill<span class="op">=</span><span class="st">"white"</span>, color<span class="op">=</span><span class="st">"black"</span>, outlier.shape<span class="op">=</span><span class="cn">NA</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_jitter</span><span class="op">(</span>width<span class="op">=</span><span class="fl">0.1</span>, size<span class="op">=</span><span class="fl">0.5</span>, alpha<span class="op">=</span><span class="fl">0.5</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_log10</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Viral read fraction"</span>, labels <span class="op">=</span> <span class="fu">label_log</span><span class="op">(</span>digits<span class="op">=</span><span class="fl">2</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_discrete</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Location"</span>, labels <span class="op">=</span> <span class="va">location_labels</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Dark2"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme_light</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">theme</span><span class="op">(</span></span>
<span>    axis.text.y <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">12</span>, angle <span class="op">=</span> <span class="fl">45</span>, hjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>,</span>
<span>    axis.title.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">15</span><span class="op">)</span>,    </span>
<span>    axis.title.y <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">15</span><span class="op">)</span>,  </span>
<span>    legend.position <span class="op">=</span> <span class="st">"none"</span>,</span>
<span>    panel.grid.minor <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    panel.border <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    panel.background <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span> <span class="op">+</span></span>
<span>  <span class="fu">coord_flip</span><span class="op">(</span><span class="op">)</span>  <span class="co"># This flips the coordinates to make the plot horizontal</span></span>
<span></span>
<span><span class="va">g_viral</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-23-mengyi_files/figure-html/count-hv-reads-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section><section id="overall-taxonomy-and-composition" class="level2" data-number="4.2"><h2 data-number="4.2" class="anchored" data-anchor-id="overall-taxonomy-and-composition">
<span class="header-section-number">4.2</span> Overall taxonomy and composition</h2>
<p>The two dominant viruses we see are Anellovirdae and Hepadnaviridae, with Parvoviridae . The threshold for the label “other” are the set of families that make up less than 5% composition in all samples.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb21"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">threshold_major_family</span> <span class="op">&lt;-</span> <span class="fl">0.05</span></span>
<span></span>
<span><span class="co"># Count reads for each human-viral family</span></span>
<span><span class="va">hv_family_counts</span> <span class="op">&lt;-</span> <span class="va">hv_reads_family</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">count</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"n_reads_hv"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_hv <span class="op">=</span> <span class="va">n_reads_hv</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Identify high-ranking families and group others</span></span>
<span><span class="va">hv_family_major_tab</span> <span class="op">&lt;-</span> <span class="va">hv_family_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_hv</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_hv</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">row_number</span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p_reads_hv</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_hv</span> <span class="op">&gt;</span> <span class="va">threshold_major_family</span><span class="op">)</span></span>
<span><span class="va">hv_family_counts_major</span> <span class="op">&lt;-</span> <span class="va">hv_family_counts</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">hv_family_major_tab</span><span class="op">$</span><span class="va">name</span>, <span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">name_display</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_hv <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>, p_reads_hv <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads_hv</span><span class="op">)</span>, </span>
<span>            .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">name_display</span>, </span>
<span>                               levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">hv_family_major_tab</span><span class="op">$</span><span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">hv_family_counts_display</span> <span class="op">&lt;-</span> <span class="va">hv_family_counts_major</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">p_reads_hv</span>, classification <span class="op">=</span> <span class="va">name_display</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by <span class="op">=</span> <span class="st">'sample'</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_hv_family</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>data<span class="op">=</span><span class="va">hv_family_counts_display</span>, position <span class="op">=</span> <span class="st">"stack"</span>, width<span class="op">=</span><span class="fl">1</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% HV Reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, </span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set1"</span>, name <span class="op">=</span> <span class="st">"Viral family"</span><span class="op">)</span> <span class="op">+</span></span>
<span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title<span class="op">=</span><span class="st">"Family composition of human-viral reads"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>fill<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>ncol<span class="op">=</span><span class="fl">4</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fu">rel</span><span class="op">(</span><span class="fl">1.4</span><span class="op">)</span>, hjust<span class="op">=</span><span class="fl">0</span>, face<span class="op">=</span><span class="st">"plain"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>  </span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span> <span class="va">location</span>, scales <span class="op">=</span> <span class="st">"free_x"</span>, ncol <span class="op">=</span> <span class="fl">2</span><span class="op">)</span> </span>
<span><span class="va">g_hv_family</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-23-mengyi_files/figure-html/hv-family-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb22"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get most prominent families for text</span></span>
<span><span class="va">hv_family_collate</span> <span class="op">&lt;-</span> <span class="va">hv_family_counts</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_tot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>,</span>
<span>            p_reads_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_hv</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">n_reads_tot</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">display_hv_family_collate</span> <span class="op">&lt;-</span> <span class="va">hv_family_collate</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">name</span>, <span class="va">n_reads_tot</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>`Family` <span class="op">=</span> <span class="va">name</span>, `Total number of reads` <span class="op">=</span> <span class="va">n_reads_tot</span><span class="op">)</span></span>
<span><span class="va">display_hv_family_collate</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["Family"],"name":[1],"type":["chr"],"align":["left"]},{"label":["Total number of reads"],"name":[2],"type":["int"],"align":["right"]}],"data":[{"1":"Hepadnaviridae","2":"339427"},{"1":"Parvoviridae","2":"17401"},{"1":"Anelloviridae","2":"3001"},{"1":"Orthoherpesviridae","2":"298"},{"1":"Retroviridae","2":"115"},{"1":"Microviridae","2":"61"},{"1":"Adenoviridae","2":"28"},{"1":"Coronaviridae","2":"2"},{"1":"Poxviridae","2":"2"},{"1":"Caudovirus D_HF5_3","2":"1"},{"1":"Flaviviridae","2":"1"},{"1":"Papillomaviridae","2":"1"},{"1":"Paramyxoviridae","2":"1"},{"1":"Polyomaviridae","2":"1"},{"1":"Rhabdoviridae","2":"1"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
</section><section id="analyzing-specific-families" class="level2" data-number="4.3"><h2 data-number="4.3" class="anchored" data-anchor-id="analyzing-specific-families">
<span class="header-section-number">4.3</span> Analyzing specific families</h2>
<p>We now investigate the composition of specific families that had more than 5 viral reads. In investigating individual viral families, to avoid distortions from a few rare reads, I restricted myself to samples where that family made up at least 10% of human-viral reads:</p>
<section id="hepadnaviridae-number-of-reads-339427" class="level3" data-number="4.3.1"><h3 data-number="4.3.1" class="anchored" data-anchor-id="hepadnaviridae-number-of-reads-339427">
<span class="header-section-number">4.3.1</span> Hepadnaviridae (Number of reads: 339,427)</h3>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb23"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Hepadnaviridae</span></span>
<span><span class="fu">plot_viral_family_histogram</span><span class="op">(</span>taxid_chosen<span class="op">=</span><span class="fl">10404</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-23-mengyi_files/figure-html/plot-hepadnaviridae-histogram-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb24"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Hepadnaviridae</span></span>
<span><span class="fu">plot_viral_family_composition</span><span class="op">(</span>taxid_chosen<span class="op">=</span><span class="fl">10404</span>, threshold_major_species <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-23-mengyi_files/figure-html/plot-hepadnaviridae-composition-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section><section id="parvoviridae-number-of-reads-17401" class="level3" data-number="4.3.2"><h3 data-number="4.3.2" class="anchored" data-anchor-id="parvoviridae-number-of-reads-17401">
<span class="header-section-number">4.3.2</span> Parvoviridae (Number of reads: 17,401)</h3>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb25"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Parvoviridae</span></span>
<span><span class="fu">plot_viral_family_histogram</span><span class="op">(</span>taxid_chosen<span class="op">=</span><span class="fl">10780</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-23-mengyi_files/figure-html/plot-parvoviridae-histogram-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb26"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Parvoviridae</span></span>
<span><span class="fu">plot_viral_family_composition</span><span class="op">(</span>taxid_chosen<span class="op">=</span><span class="fl">10780</span>, threshold_major_species <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-23-mengyi_files/figure-html/plot-parvoviridae-composition-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section><section id="anelloviridae-number-of-reads-3001" class="level3" data-number="4.3.3"><h3 data-number="4.3.3" class="anchored" data-anchor-id="anelloviridae-number-of-reads-3001">
<span class="header-section-number">4.3.3</span> Anelloviridae (Number of reads: 3001)</h3>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb27"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Anelloviridae</span></span>
<span><span class="fu">plot_viral_family_histogram</span><span class="op">(</span>taxid_chosen<span class="op">=</span><span class="fl">687329</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-23-mengyi_files/figure-html/plot-anelloviridae-histogram-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb28"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Anelloviridae</span></span>
<span><span class="fu">plot_viral_family_composition</span><span class="op">(</span>taxid_chosen<span class="op">=</span><span class="fl">687329</span>, threshold_major_species <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-23-mengyi_files/figure-html/plot-anelloviridae-composition-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section><section id="orthoherpesviridae-number-of-reads-298" class="level3" data-number="4.3.4"><h3 data-number="4.3.4" class="anchored" data-anchor-id="orthoherpesviridae-number-of-reads-298">
<span class="header-section-number">4.3.4</span> Orthoherpesviridae (Number of reads: 298)</h3>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb29"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Orthoherpesviridae</span></span>
<span><span class="fu">plot_viral_family_histogram</span><span class="op">(</span>taxid_chosen<span class="op">=</span><span class="fl">3044472</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-23-mengyi_files/figure-html/plot-orthoherpesviridae-histogram-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb30"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Orthoherpesviridae</span></span>
<span><span class="fu">plot_viral_family_composition</span><span class="op">(</span>taxid_chosen<span class="op">=</span><span class="fl">3044472</span>, threshold_major_species <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-23-mengyi_files/figure-html/plot-orthoherpesviridae-composition-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section><section id="retroviridae-number-of-reads-115" class="level3" data-number="4.3.5"><h3 data-number="4.3.5" class="anchored" data-anchor-id="retroviridae-number-of-reads-115">
<span class="header-section-number">4.3.5</span> Retroviridae (Number of reads: 115)</h3>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb31"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Retroviridae</span></span>
<span><span class="fu">plot_viral_family_histogram</span><span class="op">(</span>taxid_chosen<span class="op">=</span><span class="fl">11632</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-23-mengyi_files/figure-html/plot-retroviridae-histogram-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb32"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Retroviridae</span></span>
<span><span class="fu">plot_viral_family_composition</span><span class="op">(</span>taxid_chosen<span class="op">=</span><span class="fl">11632</span>, threshold_major_species <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-23-mengyi_files/figure-html/plot-retroviridae-composition-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section><section id="microviridae-number-of-reads-61" class="level3" data-number="4.3.6"><h3 data-number="4.3.6" class="anchored" data-anchor-id="microviridae-number-of-reads-61">
<span class="header-section-number">4.3.6</span> Microviridae (Number of reads: 61)</h3>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb33"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Microviridae</span></span>
<span><span class="fu">plot_viral_family_histogram</span><span class="op">(</span>taxid_chosen<span class="op">=</span><span class="fl">10841</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-23-mengyi_files/figure-html/plot-microviridae-histogram-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb34"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Microviridae</span></span>
<span><span class="fu">plot_viral_family_composition</span><span class="op">(</span>taxid_chosen<span class="op">=</span><span class="fl">10841</span>, threshold_major_species <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-23-mengyi_files/figure-html/plot-microviridae-composition-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section><section id="adenoviridae-number-of-reads-28" class="level3" data-number="4.3.7"><h3 data-number="4.3.7" class="anchored" data-anchor-id="adenoviridae-number-of-reads-28">
<span class="header-section-number">4.3.7</span> Adenoviridae (Number of reads: 28)</h3>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb35"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Adenoviridae</span></span>
<span><span class="fu">plot_viral_family_histogram</span><span class="op">(</span>taxid_chosen<span class="op">=</span><span class="fl">10508</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-23-mengyi_files/figure-html/plot-adenoviridae-histogram-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb36"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Adenoviridae</span></span>
<span><span class="fu">plot_viral_family_composition</span><span class="op">(</span>taxid_chosen<span class="op">=</span><span class="fl">10508</span>, threshold_major_species <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-23-mengyi_files/figure-html/plot-adenoviridae-composition-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section></section><section id="relative-abundance-of-pathogenic-viruses-of-interest" class="level2" data-number="4.4"><h2 data-number="4.4" class="anchored" data-anchor-id="relative-abundance-of-pathogenic-viruses-of-interest">
<span class="header-section-number">4.4</span> Relative abundance of pathogenic viruses of interest</h2>
<p>Each dot represents a sample, colored by viral family. The x-axis shows the relative abundance of human-infecting viruses, and the y-axis shows the species.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb37"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">species_family</span> <span class="op">&lt;-</span> <span class="va">result</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">species</span>, <span class="va">family</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">rename</span><span class="op">(</span><span class="st">'name'</span> <span class="op">=</span> <span class="st">'species'</span><span class="op">)</span></span>
<span></span>
<span><span class="va">play</span> <span class="op">&lt;-</span> <span class="va">hv_species_counts</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">ungroup</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by <span class="op">=</span> <span class="st">'sample'</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">species_family</span>, by <span class="op">=</span> <span class="st">'name'</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">name</span> <span class="op">==</span> <span class="st">"Severe acute respiratory syndrome-related coronavirus"</span>, <span class="st">"SARS-related coronavirus"</span>, <span class="va">name</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span><span class="op">(</span><span class="va">family</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Anelloviridae"</span>, <span class="cn">NA</span>, <span class="st">"Microviridae"</span>, <span class="st">"Rhabdoviridae"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> </span>
<span></span>
<span><span class="va">adjusted_play</span> <span class="op">&lt;-</span> <span class="va">play</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>virus_prevalence_num <span class="op">=</span> <span class="fu">n_distinct</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span><span class="op">/</span><span class="fu">n_distinct</span><span class="op">(</span><span class="va">libraries</span><span class="op">)</span>,</span>
<span>         total_reads_hv <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ungroup</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name <span class="op">=</span> <span class="fu">fct_reorder</span><span class="op">(</span><span class="va">name</span>, <span class="va">virus_prevalence_num</span>, .desc<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">name</span>, <span class="va">ra_reads_hv</span>, <span class="va">family</span>, <span class="va">location</span>, <span class="va">virus_prevalence_num</span>, <span class="va">total_reads_hv</span>, <span class="va">n_reads_hv</span><span class="op">)</span></span>
<span></span>
<span><span class="va">pal</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu">brewer.pal</span><span class="op">(</span><span class="fl">8</span>, <span class="st">'Dark2'</span><span class="op">)</span>,<span class="fu">brewer.pal</span><span class="op">(</span><span class="fl">8</span>, <span class="st">'Accent'</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#, labels = label_log(digits=2)</span></span>
<span></span>
<span><span class="va">ra_dot</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">adjusted_play</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">ra_reads_hv</span>, y<span class="op">=</span><span class="va">name</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_quasirandom</span><span class="op">(</span>orientation <span class="op">=</span> <span class="st">'y'</span>, <span class="fu">aes</span><span class="op">(</span>color <span class="op">=</span> <span class="va">family</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_manual</span><span class="op">(</span>values <span class="op">=</span> <span class="va">pal</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu">scale_x_log10</span><span class="op">(</span></span>
<span>    <span class="st">"Relative abundance human virus reads"</span>,</span>
<span>    labels<span class="op">=</span><span class="fu">label_log</span><span class="op">(</span>digits<span class="op">=</span><span class="fl">3</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>y <span class="op">=</span>  <span class="st">""</span>,</span>
<span>       color <span class="op">=</span> <span class="st">'Viral family'</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">guides</span><span class="op">(</span>color <span class="op">=</span> <span class="fu">guide_legend</span><span class="op">(</span>ncol<span class="op">=</span><span class="fl">2</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme_light</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu">theme</span><span class="op">(</span></span>
<span>    axis.text.y <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,</span>
<span>    axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    axis.ticks.y <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,axis.line <span class="op">=</span> <span class="fu">element_line</span><span class="op">(</span>colour <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span>,</span>
<span>    axis.title.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">14</span><span class="op">)</span>,    </span>
<span>    legend.text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,</span>
<span>    legend.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span>, face<span class="op">=</span><span class="st">"bold"</span><span class="op">)</span>,</span>
<span>    legend.position <span class="op">=</span> <span class="st">'bottom'</span>,</span>
<span>    <span class="co">#legend.position = c(1, 1),  # Move legend to top right</span></span>
<span>    <span class="co">#legend.justification = c(1, 1),  # Align legend to top right</span></span>
<span>    panel.grid.minor <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    panel.border <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    panel.background <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">ra_dot</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-07-23-mengyi_files/figure-html/plot-pathogenic-viruses-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb38"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co">#  geom_text(aes(label = total_reads_hv, y = name), </span></span>
<span><span class="co">#            x = 1e-9, hjust = 0, vjust = 0.5, size = 3, </span></span>
<span><span class="co">#            check_overlap = TRUE)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>We can ask Claude to analyze the pathogenicity of these viruses along with their popular names.</p>
<table class="caption-top table">
<colgroup>
<col style="width: 25%">
<col style="width: 37%">
<col style="width: 37%">
</colgroup>
<thead><tr class="header">
<th>Scientific Name</th>
<th>Popular/Well-known Name</th>
<th>Pathogenicity to Humans</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Human immunodeficiency virus 1</td>
<td>HIV-1</td>
<td>Very High</td>
</tr>
<tr class="even">
<td>SARS-related coronavirus</td>
<td>SARS-CoV</td>
<td>High</td>
</tr>
<tr class="odd">
<td>Hepatitis B virus</td>
<td>HBV</td>
<td>High</td>
</tr>
<tr class="even">
<td>Hepacivirus hominis</td>
<td>Hepatitis C virus (HCV)</td>
<td>High</td>
</tr>
<tr class="odd">
<td>Simplexvirus humanalpha1</td>
<td>Herpes simplex virus 1 (HSV-1)</td>
<td>Moderate to High</td>
</tr>
<tr class="even">
<td>Cytomegalovirus humanbeta5</td>
<td>Human cytomegalovirus (HCMV)</td>
<td>Moderate to High</td>
</tr>
<tr class="odd">
<td>Lymphocryptovirus humangamma4</td>
<td>Epstein-Barr virus (EBV)</td>
<td>Moderate</td>
</tr>
<tr class="even">
<td>Human mastadenovirus C</td>
<td>Adenovirus C</td>
<td>Moderate</td>
</tr>
<tr class="odd">
<td>Human mastadenovirus F</td>
<td>Adenovirus F</td>
<td>Moderate</td>
</tr>
<tr class="even">
<td>Roseolovirus humanbeta6a</td>
<td>Human herpesvirus 6A (HHV-6A)</td>
<td>Low to Moderate</td>
</tr>
<tr class="odd">
<td>Roseolovirus humanbeta6b</td>
<td>Human herpesvirus 6B (HHV-6B)</td>
<td>Low to Moderate</td>
</tr>
<tr class="even">
<td>Roseolovirus humanbeta7</td>
<td>Human herpesvirus 7 (HHV-7)</td>
<td>Low to Moderate</td>
</tr>
<tr class="odd">
<td>Rhadinovirus humangamma8</td>
<td>Kaposi’s sarcoma-associated herpesvirus (KSHV)</td>
<td>Low to Moderate</td>
</tr>
<tr class="even">
<td>Molluscum contagiosum virus</td>
<td>MCV</td>
<td>Low</td>
</tr>
<tr class="odd">
<td>Erythroparvovirus primate1</td>
<td>Parvovirus B19</td>
<td>Low</td>
</tr>
<tr class="even">
<td>Human erythrovirus V9</td>
<td>Erythrovirus V9</td>
<td>Low</td>
</tr>
<tr class="odd">
<td>Alphapapillomavirus 4</td>
<td>Human papillomavirus 4 (HPV-4)</td>
<td>Low</td>
</tr>
<tr class="even">
<td>Betapolyomavirus hominis</td>
<td>Human polyomavirus</td>
<td>Low</td>
</tr>
<tr class="odd">
<td>Dependoparvovirus primate1</td>
<td>Adeno-associated virus (AAV)</td>
<td>Very Low</td>
</tr>
<tr class="even">
<td>Murine leukemia virus</td>
<td>MLV</td>
<td>Very Low (not a human pathogen)</td>
</tr>
<tr class="odd">
<td>Murine leukemia-related retroviruses</td>
<td>MLV-related viruses</td>
<td>Very Low (not typical human pathogens)</td>
</tr>
</tbody>
</table>
<p>We get a variety of pathogenic viruses, none of which are a suprise to see in plasma (other than SARS-CoV which we determined was contamination <a href="https://twist.com/a/197793/ch/619193/t/6323010/">on twist</a>).</p>
</section><section id="relative-abundance-assuming-perfect-human-read-removal" class="level2" data-number="4.5"><h2 data-number="4.5" class="anchored" data-anchor-id="relative-abundance-assuming-perfect-human-read-removal">
<span class="header-section-number">4.5</span> Relative abundance assuming perfect human read removal</h2>
<p>Assuming we’re able to perfectly remove all human reads, the average relative abundance of known human infecting virus is <span class="math inline">\(5.37 \times 10^{-4}\)</span>.</p>
</section></section><section id="conclusion" class="level1" data-number="5"><h1 data-number="5">
<span class="header-section-number">5</span> Conclusion</h1>
<p>There were some interesting takeways from this analysis:</p>
<ol type="1">
<li>Human read fraction remains quite high in plasma if special sample preparation/library preparation is not taken (see <a href="https://pubmed.ncbi.nlm.nih.gov/33767340/">here</a>).</li>
<li>There is a lot of Anelloviridae in blood, including that which has not been classified by species.</li>
<li>Pathogenic viruses such as HIV and HBV, and even latent viruses like Herpes are picked up by plasma MGS</li>
<li>Respiratory viruses are also picked up by plasma MGS (SARS), but not as much as other viruses.</li>
</ol>
<p><em>We do note that we’ve analyzed other datasets and it seems that having a large catchment is crucial, as we’d expect, for picking up these more rare, pathogenic viruses</em>. Using this data, we can also look at China’s HIV/HBV reported incidence to do a P2RA/swab sampling type analysis. We plan to do more analysis on this data, but wanted to share some preliminary results.</p>
</section><section id="appendix" class="level1" data-number="6"><h1 data-number="6">
<span class="header-section-number">6</span> Appendix</h1>
<section id="human-infecting-virus-families-genera-and-species" class="level2" data-number="6.1"><h2 data-number="6.1" class="anchored" data-anchor-id="human-infecting-virus-families-genera-and-species">
<span class="header-section-number">6.1</span> Human-infecting virus families, genera, and species</h2>
<p>To get a good overview of families, genera, and species, we can look at a Sankey plot where the magnitude of relative abundance, averaged over all samples, is shown in parentheses.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb39"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Function to create links</span></span>
<span><span class="va">create_links</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">data</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">family_to_genus</span> <span class="op">&lt;-</span> <span class="va">data</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">genus</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">group_by</span><span class="op">(</span><span class="va">family</span>, <span class="va">genus</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">summarise</span><span class="op">(</span>value <span class="op">=</span> <span class="fu">n</span><span class="op">(</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>source <span class="op">=</span> <span class="va">family</span>, target <span class="op">=</span> <span class="va">genus</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="va">genus_to_species</span> <span class="op">&lt;-</span> <span class="va">data</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">group_by</span><span class="op">(</span><span class="va">genus</span>, <span class="va">species</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">summarise</span><span class="op">(</span>value <span class="op">=</span> <span class="fu">n</span><span class="op">(</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>source <span class="op">=</span> <span class="va">genus</span>, target <span class="op">=</span> <span class="va">species</span><span class="op">)</span></span>
<span></span>
<span>  <span class="va">family_to_species</span> <span class="op">&lt;-</span> <span class="va">data</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">genus</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">group_by</span><span class="op">(</span><span class="va">family</span>, <span class="va">species</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">summarise</span><span class="op">(</span>value <span class="op">=</span> <span class="fu">n</span><span class="op">(</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>source <span class="op">=</span> <span class="va">family</span>, target <span class="op">=</span> <span class="va">species</span><span class="op">)</span></span>
<span></span>
<span>  <span class="fu">bind_rows</span><span class="op">(</span><span class="va">family_to_genus</span>, <span class="va">genus_to_species</span>, <span class="va">family_to_species</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">source</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Function to create nodes</span></span>
<span><span class="va">create_nodes</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">links</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span></span>
<span>    name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">links</span><span class="op">$</span><span class="va">source</span>, <span class="va">links</span><span class="op">$</span><span class="va">target</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Function to prepare data for Sankey diagram</span></span>
<span><span class="va">prepare_sankey_data</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">links</span>, <span class="va">nodes</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">links</span><span class="op">$</span><span class="va">IDsource</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/match.html">match</a></span><span class="op">(</span><span class="va">links</span><span class="op">$</span><span class="va">source</span>, <span class="va">nodes</span><span class="op">$</span><span class="va">name</span><span class="op">)</span> <span class="op">-</span> <span class="fl">1</span></span>
<span>  <span class="va">links</span><span class="op">$</span><span class="va">IDtarget</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/match.html">match</a></span><span class="op">(</span><span class="va">links</span><span class="op">$</span><span class="va">target</span>, <span class="va">nodes</span><span class="op">$</span><span class="va">name</span><span class="op">)</span> <span class="op">-</span> <span class="fl">1</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>links <span class="op">=</span> <span class="va">links</span>, nodes <span class="op">=</span> <span class="va">nodes</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Function to create Sankey plot</span></span>
<span><span class="va">create_sankey_plot</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">sankey_data</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu">sankeyNetwork</span><span class="op">(</span></span>
<span>    Links <span class="op">=</span> <span class="va">sankey_data</span><span class="op">$</span><span class="va">links</span>, </span>
<span>    Nodes <span class="op">=</span> <span class="va">sankey_data</span><span class="op">$</span><span class="va">nodes</span>,</span>
<span>    Source <span class="op">=</span> <span class="st">"IDsource"</span>, </span>
<span>    Target <span class="op">=</span> <span class="st">"IDtarget"</span>,</span>
<span>    Value <span class="op">=</span> <span class="st">"value"</span>, </span>
<span>    NodeID <span class="op">=</span> <span class="st">"name"</span>,</span>
<span>    sinksRight <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    nodeWidth <span class="op">=</span> <span class="fl">25</span>,</span>
<span>    fontSize <span class="op">=</span> <span class="fl">14</span>,</span>
<span>  <span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="va">save_sankey_as_png</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">sankey_plot</span>, <span class="va">width</span> <span class="op">=</span> <span class="fl">1000</span>, <span class="va">height</span> <span class="op">=</span> <span class="fl">800</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="co"># Save the plot as an HTML file</span></span>
<span>  <span class="fu">saveWidget</span><span class="op">(</span><span class="va">sankey_plot</span>, <span class="fu"><a href="https://rdrr.io/r/base/sprintf.html">sprintf</a></span><span class="op">(</span><span class="st">'%s/sankey.html'</span>,<span class="va">data_dir</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="va">format_scientific</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">digits</span><span class="op">=</span><span class="fl">2</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">x</span>, <span class="kw">function</span><span class="op">(</span><span class="va">val</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">val</span><span class="op">)</span> <span class="op">||</span> <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span><span class="op">(</span><span class="va">val</span><span class="op">)</span> <span class="op">&lt;</span> <span class="fl">1e-15</span><span class="op">)</span> <span class="op">{</span></span>
<span>      <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="st">"0"</span><span class="op">)</span></span>
<span>    <span class="op">}</span> <span class="kw">else</span> <span class="op">{</span></span>
<span>      <span class="va">exponent</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">floor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span><span class="op">(</span><span class="va">val</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span>      <span class="va">coef</span> <span class="op">&lt;-</span> <span class="va">val</span> <span class="op">/</span> <span class="fl">10</span><span class="op">^</span><span class="va">exponent</span></span>
<span>      <span class="co">#return(sprintf("%.1f × 10^%d", round(coef, digits), exponent))</span></span>
<span>      <span class="co"># May or may not be smart, just keeping magnitude</span></span>
<span>      <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sprintf.html">sprintf</a></span><span class="op">(</span><span class="st">"10^%d"</span>, <span class="va">exponent</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="op">}</span></span>
<span>  <span class="op">}</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="va">result</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span><span class="fu">across</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">genus_n_reads_tot</span>, <span class="va">genus_ra_reads_tot</span><span class="op">)</span>, <span class="op">~</span><span class="fu">replace_na</span><span class="op">(</span><span class="va">.</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span>,</span>
<span>         genus <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">genus</span><span class="op">)</span>, <span class="st">"Unknown Genus"</span>, <span class="va">genus</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>  species <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">species</span>, <span class="fu"><a href="https://rdrr.io/r/base/sprintf.html">sprintf</a></span><span class="op">(</span><span class="st">' (%s)'</span>, <span class="fu">format_scientific</span><span class="op">(</span><span class="va">species_ra_reads_tot</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,</span>
<span>  genus <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">genus</span>, <span class="fu"><a href="https://rdrr.io/r/base/sprintf.html">sprintf</a></span><span class="op">(</span><span class="st">' (%s)'</span>, <span class="fu">format_scientific</span><span class="op">(</span><span class="va">genus_ra_reads_tot</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,</span>
<span>  family <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">family</span>, <span class="fu"><a href="https://rdrr.io/r/base/sprintf.html">sprintf</a></span><span class="op">(</span><span class="st">' (%s)'</span>, <span class="fu">format_scientific</span><span class="op">(</span><span class="va">family_ra_reads_tot</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="va">links</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="fu">create_links</span><span class="op">(</span><span class="va">data</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">nodes</span> <span class="op">&lt;-</span> <span class="fu">create_nodes</span><span class="op">(</span><span class="va">links</span><span class="op">)</span></span>
<span><span class="va">sankey_data</span> <span class="op">&lt;-</span> <span class="fu">prepare_sankey_data</span><span class="op">(</span><span class="va">links</span>, <span class="va">nodes</span><span class="op">)</span></span>
<span><span class="va">sankey</span> <span class="op">&lt;-</span> <span class="fu">create_sankey_plot</span><span class="op">(</span><span class="va">sankey_data</span><span class="op">)</span></span>
<span></span>
<span><span class="va">sankey</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div id="htmlwidget-ac1f7e7e5c990ca74b39" style="width:100%;height:1462px;" class="sankeyNetwork html-widget"></div>
<script type="application/json" data-for="htmlwidget-ac1f7e7e5c990ca74b39">{"x":{"links":{"source":[0,1,1,1,1,2,3,4,5,6,7,7,7,7,7,8,9,10,10,11,12,13,13,13,14,15,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,17,18,19,19,19,19,19,20,21,22,22,23,23,24,24,24,24,24,24,24,24,24,24,24,24,24,24,25,26,26,27,28,28,29,30,31,32,33,33,33,34,35,35,35,35,35,35,35,35,35,35,36],"target":[28,16,19,24,35,17,25,30,35,35,20,27,32,33,34,15,31,21,22,18,29,23,26,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111],"value":[2,22,5,14,4,1,1,1,4,1,1,1,1,3,1,1,1,1,2,1,1,2,2,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1]},"nodes":{"name":["Adenoviridae (10^-7)","Anelloviridae (10^-6)","Coronaviridae (10^-8)","Flaviviridae (10^-8)","Hepadnaviridae (10^-4)","Microviridae (10^-7)","NA (0)","Orthoherpesviridae (10^-7)","Papillomaviridae (10^-8)","Paramyxoviridae (10^-8)","Parvoviridae (10^-5)","Polyomaviridae (10^-8)","Poxviridae (10^-8)","Retroviridae (10^-7)","Rhabdoviridae (10^-8)","Alphapapillomavirus (10^-8)","Alphatorquevirus (10^-7)","Betacoronavirus (10^-8)","Betapolyomavirus (10^-8)","Betatorquevirus (10^-8)","Cytomegalovirus (10^-8)","Dependoparvovirus (10^-8)","Erythroparvovirus (10^-5)","Gammaretrovirus (10^-8)","Gammatorquevirus (10^-7)","Hepacivirus (10^-8)","Lentivirus (10^-7)","Lymphocryptovirus (10^-7)","Mastadenovirus (10^-7)","Molluscipoxvirus (10^-8)","Orthohepadnavirus (10^-4)","Orthorubulavirus (10^-8)","Rhadinovirus (10^-8)","Roseolovirus (10^-7)","Simplexvirus (10^-7)","Unknown Genus (0)","Vesiculovirus (10^-8)","Alphapapillomavirus 4 (10^-8)","Alphatorquevirus homin1 (10^-8)","Alphatorquevirus homin10 (10^-7)","Alphatorquevirus homin13 (10^-8)","Alphatorquevirus homin15 (10^-8)","Alphatorquevirus homin17 (10^-8)","Alphatorquevirus homin18 (10^-8)","Alphatorquevirus homin19 (10^-8)","Alphatorquevirus homin20 (10^-8)","Alphatorquevirus homin21 (10^-8)","Alphatorquevirus homin24 (10^-8)","Alphatorquevirus homin29 (10^-8)","Alphatorquevirus homin3 (10^-8)","Alphatorquevirus homin5 (10^-8)","Alphatorquevirus homin7 (10^-8)","Alphatorquevirus homin9 (10^-7)","Torque teno virus 11 (10^-8)","Torque teno virus 12 (10^-8)","Torque teno virus 16 (10^-8)","Torque teno virus 22 (10^-8)","Torque teno virus 27 (10^-8)","Torque teno virus 28 (10^-8)","Torque teno virus 8 (10^-8)","Severe acute respiratory syndrome-related coronavirus (10^-8)","Betapolyomavirus hominis (10^-8)","Betatorquevirus homini1 (10^-8)","Betatorquevirus homini5 (10^-8)","Betatorquevirus homini6 (10^-8)","Betatorquevirus homini7 (10^-8)","TTV-like mini virus (10^-8)","Cytomegalovirus humanbeta5 (10^-8)","Dependoparvovirus primate1 (10^-8)","Erythroparvovirus primate1 (10^-5)","Human erythrovirus V9 (10^-7)","Murine leukemia virus (10^-8)","Murine leukemia-related retroviruses (10^-8)","Gammatorquevirus homidi1 (10^-8)","Gammatorquevirus homidi10 (10^-8)","Gammatorquevirus homidi11 (10^-8)","Gammatorquevirus homidi12 (10^-8)","Gammatorquevirus homidi13 (10^-8)","Gammatorquevirus homidi14 (10^-8)","Gammatorquevirus homidi2 (10^-8)","Gammatorquevirus homidi3 (10^-8)","Gammatorquevirus homidi4 (10^-8)","Gammatorquevirus homidi5 (10^-8)","Gammatorquevirus homidi6 (10^-8)","Gammatorquevirus homidi7 (10^-8)","Gammatorquevirus homidi8 (10^-7)","Gammatorquevirus homidi9 (10^-8)","Hepacivirus hominis (10^-8)","Human immunodeficiency virus (10^-8)","Human immunodeficiency virus 1 (10^-7)","Lymphocryptovirus humangamma4 (10^-7)","Human mastadenovirus C (10^-7)","Human mastadenovirus F (10^-7)","Molluscum contagiosum virus (10^-8)","Hepatitis B virus (10^-4)","Orthorubulavirus mammalis (10^-8)","Rhadinovirus humangamma8 (10^-8)","Roseolovirus humanbeta6a (10^-7)","Roseolovirus humanbeta6b (10^-7)","Roseolovirus humanbeta7 (10^-8)","Simplexvirus humanalpha1 (10^-7)","Anelloviridae sp. (10^-7)","Caudovirus D_HF5_3 (10^-8)","Gokushovirus WZ-2015a (10^-8)","Human endogenous retrovirus K (10^-8)","Human gut gokushovirus (10^-8)","Micro Torque teno virus (10^-8)","Microviridae sp. (10^-7)","Microvirus sp. (10^-8)","Small anellovirus (10^-8)","Torque teno virus (10^-7)","Vesiculovirus indiana (10^-8)"],"group":["Adenoviridae (10^-7)","Anelloviridae (10^-6)","Coronaviridae (10^-8)","Flaviviridae (10^-8)","Hepadnaviridae (10^-4)","Microviridae (10^-7)","NA (0)","Orthoherpesviridae (10^-7)","Papillomaviridae (10^-8)","Paramyxoviridae (10^-8)","Parvoviridae (10^-5)","Polyomaviridae (10^-8)","Poxviridae (10^-8)","Retroviridae (10^-7)","Rhabdoviridae (10^-8)","Alphapapillomavirus (10^-8)","Alphatorquevirus (10^-7)","Betacoronavirus (10^-8)","Betapolyomavirus (10^-8)","Betatorquevirus (10^-8)","Cytomegalovirus (10^-8)","Dependoparvovirus (10^-8)","Erythroparvovirus (10^-5)","Gammaretrovirus (10^-8)","Gammatorquevirus (10^-7)","Hepacivirus (10^-8)","Lentivirus (10^-7)","Lymphocryptovirus (10^-7)","Mastadenovirus (10^-7)","Molluscipoxvirus (10^-8)","Orthohepadnavirus (10^-4)","Orthorubulavirus (10^-8)","Rhadinovirus (10^-8)","Roseolovirus (10^-7)","Simplexvirus (10^-7)","Unknown Genus (0)","Vesiculovirus (10^-8)","Alphapapillomavirus 4 (10^-8)","Alphatorquevirus homin1 (10^-8)","Alphatorquevirus homin10 (10^-7)","Alphatorquevirus homin13 (10^-8)","Alphatorquevirus homin15 (10^-8)","Alphatorquevirus homin17 (10^-8)","Alphatorquevirus homin18 (10^-8)","Alphatorquevirus homin19 (10^-8)","Alphatorquevirus homin20 (10^-8)","Alphatorquevirus homin21 (10^-8)","Alphatorquevirus homin24 (10^-8)","Alphatorquevirus homin29 (10^-8)","Alphatorquevirus homin3 (10^-8)","Alphatorquevirus homin5 (10^-8)","Alphatorquevirus homin7 (10^-8)","Alphatorquevirus homin9 (10^-7)","Torque teno virus 11 (10^-8)","Torque teno virus 12 (10^-8)","Torque teno virus 16 (10^-8)","Torque teno virus 22 (10^-8)","Torque teno virus 27 (10^-8)","Torque teno virus 28 (10^-8)","Torque teno virus 8 (10^-8)","Severe acute respiratory syndrome-related coronavirus (10^-8)","Betapolyomavirus hominis (10^-8)","Betatorquevirus homini1 (10^-8)","Betatorquevirus homini5 (10^-8)","Betatorquevirus homini6 (10^-8)","Betatorquevirus homini7 (10^-8)","TTV-like mini virus (10^-8)","Cytomegalovirus humanbeta5 (10^-8)","Dependoparvovirus primate1 (10^-8)","Erythroparvovirus primate1 (10^-5)","Human erythrovirus V9 (10^-7)","Murine leukemia virus (10^-8)","Murine leukemia-related retroviruses (10^-8)","Gammatorquevirus homidi1 (10^-8)","Gammatorquevirus homidi10 (10^-8)","Gammatorquevirus homidi11 (10^-8)","Gammatorquevirus homidi12 (10^-8)","Gammatorquevirus homidi13 (10^-8)","Gammatorquevirus homidi14 (10^-8)","Gammatorquevirus homidi2 (10^-8)","Gammatorquevirus homidi3 (10^-8)","Gammatorquevirus homidi4 (10^-8)","Gammatorquevirus homidi5 (10^-8)","Gammatorquevirus homidi6 (10^-8)","Gammatorquevirus homidi7 (10^-8)","Gammatorquevirus homidi8 (10^-7)","Gammatorquevirus homidi9 (10^-8)","Hepacivirus hominis (10^-8)","Human immunodeficiency virus (10^-8)","Human immunodeficiency virus 1 (10^-7)","Lymphocryptovirus humangamma4 (10^-7)","Human mastadenovirus C (10^-7)","Human mastadenovirus F (10^-7)","Molluscum contagiosum virus (10^-8)","Hepatitis B virus (10^-4)","Orthorubulavirus mammalis (10^-8)","Rhadinovirus humangamma8 (10^-8)","Roseolovirus humanbeta6a (10^-7)","Roseolovirus humanbeta6b (10^-7)","Roseolovirus humanbeta7 (10^-8)","Simplexvirus humanalpha1 (10^-7)","Anelloviridae sp. (10^-7)","Caudovirus D_HF5_3 (10^-8)","Gokushovirus WZ-2015a (10^-8)","Human endogenous retrovirus K (10^-8)","Human gut gokushovirus (10^-8)","Micro Torque teno virus (10^-8)","Microviridae sp. (10^-7)","Microvirus sp. (10^-8)","Small anellovirus (10^-8)","Torque teno virus (10^-7)","Vesiculovirus indiana (10^-8)"]},"options":{"NodeID":"name","NodeGroup":"name","LinkGroup":null,"colourScale":"d3.scaleOrdinal(d3.schemeCategory20);","fontSize":14,"fontFamily":null,"nodeWidth":25,"nodePadding":10,"units":"","margin":{"top":null,"right":null,"bottom":null,"left":null},"iterations":32,"sinksRight":true}},"evals":[],"jsHooks":[]}</script>
</div>
</div>


<!-- -->

</section></section></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb40" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "Workflow of Mengyi et al. (2023)"</span></span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a><span class="an">subtitle:</span><span class="co"> "Pooled plasma from China (DNA)"</span></span>
<span id="cb40-4"><a href="#cb40-4" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> "Harmon Bhasin"</span></span>
<span id="cb40-5"><a href="#cb40-5" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> 2024-07-23</span></span>
<span id="cb40-6"><a href="#cb40-6" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span></span>
<span id="cb40-7"><a href="#cb40-7" aria-hidden="true" tabindex="-1"></a><span class="co">  html:</span></span>
<span id="cb40-8"><a href="#cb40-8" aria-hidden="true" tabindex="-1"></a><span class="co">    toc: true # table of contents</span></span>
<span id="cb40-9"><a href="#cb40-9" aria-hidden="true" tabindex="-1"></a><span class="co">    toc-title: "Table of contents" # table of contents title</span></span>
<span id="cb40-10"><a href="#cb40-10" aria-hidden="true" tabindex="-1"></a><span class="co">    number-sections: true # number sections</span></span>
<span id="cb40-11"><a href="#cb40-11" aria-hidden="true" tabindex="-1"></a><span class="co">    number-depth: 3 # number depth to show in table of contents</span></span>
<span id="cb40-12"><a href="#cb40-12" aria-hidden="true" tabindex="-1"></a><span class="co">    toc-location: right # table of contents location</span></span>
<span id="cb40-13"><a href="#cb40-13" aria-hidden="true" tabindex="-1"></a><span class="co">    page-layout: full # full page layout</span></span>
<span id="cb40-14"><a href="#cb40-14" aria-hidden="true" tabindex="-1"></a><span class="co">    code-fold: true # Keep option to fold code (i.e. show or hide it; default: hide)</span></span>
<span id="cb40-15"><a href="#cb40-15" aria-hidden="true" tabindex="-1"></a><span class="co">    code-tools: true # Code menu in the header of your document that provides various tools for readers to interact with the source code</span></span>
<span id="cb40-16"><a href="#cb40-16" aria-hidden="true" tabindex="-1"></a><span class="co">    code-link: true # Enables hyper-linking of functions within code blocks to their online documentation</span></span>
<span id="cb40-17"><a href="#cb40-17" aria-hidden="true" tabindex="-1"></a><span class="co">    df-print: paged # print data frame</span></span>
<span id="cb40-18"><a href="#cb40-18" aria-hidden="true" tabindex="-1"></a><span class="co">    fig-format: svg</span></span>
<span id="cb40-19"><a href="#cb40-19" aria-hidden="true" tabindex="-1"></a><span class="co">    other-links:</span></span>
<span id="cb40-20"><a href="#cb40-20" aria-hidden="true" tabindex="-1"></a><span class="co">      - text: Paper</span></span>
<span id="cb40-21"><a href="#cb40-21" aria-hidden="true" tabindex="-1"></a><span class="co">        href: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10372899/</span></span>
<span id="cb40-22"><a href="#cb40-22" aria-hidden="true" tabindex="-1"></a><span class="co">      - text: Data</span></span>
<span id="cb40-23"><a href="#cb40-23" aria-hidden="true" tabindex="-1"></a><span class="co">        href: https://bigd.big.ac.cn/gsa/browse/CRA006191</span></span>
<span id="cb40-24"><a href="#cb40-24" aria-hidden="true" tabindex="-1"></a><span class="co">    code-links:</span></span>
<span id="cb40-25"><a href="#cb40-25" aria-hidden="true" tabindex="-1"></a><span class="co">      - text: Code for this post</span></span>
<span id="cb40-26"><a href="#cb40-26" aria-hidden="true" tabindex="-1"></a><span class="co">        icon: file-code</span></span>
<span id="cb40-27"><a href="#cb40-27" aria-hidden="true" tabindex="-1"></a><span class="co">        href: https://github.com/naobservatory/harmons-public-notebook/blob/main/notebooks/2024-07-23-mengyi.qmd</span></span>
<span id="cb40-28"><a href="#cb40-28" aria-hidden="true" tabindex="-1"></a><span class="an">editor:</span></span>
<span id="cb40-29"><a href="#cb40-29" aria-hidden="true" tabindex="-1"></a><span class="co">  visual: true # visual editor</span></span>
<span id="cb40-30"><a href="#cb40-30" aria-hidden="true" tabindex="-1"></a><span class="co">  render-on-save: true # render on save</span></span>
<span id="cb40-31"><a href="#cb40-31" aria-hidden="true" tabindex="-1"></a><span class="an">comments:</span></span>
<span id="cb40-32"><a href="#cb40-32" aria-hidden="true" tabindex="-1"></a><span class="co">  hypothesis: true # hypothesis</span></span>
<span id="cb40-33"><a href="#cb40-33" aria-hidden="true" tabindex="-1"></a><span class="an">execute:</span><span class="co"> </span></span>
<span id="cb40-34"><a href="#cb40-34" aria-hidden="true" tabindex="-1"></a><span class="co">  freeze: auto</span></span>
<span id="cb40-35"><a href="#cb40-35" aria-hidden="true" tabindex="-1"></a><span class="co">  cache: true</span></span>
<span id="cb40-36"><a href="#cb40-36" aria-hidden="true" tabindex="-1"></a><span class="an">title-block-banner:</span><span class="co"> "#de2d26"</span></span>
<span id="cb40-37"><a href="#cb40-37" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb40-38"><a href="#cb40-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-41"><a href="#cb40-41" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb40-42"><a href="#cb40-42" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: load-packages</span></span>
<span id="cb40-43"><a href="#cb40-43" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb40-44"><a href="#cb40-44" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb40-45"><a href="#cb40-45" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(grDevices)</span>
<span id="cb40-46"><a href="#cb40-46" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(svglite)</span>
<span id="cb40-47"><a href="#cb40-47" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cowplot)</span>
<span id="cb40-48"><a href="#cb40-48" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb40-49"><a href="#cb40-49" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(fastqcr)</span>
<span id="cb40-50"><a href="#cb40-50" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)</span>
<span id="cb40-51"><a href="#cb40-51" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(networkD3)</span>
<span id="cb40-52"><a href="#cb40-52" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb40-53"><a href="#cb40-53" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(htmlwidgets)</span>
<span id="cb40-54"><a href="#cb40-54" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggbeeswarm)</span>
<span id="cb40-55"><a href="#cb40-55" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(latex2exp)</span>
<span id="cb40-56"><a href="#cb40-56" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(forcats)</span>
<span id="cb40-57"><a href="#cb40-57" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scales)</span>
<span id="cb40-58"><a href="#cb40-58" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb40-59"><a href="#cb40-59" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggbeeswarm)</span>
<span id="cb40-60"><a href="#cb40-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-61"><a href="#cb40-61" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"/Users/harmonbhasin/work/securebio/sampling-strategies/scripts/aux_plot-theme.R"</span>)</span>
<span id="cb40-62"><a href="#cb40-62" aria-hidden="true" tabindex="-1"></a>theme_base <span class="ot">&lt;-</span> theme_base <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio =</span> <span class="cn">NULL</span>)</span>
<span id="cb40-63"><a href="#cb40-63" aria-hidden="true" tabindex="-1"></a>theme_kit <span class="ot">&lt;-</span> theme_base <span class="sc">+</span> <span class="fu">theme</span>(</span>
<span id="cb40-64"><a href="#cb40-64" aria-hidden="true" tabindex="-1"></a>  <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">angle =</span> <span class="dv">45</span>),</span>
<span id="cb40-65"><a href="#cb40-65" aria-hidden="true" tabindex="-1"></a>  <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb40-66"><a href="#cb40-66" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb40-67"><a href="#cb40-67" aria-hidden="true" tabindex="-1"></a>tnl <span class="ot">&lt;-</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb40-68"><a href="#cb40-68" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-69"><a href="#cb40-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-72"><a href="#cb40-72" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb40-73"><a href="#cb40-73" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb40-74"><a href="#cb40-74" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: set-up-paths</span></span>
<span id="cb40-75"><a href="#cb40-75" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb40-76"><a href="#cb40-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-77"><a href="#cb40-77" aria-hidden="true" tabindex="-1"></a><span class="co"># Data input paths</span></span>
<span id="cb40-78"><a href="#cb40-78" aria-hidden="true" tabindex="-1"></a>data_dir <span class="ot">&lt;-</span> <span class="st">"/Users/harmonbhasin/work/securebio/nao-harmon/mengyi2023/analysis/"</span></span>
<span id="cb40-79"><a href="#cb40-79" aria-hidden="true" tabindex="-1"></a>input_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"input"</span>)</span>
<span id="cb40-80"><a href="#cb40-80" aria-hidden="true" tabindex="-1"></a>results_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"results"</span>)</span>
<span id="cb40-81"><a href="#cb40-81" aria-hidden="true" tabindex="-1"></a>qc_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(results_dir, <span class="st">"qc"</span>)</span>
<span id="cb40-82"><a href="#cb40-82" aria-hidden="true" tabindex="-1"></a>hv_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(results_dir, <span class="st">"hv"</span>)</span>
<span id="cb40-83"><a href="#cb40-83" aria-hidden="true" tabindex="-1"></a>libraries_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(input_dir, <span class="st">"libraries.csv"</span>)</span>
<span id="cb40-84"><a href="#cb40-84" aria-hidden="true" tabindex="-1"></a>basic_stats_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(qc_dir, <span class="st">"qc_basic_stats.tsv.gz"</span>)</span>
<span id="cb40-85"><a href="#cb40-85" aria-hidden="true" tabindex="-1"></a>adapter_stats_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(qc_dir, <span class="st">"qc_adapter_stats.tsv.gz"</span>)</span>
<span id="cb40-86"><a href="#cb40-86" aria-hidden="true" tabindex="-1"></a>quality_base_stats_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(qc_dir, <span class="st">"qc_quality_base_stats.tsv.gz"</span>)</span>
<span id="cb40-87"><a href="#cb40-87" aria-hidden="true" tabindex="-1"></a>quality_seq_stats_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(qc_dir, <span class="st">"qc_quality_sequence_stats.tsv.gz"</span>)</span>
<span id="cb40-88"><a href="#cb40-88" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-89"><a href="#cb40-89" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-90"><a href="#cb40-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-91"><a href="#cb40-91" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-92"><a href="#cb40-92" aria-hidden="true" tabindex="-1"></a>This one of the studies that we hope to discuss in <span class="co">[</span><span class="ot">our third blog post</span><span class="co">](https://naobservatory.org/blog/exploring-blood-biosurveillance-part1)</span>, which will cover the metagenomic analysis of whole blood/plasma. In this notebook, I analyze <span class="co">[</span><span class="ot">Mengyi 2023</span><span class="co">](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10372899/)</span>, a dataset from China with 201 pools of **plasma**, where each pool contains 160 samples (we don't know whether 1 sample = 1 donation) from 7 different locations, for a total of 10,720 samples. </span>
<span id="cb40-93"><a href="#cb40-93" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-94"><a href="#cb40-94" aria-hidden="true" tabindex="-1"></a>*I'd like to thank Lenni for giving me feedback on the notebook, Will for providing me with a boilerplate rmarkdown file, and Simon for giving me feedback on my figures. This notebook uses the MGS Workflow v2.2.1 (note that this is old).*</span>
<span id="cb40-95"><a href="#cb40-95" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-96"><a href="#cb40-96" aria-hidden="true" tabindex="-1"></a><span class="fu"># Raw data</span></span>
<span id="cb40-97"><a href="#cb40-97" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-98"><a href="#cb40-98" aria-hidden="true" tabindex="-1"></a><span class="fu">## About </span></span>
<span id="cb40-99"><a href="#cb40-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-100"><a href="#cb40-100" aria-hidden="true" tabindex="-1"></a><span class="co">[</span><span class="ot">This dataset</span><span class="co">](https://bigd.big.ac.cn/gsa/browse/CRA006191)</span> from China has 201 pools of plasma, where each pool contains 160 samples from 7 different locations between 2012-2018, for a total of 10,720 samples. This paper did not discuss the number of individuals that contributed to the samples, but we're going to attempt to contact the authors to get this information (this shouldn't hold up any further analysis, but would be good to have this information). They did DNA-sequencing for each pool, with Illumina HiSeq 4500, producing 2x150 bp reads.</span>
<span id="cb40-101"><a href="#cb40-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-104"><a href="#cb40-104" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb40-105"><a href="#cb40-105" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: load-libraries</span></span>
<span id="cb40-106"><a href="#cb40-106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-107"><a href="#cb40-107" aria-hidden="true" tabindex="-1"></a><span class="co"># Import libraries and extract metadata from sample names</span></span>
<span id="cb40-108"><a href="#cb40-108" aria-hidden="true" tabindex="-1"></a>libraries <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(libraries_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb40-109"><a href="#cb40-109" aria-hidden="true" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="fu">sprintf</span>(<span class="st">'%s/metadata.tsv'</span>, data_dir), <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-110"><a href="#cb40-110" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="st">'sample'</span><span class="ot">=</span><span class="st">`</span><span class="at">Library</span><span class="st">`</span>, <span class="st">`</span><span class="at">library</span><span class="st">`</span><span class="ot">=</span><span class="st">`</span><span class="at">Sample title</span><span class="st">`</span>, <span class="st">'Geographic location'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-111"><a href="#cb40-111" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">'location'</span> <span class="ot">=</span> <span class="fu">str_split</span>(<span class="st">`</span><span class="at">Geographic location</span><span class="st">`</span>, <span class="st">":"</span>, <span class="at">simplify =</span> <span class="cn">TRUE</span>)[, <span class="fu">ncol</span>(<span class="fu">str_split</span>(<span class="st">`</span><span class="at">Geographic location</span><span class="st">`</span>, <span class="st">":"</span>, <span class="at">simplify =</span> <span class="cn">TRUE</span>))])</span>
<span id="cb40-112"><a href="#cb40-112" aria-hidden="true" tabindex="-1"></a>libraries <span class="ot">&lt;-</span> libraries <span class="sc">%&gt;%</span></span>
<span id="cb40-113"><a href="#cb40-113" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(metadata, <span class="at">by =</span> <span class="st">'sample'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-114"><a href="#cb40-114" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, <span class="at">library=</span>library.y, location) <span class="sc">%&gt;%</span></span>
<span id="cb40-115"><a href="#cb40-115" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">library =</span> <span class="fu">factor</span>(library))</span>
<span id="cb40-116"><a href="#cb40-116" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-117"><a href="#cb40-117" aria-hidden="true" tabindex="-1"></a>libraries <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(location) <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(<span class="at">n_pools =</span> <span class="fu">n</span>())</span>
<span id="cb40-118"><a href="#cb40-118" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-119"><a href="#cb40-119" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-120"><a href="#cb40-120" aria-hidden="true" tabindex="-1"></a><span class="fu">### Sample + library preparation</span></span>
<span id="cb40-121"><a href="#cb40-121" aria-hidden="true" tabindex="-1"></a>The following excerpt from the paper describes the sample and library preparation process. It's crucial to note that while blood samples were initially collected from volunteers, these samples were converted to plasma through ultracentrifugation prior to sequencing:</span>
<span id="cb40-122"><a href="#cb40-122" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-123"><a href="#cb40-123" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; From January 1, 2012, to December 31, 2018, a total of 10,720 blood samples of 10 ml each were randomly selected from voluntary blood donors in 7 regions. The blood samples taken from various places were mixed in units of 160 (each 100 μl) for ultracentrifugation (32,000 rpm, 120 min, maximum centrifugal radius of 91.9 mm). Afterward, we rinsed and resuspended the precipitate with 500 μl PBS.</span></span>
<span id="cb40-124"><a href="#cb40-124" aria-hidden="true" tabindex="-1"></a><span class="at">&gt;</span></span>
<span id="cb40-125"><a href="#cb40-125" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; The pooled suspensions were subjected to extraction of total DNA using QIAamp® DNA Blood mini Kit (QIAGEN Cat. NO.160019269, Frankfurt, Germany), DNA concentration was measured by Equalbit® 1 × dsDNA HS Assay Kit (Vazyme Cat. NO. 7E302K9, Nanjing, China).</span></span>
<span id="cb40-126"><a href="#cb40-126" aria-hidden="true" tabindex="-1"></a><span class="at">&gt;</span></span>
<span id="cb40-127"><a href="#cb40-127" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; The metagenomic library was constructed using KAPA HyperPlus Kit (KAPA Cat. NO. 0000097583, Boston, USA) with dual-indexed Adapters (KAPA Cat. NO. 0000093370, Boston, USA), the DNA was fragmented to 250 bp approximately by the enzyme at 37 °C for 20 min, after end repair and A-tailing, adapter ligation, post-ligation cleanup, library amplification, and post-amplification cleanup, the library was constructed.</span></span>
<span id="cb40-128"><a href="#cb40-128" aria-hidden="true" tabindex="-1"></a><span class="at">&gt;</span></span>
<span id="cb40-129"><a href="#cb40-129" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; Agilent 2100 Bioanalyzer (Agilent Technologies, Beijing, China) was used for library quality control, and qualified DNA library was sent to the Novogene company to sequence in HiSeq 4500.</span></span>
<span id="cb40-130"><a href="#cb40-130" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-131"><a href="#cb40-131" aria-hidden="true" tabindex="-1"></a><span class="fu">## Quality control metrics</span></span>
<span id="cb40-132"><a href="#cb40-132" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-135"><a href="#cb40-135" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb40-136"><a href="#cb40-136" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb40-137"><a href="#cb40-137" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: read-qc-data</span></span>
<span id="cb40-138"><a href="#cb40-138" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb40-139"><a href="#cb40-139" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-140"><a href="#cb40-140" aria-hidden="true" tabindex="-1"></a><span class="co"># Import QC data</span></span>
<span id="cb40-141"><a href="#cb40-141" aria-hidden="true" tabindex="-1"></a>stages <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"raw_concat"</span>, <span class="st">"cleaned"</span>, <span class="st">"dedup"</span>, <span class="st">"ribo_initial"</span>, <span class="st">"ribo_secondary"</span>)</span>
<span id="cb40-142"><a href="#cb40-142" aria-hidden="true" tabindex="-1"></a>basic_stats <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(basic_stats_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-143"><a href="#cb40-143" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb40-144"><a href="#cb40-144" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">factor</span>(stage, <span class="at">levels =</span> stages),</span>
<span id="cb40-145"><a href="#cb40-145" aria-hidden="true" tabindex="-1"></a>         <span class="at">sample =</span> <span class="fu">fct_inorder</span>(sample))</span>
<span id="cb40-146"><a href="#cb40-146" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-147"><a href="#cb40-147" aria-hidden="true" tabindex="-1"></a>adapter_stats <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(adapter_stats_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-148"><a href="#cb40-148" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb40-149"><a href="#cb40-149" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">factor</span>(stage, <span class="at">levels =</span> stages),</span>
<span id="cb40-150"><a href="#cb40-150" aria-hidden="true" tabindex="-1"></a>         <span class="at">read_pair =</span> <span class="fu">fct_inorder</span>(<span class="fu">as.character</span>(read_pair)))</span>
<span id="cb40-151"><a href="#cb40-151" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-152"><a href="#cb40-152" aria-hidden="true" tabindex="-1"></a>quality_base_stats <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(quality_base_stats_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-153"><a href="#cb40-153" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb40-154"><a href="#cb40-154" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">factor</span>(stage, <span class="at">levels =</span> stages),</span>
<span id="cb40-155"><a href="#cb40-155" aria-hidden="true" tabindex="-1"></a>         <span class="at">read_pair =</span> <span class="fu">fct_inorder</span>(<span class="fu">as.character</span>(read_pair)))</span>
<span id="cb40-156"><a href="#cb40-156" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb40-157"><a href="#cb40-157" aria-hidden="true" tabindex="-1"></a>quality_seq_stats <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(quality_seq_stats_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-158"><a href="#cb40-158" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb40-159"><a href="#cb40-159" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">factor</span>(stage, <span class="at">levels =</span> stages),</span>
<span id="cb40-160"><a href="#cb40-160" aria-hidden="true" tabindex="-1"></a>         <span class="at">read_pair =</span> <span class="fu">fct_inorder</span>(<span class="fu">as.character</span>(read_pair)))</span>
<span id="cb40-161"><a href="#cb40-161" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb40-162"><a href="#cb40-162" aria-hidden="true" tabindex="-1"></a> <span class="co"># Filter to raw data</span></span>
<span id="cb40-163"><a href="#cb40-163" aria-hidden="true" tabindex="-1"></a> basic_stats_raw <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"raw_concat"</span>)</span>
<span id="cb40-164"><a href="#cb40-164" aria-hidden="true" tabindex="-1"></a> adapter_stats_raw <span class="ot">&lt;-</span> adapter_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"raw_concat"</span>)</span>
<span id="cb40-165"><a href="#cb40-165" aria-hidden="true" tabindex="-1"></a> quality_base_stats_raw <span class="ot">&lt;-</span> quality_base_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"raw_concat"</span>)</span>
<span id="cb40-166"><a href="#cb40-166" aria-hidden="true" tabindex="-1"></a> quality_seq_stats_raw <span class="ot">&lt;-</span> quality_seq_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"raw_concat"</span>)</span>
<span id="cb40-167"><a href="#cb40-167" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb40-168"><a href="#cb40-168" aria-hidden="true" tabindex="-1"></a> <span class="co"># Get key values for readout</span></span>
<span id="cb40-169"><a href="#cb40-169" aria-hidden="true" tabindex="-1"></a>raw_read_counts <span class="ot">&lt;-</span> basic_stats_raw <span class="sc">%&gt;%</span> </span>
<span id="cb40-170"><a href="#cb40-170" aria-hidden="true" tabindex="-1"></a>   <span class="fu">summarize</span>(<span class="at">rmin =</span> <span class="fu">min</span>(n_read_pairs), <span class="at">rmax=</span><span class="fu">max</span>(n_read_pairs),</span>
<span id="cb40-171"><a href="#cb40-171" aria-hidden="true" tabindex="-1"></a>             <span class="at">rmean=</span><span class="fu">mean</span>(n_read_pairs), </span>
<span id="cb40-172"><a href="#cb40-172" aria-hidden="true" tabindex="-1"></a>             <span class="at">rtot =</span> <span class="fu">sum</span>(n_read_pairs),</span>
<span id="cb40-173"><a href="#cb40-173" aria-hidden="true" tabindex="-1"></a>             <span class="at">btot =</span> <span class="fu">sum</span>(n_bases_approx),</span>
<span id="cb40-174"><a href="#cb40-174" aria-hidden="true" tabindex="-1"></a>             <span class="at">dmin =</span> <span class="fu">min</span>(percent_duplicates), <span class="at">dmax=</span><span class="fu">max</span>(percent_duplicates),</span>
<span id="cb40-175"><a href="#cb40-175" aria-hidden="true" tabindex="-1"></a>             <span class="at">dmean=</span><span class="fu">mean</span>(percent_duplicates), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb40-176"><a href="#cb40-176" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-177"><a href="#cb40-177" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-178"><a href="#cb40-178" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-179"><a href="#cb40-179" aria-hidden="true" tabindex="-1"></a>These 201 samples contained 3.4B read pairs. The samples had 8.2M - 30.1M (mean 17.1M) read pairs each. The number of read pairs and total bases look relatively evenly distributed across the locations. The duplication rate is also quite low, around ~10%. Adapter content is low. As we'd expect, we see higher quality Q scores near the beginning of the read and a gradual decline towards the end of the read, however all positions seem to have a Q score of 35 which means that our reads are \~99.97% accurate. When looking at the Q score over all sequences, we can see a sharp peak after 35, which closely follows the previous plot, indicating high read quality.</span>
<span id="cb40-180"><a href="#cb40-180" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-183"><a href="#cb40-183" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb40-184"><a href="#cb40-184" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 15</span></span>
<span id="cb40-185"><a href="#cb40-185" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 10</span></span>
<span id="cb40-186"><a href="#cb40-186" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb40-187"><a href="#cb40-187" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-basic-stats</span></span>
<span id="cb40-188"><a href="#cb40-188" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-189"><a href="#cb40-189" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare data</span></span>
<span id="cb40-190"><a href="#cb40-190" aria-hidden="true" tabindex="-1"></a>basic_stats_raw_metrics <span class="ot">&lt;-</span> basic_stats_raw <span class="sc">%&gt;%</span></span>
<span id="cb40-191"><a href="#cb40-191" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(library,</span>
<span id="cb40-192"><a href="#cb40-192" aria-hidden="true" tabindex="-1"></a>         location,</span>
<span id="cb40-193"><a href="#cb40-193" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at"># Read pairs</span><span class="st">`</span> <span class="ot">=</span> n_read_pairs,</span>
<span id="cb40-194"><a href="#cb40-194" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Total base pairs</span><span class="sc">\n</span><span class="at">(approx)</span><span class="st">`</span> <span class="ot">=</span> n_bases_approx,</span>
<span id="cb40-195"><a href="#cb40-195" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">% Duplicates</span><span class="sc">\n</span><span class="at">(FASTQC)</span><span class="st">`</span> <span class="ot">=</span> percent_duplicates) <span class="sc">%&gt;%</span></span>
<span id="cb40-196"><a href="#cb40-196" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>(library<span class="sc">:</span>location), <span class="at">names_to =</span> <span class="st">"metric"</span>, <span class="at">values_to =</span> <span class="st">"value"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-197"><a href="#cb40-197" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">metric =</span> <span class="fu">fct_inorder</span>(metric)) </span>
<span id="cb40-198"><a href="#cb40-198" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-199"><a href="#cb40-199" aria-hidden="true" tabindex="-1"></a><span class="co"># Set up plot templates</span></span>
<span id="cb40-200"><a href="#cb40-200" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-201"><a href="#cb40-201" aria-hidden="true" tabindex="-1"></a>g_basic <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(basic_stats_raw_metrics, <span class="fu">aes</span>(<span class="at">x =</span> library, <span class="at">y =</span> value, <span class="at">fill=</span>location)) <span class="sc">+</span></span>
<span id="cb40-202"><a href="#cb40-202" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position =</span> <span class="st">"dodge"</span>) <span class="sc">+</span></span>
<span id="cb40-203"><a href="#cb40-203" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>() <span class="sc">+</span></span>
<span id="cb40-204"><a href="#cb40-204" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Dark2"</span>) <span class="sc">+</span></span>
<span id="cb40-205"><a href="#cb40-205" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb40-206"><a href="#cb40-206" aria-hidden="true" tabindex="-1"></a>  <span class="fu">expand_limits</span>(<span class="at">y =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">100</span>)) <span class="sc">+</span></span>
<span id="cb40-207"><a href="#cb40-207" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(metric <span class="sc">~</span> ., <span class="at">scales =</span> <span class="st">"free"</span>, <span class="at">space =</span> <span class="st">"free_x"</span>, <span class="at">switch =</span> <span class="st">"y"</span>) <span class="sc">+</span></span>
<span id="cb40-208"><a href="#cb40-208" aria-hidden="true" tabindex="-1"></a>  theme_kit <span class="sc">+</span> <span class="fu">theme</span>(</span>
<span id="cb40-209"><a href="#cb40-209" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb40-210"><a href="#cb40-210" aria-hidden="true" tabindex="-1"></a>    <span class="at">strip.text.y =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"plain"</span>),</span>
<span id="cb40-211"><a href="#cb40-211" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>()</span>
<span id="cb40-212"><a href="#cb40-212" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb40-213"><a href="#cb40-213" aria-hidden="true" tabindex="-1"></a>g_basic</span>
<span id="cb40-214"><a href="#cb40-214" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-215"><a href="#cb40-215" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-216"><a href="#cb40-216" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-219"><a href="#cb40-219" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb40-220"><a href="#cb40-220" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-raw-quality</span></span>
<span id="cb40-221"><a href="#cb40-221" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 15</span></span>
<span id="cb40-222"><a href="#cb40-222" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 10</span></span>
<span id="cb40-223"><a href="#cb40-223" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-224"><a href="#cb40-224" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-225"><a href="#cb40-225" aria-hidden="true" tabindex="-1"></a><span class="co"># Set up plotting templates</span></span>
<span id="cb40-226"><a href="#cb40-226" aria-hidden="true" tabindex="-1"></a>g_qual_raw <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">color =</span> location, <span class="at">linetype=</span>read_pair, <span class="at">group=</span><span class="fu">interaction</span>(sample,read_pair))) <span class="sc">+</span> </span>
<span id="cb40-227"><a href="#cb40-227" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_linetype_discrete</span>(<span class="at">name =</span> <span class="st">"Read Pair"</span>) <span class="sc">+</span></span>
<span id="cb40-228"><a href="#cb40-228" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_brewer</span>(<span class="at">palette =</span> <span class="st">"Dark2"</span>) <span class="sc">+</span></span>
<span id="cb40-229"><a href="#cb40-229" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color=</span><span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>),</span>
<span id="cb40-230"><a href="#cb40-230" aria-hidden="true" tabindex="-1"></a>         <span class="at">linetype =</span> <span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>)) <span class="sc">+</span></span>
<span id="cb40-231"><a href="#cb40-231" aria-hidden="true" tabindex="-1"></a>  theme_base</span>
<span id="cb40-232"><a href="#cb40-232" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-233"><a href="#cb40-233" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize adapters</span></span>
<span id="cb40-234"><a href="#cb40-234" aria-hidden="true" tabindex="-1"></a>g_adapters_raw <span class="ot">&lt;-</span> g_qual_raw <span class="sc">+</span> </span>
<span id="cb40-235"><a href="#cb40-235" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>pc_adapters), <span class="at">data=</span>adapter_stats_raw) <span class="sc">+</span></span>
<span id="cb40-236"><a href="#cb40-236" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Adapters"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">5</span>),</span>
<span id="cb40-237"><a href="#cb40-237" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb40-238"><a href="#cb40-238" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Position"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb40-239"><a href="#cb40-239" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">500</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb40-240"><a href="#cb40-240" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(.<span class="sc">~</span>adapter)</span>
<span id="cb40-241"><a href="#cb40-241" aria-hidden="true" tabindex="-1"></a>g_adapters_raw</span>
<span id="cb40-242"><a href="#cb40-242" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-243"><a href="#cb40-243" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize quality</span></span>
<span id="cb40-244"><a href="#cb40-244" aria-hidden="true" tabindex="-1"></a>g_quality_base_raw <span class="ot">&lt;-</span> g_qual_raw <span class="sc">+</span></span>
<span id="cb40-245"><a href="#cb40-245" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb40-246"><a href="#cb40-246" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb40-247"><a href="#cb40-247" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>mean_phred_score), <span class="at">data=</span>quality_base_stats_raw) <span class="sc">+</span></span>
<span id="cb40-248"><a href="#cb40-248" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"Mean Phred score"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">10</span>,<span class="dv">45</span>)) <span class="sc">+</span></span>
<span id="cb40-249"><a href="#cb40-249" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Position"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb40-250"><a href="#cb40-250" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">500</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb40-251"><a href="#cb40-251" aria-hidden="true" tabindex="-1"></a>g_quality_base_raw</span>
<span id="cb40-252"><a href="#cb40-252" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-253"><a href="#cb40-253" aria-hidden="true" tabindex="-1"></a>g_quality_seq_raw <span class="ot">&lt;-</span> g_qual_raw <span class="sc">+</span></span>
<span id="cb40-254"><a href="#cb40-254" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb40-255"><a href="#cb40-255" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb40-256"><a href="#cb40-256" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>mean_phred_score, <span class="at">y=</span>n_sequences), <span class="at">data=</span>quality_seq_stats_raw) <span class="sc">+</span></span>
<span id="cb40-257"><a href="#cb40-257" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Mean Phred score"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb40-258"><a href="#cb40-258" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"# Sequences"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb40-259"><a href="#cb40-259" aria-hidden="true" tabindex="-1"></a>g_quality_seq_raw</span>
<span id="cb40-260"><a href="#cb40-260" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-261"><a href="#cb40-261" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-262"><a href="#cb40-262" aria-hidden="true" tabindex="-1"></a><span class="fu"># Preprocessing</span></span>
<span id="cb40-263"><a href="#cb40-263" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-264"><a href="#cb40-264" aria-hidden="true" tabindex="-1"></a><span class="fu">## Summary</span></span>
<span id="cb40-265"><a href="#cb40-265" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-266"><a href="#cb40-266" aria-hidden="true" tabindex="-1"></a>The average fraction of reads at each stage in the preprocessing pipeline is shown in the following table. Reads lost during trimming and filtering approximately matches what we'd expect based on the raw adapter content. Deduplication loses us about 10% of reads which matched the amount of estimated duplicated reads by QC. Low ribodepletion is observed which makes sense because they only sequenced DNA.</span>
<span id="cb40-267"><a href="#cb40-267" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-270"><a href="#cb40-270" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb40-271"><a href="#cb40-271" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: preproc-table</span></span>
<span id="cb40-272"><a href="#cb40-272" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-273"><a href="#cb40-273" aria-hidden="true" tabindex="-1"></a><span class="co"># </span><span class="al">TODO</span><span class="co">: Group by pool size as well</span></span>
<span id="cb40-274"><a href="#cb40-274" aria-hidden="true" tabindex="-1"></a><span class="co"># Count read losses</span></span>
<span id="cb40-275"><a href="#cb40-275" aria-hidden="true" tabindex="-1"></a>n_reads_rel <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> </span>
<span id="cb40-276"><a href="#cb40-276" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample,location, stage, percent_duplicates, n_read_pairs) <span class="sc">%&gt;%</span></span>
<span id="cb40-277"><a href="#cb40-277" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb40-278"><a href="#cb40-278" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(sample, stage) <span class="sc">%&gt;%</span></span>
<span id="cb40-279"><a href="#cb40-279" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_retained =</span> n_read_pairs <span class="sc">/</span> <span class="fu">lag</span>(n_read_pairs),</span>
<span id="cb40-280"><a href="#cb40-280" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_lost =</span> <span class="dv">1</span> <span class="sc">-</span> p_reads_retained,</span>
<span id="cb40-281"><a href="#cb40-281" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_retained_abs =</span> n_read_pairs <span class="sc">/</span> n_read_pairs[<span class="dv">1</span>],</span>
<span id="cb40-282"><a href="#cb40-282" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_lost_abs =</span> <span class="dv">1</span><span class="sc">-</span>p_reads_retained_abs,</span>
<span id="cb40-283"><a href="#cb40-283" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_lost_abs_marginal =</span> p_reads_lost_abs <span class="sc">-</span> <span class="fu">lag</span>(p_reads_lost_abs))</span>
<span id="cb40-284"><a href="#cb40-284" aria-hidden="true" tabindex="-1"></a>n_reads_rel_display <span class="ot">&lt;-</span> n_reads_rel <span class="sc">%&gt;%</span> </span>
<span id="cb40-285"><a href="#cb40-285" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">Stage=</span>stage) <span class="sc">%&gt;%</span> </span>
<span id="cb40-286"><a href="#cb40-286" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Stage) <span class="sc">%&gt;%</span> </span>
<span id="cb40-287"><a href="#cb40-287" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="st">`</span><span class="at">% Total Reads Lost (Cumulative)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">round</span>(<span class="fu">min</span>(p_reads_lost_abs<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">"-"</span>, <span class="fu">round</span>(<span class="fu">max</span>(p_reads_lost_abs<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">" (mean "</span>, <span class="fu">round</span>(<span class="fu">mean</span>(p_reads_lost_abs<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">")"</span>),</span>
<span id="cb40-288"><a href="#cb40-288" aria-hidden="true" tabindex="-1"></a>            <span class="st">`</span><span class="at">% Total Reads Lost (Marginal)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">round</span>(<span class="fu">min</span>(p_reads_lost_abs_marginal<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">"-"</span>, <span class="fu">round</span>(<span class="fu">max</span>(p_reads_lost_abs_marginal<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">" (mean "</span>, <span class="fu">round</span>(<span class="fu">mean</span>(p_reads_lost_abs_marginal<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">")"</span>), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb40-289"><a href="#cb40-289" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Stage <span class="sc">!=</span> <span class="st">"raw_concat"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-290"><a href="#cb40-290" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Stage =</span> Stage <span class="sc">%&gt;%</span> as.numeric <span class="sc">%&gt;%</span> <span class="fu">factor</span>(<span class="at">labels=</span><span class="fu">c</span>(<span class="st">"Trimming &amp; filtering"</span>, <span class="st">"Deduplication"</span>, <span class="st">"Initial ribodepletion"</span>, <span class="st">"Secondary ribodepletion"</span>)))</span>
<span id="cb40-291"><a href="#cb40-291" aria-hidden="true" tabindex="-1"></a>n_reads_rel_display</span>
<span id="cb40-292"><a href="#cb40-292" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-293"><a href="#cb40-293" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-294"><a href="#cb40-294" aria-hidden="true" tabindex="-1"></a><span class="fu">## Quality control metrics</span></span>
<span id="cb40-295"><a href="#cb40-295" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-296"><a href="#cb40-296" aria-hidden="true" tabindex="-1"></a>Trimming and cleaning gets rid of the Illumina adapter as well as polyg, and we see a decrease in polya. Q score remain the same during read cleaning when looking at the positions, with the end of the read actually improving in score. Q scores across all sequences look pretty much the same throughout cleaning.</span>
<span id="cb40-297"><a href="#cb40-297" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-300"><a href="#cb40-300" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb40-301"><a href="#cb40-301" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb40-302"><a href="#cb40-302" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-quality</span></span>
<span id="cb40-303"><a href="#cb40-303" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 15</span></span>
<span id="cb40-304"><a href="#cb40-304" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 10</span></span>
<span id="cb40-305"><a href="#cb40-305" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-306"><a href="#cb40-306" aria-hidden="true" tabindex="-1"></a>g_qual <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">color =</span> location, <span class="at">linetype=</span>read_pair, <span class="at">group=</span><span class="fu">interaction</span>(sample,read_pair))) <span class="sc">+</span> </span>
<span id="cb40-307"><a href="#cb40-307" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_linetype_discrete</span>(<span class="at">name =</span> <span class="st">"Read Pair"</span>) <span class="sc">+</span></span>
<span id="cb40-308"><a href="#cb40-308" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_brewer</span>(<span class="at">palette =</span> <span class="st">"Dark2"</span>) <span class="sc">+</span></span>
<span id="cb40-309"><a href="#cb40-309" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color=</span><span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>),</span>
<span id="cb40-310"><a href="#cb40-310" aria-hidden="true" tabindex="-1"></a>         <span class="at">linetype =</span> <span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>)) <span class="sc">+</span></span>
<span id="cb40-311"><a href="#cb40-311" aria-hidden="true" tabindex="-1"></a>  theme_base</span>
<span id="cb40-312"><a href="#cb40-312" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-313"><a href="#cb40-313" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize adapters</span></span>
<span id="cb40-314"><a href="#cb40-314" aria-hidden="true" tabindex="-1"></a>g_adapters <span class="ot">&lt;-</span> g_qual <span class="sc">+</span> </span>
<span id="cb40-315"><a href="#cb40-315" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>pc_adapters), <span class="at">data=</span>adapter_stats) <span class="sc">+</span></span>
<span id="cb40-316"><a href="#cb40-316" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Adapters"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">5</span>),</span>
<span id="cb40-317"><a href="#cb40-317" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb40-318"><a href="#cb40-318" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Position"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb40-319"><a href="#cb40-319" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">140</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb40-320"><a href="#cb40-320" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(stage<span class="sc">~</span>adapter)</span>
<span id="cb40-321"><a href="#cb40-321" aria-hidden="true" tabindex="-1"></a>g_adapters</span>
<span id="cb40-322"><a href="#cb40-322" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-323"><a href="#cb40-323" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize quality</span></span>
<span id="cb40-324"><a href="#cb40-324" aria-hidden="true" tabindex="-1"></a>g_quality_base <span class="ot">&lt;-</span> g_qual <span class="sc">+</span></span>
<span id="cb40-325"><a href="#cb40-325" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb40-326"><a href="#cb40-326" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb40-327"><a href="#cb40-327" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>mean_phred_score), <span class="at">data=</span>quality_base_stats) <span class="sc">+</span></span>
<span id="cb40-328"><a href="#cb40-328" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"Mean Phred score"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">10</span>,<span class="dv">45</span>)) <span class="sc">+</span></span>
<span id="cb40-329"><a href="#cb40-329" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Position"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb40-330"><a href="#cb40-330" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">140</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb40-331"><a href="#cb40-331" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(stage<span class="sc">~</span>.)</span>
<span id="cb40-332"><a href="#cb40-332" aria-hidden="true" tabindex="-1"></a>g_quality_base</span>
<span id="cb40-333"><a href="#cb40-333" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-334"><a href="#cb40-334" aria-hidden="true" tabindex="-1"></a>g_quality_seq <span class="ot">&lt;-</span> g_qual <span class="sc">+</span></span>
<span id="cb40-335"><a href="#cb40-335" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb40-336"><a href="#cb40-336" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb40-337"><a href="#cb40-337" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>mean_phred_score, <span class="at">y=</span>n_sequences), <span class="at">data=</span>quality_seq_stats) <span class="sc">+</span></span>
<span id="cb40-338"><a href="#cb40-338" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Mean Phred score"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb40-339"><a href="#cb40-339" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"# Sequences"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb40-340"><a href="#cb40-340" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(stage<span class="sc">~</span>., <span class="at">scales =</span> <span class="st">"free_y"</span>)</span>
<span id="cb40-341"><a href="#cb40-341" aria-hidden="true" tabindex="-1"></a>g_quality_seq</span>
<span id="cb40-342"><a href="#cb40-342" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-343"><a href="#cb40-343" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-344"><a href="#cb40-344" aria-hidden="true" tabindex="-1"></a>Number of read pairs look reasonable, same with a large portion being lost during deduplication. At the end of deduplication it seems that only 5% of reads are duplicates which doesn't seem too bad. Mean read length stays around 150 bp which is pretty good.</span>
<span id="cb40-345"><a href="#cb40-345" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-348"><a href="#cb40-348" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb40-349"><a href="#cb40-349" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: preproc-figures</span></span>
<span id="cb40-350"><a href="#cb40-350" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb40-351"><a href="#cb40-351" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 15</span></span>
<span id="cb40-352"><a href="#cb40-352" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 10</span></span>
<span id="cb40-353"><a href="#cb40-353" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-354"><a href="#cb40-354" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-355"><a href="#cb40-355" aria-hidden="true" tabindex="-1"></a>g_stage_trace <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(basic_stats, <span class="fu">aes</span>(<span class="at">x=</span>stage, <span class="at">group=</span>sample, <span class="at">color =</span> location)) <span class="sc">+</span></span>
<span id="cb40-356"><a href="#cb40-356" aria-hidden="true" tabindex="-1"></a>  theme_kit <span class="sc">+</span></span>
<span id="cb40-357"><a href="#cb40-357" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_brewer</span>(<span class="at">palette =</span> <span class="st">"Dark2"</span>)</span>
<span id="cb40-358"><a href="#cb40-358" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot reads over preprocessing</span></span>
<span id="cb40-359"><a href="#cb40-359" aria-hidden="true" tabindex="-1"></a>g_reads_stages <span class="ot">&lt;-</span> g_stage_trace <span class="sc">+</span></span>
<span id="cb40-360"><a href="#cb40-360" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>n_read_pairs)) <span class="sc">+</span></span>
<span id="cb40-361"><a href="#cb40-361" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"# Read pairs"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>))</span>
<span id="cb40-362"><a href="#cb40-362" aria-hidden="true" tabindex="-1"></a>g_reads_stages</span>
<span id="cb40-363"><a href="#cb40-363" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-364"><a href="#cb40-364" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot relative read losses during preprocessing</span></span>
<span id="cb40-365"><a href="#cb40-365" aria-hidden="true" tabindex="-1"></a>g_reads_rel <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(n_reads_rel, </span>
<span id="cb40-366"><a href="#cb40-366" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">aes</span>(<span class="at">x=</span>stage, <span class="at">group=</span>sample, <span class="at">color =</span> location)) <span class="sc">+</span></span>
<span id="cb40-367"><a href="#cb40-367" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>p_reads_lost_abs_marginal)) <span class="sc">+</span></span>
<span id="cb40-368"><a href="#cb40-368" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"% Total Reads Lost"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), </span>
<span id="cb40-369"><a href="#cb40-369" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="cf">function</span>(x) x<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb40-370"><a href="#cb40-370" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_brewer</span>(<span class="at">palette =</span> <span class="st">"Dark2"</span>) <span class="sc">+</span></span>
<span id="cb40-371"><a href="#cb40-371" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb40-372"><a href="#cb40-372" aria-hidden="true" tabindex="-1"></a>g_reads_rel</span>
<span id="cb40-373"><a href="#cb40-373" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-374"><a href="#cb40-374" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-377"><a href="#cb40-377" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb40-378"><a href="#cb40-378" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: preproc-dedup</span></span>
<span id="cb40-379"><a href="#cb40-379" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 15</span></span>
<span id="cb40-380"><a href="#cb40-380" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 10</span></span>
<span id="cb40-381"><a href="#cb40-381" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-382"><a href="#cb40-382" aria-hidden="true" tabindex="-1"></a>stage_dup <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(stage) <span class="sc">%&gt;%</span> </span>
<span id="cb40-383"><a href="#cb40-383" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">dmin =</span> <span class="fu">min</span>(percent_duplicates), <span class="at">dmax=</span><span class="fu">max</span>(percent_duplicates),</span>
<span id="cb40-384"><a href="#cb40-384" aria-hidden="true" tabindex="-1"></a>            <span class="at">dmean=</span><span class="fu">mean</span>(percent_duplicates), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb40-385"><a href="#cb40-385" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-386"><a href="#cb40-386" aria-hidden="true" tabindex="-1"></a>g_dup_stages <span class="ot">&lt;-</span> g_stage_trace <span class="sc">+</span></span>
<span id="cb40-387"><a href="#cb40-387" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>percent_duplicates)) <span class="sc">+</span></span>
<span id="cb40-388"><a href="#cb40-388" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"% Duplicates"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb40-389"><a href="#cb40-389" aria-hidden="true" tabindex="-1"></a>g_dup_stages</span>
<span id="cb40-390"><a href="#cb40-390" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-391"><a href="#cb40-391" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-394"><a href="#cb40-394" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb40-395"><a href="#cb40-395" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: preproc-mean-readlen</span></span>
<span id="cb40-396"><a href="#cb40-396" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 15</span></span>
<span id="cb40-397"><a href="#cb40-397" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 10</span></span>
<span id="cb40-398"><a href="#cb40-398" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "Reminder that this data is 2x150 bp."</span></span>
<span id="cb40-399"><a href="#cb40-399" aria-hidden="true" tabindex="-1"></a><span class="co">#| cap-location: margin</span></span>
<span id="cb40-400"><a href="#cb40-400" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-401"><a href="#cb40-401" aria-hidden="true" tabindex="-1"></a>g_readlen_stages <span class="ot">&lt;-</span> g_stage_trace <span class="sc">+</span> <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>mean_seq_len)) <span class="sc">+</span></span>
<span id="cb40-402"><a href="#cb40-402" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"Mean read length (nt)"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>))</span>
<span id="cb40-403"><a href="#cb40-403" aria-hidden="true" tabindex="-1"></a>g_readlen_stages</span>
<span id="cb40-404"><a href="#cb40-404" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-405"><a href="#cb40-405" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-406"><a href="#cb40-406" aria-hidden="true" tabindex="-1"></a><span class="fu"># Taxonomic composition</span></span>
<span id="cb40-407"><a href="#cb40-407" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-408"><a href="#cb40-408" aria-hidden="true" tabindex="-1"></a><span class="fu">## High-level composition</span></span>
<span id="cb40-409"><a href="#cb40-409" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-410"><a href="#cb40-410" aria-hidden="true" tabindex="-1"></a>To assess the high-level composition of the reads, I ran the files through Kraken2 and summarized the results with Bracken.</span>
<span id="cb40-411"><a href="#cb40-411" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-412"><a href="#cb40-412" aria-hidden="true" tabindex="-1"></a>The groups listed below were created by Will:</span>
<span id="cb40-413"><a href="#cb40-413" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-414"><a href="#cb40-414" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Filtered (removed during cleaning)</span>
<span id="cb40-415"><a href="#cb40-415" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Duplicate (removed during deduplication)</span>
<span id="cb40-416"><a href="#cb40-416" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Ribosomal (removed during ribodepletion)</span>
<span id="cb40-417"><a href="#cb40-417" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Unassigned (non-ribosomal reads that were not assigned to any taxon by Kraken/Bracken)</span>
<span id="cb40-418"><a href="#cb40-418" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Bacterial (non-ribosomal reads assigned to the Bacteria domain by Kraken/Bracken)</span>
<span id="cb40-419"><a href="#cb40-419" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Archaeal (non-ribosomal reads assigned to the Archaea domain by Kraken/Bracken)</span>
<span id="cb40-420"><a href="#cb40-420" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Viral (non-ribosomal reads assigned to the Viruses domain by Kraken/Bracken)</span>
<span id="cb40-421"><a href="#cb40-421" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Human (non-ribosomal reads assigned to the Eukarya domain by Kraken/Bracken)</span>
<span id="cb40-422"><a href="#cb40-422" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-423"><a href="#cb40-423" aria-hidden="true" tabindex="-1"></a>Human reads make up a large proportion of all reads at 82% total composition, whereas viruses only account for 0.01% of all reads.</span>
<span id="cb40-424"><a href="#cb40-424" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-427"><a href="#cb40-427" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb40-428"><a href="#cb40-428" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: prepare-composition</span></span>
<span id="cb40-429"><a href="#cb40-429" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb40-430"><a href="#cb40-430" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-431"><a href="#cb40-431" aria-hidden="true" tabindex="-1"></a>classifications <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Filtered"</span>, <span class="st">"Duplicate"</span>, <span class="st">"Ribosomal"</span>, <span class="st">"Unassigned"</span>,</span>
<span id="cb40-432"><a href="#cb40-432" aria-hidden="true" tabindex="-1"></a>                     <span class="st">"Bacterial"</span>, <span class="st">"Archaeal"</span>, <span class="st">"Viral"</span>, <span class="st">"Human"</span>)</span>
<span id="cb40-433"><a href="#cb40-433" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-434"><a href="#cb40-434" aria-hidden="true" tabindex="-1"></a><span class="co"># Import composition data</span></span>
<span id="cb40-435"><a href="#cb40-435" aria-hidden="true" tabindex="-1"></a>tax_final_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(results_dir, <span class="st">"taxonomy_final"</span>)</span>
<span id="cb40-436"><a href="#cb40-436" aria-hidden="true" tabindex="-1"></a>comp <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="fu">sprintf</span>(<span class="st">"%s/taxonomic_composition.tsv.gz"</span>, tax_final_dir)) <span class="sc">%&gt;%</span> </span>
<span id="cb40-437"><a href="#cb40-437" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">'sample'</span>)</span>
<span id="cb40-438"><a href="#cb40-438" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-439"><a href="#cb40-439" aria-hidden="true" tabindex="-1"></a>comp_minor <span class="ot">&lt;-</span> comp <span class="sc">%&gt;%</span> </span>
<span id="cb40-440"><a href="#cb40-440" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(classification <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Archaeal"</span>, <span class="st">"Viral"</span>, <span class="st">"Human"</span>, <span class="st">"Other"</span>))</span>
<span id="cb40-441"><a href="#cb40-441" aria-hidden="true" tabindex="-1"></a>comp_assigned <span class="ot">&lt;-</span> comp <span class="sc">%&gt;%</span></span>
<span id="cb40-442"><a href="#cb40-442" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span> classification <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Filtered"</span>, <span class="st">"Duplicate"</span>, </span>
<span id="cb40-443"><a href="#cb40-443" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">"Ribosomal"</span>, <span class="st">"Unassigned"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb40-444"><a href="#cb40-444" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb40-445"><a href="#cb40-445" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads =</span> n_reads<span class="sc">/</span><span class="fu">sum</span>(n_reads))</span>
<span id="cb40-446"><a href="#cb40-446" aria-hidden="true" tabindex="-1"></a>comp_assigned_minor <span class="ot">&lt;-</span> comp_assigned <span class="sc">%&gt;%</span> </span>
<span id="cb40-447"><a href="#cb40-447" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(classification <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Archaeal"</span>, <span class="st">"Viral"</span>, <span class="st">"Human"</span>, <span class="st">"Other"</span>))</span>
<span id="cb40-448"><a href="#cb40-448" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-449"><a href="#cb40-449" aria-hidden="true" tabindex="-1"></a><span class="co"># Summarize composition</span></span>
<span id="cb40-450"><a href="#cb40-450" aria-hidden="true" tabindex="-1"></a>read_comp_summ <span class="ot">&lt;-</span> comp <span class="sc">%&gt;%</span> </span>
<span id="cb40-451"><a href="#cb40-451" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(classification) <span class="sc">%&gt;%</span></span>
<span id="cb40-452"><a href="#cb40-452" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads =</span> <span class="fu">sum</span>(n_reads), <span class="at">.groups =</span> <span class="st">"drop_last"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-453"><a href="#cb40-453" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">n_reads =</span> <span class="fu">replace_na</span>(n_reads,<span class="dv">0</span>),</span>
<span id="cb40-454"><a href="#cb40-454" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads =</span> n_reads<span class="sc">/</span><span class="fu">sum</span>(n_reads),</span>
<span id="cb40-455"><a href="#cb40-455" aria-hidden="true" tabindex="-1"></a>         <span class="at">pc_reads =</span> p_reads<span class="sc">*</span><span class="dv">100</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-456"><a href="#cb40-456" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">classification =</span> <span class="fu">factor</span>(classification, <span class="at">levels=</span>classifications)) <span class="sc">%&gt;%</span></span>
<span id="cb40-457"><a href="#cb40-457" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(classification, n_reads, pc_reads) <span class="sc">%&gt;%</span></span>
<span id="cb40-458"><a href="#cb40-458" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">`</span><span class="at"># of reads</span><span class="st">`</span> <span class="ot">=</span> n_reads, <span class="st">"% of reads"</span> <span class="ot">=</span> pc_reads) <span class="sc">%&gt;%</span></span>
<span id="cb40-459"><a href="#cb40-459" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">% of reads</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sprintf</span>(<span class="st">"%.2f"</span>, <span class="st">`</span><span class="at">% of reads</span><span class="st">`</span>))</span>
<span id="cb40-460"><a href="#cb40-460" aria-hidden="true" tabindex="-1"></a>read_comp_summ</span>
<span id="cb40-461"><a href="#cb40-461" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-462"><a href="#cb40-462" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-465"><a href="#cb40-465" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb40-466"><a href="#cb40-466" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-composition-all</span></span>
<span id="cb40-467"><a href="#cb40-467" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 15</span></span>
<span id="cb40-468"><a href="#cb40-468" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 10</span></span>
<span id="cb40-469"><a href="#cb40-469" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-470"><a href="#cb40-470" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare plotting templates</span></span>
<span id="cb40-471"><a href="#cb40-471" aria-hidden="true" tabindex="-1"></a>g_comp_base <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">x=</span>library, <span class="at">y=</span>p_reads, <span class="at">fill=</span>classification)) <span class="sc">+</span></span>
<span id="cb40-472"><a href="#cb40-472" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">name=</span><span class="st">"Plasma pool"</span>) <span class="sc">+</span></span>
<span id="cb40-473"><a href="#cb40-473" aria-hidden="true" tabindex="-1"></a>  theme_kit <span class="sc">+</span> </span>
<span id="cb40-474"><a href="#cb40-474" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust=</span><span class="dv">0</span>, <span class="at">face=</span><span class="st">"plain"</span>, <span class="at">size=</span><span class="fu">rel</span>(<span class="fl">1.5</span>)),</span>
<span id="cb40-475"><a href="#cb40-475" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>())</span>
<span id="cb40-476"><a href="#cb40-476" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-477"><a href="#cb40-477" aria-hidden="true" tabindex="-1"></a>scale_y_pc_reads <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">partial</span>(scale_y_continuous, <span class="at">name =</span> <span class="st">"% Reads"</span>,</span>
<span id="cb40-478"><a href="#cb40-478" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>)</span>
<span id="cb40-479"><a href="#cb40-479" aria-hidden="true" tabindex="-1"></a>geom_comp <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">partial</span>(geom_col, <span class="at">position =</span> <span class="st">"stack"</span>, <span class="at">width =</span> <span class="dv">1</span>)</span>
<span id="cb40-480"><a href="#cb40-480" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-481"><a href="#cb40-481" aria-hidden="true" tabindex="-1"></a><span class="co"># Define a color palette for the classification</span></span>
<span id="cb40-482"><a href="#cb40-482" aria-hidden="true" tabindex="-1"></a>classification_colors <span class="ot">&lt;-</span> <span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">"Accent"</span>)</span>
<span id="cb40-483"><a href="#cb40-483" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(classification_colors) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Filtered"</span>, <span class="st">"Duplicate"</span>, <span class="st">"Ribosomal"</span>, <span class="st">"Unassigned"</span>, <span class="st">"Bacterial"</span>, <span class="st">"Archaeal"</span>, <span class="st">"Viral"</span>, <span class="st">"Human"</span>)</span>
<span id="cb40-484"><a href="#cb40-484" aria-hidden="true" tabindex="-1"></a>scale_fill_classification <span class="ot">&lt;-</span> <span class="cf">function</span>() {</span>
<span id="cb40-485"><a href="#cb40-485" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> classification_colors, <span class="at">name =</span> <span class="st">"Classification"</span>)</span>
<span id="cb40-486"><a href="#cb40-486" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb40-487"><a href="#cb40-487" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-488"><a href="#cb40-488" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot overall composition</span></span>
<span id="cb40-489"><a href="#cb40-489" aria-hidden="true" tabindex="-1"></a>g_comp <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> <span class="fu">geom_comp</span>(<span class="at">data =</span> comp) <span class="sc">+</span></span>
<span id="cb40-490"><a href="#cb40-490" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_pc_reads</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>)) <span class="sc">+</span></span>
<span id="cb40-491"><a href="#cb40-491" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_classification</span>() <span class="sc">+</span> </span>
<span id="cb40-492"><a href="#cb40-492" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Read composition (all reads, all groups)"</span>)</span>
<span id="cb40-493"><a href="#cb40-493" aria-hidden="true" tabindex="-1"></a>g_comp</span>
<span id="cb40-494"><a href="#cb40-494" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-495"><a href="#cb40-495" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a faceted plot of composition by location</span></span>
<span id="cb40-496"><a href="#cb40-496" aria-hidden="true" tabindex="-1"></a><span class="co">#g_comp_by_location &lt;- g_comp_base +</span></span>
<span id="cb40-497"><a href="#cb40-497" aria-hidden="true" tabindex="-1"></a><span class="co">#  geom_comp(data = comp) +</span></span>
<span id="cb40-498"><a href="#cb40-498" aria-hidden="true" tabindex="-1"></a><span class="co">#  scale_y_pc_reads(limits = c(0,1.01), breaks = seq(0,1,0.2)) +</span></span>
<span id="cb40-499"><a href="#cb40-499" aria-hidden="true" tabindex="-1"></a><span class="co">#  scale_fill_classification() +</span></span>
<span id="cb40-500"><a href="#cb40-500" aria-hidden="true" tabindex="-1"></a><span class="co">#  facet_wrap(~ location, scales = "free_x", ncol = 2) +</span></span>
<span id="cb40-501"><a href="#cb40-501" aria-hidden="true" tabindex="-1"></a><span class="co">#  theme(axis.text.x = element_blank(),  # Remove x-axis labels</span></span>
<span id="cb40-502"><a href="#cb40-502" aria-hidden="true" tabindex="-1"></a><span class="co">#        axis.ticks.x = element_blank()) +  # Remove x-axis ticks</span></span>
<span id="cb40-503"><a href="#cb40-503" aria-hidden="true" tabindex="-1"></a><span class="co">#  ggtitle("Read composition (all reads, all groups)")</span></span>
<span id="cb40-504"><a href="#cb40-504" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-505"><a href="#cb40-505" aria-hidden="true" tabindex="-1"></a><span class="co"># Display the plot</span></span>
<span id="cb40-506"><a href="#cb40-506" aria-hidden="true" tabindex="-1"></a><span class="co">#g_comp_by_location</span></span>
<span id="cb40-507"><a href="#cb40-507" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-508"><a href="#cb40-508" aria-hidden="true" tabindex="-1"></a><span class="co"># Repeat for classified reads only</span></span>
<span id="cb40-509"><a href="#cb40-509" aria-hidden="true" tabindex="-1"></a>g_comp_assigned <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb40-510"><a href="#cb40-510" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_comp</span>(<span class="at">data =</span> comp_assigned) <span class="sc">+</span></span>
<span id="cb40-511"><a href="#cb40-511" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_pc_reads</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>)) <span class="sc">+</span></span>
<span id="cb40-512"><a href="#cb40-512" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_classification</span>() <span class="sc">+</span> </span>
<span id="cb40-513"><a href="#cb40-513" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Read composition (assigned reads, all groups)"</span>)</span>
<span id="cb40-514"><a href="#cb40-514" aria-hidden="true" tabindex="-1"></a>g_comp_assigned</span>
<span id="cb40-515"><a href="#cb40-515" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-516"><a href="#cb40-516" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot composition of minor components</span></span>
<span id="cb40-517"><a href="#cb40-517" aria-hidden="true" tabindex="-1"></a><span class="co">#g_comp_minor &lt;- g_comp_base + </span></span>
<span id="cb40-518"><a href="#cb40-518" aria-hidden="true" tabindex="-1"></a><span class="co">#  geom_comp(data=comp_minor) +</span></span>
<span id="cb40-519"><a href="#cb40-519" aria-hidden="true" tabindex="-1"></a><span class="co">#  scale_y_pc_reads() +</span></span>
<span id="cb40-520"><a href="#cb40-520" aria-hidden="true" tabindex="-1"></a><span class="co">#  scale_fill_classification() + </span></span>
<span id="cb40-521"><a href="#cb40-521" aria-hidden="true" tabindex="-1"></a><span class="co">#  ggtitle("Read composition (all reads, minor groups)")</span></span>
<span id="cb40-522"><a href="#cb40-522" aria-hidden="true" tabindex="-1"></a><span class="co">#g_comp_minor</span></span>
<span id="cb40-523"><a href="#cb40-523" aria-hidden="true" tabindex="-1"></a><span class="co">#g_comp_assigned_minor &lt;- g_comp_base + </span></span>
<span id="cb40-524"><a href="#cb40-524" aria-hidden="true" tabindex="-1"></a><span class="co">#  geom_comp(data=comp_assigned_minor) +</span></span>
<span id="cb40-525"><a href="#cb40-525" aria-hidden="true" tabindex="-1"></a><span class="co">#  scale_y_pc_reads() +</span></span>
<span id="cb40-526"><a href="#cb40-526" aria-hidden="true" tabindex="-1"></a><span class="co">#  scale_fill_classification() + </span></span>
<span id="cb40-527"><a href="#cb40-527" aria-hidden="true" tabindex="-1"></a><span class="co">#  ggtitle("Read composition (assigned reads, minor groups)")</span></span>
<span id="cb40-528"><a href="#cb40-528" aria-hidden="true" tabindex="-1"></a><span class="co">#g_comp_assigned_minor</span></span>
<span id="cb40-529"><a href="#cb40-529" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-530"><a href="#cb40-530" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-531"><a href="#cb40-531" aria-hidden="true" tabindex="-1"></a><span class="fu">## Total viral content</span></span>
<span id="cb40-532"><a href="#cb40-532" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-533"><a href="#cb40-533" aria-hidden="true" tabindex="-1"></a>Total viral fraction average $1.03 \times 10^{-4}$ across samples. As a fraction of assigned (rather than total) reads, this jumped to $1.28 \times 10^{-4}$. The increase in viral fraction from assigned reads to total reads was constant across all samples, however, we can see that the viral fractions differed by location. Note that one of the pools from "Heilongjiang,Mudanjiang" had no viral reads.</span>
<span id="cb40-534"><a href="#cb40-534" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-537"><a href="#cb40-537" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb40-538"><a href="#cb40-538" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: p-viral-scatter</span></span>
<span id="cb40-539"><a href="#cb40-539" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb40-540"><a href="#cb40-540" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 10</span></span>
<span id="cb40-541"><a href="#cb40-541" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb40-542"><a href="#cb40-542" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-543"><a href="#cb40-543" aria-hidden="true" tabindex="-1"></a><span class="co">#p_reads_viral_all &lt;- comp %&gt;% filter(classification == "Viral") %&gt;%</span></span>
<span id="cb40-544"><a href="#cb40-544" aria-hidden="true" tabindex="-1"></a><span class="co">#  mutate(read_group = "All reads")</span></span>
<span id="cb40-545"><a href="#cb40-545" aria-hidden="true" tabindex="-1"></a>p_reads_viral_assigned <span class="ot">&lt;-</span> comp_assigned <span class="sc">%&gt;%</span> <span class="fu">filter</span>(classification <span class="sc">==</span> <span class="st">"Viral"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-546"><a href="#cb40-546" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">read_group =</span> <span class="st">"Classified reads"</span>)</span>
<span id="cb40-547"><a href="#cb40-547" aria-hidden="true" tabindex="-1"></a><span class="co">#p_reads_viral &lt;- bind_rows(p_reads_viral_all, p_reads_viral_assigned)</span></span>
<span id="cb40-548"><a href="#cb40-548" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-549"><a href="#cb40-549" aria-hidden="true" tabindex="-1"></a>location_viral <span class="ot">&lt;-</span> comp_assigned <span class="sc">%&gt;%</span> <span class="fu">filter</span>(classification <span class="sc">==</span> <span class="st">"Viral"</span> ) <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(location) </span>
<span id="cb40-550"><a href="#cb40-550" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-551"><a href="#cb40-551" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb40-552"><a href="#cb40-552" aria-hidden="true" tabindex="-1"></a><span class="co"># g_viral &lt;- ggplot(p_reads_viral, aes(x=p_reads, y=library, color=location, shape=read_group)) +</span></span>
<span id="cb40-553"><a href="#cb40-553" aria-hidden="true" tabindex="-1"></a><span class="co">#   geom_point() +</span></span>
<span id="cb40-554"><a href="#cb40-554" aria-hidden="true" tabindex="-1"></a><span class="co">#   scale_y_discrete(name="Plasma pool") +</span></span>
<span id="cb40-555"><a href="#cb40-555" aria-hidden="true" tabindex="-1"></a><span class="co">#   scale_x_log10(name="Viral read fraction") +</span></span>
<span id="cb40-556"><a href="#cb40-556" aria-hidden="true" tabindex="-1"></a><span class="co">#   scale_color_brewer(palette = "Dark2") +</span></span>
<span id="cb40-557"><a href="#cb40-557" aria-hidden="true" tabindex="-1"></a><span class="co">#   #facet_grid(.~read_group, scales = "free") +</span></span>
<span id="cb40-558"><a href="#cb40-558" aria-hidden="true" tabindex="-1"></a><span class="co">#   guides(color=guide_legend(nrow=2), shape=guide_legend(nrow=2),</span></span>
<span id="cb40-559"><a href="#cb40-559" aria-hidden="true" tabindex="-1"></a><span class="co">#          linetype=guide_legend(nrow=2)) +</span></span>
<span id="cb40-560"><a href="#cb40-560" aria-hidden="true" tabindex="-1"></a><span class="co">#   theme_kit</span></span>
<span id="cb40-561"><a href="#cb40-561" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-562"><a href="#cb40-562" aria-hidden="true" tabindex="-1"></a><span class="co">#ggplot(location_viral, aes(x=p_reads, y=location, color = location)) +</span></span>
<span id="cb40-563"><a href="#cb40-563" aria-hidden="true" tabindex="-1"></a><span class="co">#  geom_quasirandom(size=2) +</span></span>
<span id="cb40-564"><a href="#cb40-564" aria-hidden="true" tabindex="-1"></a><span class="co">#  scale_x_log10(name="Viral read fraction", labels = label_log(digits=2)) +</span></span>
<span id="cb40-565"><a href="#cb40-565" aria-hidden="true" tabindex="-1"></a><span class="co">#  scale_y_discrete(name="Location") +</span></span>
<span id="cb40-566"><a href="#cb40-566" aria-hidden="true" tabindex="-1"></a><span class="co">#  scale_color_brewer(palette = "Dark2") +</span></span>
<span id="cb40-567"><a href="#cb40-567" aria-hidden="true" tabindex="-1"></a><span class="co">#  theme_light() + </span></span>
<span id="cb40-568"><a href="#cb40-568" aria-hidden="true" tabindex="-1"></a><span class="co">#    theme(</span></span>
<span id="cb40-569"><a href="#cb40-569" aria-hidden="true" tabindex="-1"></a><span class="co">#    axis.text.y = element_text(size = 8),</span></span>
<span id="cb40-570"><a href="#cb40-570" aria-hidden="true" tabindex="-1"></a><span class="co">#    axis.text.x = element_text(size = 14),</span></span>
<span id="cb40-571"><a href="#cb40-571" aria-hidden="true" tabindex="-1"></a><span class="co">#    axis.ticks.y = element_blank(),</span></span>
<span id="cb40-572"><a href="#cb40-572" aria-hidden="true" tabindex="-1"></a><span class="co">#    axis.line = element_line(colour = "black"),</span></span>
<span id="cb40-573"><a href="#cb40-573" aria-hidden="true" tabindex="-1"></a><span class="co">#    axis.title.x = element_text(size = 15),    </span></span>
<span id="cb40-574"><a href="#cb40-574" aria-hidden="true" tabindex="-1"></a><span class="co">#    axis.title.y = element_text(size = 15),  </span></span>
<span id="cb40-575"><a href="#cb40-575" aria-hidden="true" tabindex="-1"></a><span class="co">#    legend.text = element_text(size = 13),</span></span>
<span id="cb40-576"><a href="#cb40-576" aria-hidden="true" tabindex="-1"></a><span class="co">#    legend.title = element_text(size = 16),</span></span>
<span id="cb40-577"><a href="#cb40-577" aria-hidden="true" tabindex="-1"></a><span class="co">#    legend.position = c(1, 1),  # Move legend to top right</span></span>
<span id="cb40-578"><a href="#cb40-578" aria-hidden="true" tabindex="-1"></a><span class="do">##    legend.justification = c(1, 1),  # Align legend to top right</span></span>
<span id="cb40-579"><a href="#cb40-579" aria-hidden="true" tabindex="-1"></a><span class="co">#    panel.grid.minor = element_blank(),</span></span>
<span id="cb40-580"><a href="#cb40-580" aria-hidden="true" tabindex="-1"></a><span class="co">#    panel.border = element_blank(),</span></span>
<span id="cb40-581"><a href="#cb40-581" aria-hidden="true" tabindex="-1"></a><span class="co">#    panel.background = element_blank())</span></span>
<span id="cb40-582"><a href="#cb40-582" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-583"><a href="#cb40-583" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate the number of samples per location</span></span>
<span id="cb40-584"><a href="#cb40-584" aria-hidden="true" tabindex="-1"></a>location_counts <span class="ot">&lt;-</span> location_viral <span class="sc">%&gt;%</span></span>
<span id="cb40-585"><a href="#cb40-585" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(location) <span class="sc">%&gt;%</span></span>
<span id="cb40-586"><a href="#cb40-586" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">count =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb40-587"><a href="#cb40-587" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">label =</span> <span class="fu">paste0</span>(location, <span class="st">" (n="</span>, count, <span class="st">")"</span>))</span>
<span id="cb40-588"><a href="#cb40-588" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-589"><a href="#cb40-589" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a named vector for easy mapping</span></span>
<span id="cb40-590"><a href="#cb40-590" aria-hidden="true" tabindex="-1"></a>location_labels <span class="ot">&lt;-</span> <span class="fu">setNames</span>(location_counts<span class="sc">$</span>label, location_counts<span class="sc">$</span>location)</span>
<span id="cb40-591"><a href="#cb40-591" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-592"><a href="#cb40-592" aria-hidden="true" tabindex="-1"></a>g_viral <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(location_viral, <span class="fu">aes</span>(<span class="at">x=</span>location, <span class="at">y=</span>p_reads, <span class="at">fill=</span>location)) <span class="sc">+</span></span>
<span id="cb40-593"><a href="#cb40-593" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_violin</span>(<span class="at">trim=</span><span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb40-594"><a href="#cb40-594" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">width=</span><span class="fl">0.1</span>, <span class="at">fill=</span><span class="st">"white"</span>, <span class="at">color=</span><span class="st">"black"</span>, <span class="at">outlier.shape=</span><span class="cn">NA</span>) <span class="sc">+</span></span>
<span id="cb40-595"><a href="#cb40-595" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">width=</span><span class="fl">0.1</span>, <span class="at">size=</span><span class="fl">0.5</span>, <span class="at">alpha=</span><span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb40-596"><a href="#cb40-596" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>(<span class="at">name=</span><span class="st">"Viral read fraction"</span>, <span class="at">labels =</span> <span class="fu">label_log</span>(<span class="at">digits=</span><span class="dv">2</span>)) <span class="sc">+</span></span>
<span id="cb40-597"><a href="#cb40-597" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">name=</span><span class="st">"Location"</span>, <span class="at">labels =</span> location_labels) <span class="sc">+</span></span>
<span id="cb40-598"><a href="#cb40-598" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Dark2"</span>) <span class="sc">+</span></span>
<span id="cb40-599"><a href="#cb40-599" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">"Total viral fraction based on assigned reads over all locations"</span>) <span class="sc">+</span></span>
<span id="cb40-600"><a href="#cb40-600" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_light</span>() <span class="sc">+</span> </span>
<span id="cb40-601"><a href="#cb40-601" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb40-602"><a href="#cb40-602" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>),</span>
<span id="cb40-603"><a href="#cb40-603" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>, <span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>),</span>
<span id="cb40-604"><a href="#cb40-604" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">15</span>),    </span>
<span id="cb40-605"><a href="#cb40-605" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">15</span>),  </span>
<span id="cb40-606"><a href="#cb40-606" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">"none"</span>,</span>
<span id="cb40-607"><a href="#cb40-607" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb40-608"><a href="#cb40-608" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.border =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb40-609"><a href="#cb40-609" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.background =</span> <span class="fu">element_blank</span>()</span>
<span id="cb40-610"><a href="#cb40-610" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb40-611"><a href="#cb40-611" aria-hidden="true" tabindex="-1"></a>  theme_kit <span class="sc">+</span></span>
<span id="cb40-612"><a href="#cb40-612" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>()  <span class="co"># This flips the coordinates to make the plot horizontal</span></span>
<span id="cb40-613"><a href="#cb40-613" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-614"><a href="#cb40-614" aria-hidden="true" tabindex="-1"></a>g_viral</span>
<span id="cb40-615"><a href="#cb40-615" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-616"><a href="#cb40-616" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-617"><a href="#cb40-617" aria-hidden="true" tabindex="-1"></a><span class="fu">## Taxonomic composition of viruses</span></span>
<span id="cb40-618"><a href="#cb40-618" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-619"><a href="#cb40-619" aria-hidden="true" tabindex="-1"></a>The threshold for the label "other" are the set of families that make up less than 20% composition in all samples (the only reason I did this was because there were too many matches). We can see that other makes up a big portion of the viral composition. Outside of the "other" label, Hepadnaviridae makes up the largest portion of the viral composition.</span>
<span id="cb40-620"><a href="#cb40-620" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-623"><a href="#cb40-623" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb40-624"><a href="#cb40-624" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: extract-viral-taxa</span></span>
<span id="cb40-625"><a href="#cb40-625" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb40-626"><a href="#cb40-626" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-627"><a href="#cb40-627" aria-hidden="true" tabindex="-1"></a><span class="co"># Get viral taxonomy</span></span>
<span id="cb40-628"><a href="#cb40-628" aria-hidden="true" tabindex="-1"></a>viral_taxa_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"total-virus-db.tsv.gz"</span>)</span>
<span id="cb40-629"><a href="#cb40-629" aria-hidden="true" tabindex="-1"></a>viral_taxa <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(viral_taxa_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb40-630"><a href="#cb40-630" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-631"><a href="#cb40-631" aria-hidden="true" tabindex="-1"></a><span class="co"># Get Kraken reports</span></span>
<span id="cb40-632"><a href="#cb40-632" aria-hidden="true" tabindex="-1"></a>reports_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(tax_final_dir, <span class="st">"kraken_reports.tsv.gz"</span>)</span>
<span id="cb40-633"><a href="#cb40-633" aria-hidden="true" tabindex="-1"></a>reports <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(reports_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-634"><a href="#cb40-634" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(sample)</span>
<span id="cb40-635"><a href="#cb40-635" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-636"><a href="#cb40-636" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter to viral taxa</span></span>
<span id="cb40-637"><a href="#cb40-637" aria-hidden="true" tabindex="-1"></a>kraken_reports_viral <span class="ot">&lt;-</span> <span class="fu">filter</span>(reports, taxid <span class="sc">%in%</span> viral_taxa<span class="sc">$</span>taxid) <span class="sc">%&gt;%</span></span>
<span id="cb40-638"><a href="#cb40-638" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb40-639"><a href="#cb40-639" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_viral =</span> n_reads_clade<span class="sc">/</span>n_reads_clade[<span class="dv">1</span>])</span>
<span id="cb40-640"><a href="#cb40-640" aria-hidden="true" tabindex="-1"></a>kraken_reports_viral_cleaned <span class="ot">&lt;-</span> kraken_reports_viral <span class="sc">%&gt;%</span></span>
<span id="cb40-641"><a href="#cb40-641" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>pc_reads_total, <span class="sc">-</span>n_reads_direct, <span class="sc">-</span><span class="fu">contains</span>(<span class="st">"minimizers"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb40-642"><a href="#cb40-642" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, taxid, p_reads_viral, n_reads_clade, <span class="fu">everything</span>()) <span class="sc">%&gt;%</span> ungroup</span>
<span id="cb40-643"><a href="#cb40-643" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-644"><a href="#cb40-644" aria-hidden="true" tabindex="-1"></a>viral_classes <span class="ot">&lt;-</span> kraken_reports_viral_cleaned <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">==</span> <span class="st">"C"</span>)</span>
<span id="cb40-645"><a href="#cb40-645" aria-hidden="true" tabindex="-1"></a>viral_families <span class="ot">&lt;-</span> kraken_reports_viral_cleaned <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">==</span> <span class="st">"F"</span>)</span>
<span id="cb40-646"><a href="#cb40-646" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-647"><a href="#cb40-647" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-650"><a href="#cb40-650" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb40-651"><a href="#cb40-651" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: viral-family-composition</span></span>
<span id="cb40-652"><a href="#cb40-652" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb40-653"><a href="#cb40-653" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 15</span></span>
<span id="cb40-654"><a href="#cb40-654" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-655"><a href="#cb40-655" aria-hidden="true" tabindex="-1"></a>major_threshold <span class="ot">&lt;-</span> <span class="fl">0.20</span></span>
<span id="cb40-656"><a href="#cb40-656" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-657"><a href="#cb40-657" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify major viral families</span></span>
<span id="cb40-658"><a href="#cb40-658" aria-hidden="true" tabindex="-1"></a>viral_families_major_tab <span class="ot">&lt;-</span> viral_families <span class="sc">%&gt;%</span> </span>
<span id="cb40-659"><a href="#cb40-659" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb40-660"><a href="#cb40-660" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads_viral_max =</span> <span class="fu">max</span>(p_reads_viral), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-661"><a href="#cb40-661" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_viral_max <span class="sc">&gt;=</span> major_threshold)</span>
<span id="cb40-662"><a href="#cb40-662" aria-hidden="true" tabindex="-1"></a>viral_families_major_list <span class="ot">&lt;-</span> viral_families_major_tab <span class="sc">%&gt;%</span> <span class="fu">pull</span>(name)</span>
<span id="cb40-663"><a href="#cb40-663" aria-hidden="true" tabindex="-1"></a>viral_families_major <span class="ot">&lt;-</span> viral_families <span class="sc">%&gt;%</span> </span>
<span id="cb40-664"><a href="#cb40-664" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(name <span class="sc">%in%</span> viral_families_major_list) <span class="sc">%&gt;%</span></span>
<span id="cb40-665"><a href="#cb40-665" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, taxid, sample, p_reads_viral)</span>
<span id="cb40-666"><a href="#cb40-666" aria-hidden="true" tabindex="-1"></a>viral_families_minor <span class="ot">&lt;-</span> viral_families_major <span class="sc">%&gt;%</span> </span>
<span id="cb40-667"><a href="#cb40-667" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb40-668"><a href="#cb40-668" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads_viral_major =</span> <span class="fu">sum</span>(p_reads_viral), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-669"><a href="#cb40-669" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="st">"Other"</span>, <span class="at">taxid=</span><span class="cn">NA</span>, <span class="at">p_reads_viral =</span> <span class="dv">1</span><span class="sc">-</span>p_reads_viral_major) <span class="sc">%&gt;%</span></span>
<span id="cb40-670"><a href="#cb40-670" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, taxid, sample, p_reads_viral)</span>
<span id="cb40-671"><a href="#cb40-671" aria-hidden="true" tabindex="-1"></a>viral_families_display <span class="ot">&lt;-</span> viral_families_major <span class="sc">%&gt;%</span> </span>
<span id="cb40-672"><a href="#cb40-672" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_rows</span>(viral_families_minor) <span class="sc">%&gt;%</span></span>
<span id="cb40-673"><a href="#cb40-673" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_viral)) <span class="sc">%&gt;%</span> </span>
<span id="cb40-674"><a href="#cb40-674" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">factor</span>(name, <span class="at">levels=</span><span class="fu">c</span>(viral_families_major_list, <span class="st">"Other"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb40-675"><a href="#cb40-675" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_viral, <span class="at">classification=</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb40-676"><a href="#cb40-676" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">'sample'</span>)</span>
<span id="cb40-677"><a href="#cb40-677" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-678"><a href="#cb40-678" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a custom color palette with up to 20 colors</span></span>
<span id="cb40-679"><a href="#cb40-679" aria-hidden="true" tabindex="-1"></a>custom_palette <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb40-680"><a href="#cb40-680" aria-hidden="true" tabindex="-1"></a>  <span class="fu">brewer.pal</span>(<span class="dv">12</span>, <span class="st">"Paired"</span>),</span>
<span id="cb40-681"><a href="#cb40-681" aria-hidden="true" tabindex="-1"></a>  <span class="fu">brewer.pal</span>(<span class="dv">12</span>, <span class="st">"Set3"</span>)</span>
<span id="cb40-682"><a href="#cb40-682" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb40-683"><a href="#cb40-683" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-684"><a href="#cb40-684" aria-hidden="true" tabindex="-1"></a>n_classifications <span class="ot">&lt;-</span> <span class="fu">length</span>(<span class="fu">unique</span>(viral_families_display<span class="sc">$</span>classification)) <span class="sc">-</span> <span class="dv">1</span></span>
<span id="cb40-685"><a href="#cb40-685" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-686"><a href="#cb40-686" aria-hidden="true" tabindex="-1"></a>custom_palette_with_black <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb40-687"><a href="#cb40-687" aria-hidden="true" tabindex="-1"></a>  custom_palette[<span class="dv">1</span><span class="sc">:</span>n_classifications],</span>
<span id="cb40-688"><a href="#cb40-688" aria-hidden="true" tabindex="-1"></a>  <span class="st">"black"</span></span>
<span id="cb40-689"><a href="#cb40-689" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb40-690"><a href="#cb40-690" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-691"><a href="#cb40-691" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a new color palette with black for "Other"</span></span>
<span id="cb40-692"><a href="#cb40-692" aria-hidden="true" tabindex="-1"></a>custom_palette_with_black <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb40-693"><a href="#cb40-693" aria-hidden="true" tabindex="-1"></a>  custom_palette[<span class="dv">1</span><span class="sc">:</span>n_classifications],</span>
<span id="cb40-694"><a href="#cb40-694" aria-hidden="true" tabindex="-1"></a>  <span class="st">"black"</span></span>
<span id="cb40-695"><a href="#cb40-695" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb40-696"><a href="#cb40-696" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-697"><a href="#cb40-697" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb40-698"><a href="#cb40-698" aria-hidden="true" tabindex="-1"></a>g_families <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb40-699"><a href="#cb40-699" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_comp</span>(<span class="at">data=</span>viral_families_display) <span class="sc">+</span></span>
<span id="cb40-700"><a href="#cb40-700" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Viral Reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb40-701"><a href="#cb40-701" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb40-702"><a href="#cb40-702" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb40-703"><a href="#cb40-703" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> custom_palette_with_black) <span class="sc">+</span></span>
<span id="cb40-704"><a href="#cb40-704" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span> location, <span class="at">scales =</span> <span class="st">"free_x"</span>, <span class="at">ncol =</span> <span class="dv">2</span>) </span>
<span id="cb40-705"><a href="#cb40-705" aria-hidden="true" tabindex="-1"></a>g_families</span>
<span id="cb40-706"><a href="#cb40-706" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-707"><a href="#cb40-707" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-708"><a href="#cb40-708" aria-hidden="true" tabindex="-1"></a><span class="fu"># Human-infecting virus reads</span></span>
<span id="cb40-709"><a href="#cb40-709" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-710"><a href="#cb40-710" aria-hidden="true" tabindex="-1"></a><span class="fu">## Overall relative abundance</span></span>
<span id="cb40-711"><a href="#cb40-711" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-712"><a href="#cb40-712" aria-hidden="true" tabindex="-1"></a>I calculated the relative abundance of human-infecting viruses in two ways:</span>
<span id="cb40-713"><a href="#cb40-713" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-714"><a href="#cb40-714" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>First, as the total number of deduplicated human-virus reads in each sample, divided by the number of raw reads ("All reads"). This results in a viral fraction of $9.84 \times 10^{-5}$ across all samples.</span>
<span id="cb40-715"><a href="#cb40-715" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-716"><a href="#cb40-716" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Second, as a fraction of preprocessed (cleaned, deduplicated, computationally ribodepleted) reads ("Preprocessed reads"). This results in a viral fraction of $1.18 \times 10^{-4}$ across all samples.</span>
<span id="cb40-717"><a href="#cb40-717" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-718"><a href="#cb40-718" aria-hidden="true" tabindex="-1"></a>Note that 10 of the pools had no viral reads: 5 from "Heilongjiang, Mudanjiang", 4 from "Jiangsu, Nanjing", and 1 from "Xinjiang, Urmqi".</span>
<span id="cb40-719"><a href="#cb40-719" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-722"><a href="#cb40-722" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb40-723"><a href="#cb40-723" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: prepare-hv</span></span>
<span id="cb40-724"><a href="#cb40-724" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb40-725"><a href="#cb40-725" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-726"><a href="#cb40-726" aria-hidden="true" tabindex="-1"></a><span class="co"># Import and format reads</span></span>
<span id="cb40-727"><a href="#cb40-727" aria-hidden="true" tabindex="-1"></a>hv_reads_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(hv_dir, <span class="st">"hv_hits_putative_collapsed.tsv.gz"</span>)</span>
<span id="cb40-728"><a href="#cb40-728" aria-hidden="true" tabindex="-1"></a>mrg_hv <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(hv_reads_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-729"><a href="#cb40-729" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb40-730"><a href="#cb40-730" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">kraken_label =</span> <span class="fu">ifelse</span>(assigned_hv, <span class="st">"Kraken2 HV assignment"</span>,</span>
<span id="cb40-731"><a href="#cb40-731" aria-hidden="true" tabindex="-1"></a>                               <span class="st">"No Kraken2 assignment"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb40-732"><a href="#cb40-732" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">adj_score_max =</span> <span class="fu">pmax</span>(adj_score_fwd, adj_score_rev),</span>
<span id="cb40-733"><a href="#cb40-733" aria-hidden="true" tabindex="-1"></a>         <span class="at">highscore =</span> adj_score_max <span class="sc">&gt;=</span> <span class="dv">20</span>,</span>
<span id="cb40-734"><a href="#cb40-734" aria-hidden="true" tabindex="-1"></a>         <span class="at">hv_status =</span> assigned_hv <span class="sc">|</span> highscore) <span class="sc">%&gt;%</span></span>
<span id="cb40-735"><a href="#cb40-735" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">taxid_all =</span> taxid, <span class="at">taxid =</span> taxid_best)</span>
<span id="cb40-736"><a href="#cb40-736" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-737"><a href="#cb40-737" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-740"><a href="#cb40-740" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb40-741"><a href="#cb40-741" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: count-hv-reads</span></span>
<span id="cb40-742"><a href="#cb40-742" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb40-743"><a href="#cb40-743" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 10</span></span>
<span id="cb40-744"><a href="#cb40-744" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb40-745"><a href="#cb40-745" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-746"><a href="#cb40-746" aria-hidden="true" tabindex="-1"></a><span class="co"># Get raw read counts</span></span>
<span id="cb40-747"><a href="#cb40-747" aria-hidden="true" tabindex="-1"></a>read_counts_raw <span class="ot">&lt;-</span> <span class="fu">filter</span>(basic_stats_raw) <span class="sc">%&gt;%</span></span>
<span id="cb40-748"><a href="#cb40-748" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, <span class="at">n_reads_raw =</span> n_read_pairs)</span>
<span id="cb40-749"><a href="#cb40-749" aria-hidden="true" tabindex="-1"></a>read_counts_preproc <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"ribo_initial"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-750"><a href="#cb40-750" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, <span class="at">n_reads_preproc =</span> n_read_pairs)</span>
<span id="cb40-751"><a href="#cb40-751" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-752"><a href="#cb40-752" aria-hidden="true" tabindex="-1"></a><span class="co"># Get HV read counts</span></span>
<span id="cb40-753"><a href="#cb40-753" aria-hidden="true" tabindex="-1"></a>read_counts_hv <span class="ot">&lt;-</span> mrg_hv <span class="sc">%&gt;%</span> <span class="fu">filter</span>(hv_status) <span class="sc">%&gt;%</span> </span>
<span id="cb40-754"><a href="#cb40-754" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb40-755"><a href="#cb40-755" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name=</span><span class="st">"n_reads_hv"</span>)</span>
<span id="cb40-756"><a href="#cb40-756" aria-hidden="true" tabindex="-1"></a>read_counts <span class="ot">&lt;-</span> read_counts_raw <span class="sc">%&gt;%</span></span>
<span id="cb40-757"><a href="#cb40-757" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(read_counts_hv, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"sample"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb40-758"><a href="#cb40-758" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">n_reads_hv =</span> <span class="fu">replace_na</span>(n_reads_hv, <span class="dv">0</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb40-759"><a href="#cb40-759" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(read_counts_preproc, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"sample"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb40-760"><a href="#cb40-760" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"sample"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb40-761"><a href="#cb40-761" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, n_reads_raw, n_reads_preproc, n_reads_hv) <span class="sc">%&gt;%</span></span>
<span id="cb40-762"><a href="#cb40-762" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">n_samples =</span> <span class="dv">1</span>,</span>
<span id="cb40-763"><a href="#cb40-763" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_total =</span> n_reads_hv<span class="sc">/</span>n_reads_raw,</span>
<span id="cb40-764"><a href="#cb40-764" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_preproc =</span> n_reads_hv<span class="sc">/</span>n_reads_preproc)</span>
<span id="cb40-765"><a href="#cb40-765" aria-hidden="true" tabindex="-1"></a>read_counts_long <span class="ot">&lt;-</span> read_counts <span class="sc">%&gt;%</span></span>
<span id="cb40-766"><a href="#cb40-766" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="fu">starts_with</span>(<span class="st">"p_reads"</span>), <span class="at">names_to=</span><span class="st">"read_group"</span>, <span class="at">values_to=</span><span class="st">"p_reads"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-767"><a href="#cb40-767" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">read_group =</span> <span class="fu">ifelse</span>(read_group <span class="sc">==</span> <span class="st">"p_reads_total"</span>, <span class="st">"All reads"</span>, <span class="st">"Preprocessed reads"</span>))</span>
<span id="cb40-768"><a href="#cb40-768" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-769"><a href="#cb40-769" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine for display</span></span>
<span id="cb40-770"><a href="#cb40-770" aria-hidden="true" tabindex="-1"></a>read_counts_agg <span class="ot">&lt;-</span> read_counts <span class="sc">%&gt;%</span></span>
<span id="cb40-771"><a href="#cb40-771" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_total =</span> n_reads_hv<span class="sc">/</span>n_reads_raw,</span>
<span id="cb40-772"><a href="#cb40-772" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_preproc =</span> n_reads_hv<span class="sc">/</span>n_reads_preproc) <span class="sc">%&gt;%</span></span>
<span id="cb40-773"><a href="#cb40-773" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"sample"</span>))</span>
<span id="cb40-774"><a href="#cb40-774" aria-hidden="true" tabindex="-1"></a>read_counts_agg_long <span class="ot">&lt;-</span> read_counts_agg <span class="sc">%&gt;%</span></span>
<span id="cb40-775"><a href="#cb40-775" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="fu">starts_with</span>(<span class="st">"p_reads"</span>), <span class="at">names_to=</span><span class="st">"read_group"</span>, <span class="at">values_to=</span><span class="st">"p_reads"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-776"><a href="#cb40-776" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">read_group =</span> <span class="fu">ifelse</span>(read_group <span class="sc">==</span> <span class="st">"p_reads_total"</span>, <span class="st">"All reads"</span>, <span class="st">"Preprocessed reads"</span>)) </span>
<span id="cb40-777"><a href="#cb40-777" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-778"><a href="#cb40-778" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize</span></span>
<span id="cb40-779"><a href="#cb40-779" aria-hidden="true" tabindex="-1"></a><span class="co">#g_read_counts &lt;- ggplot(read_counts_agg_long, aes(x=library, y=p_reads)) +</span></span>
<span id="cb40-780"><a href="#cb40-780" aria-hidden="true" tabindex="-1"></a><span class="co">#  geom_point() +</span></span>
<span id="cb40-781"><a href="#cb40-781" aria-hidden="true" tabindex="-1"></a><span class="co">#  scale_y_log10(name = "Unique human-viral read fraction") +</span></span>
<span id="cb40-782"><a href="#cb40-782" aria-hidden="true" tabindex="-1"></a><span class="co">#  facet_grid(.~read_group, scales = "free") +</span></span>
<span id="cb40-783"><a href="#cb40-783" aria-hidden="true" tabindex="-1"></a><span class="co">#  theme_kit</span></span>
<span id="cb40-784"><a href="#cb40-784" aria-hidden="true" tabindex="-1"></a><span class="co">#g_read_counts</span></span>
<span id="cb40-785"><a href="#cb40-785" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-786"><a href="#cb40-786" aria-hidden="true" tabindex="-1"></a><span class="co">#g_viral &lt;- ggplot(read_counts_agg_long, aes(x=p_reads, y=library, color=location, shape=read_group)) +</span></span>
<span id="cb40-787"><a href="#cb40-787" aria-hidden="true" tabindex="-1"></a><span class="co">#  geom_point() +</span></span>
<span id="cb40-788"><a href="#cb40-788" aria-hidden="true" tabindex="-1"></a><span class="co">#  scale_y_discrete(name="Plasma pool") +</span></span>
<span id="cb40-789"><a href="#cb40-789" aria-hidden="true" tabindex="-1"></a><span class="co">#  scale_x_log10(name="Viral read fraction") +</span></span>
<span id="cb40-790"><a href="#cb40-790" aria-hidden="true" tabindex="-1"></a><span class="co">#  scale_color_brewer(palette = "Dark2") +</span></span>
<span id="cb40-791"><a href="#cb40-791" aria-hidden="true" tabindex="-1"></a><span class="co">#  facet_grid(.~read_group, scales = "free") +</span></span>
<span id="cb40-792"><a href="#cb40-792" aria-hidden="true" tabindex="-1"></a><span class="co">#  guides(color=guide_legend(nrow=2), shape=guide_legend(nrow=2),</span></span>
<span id="cb40-793"><a href="#cb40-793" aria-hidden="true" tabindex="-1"></a><span class="co">#         linetype=guide_legend(nrow=2)) +</span></span>
<span id="cb40-794"><a href="#cb40-794" aria-hidden="true" tabindex="-1"></a><span class="co">#  theme_kit</span></span>
<span id="cb40-795"><a href="#cb40-795" aria-hidden="true" tabindex="-1"></a><span class="co">#g_viral</span></span>
<span id="cb40-796"><a href="#cb40-796" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-797"><a href="#cb40-797" aria-hidden="true" tabindex="-1"></a>hv_location <span class="ot">&lt;-</span> read_counts_agg_long <span class="sc">%&gt;%</span> <span class="fu">filter</span>(read_group <span class="sc">==</span> <span class="st">"Preprocessed reads"</span> ) <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(location) </span>
<span id="cb40-798"><a href="#cb40-798" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-799"><a href="#cb40-799" aria-hidden="true" tabindex="-1"></a>hv_location_counts <span class="ot">&lt;-</span> hv_location <span class="sc">%&gt;%</span></span>
<span id="cb40-800"><a href="#cb40-800" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(location) <span class="sc">%&gt;%</span></span>
<span id="cb40-801"><a href="#cb40-801" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">count =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb40-802"><a href="#cb40-802" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">label =</span> <span class="fu">paste0</span>(location, <span class="st">" (n="</span>, count, <span class="st">")"</span>))</span>
<span id="cb40-803"><a href="#cb40-803" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-804"><a href="#cb40-804" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create a named vector for easy mapping</span></span>
<span id="cb40-805"><a href="#cb40-805" aria-hidden="true" tabindex="-1"></a>location_labels <span class="ot">&lt;-</span> <span class="fu">setNames</span>(hv_location_counts<span class="sc">$</span>label, hv_location_counts<span class="sc">$</span>location)</span>
<span id="cb40-806"><a href="#cb40-806" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-807"><a href="#cb40-807" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a named vector for easy mapping</span></span>
<span id="cb40-808"><a href="#cb40-808" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb40-809"><a href="#cb40-809" aria-hidden="true" tabindex="-1"></a><span class="co"># g_viral &lt;- ggplot(p_reads_viral, aes(x=p_reads, y=library, color=location, shape=read_group)) +</span></span>
<span id="cb40-810"><a href="#cb40-810" aria-hidden="true" tabindex="-1"></a><span class="co">#   geom_point() +</span></span>
<span id="cb40-811"><a href="#cb40-811" aria-hidden="true" tabindex="-1"></a><span class="co">#   scale_y_discrete(name="Plasma pool") +</span></span>
<span id="cb40-812"><a href="#cb40-812" aria-hidden="true" tabindex="-1"></a><span class="co">#   scale_x_log10(name="Viral read fraction") +</span></span>
<span id="cb40-813"><a href="#cb40-813" aria-hidden="true" tabindex="-1"></a><span class="co">#   scale_color_brewer(palette = "Dark2") +</span></span>
<span id="cb40-814"><a href="#cb40-814" aria-hidden="true" tabindex="-1"></a><span class="co">#   #facet_grid(.~read_group, scales = "free") +</span></span>
<span id="cb40-815"><a href="#cb40-815" aria-hidden="true" tabindex="-1"></a><span class="co">#   guides(color=guide_legend(nrow=2), shape=guide_legend(nrow=2),</span></span>
<span id="cb40-816"><a href="#cb40-816" aria-hidden="true" tabindex="-1"></a><span class="co">#          linetype=guide_legend(nrow=2)) +</span></span>
<span id="cb40-817"><a href="#cb40-817" aria-hidden="true" tabindex="-1"></a><span class="co">#   theme_kit</span></span>
<span id="cb40-818"><a href="#cb40-818" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-819"><a href="#cb40-819" aria-hidden="true" tabindex="-1"></a><span class="co">#ggplot(location_viral, aes(x=p_reads, y=location, color = location)) +</span></span>
<span id="cb40-820"><a href="#cb40-820" aria-hidden="true" tabindex="-1"></a><span class="co">#  geom_quasirandom(size=2) +</span></span>
<span id="cb40-821"><a href="#cb40-821" aria-hidden="true" tabindex="-1"></a><span class="co">#  scale_x_log10(name="Viral read fraction", labels = label_log(digits=2)) +</span></span>
<span id="cb40-822"><a href="#cb40-822" aria-hidden="true" tabindex="-1"></a><span class="co">#  scale_y_discrete(name="Location") +</span></span>
<span id="cb40-823"><a href="#cb40-823" aria-hidden="true" tabindex="-1"></a><span class="co">#  scale_color_brewer(palette = "Dark2") +</span></span>
<span id="cb40-824"><a href="#cb40-824" aria-hidden="true" tabindex="-1"></a><span class="co">#  theme_light() + </span></span>
<span id="cb40-825"><a href="#cb40-825" aria-hidden="true" tabindex="-1"></a><span class="co">#    theme(</span></span>
<span id="cb40-826"><a href="#cb40-826" aria-hidden="true" tabindex="-1"></a><span class="co">#    axis.text.y = element_text(size = 8),</span></span>
<span id="cb40-827"><a href="#cb40-827" aria-hidden="true" tabindex="-1"></a><span class="co">#    axis.text.x = element_text(size = 14),</span></span>
<span id="cb40-828"><a href="#cb40-828" aria-hidden="true" tabindex="-1"></a><span class="co">#    axis.ticks.y = element_blank(),</span></span>
<span id="cb40-829"><a href="#cb40-829" aria-hidden="true" tabindex="-1"></a><span class="co">#    axis.line = element_line(colour = "black"),</span></span>
<span id="cb40-830"><a href="#cb40-830" aria-hidden="true" tabindex="-1"></a><span class="co">#    axis.title.x = element_text(size = 15),    </span></span>
<span id="cb40-831"><a href="#cb40-831" aria-hidden="true" tabindex="-1"></a><span class="co">#    axis.title.y = element_text(size = 15),  </span></span>
<span id="cb40-832"><a href="#cb40-832" aria-hidden="true" tabindex="-1"></a><span class="co">#    legend.text = element_text(size = 13),</span></span>
<span id="cb40-833"><a href="#cb40-833" aria-hidden="true" tabindex="-1"></a><span class="co">#    legend.title = element_text(size = 16),</span></span>
<span id="cb40-834"><a href="#cb40-834" aria-hidden="true" tabindex="-1"></a><span class="co">#    legend.position = c(1, 1),  # Move legend to top right</span></span>
<span id="cb40-835"><a href="#cb40-835" aria-hidden="true" tabindex="-1"></a><span class="do">##    legend.justification = c(1, 1),  # Align legend to top right</span></span>
<span id="cb40-836"><a href="#cb40-836" aria-hidden="true" tabindex="-1"></a><span class="co">#    panel.grid.minor = element_blank(),</span></span>
<span id="cb40-837"><a href="#cb40-837" aria-hidden="true" tabindex="-1"></a><span class="co">#    panel.border = element_blank(),</span></span>
<span id="cb40-838"><a href="#cb40-838" aria-hidden="true" tabindex="-1"></a><span class="co">#    panel.background = element_blank())</span></span>
<span id="cb40-839"><a href="#cb40-839" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-840"><a href="#cb40-840" aria-hidden="true" tabindex="-1"></a>g_viral <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(hv_location, <span class="fu">aes</span>(<span class="at">x=</span>location, <span class="at">y=</span>p_reads, <span class="at">fill=</span>location)) <span class="sc">+</span></span>
<span id="cb40-841"><a href="#cb40-841" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_violin</span>(<span class="at">trim=</span><span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb40-842"><a href="#cb40-842" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">width=</span><span class="fl">0.1</span>, <span class="at">fill=</span><span class="st">"white"</span>, <span class="at">color=</span><span class="st">"black"</span>, <span class="at">outlier.shape=</span><span class="cn">NA</span>) <span class="sc">+</span></span>
<span id="cb40-843"><a href="#cb40-843" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">width=</span><span class="fl">0.1</span>, <span class="at">size=</span><span class="fl">0.5</span>, <span class="at">alpha=</span><span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb40-844"><a href="#cb40-844" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>(<span class="at">name=</span><span class="st">"Viral read fraction"</span>, <span class="at">labels =</span> <span class="fu">label_log</span>(<span class="at">digits=</span><span class="dv">2</span>)) <span class="sc">+</span></span>
<span id="cb40-845"><a href="#cb40-845" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">name=</span><span class="st">"Location"</span>, <span class="at">labels =</span> location_labels) <span class="sc">+</span></span>
<span id="cb40-846"><a href="#cb40-846" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Dark2"</span>) <span class="sc">+</span></span>
<span id="cb40-847"><a href="#cb40-847" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_light</span>() <span class="sc">+</span> </span>
<span id="cb40-848"><a href="#cb40-848" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb40-849"><a href="#cb40-849" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>),</span>
<span id="cb40-850"><a href="#cb40-850" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>, <span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>),</span>
<span id="cb40-851"><a href="#cb40-851" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">15</span>),    </span>
<span id="cb40-852"><a href="#cb40-852" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">15</span>),  </span>
<span id="cb40-853"><a href="#cb40-853" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">"none"</span>,</span>
<span id="cb40-854"><a href="#cb40-854" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb40-855"><a href="#cb40-855" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.border =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb40-856"><a href="#cb40-856" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.background =</span> <span class="fu">element_blank</span>()</span>
<span id="cb40-857"><a href="#cb40-857" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb40-858"><a href="#cb40-858" aria-hidden="true" tabindex="-1"></a>  theme_kit <span class="sc">+</span></span>
<span id="cb40-859"><a href="#cb40-859" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>()  <span class="co"># This flips the coordinates to make the plot horizontal</span></span>
<span id="cb40-860"><a href="#cb40-860" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-861"><a href="#cb40-861" aria-hidden="true" tabindex="-1"></a>g_viral</span>
<span id="cb40-862"><a href="#cb40-862" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-863"><a href="#cb40-863" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-864"><a href="#cb40-864" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-865"><a href="#cb40-865" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-866"><a href="#cb40-866" aria-hidden="true" tabindex="-1"></a><span class="fu">## Overall taxonomy and composition</span></span>
<span id="cb40-867"><a href="#cb40-867" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-868"><a href="#cb40-868" aria-hidden="true" tabindex="-1"></a>The two dominant viruses we see are Anellovirdae and Hepadnaviridae, with Parvoviridae . The threshold for the label "other" are the set of families that make up less than 5% composition in all samples.</span>
<span id="cb40-869"><a href="#cb40-869" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-872"><a href="#cb40-872" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb40-873"><a href="#cb40-873" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: raise-hv-taxa</span></span>
<span id="cb40-874"><a href="#cb40-874" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb40-875"><a href="#cb40-875" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-876"><a href="#cb40-876" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter samples and add viral taxa information</span></span>
<span id="cb40-877"><a href="#cb40-877" aria-hidden="true" tabindex="-1"></a><span class="co">#samples_keep &lt;- read_counts %&gt;% filter(n_reads_hv &gt; 1) %&gt;% pull(sample)</span></span>
<span id="cb40-878"><a href="#cb40-878" aria-hidden="true" tabindex="-1"></a>samples_keep <span class="ot">&lt;-</span> read_counts <span class="sc">%&gt;%</span>  <span class="fu">pull</span>(sample)</span>
<span id="cb40-879"><a href="#cb40-879" aria-hidden="true" tabindex="-1"></a>mrg_hv_named <span class="ot">&lt;-</span> mrg_hv <span class="sc">%&gt;%</span> <span class="fu">filter</span>(sample <span class="sc">%in%</span> samples_keep, hv_status) <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(viral_taxa, <span class="at">by=</span><span class="st">"taxid"</span>) </span>
<span id="cb40-880"><a href="#cb40-880" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-881"><a href="#cb40-881" aria-hidden="true" tabindex="-1"></a><span class="co"># Discover viral species &amp; genera for HV reads</span></span>
<span id="cb40-882"><a href="#cb40-882" aria-hidden="true" tabindex="-1"></a>raise_rank <span class="ot">&lt;-</span> <span class="cf">function</span>(read_db, taxid_db, <span class="at">out_rank =</span> <span class="st">"species"</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>){</span>
<span id="cb40-883"><a href="#cb40-883" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get higher ranks than search rank</span></span>
<span id="cb40-884"><a href="#cb40-884" aria-hidden="true" tabindex="-1"></a>  ranks <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"subspecies"</span>, <span class="st">"species"</span>, <span class="st">"subgenus"</span>, <span class="st">"genus"</span>, <span class="st">"subfamily"</span>, <span class="st">"family"</span>, <span class="st">"suborder"</span>, <span class="st">"order"</span>, <span class="st">"class"</span>, <span class="st">"subphylum"</span>, <span class="st">"phylum"</span>, <span class="st">"kingdom"</span>, <span class="st">"superkingdom"</span>)</span>
<span id="cb40-885"><a href="#cb40-885" aria-hidden="true" tabindex="-1"></a>  rank_match <span class="ot">&lt;-</span> <span class="fu">which.max</span>(ranks <span class="sc">==</span> out_rank)</span>
<span id="cb40-886"><a href="#cb40-886" aria-hidden="true" tabindex="-1"></a>  high_ranks <span class="ot">&lt;-</span> ranks[rank_match<span class="sc">:</span><span class="fu">length</span>(ranks)]</span>
<span id="cb40-887"><a href="#cb40-887" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Merge read DB and taxid DB</span></span>
<span id="cb40-888"><a href="#cb40-888" aria-hidden="true" tabindex="-1"></a>  reads <span class="ot">&lt;-</span> read_db <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>parent_taxid, <span class="sc">-</span>rank, <span class="sc">-</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb40-889"><a href="#cb40-889" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(taxid_db, <span class="at">by=</span><span class="st">"taxid"</span>)</span>
<span id="cb40-890"><a href="#cb40-890" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract sequences that are already at appropriate rank</span></span>
<span id="cb40-891"><a href="#cb40-891" aria-hidden="true" tabindex="-1"></a>  reads_rank <span class="ot">&lt;-</span> <span class="fu">filter</span>(reads, rank <span class="sc">==</span> out_rank)</span>
<span id="cb40-892"><a href="#cb40-892" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Drop sequences at a higher rank and return unclassified sequences</span></span>
<span id="cb40-893"><a href="#cb40-893" aria-hidden="true" tabindex="-1"></a>  reads_norank <span class="ot">&lt;-</span> reads <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">!=</span> out_rank, <span class="sc">!</span>rank <span class="sc">%in%</span> high_ranks, <span class="sc">!</span><span class="fu">is.na</span>(taxid))</span>
<span id="cb40-894"><a href="#cb40-894" aria-hidden="true" tabindex="-1"></a>  <span class="cf">while</span>(<span class="fu">nrow</span>(reads_norank) <span class="sc">&gt;</span> <span class="dv">0</span>){ <span class="co"># As long as there are unclassified sequences...</span></span>
<span id="cb40-895"><a href="#cb40-895" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Promote read taxids and re-merge with taxid DB, then re-classify and filter</span></span>
<span id="cb40-896"><a href="#cb40-896" aria-hidden="true" tabindex="-1"></a>    reads_remaining <span class="ot">&lt;-</span> reads_norank <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">taxid =</span> parent_taxid) <span class="sc">%&gt;%</span></span>
<span id="cb40-897"><a href="#cb40-897" aria-hidden="true" tabindex="-1"></a>      <span class="fu">select</span>(<span class="sc">-</span>parent_taxid, <span class="sc">-</span>rank, <span class="sc">-</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb40-898"><a href="#cb40-898" aria-hidden="true" tabindex="-1"></a>      <span class="fu">left_join</span>(taxid_db, <span class="at">by=</span><span class="st">"taxid"</span>)</span>
<span id="cb40-899"><a href="#cb40-899" aria-hidden="true" tabindex="-1"></a>    reads_rank <span class="ot">&lt;-</span> reads_remaining <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">==</span> out_rank) <span class="sc">%&gt;%</span></span>
<span id="cb40-900"><a href="#cb40-900" aria-hidden="true" tabindex="-1"></a>      <span class="fu">bind_rows</span>(reads_rank)</span>
<span id="cb40-901"><a href="#cb40-901" aria-hidden="true" tabindex="-1"></a>    reads_norank <span class="ot">&lt;-</span> reads_remaining <span class="sc">%&gt;%</span></span>
<span id="cb40-902"><a href="#cb40-902" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(rank <span class="sc">!=</span> out_rank, <span class="sc">!</span>rank <span class="sc">%in%</span> high_ranks, <span class="sc">!</span><span class="fu">is.na</span>(taxid))</span>
<span id="cb40-903"><a href="#cb40-903" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb40-904"><a href="#cb40-904" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Finally, extract and append reads that were excluded during the process</span></span>
<span id="cb40-905"><a href="#cb40-905" aria-hidden="true" tabindex="-1"></a>  reads_dropped <span class="ot">&lt;-</span> reads <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span>seq_id <span class="sc">%in%</span> reads_rank<span class="sc">$</span>seq_id)</span>
<span id="cb40-906"><a href="#cb40-906" aria-hidden="true" tabindex="-1"></a>  reads_out <span class="ot">&lt;-</span> reads_rank <span class="sc">%&gt;%</span> <span class="fu">bind_rows</span>(reads_dropped) <span class="sc">%&gt;%</span></span>
<span id="cb40-907"><a href="#cb40-907" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(<span class="sc">-</span>parent_taxid, <span class="sc">-</span>rank, <span class="sc">-</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb40-908"><a href="#cb40-908" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(taxid_db, <span class="at">by=</span><span class="st">"taxid"</span>)</span>
<span id="cb40-909"><a href="#cb40-909" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(reads_out)</span>
<span id="cb40-910"><a href="#cb40-910" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb40-911"><a href="#cb40-911" aria-hidden="true" tabindex="-1"></a>hv_reads_species <span class="ot">&lt;-</span> <span class="fu">raise_rank</span>(mrg_hv_named, viral_taxa, <span class="st">"species"</span>)</span>
<span id="cb40-912"><a href="#cb40-912" aria-hidden="true" tabindex="-1"></a>hv_reads_genus <span class="ot">&lt;-</span> <span class="fu">raise_rank</span>(mrg_hv_named, viral_taxa, <span class="st">"genus"</span>)</span>
<span id="cb40-913"><a href="#cb40-913" aria-hidden="true" tabindex="-1"></a>hv_reads_family <span class="ot">&lt;-</span> <span class="fu">raise_rank</span>(mrg_hv_named, viral_taxa, <span class="st">"family"</span>)</span>
<span id="cb40-914"><a href="#cb40-914" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-915"><a href="#cb40-915" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-918"><a href="#cb40-918" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb40-919"><a href="#cb40-919" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-family</span></span>
<span id="cb40-920"><a href="#cb40-920" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 15</span></span>
<span id="cb40-921"><a href="#cb40-921" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 15</span></span>
<span id="cb40-922"><a href="#cb40-922" aria-hidden="true" tabindex="-1"></a>threshold_major_family <span class="ot">&lt;-</span> <span class="fl">0.05</span></span>
<span id="cb40-923"><a href="#cb40-923" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-924"><a href="#cb40-924" aria-hidden="true" tabindex="-1"></a><span class="co"># Count reads for each human-viral family</span></span>
<span id="cb40-925"><a href="#cb40-925" aria-hidden="true" tabindex="-1"></a>hv_family_counts <span class="ot">&lt;-</span> hv_reads_family <span class="sc">%&gt;%</span> </span>
<span id="cb40-926"><a href="#cb40-926" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb40-927"><a href="#cb40-927" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-928"><a href="#cb40-928" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb40-929"><a href="#cb40-929" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_hv =</span> n_reads_hv<span class="sc">/</span><span class="fu">sum</span>(n_reads_hv))</span>
<span id="cb40-930"><a href="#cb40-930" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-931"><a href="#cb40-931" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify high-ranking families and group others</span></span>
<span id="cb40-932"><a href="#cb40-932" aria-hidden="true" tabindex="-1"></a>hv_family_major_tab <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span> </span>
<span id="cb40-933"><a href="#cb40-933" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_hv <span class="sc">==</span> <span class="fu">max</span>(p_reads_hv)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-934"><a href="#cb40-934" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_hv)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(p_reads_hv <span class="sc">&gt;</span> threshold_major_family)</span>
<span id="cb40-935"><a href="#cb40-935" aria-hidden="true" tabindex="-1"></a>hv_family_counts_major <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span></span>
<span id="cb40-936"><a href="#cb40-936" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">ifelse</span>(name <span class="sc">%in%</span> hv_family_major_tab<span class="sc">$</span>name, name, <span class="st">"Other"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb40-937"><a href="#cb40-937" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, name_display) <span class="sc">%&gt;%</span></span>
<span id="cb40-938"><a href="#cb40-938" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_hv =</span> <span class="fu">sum</span>(n_reads_hv), <span class="at">p_reads_hv =</span> <span class="fu">sum</span>(p_reads_hv), </span>
<span id="cb40-939"><a href="#cb40-939" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-940"><a href="#cb40-940" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">factor</span>(name_display, </span>
<span id="cb40-941"><a href="#cb40-941" aria-hidden="true" tabindex="-1"></a>                               <span class="at">levels =</span> <span class="fu">c</span>(hv_family_major_tab<span class="sc">$</span>name, <span class="st">"Other"</span>)))</span>
<span id="cb40-942"><a href="#cb40-942" aria-hidden="true" tabindex="-1"></a>hv_family_counts_display <span class="ot">&lt;-</span> hv_family_counts_major <span class="sc">%&gt;%</span></span>
<span id="cb40-943"><a href="#cb40-943" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_hv, <span class="at">classification =</span> name_display) <span class="sc">%&gt;%</span></span>
<span id="cb40-944"><a href="#cb40-944" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by =</span> <span class="st">'sample'</span>)</span>
<span id="cb40-945"><a href="#cb40-945" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-946"><a href="#cb40-946" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb40-947"><a href="#cb40-947" aria-hidden="true" tabindex="-1"></a>g_hv_family <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb40-948"><a href="#cb40-948" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">data=</span>hv_family_counts_display, <span class="at">position =</span> <span class="st">"stack"</span>, <span class="at">width=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb40-949"><a href="#cb40-949" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% HV Reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb40-950"><a href="#cb40-950" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb40-951"><a href="#cb40-951" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb40-952"><a href="#cb40-952" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>, <span class="at">name =</span> <span class="st">"Viral family"</span>) <span class="sc">+</span></span>
<span id="cb40-953"><a href="#cb40-953" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-954"><a href="#cb40-954" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">"Family composition of human-viral reads"</span>) <span class="sc">+</span></span>
<span id="cb40-955"><a href="#cb40-955" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">4</span>)) <span class="sc">+</span></span>
<span id="cb40-956"><a href="#cb40-956" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">1.4</span>), <span class="at">hjust=</span><span class="dv">0</span>, <span class="at">face=</span><span class="st">"plain"</span>)) <span class="sc">+</span>  </span>
<span id="cb40-957"><a href="#cb40-957" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span> location, <span class="at">scales =</span> <span class="st">"free_x"</span>, <span class="at">ncol =</span> <span class="dv">2</span>) </span>
<span id="cb40-958"><a href="#cb40-958" aria-hidden="true" tabindex="-1"></a>g_hv_family</span>
<span id="cb40-959"><a href="#cb40-959" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-960"><a href="#cb40-960" aria-hidden="true" tabindex="-1"></a><span class="co"># Get most prominent families for text</span></span>
<span id="cb40-961"><a href="#cb40-961" aria-hidden="true" tabindex="-1"></a>hv_family_collate <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span></span>
<span id="cb40-962"><a href="#cb40-962" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb40-963"><a href="#cb40-963" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_tot =</span> <span class="fu">sum</span>(n_reads_hv),</span>
<span id="cb40-964"><a href="#cb40-964" aria-hidden="true" tabindex="-1"></a>            <span class="at">p_reads_max =</span> <span class="fu">max</span>(p_reads_hv), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb40-965"><a href="#cb40-965" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n_reads_tot))</span>
<span id="cb40-966"><a href="#cb40-966" aria-hidden="true" tabindex="-1"></a>display_hv_family_collate <span class="ot">&lt;-</span> hv_family_collate <span class="sc">%&gt;%</span></span>
<span id="cb40-967"><a href="#cb40-967" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, n_reads_tot) <span class="sc">%&gt;%</span></span>
<span id="cb40-968"><a href="#cb40-968" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">`</span><span class="at">Family</span><span class="st">`</span> <span class="ot">=</span> name, <span class="st">`</span><span class="at">Total number of reads</span><span class="st">`</span> <span class="ot">=</span> n_reads_tot)</span>
<span id="cb40-969"><a href="#cb40-969" aria-hidden="true" tabindex="-1"></a>display_hv_family_collate</span>
<span id="cb40-970"><a href="#cb40-970" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-971"><a href="#cb40-971" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-972"><a href="#cb40-972" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-973"><a href="#cb40-973" aria-hidden="true" tabindex="-1"></a><span class="fu">## Analyzing specific families</span></span>
<span id="cb40-974"><a href="#cb40-974" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-975"><a href="#cb40-975" aria-hidden="true" tabindex="-1"></a>We now investigate the composition of specific families that had more than 5 viral reads. In investigating individual viral families, to avoid distortions from a few rare reads, I restricted myself to samples where that family made up at least 10% of human-viral reads:</span>
<span id="cb40-976"><a href="#cb40-976" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-979"><a href="#cb40-979" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb40-980"><a href="#cb40-980" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-viral-family-scripts</span></span>
<span id="cb40-981"><a href="#cb40-981" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb40-982"><a href="#cb40-982" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-983"><a href="#cb40-983" aria-hidden="true" tabindex="-1"></a>library_with_total <span class="ot">&lt;-</span> libraries <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(location) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">total_samples =</span> <span class="fu">n_distinct</span>(sample)) <span class="sc">%&gt;%</span> <span class="fu">ungroup</span>()</span>
<span id="cb40-984"><a href="#cb40-984" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-985"><a href="#cb40-985" aria-hidden="true" tabindex="-1"></a>plot_viral_family_composition <span class="ot">&lt;-</span> <span class="cf">function</span>(taxid_chosen, <span class="at">threshold_major_species =</span> <span class="fl">0.1</span>) {</span>
<span id="cb40-986"><a href="#cb40-986" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get set of chosen family reads</span></span>
<span id="cb40-987"><a href="#cb40-987" aria-hidden="true" tabindex="-1"></a>  family_samples <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span> <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_chosen) <span class="sc">%&gt;%</span></span>
<span id="cb40-988"><a href="#cb40-988" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(p_reads_hv <span class="sc">&gt;=</span> <span class="fl">0.1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-989"><a href="#cb40-989" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pull</span>(sample)</span>
<span id="cb40-990"><a href="#cb40-990" aria-hidden="true" tabindex="-1"></a>  family_ids <span class="ot">&lt;-</span> hv_reads_family <span class="sc">%&gt;%</span> </span>
<span id="cb40-991"><a href="#cb40-991" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_chosen, sample <span class="sc">%in%</span> family_samples) <span class="sc">%&gt;%</span></span>
<span id="cb40-992"><a href="#cb40-992" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pull</span>(seq_id)</span>
<span id="cb40-993"><a href="#cb40-993" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-994"><a href="#cb40-994" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Count reads for each species in the chosen family</span></span>
<span id="cb40-995"><a href="#cb40-995" aria-hidden="true" tabindex="-1"></a>  species_counts <span class="ot">&lt;-</span> hv_reads_species <span class="sc">%&gt;%</span></span>
<span id="cb40-996"><a href="#cb40-996" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(seq_id <span class="sc">%in%</span> family_ids) <span class="sc">%&gt;%</span></span>
<span id="cb40-997"><a href="#cb40-997" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(sample, location, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb40-998"><a href="#cb40-998" aria-hidden="true" tabindex="-1"></a>    <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-999"><a href="#cb40-999" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(sample, location) <span class="sc">%&gt;%</span></span>
<span id="cb40-1000"><a href="#cb40-1000" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">p_reads_family =</span> n_reads_hv<span class="sc">/</span><span class="fu">sum</span>(n_reads_hv))</span>
<span id="cb40-1001"><a href="#cb40-1001" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1002"><a href="#cb40-1002" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Identify high-ranking species and group others</span></span>
<span id="cb40-1003"><a href="#cb40-1003" aria-hidden="true" tabindex="-1"></a>  species_major_tab <span class="ot">&lt;-</span> species_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span> </span>
<span id="cb40-1004"><a href="#cb40-1004" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(p_reads_family <span class="sc">==</span> <span class="fu">max</span>(p_reads_family)) <span class="sc">%&gt;%</span> </span>
<span id="cb40-1005"><a href="#cb40-1005" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-1006"><a href="#cb40-1006" aria-hidden="true" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_family)) <span class="sc">%&gt;%</span> </span>
<span id="cb40-1007"><a href="#cb40-1007" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(p_reads_family <span class="sc">&gt;</span> threshold_major_species)</span>
<span id="cb40-1008"><a href="#cb40-1008" aria-hidden="true" tabindex="-1"></a>  species_counts_major <span class="ot">&lt;-</span> species_counts <span class="sc">%&gt;%</span></span>
<span id="cb40-1009"><a href="#cb40-1009" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">ifelse</span>(name <span class="sc">%in%</span> species_major_tab<span class="sc">$</span>name, </span>
<span id="cb40-1010"><a href="#cb40-1010" aria-hidden="true" tabindex="-1"></a>                                 name, <span class="st">"Other"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb40-1011"><a href="#cb40-1011" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(sample, location, name_display) <span class="sc">%&gt;%</span></span>
<span id="cb40-1012"><a href="#cb40-1012" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarize</span>(<span class="at">n_reads_family =</span> <span class="fu">sum</span>(n_reads_hv),</span>
<span id="cb40-1013"><a href="#cb40-1013" aria-hidden="true" tabindex="-1"></a>              <span class="at">p_reads_family =</span> <span class="fu">sum</span>(p_reads_family), </span>
<span id="cb40-1014"><a href="#cb40-1014" aria-hidden="true" tabindex="-1"></a>              <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-1015"><a href="#cb40-1015" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">factor</span>(name_display, </span>
<span id="cb40-1016"><a href="#cb40-1016" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">levels =</span> <span class="fu">c</span>(species_major_tab<span class="sc">$</span>name, <span class="st">"Other"</span>)))</span>
<span id="cb40-1017"><a href="#cb40-1017" aria-hidden="true" tabindex="-1"></a>  species_counts_display <span class="ot">&lt;-</span> species_counts_major <span class="sc">%&gt;%</span></span>
<span id="cb40-1018"><a href="#cb40-1018" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_family, <span class="at">classification =</span> name_display) <span class="sc">%&gt;%</span> </span>
<span id="cb40-1019"><a href="#cb40-1019" aria-hidden="true" tabindex="-1"></a>    <span class="fu">right_join</span>(library_with_total, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">'sample'</span>, <span class="st">'location'</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb40-1020"><a href="#cb40-1020" aria-hidden="true" tabindex="-1"></a>    <span class="co">#mutate(p_reads = coalesce(p_reads, 0),</span></span>
<span id="cb40-1021"><a href="#cb40-1021" aria-hidden="true" tabindex="-1"></a>    <span class="co">#       classification = coalesce(classification, species_major_tab$name[1])) %&gt;%</span></span>
<span id="cb40-1022"><a href="#cb40-1022" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">location =</span> <span class="fu">factor</span>(location))</span>
<span id="cb40-1023"><a href="#cb40-1023" aria-hidden="true" tabindex="-1"></a>    <span class="co">#group_by(location) %&gt;%</span></span>
<span id="cb40-1024"><a href="#cb40-1024" aria-hidden="true" tabindex="-1"></a>    <span class="co">#mutate(samples_with_reads = sum(p_reads &gt; 0),</span></span>
<span id="cb40-1025"><a href="#cb40-1025" aria-hidden="true" tabindex="-1"></a>    <span class="co">#  label = paste0(location, " (", sprintf("%.1f%%", 100 * samples_with_reads / total_samples), ")")) %&gt;%</span></span>
<span id="cb40-1026"><a href="#cb40-1026" aria-hidden="true" tabindex="-1"></a>    <span class="co">#ungroup()</span></span>
<span id="cb40-1027"><a href="#cb40-1027" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1028"><a href="#cb40-1028" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get family name for the title</span></span>
<span id="cb40-1029"><a href="#cb40-1029" aria-hidden="true" tabindex="-1"></a>  family_name <span class="ot">&lt;-</span> viral_taxa <span class="sc">%&gt;%</span> </span>
<span id="cb40-1030"><a href="#cb40-1030" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_chosen) <span class="sc">%&gt;%</span> </span>
<span id="cb40-1031"><a href="#cb40-1031" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pull</span>(name)</span>
<span id="cb40-1032"><a href="#cb40-1032" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1033"><a href="#cb40-1033" aria-hidden="true" tabindex="-1"></a>  species_count <span class="ot">&lt;-</span> species_counts_display <span class="sc">%&gt;%</span> <span class="fu">select</span>(classification) <span class="sc">%&gt;%</span> <span class="fu">distinct</span>() <span class="sc">%&gt;%</span> <span class="fu">pull</span>() <span class="sc">%&gt;%</span> <span class="fu">length</span>()</span>
<span id="cb40-1034"><a href="#cb40-1034" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (species_count <span class="sc">&lt;=</span> <span class="dv">8</span>) {</span>
<span id="cb40-1035"><a href="#cb40-1035" aria-hidden="true" tabindex="-1"></a>    pal <span class="ot">&lt;-</span> <span class="fu">brewer.pal</span>(<span class="at">name =</span> <span class="st">"Accent"</span>, <span class="at">n =</span> species_count)</span>
<span id="cb40-1036"><a href="#cb40-1036" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb40-1037"><a href="#cb40-1037" aria-hidden="true" tabindex="-1"></a>    pal <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">brewer.pal</span>(<span class="at">name =</span> <span class="st">"Paired"</span>, <span class="at">n =</span> <span class="dv">12</span>), <span class="fu">brewer.pal</span>(<span class="at">name =</span> <span class="st">"Set3"</span>, <span class="at">n =</span><span class="dv">12</span>), <span class="fu">brewer.pal</span>(<span class="at">name =</span> <span class="st">"Set1"</span>, <span class="at">n =</span> <span class="dv">9</span>))[<span class="dv">1</span><span class="sc">:</span>species_count]</span>
<span id="cb40-1038"><a href="#cb40-1038" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb40-1039"><a href="#cb40-1039" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1040"><a href="#cb40-1040" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Plot</span></span>
<span id="cb40-1041"><a href="#cb40-1041" aria-hidden="true" tabindex="-1"></a>  g_species <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb40-1042"><a href="#cb40-1042" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_col</span>(<span class="at">data=</span>species_counts_display, <span class="at">position =</span> <span class="st">"stack"</span>) <span class="sc">+</span></span>
<span id="cb40-1043"><a href="#cb40-1043" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="fu">paste0</span>(<span class="st">"% "</span>, family_name, <span class="st">" reads"</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb40-1044"><a href="#cb40-1044" aria-hidden="true" tabindex="-1"></a>                       <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb40-1045"><a href="#cb40-1045" aria-hidden="true" tabindex="-1"></a>                       <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb40-1046"><a href="#cb40-1046" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> pal, <span class="at">name =</span> <span class="st">"Viral species"</span>) <span class="sc">+</span></span>
<span id="cb40-1047"><a href="#cb40-1047" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title=</span><span class="fu">paste0</span>(<span class="st">"Species composition of "</span>, family_name, <span class="st">" reads"</span>)) <span class="sc">+</span></span>
<span id="cb40-1048"><a href="#cb40-1048" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_wrap</span>(<span class="sc">~</span> location, <span class="at">scales =</span> <span class="st">"free_x"</span>, <span class="at">ncol =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb40-1049"><a href="#cb40-1049" aria-hidden="true" tabindex="-1"></a>    <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">3</span>)) <span class="sc">+</span></span>
<span id="cb40-1050"><a href="#cb40-1050" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">1.4</span>), <span class="at">hjust=</span><span class="dv">0</span>, <span class="at">face=</span><span class="st">"plain"</span>))</span>
<span id="cb40-1051"><a href="#cb40-1051" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(g_species)</span>
<span id="cb40-1052"><a href="#cb40-1052" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb40-1053"><a href="#cb40-1053" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1054"><a href="#cb40-1054" aria-hidden="true" tabindex="-1"></a>plot_viral_family_histogram <span class="ot">&lt;-</span> <span class="cf">function</span>(taxid_chosen) {</span>
<span id="cb40-1055"><a href="#cb40-1055" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filter data for the specified virus family</span></span>
<span id="cb40-1056"><a href="#cb40-1056" aria-hidden="true" tabindex="-1"></a>  family_data <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span></span>
<span id="cb40-1057"><a href="#cb40-1057" aria-hidden="true" tabindex="-1"></a>    <span class="fu">right_join</span>(library_with_total, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">'sample'</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb40-1058"><a href="#cb40-1058" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb40-1059"><a href="#cb40-1059" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">n_reads_hv =</span> <span class="fu">coalesce</span>(n_reads_hv, <span class="dv">0</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb40-1060"><a href="#cb40-1060" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_chosen) <span class="sc">%&gt;%</span></span>
<span id="cb40-1061"><a href="#cb40-1061" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(location) <span class="sc">%&gt;%</span></span>
<span id="cb40-1062"><a href="#cb40-1062" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarize</span>(<span class="at">n_reads_hv =</span> <span class="fu">sum</span>(n_reads_hv), <span class="at">.groups=</span><span class="st">"drop"</span>)</span>
<span id="cb40-1063"><a href="#cb40-1063" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb40-1064"><a href="#cb40-1064" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get family name for the title</span></span>
<span id="cb40-1065"><a href="#cb40-1065" aria-hidden="true" tabindex="-1"></a>  family_name <span class="ot">&lt;-</span> viral_taxa <span class="sc">%&gt;%</span> </span>
<span id="cb40-1066"><a href="#cb40-1066" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_chosen) <span class="sc">%&gt;%</span> </span>
<span id="cb40-1067"><a href="#cb40-1067" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pull</span>(name)</span>
<span id="cb40-1068"><a href="#cb40-1068" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb40-1069"><a href="#cb40-1069" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create the histogram</span></span>
<span id="cb40-1070"><a href="#cb40-1070" aria-hidden="true" tabindex="-1"></a>  g_histogram <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(family_data, <span class="fu">aes</span>(<span class="at">x =</span> location, <span class="at">y =</span> n_reads_hv)) <span class="sc">+</span></span>
<span id="cb40-1071"><a href="#cb40-1071" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb40-1072"><a href="#cb40-1072" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label =</span> scales<span class="sc">::</span><span class="fu">comma</span>(n_reads_hv)), <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb40-1073"><a href="#cb40-1073" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb40-1074"><a href="#cb40-1074" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb40-1075"><a href="#cb40-1075" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(</span>
<span id="cb40-1076"><a href="#cb40-1076" aria-hidden="true" tabindex="-1"></a>      <span class="at">title =</span> <span class="fu">paste</span>(<span class="st">"Distribution of"</span>, family_name, <span class="st">"Reads by Location"</span>),</span>
<span id="cb40-1077"><a href="#cb40-1077" aria-hidden="true" tabindex="-1"></a>      <span class="at">x =</span> <span class="st">"Location"</span>,</span>
<span id="cb40-1078"><a href="#cb40-1078" aria-hidden="true" tabindex="-1"></a>      <span class="at">y =</span> <span class="st">"Number of reads"</span></span>
<span id="cb40-1079"><a href="#cb40-1079" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb40-1080"><a href="#cb40-1080" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_continuous</span>(<span class="at">labels =</span> scales<span class="sc">::</span>comma) <span class="sc">+</span></span>
<span id="cb40-1081"><a href="#cb40-1081" aria-hidden="true" tabindex="-1"></a>    <span class="fu">expand_limits</span>(<span class="at">y =</span> <span class="fu">max</span>(family_data<span class="sc">$</span>n_reads_hv) <span class="sc">*</span> <span class="fl">1.1</span>)<span class="sc">+</span></span>
<span id="cb40-1082"><a href="#cb40-1082" aria-hidden="true" tabindex="-1"></a>    theme_kit  <span class="co"># Expand y-axis to make room for labels</span></span>
<span id="cb40-1083"><a href="#cb40-1083" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(g_histogram)</span>
<span id="cb40-1084"><a href="#cb40-1084" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb40-1085"><a href="#cb40-1085" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1086"><a href="#cb40-1086" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1087"><a href="#cb40-1087" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-1088"><a href="#cb40-1088" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1089"><a href="#cb40-1089" aria-hidden="true" tabindex="-1"></a><span class="fu">### Hepadnaviridae (Number of reads: 339,427)</span></span>
<span id="cb40-1092"><a href="#cb40-1092" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb40-1093"><a href="#cb40-1093" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-hepadnaviridae-histogram</span></span>
<span id="cb40-1094"><a href="#cb40-1094" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb40-1095"><a href="#cb40-1095" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 2.5</span></span>
<span id="cb40-1096"><a href="#cb40-1096" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1097"><a href="#cb40-1097" aria-hidden="true" tabindex="-1"></a><span class="co"># Hepadnaviridae</span></span>
<span id="cb40-1098"><a href="#cb40-1098" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_histogram</span>(<span class="at">taxid_chosen=</span><span class="dv">10404</span>)</span>
<span id="cb40-1099"><a href="#cb40-1099" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-1102"><a href="#cb40-1102" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb40-1103"><a href="#cb40-1103" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-hepadnaviridae-composition</span></span>
<span id="cb40-1104"><a href="#cb40-1104" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb40-1105"><a href="#cb40-1105" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 7.5</span></span>
<span id="cb40-1106"><a href="#cb40-1106" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb40-1107"><a href="#cb40-1107" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb40-1108"><a href="#cb40-1108" aria-hidden="true" tabindex="-1"></a><span class="co"># Hepadnaviridae</span></span>
<span id="cb40-1109"><a href="#cb40-1109" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_composition</span>(<span class="at">taxid_chosen=</span><span class="dv">10404</span>, <span class="at">threshold_major_species =</span> <span class="fl">0.1</span>)</span>
<span id="cb40-1110"><a href="#cb40-1110" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-1111"><a href="#cb40-1111" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1112"><a href="#cb40-1112" aria-hidden="true" tabindex="-1"></a><span class="fu">### Parvoviridae (Number of reads: 17,401)</span></span>
<span id="cb40-1115"><a href="#cb40-1115" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb40-1116"><a href="#cb40-1116" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-parvoviridae-histogram</span></span>
<span id="cb40-1117"><a href="#cb40-1117" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb40-1118"><a href="#cb40-1118" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 2.5</span></span>
<span id="cb40-1119"><a href="#cb40-1119" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1120"><a href="#cb40-1120" aria-hidden="true" tabindex="-1"></a><span class="co"># Parvoviridae</span></span>
<span id="cb40-1121"><a href="#cb40-1121" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_histogram</span>(<span class="at">taxid_chosen=</span><span class="dv">10780</span>)</span>
<span id="cb40-1122"><a href="#cb40-1122" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-1125"><a href="#cb40-1125" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb40-1126"><a href="#cb40-1126" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-parvoviridae-composition</span></span>
<span id="cb40-1127"><a href="#cb40-1127" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb40-1128"><a href="#cb40-1128" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 7.5</span></span>
<span id="cb40-1129"><a href="#cb40-1129" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb40-1130"><a href="#cb40-1130" aria-hidden="true" tabindex="-1"></a><span class="co"># Parvoviridae</span></span>
<span id="cb40-1131"><a href="#cb40-1131" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_composition</span>(<span class="at">taxid_chosen=</span><span class="dv">10780</span>, <span class="at">threshold_major_species =</span> <span class="fl">0.1</span>)</span>
<span id="cb40-1132"><a href="#cb40-1132" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-1133"><a href="#cb40-1133" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1134"><a href="#cb40-1134" aria-hidden="true" tabindex="-1"></a><span class="fu">### Anelloviridae (Number of reads: 3001)</span></span>
<span id="cb40-1137"><a href="#cb40-1137" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb40-1138"><a href="#cb40-1138" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-anelloviridae-histogram</span></span>
<span id="cb40-1139"><a href="#cb40-1139" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb40-1140"><a href="#cb40-1140" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 2.5</span></span>
<span id="cb40-1141"><a href="#cb40-1141" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1142"><a href="#cb40-1142" aria-hidden="true" tabindex="-1"></a><span class="co"># Anelloviridae</span></span>
<span id="cb40-1143"><a href="#cb40-1143" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_histogram</span>(<span class="at">taxid_chosen=</span><span class="dv">687329</span>)</span>
<span id="cb40-1144"><a href="#cb40-1144" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-1147"><a href="#cb40-1147" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb40-1148"><a href="#cb40-1148" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-anelloviridae-composition</span></span>
<span id="cb40-1149"><a href="#cb40-1149" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb40-1150"><a href="#cb40-1150" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 7.5</span></span>
<span id="cb40-1151"><a href="#cb40-1151" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb40-1152"><a href="#cb40-1152" aria-hidden="true" tabindex="-1"></a><span class="co"># Anelloviridae</span></span>
<span id="cb40-1153"><a href="#cb40-1153" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_composition</span>(<span class="at">taxid_chosen=</span><span class="dv">687329</span>, <span class="at">threshold_major_species =</span> <span class="fl">0.1</span>)</span>
<span id="cb40-1154"><a href="#cb40-1154" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-1155"><a href="#cb40-1155" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1156"><a href="#cb40-1156" aria-hidden="true" tabindex="-1"></a><span class="fu">### Orthoherpesviridae (Number of reads: 298)</span></span>
<span id="cb40-1159"><a href="#cb40-1159" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb40-1160"><a href="#cb40-1160" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-orthoherpesviridae-histogram</span></span>
<span id="cb40-1161"><a href="#cb40-1161" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb40-1162"><a href="#cb40-1162" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 2.5</span></span>
<span id="cb40-1163"><a href="#cb40-1163" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1164"><a href="#cb40-1164" aria-hidden="true" tabindex="-1"></a><span class="co"># Orthoherpesviridae</span></span>
<span id="cb40-1165"><a href="#cb40-1165" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_histogram</span>(<span class="at">taxid_chosen=</span><span class="dv">3044472</span>)</span>
<span id="cb40-1166"><a href="#cb40-1166" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-1169"><a href="#cb40-1169" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb40-1170"><a href="#cb40-1170" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-orthoherpesviridae-composition</span></span>
<span id="cb40-1171"><a href="#cb40-1171" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb40-1172"><a href="#cb40-1172" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 7.5</span></span>
<span id="cb40-1173"><a href="#cb40-1173" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb40-1174"><a href="#cb40-1174" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb40-1175"><a href="#cb40-1175" aria-hidden="true" tabindex="-1"></a><span class="co"># Orthoherpesviridae</span></span>
<span id="cb40-1176"><a href="#cb40-1176" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_composition</span>(<span class="at">taxid_chosen=</span><span class="dv">3044472</span>, <span class="at">threshold_major_species =</span> <span class="fl">0.1</span>)</span>
<span id="cb40-1177"><a href="#cb40-1177" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-1178"><a href="#cb40-1178" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1179"><a href="#cb40-1179" aria-hidden="true" tabindex="-1"></a><span class="fu">### Retroviridae (Number of reads: 115)</span></span>
<span id="cb40-1182"><a href="#cb40-1182" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb40-1183"><a href="#cb40-1183" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-retroviridae-histogram</span></span>
<span id="cb40-1184"><a href="#cb40-1184" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb40-1185"><a href="#cb40-1185" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 2.5</span></span>
<span id="cb40-1186"><a href="#cb40-1186" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1187"><a href="#cb40-1187" aria-hidden="true" tabindex="-1"></a><span class="co"># Retroviridae</span></span>
<span id="cb40-1188"><a href="#cb40-1188" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_histogram</span>(<span class="at">taxid_chosen=</span><span class="dv">11632</span>)</span>
<span id="cb40-1189"><a href="#cb40-1189" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-1192"><a href="#cb40-1192" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb40-1193"><a href="#cb40-1193" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-retroviridae-composition</span></span>
<span id="cb40-1194"><a href="#cb40-1194" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb40-1195"><a href="#cb40-1195" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 7.5</span></span>
<span id="cb40-1196"><a href="#cb40-1196" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb40-1197"><a href="#cb40-1197" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb40-1198"><a href="#cb40-1198" aria-hidden="true" tabindex="-1"></a><span class="co"># Retroviridae</span></span>
<span id="cb40-1199"><a href="#cb40-1199" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_composition</span>(<span class="at">taxid_chosen=</span><span class="dv">11632</span>, <span class="at">threshold_major_species =</span> <span class="fl">0.1</span>)</span>
<span id="cb40-1200"><a href="#cb40-1200" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-1201"><a href="#cb40-1201" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1202"><a href="#cb40-1202" aria-hidden="true" tabindex="-1"></a><span class="fu">### Microviridae (Number of reads: 61)</span></span>
<span id="cb40-1205"><a href="#cb40-1205" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb40-1206"><a href="#cb40-1206" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-microviridae-histogram</span></span>
<span id="cb40-1207"><a href="#cb40-1207" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb40-1208"><a href="#cb40-1208" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 2.5</span></span>
<span id="cb40-1209"><a href="#cb40-1209" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1210"><a href="#cb40-1210" aria-hidden="true" tabindex="-1"></a><span class="co"># Microviridae</span></span>
<span id="cb40-1211"><a href="#cb40-1211" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_histogram</span>(<span class="at">taxid_chosen=</span><span class="dv">10841</span>)</span>
<span id="cb40-1212"><a href="#cb40-1212" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-1215"><a href="#cb40-1215" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb40-1216"><a href="#cb40-1216" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-microviridae-composition</span></span>
<span id="cb40-1217"><a href="#cb40-1217" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb40-1218"><a href="#cb40-1218" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 7.5</span></span>
<span id="cb40-1219"><a href="#cb40-1219" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb40-1220"><a href="#cb40-1220" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb40-1221"><a href="#cb40-1221" aria-hidden="true" tabindex="-1"></a><span class="co"># Microviridae</span></span>
<span id="cb40-1222"><a href="#cb40-1222" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_composition</span>(<span class="at">taxid_chosen=</span><span class="dv">10841</span>, <span class="at">threshold_major_species =</span> <span class="fl">0.1</span>)</span>
<span id="cb40-1223"><a href="#cb40-1223" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-1224"><a href="#cb40-1224" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1225"><a href="#cb40-1225" aria-hidden="true" tabindex="-1"></a><span class="fu">### Adenoviridae (Number of reads: 28)</span></span>
<span id="cb40-1228"><a href="#cb40-1228" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb40-1229"><a href="#cb40-1229" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-adenoviridae-histogram</span></span>
<span id="cb40-1230"><a href="#cb40-1230" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb40-1231"><a href="#cb40-1231" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 2.5</span></span>
<span id="cb40-1232"><a href="#cb40-1232" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1233"><a href="#cb40-1233" aria-hidden="true" tabindex="-1"></a><span class="co"># Adenoviridae</span></span>
<span id="cb40-1234"><a href="#cb40-1234" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_histogram</span>(<span class="at">taxid_chosen=</span><span class="dv">10508</span>)</span>
<span id="cb40-1235"><a href="#cb40-1235" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-1238"><a href="#cb40-1238" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb40-1239"><a href="#cb40-1239" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-adenoviridae-composition</span></span>
<span id="cb40-1240"><a href="#cb40-1240" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb40-1241"><a href="#cb40-1241" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 7.5</span></span>
<span id="cb40-1242"><a href="#cb40-1242" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb40-1243"><a href="#cb40-1243" aria-hidden="true" tabindex="-1"></a><span class="co">#| </span></span>
<span id="cb40-1244"><a href="#cb40-1244" aria-hidden="true" tabindex="-1"></a><span class="co"># Adenoviridae</span></span>
<span id="cb40-1245"><a href="#cb40-1245" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_composition</span>(<span class="at">taxid_chosen=</span><span class="dv">10508</span>, <span class="at">threshold_major_species =</span> <span class="fl">0.1</span>)</span>
<span id="cb40-1246"><a href="#cb40-1246" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-1247"><a href="#cb40-1247" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1248"><a href="#cb40-1248" aria-hidden="true" tabindex="-1"></a><span class="fu">## Relative abundance of pathogenic viruses of interest</span></span>
<span id="cb40-1249"><a href="#cb40-1249" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1252"><a href="#cb40-1252" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb40-1253"><a href="#cb40-1253" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-composition-exclusion</span></span>
<span id="cb40-1254"><a href="#cb40-1254" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb40-1255"><a href="#cb40-1255" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1256"><a href="#cb40-1256" aria-hidden="true" tabindex="-1"></a>total <span class="ot">&lt;-</span> basic_stats_raw <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample, n_read_pairs)</span>
<span id="cb40-1257"><a href="#cb40-1257" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1258"><a href="#cb40-1258" aria-hidden="true" tabindex="-1"></a>hv_family_counts <span class="ot">&lt;-</span> hv_reads_family <span class="sc">%&gt;%</span> </span>
<span id="cb40-1259"><a href="#cb40-1259" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb40-1260"><a href="#cb40-1260" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-1261"><a href="#cb40-1261" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(total, <span class="at">by=</span><span class="st">'sample'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-1262"><a href="#cb40-1262" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb40-1263"><a href="#cb40-1263" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">ra_reads_hv =</span> n_reads_hv<span class="sc">/</span>n_read_pairs)</span>
<span id="cb40-1264"><a href="#cb40-1264" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1265"><a href="#cb40-1265" aria-hidden="true" tabindex="-1"></a>hv_family_collate <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span></span>
<span id="cb40-1266"><a href="#cb40-1266" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb40-1267"><a href="#cb40-1267" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_tot =</span> <span class="fu">sum</span>(n_reads_hv),</span>
<span id="cb40-1268"><a href="#cb40-1268" aria-hidden="true" tabindex="-1"></a>            <span class="at">ra_reads_tot =</span> <span class="fu">mean</span>(ra_reads_hv), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb40-1269"><a href="#cb40-1269" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n_reads_tot))</span>
<span id="cb40-1270"><a href="#cb40-1270" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1271"><a href="#cb40-1271" aria-hidden="true" tabindex="-1"></a>hv_family_collate <span class="ot">&lt;-</span> hv_family_collate <span class="sc">%&gt;%</span> </span>
<span id="cb40-1272"><a href="#cb40-1272" aria-hidden="true" tabindex="-1"></a><span class="fu">rename</span>(</span>
<span id="cb40-1273"><a href="#cb40-1273" aria-hidden="true" tabindex="-1"></a><span class="st">'family'</span> <span class="ot">=</span> <span class="st">'name'</span>,</span>
<span id="cb40-1274"><a href="#cb40-1274" aria-hidden="true" tabindex="-1"></a><span class="st">'family_n_reads_tot'</span> <span class="ot">=</span> <span class="st">'n_reads_tot'</span>,</span>
<span id="cb40-1275"><a href="#cb40-1275" aria-hidden="true" tabindex="-1"></a><span class="st">'family_ra_reads_tot'</span> <span class="ot">=</span> <span class="st">'ra_reads_tot'</span>,</span>
<span id="cb40-1276"><a href="#cb40-1276" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb40-1277"><a href="#cb40-1277" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1278"><a href="#cb40-1278" aria-hidden="true" tabindex="-1"></a>hv_genus_counts <span class="ot">&lt;-</span> hv_reads_genus <span class="sc">%&gt;%</span> </span>
<span id="cb40-1279"><a href="#cb40-1279" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb40-1280"><a href="#cb40-1280" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-1281"><a href="#cb40-1281" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(total, <span class="at">by=</span><span class="st">'sample'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-1282"><a href="#cb40-1282" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb40-1283"><a href="#cb40-1283" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">ra_reads_hv =</span> n_reads_hv<span class="sc">/</span>n_read_pairs)</span>
<span id="cb40-1284"><a href="#cb40-1284" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1285"><a href="#cb40-1285" aria-hidden="true" tabindex="-1"></a>hv_genus_collate <span class="ot">&lt;-</span> hv_genus_counts <span class="sc">%&gt;%</span></span>
<span id="cb40-1286"><a href="#cb40-1286" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb40-1287"><a href="#cb40-1287" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_tot =</span> <span class="fu">sum</span>(n_reads_hv),</span>
<span id="cb40-1288"><a href="#cb40-1288" aria-hidden="true" tabindex="-1"></a>            <span class="at">ra_reads_tot =</span> <span class="fu">mean</span>(ra_reads_hv), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb40-1289"><a href="#cb40-1289" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n_reads_tot))</span>
<span id="cb40-1290"><a href="#cb40-1290" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1291"><a href="#cb40-1291" aria-hidden="true" tabindex="-1"></a>hv_genus_collate <span class="ot">&lt;-</span> hv_genus_collate <span class="sc">%&gt;%</span> </span>
<span id="cb40-1292"><a href="#cb40-1292" aria-hidden="true" tabindex="-1"></a><span class="fu">rename</span>(</span>
<span id="cb40-1293"><a href="#cb40-1293" aria-hidden="true" tabindex="-1"></a><span class="st">'genus'</span> <span class="ot">=</span> <span class="st">'name'</span>,</span>
<span id="cb40-1294"><a href="#cb40-1294" aria-hidden="true" tabindex="-1"></a><span class="st">'genus_n_reads_tot'</span> <span class="ot">=</span> <span class="st">'n_reads_tot'</span>,</span>
<span id="cb40-1295"><a href="#cb40-1295" aria-hidden="true" tabindex="-1"></a><span class="st">'genus_ra_reads_tot'</span> <span class="ot">=</span> <span class="st">'ra_reads_tot'</span>,</span>
<span id="cb40-1296"><a href="#cb40-1296" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb40-1297"><a href="#cb40-1297" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1298"><a href="#cb40-1298" aria-hidden="true" tabindex="-1"></a>hv_species_counts <span class="ot">&lt;-</span> hv_reads_species <span class="sc">%&gt;%</span> </span>
<span id="cb40-1299"><a href="#cb40-1299" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb40-1300"><a href="#cb40-1300" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-1301"><a href="#cb40-1301" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(total, <span class="at">by=</span><span class="st">'sample'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-1302"><a href="#cb40-1302" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb40-1303"><a href="#cb40-1303" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">ra_reads_hv =</span> n_reads_hv<span class="sc">/</span>n_read_pairs)</span>
<span id="cb40-1304"><a href="#cb40-1304" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1305"><a href="#cb40-1305" aria-hidden="true" tabindex="-1"></a>hv_species_collate <span class="ot">&lt;-</span> hv_species_counts <span class="sc">%&gt;%</span></span>
<span id="cb40-1306"><a href="#cb40-1306" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb40-1307"><a href="#cb40-1307" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_tot =</span> <span class="fu">sum</span>(n_reads_hv),</span>
<span id="cb40-1308"><a href="#cb40-1308" aria-hidden="true" tabindex="-1"></a>            <span class="at">ra_reads_tot =</span> <span class="fu">mean</span>(ra_reads_hv), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb40-1309"><a href="#cb40-1309" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n_reads_tot))</span>
<span id="cb40-1310"><a href="#cb40-1310" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1311"><a href="#cb40-1311" aria-hidden="true" tabindex="-1"></a>hv_species_collate <span class="ot">&lt;-</span> hv_species_collate <span class="sc">%&gt;%</span> </span>
<span id="cb40-1312"><a href="#cb40-1312" aria-hidden="true" tabindex="-1"></a><span class="fu">rename</span>(</span>
<span id="cb40-1313"><a href="#cb40-1313" aria-hidden="true" tabindex="-1"></a><span class="st">'species'</span> <span class="ot">=</span> <span class="st">'name'</span>,</span>
<span id="cb40-1314"><a href="#cb40-1314" aria-hidden="true" tabindex="-1"></a><span class="st">'species_n_reads_tot'</span> <span class="ot">=</span> <span class="st">'n_reads_tot'</span>,</span>
<span id="cb40-1315"><a href="#cb40-1315" aria-hidden="true" tabindex="-1"></a><span class="st">'species_ra_reads_tot'</span> <span class="ot">=</span> <span class="st">'ra_reads_tot'</span>,</span>
<span id="cb40-1316"><a href="#cb40-1316" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb40-1317"><a href="#cb40-1317" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1318"><a href="#cb40-1318" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-1319"><a href="#cb40-1319" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1322"><a href="#cb40-1322" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb40-1323"><a href="#cb40-1323" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-filter-exclusion</span></span>
<span id="cb40-1324"><a href="#cb40-1324" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb40-1325"><a href="#cb40-1325" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1326"><a href="#cb40-1326" aria-hidden="true" tabindex="-1"></a><span class="co"># Assuming the data is already loaded into tibbles:</span></span>
<span id="cb40-1327"><a href="#cb40-1327" aria-hidden="true" tabindex="-1"></a><span class="co">#hv_family_collate</span></span>
<span id="cb40-1328"><a href="#cb40-1328" aria-hidden="true" tabindex="-1"></a><span class="co">#hv_genus_collate</span></span>
<span id="cb40-1329"><a href="#cb40-1329" aria-hidden="true" tabindex="-1"></a><span class="co">#hv_species_collate</span></span>
<span id="cb40-1330"><a href="#cb40-1330" aria-hidden="true" tabindex="-1"></a><span class="co">#viral_taxa</span></span>
<span id="cb40-1331"><a href="#cb40-1331" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1332"><a href="#cb40-1332" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to find parent taxa recursively</span></span>
<span id="cb40-1333"><a href="#cb40-1333" aria-hidden="true" tabindex="-1"></a>find_parent_taxa <span class="ot">&lt;-</span> <span class="cf">function</span>(taxid, target_ranks, taxa_data) {</span>
<span id="cb40-1334"><a href="#cb40-1334" aria-hidden="true" tabindex="-1"></a> current_taxid <span class="ot">&lt;-</span> taxid</span>
<span id="cb40-1335"><a href="#cb40-1335" aria-hidden="true" tabindex="-1"></a>  result <span class="ot">&lt;-</span> <span class="fu">setNames</span>(<span class="fu">vector</span>(<span class="st">"list"</span>, <span class="fu">length</span>(target_ranks)), target_ranks)</span>
<span id="cb40-1336"><a href="#cb40-1336" aria-hidden="true" tabindex="-1"></a>  <span class="cf">while</span> (<span class="cn">TRUE</span>) {</span>
<span id="cb40-1337"><a href="#cb40-1337" aria-hidden="true" tabindex="-1"></a>    row <span class="ot">&lt;-</span> taxa_data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(taxid <span class="sc">==</span> current_taxid) <span class="sc">%&gt;%</span> <span class="fu">slice</span>(<span class="dv">1</span>)</span>
<span id="cb40-1338"><a href="#cb40-1338" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">nrow</span>(row) <span class="sc">==</span> <span class="dv">0</span>) <span class="cf">break</span>  <span class="co"># If no matching taxid is found, break the loop</span></span>
<span id="cb40-1339"><a href="#cb40-1339" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (row<span class="sc">$</span>rank <span class="sc">%in%</span> target_ranks) {</span>
<span id="cb40-1340"><a href="#cb40-1340" aria-hidden="true" tabindex="-1"></a>      result[[row<span class="sc">$</span>rank]] <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">taxid =</span> row<span class="sc">$</span>taxid, <span class="at">name =</span> row<span class="sc">$</span>name)</span>
<span id="cb40-1341"><a href="#cb40-1341" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span> (<span class="fu">all</span>(<span class="sc">!</span><span class="fu">sapply</span>(result, is.null))) {</span>
<span id="cb40-1342"><a href="#cb40-1342" aria-hidden="true" tabindex="-1"></a>        <span class="cf">break</span>  <span class="co"># If all target ranks are found, break the loop</span></span>
<span id="cb40-1343"><a href="#cb40-1343" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb40-1344"><a href="#cb40-1344" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb40-1345"><a href="#cb40-1345" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (row<span class="sc">$</span>taxid <span class="sc">==</span> row<span class="sc">$</span>parent_taxid) {</span>
<span id="cb40-1346"><a href="#cb40-1346" aria-hidden="true" tabindex="-1"></a>      <span class="cf">break</span>  <span class="co"># Avoid infinite loop at root</span></span>
<span id="cb40-1347"><a href="#cb40-1347" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb40-1348"><a href="#cb40-1348" aria-hidden="true" tabindex="-1"></a>    current_taxid <span class="ot">&lt;-</span> row<span class="sc">$</span>parent_taxid</span>
<span id="cb40-1349"><a href="#cb40-1349" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb40-1350"><a href="#cb40-1350" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Replace NULL with NA for any unfound ranks</span></span>
<span id="cb40-1351"><a href="#cb40-1351" aria-hidden="true" tabindex="-1"></a>  result <span class="ot">&lt;-</span> <span class="fu">lapply</span>(result, <span class="cf">function</span>(x) <span class="cf">if</span> (<span class="fu">is.null</span>(x)) <span class="fu">list</span>(<span class="at">taxid =</span> <span class="cn">NA</span>, <span class="at">name =</span> <span class="cn">NA</span>) <span class="cf">else</span> x)</span>
<span id="cb40-1352"><a href="#cb40-1352" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(result)</span>
<span id="cb40-1353"><a href="#cb40-1353" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb40-1354"><a href="#cb40-1354" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1355"><a href="#cb40-1355" aria-hidden="true" tabindex="-1"></a><span class="co"># Find all relevant parent taxa for each species</span></span>
<span id="cb40-1356"><a href="#cb40-1356" aria-hidden="true" tabindex="-1"></a>result <span class="ot">&lt;-</span> hv_species_collate <span class="sc">%&gt;%</span></span>
<span id="cb40-1357"><a href="#cb40-1357" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb40-1358"><a href="#cb40-1358" aria-hidden="true" tabindex="-1"></a>    <span class="at">parent_info =</span> <span class="fu">map</span>(taxid, <span class="sc">~</span><span class="fu">find_parent_taxa</span>(.x, <span class="fu">c</span>(<span class="st">"species"</span>,<span class="st">"subgenus"</span>, <span class="st">"genus"</span>, <span class="st">"subfamily"</span>, <span class="st">"family"</span>), viral_taxa)),</span>
<span id="cb40-1359"><a href="#cb40-1359" aria-hidden="true" tabindex="-1"></a>    <span class="at">genus_taxid =</span> <span class="fu">map_dbl</span>(parent_info, <span class="sc">~</span>.x<span class="sc">$</span>genus<span class="sc">$</span>taxid),</span>
<span id="cb40-1360"><a href="#cb40-1360" aria-hidden="true" tabindex="-1"></a>    <span class="at">genus =</span> <span class="fu">map_chr</span>(parent_info, <span class="sc">~</span>.x<span class="sc">$</span>genus<span class="sc">$</span>name),</span>
<span id="cb40-1361"><a href="#cb40-1361" aria-hidden="true" tabindex="-1"></a>    <span class="at">family_taxid =</span> <span class="fu">map_dbl</span>(parent_info, <span class="sc">~</span>.x<span class="sc">$</span>family<span class="sc">$</span>taxid),</span>
<span id="cb40-1362"><a href="#cb40-1362" aria-hidden="true" tabindex="-1"></a>    <span class="at">family =</span> <span class="fu">map_chr</span>(parent_info, <span class="sc">~</span>.x<span class="sc">$</span>family<span class="sc">$</span>name)</span>
<span id="cb40-1363"><a href="#cb40-1363" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb40-1364"><a href="#cb40-1364" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>parent_info)</span>
<span id="cb40-1365"><a href="#cb40-1365" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1366"><a href="#cb40-1366" aria-hidden="true" tabindex="-1"></a><span class="co"># Join with genus counts</span></span>
<span id="cb40-1367"><a href="#cb40-1367" aria-hidden="true" tabindex="-1"></a>result <span class="ot">&lt;-</span> result <span class="sc">%&gt;%</span></span>
<span id="cb40-1368"><a href="#cb40-1368" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(</span>
<span id="cb40-1369"><a href="#cb40-1369" aria-hidden="true" tabindex="-1"></a>    hv_genus_collate <span class="sc">%&gt;%</span> <span class="fu">select</span>(taxid, genus_n_reads_tot, genus_ra_reads_tot),</span>
<span id="cb40-1370"><a href="#cb40-1370" aria-hidden="true" tabindex="-1"></a>    <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"genus_taxid"</span> <span class="ot">=</span> <span class="st">"taxid"</span>)</span>
<span id="cb40-1371"><a href="#cb40-1371" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb40-1372"><a href="#cb40-1372" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1373"><a href="#cb40-1373" aria-hidden="true" tabindex="-1"></a><span class="co"># Join with family counts</span></span>
<span id="cb40-1374"><a href="#cb40-1374" aria-hidden="true" tabindex="-1"></a>result <span class="ot">&lt;-</span> result <span class="sc">%&gt;%</span></span>
<span id="cb40-1375"><a href="#cb40-1375" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(</span>
<span id="cb40-1376"><a href="#cb40-1376" aria-hidden="true" tabindex="-1"></a>    hv_family_collate <span class="sc">%&gt;%</span> <span class="fu">select</span>(taxid, family_n_reads_tot, family_ra_reads_tot),</span>
<span id="cb40-1377"><a href="#cb40-1377" aria-hidden="true" tabindex="-1"></a>    <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"family_taxid"</span> <span class="ot">=</span> <span class="st">"taxid"</span>)</span>
<span id="cb40-1378"><a href="#cb40-1378" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb40-1379"><a href="#cb40-1379" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1380"><a href="#cb40-1380" aria-hidden="true" tabindex="-1"></a><span class="co"># Clean up</span></span>
<span id="cb40-1381"><a href="#cb40-1381" aria-hidden="true" tabindex="-1"></a>result <span class="ot">&lt;-</span> result <span class="sc">%&gt;%</span></span>
<span id="cb40-1382"><a href="#cb40-1382" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="fu">ends_with</span>(<span class="st">"_taxid"</span>))</span>
<span id="cb40-1383"><a href="#cb40-1383" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1384"><a href="#cb40-1384" aria-hidden="true" tabindex="-1"></a><span class="co">#result</span></span>
<span id="cb40-1385"><a href="#cb40-1385" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-1386"><a href="#cb40-1386" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1387"><a href="#cb40-1387" aria-hidden="true" tabindex="-1"></a>Each dot represents a sample, colored by viral family. The x-axis shows the relative abundance of human-infecting viruses, and the y-axis shows the species.</span>
<span id="cb40-1390"><a href="#cb40-1390" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb40-1391"><a href="#cb40-1391" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-pathogenic-viruses</span></span>
<span id="cb40-1392"><a href="#cb40-1392" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb40-1393"><a href="#cb40-1393" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 10</span></span>
<span id="cb40-1394"><a href="#cb40-1394" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb40-1395"><a href="#cb40-1395" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1396"><a href="#cb40-1396" aria-hidden="true" tabindex="-1"></a>species_family <span class="ot">&lt;-</span> result <span class="sc">%&gt;%</span> <span class="fu">select</span>(species, family) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="st">'name'</span> <span class="ot">=</span> <span class="st">'species'</span>)</span>
<span id="cb40-1397"><a href="#cb40-1397" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1398"><a href="#cb40-1398" aria-hidden="true" tabindex="-1"></a>play <span class="ot">&lt;-</span> hv_species_counts <span class="sc">%&gt;%</span> </span>
<span id="cb40-1399"><a href="#cb40-1399" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb40-1400"><a href="#cb40-1400" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by =</span> <span class="st">'sample'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-1401"><a href="#cb40-1401" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(species_family, <span class="at">by =</span> <span class="st">'name'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-1402"><a href="#cb40-1402" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">ifelse</span>(name <span class="sc">==</span> <span class="st">"Severe acute respiratory syndrome-related coronavirus"</span>, <span class="st">"SARS-related coronavirus"</span>, name)) <span class="sc">%&gt;%</span></span>
<span id="cb40-1403"><a href="#cb40-1403" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span>(family <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Anelloviridae"</span>, <span class="cn">NA</span>, <span class="st">"Microviridae"</span>, <span class="st">"Rhabdoviridae"</span>))) </span>
<span id="cb40-1404"><a href="#cb40-1404" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1405"><a href="#cb40-1405" aria-hidden="true" tabindex="-1"></a>adjusted_play <span class="ot">&lt;-</span> play <span class="sc">%&gt;%</span> </span>
<span id="cb40-1406"><a href="#cb40-1406" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span></span>
<span id="cb40-1407"><a href="#cb40-1407" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">virus_prevalence_num =</span> <span class="fu">n_distinct</span>(sample)<span class="sc">/</span><span class="fu">n_distinct</span>(libraries),</span>
<span id="cb40-1408"><a href="#cb40-1408" aria-hidden="true" tabindex="-1"></a>         <span class="at">total_reads_hv =</span> <span class="fu">sum</span>(n_reads_hv)) <span class="sc">%&gt;%</span></span>
<span id="cb40-1409"><a href="#cb40-1409" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb40-1410"><a href="#cb40-1410" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">fct_reorder</span>(name, virus_prevalence_num, <span class="at">.desc=</span><span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb40-1411"><a href="#cb40-1411" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, ra_reads_hv, family, location, virus_prevalence_num, total_reads_hv, n_reads_hv)</span>
<span id="cb40-1412"><a href="#cb40-1412" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1413"><a href="#cb40-1413" aria-hidden="true" tabindex="-1"></a>pal <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">'Dark2'</span>),<span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">'Accent'</span>))</span>
<span id="cb40-1414"><a href="#cb40-1414" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1415"><a href="#cb40-1415" aria-hidden="true" tabindex="-1"></a><span class="co">#, labels = label_log(digits=2)</span></span>
<span id="cb40-1416"><a href="#cb40-1416" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1417"><a href="#cb40-1417" aria-hidden="true" tabindex="-1"></a>ra_dot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(adjusted_play, <span class="fu">aes</span>(<span class="at">x =</span> ra_reads_hv, <span class="at">y=</span>name)) <span class="sc">+</span></span>
<span id="cb40-1418"><a href="#cb40-1418" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_quasirandom</span>(<span class="at">orientation =</span> <span class="st">'y'</span>, <span class="fu">aes</span>(<span class="at">color =</span> family)) <span class="sc">+</span></span>
<span id="cb40-1419"><a href="#cb40-1419" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> pal) <span class="sc">+</span> </span>
<span id="cb40-1420"><a href="#cb40-1420" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_log10</span>(</span>
<span id="cb40-1421"><a href="#cb40-1421" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Relative abundance human virus reads"</span>,</span>
<span id="cb40-1422"><a href="#cb40-1422" aria-hidden="true" tabindex="-1"></a>    <span class="at">labels=</span><span class="fu">label_log</span>(<span class="at">digits=</span><span class="dv">3</span>)</span>
<span id="cb40-1423"><a href="#cb40-1423" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb40-1424"><a href="#cb40-1424" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span>  <span class="st">""</span>,</span>
<span id="cb40-1425"><a href="#cb40-1425" aria-hidden="true" tabindex="-1"></a>       <span class="at">color =</span> <span class="st">'Viral family'</span>) <span class="sc">+</span> </span>
<span id="cb40-1426"><a href="#cb40-1426" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color =</span> <span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">2</span>)) <span class="sc">+</span></span>
<span id="cb40-1427"><a href="#cb40-1427" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_light</span>() <span class="sc">+</span> </span>
<span id="cb40-1428"><a href="#cb40-1428" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(</span>
<span id="cb40-1429"><a href="#cb40-1429" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb40-1430"><a href="#cb40-1430" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>),</span>
<span id="cb40-1431"><a href="#cb40-1431" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.ticks.y =</span> <span class="fu">element_blank</span>(),<span class="at">axis.line =</span> <span class="fu">element_line</span>(<span class="at">colour =</span> <span class="st">"black"</span>),</span>
<span id="cb40-1432"><a href="#cb40-1432" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">14</span>),    </span>
<span id="cb40-1433"><a href="#cb40-1433" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb40-1434"><a href="#cb40-1434" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>, <span class="at">face=</span><span class="st">"bold"</span>),</span>
<span id="cb40-1435"><a href="#cb40-1435" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">'bottom'</span>,</span>
<span id="cb40-1436"><a href="#cb40-1436" aria-hidden="true" tabindex="-1"></a>    <span class="co">#legend.position = c(1, 1),  # Move legend to top right</span></span>
<span id="cb40-1437"><a href="#cb40-1437" aria-hidden="true" tabindex="-1"></a>    <span class="co">#legend.justification = c(1, 1),  # Align legend to top right</span></span>
<span id="cb40-1438"><a href="#cb40-1438" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb40-1439"><a href="#cb40-1439" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.border =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb40-1440"><a href="#cb40-1440" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.background =</span> <span class="fu">element_blank</span>())</span>
<span id="cb40-1441"><a href="#cb40-1441" aria-hidden="true" tabindex="-1"></a>ra_dot</span>
<span id="cb40-1442"><a href="#cb40-1442" aria-hidden="true" tabindex="-1"></a><span class="co">#  geom_text(aes(label = total_reads_hv, y = name), </span></span>
<span id="cb40-1443"><a href="#cb40-1443" aria-hidden="true" tabindex="-1"></a><span class="co">#            x = 1e-9, hjust = 0, vjust = 0.5, size = 3, </span></span>
<span id="cb40-1444"><a href="#cb40-1444" aria-hidden="true" tabindex="-1"></a><span class="co">#            check_overlap = TRUE)</span></span>
<span id="cb40-1445"><a href="#cb40-1445" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-1446"><a href="#cb40-1446" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1447"><a href="#cb40-1447" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1448"><a href="#cb40-1448" aria-hidden="true" tabindex="-1"></a>We can ask Claude to analyze the pathogenicity of these viruses  along with their popular names.</span>
<span id="cb40-1449"><a href="#cb40-1449" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1450"><a href="#cb40-1450" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1451"><a href="#cb40-1451" aria-hidden="true" tabindex="-1"></a>| Scientific Name | Popular/Well-known Name | Pathogenicity to Humans |</span>
<span id="cb40-1452"><a href="#cb40-1452" aria-hidden="true" tabindex="-1"></a>|-----------------|-------------------------|-------------------------|</span>
<span id="cb40-1453"><a href="#cb40-1453" aria-hidden="true" tabindex="-1"></a>| Human immunodeficiency virus 1 | HIV-1 | Very High |</span>
<span id="cb40-1454"><a href="#cb40-1454" aria-hidden="true" tabindex="-1"></a>| SARS-related coronavirus | SARS-CoV | High |</span>
<span id="cb40-1455"><a href="#cb40-1455" aria-hidden="true" tabindex="-1"></a>| Hepatitis B virus | HBV | High |</span>
<span id="cb40-1456"><a href="#cb40-1456" aria-hidden="true" tabindex="-1"></a>| Hepacivirus hominis | Hepatitis C virus (HCV) | High |</span>
<span id="cb40-1457"><a href="#cb40-1457" aria-hidden="true" tabindex="-1"></a>| Simplexvirus humanalpha1 | Herpes simplex virus 1 (HSV-1) | Moderate to High |</span>
<span id="cb40-1458"><a href="#cb40-1458" aria-hidden="true" tabindex="-1"></a>| Cytomegalovirus humanbeta5 | Human cytomegalovirus (HCMV) | Moderate to High |</span>
<span id="cb40-1459"><a href="#cb40-1459" aria-hidden="true" tabindex="-1"></a>| Lymphocryptovirus humangamma4 | Epstein-Barr virus (EBV) | Moderate |</span>
<span id="cb40-1460"><a href="#cb40-1460" aria-hidden="true" tabindex="-1"></a>| Human mastadenovirus C | Adenovirus C | Moderate |</span>
<span id="cb40-1461"><a href="#cb40-1461" aria-hidden="true" tabindex="-1"></a>| Human mastadenovirus F | Adenovirus F | Moderate |</span>
<span id="cb40-1462"><a href="#cb40-1462" aria-hidden="true" tabindex="-1"></a>| Roseolovirus humanbeta6a | Human herpesvirus 6A (HHV-6A) | Low to Moderate |</span>
<span id="cb40-1463"><a href="#cb40-1463" aria-hidden="true" tabindex="-1"></a>| Roseolovirus humanbeta6b | Human herpesvirus 6B (HHV-6B) | Low to Moderate |</span>
<span id="cb40-1464"><a href="#cb40-1464" aria-hidden="true" tabindex="-1"></a>| Roseolovirus humanbeta7 | Human herpesvirus 7 (HHV-7) | Low to Moderate |</span>
<span id="cb40-1465"><a href="#cb40-1465" aria-hidden="true" tabindex="-1"></a>| Rhadinovirus humangamma8 | Kaposi's sarcoma-associated herpesvirus (KSHV) | Low to Moderate |</span>
<span id="cb40-1466"><a href="#cb40-1466" aria-hidden="true" tabindex="-1"></a>| Molluscum contagiosum virus | MCV | Low |</span>
<span id="cb40-1467"><a href="#cb40-1467" aria-hidden="true" tabindex="-1"></a>| Erythroparvovirus primate1 | Parvovirus B19 | Low |</span>
<span id="cb40-1468"><a href="#cb40-1468" aria-hidden="true" tabindex="-1"></a>| Human erythrovirus V9 | Erythrovirus V9 | Low |</span>
<span id="cb40-1469"><a href="#cb40-1469" aria-hidden="true" tabindex="-1"></a>| Alphapapillomavirus 4 | Human papillomavirus 4 (HPV-4) | Low |</span>
<span id="cb40-1470"><a href="#cb40-1470" aria-hidden="true" tabindex="-1"></a>| Betapolyomavirus hominis | Human polyomavirus | Low |</span>
<span id="cb40-1471"><a href="#cb40-1471" aria-hidden="true" tabindex="-1"></a>| Dependoparvovirus primate1 | Adeno-associated virus (AAV) | Very Low |</span>
<span id="cb40-1472"><a href="#cb40-1472" aria-hidden="true" tabindex="-1"></a>| Murine leukemia virus | MLV | Very Low (not a human pathogen) |</span>
<span id="cb40-1473"><a href="#cb40-1473" aria-hidden="true" tabindex="-1"></a>| Murine leukemia-related retroviruses | MLV-related viruses | Very Low (not typical human pathogens) |</span>
<span id="cb40-1474"><a href="#cb40-1474" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1475"><a href="#cb40-1475" aria-hidden="true" tabindex="-1"></a>We get a variety of pathogenic viruses, none of which are a suprise to see in plasma (other than SARS-CoV which we determined was contamination <span class="co">[</span><span class="ot">on twist</span><span class="co">](https://twist.com/a/197793/ch/619193/t/6323010/)</span>).</span>
<span id="cb40-1476"><a href="#cb40-1476" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1477"><a href="#cb40-1477" aria-hidden="true" tabindex="-1"></a><span class="fu">## Relative abundance assuming perfect human read removal</span></span>
<span id="cb40-1480"><a href="#cb40-1480" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb40-1481"><a href="#cb40-1481" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: perfect-hv-removal</span></span>
<span id="cb40-1482"><a href="#cb40-1482" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb40-1483"><a href="#cb40-1483" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1484"><a href="#cb40-1484" aria-hidden="true" tabindex="-1"></a>human_reads <span class="ot">&lt;-</span> comp_assigned <span class="sc">%&gt;%</span> <span class="fu">filter</span>(classification <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Human"</span>)) <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample, <span class="at">human_reads=</span>n_reads)</span>
<span id="cb40-1485"><a href="#cb40-1485" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1486"><a href="#cb40-1486" aria-hidden="true" tabindex="-1"></a>all <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span> </span>
<span id="cb40-1487"><a href="#cb40-1487" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(human_reads, <span class="at">by=</span><span class="st">'sample'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-1488"><a href="#cb40-1488" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb40-1489"><a href="#cb40-1489" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">sum_reads=</span><span class="fu">sum</span>(n_reads_hv), <span class="at">n_read_pairs=</span><span class="fu">first</span>(n_read_pairs) <span class="sc">-</span> <span class="fu">first</span>(human_reads)) <span class="sc">%&gt;%</span></span>
<span id="cb40-1490"><a href="#cb40-1490" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">ra =</span> sum_reads<span class="sc">/</span>n_read_pairs) <span class="sc">%&gt;%</span> </span>
<span id="cb40-1491"><a href="#cb40-1491" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(ra) <span class="sc">%&gt;%</span> <span class="fu">mean</span>()</span>
<span id="cb40-1492"><a href="#cb40-1492" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1493"><a href="#cb40-1493" aria-hidden="true" tabindex="-1"></a>all</span>
<span id="cb40-1494"><a href="#cb40-1494" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-1495"><a href="#cb40-1495" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1496"><a href="#cb40-1496" aria-hidden="true" tabindex="-1"></a>Assuming we're able to perfectly remove all human reads, the average relative abundance of known human infecting virus is $5.37 \times 10^{-4}$.</span>
<span id="cb40-1497"><a href="#cb40-1497" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1498"><a href="#cb40-1498" aria-hidden="true" tabindex="-1"></a><span class="fu"># Conclusion</span></span>
<span id="cb40-1499"><a href="#cb40-1499" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1500"><a href="#cb40-1500" aria-hidden="true" tabindex="-1"></a>There were some interesting takeways from this analysis:</span>
<span id="cb40-1501"><a href="#cb40-1501" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1502"><a href="#cb40-1502" aria-hidden="true" tabindex="-1"></a><span class="ss">1. </span>Human read fraction remains quite high in plasma if special sample preparation/library preparation is not taken (see <span class="co">[</span><span class="ot">here</span><span class="co">](https://pubmed.ncbi.nlm.nih.gov/33767340/)</span>).</span>
<span id="cb40-1503"><a href="#cb40-1503" aria-hidden="true" tabindex="-1"></a><span class="ss">2. </span>There is a lot of Anelloviridae in blood, including that which has not been classified by species.</span>
<span id="cb40-1504"><a href="#cb40-1504" aria-hidden="true" tabindex="-1"></a><span class="ss">3. </span>Pathogenic viruses such as HIV and HBV, and even latent viruses like Herpes are picked up by plasma MGS</span>
<span id="cb40-1505"><a href="#cb40-1505" aria-hidden="true" tabindex="-1"></a><span class="ss">4. </span>Respiratory viruses are also picked up by plasma MGS (SARS), but not as much as other viruses.</span>
<span id="cb40-1506"><a href="#cb40-1506" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1507"><a href="#cb40-1507" aria-hidden="true" tabindex="-1"></a>*We do note that we've analyzed other datasets and it seems that having a large catchment is crucial, as we'd expect, for picking up these more rare, pathogenic viruses*. Using this data, we can also look at China's HIV/HBV reported incidence to do a P2RA/swab sampling type analysis. We plan to do more analysis on this data, but wanted to share some preliminary results.</span>
<span id="cb40-1508"><a href="#cb40-1508" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1509"><a href="#cb40-1509" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1510"><a href="#cb40-1510" aria-hidden="true" tabindex="-1"></a><span class="fu"># Appendix</span></span>
<span id="cb40-1511"><a href="#cb40-1511" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1512"><a href="#cb40-1512" aria-hidden="true" tabindex="-1"></a><span class="fu">## Human-infecting virus families, genera, and species</span></span>
<span id="cb40-1513"><a href="#cb40-1513" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1514"><a href="#cb40-1514" aria-hidden="true" tabindex="-1"></a>To get a good overview of families, genera, and species, we can look at a Sankey plot where the magnitude of relative abundance, averaged over all samples, is shown in parentheses.</span>
<span id="cb40-1515"><a href="#cb40-1515" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1516"><a href="#cb40-1516" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1519"><a href="#cb40-1519" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb40-1520"><a href="#cb40-1520" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-sankey</span></span>
<span id="cb40-1521"><a href="#cb40-1521" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb40-1522"><a href="#cb40-1522" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 22.5</span></span>
<span id="cb40-1523"><a href="#cb40-1523" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb40-1524"><a href="#cb40-1524" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1525"><a href="#cb40-1525" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to create links</span></span>
<span id="cb40-1526"><a href="#cb40-1526" aria-hidden="true" tabindex="-1"></a>create_links <span class="ot">&lt;-</span> <span class="cf">function</span>(data) {</span>
<span id="cb40-1527"><a href="#cb40-1527" aria-hidden="true" tabindex="-1"></a>  family_to_genus <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb40-1528"><a href="#cb40-1528" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(genus)) <span class="sc">%&gt;%</span></span>
<span id="cb40-1529"><a href="#cb40-1529" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(family, genus) <span class="sc">%&gt;%</span></span>
<span id="cb40-1530"><a href="#cb40-1530" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">value =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-1531"><a href="#cb40-1531" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">source =</span> family, <span class="at">target =</span> genus)</span>
<span id="cb40-1532"><a href="#cb40-1532" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb40-1533"><a href="#cb40-1533" aria-hidden="true" tabindex="-1"></a>  genus_to_species <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb40-1534"><a href="#cb40-1534" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(genus, species) <span class="sc">%&gt;%</span></span>
<span id="cb40-1535"><a href="#cb40-1535" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">value =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-1536"><a href="#cb40-1536" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">source =</span> genus, <span class="at">target =</span> species)</span>
<span id="cb40-1537"><a href="#cb40-1537" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1538"><a href="#cb40-1538" aria-hidden="true" tabindex="-1"></a>  family_to_species <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb40-1539"><a href="#cb40-1539" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="fu">is.na</span>(genus)) <span class="sc">%&gt;%</span></span>
<span id="cb40-1540"><a href="#cb40-1540" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(family, species) <span class="sc">%&gt;%</span></span>
<span id="cb40-1541"><a href="#cb40-1541" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">value =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-1542"><a href="#cb40-1542" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">source =</span> family, <span class="at">target =</span> species)</span>
<span id="cb40-1543"><a href="#cb40-1543" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1544"><a href="#cb40-1544" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_rows</span>(family_to_genus, genus_to_species, family_to_species) <span class="sc">%&gt;%</span></span>
<span id="cb40-1545"><a href="#cb40-1545" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(source))</span>
<span id="cb40-1546"><a href="#cb40-1546" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb40-1547"><a href="#cb40-1547" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1548"><a href="#cb40-1548" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to create nodes</span></span>
<span id="cb40-1549"><a href="#cb40-1549" aria-hidden="true" tabindex="-1"></a>create_nodes <span class="ot">&lt;-</span> <span class="cf">function</span>(links) {</span>
<span id="cb40-1550"><a href="#cb40-1550" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>(</span>
<span id="cb40-1551"><a href="#cb40-1551" aria-hidden="true" tabindex="-1"></a>    <span class="at">name =</span> <span class="fu">c</span>(links<span class="sc">$</span>source, links<span class="sc">$</span>target) <span class="sc">%&gt;%</span> <span class="fu">unique</span>()</span>
<span id="cb40-1552"><a href="#cb40-1552" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb40-1553"><a href="#cb40-1553" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb40-1554"><a href="#cb40-1554" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1555"><a href="#cb40-1555" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to prepare data for Sankey diagram</span></span>
<span id="cb40-1556"><a href="#cb40-1556" aria-hidden="true" tabindex="-1"></a>prepare_sankey_data <span class="ot">&lt;-</span> <span class="cf">function</span>(links, nodes) {</span>
<span id="cb40-1557"><a href="#cb40-1557" aria-hidden="true" tabindex="-1"></a>  links<span class="sc">$</span>IDsource <span class="ot">&lt;-</span> <span class="fu">match</span>(links<span class="sc">$</span>source, nodes<span class="sc">$</span>name) <span class="sc">-</span> <span class="dv">1</span></span>
<span id="cb40-1558"><a href="#cb40-1558" aria-hidden="true" tabindex="-1"></a>  links<span class="sc">$</span>IDtarget <span class="ot">&lt;-</span> <span class="fu">match</span>(links<span class="sc">$</span>target, nodes<span class="sc">$</span>name) <span class="sc">-</span> <span class="dv">1</span></span>
<span id="cb40-1559"><a href="#cb40-1559" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list</span>(<span class="at">links =</span> links, <span class="at">nodes =</span> nodes)</span>
<span id="cb40-1560"><a href="#cb40-1560" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb40-1561"><a href="#cb40-1561" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1562"><a href="#cb40-1562" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to create Sankey plot</span></span>
<span id="cb40-1563"><a href="#cb40-1563" aria-hidden="true" tabindex="-1"></a>create_sankey_plot <span class="ot">&lt;-</span> <span class="cf">function</span>(sankey_data) {</span>
<span id="cb40-1564"><a href="#cb40-1564" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sankeyNetwork</span>(</span>
<span id="cb40-1565"><a href="#cb40-1565" aria-hidden="true" tabindex="-1"></a>    <span class="at">Links =</span> sankey_data<span class="sc">$</span>links, </span>
<span id="cb40-1566"><a href="#cb40-1566" aria-hidden="true" tabindex="-1"></a>    <span class="at">Nodes =</span> sankey_data<span class="sc">$</span>nodes,</span>
<span id="cb40-1567"><a href="#cb40-1567" aria-hidden="true" tabindex="-1"></a>    <span class="at">Source =</span> <span class="st">"IDsource"</span>, </span>
<span id="cb40-1568"><a href="#cb40-1568" aria-hidden="true" tabindex="-1"></a>    <span class="at">Target =</span> <span class="st">"IDtarget"</span>,</span>
<span id="cb40-1569"><a href="#cb40-1569" aria-hidden="true" tabindex="-1"></a>    <span class="at">Value =</span> <span class="st">"value"</span>, </span>
<span id="cb40-1570"><a href="#cb40-1570" aria-hidden="true" tabindex="-1"></a>    <span class="at">NodeID =</span> <span class="st">"name"</span>,</span>
<span id="cb40-1571"><a href="#cb40-1571" aria-hidden="true" tabindex="-1"></a>    <span class="at">sinksRight =</span> <span class="cn">TRUE</span>,</span>
<span id="cb40-1572"><a href="#cb40-1572" aria-hidden="true" tabindex="-1"></a>    <span class="at">nodeWidth =</span> <span class="dv">25</span>,</span>
<span id="cb40-1573"><a href="#cb40-1573" aria-hidden="true" tabindex="-1"></a>    <span class="at">fontSize =</span> <span class="dv">14</span>,</span>
<span id="cb40-1574"><a href="#cb40-1574" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb40-1575"><a href="#cb40-1575" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb40-1576"><a href="#cb40-1576" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1577"><a href="#cb40-1577" aria-hidden="true" tabindex="-1"></a>save_sankey_as_png <span class="ot">&lt;-</span> <span class="cf">function</span>(sankey_plot, <span class="at">width =</span> <span class="dv">1000</span>, <span class="at">height =</span> <span class="dv">800</span>) {</span>
<span id="cb40-1578"><a href="#cb40-1578" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Save the plot as an HTML file</span></span>
<span id="cb40-1579"><a href="#cb40-1579" aria-hidden="true" tabindex="-1"></a>  <span class="fu">saveWidget</span>(sankey_plot, <span class="fu">sprintf</span>(<span class="st">'%s/sankey.html'</span>,data_dir))</span>
<span id="cb40-1580"><a href="#cb40-1580" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb40-1581"><a href="#cb40-1581" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1582"><a href="#cb40-1582" aria-hidden="true" tabindex="-1"></a>format_scientific <span class="ot">&lt;-</span> <span class="cf">function</span>(x, <span class="at">digits=</span><span class="dv">2</span>) {</span>
<span id="cb40-1583"><a href="#cb40-1583" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sapply</span>(x, <span class="cf">function</span>(val) {</span>
<span id="cb40-1584"><a href="#cb40-1584" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">is.na</span>(val) <span class="sc">||</span> <span class="fu">abs</span>(val) <span class="sc">&lt;</span> <span class="fl">1e-15</span>) {</span>
<span id="cb40-1585"><a href="#cb40-1585" aria-hidden="true" tabindex="-1"></a>      <span class="fu">return</span>(<span class="st">"0"</span>)</span>
<span id="cb40-1586"><a href="#cb40-1586" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb40-1587"><a href="#cb40-1587" aria-hidden="true" tabindex="-1"></a>      exponent <span class="ot">&lt;-</span> <span class="fu">floor</span>(<span class="fu">log10</span>(<span class="fu">abs</span>(val)))</span>
<span id="cb40-1588"><a href="#cb40-1588" aria-hidden="true" tabindex="-1"></a>      coef <span class="ot">&lt;-</span> val <span class="sc">/</span> <span class="dv">10</span><span class="sc">^</span>exponent</span>
<span id="cb40-1589"><a href="#cb40-1589" aria-hidden="true" tabindex="-1"></a>      <span class="co">#return(sprintf("%.1f × 10^%d", round(coef, digits), exponent))</span></span>
<span id="cb40-1590"><a href="#cb40-1590" aria-hidden="true" tabindex="-1"></a>      <span class="co"># May or may not be smart, just keeping magnitude</span></span>
<span id="cb40-1591"><a href="#cb40-1591" aria-hidden="true" tabindex="-1"></a>      <span class="fu">return</span>(<span class="fu">sprintf</span>(<span class="st">"10^%d"</span>, exponent))</span>
<span id="cb40-1592"><a href="#cb40-1592" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb40-1593"><a href="#cb40-1593" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb40-1594"><a href="#cb40-1594" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb40-1595"><a href="#cb40-1595" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1596"><a href="#cb40-1596" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> result <span class="sc">%&gt;%</span> </span>
<span id="cb40-1597"><a href="#cb40-1597" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">c</span>(genus_n_reads_tot, genus_ra_reads_tot), <span class="sc">~</span><span class="fu">replace_na</span>(., <span class="dv">0</span>)),</span>
<span id="cb40-1598"><a href="#cb40-1598" aria-hidden="true" tabindex="-1"></a>         <span class="at">genus =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(genus), <span class="st">"Unknown Genus"</span>, genus)) <span class="sc">%&gt;%</span></span>
<span id="cb40-1599"><a href="#cb40-1599" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb40-1600"><a href="#cb40-1600" aria-hidden="true" tabindex="-1"></a>  <span class="at">species =</span> <span class="fu">paste0</span>(species, <span class="fu">sprintf</span>(<span class="st">' (%s)'</span>, <span class="fu">format_scientific</span>(species_ra_reads_tot))),</span>
<span id="cb40-1601"><a href="#cb40-1601" aria-hidden="true" tabindex="-1"></a>  <span class="at">genus =</span> <span class="fu">paste0</span>(genus, <span class="fu">sprintf</span>(<span class="st">' (%s)'</span>, <span class="fu">format_scientific</span>(genus_ra_reads_tot))),</span>
<span id="cb40-1602"><a href="#cb40-1602" aria-hidden="true" tabindex="-1"></a>  <span class="at">family =</span> <span class="fu">paste0</span>(family, <span class="fu">sprintf</span>(<span class="st">' (%s)'</span>, <span class="fu">format_scientific</span>(family_ra_reads_tot)))</span>
<span id="cb40-1603"><a href="#cb40-1603" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb40-1604"><a href="#cb40-1604" aria-hidden="true" tabindex="-1"></a>links <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">create_links</span>(data))</span>
<span id="cb40-1605"><a href="#cb40-1605" aria-hidden="true" tabindex="-1"></a>nodes <span class="ot">&lt;-</span> <span class="fu">create_nodes</span>(links)</span>
<span id="cb40-1606"><a href="#cb40-1606" aria-hidden="true" tabindex="-1"></a>sankey_data <span class="ot">&lt;-</span> <span class="fu">prepare_sankey_data</span>(links, nodes)</span>
<span id="cb40-1607"><a href="#cb40-1607" aria-hidden="true" tabindex="-1"></a>sankey <span class="ot">&lt;-</span> <span class="fu">create_sankey_plot</span>(sankey_data)</span>
<span id="cb40-1608"><a href="#cb40-1608" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1609"><a href="#cb40-1609" aria-hidden="true" tabindex="-1"></a>sankey</span>
<span id="cb40-1610"><a href="#cb40-1610" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-1611"><a href="#cb40-1611" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1612"><a href="#cb40-1612" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1615"><a href="#cb40-1615" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb40-1616"><a href="#cb40-1616" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-species-dot</span></span>
<span id="cb40-1617"><a href="#cb40-1617" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 10</span></span>
<span id="cb40-1618"><a href="#cb40-1618" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 15</span></span>
<span id="cb40-1619"><a href="#cb40-1619" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb40-1620"><a href="#cb40-1620" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb40-1621"><a href="#cb40-1621" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1622"><a href="#cb40-1622" aria-hidden="true" tabindex="-1"></a>species_family <span class="ot">&lt;-</span> result <span class="sc">%&gt;%</span> <span class="fu">select</span>(species, family) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="st">'name'</span> <span class="ot">=</span> <span class="st">'species'</span>)</span>
<span id="cb40-1623"><a href="#cb40-1623" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1624"><a href="#cb40-1624" aria-hidden="true" tabindex="-1"></a>play <span class="ot">&lt;-</span> hv_species_counts <span class="sc">%&gt;%</span> </span>
<span id="cb40-1625"><a href="#cb40-1625" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb40-1626"><a href="#cb40-1626" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by =</span> <span class="st">'sample'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-1627"><a href="#cb40-1627" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(species_family, <span class="at">by =</span> <span class="st">'name'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-1628"><a href="#cb40-1628" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">ifelse</span>(name <span class="sc">==</span> <span class="st">"Severe acute respiratory syndrome-related coronavirus"</span>, <span class="st">"SARS-related coronavirus"</span>, name))</span>
<span id="cb40-1629"><a href="#cb40-1629" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1630"><a href="#cb40-1630" aria-hidden="true" tabindex="-1"></a>adjusted_play <span class="ot">&lt;-</span> play <span class="sc">%&gt;%</span> </span>
<span id="cb40-1631"><a href="#cb40-1631" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span></span>
<span id="cb40-1632"><a href="#cb40-1632" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">virus_prevalence_num =</span> <span class="fu">n_distinct</span>(sample)<span class="sc">/</span><span class="fu">n_distinct</span>(libraries),</span>
<span id="cb40-1633"><a href="#cb40-1633" aria-hidden="true" tabindex="-1"></a>         <span class="at">total_reads_hv =</span> <span class="fu">sum</span>(n_reads_hv)) <span class="sc">%&gt;%</span></span>
<span id="cb40-1634"><a href="#cb40-1634" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb40-1635"><a href="#cb40-1635" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">fct_reorder</span>(name, virus_prevalence_num, <span class="at">.desc=</span><span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb40-1636"><a href="#cb40-1636" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, ra_reads_hv, family, location, virus_prevalence_num, total_reads_hv, n_reads_hv)</span>
<span id="cb40-1637"><a href="#cb40-1637" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1638"><a href="#cb40-1638" aria-hidden="true" tabindex="-1"></a>pal <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">'Dark2'</span>),<span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">'Accent'</span>))</span>
<span id="cb40-1639"><a href="#cb40-1639" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1640"><a href="#cb40-1640" aria-hidden="true" tabindex="-1"></a><span class="co">#, labels = label_log(digits=2)</span></span>
<span id="cb40-1641"><a href="#cb40-1641" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1642"><a href="#cb40-1642" aria-hidden="true" tabindex="-1"></a>ra_dot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(adjusted_play, <span class="fu">aes</span>(<span class="at">x =</span> ra_reads_hv, <span class="at">y=</span>name)) <span class="sc">+</span></span>
<span id="cb40-1643"><a href="#cb40-1643" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_quasirandom</span>(<span class="at">orientation =</span> <span class="st">'y'</span>, <span class="fu">aes</span>(<span class="at">color =</span> family)) <span class="sc">+</span></span>
<span id="cb40-1644"><a href="#cb40-1644" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> pal) <span class="sc">+</span> </span>
<span id="cb40-1645"><a href="#cb40-1645" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_log10</span>(</span>
<span id="cb40-1646"><a href="#cb40-1646" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Relative abundance human virus reads"</span>,</span>
<span id="cb40-1647"><a href="#cb40-1647" aria-hidden="true" tabindex="-1"></a>    <span class="at">labels=</span><span class="fu">label_log</span>(<span class="at">digits=</span><span class="dv">3</span>)</span>
<span id="cb40-1648"><a href="#cb40-1648" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb40-1649"><a href="#cb40-1649" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span>  <span class="st">""</span>,</span>
<span id="cb40-1650"><a href="#cb40-1650" aria-hidden="true" tabindex="-1"></a>       <span class="at">color =</span> <span class="st">'Viral family'</span>) <span class="sc">+</span> </span>
<span id="cb40-1651"><a href="#cb40-1651" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color =</span> <span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">2</span>)) <span class="sc">+</span></span>
<span id="cb40-1652"><a href="#cb40-1652" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_light</span>() <span class="sc">+</span> </span>
<span id="cb40-1653"><a href="#cb40-1653" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(</span>
<span id="cb40-1654"><a href="#cb40-1654" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb40-1655"><a href="#cb40-1655" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>),</span>
<span id="cb40-1656"><a href="#cb40-1656" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.ticks.y =</span> <span class="fu">element_blank</span>(),<span class="at">axis.line =</span> <span class="fu">element_line</span>(<span class="at">colour =</span> <span class="st">"black"</span>),</span>
<span id="cb40-1657"><a href="#cb40-1657" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">14</span>),    </span>
<span id="cb40-1658"><a href="#cb40-1658" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb40-1659"><a href="#cb40-1659" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>, <span class="at">face=</span><span class="st">"bold"</span>),</span>
<span id="cb40-1660"><a href="#cb40-1660" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">'bottom'</span>,</span>
<span id="cb40-1661"><a href="#cb40-1661" aria-hidden="true" tabindex="-1"></a>    <span class="co">#legend.position = c(1, 1),  # Move legend to top right</span></span>
<span id="cb40-1662"><a href="#cb40-1662" aria-hidden="true" tabindex="-1"></a>    <span class="co">#legend.justification = c(1, 1),  # Align legend to top right</span></span>
<span id="cb40-1663"><a href="#cb40-1663" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb40-1664"><a href="#cb40-1664" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.border =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb40-1665"><a href="#cb40-1665" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.background =</span> <span class="fu">element_blank</span>())</span>
<span id="cb40-1666"><a href="#cb40-1666" aria-hidden="true" tabindex="-1"></a><span class="co">#  geom_text(aes(label = total_reads_hv, y = name), </span></span>
<span id="cb40-1667"><a href="#cb40-1667" aria-hidden="true" tabindex="-1"></a><span class="co">#            x = 1e-9, hjust = 0, vjust = 0.5, size = 3, </span></span>
<span id="cb40-1668"><a href="#cb40-1668" aria-hidden="true" tabindex="-1"></a><span class="co">#            check_overlap = TRUE)</span></span>
<span id="cb40-1669"><a href="#cb40-1669" aria-hidden="true" tabindex="-1"></a>prev_play <span class="ot">&lt;-</span> adjusted_play <span class="sc">%&gt;%</span></span>
<span id="cb40-1670"><a href="#cb40-1670" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, location) <span class="sc">%&gt;%</span></span>
<span id="cb40-1671"><a href="#cb40-1671" aria-hidden="true" tabindex="-1"></a>  <span class="fu">reframe</span>(<span class="at">location_of_reads =</span> n_reads_hv<span class="sc">/</span>total_reads_hv,</span>
<span id="cb40-1672"><a href="#cb40-1672" aria-hidden="true" tabindex="-1"></a>          <span class="at">total_reads_hv =</span> <span class="fu">first</span>(total_reads_hv))</span>
<span id="cb40-1673"><a href="#cb40-1673" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1674"><a href="#cb40-1674" aria-hidden="true" tabindex="-1"></a>prevalence_bar <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(prev_play, <span class="fu">aes</span>(<span class="at">x =</span> location_of_reads, <span class="at">y =</span> name, <span class="at">fill =</span> location)) <span class="sc">+</span></span>
<span id="cb40-1675"><a href="#cb40-1675" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">position =</span> <span class="st">"stack"</span>) <span class="sc">+</span></span>
<span id="cb40-1676"><a href="#cb40-1676" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Accent"</span>) <span class="sc">+</span></span>
<span id="cb40-1677"><a href="#cb40-1677" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(</span>
<span id="cb40-1678"><a href="#cb40-1678" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Relative composition of locations with reads"</span>,</span>
<span id="cb40-1679"><a href="#cb40-1679" aria-hidden="true" tabindex="-1"></a>    <span class="at">labels =</span> scales<span class="sc">::</span><span class="fu">percent_format</span>(<span class="at">scale =</span> <span class="dv">100</span>, <span class="at">accuracy =</span> <span class="dv">1</span>),</span>
<span id="cb40-1680"><a href="#cb40-1680" aria-hidden="true" tabindex="-1"></a>    <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>)</span>
<span id="cb40-1681"><a href="#cb40-1681" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb40-1682"><a href="#cb40-1682" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">''</span>,</span>
<span id="cb40-1683"><a href="#cb40-1683" aria-hidden="true" tabindex="-1"></a>       <span class="at">fill =</span> <span class="st">'Location'</span>) <span class="sc">+</span></span>
<span id="cb40-1684"><a href="#cb40-1684" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill =</span> <span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb40-1685"><a href="#cb40-1685" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_light</span>() <span class="sc">+</span> </span>
<span id="cb40-1686"><a href="#cb40-1686" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb40-1687"><a href="#cb40-1687" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb40-1688"><a href="#cb40-1688" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>),</span>
<span id="cb40-1689"><a href="#cb40-1689" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.ticks.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb40-1690"><a href="#cb40-1690" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.line =</span> <span class="fu">element_line</span>(<span class="at">colour =</span> <span class="st">"black"</span>),</span>
<span id="cb40-1691"><a href="#cb40-1691" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">14</span>),    </span>
<span id="cb40-1692"><a href="#cb40-1692" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(),  </span>
<span id="cb40-1693"><a href="#cb40-1693" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb40-1694"><a href="#cb40-1694" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>, <span class="at">face =</span> <span class="st">"bold"</span>),</span>
<span id="cb40-1695"><a href="#cb40-1695" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">'bottom'</span>, </span>
<span id="cb40-1696"><a href="#cb40-1696" aria-hidden="true" tabindex="-1"></a>    <span class="co">#legend.position = c(1, 1),  # Move legend to top right</span></span>
<span id="cb40-1697"><a href="#cb40-1697" aria-hidden="true" tabindex="-1"></a>    <span class="co">#legend.justification = c(1, 1),  # Align legend to top right</span></span>
<span id="cb40-1698"><a href="#cb40-1698" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb40-1699"><a href="#cb40-1699" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.border =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb40-1700"><a href="#cb40-1700" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.background =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span> </span>
<span id="cb40-1701"><a href="#cb40-1701" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label =</span> total_reads_hv, <span class="at">y =</span> name), </span>
<span id="cb40-1702"><a href="#cb40-1702" aria-hidden="true" tabindex="-1"></a>            <span class="at">x =</span> <span class="fl">1e-9</span>, <span class="at">hjust =</span> <span class="dv">0</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">3</span>, </span>
<span id="cb40-1703"><a href="#cb40-1703" aria-hidden="true" tabindex="-1"></a>            <span class="at">check_overlap =</span> <span class="cn">TRUE</span>, <span class="at">fontface =</span> <span class="st">"bold"</span>)</span>
<span id="cb40-1704"><a href="#cb40-1704" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine the plots using ggarrange</span></span>
<span id="cb40-1705"><a href="#cb40-1705" aria-hidden="true" tabindex="-1"></a>combined_plot <span class="ot">&lt;-</span> <span class="fu">ggarrange</span>(ra_dot, prevalence_bar, </span>
<span id="cb40-1706"><a href="#cb40-1706" aria-hidden="true" tabindex="-1"></a>                           <span class="at">ncol =</span> <span class="dv">2</span>, </span>
<span id="cb40-1707"><a href="#cb40-1707" aria-hidden="true" tabindex="-1"></a>                           <span class="at">widths =</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">1</span>),</span>
<span id="cb40-1708"><a href="#cb40-1708" aria-hidden="true" tabindex="-1"></a>                           <span class="at">common.legend =</span> <span class="cn">FALSE</span>,</span>
<span id="cb40-1709"><a href="#cb40-1709" aria-hidden="true" tabindex="-1"></a>                           <span class="at">align =</span> <span class="st">"h"</span>)</span>
<span id="cb40-1710"><a href="#cb40-1710" aria-hidden="true" tabindex="-1"></a>combined_plot</span>
<span id="cb40-1711"><a href="#cb40-1711" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>