<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en">

<head>

<meta charset="utf-8" />
<meta name="generator" content="quarto-1.5.57" />

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />

<meta name="author" content="Harmon Bhasin" />
<meta name="dcterms.date" content="2024-07-22" />

<title>Workflow of Thijssen et al. (2023) – Harmon's Public Notebook</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>

<!-- htmldependencies:E3FAD763 -->
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script async src="https://hypothes.is/embed.js"></script>
<script>
  window.document.addEventListener("DOMContentLoaded", function (_event) {
    document.body.classList.add('hypothesis-enabled');
  });
</script>
<style>

      .quarto-title-block .quarto-title-banner {
        background: #de2d26;
      }
</style>
<script src="../site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<script src="../site_libs/d3-4.5.0/d3.min.js"></script>
<script src="../site_libs/sankey-1/sankey.js"></script>
<script src="../site_libs/sankeyNetwork-binding-0.4/sankeyNetwork.js"></script>
<link href="../site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="../site_libs/pagedtable-1.1/js/pagedtable.js"></script>

  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

<link rel="stylesheet" href="../styles.css" />
</head>

<body>

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="/index.html">
    <span class="navbar-title">Harmon&#39;s Public Notebook</span>
    </a>
  </div>
        <div class="quarto-navbar-tools tools-end">
</div>
          <div id="quarto-search" class="" title="Search"></div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <div id="quarto-toc-target"></div>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">


<header id="title-block-header" class="quarto-title-block default">
  <div class="quarto-title-banner">
    <div class="quarto-title column-body">
      <h1 class="title">Workflow of Thijssen et al. (2023)</h1>
            <p class="subtitle lead">Pooled plasma from Iran (DNA + RNA)</p>
                      </div>
  </div>
    
  
  <div class="quarto-title-meta">

      <div>
      <div class="quarto-title-meta-heading">Author</div>
      <div class="quarto-title-meta-contents">
               <p>Harmon Bhasin </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">July 22, 2024</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header>

<nav id="TOC" role="doc-toc">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#the-raw-data" id="toc-the-raw-data"><span class="header-section-number">1</span> The raw data</a>
  <ul>
  <li><a href="#about" id="toc-about"><span class="header-section-number">1.1</span> About</a>
  <ul>
  <li><a href="#sample-library-preparation" id="toc-sample-library-preparation"><span class="header-section-number">1.1.1</span> Sample + library preparation</a></li>
  </ul></li>
  <li><a href="#quality-control-metrics" id="toc-quality-control-metrics"><span class="header-section-number">1.2</span> Quality control metrics</a></li>
  </ul></li>
  <li><a href="#preprocessing" id="toc-preprocessing"><span class="header-section-number">2</span> Preprocessing</a>
  <ul>
  <li><a href="#summary" id="toc-summary"><span class="header-section-number">2.1</span> Summary</a></li>
  <li><a href="#quality-control-metrics-1" id="toc-quality-control-metrics-1"><span class="header-section-number">2.2</span> Quality control metrics</a></li>
  </ul></li>
  <li><a href="#taxonomic-composition" id="toc-taxonomic-composition"><span class="header-section-number">3</span> Taxonomic composition</a>
  <ul>
  <li><a href="#high-level-composition" id="toc-high-level-composition"><span class="header-section-number">3.1</span> High-level composition</a></li>
  <li><a href="#total-viral-content" id="toc-total-viral-content"><span class="header-section-number">3.2</span> Total viral content</a></li>
  <li><a href="#taxonomic-composition-of-viruses" id="toc-taxonomic-composition-of-viruses"><span class="header-section-number">3.3</span> Taxonomic composition of viruses</a></li>
  </ul></li>
  <li><a href="#human-infecting-virus-reads" id="toc-human-infecting-virus-reads"><span class="header-section-number">4</span> Human-infecting virus reads</a>
  <ul>
  <li><a href="#overall-relative-abundance" id="toc-overall-relative-abundance"><span class="header-section-number">4.1</span> Overall relative abundance</a></li>
  <li><a href="#overall-taxonomy-and-composition" id="toc-overall-taxonomy-and-composition"><span class="header-section-number">4.2</span> Overall taxonomy and composition</a></li>
  <li><a href="#analyzing-specific-families" id="toc-analyzing-specific-families"><span class="header-section-number">4.3</span> Analyzing specific families</a>
  <ul>
  <li><a href="#anelloviridae-number-of-reads-12804" id="toc-anelloviridae-number-of-reads-12804"><span class="header-section-number">4.3.1</span> Anelloviridae (Number of reads: 12,804)</a></li>
  <li><a href="#microviridae-number-of-reads-5194" id="toc-microviridae-number-of-reads-5194"><span class="header-section-number">4.3.2</span> Microviridae (Number of reads: 5,194)</a></li>
  <li><a href="#sedreoviridae-number-of-reads-278" id="toc-sedreoviridae-number-of-reads-278"><span class="header-section-number">4.3.3</span> Sedreoviridae (Number of reads: 278)</a></li>
  <li><a href="#parvoviridae-number-of-reads-266" id="toc-parvoviridae-number-of-reads-266"><span class="header-section-number">4.3.4</span> Parvoviridae (Number of reads: 266)</a></li>
  <li><a href="#adenoviridae-number-of-reads-104" id="toc-adenoviridae-number-of-reads-104"><span class="header-section-number">4.3.5</span> Adenoviridae (Number of reads: 104)</a></li>
  <li><a href="#polyomaviridae-number-of-reads-95" id="toc-polyomaviridae-number-of-reads-95"><span class="header-section-number">4.3.6</span> Polyomaviridae (Number of reads: 95)</a></li>
  <li><a href="#papillomaviridae-number-of-reads-16" id="toc-papillomaviridae-number-of-reads-16"><span class="header-section-number">4.3.7</span> Papillomaviridae (Number of reads: 16)</a></li>
  </ul></li>
  <li><a href="#relative-abundance-of-pathogenic-viruses-of-interest" id="toc-relative-abundance-of-pathogenic-viruses-of-interest"><span class="header-section-number">4.4</span> Relative abundance of pathogenic viruses of interest</a></li>
  <li><a href="#relative-abundance-assuming-perfect-human-read-removal" id="toc-relative-abundance-assuming-perfect-human-read-removal"><span class="header-section-number">4.5</span> Relative abundance assuming perfect human read removal</a></li>
  </ul></li>
  <li><a href="#conclusion" id="toc-conclusion"><span class="header-section-number">5</span> Conclusion</a></li>
  <li><a href="#appendix" id="toc-appendix"><span class="header-section-number">6</span> Appendix</a>
  <ul>
  <li><a href="#human-infecting-virus-families-genera-and-species" id="toc-human-infecting-virus-families-genera-and-species"><span class="header-section-number">6.1</span> Human-infecting virus families, genera, and species</a></li>
  </ul></li>
  </ul>
</nav>
<p>This is another potential study of <a href="https://data.securebio.org/harmons-public-notebook/notebooks/2024-07-08_cebria-mendoza.html">this series</a>. In this post, I analyze <a href="https://doi.org/10.3390/v15071425">Thijssen 2023</a>, a dataset with 20 pooled samples from 100 healthy individuals in Iran.</p>
<p><em>This notebook uses the MGS Workflow v2.2.1 (note that this is old).</em></p>
<section id="the-raw-data" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> The raw data</h1>
<section id="about" class="level2" data-number="1.1">
<h2 data-number="1.1"><span class="header-section-number">1.1</span> About</h2>
<p><a href="https://www.ncbi.nlm.nih.gov/bioproject/?term=PRJNA983534">This dataset</a> comprises 20 samples, each derived from plasma pools of 5 people from Iran. In total, 100 healthy individuals contributed to these pools. For each pooled sample, a combined DNA and RNA library preparation was performed, resulting in a single sequencing output that captures both nucleic acid types. This approach provides comprehensive genetic information but precludes separate analysis of DNA and RNA from individual samples.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode" id="cb1"><pre class="sourceCode r cell-code"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Import libraries and extract metadata from sample names</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>libraries <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(libraries_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>meta_data <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&#39;/Users/harmonbhasin/work/securebio/nao-harmon/thijssen2023/preprocessing/SraRunTable.txt&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="st">&#39;Library Name&#39;</span>, Run) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">sample =</span> Run, <span class="at">library =</span> <span class="st">&#39;Library Name&#39;</span>)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>libraries <span class="ot">&lt;-</span> libraries <span class="sc">%&gt;%</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(meta_data, <span class="at">by =</span> <span class="st">&#39;sample&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, <span class="at">library=</span>library.y) <span class="sc">%&gt;%</span> </span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(library <span class="sc">!=</span> <span class="st">&#39;PCL21&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">library =</span> <span class="fu">factor</span>(library, <span class="at">levels =</span> <span class="fu">c</span>(<span class="fu">paste0</span>(<span class="st">&#39;PCL&#39;</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>))))</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>libraries</span></code></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source type="application/json">
{"columns":[{"label":["sample"],"name":[1],"type":["chr"],"align":["left"]},{"label":["library"],"name":[2],"type":["fct"],"align":["left"]}],"data":[{"1":"SRR24991051","2":"PCL16"},{"1":"SRR24991062","2":"PCL15"},{"1":"SRR24991071","2":"PCL18"},{"1":"SRR24991073","2":"PCL17"},{"1":"SRR24991082","2":"PCL14"},{"1":"SRR24991093","2":"PCL13"},{"1":"SRR24991104","2":"PCL12"},{"1":"SRR24991114","2":"PCL9"},{"1":"SRR24991115","2":"PCL11"},{"1":"SRR24991116","2":"PCL8"},{"1":"SRR24991117","2":"PCL7"},{"1":"SRR24991118","2":"PCL6"},{"1":"SRR24991119","2":"PCL5"},{"1":"SRR24991120","2":"PCL4"},{"1":"SRR24991121","2":"PCL3"},{"1":"SRR24991123","2":"PCL20"},{"1":"SRR24991124","2":"PCL2"},{"1":"SRR24991125","2":"PCL19"},{"1":"SRR24991126","2":"PCL10"},{"1":"SRR24991127","2":"PCL1"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<section id="sample-library-preparation" class="level3" data-number="1.1.1">
<h3 data-number="1.1.1"><span class="header-section-number">1.1.1</span> Sample + library preparation</h3>
<p>Paraphrased from the <a href="https://www.mdpi.com/1999-4915/15/7/1425">paper</a>:</p>
<blockquote>
<p>Study was conducted from 2017-2018 in the Boushehr Province, Iran. Participants were recruited in one of the five transfusion clinics located in Boushehr. Seven milliliters of blood were collected from each individual during blood transfusion or donation. Immediately after collection, plasma was separated from the samples and stored at −70 °C.</p>
<p>Initially, the samples were centrifuged and 100 µL of the supernatant was pooled with five samples per pool. The pooled plasma samples were subjected to an adapted version of the NetoVIR protocol for viral particle enrichment and metagenomic sequencing (<a href="https://www.mdpi.com/1999-4915/15/7/1425#B20-viruses-15-01425">more information can be found here</a>). Pooled plasma samples were centrifuged for 3 min at 17,000× g and filtered through 0.8 µm polyether sulphone filters. To remove free-floating nucleic acids, the filtered samples were subjected to a nuclease treatment with a cocktail of 1 µL micrococcal nuclease and 2 µL benzonase for 2 h at 37 °C. Both viral DNA and RNA were extracted and randomly amplified (including primary step of reverse transcription) with the Whole Transcriptome Amplification 2 kit for 20 cycles.</p>
<p>The amplification product was purified with the MSB SPIN PCRAPACE kit and prepared for sequencing by using the Nextera XT kit. DNA products were quantified with the Qubit fluorometer, and the High-Sensitivity DNA kit for the Bioanalyzer 2100 was used to determine the average library fragment size. Samples were pooled in equimolar ratios, and paired-end sequencing (2 × 150 bp) was performed on a NextSeq 500 Illumina platform with an average of 10 million reads per sample.</p>
</blockquote>
</section>
</section>
<section id="quality-control-metrics" class="level2" data-number="1.2">
<h2 data-number="1.2"><span class="header-section-number">1.2</span> Quality control metrics</h2>
<p>In total, these 20 samples contained ~116M read pairs. The samples had 1.3M - 11.1M (mean ~5.8M) read pairs each.</p>
<p>The number of reads looks pretty good, although a few samples have a much lower number of reads. The total number of base pairs also looks reasonable, matching the trends of the number of reads. The duplication rate is low. Adapter content is quite high for nextera-transposase-sequence (<a href="https://www.biostars.org/p/119774/">library contamination</a>).</p>
<p>As we’d expect, we see higher quality Q scores near the beginning of the read and a gradual decline towards the end of the read, however all positions seem to have a Q score of 35 which means that our reads are ~99.97% accurate. When looking at the Q score over all sequences, we can see a sharp peak around 35, which corresponds to the previous plot, indicating high read quality.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode" id="cb2"><pre class="sourceCode r cell-code"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare data</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>basic_stats_raw_metrics <span class="ot">&lt;-</span> basic_stats_raw <span class="sc">%&gt;%</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(library,</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at"># Read pairs</span><span class="st">`</span> <span class="ot">=</span> n_read_pairs,</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Total base pairs</span><span class="sc">\n</span><span class="at">(approx)</span><span class="st">`</span> <span class="ot">=</span> n_bases_approx,</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">% Duplicates</span><span class="sc">\n</span><span class="at">(FASTQC)</span><span class="st">`</span> <span class="ot">=</span> percent_duplicates) <span class="sc">%&gt;%</span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>library, <span class="at">names_to =</span> <span class="st">&quot;metric&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;value&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">metric =</span> <span class="fu">fct_inorder</span>(metric)) </span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Set up plot templates</span></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>g_basic <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(basic_stats_raw_metrics, <span class="fu">aes</span>(<span class="at">x =</span> library, <span class="at">y =</span> value)) <span class="sc">+</span></span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position =</span> <span class="st">&quot;dodge&quot;</span>) <span class="sc">+</span></span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>() <span class="sc">+</span></span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">expand_limits</span>(<span class="at">y =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">100</span>)) <span class="sc">+</span></span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(metric <span class="sc">~</span> ., <span class="at">scales =</span> <span class="st">&quot;free&quot;</span>, <span class="at">space =</span> <span class="st">&quot;free_x&quot;</span>, <span class="at">switch =</span> <span class="st">&quot;y&quot;</span>) <span class="sc">+</span></span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a>  theme_kit <span class="sc">+</span> <span class="fu">theme</span>(</span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">strip.text.y =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">&quot;plain&quot;</span>)</span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a>g_basic</span></code></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure>
<p><img src="2024-07-22-thijssen_files/figure-html/plot-basic-stats-1.svg" class="img-fluid" /></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode" id="cb3"><pre class="sourceCode r cell-code"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Set up plotting templates</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>g_qual_raw <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">linetype=</span>read_pair, <span class="at">group=</span><span class="fu">interaction</span>(sample,read_pair))) <span class="sc">+</span> </span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_linetype_discrete</span>(<span class="at">name =</span> <span class="st">&quot;Read Pair&quot;</span>) <span class="sc">+</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color=</span><span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>),</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">linetype =</span> <span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>)) <span class="sc">+</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>  theme_base</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize adapters</span></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>g_adapters_raw <span class="ot">&lt;-</span> g_qual_raw <span class="sc">+</span> </span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>pc_adapters), <span class="at">data=</span>adapter_stats_raw) <span class="sc">+</span></span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">&quot;% Adapters&quot;</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">100</span>,<span class="dv">10</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">&quot;Position&quot;</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">500</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(.<span class="sc">~</span>adapter)</span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a>g_adapters_raw</span></code></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure>
<p><img src="2024-07-22-thijssen_files/figure-html/plot-raw-quality-1.svg" class="img-fluid" /></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode" id="cb4"><pre class="sourceCode r cell-code"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize quality</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>g_quality_base_raw <span class="ot">&lt;-</span> g_qual_raw <span class="sc">+</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">&quot;dashed&quot;</span>, <span class="at">color=</span><span class="st">&quot;red&quot;</span>) <span class="sc">+</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">&quot;dashed&quot;</span>, <span class="at">color=</span><span class="st">&quot;red&quot;</span>) <span class="sc">+</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>mean_phred_score), <span class="at">data=</span>quality_base_stats_raw) <span class="sc">+</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">&quot;Mean Phred score&quot;</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">10</span>,<span class="dv">45</span>)) <span class="sc">+</span></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">&quot;Position&quot;</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">500</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>g_quality_base_raw</span></code></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure>
<p><img src="2024-07-22-thijssen_files/figure-html/plot-raw-quality-2.svg" class="img-fluid" /></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode" id="cb5"><pre class="sourceCode r cell-code"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>g_quality_seq_raw <span class="ot">&lt;-</span> g_qual_raw <span class="sc">+</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">&quot;dashed&quot;</span>, <span class="at">color=</span><span class="st">&quot;red&quot;</span>) <span class="sc">+</span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">&quot;dashed&quot;</span>, <span class="at">color=</span><span class="st">&quot;red&quot;</span>) <span class="sc">+</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>mean_phred_score, <span class="at">y=</span>n_sequences), <span class="at">data=</span>quality_seq_stats_raw) <span class="sc">+</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">&quot;Mean Phred score&quot;</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">&quot;# Sequences&quot;</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>g_quality_seq_raw</span></code></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure>
<p><img src="2024-07-22-thijssen_files/figure-html/plot-raw-quality-3.svg" class="img-fluid" /></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="preprocessing" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Preprocessing</h1>
<section id="summary" class="level2" data-number="2.1">
<h2 data-number="2.1"><span class="header-section-number">2.1</span> Summary</h2>
<p>The average fraction of reads at each stage in the preprocessing pipeline is shown in the following table. Adapter trimming &amp; filtering doesn’t lose too many reads. Deduplication loses us about 30% of reads on average, then ribodepletion only loses as about 0.7% on average.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode" id="cb6"><pre class="sourceCode r cell-code"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># </span><span class="al">TODO</span><span class="co">: Group by pool size as well</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Count read losses</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>n_reads_rel <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> </span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, stage, percent_duplicates, n_read_pairs) <span class="sc">%&gt;%</span></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(sample, stage) <span class="sc">%&gt;%</span></span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_retained =</span> n_read_pairs <span class="sc">/</span> <span class="fu">lag</span>(n_read_pairs),</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_lost =</span> <span class="dv">1</span> <span class="sc">-</span> p_reads_retained,</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_retained_abs =</span> n_read_pairs <span class="sc">/</span> n_read_pairs[<span class="dv">1</span>],</span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_lost_abs =</span> <span class="dv">1</span><span class="sc">-</span>p_reads_retained_abs,</span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_lost_abs_marginal =</span> p_reads_lost_abs <span class="sc">-</span> <span class="fu">lag</span>(p_reads_lost_abs))</span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>n_reads_rel_display <span class="ot">&lt;-</span> n_reads_rel <span class="sc">%&gt;%</span> </span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">Stage=</span>stage) <span class="sc">%&gt;%</span> </span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Stage) <span class="sc">%&gt;%</span> </span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="st">`</span><span class="at">% Total Reads Lost (Cumulative)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">round</span>(<span class="fu">min</span>(p_reads_lost_abs<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">&quot;-&quot;</span>, <span class="fu">round</span>(<span class="fu">max</span>(p_reads_lost_abs<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">&quot; (mean &quot;</span>, <span class="fu">round</span>(<span class="fu">mean</span>(p_reads_lost_abs<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">&quot;)&quot;</span>),</span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a>            <span class="st">`</span><span class="at">% Total Reads Lost (Marginal)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">round</span>(<span class="fu">min</span>(p_reads_lost_abs_marginal<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">&quot;-&quot;</span>, <span class="fu">round</span>(<span class="fu">max</span>(p_reads_lost_abs_marginal<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">&quot; (mean &quot;</span>, <span class="fu">round</span>(<span class="fu">mean</span>(p_reads_lost_abs_marginal<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">&quot;)&quot;</span>), <span class="at">.groups=</span><span class="st">&quot;drop&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Stage <span class="sc">!=</span> <span class="st">&quot;raw_concat&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Stage =</span> Stage <span class="sc">%&gt;%</span> as.numeric <span class="sc">%&gt;%</span> <span class="fu">factor</span>(<span class="at">labels=</span><span class="fu">c</span>(<span class="st">&quot;Trimming &amp; filtering&quot;</span>, <span class="st">&quot;Deduplication&quot;</span>, <span class="st">&quot;Initial ribodepletion&quot;</span>, <span class="st">&quot;Secondary ribodepletion&quot;</span>)))</span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a>n_reads_rel_display</span></code></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source type="application/json">
{"columns":[{"label":["Stage"],"name":[1],"type":["fct"],"align":["left"]},{"label":["% Total Reads Lost (Cumulative)"],"name":[2],"type":["chr"],"align":["left"]},{"label":["% Total Reads Lost (Marginal)"],"name":[3],"type":["chr"],"align":["left"]}],"data":[{"1":"Trimming & filtering","2":"0.8-1.8 (mean 1.2)","3":"0.8-1.8 (mean 1.2)"},{"1":"Deduplication","2":"18.8-45.3 (mean 31.7)","3":"17.9-44.4 (mean 30.6)"},{"1":"Initial ribodepletion","2":"19-46.4 (mean 32.2)","3":"0-1 (mean 0.4)"},{"1":"Secondary ribodepletion","2":"19-46.7 (mean 32.5)","3":"0-0.7 (mean 0.3)"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
</section>
<section id="quality-control-metrics-1" class="level2" data-number="2.2">
<h2 data-number="2.2"><span class="header-section-number">2.2</span> Quality control metrics</h2>
<p>Adapter trimming gets rid of nextera_transposase_sequenece and poly_g, although there is still a bit of poly_a adapter contamination (<a href="https://github.com/s-andrews/FastQC/blob/master/Configuration/adapter_list.txt">overivew of fastqc adapters</a>). Not sure how to interpret the poly_a contamination, but it’s not that high so I’m not going to worry about it for now. Q score remain the same during read cleaning when looking at the positions, with the end of the read actually improving in score. Q scores across all sequences look pretty much the same throughout cleaning.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode" id="cb7"><pre class="sourceCode r cell-code"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>g_qual <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">linetype=</span>read_pair, <span class="at">group=</span><span class="fu">interaction</span>(sample,read_pair))) <span class="sc">+</span> </span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_linetype_discrete</span>(<span class="at">name =</span> <span class="st">&quot;Read Pair&quot;</span>) <span class="sc">+</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color=</span><span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>),</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">linetype =</span> <span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>)) <span class="sc">+</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>  theme_base</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize adapters</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>g_adapters <span class="ot">&lt;-</span> g_qual <span class="sc">+</span> </span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>pc_adapters), <span class="at">data=</span>adapter_stats) <span class="sc">+</span></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">&quot;% Adapters&quot;</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">20</span>),</span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">50</span>,<span class="dv">10</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">&quot;Position&quot;</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">140</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(stage<span class="sc">~</span>adapter)</span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a>g_adapters</span></code></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure>
<p><img src="2024-07-22-thijssen_files/figure-html/plot-quality-1.svg" class="img-fluid" /></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode" id="cb8"><pre class="sourceCode r cell-code"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize quality</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>g_quality_base <span class="ot">&lt;-</span> g_qual <span class="sc">+</span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">&quot;dashed&quot;</span>, <span class="at">color=</span><span class="st">&quot;red&quot;</span>) <span class="sc">+</span></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">&quot;dashed&quot;</span>, <span class="at">color=</span><span class="st">&quot;red&quot;</span>) <span class="sc">+</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>mean_phred_score), <span class="at">data=</span>quality_base_stats) <span class="sc">+</span></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">&quot;Mean Phred score&quot;</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">10</span>,<span class="dv">45</span>)) <span class="sc">+</span></span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">&quot;Position&quot;</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">140</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(stage<span class="sc">~</span>.)</span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>g_quality_base</span></code></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure>
<p><img src="2024-07-22-thijssen_files/figure-html/plot-quality-2.svg" class="img-fluid" /></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode" id="cb9"><pre class="sourceCode r cell-code"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>g_quality_seq <span class="ot">&lt;-</span> g_qual <span class="sc">+</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">&quot;dashed&quot;</span>, <span class="at">color=</span><span class="st">&quot;red&quot;</span>) <span class="sc">+</span></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">&quot;dashed&quot;</span>, <span class="at">color=</span><span class="st">&quot;red&quot;</span>) <span class="sc">+</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>mean_phred_score, <span class="at">y=</span>n_sequences), <span class="at">data=</span>quality_seq_stats) <span class="sc">+</span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">&quot;Mean Phred score&quot;</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">&quot;# Sequences&quot;</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(stage<span class="sc">~</span>., <span class="at">scales =</span> <span class="st">&quot;free_y&quot;</span>)</span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>g_quality_seq</span></code></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure>
<p><img src="2024-07-22-thijssen_files/figure-html/plot-quality-3.svg" class="img-fluid" /></p>
</figure>
</div>
</div>
</div>
<p>These plots below show the trends from above in each sample. All samples tend to follow similar trends for deduplication, with a large decrease in read length post adapter trimming. Ribosomal reads were quite low, near 1% for every sample.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode" id="cb10"><pre class="sourceCode r cell-code"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>g_stage_trace <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(basic_stats, <span class="fu">aes</span>(<span class="at">x=</span>stage, <span class="at">group=</span>sample)) <span class="sc">+</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot reads over preprocessing</span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>g_reads_stages <span class="ot">&lt;-</span> g_stage_trace <span class="sc">+</span></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>n_read_pairs)) <span class="sc">+</span></span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">&quot;# Read pairs&quot;</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>))</span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>g_reads_stages</span></code></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure>
<p><img src="2024-07-22-thijssen_files/figure-html/preproc-figures-1.svg" class="img-fluid" /></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode" id="cb11"><pre class="sourceCode r cell-code"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot relative read losses during preprocessing</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>g_reads_rel <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(n_reads_rel, </span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">aes</span>(<span class="at">x=</span>stage, <span class="at">group=</span>sample)) <span class="sc">+</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>p_reads_lost_abs_marginal)) <span class="sc">+</span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">&quot;% Total Reads Lost&quot;</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), </span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="cf">function</span>(x) x<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>g_reads_rel</span></code></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure>
<p><img src="2024-07-22-thijssen_files/figure-html/preproc-figures-2.svg" class="img-fluid" /></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode" id="cb12"><pre class="sourceCode r cell-code"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>stage_dup <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(stage) <span class="sc">%&gt;%</span> </span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">dmin =</span> <span class="fu">min</span>(percent_duplicates), <span class="at">dmax=</span><span class="fu">max</span>(percent_duplicates),</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>            <span class="at">dmean=</span><span class="fu">mean</span>(percent_duplicates), <span class="at">.groups =</span> <span class="st">&quot;drop&quot;</span>)</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>g_dup_stages <span class="ot">&lt;-</span> g_stage_trace <span class="sc">+</span></span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>percent_duplicates)) <span class="sc">+</span></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">&quot;% Duplicates&quot;</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>g_dup_stages</span></code></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure>
<p><img src="2024-07-22-thijssen_files/figure-html/preproc-dedup-1.svg" class="img-fluid" /></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode" id="cb13"><pre class="sourceCode r cell-code"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>g_readlen_stages <span class="ot">&lt;-</span> g_stage_trace <span class="sc">+</span> <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>mean_seq_len)) <span class="sc">+</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">&quot;Mean read length (nt)&quot;</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>))</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>g_readlen_stages</span></code></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure>
<p><img src="2024-07-22-thijssen_files/figure-html/preproc-dedup-2.svg" class="img-fluid" /></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode" id="cb14"><pre class="sourceCode r cell-code"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate reads lost during ribodepletion (approximation for % ribosomal reads)</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>reads_ribo <span class="ot">&lt;-</span> n_reads_rel <span class="sc">%&gt;%</span> </span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(stage <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;dedup&quot;</span>, <span class="st">&quot;ribo_secondary&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads_ribo=</span><span class="dv">1</span><span class="sc">-</span>n_read_pairs[<span class="dv">2</span>]<span class="sc">/</span>n_read_pairs[<span class="dv">1</span>], <span class="at">.groups =</span> <span class="st">&quot;drop&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by =</span> <span class="st">&#39;sample&#39;</span>)</span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>reads_ribo_summ <span class="ot">&lt;-</span> reads_ribo <span class="sc">%&gt;%</span></span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">min=</span><span class="fu">min</span>(p_reads_ribo), <span class="at">max=</span><span class="fu">max</span>(p_reads_ribo),</span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>            <span class="at">mean=</span><span class="fu">mean</span>(p_reads_ribo), <span class="at">.groups =</span> <span class="st">&quot;drop&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by =</span> <span class="st">&#39;sample&#39;</span>)</span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a>g_reads_ribo <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(reads_ribo, </span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">aes</span>(<span class="at">x=</span>library, <span class="at">y=</span>p_reads_ribo)) <span class="sc">+</span></span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span> </span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">&quot;Approx % ribosomal reads&quot;</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>),</span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>)<span class="sc">+</span></span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb14-18"><a href="#cb14-18" aria-hidden="true" tabindex="-1"></a>g_reads_ribo</span></code></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure>
<p><img src="2024-07-22-thijssen_files/figure-html/ribo-frac-1.svg" class="img-fluid" /></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="taxonomic-composition" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> Taxonomic composition</h1>
<section id="high-level-composition" class="level2" data-number="3.1">
<h2 data-number="3.1"><span class="header-section-number">3.1</span> High-level composition</h2>
<p>To assess the high-level composition of the reads, I ran the ribodepleted files through Kraken2 and summarized the results with Bracken.</p>
<p>The groups listed below were created by Will:</p>
<ul>
<li>Filtered (removed during cleaning)</li>
<li>Duplicate (removed during deduplication)</li>
<li>Ribosomal (removed during ribodepletion)</li>
<li>Unassigned (non-ribosomal reads that were not assigned to any taxon by Kraken/Bracken)</li>
<li>Bacterial (non-ribosomal reads assigned to the Bacteria domain by Kraken/Bracken)</li>
<li>Archaeal (non-ribosomal reads assigned to the Archaea domain by Kraken/Bracken)</li>
<li>Viral (non-ribosomal reads assigned to the Viruses domain by Kraken/Bracken)</li>
<li>Human (non-ribosomal reads assigned to the Eukarya domain by Kraken/Bracken)</li>
</ul>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode" id="cb15"><pre class="sourceCode r cell-code"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare plotting templates</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>g_comp_base <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">x=</span>library, <span class="at">y=</span>p_reads, <span class="at">fill=</span>classification)) <span class="sc">+</span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">name=</span><span class="st">&quot;Plasma pool&quot;</span>) <span class="sc">+</span></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>  theme_kit <span class="sc">+</span> </span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust=</span><span class="dv">0</span>, <span class="at">face=</span><span class="st">&quot;plain&quot;</span>, <span class="at">size=</span><span class="fu">rel</span>(<span class="fl">1.5</span>)))</span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>scale_y_pc_reads <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">partial</span>(scale_y_continuous, <span class="at">name =</span> <span class="st">&quot;% Reads&quot;</span>,</span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>)</span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a>geom_comp <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">partial</span>(geom_col, <span class="at">position =</span> <span class="st">&quot;stack&quot;</span>, <span class="at">width =</span> <span class="dv">1</span>)</span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Define a color palette for the classification</span></span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a>classification_colors <span class="ot">&lt;-</span> <span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">&quot;Accent&quot;</span>)</span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(classification_colors) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Filtered&quot;</span>, <span class="st">&quot;Duplicate&quot;</span>, <span class="st">&quot;Ribosomal&quot;</span>, <span class="st">&quot;Unassigned&quot;</span>, <span class="st">&quot;Bacterial&quot;</span>, <span class="st">&quot;Archaeal&quot;</span>, <span class="st">&quot;Viral&quot;</span>, <span class="st">&quot;Human&quot;</span>)</span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a>scale_fill_classification <span class="ot">&lt;-</span> <span class="cf">function</span>() {</span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> classification_colors, <span class="at">name =</span> <span class="st">&quot;Classification&quot;</span>)</span>
<span id="cb15-15"><a href="#cb15-15" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb15-16"><a href="#cb15-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-17"><a href="#cb15-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot overall composition</span></span>
<span id="cb15-18"><a href="#cb15-18" aria-hidden="true" tabindex="-1"></a>g_comp <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> <span class="fu">geom_comp</span>(<span class="at">data =</span> comp) <span class="sc">+</span></span>
<span id="cb15-19"><a href="#cb15-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_pc_reads</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>)) <span class="sc">+</span></span>
<span id="cb15-20"><a href="#cb15-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_classification</span>() <span class="sc">+</span> </span>
<span id="cb15-21"><a href="#cb15-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Read composition (all reads, all groups)&quot;</span>)</span>
<span id="cb15-22"><a href="#cb15-22" aria-hidden="true" tabindex="-1"></a>g_comp</span></code></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure>
<p><img src="2024-07-22-thijssen_files/figure-html/plot-composition-all-1.svg" class="img-fluid" /></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode" id="cb16"><pre class="sourceCode r cell-code"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>g_comp_assigned <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_comp</span>(<span class="at">data =</span> comp_assigned) <span class="sc">+</span></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_pc_reads</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>)) <span class="sc">+</span></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_classification</span>() <span class="sc">+</span> </span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Read composition (assigned reads, all groups)&quot;</span>)</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>g_comp_assigned</span></code></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure>
<p><img src="2024-07-22-thijssen_files/figure-html/plot-composition-all-2.svg" class="img-fluid" /></p>
</figure>
</div>
</div>
</div>
</section>
<section id="total-viral-content" class="level2" data-number="3.2">
<h2 data-number="3.2"><span class="header-section-number">3.2</span> Total viral content</h2>
<p>Total viral fraction average <span class="math inline">\(2.36 \times 10^{-4}\)</span> across samples. As a fraction of assigned (rather than total) reads, this jumped to <span class="math inline">\(3.19 \times 10^{-3}\)</span>:</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode" id="cb17"><pre class="sourceCode r cell-code"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>p_reads_viral_all <span class="ot">&lt;-</span> comp <span class="sc">%&gt;%</span> <span class="fu">filter</span>(classification <span class="sc">==</span> <span class="st">&quot;Viral&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">read_group =</span> <span class="st">&quot;All reads&quot;</span>)</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>p_reads_viral_assigned <span class="ot">&lt;-</span> comp_assigned <span class="sc">%&gt;%</span> <span class="fu">filter</span>(classification <span class="sc">==</span> <span class="st">&quot;Viral&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">read_group =</span> <span class="st">&quot;Classified reads&quot;</span>)</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>p_reads_viral <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(p_reads_viral_all, p_reads_viral_assigned)</span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>g_viral <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(p_reads_viral, <span class="fu">aes</span>(<span class="at">x=</span>library, <span class="at">y=</span>p_reads, <span class="at">color =</span> read_group, <span class="at">group =</span> library)) <span class="sc">+</span></span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">color =</span> <span class="st">&#39;black&#39;</span>) <span class="sc">+</span></span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">name=</span><span class="st">&quot;Plasma pool&quot;</span>) <span class="sc">+</span></span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>(<span class="at">name=</span><span class="st">&quot;Viral read fraction&quot;</span>, <span class="at">labels =</span> <span class="fu">label_log</span>(<span class="at">digits=</span><span class="dv">2</span>), <span class="at">limits =</span> <span class="fu">c</span>(<span class="fl">1e-6</span>, <span class="fl">1e-2</span>)) <span class="sc">+</span></span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a>  theme_kit <span class="sc">+</span> </span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>()</span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a>g_viral</span></code></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure>
<p><img src="2024-07-22-thijssen_files/figure-html/p-viral-1.svg" class="img-fluid" /></p>
</figure>
</div>
</div>
</div>
</section>
<section id="taxonomic-composition-of-viruses" class="level2" data-number="3.3">
<h2 data-number="3.3"><span class="header-section-number">3.3</span> Taxonomic composition of viruses</h2>
<p>The two dominant viruses we see are Anellovirdae and Phycodnaviridae. The threshold for the label “other” are the set of families that make up less than 5% composition in all samples.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode" id="cb18"><pre class="sourceCode r cell-code"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>major_threshold <span class="ot">&lt;-</span> <span class="fl">0.05</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify major viral families</span></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>viral_families_major_tab <span class="ot">&lt;-</span> viral_families <span class="sc">%&gt;%</span> </span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads_viral_max =</span> <span class="fu">max</span>(p_reads_viral), <span class="at">.groups=</span><span class="st">&quot;drop&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_viral_max <span class="sc">&gt;=</span> major_threshold)</span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>viral_families_major_list <span class="ot">&lt;-</span> viral_families_major_tab <span class="sc">%&gt;%</span> <span class="fu">pull</span>(name)</span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>viral_families_major <span class="ot">&lt;-</span> viral_families <span class="sc">%&gt;%</span> </span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(name <span class="sc">%in%</span> viral_families_major_list) <span class="sc">%&gt;%</span></span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, taxid, sample, p_reads_viral)</span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a>viral_families_minor <span class="ot">&lt;-</span> viral_families_major <span class="sc">%&gt;%</span> </span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads_viral_major =</span> <span class="fu">sum</span>(p_reads_viral), <span class="at">.groups =</span> <span class="st">&quot;drop&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="st">&quot;Other&quot;</span>, <span class="at">taxid=</span><span class="cn">NA</span>, <span class="at">p_reads_viral =</span> <span class="dv">1</span><span class="sc">-</span>p_reads_viral_major) <span class="sc">%&gt;%</span></span>
<span id="cb18-16"><a href="#cb18-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, taxid, sample, p_reads_viral)</span>
<span id="cb18-17"><a href="#cb18-17" aria-hidden="true" tabindex="-1"></a>viral_families_display <span class="ot">&lt;-</span> viral_families_major <span class="sc">%&gt;%</span> </span>
<span id="cb18-18"><a href="#cb18-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_rows</span>(viral_families_minor) <span class="sc">%&gt;%</span></span>
<span id="cb18-19"><a href="#cb18-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_viral)) <span class="sc">%&gt;%</span> </span>
<span id="cb18-20"><a href="#cb18-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">factor</span>(name, <span class="at">levels=</span><span class="fu">c</span>(viral_families_major_list, <span class="st">&quot;Other&quot;</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb18-21"><a href="#cb18-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_viral, <span class="at">classification=</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb18-22"><a href="#cb18-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">&#39;sample&#39;</span>)</span>
<span id="cb18-23"><a href="#cb18-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-24"><a href="#cb18-24" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb18-25"><a href="#cb18-25" aria-hidden="true" tabindex="-1"></a>g_families <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb18-26"><a href="#cb18-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_comp</span>(<span class="at">data=</span>viral_families_display) <span class="sc">+</span></span>
<span id="cb18-27"><a href="#cb18-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">&quot;% Viral Reads&quot;</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb18-28"><a href="#cb18-28" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb18-29"><a href="#cb18-29" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb18-30"><a href="#cb18-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">&#39;Accent&#39;</span>)</span>
<span id="cb18-31"><a href="#cb18-31" aria-hidden="true" tabindex="-1"></a>g_families</span></code></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure>
<p><img src="2024-07-22-thijssen_files/figure-html/viral-family-composition-1.svg" class="img-fluid" /></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="human-infecting-virus-reads" class="level1" data-number="4">
<h1 data-number="4"><span class="header-section-number">4</span> Human-infecting virus reads</h1>
<section id="overall-relative-abundance" class="level2" data-number="4.1">
<h2 data-number="4.1"><span class="header-section-number">4.1</span> Overall relative abundance</h2>
<p>I calculated the relative abundance of human-infecting viruses in two ways:</p>
<ul>
<li><p>First, as the total number of deduplicated human-virus reads in each sample, divided by the number of raw reads (“All reads”). This results in a viral fraction of <span class="math inline">\(1.64 \times 10^{-4}\)</span> across all samples</p></li>
<li><p>Second, as a fraction of preprocessed (cleaned, deduplicated, computationally ribodepleted) reads (“Preprocessed reads”). This results in a viral fraction of <span class="math inline">\(2.42 \times 10^{-4}\)</span> across all samples.</p></li>
</ul>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode" id="cb19"><pre class="sourceCode r cell-code"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get raw read counts</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>read_counts_raw <span class="ot">&lt;-</span> <span class="fu">filter</span>(basic_stats_raw) <span class="sc">%&gt;%</span></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, <span class="at">n_reads_raw =</span> n_read_pairs)</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>read_counts_preproc <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">&quot;ribo_initial&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, <span class="at">n_reads_preproc =</span> n_read_pairs)</span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Get HV read counts</span></span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a>read_counts_hv <span class="ot">&lt;-</span> mrg_hv <span class="sc">%&gt;%</span> <span class="fu">filter</span>(hv_status) <span class="sc">%&gt;%</span> </span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name=</span><span class="st">&quot;n_reads_hv&quot;</span>)</span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a>read_counts <span class="ot">&lt;-</span> read_counts_raw <span class="sc">%&gt;%</span></span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(read_counts_hv, <span class="at">by=</span><span class="fu">c</span>(<span class="st">&quot;sample&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">n_reads_hv =</span> <span class="fu">replace_na</span>(n_reads_hv, <span class="dv">0</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb19-14"><a href="#cb19-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(read_counts_preproc, <span class="at">by=</span><span class="fu">c</span>(<span class="st">&quot;sample&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb19-15"><a href="#cb19-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="fu">c</span>(<span class="st">&quot;sample&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb19-16"><a href="#cb19-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, n_reads_raw, n_reads_preproc, n_reads_hv) <span class="sc">%&gt;%</span></span>
<span id="cb19-17"><a href="#cb19-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">n_samples =</span> <span class="dv">1</span>,</span>
<span id="cb19-18"><a href="#cb19-18" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_total =</span> n_reads_hv<span class="sc">/</span>n_reads_raw,</span>
<span id="cb19-19"><a href="#cb19-19" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_preproc =</span> n_reads_hv<span class="sc">/</span>n_reads_preproc)</span>
<span id="cb19-20"><a href="#cb19-20" aria-hidden="true" tabindex="-1"></a>read_counts_long <span class="ot">&lt;-</span> read_counts <span class="sc">%&gt;%</span></span>
<span id="cb19-21"><a href="#cb19-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="fu">starts_with</span>(<span class="st">&quot;p_reads&quot;</span>), <span class="at">names_to=</span><span class="st">&quot;read_group&quot;</span>, <span class="at">values_to=</span><span class="st">&quot;p_reads&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-22"><a href="#cb19-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">read_group =</span> <span class="fu">ifelse</span>(read_group <span class="sc">==</span> <span class="st">&quot;p_reads_total&quot;</span>, <span class="st">&quot;All reads&quot;</span>, <span class="st">&quot;Preprocessed reads&quot;</span>))</span>
<span id="cb19-23"><a href="#cb19-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-24"><a href="#cb19-24" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine for display</span></span>
<span id="cb19-25"><a href="#cb19-25" aria-hidden="true" tabindex="-1"></a>read_counts_agg <span class="ot">&lt;-</span> read_counts <span class="sc">%&gt;%</span></span>
<span id="cb19-26"><a href="#cb19-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_total =</span> n_reads_hv<span class="sc">/</span>n_reads_raw,</span>
<span id="cb19-27"><a href="#cb19-27" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_preproc =</span> n_reads_hv<span class="sc">/</span>n_reads_preproc) <span class="sc">%&gt;%</span></span>
<span id="cb19-28"><a href="#cb19-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="fu">c</span>(<span class="st">&quot;sample&quot;</span>))</span>
<span id="cb19-29"><a href="#cb19-29" aria-hidden="true" tabindex="-1"></a>read_counts_agg_long <span class="ot">&lt;-</span> read_counts_agg <span class="sc">%&gt;%</span></span>
<span id="cb19-30"><a href="#cb19-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="fu">starts_with</span>(<span class="st">&quot;p_reads&quot;</span>), <span class="at">names_to=</span><span class="st">&quot;read_group&quot;</span>, <span class="at">values_to=</span><span class="st">&quot;p_reads&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-31"><a href="#cb19-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">read_group =</span> <span class="fu">ifelse</span>(read_group <span class="sc">==</span> <span class="st">&quot;p_reads_total&quot;</span>, <span class="st">&quot;All reads (HV)&quot;</span>, <span class="st">&quot;Preprocessed reads (HV)&quot;</span>)) </span>
<span id="cb19-32"><a href="#cb19-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-33"><a href="#cb19-33" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize</span></span>
<span id="cb19-34"><a href="#cb19-34" aria-hidden="true" tabindex="-1"></a>g_read_counts <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(read_counts_agg_long, <span class="fu">aes</span>(<span class="at">x=</span>library, <span class="at">y=</span>p_reads, <span class="at">color =</span> read_group, <span class="at">group =</span> library)) <span class="sc">+</span></span>
<span id="cb19-35"><a href="#cb19-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb19-36"><a href="#cb19-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">color =</span> <span class="st">&#39;black&#39;</span>) <span class="sc">+</span></span>
<span id="cb19-37"><a href="#cb19-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>(<span class="at">name =</span> <span class="st">&quot;Unique human-viral read fraction&quot;</span>, <span class="at">labels =</span> <span class="fu">label_log</span>(<span class="at">digits=</span><span class="dv">2</span>), <span class="at">limits =</span> <span class="fu">c</span>(<span class="fl">1e-6</span>, <span class="fl">1e-2</span>)) <span class="sc">+</span></span>
<span id="cb19-38"><a href="#cb19-38" aria-hidden="true" tabindex="-1"></a>  theme_kit <span class="sc">+</span> </span>
<span id="cb19-39"><a href="#cb19-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>()</span>
<span id="cb19-40"><a href="#cb19-40" aria-hidden="true" tabindex="-1"></a>g_read_counts</span></code></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure>
<p><img src="2024-07-22-thijssen_files/figure-html/count-hv-reads-1.svg" class="img-fluid" /></p>
</figure>
</div>
</div>
</div>
</section>
<section id="overall-taxonomy-and-composition" class="level2" data-number="4.2">
<h2 data-number="4.2"><span class="header-section-number">4.2</span> Overall taxonomy and composition</h2>
<p>Composition of HV reads was changed from when looking at all viral reads. The two dominant viruses we see are Anellovirdae and Microviridae (bacteriophage). The threshold for the label “other” are the set of families that make up less than 5% composition in all samples.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode" id="cb20"><pre class="sourceCode r cell-code"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>threshold_major_family <span class="ot">&lt;-</span> <span class="fl">0.05</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Count reads for each human-viral family</span></span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>hv_family_counts <span class="ot">&lt;-</span> hv_reads_family <span class="sc">%&gt;%</span> </span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">&quot;n_reads_hv&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_hv =</span> n_reads_hv<span class="sc">/</span><span class="fu">sum</span>(n_reads_hv))</span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify high-ranking families and group others</span></span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a>hv_family_major_tab <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span> </span>
<span id="cb20-12"><a href="#cb20-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_hv <span class="sc">==</span> <span class="fu">max</span>(p_reads_hv)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb20-13"><a href="#cb20-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_hv)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(p_reads_hv <span class="sc">&gt;</span> threshold_major_family)</span>
<span id="cb20-14"><a href="#cb20-14" aria-hidden="true" tabindex="-1"></a>hv_family_counts_major <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span></span>
<span id="cb20-15"><a href="#cb20-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">ifelse</span>(name <span class="sc">%in%</span> hv_family_major_tab<span class="sc">$</span>name, name, <span class="st">&quot;Other&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb20-16"><a href="#cb20-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, name_display) <span class="sc">%&gt;%</span></span>
<span id="cb20-17"><a href="#cb20-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_hv =</span> <span class="fu">sum</span>(n_reads_hv), <span class="at">p_reads_hv =</span> <span class="fu">sum</span>(p_reads_hv), </span>
<span id="cb20-18"><a href="#cb20-18" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups=</span><span class="st">&quot;drop&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb20-19"><a href="#cb20-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">factor</span>(name_display, </span>
<span id="cb20-20"><a href="#cb20-20" aria-hidden="true" tabindex="-1"></a>                               <span class="at">levels =</span> <span class="fu">c</span>(hv_family_major_tab<span class="sc">$</span>name, <span class="st">&quot;Other&quot;</span>)))</span>
<span id="cb20-21"><a href="#cb20-21" aria-hidden="true" tabindex="-1"></a>hv_family_counts_display <span class="ot">&lt;-</span> hv_family_counts_major <span class="sc">%&gt;%</span></span>
<span id="cb20-22"><a href="#cb20-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_hv, <span class="at">classification =</span> name_display) <span class="sc">%&gt;%</span></span>
<span id="cb20-23"><a href="#cb20-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by =</span> <span class="st">&#39;sample&#39;</span>)</span>
<span id="cb20-24"><a href="#cb20-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-25"><a href="#cb20-25" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb20-26"><a href="#cb20-26" aria-hidden="true" tabindex="-1"></a>g_hv_family <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb20-27"><a href="#cb20-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">data=</span>hv_family_counts_display, <span class="at">position =</span> <span class="st">&quot;stack&quot;</span>, <span class="at">width=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb20-28"><a href="#cb20-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">&quot;% HV Reads&quot;</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb20-29"><a href="#cb20-29" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb20-30"><a href="#cb20-30" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb20-31"><a href="#cb20-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">&#39;Accent&#39;</span>, <span class="at">name =</span> <span class="st">&quot;Viral family&quot;</span>) <span class="sc">+</span></span>
<span id="cb20-32"><a href="#cb20-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">&quot;Family composition of human-viral reads&quot;</span>) <span class="sc">+</span></span>
<span id="cb20-33"><a href="#cb20-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">4</span>)) <span class="sc">+</span></span>
<span id="cb20-34"><a href="#cb20-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">1.4</span>), <span class="at">hjust=</span><span class="dv">0</span>, <span class="at">face=</span><span class="st">&quot;plain&quot;</span>))</span>
<span id="cb20-35"><a href="#cb20-35" aria-hidden="true" tabindex="-1"></a>g_hv_family</span></code></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure>
<p><img src="2024-07-22-thijssen_files/figure-html/hv-family-1.svg" class="img-fluid" /></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode" id="cb21"><pre class="sourceCode r cell-code"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get most prominent families for text</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>hv_family_collate <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_tot =</span> <span class="fu">sum</span>(n_reads_hv),</span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>            <span class="at">p_reads_max =</span> <span class="fu">max</span>(p_reads_hv), <span class="at">.groups=</span><span class="st">&quot;drop&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n_reads_tot))</span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>hv_family_collate <span class="sc">%&gt;%</span></span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a> <span class="fu">select</span>(name,taxid, n_reads_tot) <span class="sc">%&gt;%</span></span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a> <span class="fu">rename</span>(</span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a>  <span class="st">&#39;family&#39;</span> <span class="ot">=</span> <span class="st">&#39;name&#39;</span>,</span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a>  <span class="st">&#39;# of total reads&#39;</span> <span class="ot">=</span> <span class="st">&#39;n_reads_tot&#39;</span>,</span>
<span id="cb21-12"><a href="#cb21-12" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source type="application/json">
{"columns":[{"label":["family"],"name":[1],"type":["chr"],"align":["left"]},{"label":["taxid"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["# of total reads"],"name":[3],"type":["int"],"align":["right"]}],"data":[{"1":"Anelloviridae","2":"687329","3":"12804"},{"1":"Microviridae","2":"10841","3":"5194"},{"1":"Sedoreoviridae","2":"2946186","3":"278"},{"1":"Parvoviridae","2":"10780","3":"266"},{"1":"Adenoviridae","2":"10508","3":"104"},{"1":"Polyomaviridae","2":"151341","3":"95"},{"1":"Papillomaviridae","2":"151340","3":"16"},{"1":"Intestiviridae","2":"2942964","3":"4"},{"1":"Picornaviridae","2":"12058","3":"4"},{"1":"Caliciviridae","2":"11974","3":"3"},{"1":"Coronaviridae","2":"11118","3":"3"},{"1":"Poxviridae","2":"10240","3":"3"},{"1":"Inoviridae","2":"10860","3":"2"},{"1":"Hudisavirus sp.","2":"2021738","3":"1"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
</section>
<section id="analyzing-specific-families" class="level2" data-number="4.3">
<h2 data-number="4.3"><span class="header-section-number">4.3</span> Analyzing specific families</h2>
<p>We now investigate the composition of specific families that had more than 5 viral reads. In investigating individual viral families, to avoid distortions from a few rare reads, I restricted myself to samples where that family made up at least 1% of human-viral reads:</p>
<section id="anelloviridae-number-of-reads-12804" class="level3" data-number="4.3.1">
<h3 data-number="4.3.1"><span class="header-section-number">4.3.1</span> Anelloviridae (Number of reads: 12,804)</h3>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode" id="cb22"><pre class="sourceCode r cell-code"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Anelloviridae</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_histogram</span>(<span class="at">taxid_chosen=</span><span class="dv">687329</span>)</span></code></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure>
<p><img src="2024-07-22-thijssen_files/figure-html/plot-anelloviridae-histogram-1.svg" class="img-fluid" /></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode" id="cb23"><pre class="sourceCode r cell-code"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Anelloviridae</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_composition</span>(<span class="at">taxid_chosen=</span><span class="dv">687329</span>, <span class="at">threshold_major_species =</span> <span class="fl">0.01</span>)</span></code></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure>
<p><img src="2024-07-22-thijssen_files/figure-html/plot-anelloviridae-composition-1.svg" class="img-fluid" /></p>
</figure>
</div>
</div>
</div>
</section>
<section id="microviridae-number-of-reads-5194" class="level3" data-number="4.3.2">
<h3 data-number="4.3.2"><span class="header-section-number">4.3.2</span> Microviridae (Number of reads: 5,194)</h3>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode" id="cb24"><pre class="sourceCode r cell-code"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Microviridae</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_histogram</span>(<span class="at">taxid_chosen=</span><span class="dv">10841</span>)</span></code></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure>
<p><img src="2024-07-22-thijssen_files/figure-html/plot-microviridae-histogram-1.svg" class="img-fluid" /></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode" id="cb25"><pre class="sourceCode r cell-code"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Microviridae</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_composition</span>(<span class="at">taxid_chosen=</span><span class="dv">10841</span>, <span class="at">threshold_major_species =</span> <span class="fl">0.01</span>)</span></code></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure>
<p><img src="2024-07-22-thijssen_files/figure-html/plot-microviridae-composition-1.svg" class="img-fluid" /></p>
</figure>
</div>
</div>
</div>
</section>
<section id="sedreoviridae-number-of-reads-278" class="level3" data-number="4.3.3">
<h3 data-number="4.3.3"><span class="header-section-number">4.3.3</span> Sedreoviridae (Number of reads: 278)</h3>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode" id="cb26"><pre class="sourceCode r cell-code"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Sedreoviridae</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_histogram</span>(<span class="at">taxid_chosen=</span><span class="dv">2946186</span>)</span></code></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure>
<p><img src="2024-07-22-thijssen_files/figure-html/plot-sedreoviridae-histogram-1.svg" class="img-fluid" /></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode" id="cb27"><pre class="sourceCode r cell-code"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Sedreoviridae</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_composition</span>(<span class="at">taxid_chosen=</span><span class="dv">2946186</span>, <span class="at">threshold_major_species =</span> <span class="fl">0.01</span>)</span></code></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure>
<p><img src="2024-07-22-thijssen_files/figure-html/plot-sedreoviridae-composition-1.svg" class="img-fluid" /></p>
</figure>
</div>
</div>
</div>
</section>
<section id="parvoviridae-number-of-reads-266" class="level3" data-number="4.3.4">
<h3 data-number="4.3.4"><span class="header-section-number">4.3.4</span> Parvoviridae (Number of reads: 266)</h3>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode" id="cb28"><pre class="sourceCode r cell-code"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Parvoviridae</span></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_histogram</span>(<span class="at">taxid_chosen=</span><span class="dv">10780</span>)</span></code></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure>
<p><img src="2024-07-22-thijssen_files/figure-html/plot-parvoviridae-histogram-1.svg" class="img-fluid" /></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode" id="cb29"><pre class="sourceCode r cell-code"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Parvoviridae</span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_composition</span>(<span class="at">taxid_chosen=</span><span class="dv">10780</span>, <span class="at">threshold_major_species =</span> <span class="fl">0.01</span>)</span></code></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure>
<p><img src="2024-07-22-thijssen_files/figure-html/plot-parvoviridae-composition-1.svg" class="img-fluid" /></p>
</figure>
</div>
</div>
</div>
</section>
<section id="adenoviridae-number-of-reads-104" class="level3" data-number="4.3.5">
<h3 data-number="4.3.5"><span class="header-section-number">4.3.5</span> Adenoviridae (Number of reads: 104)</h3>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode" id="cb30"><pre class="sourceCode r cell-code"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Adenoviridae</span></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_histogram</span>(<span class="at">taxid_chosen=</span><span class="dv">10508</span>)</span></code></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure>
<p><img src="2024-07-22-thijssen_files/figure-html/plot-adenoviridae-histogram-1.svg" class="img-fluid" /></p>
</figure>
</div>
</div>
</div>
<p>None of these reads made up more than 1% of the total reads of any sample.</p>
</section>
<section id="polyomaviridae-number-of-reads-95" class="level3" data-number="4.3.6">
<h3 data-number="4.3.6"><span class="header-section-number">4.3.6</span> Polyomaviridae (Number of reads: 95)</h3>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode" id="cb31"><pre class="sourceCode r cell-code"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Polyomaviridae</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_histogram</span>(<span class="at">taxid_chosen=</span><span class="dv">151341</span>)</span></code></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure>
<p><img src="2024-07-22-thijssen_files/figure-html/plot-polyomaviridae-histogram-1.svg" class="img-fluid" /></p>
</figure>
</div>
</div>
</div>
<p>None of these reads made up more than 1% of the total reads of any sample.</p>
</section>
<section id="papillomaviridae-number-of-reads-16" class="level3" data-number="4.3.7">
<h3 data-number="4.3.7"><span class="header-section-number">4.3.7</span> Papillomaviridae (Number of reads: 16)</h3>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode" id="cb32"><pre class="sourceCode r cell-code"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Polyomaviridae</span></span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_histogram</span>(<span class="at">taxid_chosen=</span><span class="dv">151340</span>)</span></code></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure>
<p><img src="2024-07-22-thijssen_files/figure-html/plot-papillomaviridae-histogram-1.svg" class="img-fluid" /></p>
</figure>
</div>
</div>
</div>
<p>None of these reads made up more than 1% of the total reads of any sample.</p>
</section>
</section>
<section id="relative-abundance-of-pathogenic-viruses-of-interest" class="level2" data-number="4.4">
<h2 data-number="4.4"><span class="header-section-number">4.4</span> Relative abundance of pathogenic viruses of interest</h2>
<p>Each dot represents a sample, colored by viral family. The x-axis shows the relative abundance of human-infecting viruses, and the y-axis shows the species.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode" id="cb33"><pre class="sourceCode r cell-code"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>species_family <span class="ot">&lt;-</span> result <span class="sc">%&gt;%</span> <span class="fu">select</span>(species, family) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="st">&#39;name&#39;</span> <span class="ot">=</span> <span class="st">&#39;species&#39;</span>)</span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>play <span class="ot">&lt;-</span> hv_species_counts <span class="sc">%&gt;%</span> </span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by =</span> <span class="st">&#39;sample&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(species_family, <span class="at">by =</span> <span class="st">&#39;name&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb33-7"><a href="#cb33-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">ifelse</span>(name <span class="sc">==</span> <span class="st">&quot;Severe acute respiratory syndrome-related coronavirus&quot;</span>, <span class="st">&quot;SARS-related coronavirus&quot;</span>, name)) <span class="sc">%&gt;%</span></span>
<span id="cb33-8"><a href="#cb33-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(family <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;Sedoreoviridae&quot;</span>, <span class="st">&quot;Parvoviridae&quot;</span>, <span class="st">&quot;Adenoviridae&quot;</span>, <span class="st">&quot;Papillomaviridae&quot;</span>, <span class="st">&quot;Polyomaviridae&quot;</span>, <span class="st">&quot;Coronaviridae&quot;</span>, <span class="st">&quot;Caliciviridae&quot;</span>, <span class="st">&quot;Picornaviridae&quot;</span>)) </span>
<span id="cb33-9"><a href="#cb33-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-10"><a href="#cb33-10" aria-hidden="true" tabindex="-1"></a>adjusted_play <span class="ot">&lt;-</span> play <span class="sc">%&gt;%</span> </span>
<span id="cb33-11"><a href="#cb33-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span></span>
<span id="cb33-12"><a href="#cb33-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">virus_prevalence_num =</span> <span class="fu">n_distinct</span>(sample)<span class="sc">/</span><span class="fu">n_distinct</span>(libraries),</span>
<span id="cb33-13"><a href="#cb33-13" aria-hidden="true" tabindex="-1"></a>         <span class="at">total_reads_hv =</span> <span class="fu">sum</span>(n_reads_hv)) <span class="sc">%&gt;%</span></span>
<span id="cb33-14"><a href="#cb33-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb33-15"><a href="#cb33-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">fct_reorder</span>(name, virus_prevalence_num, <span class="at">.desc=</span><span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb33-16"><a href="#cb33-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, ra_reads_hv, family, virus_prevalence_num, total_reads_hv, n_reads_hv)</span>
<span id="cb33-17"><a href="#cb33-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-18"><a href="#cb33-18" aria-hidden="true" tabindex="-1"></a>pal <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">&#39;Dark2&#39;</span>),<span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">&#39;Accent&#39;</span>))</span>
<span id="cb33-19"><a href="#cb33-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-20"><a href="#cb33-20" aria-hidden="true" tabindex="-1"></a><span class="co">#, labels = label_log(digits=2)</span></span>
<span id="cb33-21"><a href="#cb33-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-22"><a href="#cb33-22" aria-hidden="true" tabindex="-1"></a>ra_dot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(adjusted_play, <span class="fu">aes</span>(<span class="at">x =</span> ra_reads_hv, <span class="at">y=</span>name)) <span class="sc">+</span></span>
<span id="cb33-23"><a href="#cb33-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_quasirandom</span>(<span class="at">orientation =</span> <span class="st">&#39;y&#39;</span>, <span class="fu">aes</span>(<span class="at">color =</span> family)) <span class="sc">+</span></span>
<span id="cb33-24"><a href="#cb33-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> pal) <span class="sc">+</span> </span>
<span id="cb33-25"><a href="#cb33-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_log10</span>(</span>
<span id="cb33-26"><a href="#cb33-26" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;Relative abundance human virus reads&quot;</span>,</span>
<span id="cb33-27"><a href="#cb33-27" aria-hidden="true" tabindex="-1"></a>    <span class="at">labels=</span><span class="fu">label_log</span>(<span class="at">digits=</span><span class="dv">3</span>)</span>
<span id="cb33-28"><a href="#cb33-28" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb33-29"><a href="#cb33-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span>  <span class="st">&quot;&quot;</span>,</span>
<span id="cb33-30"><a href="#cb33-30" aria-hidden="true" tabindex="-1"></a>       <span class="at">color =</span> <span class="st">&#39;Viral family&#39;</span>) <span class="sc">+</span> </span>
<span id="cb33-31"><a href="#cb33-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color =</span> <span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">2</span>)) <span class="sc">+</span></span>
<span id="cb33-32"><a href="#cb33-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_light</span>() <span class="sc">+</span> </span>
<span id="cb33-33"><a href="#cb33-33" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(</span>
<span id="cb33-34"><a href="#cb33-34" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb33-35"><a href="#cb33-35" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>),</span>
<span id="cb33-36"><a href="#cb33-36" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.ticks.y =</span> <span class="fu">element_blank</span>(),<span class="at">axis.line =</span> <span class="fu">element_line</span>(<span class="at">colour =</span> <span class="st">&quot;black&quot;</span>),</span>
<span id="cb33-37"><a href="#cb33-37" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">14</span>),    </span>
<span id="cb33-38"><a href="#cb33-38" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb33-39"><a href="#cb33-39" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>, <span class="at">face=</span><span class="st">&quot;bold&quot;</span>),</span>
<span id="cb33-40"><a href="#cb33-40" aria-hidden="true" tabindex="-1"></a>    <span class="co">#legend.position = &#39;bottom&#39;,</span></span>
<span id="cb33-41"><a href="#cb33-41" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>),  <span class="co"># Move legend to top right</span></span>
<span id="cb33-42"><a href="#cb33-42" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.justification =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>),  <span class="co"># Align legend to top right</span></span>
<span id="cb33-43"><a href="#cb33-43" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb33-44"><a href="#cb33-44" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.border =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb33-45"><a href="#cb33-45" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.background =</span> <span class="fu">element_blank</span>())</span>
<span id="cb33-46"><a href="#cb33-46" aria-hidden="true" tabindex="-1"></a>ra_dot</span></code></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure>
<p><img src="2024-07-22-thijssen_files/figure-html/plot-pathogenic-viruses-1.svg" class="img-fluid" /></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode" id="cb34"><pre class="sourceCode r cell-code"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="co">#  geom_text(aes(label = total_reads_hv, y = name), </span></span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a><span class="co">#            x = 1e-9, hjust = 0, vjust = 0.5, size = 3, </span></span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a><span class="co">#            check_overlap = TRUE)</span></span></code></pre></div>
</details>
</div>
<p>I can then take all of the viruses that we found and look up what they’re responsible for and whether they’re dangerous.</p>
<table class="caption-top">
<colgroup>
<col style="width: 30%" />
<col style="width: 25%" />
<col style="width: 44%" />
</colgroup>
<thead>
<tr class="header">
<th>Virus Name</th>
<th>Common Name</th>
<th>Pathogenic Potential</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Rotavirus A</td>
<td>Rotavirus</td>
<td>High, causes severe diarrhea in children</td>
</tr>
<tr class="even">
<td>Enterovirus C</td>
<td>Includes poliovirus and some coxsackieviruses</td>
<td>High, can cause various diseases including polio</td>
</tr>
<tr class="odd">
<td>Severe acute respiratory syndrome-related coronavirus</td>
<td>SARS coronavirus</td>
<td>High, causes severe respiratory illness</td>
</tr>
<tr class="even">
<td>Sapporo virus</td>
<td>Sapporo virus or Sapovirus</td>
<td>Moderate, causes gastroenteritis</td>
</tr>
<tr class="odd">
<td>Human mastadenovirus F</td>
<td>Adenovirus F</td>
<td>Moderate, can cause gastroenteritis</td>
</tr>
<tr class="even">
<td>Alphapolyomavirus quintihominis</td>
<td>Not widely known</td>
<td>Low, but some polyomaviruses can cause disease in immunocompromised individuals</td>
</tr>
<tr class="odd">
<td>Alphapapillomavirus 4</td>
<td>Human papillomavirus (HPV) type 4</td>
<td>Low to moderate, can cause warts</td>
</tr>
</tbody>
</table>
<p>None of these viruses are too suprising to see other than the Severe acute respiratory syndrome-related coronavirus, which when BLASTED, comes up as SARS-CoV-2. We believe that this is due to contamination with the COVID-19 pandemic since these samples were taken pre-pandemic.</p>
</section>
<section id="relative-abundance-assuming-perfect-human-read-removal" class="level2" data-number="4.5">
<h2 data-number="4.5"><span class="header-section-number">4.5</span> Relative abundance assuming perfect human read removal</h2>
<p>Assuming we’re able to perfectly remove all human reads, the average relative abundance of known human infecting virus is <span class="math inline">\(1.87 \times 10^{-4}\)</span>.</p>
</section>
</section>
<section id="conclusion" class="level1" data-number="5">
<h1 data-number="5"><span class="header-section-number">5</span> Conclusion</h1>
<p>There were some interesting takeways from this analysis:</p>
<ol type="1">
<li>Contamination with SARS-Cov-2 seems to be a common occurence and should be accounted for when interpretting these results.</li>
<li>Gasterointestinal diseases are picked up in plasma.</li>
<li>Sexually transmitted infections are picked up in plasma.</li>
<li>We pick up a decent amount of baceriophage (i.e. Microviridae).</li>
<li>A lot of the Microviridae and Anelloviridae are largely uncharacterized (denoted by the family name followed by the abbreviation “sp.”).</li>
</ol>
</section>
<section id="appendix" class="level1" data-number="6">
<h1 data-number="6"><span class="header-section-number">6</span> Appendix</h1>
<section id="human-infecting-virus-families-genera-and-species" class="level2" data-number="6.1">
<h2 data-number="6.1"><span class="header-section-number">6.1</span> Human-infecting virus families, genera, and species</h2>
<p>To get a good overview of families, genera, and species, we can look at a Sankey plot where the magnitude of relative abundance, averaged over all samples, is shown in parentheses.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode" id="cb35"><pre class="sourceCode r cell-code"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to create links</span></span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>create_links <span class="ot">&lt;-</span> <span class="cf">function</span>(data) {</span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>  family_to_genus <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(genus)) <span class="sc">%&gt;%</span></span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(family, genus) <span class="sc">%&gt;%</span></span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">value =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">&quot;drop&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb35-7"><a href="#cb35-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">source =</span> family, <span class="at">target =</span> genus)</span>
<span id="cb35-8"><a href="#cb35-8" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb35-9"><a href="#cb35-9" aria-hidden="true" tabindex="-1"></a>  genus_to_species <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb35-10"><a href="#cb35-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(genus, species) <span class="sc">%&gt;%</span></span>
<span id="cb35-11"><a href="#cb35-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">value =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">&quot;drop&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb35-12"><a href="#cb35-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">source =</span> genus, <span class="at">target =</span> species)</span>
<span id="cb35-13"><a href="#cb35-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-14"><a href="#cb35-14" aria-hidden="true" tabindex="-1"></a>  family_to_species <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb35-15"><a href="#cb35-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="fu">is.na</span>(genus)) <span class="sc">%&gt;%</span></span>
<span id="cb35-16"><a href="#cb35-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(family, species) <span class="sc">%&gt;%</span></span>
<span id="cb35-17"><a href="#cb35-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">value =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">&quot;drop&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb35-18"><a href="#cb35-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">source =</span> family, <span class="at">target =</span> species)</span>
<span id="cb35-19"><a href="#cb35-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-20"><a href="#cb35-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_rows</span>(family_to_genus, genus_to_species, family_to_species) <span class="sc">%&gt;%</span></span>
<span id="cb35-21"><a href="#cb35-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(source))</span>
<span id="cb35-22"><a href="#cb35-22" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb35-23"><a href="#cb35-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-24"><a href="#cb35-24" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to create nodes</span></span>
<span id="cb35-25"><a href="#cb35-25" aria-hidden="true" tabindex="-1"></a>create_nodes <span class="ot">&lt;-</span> <span class="cf">function</span>(links) {</span>
<span id="cb35-26"><a href="#cb35-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>(</span>
<span id="cb35-27"><a href="#cb35-27" aria-hidden="true" tabindex="-1"></a>    <span class="at">name =</span> <span class="fu">c</span>(links<span class="sc">$</span>source, links<span class="sc">$</span>target) <span class="sc">%&gt;%</span> <span class="fu">unique</span>()</span>
<span id="cb35-28"><a href="#cb35-28" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb35-29"><a href="#cb35-29" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb35-30"><a href="#cb35-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-31"><a href="#cb35-31" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to prepare data for Sankey diagram</span></span>
<span id="cb35-32"><a href="#cb35-32" aria-hidden="true" tabindex="-1"></a>prepare_sankey_data <span class="ot">&lt;-</span> <span class="cf">function</span>(links, nodes) {</span>
<span id="cb35-33"><a href="#cb35-33" aria-hidden="true" tabindex="-1"></a>  links<span class="sc">$</span>IDsource <span class="ot">&lt;-</span> <span class="fu">match</span>(links<span class="sc">$</span>source, nodes<span class="sc">$</span>name) <span class="sc">-</span> <span class="dv">1</span></span>
<span id="cb35-34"><a href="#cb35-34" aria-hidden="true" tabindex="-1"></a>  links<span class="sc">$</span>IDtarget <span class="ot">&lt;-</span> <span class="fu">match</span>(links<span class="sc">$</span>target, nodes<span class="sc">$</span>name) <span class="sc">-</span> <span class="dv">1</span></span>
<span id="cb35-35"><a href="#cb35-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list</span>(<span class="at">links =</span> links, <span class="at">nodes =</span> nodes)</span>
<span id="cb35-36"><a href="#cb35-36" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb35-37"><a href="#cb35-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-38"><a href="#cb35-38" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to create Sankey plot</span></span>
<span id="cb35-39"><a href="#cb35-39" aria-hidden="true" tabindex="-1"></a>create_sankey_plot <span class="ot">&lt;-</span> <span class="cf">function</span>(sankey_data) {</span>
<span id="cb35-40"><a href="#cb35-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sankeyNetwork</span>(</span>
<span id="cb35-41"><a href="#cb35-41" aria-hidden="true" tabindex="-1"></a>    <span class="at">Links =</span> sankey_data<span class="sc">$</span>links, </span>
<span id="cb35-42"><a href="#cb35-42" aria-hidden="true" tabindex="-1"></a>    <span class="at">Nodes =</span> sankey_data<span class="sc">$</span>nodes,</span>
<span id="cb35-43"><a href="#cb35-43" aria-hidden="true" tabindex="-1"></a>    <span class="at">Source =</span> <span class="st">&quot;IDsource&quot;</span>, </span>
<span id="cb35-44"><a href="#cb35-44" aria-hidden="true" tabindex="-1"></a>    <span class="at">Target =</span> <span class="st">&quot;IDtarget&quot;</span>,</span>
<span id="cb35-45"><a href="#cb35-45" aria-hidden="true" tabindex="-1"></a>    <span class="at">Value =</span> <span class="st">&quot;value&quot;</span>, </span>
<span id="cb35-46"><a href="#cb35-46" aria-hidden="true" tabindex="-1"></a>    <span class="at">NodeID =</span> <span class="st">&quot;name&quot;</span>,</span>
<span id="cb35-47"><a href="#cb35-47" aria-hidden="true" tabindex="-1"></a>    <span class="at">sinksRight =</span> <span class="cn">TRUE</span>,</span>
<span id="cb35-48"><a href="#cb35-48" aria-hidden="true" tabindex="-1"></a>    <span class="at">nodeWidth =</span> <span class="dv">25</span>,</span>
<span id="cb35-49"><a href="#cb35-49" aria-hidden="true" tabindex="-1"></a>    <span class="at">fontSize =</span> <span class="dv">14</span>,</span>
<span id="cb35-50"><a href="#cb35-50" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb35-51"><a href="#cb35-51" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb35-52"><a href="#cb35-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-53"><a href="#cb35-53" aria-hidden="true" tabindex="-1"></a>save_sankey_as_png <span class="ot">&lt;-</span> <span class="cf">function</span>(sankey_plot, <span class="at">width =</span> <span class="dv">1000</span>, <span class="at">height =</span> <span class="dv">800</span>) {</span>
<span id="cb35-54"><a href="#cb35-54" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Save the plot as an HTML file</span></span>
<span id="cb35-55"><a href="#cb35-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">saveWidget</span>(sankey_plot, <span class="fu">sprintf</span>(<span class="st">&#39;%s/sankey.html&#39;</span>,data_dir))</span>
<span id="cb35-56"><a href="#cb35-56" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb35-57"><a href="#cb35-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-58"><a href="#cb35-58" aria-hidden="true" tabindex="-1"></a>format_scientific <span class="ot">&lt;-</span> <span class="cf">function</span>(x, <span class="at">digits=</span><span class="dv">2</span>) {</span>
<span id="cb35-59"><a href="#cb35-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sapply</span>(x, <span class="cf">function</span>(val) {</span>
<span id="cb35-60"><a href="#cb35-60" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">is.na</span>(val) <span class="sc">||</span> <span class="fu">abs</span>(val) <span class="sc">&lt;</span> <span class="fl">1e-15</span>) {</span>
<span id="cb35-61"><a href="#cb35-61" aria-hidden="true" tabindex="-1"></a>      <span class="fu">return</span>(<span class="st">&quot;0&quot;</span>)</span>
<span id="cb35-62"><a href="#cb35-62" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb35-63"><a href="#cb35-63" aria-hidden="true" tabindex="-1"></a>      exponent <span class="ot">&lt;-</span> <span class="fu">floor</span>(<span class="fu">log10</span>(<span class="fu">abs</span>(val)))</span>
<span id="cb35-64"><a href="#cb35-64" aria-hidden="true" tabindex="-1"></a>      coef <span class="ot">&lt;-</span> val <span class="sc">/</span> <span class="dv">10</span><span class="sc">^</span>exponent</span>
<span id="cb35-65"><a href="#cb35-65" aria-hidden="true" tabindex="-1"></a>      <span class="co">#return(sprintf(&quot;%.1f × 10^%d&quot;, round(coef, digits), exponent))</span></span>
<span id="cb35-66"><a href="#cb35-66" aria-hidden="true" tabindex="-1"></a>      <span class="co"># May or may not be smart, just keeping magnitude</span></span>
<span id="cb35-67"><a href="#cb35-67" aria-hidden="true" tabindex="-1"></a>      <span class="fu">return</span>(<span class="fu">sprintf</span>(<span class="st">&quot;10^%d&quot;</span>, exponent))</span>
<span id="cb35-68"><a href="#cb35-68" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb35-69"><a href="#cb35-69" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb35-70"><a href="#cb35-70" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb35-71"><a href="#cb35-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-72"><a href="#cb35-72" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> result <span class="sc">%&gt;%</span> </span>
<span id="cb35-73"><a href="#cb35-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">c</span>(genus_n_reads_tot, genus_ra_reads_tot), <span class="sc">~</span><span class="fu">replace_na</span>(., <span class="dv">0</span>)),</span>
<span id="cb35-74"><a href="#cb35-74" aria-hidden="true" tabindex="-1"></a>         <span class="at">genus =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(genus), <span class="st">&quot;Unknown Genus&quot;</span>, genus)) <span class="sc">%&gt;%</span></span>
<span id="cb35-75"><a href="#cb35-75" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb35-76"><a href="#cb35-76" aria-hidden="true" tabindex="-1"></a>  <span class="at">species =</span> <span class="fu">paste0</span>(species, <span class="fu">sprintf</span>(<span class="st">&#39; (%s)&#39;</span>, <span class="fu">format_scientific</span>(species_ra_reads_tot))),</span>
<span id="cb35-77"><a href="#cb35-77" aria-hidden="true" tabindex="-1"></a>  <span class="at">genus =</span> <span class="fu">paste0</span>(genus, <span class="fu">sprintf</span>(<span class="st">&#39; (%s)&#39;</span>, <span class="fu">format_scientific</span>(genus_ra_reads_tot))),</span>
<span id="cb35-78"><a href="#cb35-78" aria-hidden="true" tabindex="-1"></a>  <span class="at">family =</span> <span class="fu">paste0</span>(family, <span class="fu">sprintf</span>(<span class="st">&#39; (%s)&#39;</span>, <span class="fu">format_scientific</span>(family_ra_reads_tot)))</span>
<span id="cb35-79"><a href="#cb35-79" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb35-80"><a href="#cb35-80" aria-hidden="true" tabindex="-1"></a>links <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">create_links</span>(data))</span>
<span id="cb35-81"><a href="#cb35-81" aria-hidden="true" tabindex="-1"></a>nodes <span class="ot">&lt;-</span> <span class="fu">create_nodes</span>(links)</span>
<span id="cb35-82"><a href="#cb35-82" aria-hidden="true" tabindex="-1"></a>sankey_data <span class="ot">&lt;-</span> <span class="fu">prepare_sankey_data</span>(links, nodes)</span>
<span id="cb35-83"><a href="#cb35-83" aria-hidden="true" tabindex="-1"></a>sankey <span class="ot">&lt;-</span> <span class="fu">create_sankey_plot</span>(sankey_data)</span>
<span id="cb35-84"><a href="#cb35-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-85"><a href="#cb35-85" aria-hidden="true" tabindex="-1"></a>sankey</span></code></pre></div>
</details>
<div class="cell-output-display">
<div id="htmlwidget-6c96f1f596e7e621474b" style="width:100%;height:1462px;" class="sankeyNetwork html-widget"></div>
<script type="application/json" data-for="htmlwidget-6c96f1f596e7e621474b">{"x":{"links":{"source":[0,1,1,1,1,2,3,4,5,6,7,8,9,9,10,10,10,11,12,13,14,15,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,17,18,19,19,19,19,19,19,19,19,19,20,21,22,23,24,25,25,25,25,25,25,25,25,25,25,25,25,25,26,26,27,28,29,30,31,31,31,31,31,31,31,31,31],"target":[27,16,19,25,31,30,18,26,21,31,31,14,17,23,20,22,24,15,28,29,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100],"value":[1,23,9,13,4,1,1,2,1,4,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1]},"nodes":{"name":["Adenoviridae (10^-6)","Anelloviridae (10^-4)","Caliciviridae (10^-7)","Coronaviridae (10^-7)","Inoviridae (10^-7)","Intestiviridae (10^-7)","Microviridae (10^-5)","NA (0)","Papillomaviridae (10^-6)","Parvoviridae (10^-6)","Picornaviridae (10^-7)","Polyomaviridae (10^-6)","Poxviridae (10^-7)","Sedoreoviridae (10^-6)","Alphapapillomavirus (10^-6)","Alphapolyomavirus (10^-6)","Alphatorquevirus (10^-6)","Ambidensovirus (10^-7)","Betacoronavirus (10^-7)","Betatorquevirus (10^-7)","Cardiovirus (10^-7)","Carjivirus (10^-7)","Cosavirus (10^-7)","Dependoparvovirus (10^-5)","Enterovirus (10^-7)","Gammatorquevirus (10^-6)","Inovirus (10^-7)","Mastadenovirus (10^-6)","Orthopoxvirus (10^-7)","Rotavirus (10^-6)","Sapovirus (10^-7)","Unknown Genus (0)","Alphapapillomavirus 4 (10^-6)","Alphapolyomavirus quintihominis (10^-6)","Alphatorquevirus homin1 (10^-6)","Alphatorquevirus homin10 (10^-6)","Alphatorquevirus homin13 (10^-6)","Alphatorquevirus homin15 (10^-7)","Alphatorquevirus homin17 (10^-7)","Alphatorquevirus homin18 (10^-6)","Alphatorquevirus homin19 (10^-6)","Alphatorquevirus homin20 (10^-6)","Alphatorquevirus homin21 (10^-7)","Alphatorquevirus homin24 (10^-7)","Alphatorquevirus homin29 (10^-7)","Alphatorquevirus homin3 (10^-6)","Alphatorquevirus homin5 (10^-7)","Alphatorquevirus homin6 (10^-6)","Alphatorquevirus homin7 (10^-7)","Alphatorquevirus homin9 (10^-7)","Torque teno virus 11 (10^-7)","Torque teno virus 12 (10^-6)","Torque teno virus 16 (10^-7)","Torque teno virus 22 (10^-6)","Torque teno virus 27 (10^-6)","Torque teno virus 28 (10^-7)","Torque teno virus 8 (10^-8)","Ambidensovirus sp. (10^-7)","Severe acute respiratory syndrome-related coronavirus (10^-7)","Betatorquevirus homini1 (10^-7)","Betatorquevirus homini10 (10^-7)","Betatorquevirus homini12 (10^-7)","Betatorquevirus homini26 (10^-7)","Betatorquevirus homini5 (10^-7)","Betatorquevirus homini6 (10^-7)","Betatorquevirus homini7 (10^-7)","Betatorquevirus homini8 (10^-7)","TTV-like mini virus (10^-7)","Cardiovirus D (10^-7)","CrAssphage ZA (10^-7)","Cosavirus A (10^-7)","Dependoparvovirus primate1 (10^-5)","Enterovirus C (10^-7)","Gammatorquevirus homidi1 (10^-7)","Gammatorquevirus homidi10 (10^-7)","Gammatorquevirus homidi11 (10^-6)","Gammatorquevirus homidi12 (10^-7)","Gammatorquevirus homidi13 (10^-7)","Gammatorquevirus homidi14 (10^-7)","Gammatorquevirus homidi2 (10^-7)","Gammatorquevirus homidi3 (10^-6)","Gammatorquevirus homidi4 (10^-7)","Gammatorquevirus homidi6 (10^-7)","Gammatorquevirus homidi7 (10^-7)","Gammatorquevirus homidi8 (10^-6)","Gammatorquevirus homidi9 (10^-7)","Inovirus D_HF3_19 (10^-7)","Inovirus D_HF5_49 (10^-7)","Human mastadenovirus F (10^-6)","Vaccinia virus (10^-7)","Rotavirus A (10^-6)","Sapporo virus (10^-7)","Anelloviridae sp. (10^-5)","Gokushovirus WZ-2015a (10^-7)","Hudisavirus sp. (10^-7)","Human gut gokushovirus (10^-6)","Micro Torque teno virus (10^-7)","Microviridae sp. (10^-5)","Microvirus sp. (10^-6)","Small anellovirus (10^-7)","Torque teno virus (10^-6)"],"group":["Adenoviridae (10^-6)","Anelloviridae (10^-4)","Caliciviridae (10^-7)","Coronaviridae (10^-7)","Inoviridae (10^-7)","Intestiviridae (10^-7)","Microviridae (10^-5)","NA (0)","Papillomaviridae (10^-6)","Parvoviridae (10^-6)","Picornaviridae (10^-7)","Polyomaviridae (10^-6)","Poxviridae (10^-7)","Sedoreoviridae (10^-6)","Alphapapillomavirus (10^-6)","Alphapolyomavirus (10^-6)","Alphatorquevirus (10^-6)","Ambidensovirus (10^-7)","Betacoronavirus (10^-7)","Betatorquevirus (10^-7)","Cardiovirus (10^-7)","Carjivirus (10^-7)","Cosavirus (10^-7)","Dependoparvovirus (10^-5)","Enterovirus (10^-7)","Gammatorquevirus (10^-6)","Inovirus (10^-7)","Mastadenovirus (10^-6)","Orthopoxvirus (10^-7)","Rotavirus (10^-6)","Sapovirus (10^-7)","Unknown Genus (0)","Alphapapillomavirus 4 (10^-6)","Alphapolyomavirus quintihominis (10^-6)","Alphatorquevirus homin1 (10^-6)","Alphatorquevirus homin10 (10^-6)","Alphatorquevirus homin13 (10^-6)","Alphatorquevirus homin15 (10^-7)","Alphatorquevirus homin17 (10^-7)","Alphatorquevirus homin18 (10^-6)","Alphatorquevirus homin19 (10^-6)","Alphatorquevirus homin20 (10^-6)","Alphatorquevirus homin21 (10^-7)","Alphatorquevirus homin24 (10^-7)","Alphatorquevirus homin29 (10^-7)","Alphatorquevirus homin3 (10^-6)","Alphatorquevirus homin5 (10^-7)","Alphatorquevirus homin6 (10^-6)","Alphatorquevirus homin7 (10^-7)","Alphatorquevirus homin9 (10^-7)","Torque teno virus 11 (10^-7)","Torque teno virus 12 (10^-6)","Torque teno virus 16 (10^-7)","Torque teno virus 22 (10^-6)","Torque teno virus 27 (10^-6)","Torque teno virus 28 (10^-7)","Torque teno virus 8 (10^-8)","Ambidensovirus sp. (10^-7)","Severe acute respiratory syndrome-related coronavirus (10^-7)","Betatorquevirus homini1 (10^-7)","Betatorquevirus homini10 (10^-7)","Betatorquevirus homini12 (10^-7)","Betatorquevirus homini26 (10^-7)","Betatorquevirus homini5 (10^-7)","Betatorquevirus homini6 (10^-7)","Betatorquevirus homini7 (10^-7)","Betatorquevirus homini8 (10^-7)","TTV-like mini virus (10^-7)","Cardiovirus D (10^-7)","CrAssphage ZA (10^-7)","Cosavirus A (10^-7)","Dependoparvovirus primate1 (10^-5)","Enterovirus C (10^-7)","Gammatorquevirus homidi1 (10^-7)","Gammatorquevirus homidi10 (10^-7)","Gammatorquevirus homidi11 (10^-6)","Gammatorquevirus homidi12 (10^-7)","Gammatorquevirus homidi13 (10^-7)","Gammatorquevirus homidi14 (10^-7)","Gammatorquevirus homidi2 (10^-7)","Gammatorquevirus homidi3 (10^-6)","Gammatorquevirus homidi4 (10^-7)","Gammatorquevirus homidi6 (10^-7)","Gammatorquevirus homidi7 (10^-7)","Gammatorquevirus homidi8 (10^-6)","Gammatorquevirus homidi9 (10^-7)","Inovirus D_HF3_19 (10^-7)","Inovirus D_HF5_49 (10^-7)","Human mastadenovirus F (10^-6)","Vaccinia virus (10^-7)","Rotavirus A (10^-6)","Sapporo virus (10^-7)","Anelloviridae sp. (10^-5)","Gokushovirus WZ-2015a (10^-7)","Hudisavirus sp. (10^-7)","Human gut gokushovirus (10^-6)","Micro Torque teno virus (10^-7)","Microviridae sp. (10^-5)","Microvirus sp. (10^-6)","Small anellovirus (10^-7)","Torque teno virus (10^-6)"]},"options":{"NodeID":"name","NodeGroup":"name","LinkGroup":null,"colourScale":"d3.scaleOrdinal(d3.schemeCategory20);","fontSize":14,"fontFamily":null,"nodeWidth":25,"nodePadding":10,"units":"","margin":{"top":null,"right":null,"bottom":null,"left":null},"iterations":32,"sinksRight":true}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div id="quarto-navigation-envelope" class="hidden">
<p><span class="hidden quarto-markdown-envelope-contents" data-render-id="quarto-int-sidebar-title">Harmon’s Public Notebook</span> <span class="hidden quarto-markdown-envelope-contents" data-render-id="quarto-int-navbar-title">Harmon’s Public Notebook</span></p>
</div>
<div id="quarto-meta-markdown" class="hidden">
<p><span class="hidden quarto-markdown-envelope-contents" data-render-id="quarto-metatitle">Workflow of Thijssen et al. (2023) – Harmon’s Public Notebook</span> <span class="hidden quarto-markdown-envelope-contents" data-render-id="quarto-twittercardtitle">Workflow of Thijssen et al. (2023) – Harmon’s Public Notebook</span> <span class="hidden quarto-markdown-envelope-contents" data-render-id="quarto-ogcardtitle">Workflow of Thijssen et al. (2023) – Harmon’s Public Notebook</span> <span class="hidden quarto-markdown-envelope-contents" data-render-id="quarto-metasitename">Harmon’s Public Notebook</span> <span class="hidden quarto-markdown-envelope-contents" data-render-id="quarto-twittercarddesc">Pooled plasma from Iran (DNA + RNA)</span> <span class="hidden quarto-markdown-envelope-contents" data-render-id="quarto-ogcardddesc">Pooled plasma from Iran (DNA + RNA)</span></p>
</div>
<!-- -->
<div class="quarto-embedded-source-code">
<div class="sourceCode" id="cb36" data-shortcodes="false"><pre class="sourceCode markdown"><code class="sourceCode markdown"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> &quot;Workflow of Thijssen et al. (2023)&quot;</span></span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a><span class="an">subtitle:</span><span class="co"> &quot;Pooled plasma from Iran (DNA + RNA)&quot;</span></span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> &quot;Harmon Bhasin&quot;</span></span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> 2024-07-22</span></span>
<span id="cb36-6"><a href="#cb36-6" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span></span>
<span id="cb36-7"><a href="#cb36-7" aria-hidden="true" tabindex="-1"></a><span class="co">  html:</span></span>
<span id="cb36-8"><a href="#cb36-8" aria-hidden="true" tabindex="-1"></a><span class="co">    toc: true # table of contents</span></span>
<span id="cb36-9"><a href="#cb36-9" aria-hidden="true" tabindex="-1"></a><span class="co">    toc-title: &quot;Table of contents&quot; # table of contents title</span></span>
<span id="cb36-10"><a href="#cb36-10" aria-hidden="true" tabindex="-1"></a><span class="co">    number-sections: true # number sections</span></span>
<span id="cb36-11"><a href="#cb36-11" aria-hidden="true" tabindex="-1"></a><span class="co">    number-depth: 3 # number depth to show in table of contents</span></span>
<span id="cb36-12"><a href="#cb36-12" aria-hidden="true" tabindex="-1"></a><span class="co">    toc-location: right # table of contents location</span></span>
<span id="cb36-13"><a href="#cb36-13" aria-hidden="true" tabindex="-1"></a><span class="co">    page-layout: full # full page layout</span></span>
<span id="cb36-14"><a href="#cb36-14" aria-hidden="true" tabindex="-1"></a><span class="co">    code-fold: true # Keep option to fold code (i.e. show or hide it; default: hide)</span></span>
<span id="cb36-15"><a href="#cb36-15" aria-hidden="true" tabindex="-1"></a><span class="co">    code-tools: true # Code menu in the header of your document that provides various tools for readers to interact with the source code</span></span>
<span id="cb36-16"><a href="#cb36-16" aria-hidden="true" tabindex="-1"></a><span class="co">    code-link: true # Enables hyper-linking of functions within code blocks to their online documentation</span></span>
<span id="cb36-17"><a href="#cb36-17" aria-hidden="true" tabindex="-1"></a><span class="co">    df-print: paged # print data frame</span></span>
<span id="cb36-18"><a href="#cb36-18" aria-hidden="true" tabindex="-1"></a><span class="co">    fig-format: svg</span></span>
<span id="cb36-19"><a href="#cb36-19" aria-hidden="true" tabindex="-1"></a><span class="co">    other-links:</span></span>
<span id="cb36-20"><a href="#cb36-20" aria-hidden="true" tabindex="-1"></a><span class="co">      - text: Paper</span></span>
<span id="cb36-21"><a href="#cb36-21" aria-hidden="true" tabindex="-1"></a><span class="co">        href: https://doi.org/10.3390/v15071425</span></span>
<span id="cb36-22"><a href="#cb36-22" aria-hidden="true" tabindex="-1"></a><span class="co">      - text: Data</span></span>
<span id="cb36-23"><a href="#cb36-23" aria-hidden="true" tabindex="-1"></a><span class="co">        href: https://www.ncbi.nlm.nih.gov/bioproject/?term=PRJNA983534</span></span>
<span id="cb36-24"><a href="#cb36-24" aria-hidden="true" tabindex="-1"></a><span class="co">    code-links:</span></span>
<span id="cb36-25"><a href="#cb36-25" aria-hidden="true" tabindex="-1"></a><span class="co">      - text: Code for this post</span></span>
<span id="cb36-26"><a href="#cb36-26" aria-hidden="true" tabindex="-1"></a><span class="co">        icon: file-code</span></span>
<span id="cb36-27"><a href="#cb36-27" aria-hidden="true" tabindex="-1"></a><span class="co">        href: https://github.com/naobservatory/harmons-public-notebook/blob/main/notebooks/2024-07-22-thijssen.qmd</span></span>
<span id="cb36-28"><a href="#cb36-28" aria-hidden="true" tabindex="-1"></a><span class="an">editor:</span><span class="co"> </span></span>
<span id="cb36-29"><a href="#cb36-29" aria-hidden="true" tabindex="-1"></a><span class="co">  visual: true</span></span>
<span id="cb36-30"><a href="#cb36-30" aria-hidden="true" tabindex="-1"></a><span class="co">  render-on-save: true</span></span>
<span id="cb36-31"><a href="#cb36-31" aria-hidden="true" tabindex="-1"></a><span class="an">comments:</span></span>
<span id="cb36-32"><a href="#cb36-32" aria-hidden="true" tabindex="-1"></a><span class="co">  hypothesis: true # hypothesis</span></span>
<span id="cb36-33"><a href="#cb36-33" aria-hidden="true" tabindex="-1"></a><span class="an">execute:</span><span class="co"> </span></span>
<span id="cb36-34"><a href="#cb36-34" aria-hidden="true" tabindex="-1"></a><span class="co">  freeze: auto</span></span>
<span id="cb36-35"><a href="#cb36-35" aria-hidden="true" tabindex="-1"></a><span class="co">  cache: true</span></span>
<span id="cb36-36"><a href="#cb36-36" aria-hidden="true" tabindex="-1"></a><span class="an">title-block-banner:</span><span class="co"> &quot;#de2d26&quot;</span></span>
<span id="cb36-37"><a href="#cb36-37" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb36-38"><a href="#cb36-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-39"><a href="#cb36-39" aria-hidden="true" tabindex="-1"></a>quarto-executable-code-5450563D</span>
<span id="cb36-40"><a href="#cb36-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-41"><a href="#cb36-41" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb36-42"><a href="#cb36-42" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: load-packages</span></span>
<span id="cb36-43"><a href="#cb36-43" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb36-44"><a href="#cb36-44" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb36-45"><a href="#cb36-45" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb36-46"><a href="#cb36-46" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cowplot)</span>
<span id="cb36-47"><a href="#cb36-47" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb36-48"><a href="#cb36-48" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(fastqcr)</span>
<span id="cb36-49"><a href="#cb36-49" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)</span>
<span id="cb36-50"><a href="#cb36-50" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(networkD3)</span>
<span id="cb36-51"><a href="#cb36-51" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb36-52"><a href="#cb36-52" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(htmlwidgets)</span>
<span id="cb36-53"><a href="#cb36-53" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggbeeswarm)</span>
<span id="cb36-54"><a href="#cb36-54" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(latex2exp)</span>
<span id="cb36-55"><a href="#cb36-55" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(languageserver)</span>
<span id="cb36-56"><a href="#cb36-56" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scales)</span>
<span id="cb36-57"><a href="#cb36-57" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb36-58"><a href="#cb36-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-59"><a href="#cb36-59" aria-hidden="true" tabindex="-1"></a>head_dir <span class="ot">&lt;-</span> <span class="st">&quot;/Users/harmonbhasin/work/securebio/&quot;</span></span>
<span id="cb36-60"><a href="#cb36-60" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="fu">sprintf</span>(<span class="st">&quot;%s/sampling-strategies/scripts/aux_plot-theme.R&quot;</span>, head_dir))</span>
<span id="cb36-61"><a href="#cb36-61" aria-hidden="true" tabindex="-1"></a>image_dir <span class="ot">&lt;-</span> <span class="fu">sprintf</span>(<span class="st">&#39;%s/lab_meeting/thijssen&#39;</span>, head_dir)</span>
<span id="cb36-62"><a href="#cb36-62" aria-hidden="true" tabindex="-1"></a>theme_base <span class="ot">&lt;-</span> theme_base <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio =</span> <span class="cn">NULL</span>)</span>
<span id="cb36-63"><a href="#cb36-63" aria-hidden="true" tabindex="-1"></a>theme_kit <span class="ot">&lt;-</span> theme_base <span class="sc">+</span> <span class="fu">theme</span>(</span>
<span id="cb36-64"><a href="#cb36-64" aria-hidden="true" tabindex="-1"></a>  <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">angle =</span> <span class="dv">45</span>),</span>
<span id="cb36-65"><a href="#cb36-65" aria-hidden="true" tabindex="-1"></a>  <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb36-66"><a href="#cb36-66" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb36-67"><a href="#cb36-67" aria-hidden="true" tabindex="-1"></a>tnl <span class="ot">&lt;-</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>)</span>
<span id="cb36-68"><a href="#cb36-68" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb36-69"><a href="#cb36-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-70"><a href="#cb36-70" aria-hidden="true" tabindex="-1"></a>quarto-executable-code-5450563D</span>
<span id="cb36-71"><a href="#cb36-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-72"><a href="#cb36-72" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb36-73"><a href="#cb36-73" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: set-up-paths</span></span>
<span id="cb36-74"><a href="#cb36-74" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb36-75"><a href="#cb36-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-76"><a href="#cb36-76" aria-hidden="true" tabindex="-1"></a><span class="co"># Data input paths</span></span>
<span id="cb36-77"><a href="#cb36-77" aria-hidden="true" tabindex="-1"></a>data_dir <span class="ot">&lt;-</span> <span class="st">&quot;/Users/harmonbhasin/work/securebio/nao-harmon/thijssen2023/analysis/&quot;</span></span>
<span id="cb36-78"><a href="#cb36-78" aria-hidden="true" tabindex="-1"></a>input_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">&quot;input&quot;</span>)</span>
<span id="cb36-79"><a href="#cb36-79" aria-hidden="true" tabindex="-1"></a>results_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">&quot;results&quot;</span>)</span>
<span id="cb36-80"><a href="#cb36-80" aria-hidden="true" tabindex="-1"></a>qc_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(results_dir, <span class="st">&quot;qc&quot;</span>)</span>
<span id="cb36-81"><a href="#cb36-81" aria-hidden="true" tabindex="-1"></a>hv_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(results_dir, <span class="st">&quot;hv&quot;</span>)</span>
<span id="cb36-82"><a href="#cb36-82" aria-hidden="true" tabindex="-1"></a>libraries_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(input_dir, <span class="st">&quot;libraries.csv&quot;</span>)</span>
<span id="cb36-83"><a href="#cb36-83" aria-hidden="true" tabindex="-1"></a>basic_stats_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(qc_dir, <span class="st">&quot;qc_basic_stats.tsv.gz&quot;</span>)</span>
<span id="cb36-84"><a href="#cb36-84" aria-hidden="true" tabindex="-1"></a>adapter_stats_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(qc_dir, <span class="st">&quot;qc_adapter_stats.tsv.gz&quot;</span>)</span>
<span id="cb36-85"><a href="#cb36-85" aria-hidden="true" tabindex="-1"></a>quality_base_stats_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(qc_dir, <span class="st">&quot;qc_quality_base_stats.tsv.gz&quot;</span>)</span>
<span id="cb36-86"><a href="#cb36-86" aria-hidden="true" tabindex="-1"></a>quality_seq_stats_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(qc_dir, <span class="st">&quot;qc_quality_sequence_stats.tsv.gz&quot;</span>)</span>
<span id="cb36-87"><a href="#cb36-87" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb36-88"><a href="#cb36-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-89"><a href="#cb36-89" aria-hidden="true" tabindex="-1"></a>This is another potential study of <span class="co">[</span><span class="ot">this series</span><span class="co">](https://data.securebio.org/harmons-public-notebook/notebooks/2024-07-08_cebria-mendoza.html)</span>. In this post, I analyze <span class="co">[</span><span class="ot">Thijssen 2023</span><span class="co">](https://doi.org/10.3390/v15071425)</span>, a dataset with 20 pooled samples from 100 healthy individuals in Iran.</span>
<span id="cb36-90"><a href="#cb36-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-91"><a href="#cb36-91" aria-hidden="true" tabindex="-1"></a>*This notebook uses the MGS Workflow v2.2.1 (note that this is old).*</span>
<span id="cb36-92"><a href="#cb36-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-93"><a href="#cb36-93" aria-hidden="true" tabindex="-1"></a><span class="fu"># The raw data</span></span>
<span id="cb36-94"><a href="#cb36-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-95"><a href="#cb36-95" aria-hidden="true" tabindex="-1"></a><span class="fu">## About</span></span>
<span id="cb36-96"><a href="#cb36-96" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-97"><a href="#cb36-97" aria-hidden="true" tabindex="-1"></a><span class="co">[</span><span class="ot">This dataset</span><span class="co">](https://www.ncbi.nlm.nih.gov/bioproject/?term=PRJNA983534)</span> comprises 20 samples, each derived from plasma pools of 5 people from Iran. In total, 100 healthy individuals contributed to these pools. For each pooled sample, a combined DNA and RNA library preparation was performed, resulting in a single sequencing output that captures both nucleic acid types. This approach provides comprehensive genetic information but precludes separate analysis of DNA and RNA from individual samples.</span>
<span id="cb36-98"><a href="#cb36-98" aria-hidden="true" tabindex="-1"></a>quarto-executable-code-5450563D</span>
<span id="cb36-99"><a href="#cb36-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-100"><a href="#cb36-100" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb36-101"><a href="#cb36-101" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb36-102"><a href="#cb36-102" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: load-libraries</span></span>
<span id="cb36-103"><a href="#cb36-103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-104"><a href="#cb36-104" aria-hidden="true" tabindex="-1"></a><span class="co"># Import libraries and extract metadata from sample names</span></span>
<span id="cb36-105"><a href="#cb36-105" aria-hidden="true" tabindex="-1"></a>libraries <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(libraries_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb36-106"><a href="#cb36-106" aria-hidden="true" tabindex="-1"></a>meta_data <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&#39;/Users/harmonbhasin/work/securebio/nao-harmon/thijssen2023/preprocessing/SraRunTable.txt&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-107"><a href="#cb36-107" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="st">&#39;Library Name&#39;</span>, Run) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">sample =</span> Run, <span class="at">library =</span> <span class="st">&#39;Library Name&#39;</span>)</span>
<span id="cb36-108"><a href="#cb36-108" aria-hidden="true" tabindex="-1"></a>libraries <span class="ot">&lt;-</span> libraries <span class="sc">%&gt;%</span></span>
<span id="cb36-109"><a href="#cb36-109" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(meta_data, <span class="at">by =</span> <span class="st">&#39;sample&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-110"><a href="#cb36-110" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, <span class="at">library=</span>library.y) <span class="sc">%&gt;%</span> </span>
<span id="cb36-111"><a href="#cb36-111" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(library <span class="sc">!=</span> <span class="st">&#39;PCL21&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-112"><a href="#cb36-112" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">library =</span> <span class="fu">factor</span>(library, <span class="at">levels =</span> <span class="fu">c</span>(<span class="fu">paste0</span>(<span class="st">&#39;PCL&#39;</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>))))</span>
<span id="cb36-113"><a href="#cb36-113" aria-hidden="true" tabindex="-1"></a>libraries</span>
<span id="cb36-114"><a href="#cb36-114" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-115"><a href="#cb36-115" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb36-116"><a href="#cb36-116" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-117"><a href="#cb36-117" aria-hidden="true" tabindex="-1"></a><span class="fu">### Sample + library preparation </span></span>
<span id="cb36-118"><a href="#cb36-118" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-119"><a href="#cb36-119" aria-hidden="true" tabindex="-1"></a>Paraphrased from the <span class="co">[</span><span class="ot">paper</span><span class="co">](https://www.mdpi.com/1999-4915/15/7/1425)</span>:</span>
<span id="cb36-120"><a href="#cb36-120" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-121"><a href="#cb36-121" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; Study was conducted from 2017-2018 in the Boushehr Province, Iran. Participants were recruited in one of the five transfusion clinics located in Boushehr. Seven milliliters of blood were collected from each individual during blood transfusion or donation. Immediately after collection, plasma was separated from the samples and stored at −70 °C. </span></span>
<span id="cb36-122"><a href="#cb36-122" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; </span></span>
<span id="cb36-123"><a href="#cb36-123" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; Initially, the samples were centrifuged and 100 µL of the supernatant was pooled with five samples per pool. The pooled plasma samples were subjected to an adapted version of the NetoVIR protocol for viral particle enrichment and metagenomic sequencing (</span><span class="co">[</span><span class="ot">more information can be found here</span><span class="co">](https://www.mdpi.com/1999-4915/15/7/1425#B20-viruses-15-01425)</span><span class="at">). Pooled plasma samples were centrifuged for 3 min at 17,000× g and filtered through 0.8 µm polyether sulphone filters. To remove free-floating nucleic acids, the filtered samples were subjected to a nuclease treatment with a cocktail of 1 µL micrococcal nuclease and 2 µL benzonase for 2 h at 37 °C. Both viral DNA and RNA were extracted and randomly amplified (including primary step of reverse transcription) with the Whole Transcriptome Amplification 2 kit for 20 cycles. </span></span>
<span id="cb36-124"><a href="#cb36-124" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; </span></span>
<span id="cb36-125"><a href="#cb36-125" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; The amplification product was purified with the MSB SPIN PCRAPACE kit and prepared for sequencing by using the Nextera XT kit. DNA products were quantified with the Qubit fluorometer, and the High-Sensitivity DNA kit for the Bioanalyzer 2100 was used to determine the average library fragment size. Samples were pooled in equimolar ratios, and paired-end sequencing (2 × 150 bp) was performed on a NextSeq 500 Illumina platform with an average of 10 million reads per sample.</span></span>
<span id="cb36-126"><a href="#cb36-126" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-127"><a href="#cb36-127" aria-hidden="true" tabindex="-1"></a><span class="fu">## Quality control metrics</span></span>
<span id="cb36-128"><a href="#cb36-128" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-129"><a href="#cb36-129" aria-hidden="true" tabindex="-1"></a>quarto-executable-code-5450563D</span>
<span id="cb36-130"><a href="#cb36-130" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-131"><a href="#cb36-131" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb36-132"><a href="#cb36-132" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: read-qc-data</span></span>
<span id="cb36-133"><a href="#cb36-133" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb36-134"><a href="#cb36-134" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-135"><a href="#cb36-135" aria-hidden="true" tabindex="-1"></a><span class="co"># Import QC data</span></span>
<span id="cb36-136"><a href="#cb36-136" aria-hidden="true" tabindex="-1"></a>stages <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;raw_concat&quot;</span>, <span class="st">&quot;cleaned&quot;</span>, <span class="st">&quot;dedup&quot;</span>, <span class="st">&quot;ribo_initial&quot;</span>, <span class="st">&quot;ribo_secondary&quot;</span>)</span>
<span id="cb36-137"><a href="#cb36-137" aria-hidden="true" tabindex="-1"></a>basic_stats <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(basic_stats_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-138"><a href="#cb36-138" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">&quot;sample&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb36-139"><a href="#cb36-139" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">factor</span>(stage, <span class="at">levels =</span> stages),</span>
<span id="cb36-140"><a href="#cb36-140" aria-hidden="true" tabindex="-1"></a>         <span class="at">sample =</span> <span class="fu">fct_inorder</span>(sample))</span>
<span id="cb36-141"><a href="#cb36-141" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-142"><a href="#cb36-142" aria-hidden="true" tabindex="-1"></a>adapter_stats <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(adapter_stats_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-143"><a href="#cb36-143" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">&quot;sample&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb36-144"><a href="#cb36-144" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">factor</span>(stage, <span class="at">levels =</span> stages),</span>
<span id="cb36-145"><a href="#cb36-145" aria-hidden="true" tabindex="-1"></a>         <span class="at">read_pair =</span> <span class="fu">fct_inorder</span>(<span class="fu">as.character</span>(read_pair)))</span>
<span id="cb36-146"><a href="#cb36-146" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-147"><a href="#cb36-147" aria-hidden="true" tabindex="-1"></a>quality_base_stats <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(quality_base_stats_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-148"><a href="#cb36-148" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">&quot;sample&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb36-149"><a href="#cb36-149" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">factor</span>(stage, <span class="at">levels =</span> stages),</span>
<span id="cb36-150"><a href="#cb36-150" aria-hidden="true" tabindex="-1"></a>         <span class="at">read_pair =</span> <span class="fu">fct_inorder</span>(<span class="fu">as.character</span>(read_pair)))</span>
<span id="cb36-151"><a href="#cb36-151" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb36-152"><a href="#cb36-152" aria-hidden="true" tabindex="-1"></a>quality_seq_stats <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(quality_seq_stats_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-153"><a href="#cb36-153" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">&quot;sample&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb36-154"><a href="#cb36-154" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">factor</span>(stage, <span class="at">levels =</span> stages),</span>
<span id="cb36-155"><a href="#cb36-155" aria-hidden="true" tabindex="-1"></a>         <span class="at">read_pair =</span> <span class="fu">fct_inorder</span>(<span class="fu">as.character</span>(read_pair)))</span>
<span id="cb36-156"><a href="#cb36-156" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb36-157"><a href="#cb36-157" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb36-158"><a href="#cb36-158" aria-hidden="true" tabindex="-1"></a> <span class="co"># Filter to raw data</span></span>
<span id="cb36-159"><a href="#cb36-159" aria-hidden="true" tabindex="-1"></a> basic_stats_raw <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">&quot;raw_concat&quot;</span>)</span>
<span id="cb36-160"><a href="#cb36-160" aria-hidden="true" tabindex="-1"></a> adapter_stats_raw <span class="ot">&lt;-</span> adapter_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">&quot;raw_concat&quot;</span>)</span>
<span id="cb36-161"><a href="#cb36-161" aria-hidden="true" tabindex="-1"></a> quality_base_stats_raw <span class="ot">&lt;-</span> quality_base_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">&quot;raw_concat&quot;</span>)</span>
<span id="cb36-162"><a href="#cb36-162" aria-hidden="true" tabindex="-1"></a> quality_seq_stats_raw <span class="ot">&lt;-</span> quality_seq_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">&quot;raw_concat&quot;</span>)</span>
<span id="cb36-163"><a href="#cb36-163" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb36-164"><a href="#cb36-164" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb36-165"><a href="#cb36-165" aria-hidden="true" tabindex="-1"></a> <span class="co"># Get key values for readout</span></span>
<span id="cb36-166"><a href="#cb36-166" aria-hidden="true" tabindex="-1"></a>raw_read_counts <span class="ot">&lt;-</span> basic_stats_raw <span class="sc">%&gt;%</span> </span>
<span id="cb36-167"><a href="#cb36-167" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">rmin =</span> <span class="fu">min</span>(n_read_pairs), </span>
<span id="cb36-168"><a href="#cb36-168" aria-hidden="true" tabindex="-1"></a>            <span class="at">rmax=</span><span class="fu">max</span>(n_read_pairs),</span>
<span id="cb36-169"><a href="#cb36-169" aria-hidden="true" tabindex="-1"></a>            <span class="at">rmean=</span><span class="fu">mean</span>(n_read_pairs), </span>
<span id="cb36-170"><a href="#cb36-170" aria-hidden="true" tabindex="-1"></a>            <span class="at">rtot =</span> <span class="fu">sum</span>(n_read_pairs),</span>
<span id="cb36-171"><a href="#cb36-171" aria-hidden="true" tabindex="-1"></a>            <span class="at">btot =</span> <span class="fu">sum</span>(n_bases_approx),</span>
<span id="cb36-172"><a href="#cb36-172" aria-hidden="true" tabindex="-1"></a>            <span class="at">dmin =</span> <span class="fu">min</span>(percent_duplicates), </span>
<span id="cb36-173"><a href="#cb36-173" aria-hidden="true" tabindex="-1"></a>            <span class="at">dmax=</span><span class="fu">max</span>(percent_duplicates),</span>
<span id="cb36-174"><a href="#cb36-174" aria-hidden="true" tabindex="-1"></a>            <span class="at">dmean=</span><span class="fu">mean</span>(percent_duplicates), </span>
<span id="cb36-175"><a href="#cb36-175" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups =</span> <span class="st">&quot;drop&quot;</span>)</span>
<span id="cb36-176"><a href="#cb36-176" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb36-177"><a href="#cb36-177" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-178"><a href="#cb36-178" aria-hidden="true" tabindex="-1"></a>In total, these 20 samples contained ~116M read pairs. The samples had 1.3M - 11.1M (mean ~5.8M) read pairs each. </span>
<span id="cb36-179"><a href="#cb36-179" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-180"><a href="#cb36-180" aria-hidden="true" tabindex="-1"></a>The number of reads looks pretty good, although a few samples have a much lower number of reads. The total number of base pairs also looks reasonable, matching the trends of the number of reads. The duplication rate is low. Adapter content is quite high for nextera-transposase-sequence (<span class="co">[</span><span class="ot">library contamination</span><span class="co">](https://www.biostars.org/p/119774/)</span>).</span>
<span id="cb36-181"><a href="#cb36-181" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-182"><a href="#cb36-182" aria-hidden="true" tabindex="-1"></a>As we&#39;d expect, we see higher quality Q scores near the beginning of the read and a gradual decline towards the end of the read, however all positions seem to have a Q score of 35 which means that our reads are \~99.97% accurate. When looking at the Q score over all sequences, we can see a sharp peak around 35, which corresponds to the previous plot, indicating high read quality. </span>
<span id="cb36-183"><a href="#cb36-183" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-184"><a href="#cb36-184" aria-hidden="true" tabindex="-1"></a>quarto-executable-code-5450563D</span>
<span id="cb36-185"><a href="#cb36-185" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-186"><a href="#cb36-186" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb36-187"><a href="#cb36-187" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 9</span></span>
<span id="cb36-188"><a href="#cb36-188" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb36-189"><a href="#cb36-189" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-basic-stats</span></span>
<span id="cb36-190"><a href="#cb36-190" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-191"><a href="#cb36-191" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare data</span></span>
<span id="cb36-192"><a href="#cb36-192" aria-hidden="true" tabindex="-1"></a>basic_stats_raw_metrics <span class="ot">&lt;-</span> basic_stats_raw <span class="sc">%&gt;%</span></span>
<span id="cb36-193"><a href="#cb36-193" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(library,</span>
<span id="cb36-194"><a href="#cb36-194" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at"># Read pairs</span><span class="st">`</span> <span class="ot">=</span> n_read_pairs,</span>
<span id="cb36-195"><a href="#cb36-195" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Total base pairs</span><span class="sc">\n</span><span class="at">(approx)</span><span class="st">`</span> <span class="ot">=</span> n_bases_approx,</span>
<span id="cb36-196"><a href="#cb36-196" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">% Duplicates</span><span class="sc">\n</span><span class="at">(FASTQC)</span><span class="st">`</span> <span class="ot">=</span> percent_duplicates) <span class="sc">%&gt;%</span></span>
<span id="cb36-197"><a href="#cb36-197" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>library, <span class="at">names_to =</span> <span class="st">&quot;metric&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;value&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-198"><a href="#cb36-198" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">metric =</span> <span class="fu">fct_inorder</span>(metric)) </span>
<span id="cb36-199"><a href="#cb36-199" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-200"><a href="#cb36-200" aria-hidden="true" tabindex="-1"></a><span class="co"># Set up plot templates</span></span>
<span id="cb36-201"><a href="#cb36-201" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-202"><a href="#cb36-202" aria-hidden="true" tabindex="-1"></a>g_basic <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(basic_stats_raw_metrics, <span class="fu">aes</span>(<span class="at">x =</span> library, <span class="at">y =</span> value)) <span class="sc">+</span></span>
<span id="cb36-203"><a href="#cb36-203" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position =</span> <span class="st">&quot;dodge&quot;</span>) <span class="sc">+</span></span>
<span id="cb36-204"><a href="#cb36-204" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>() <span class="sc">+</span></span>
<span id="cb36-205"><a href="#cb36-205" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb36-206"><a href="#cb36-206" aria-hidden="true" tabindex="-1"></a>  <span class="fu">expand_limits</span>(<span class="at">y =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">100</span>)) <span class="sc">+</span></span>
<span id="cb36-207"><a href="#cb36-207" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(metric <span class="sc">~</span> ., <span class="at">scales =</span> <span class="st">&quot;free&quot;</span>, <span class="at">space =</span> <span class="st">&quot;free_x&quot;</span>, <span class="at">switch =</span> <span class="st">&quot;y&quot;</span>) <span class="sc">+</span></span>
<span id="cb36-208"><a href="#cb36-208" aria-hidden="true" tabindex="-1"></a>  theme_kit <span class="sc">+</span> <span class="fu">theme</span>(</span>
<span id="cb36-209"><a href="#cb36-209" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb36-210"><a href="#cb36-210" aria-hidden="true" tabindex="-1"></a>    <span class="at">strip.text.y =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">&quot;plain&quot;</span>)</span>
<span id="cb36-211"><a href="#cb36-211" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb36-212"><a href="#cb36-212" aria-hidden="true" tabindex="-1"></a>g_basic</span>
<span id="cb36-213"><a href="#cb36-213" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb36-214"><a href="#cb36-214" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-215"><a href="#cb36-215" aria-hidden="true" tabindex="-1"></a>quarto-executable-code-5450563D</span>
<span id="cb36-216"><a href="#cb36-216" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-217"><a href="#cb36-217" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb36-218"><a href="#cb36-218" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-raw-quality</span></span>
<span id="cb36-219"><a href="#cb36-219" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 8</span></span>
<span id="cb36-220"><a href="#cb36-220" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-221"><a href="#cb36-221" aria-hidden="true" tabindex="-1"></a><span class="co"># Set up plotting templates</span></span>
<span id="cb36-222"><a href="#cb36-222" aria-hidden="true" tabindex="-1"></a>g_qual_raw <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">linetype=</span>read_pair, <span class="at">group=</span><span class="fu">interaction</span>(sample,read_pair))) <span class="sc">+</span> </span>
<span id="cb36-223"><a href="#cb36-223" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_linetype_discrete</span>(<span class="at">name =</span> <span class="st">&quot;Read Pair&quot;</span>) <span class="sc">+</span></span>
<span id="cb36-224"><a href="#cb36-224" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color=</span><span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>),</span>
<span id="cb36-225"><a href="#cb36-225" aria-hidden="true" tabindex="-1"></a>         <span class="at">linetype =</span> <span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>)) <span class="sc">+</span></span>
<span id="cb36-226"><a href="#cb36-226" aria-hidden="true" tabindex="-1"></a>  theme_base</span>
<span id="cb36-227"><a href="#cb36-227" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-228"><a href="#cb36-228" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize adapters</span></span>
<span id="cb36-229"><a href="#cb36-229" aria-hidden="true" tabindex="-1"></a>g_adapters_raw <span class="ot">&lt;-</span> g_qual_raw <span class="sc">+</span> </span>
<span id="cb36-230"><a href="#cb36-230" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>pc_adapters), <span class="at">data=</span>adapter_stats_raw) <span class="sc">+</span></span>
<span id="cb36-231"><a href="#cb36-231" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">&quot;% Adapters&quot;</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb36-232"><a href="#cb36-232" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">100</span>,<span class="dv">10</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb36-233"><a href="#cb36-233" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">&quot;Position&quot;</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb36-234"><a href="#cb36-234" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">500</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb36-235"><a href="#cb36-235" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(.<span class="sc">~</span>adapter)</span>
<span id="cb36-236"><a href="#cb36-236" aria-hidden="true" tabindex="-1"></a>g_adapters_raw</span>
<span id="cb36-237"><a href="#cb36-237" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-238"><a href="#cb36-238" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize quality</span></span>
<span id="cb36-239"><a href="#cb36-239" aria-hidden="true" tabindex="-1"></a>g_quality_base_raw <span class="ot">&lt;-</span> g_qual_raw <span class="sc">+</span></span>
<span id="cb36-240"><a href="#cb36-240" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">&quot;dashed&quot;</span>, <span class="at">color=</span><span class="st">&quot;red&quot;</span>) <span class="sc">+</span></span>
<span id="cb36-241"><a href="#cb36-241" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">&quot;dashed&quot;</span>, <span class="at">color=</span><span class="st">&quot;red&quot;</span>) <span class="sc">+</span></span>
<span id="cb36-242"><a href="#cb36-242" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>mean_phred_score), <span class="at">data=</span>quality_base_stats_raw) <span class="sc">+</span></span>
<span id="cb36-243"><a href="#cb36-243" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">&quot;Mean Phred score&quot;</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">10</span>,<span class="dv">45</span>)) <span class="sc">+</span></span>
<span id="cb36-244"><a href="#cb36-244" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">&quot;Position&quot;</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb36-245"><a href="#cb36-245" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">500</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb36-246"><a href="#cb36-246" aria-hidden="true" tabindex="-1"></a>g_quality_base_raw</span>
<span id="cb36-247"><a href="#cb36-247" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-248"><a href="#cb36-248" aria-hidden="true" tabindex="-1"></a>g_quality_seq_raw <span class="ot">&lt;-</span> g_qual_raw <span class="sc">+</span></span>
<span id="cb36-249"><a href="#cb36-249" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">&quot;dashed&quot;</span>, <span class="at">color=</span><span class="st">&quot;red&quot;</span>) <span class="sc">+</span></span>
<span id="cb36-250"><a href="#cb36-250" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">&quot;dashed&quot;</span>, <span class="at">color=</span><span class="st">&quot;red&quot;</span>) <span class="sc">+</span></span>
<span id="cb36-251"><a href="#cb36-251" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>mean_phred_score, <span class="at">y=</span>n_sequences), <span class="at">data=</span>quality_seq_stats_raw) <span class="sc">+</span></span>
<span id="cb36-252"><a href="#cb36-252" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">&quot;Mean Phred score&quot;</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb36-253"><a href="#cb36-253" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">&quot;# Sequences&quot;</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb36-254"><a href="#cb36-254" aria-hidden="true" tabindex="-1"></a>g_quality_seq_raw</span>
<span id="cb36-255"><a href="#cb36-255" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb36-256"><a href="#cb36-256" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-257"><a href="#cb36-257" aria-hidden="true" tabindex="-1"></a><span class="fu"># Preprocessing</span></span>
<span id="cb36-258"><a href="#cb36-258" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-259"><a href="#cb36-259" aria-hidden="true" tabindex="-1"></a><span class="fu">## Summary</span></span>
<span id="cb36-260"><a href="#cb36-260" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-261"><a href="#cb36-261" aria-hidden="true" tabindex="-1"></a>The average fraction of reads at each stage in the preprocessing pipeline is shown in the following table. Adapter trimming &amp; filtering doesn&#39;t lose too many reads. Deduplication loses us about 30% of reads on average, then ribodepletion only loses as about 0.7% on average.</span>
<span id="cb36-262"><a href="#cb36-262" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-263"><a href="#cb36-263" aria-hidden="true" tabindex="-1"></a>quarto-executable-code-5450563D</span>
<span id="cb36-264"><a href="#cb36-264" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-265"><a href="#cb36-265" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb36-266"><a href="#cb36-266" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: preproc-table</span></span>
<span id="cb36-267"><a href="#cb36-267" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-268"><a href="#cb36-268" aria-hidden="true" tabindex="-1"></a><span class="co"># </span><span class="al">TODO</span><span class="co">: Group by pool size as well</span></span>
<span id="cb36-269"><a href="#cb36-269" aria-hidden="true" tabindex="-1"></a><span class="co"># Count read losses</span></span>
<span id="cb36-270"><a href="#cb36-270" aria-hidden="true" tabindex="-1"></a>n_reads_rel <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> </span>
<span id="cb36-271"><a href="#cb36-271" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, stage, percent_duplicates, n_read_pairs) <span class="sc">%&gt;%</span></span>
<span id="cb36-272"><a href="#cb36-272" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb36-273"><a href="#cb36-273" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(sample, stage) <span class="sc">%&gt;%</span></span>
<span id="cb36-274"><a href="#cb36-274" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_retained =</span> n_read_pairs <span class="sc">/</span> <span class="fu">lag</span>(n_read_pairs),</span>
<span id="cb36-275"><a href="#cb36-275" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_lost =</span> <span class="dv">1</span> <span class="sc">-</span> p_reads_retained,</span>
<span id="cb36-276"><a href="#cb36-276" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_retained_abs =</span> n_read_pairs <span class="sc">/</span> n_read_pairs[<span class="dv">1</span>],</span>
<span id="cb36-277"><a href="#cb36-277" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_lost_abs =</span> <span class="dv">1</span><span class="sc">-</span>p_reads_retained_abs,</span>
<span id="cb36-278"><a href="#cb36-278" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_lost_abs_marginal =</span> p_reads_lost_abs <span class="sc">-</span> <span class="fu">lag</span>(p_reads_lost_abs))</span>
<span id="cb36-279"><a href="#cb36-279" aria-hidden="true" tabindex="-1"></a>n_reads_rel_display <span class="ot">&lt;-</span> n_reads_rel <span class="sc">%&gt;%</span> </span>
<span id="cb36-280"><a href="#cb36-280" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">Stage=</span>stage) <span class="sc">%&gt;%</span> </span>
<span id="cb36-281"><a href="#cb36-281" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Stage) <span class="sc">%&gt;%</span> </span>
<span id="cb36-282"><a href="#cb36-282" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="st">`</span><span class="at">% Total Reads Lost (Cumulative)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">round</span>(<span class="fu">min</span>(p_reads_lost_abs<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">&quot;-&quot;</span>, <span class="fu">round</span>(<span class="fu">max</span>(p_reads_lost_abs<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">&quot; (mean &quot;</span>, <span class="fu">round</span>(<span class="fu">mean</span>(p_reads_lost_abs<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">&quot;)&quot;</span>),</span>
<span id="cb36-283"><a href="#cb36-283" aria-hidden="true" tabindex="-1"></a>            <span class="st">`</span><span class="at">% Total Reads Lost (Marginal)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">round</span>(<span class="fu">min</span>(p_reads_lost_abs_marginal<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">&quot;-&quot;</span>, <span class="fu">round</span>(<span class="fu">max</span>(p_reads_lost_abs_marginal<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">&quot; (mean &quot;</span>, <span class="fu">round</span>(<span class="fu">mean</span>(p_reads_lost_abs_marginal<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">&quot;)&quot;</span>), <span class="at">.groups=</span><span class="st">&quot;drop&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-284"><a href="#cb36-284" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Stage <span class="sc">!=</span> <span class="st">&quot;raw_concat&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-285"><a href="#cb36-285" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Stage =</span> Stage <span class="sc">%&gt;%</span> as.numeric <span class="sc">%&gt;%</span> <span class="fu">factor</span>(<span class="at">labels=</span><span class="fu">c</span>(<span class="st">&quot;Trimming &amp; filtering&quot;</span>, <span class="st">&quot;Deduplication&quot;</span>, <span class="st">&quot;Initial ribodepletion&quot;</span>, <span class="st">&quot;Secondary ribodepletion&quot;</span>)))</span>
<span id="cb36-286"><a href="#cb36-286" aria-hidden="true" tabindex="-1"></a>n_reads_rel_display</span>
<span id="cb36-287"><a href="#cb36-287" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb36-288"><a href="#cb36-288" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-289"><a href="#cb36-289" aria-hidden="true" tabindex="-1"></a><span class="fu">## Quality control metrics</span></span>
<span id="cb36-290"><a href="#cb36-290" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-291"><a href="#cb36-291" aria-hidden="true" tabindex="-1"></a>Adapter trimming gets rid of nextera_transposase_sequenece and poly_g, although there is still a bit of poly_a adapter contamination (<span class="co">[</span><span class="ot">overivew of fastqc adapters</span><span class="co">](https://github.com/s-andrews/FastQC/blob/master/Configuration/adapter_list.txt)</span>). Not sure how to interpret the poly_a contamination, but it&#39;s not that high so I&#39;m not going to worry about it for now. Q score remain the same during read cleaning when looking at the positions, with the end of the read actually improving in score. Q scores across all sequences look pretty much the same throughout cleaning.</span>
<span id="cb36-292"><a href="#cb36-292" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-293"><a href="#cb36-293" aria-hidden="true" tabindex="-1"></a>quarto-executable-code-5450563D</span>
<span id="cb36-294"><a href="#cb36-294" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-295"><a href="#cb36-295" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb36-296"><a href="#cb36-296" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb36-297"><a href="#cb36-297" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-quality</span></span>
<span id="cb36-298"><a href="#cb36-298" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 8</span></span>
<span id="cb36-299"><a href="#cb36-299" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb36-300"><a href="#cb36-300" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-301"><a href="#cb36-301" aria-hidden="true" tabindex="-1"></a>g_qual <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">linetype=</span>read_pair, <span class="at">group=</span><span class="fu">interaction</span>(sample,read_pair))) <span class="sc">+</span> </span>
<span id="cb36-302"><a href="#cb36-302" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_linetype_discrete</span>(<span class="at">name =</span> <span class="st">&quot;Read Pair&quot;</span>) <span class="sc">+</span></span>
<span id="cb36-303"><a href="#cb36-303" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color=</span><span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>),</span>
<span id="cb36-304"><a href="#cb36-304" aria-hidden="true" tabindex="-1"></a>         <span class="at">linetype =</span> <span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>)) <span class="sc">+</span></span>
<span id="cb36-305"><a href="#cb36-305" aria-hidden="true" tabindex="-1"></a>  theme_base</span>
<span id="cb36-306"><a href="#cb36-306" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-307"><a href="#cb36-307" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize adapters</span></span>
<span id="cb36-308"><a href="#cb36-308" aria-hidden="true" tabindex="-1"></a>g_adapters <span class="ot">&lt;-</span> g_qual <span class="sc">+</span> </span>
<span id="cb36-309"><a href="#cb36-309" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>pc_adapters), <span class="at">data=</span>adapter_stats) <span class="sc">+</span></span>
<span id="cb36-310"><a href="#cb36-310" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">&quot;% Adapters&quot;</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">20</span>),</span>
<span id="cb36-311"><a href="#cb36-311" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">50</span>,<span class="dv">10</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb36-312"><a href="#cb36-312" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">&quot;Position&quot;</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb36-313"><a href="#cb36-313" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">140</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb36-314"><a href="#cb36-314" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(stage<span class="sc">~</span>adapter)</span>
<span id="cb36-315"><a href="#cb36-315" aria-hidden="true" tabindex="-1"></a>g_adapters</span>
<span id="cb36-316"><a href="#cb36-316" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-317"><a href="#cb36-317" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize quality</span></span>
<span id="cb36-318"><a href="#cb36-318" aria-hidden="true" tabindex="-1"></a>g_quality_base <span class="ot">&lt;-</span> g_qual <span class="sc">+</span></span>
<span id="cb36-319"><a href="#cb36-319" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">&quot;dashed&quot;</span>, <span class="at">color=</span><span class="st">&quot;red&quot;</span>) <span class="sc">+</span></span>
<span id="cb36-320"><a href="#cb36-320" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">&quot;dashed&quot;</span>, <span class="at">color=</span><span class="st">&quot;red&quot;</span>) <span class="sc">+</span></span>
<span id="cb36-321"><a href="#cb36-321" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>mean_phred_score), <span class="at">data=</span>quality_base_stats) <span class="sc">+</span></span>
<span id="cb36-322"><a href="#cb36-322" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">&quot;Mean Phred score&quot;</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">10</span>,<span class="dv">45</span>)) <span class="sc">+</span></span>
<span id="cb36-323"><a href="#cb36-323" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">&quot;Position&quot;</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb36-324"><a href="#cb36-324" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">140</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb36-325"><a href="#cb36-325" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(stage<span class="sc">~</span>.)</span>
<span id="cb36-326"><a href="#cb36-326" aria-hidden="true" tabindex="-1"></a>g_quality_base</span>
<span id="cb36-327"><a href="#cb36-327" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-328"><a href="#cb36-328" aria-hidden="true" tabindex="-1"></a>g_quality_seq <span class="ot">&lt;-</span> g_qual <span class="sc">+</span></span>
<span id="cb36-329"><a href="#cb36-329" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">&quot;dashed&quot;</span>, <span class="at">color=</span><span class="st">&quot;red&quot;</span>) <span class="sc">+</span></span>
<span id="cb36-330"><a href="#cb36-330" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">&quot;dashed&quot;</span>, <span class="at">color=</span><span class="st">&quot;red&quot;</span>) <span class="sc">+</span></span>
<span id="cb36-331"><a href="#cb36-331" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>mean_phred_score, <span class="at">y=</span>n_sequences), <span class="at">data=</span>quality_seq_stats) <span class="sc">+</span></span>
<span id="cb36-332"><a href="#cb36-332" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">&quot;Mean Phred score&quot;</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb36-333"><a href="#cb36-333" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">&quot;# Sequences&quot;</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb36-334"><a href="#cb36-334" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(stage<span class="sc">~</span>., <span class="at">scales =</span> <span class="st">&quot;free_y&quot;</span>)</span>
<span id="cb36-335"><a href="#cb36-335" aria-hidden="true" tabindex="-1"></a>g_quality_seq</span>
<span id="cb36-336"><a href="#cb36-336" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb36-337"><a href="#cb36-337" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-338"><a href="#cb36-338" aria-hidden="true" tabindex="-1"></a>These plots below show the trends from above in each sample. All samples tend to follow similar trends for deduplication, with a large decrease in read length post adapter trimming. Ribosomal reads were quite low, near 1% for every sample.</span>
<span id="cb36-339"><a href="#cb36-339" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-340"><a href="#cb36-340" aria-hidden="true" tabindex="-1"></a>quarto-executable-code-5450563D</span>
<span id="cb36-341"><a href="#cb36-341" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-342"><a href="#cb36-342" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb36-343"><a href="#cb36-343" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: preproc-figures</span></span>
<span id="cb36-344"><a href="#cb36-344" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb36-345"><a href="#cb36-345" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 3</span></span>
<span id="cb36-346"><a href="#cb36-346" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 6</span></span>
<span id="cb36-347"><a href="#cb36-347" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-348"><a href="#cb36-348" aria-hidden="true" tabindex="-1"></a>g_stage_trace <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(basic_stats, <span class="fu">aes</span>(<span class="at">x=</span>stage, <span class="at">group=</span>sample)) <span class="sc">+</span></span>
<span id="cb36-349"><a href="#cb36-349" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb36-350"><a href="#cb36-350" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-351"><a href="#cb36-351" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot reads over preprocessing</span></span>
<span id="cb36-352"><a href="#cb36-352" aria-hidden="true" tabindex="-1"></a>g_reads_stages <span class="ot">&lt;-</span> g_stage_trace <span class="sc">+</span></span>
<span id="cb36-353"><a href="#cb36-353" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>n_read_pairs)) <span class="sc">+</span></span>
<span id="cb36-354"><a href="#cb36-354" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">&quot;# Read pairs&quot;</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>))</span>
<span id="cb36-355"><a href="#cb36-355" aria-hidden="true" tabindex="-1"></a>g_reads_stages</span>
<span id="cb36-356"><a href="#cb36-356" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-357"><a href="#cb36-357" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot relative read losses during preprocessing</span></span>
<span id="cb36-358"><a href="#cb36-358" aria-hidden="true" tabindex="-1"></a>g_reads_rel <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(n_reads_rel, </span>
<span id="cb36-359"><a href="#cb36-359" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">aes</span>(<span class="at">x=</span>stage, <span class="at">group=</span>sample)) <span class="sc">+</span></span>
<span id="cb36-360"><a href="#cb36-360" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>p_reads_lost_abs_marginal)) <span class="sc">+</span></span>
<span id="cb36-361"><a href="#cb36-361" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">&quot;% Total Reads Lost&quot;</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), </span>
<span id="cb36-362"><a href="#cb36-362" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="cf">function</span>(x) x<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb36-363"><a href="#cb36-363" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb36-364"><a href="#cb36-364" aria-hidden="true" tabindex="-1"></a>g_reads_rel</span>
<span id="cb36-365"><a href="#cb36-365" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb36-366"><a href="#cb36-366" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-367"><a href="#cb36-367" aria-hidden="true" tabindex="-1"></a>quarto-executable-code-5450563D</span>
<span id="cb36-368"><a href="#cb36-368" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-369"><a href="#cb36-369" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb36-370"><a href="#cb36-370" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: preproc-dedup</span></span>
<span id="cb36-371"><a href="#cb36-371" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb36-372"><a href="#cb36-372" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-373"><a href="#cb36-373" aria-hidden="true" tabindex="-1"></a>stage_dup <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(stage) <span class="sc">%&gt;%</span> </span>
<span id="cb36-374"><a href="#cb36-374" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">dmin =</span> <span class="fu">min</span>(percent_duplicates), <span class="at">dmax=</span><span class="fu">max</span>(percent_duplicates),</span>
<span id="cb36-375"><a href="#cb36-375" aria-hidden="true" tabindex="-1"></a>            <span class="at">dmean=</span><span class="fu">mean</span>(percent_duplicates), <span class="at">.groups =</span> <span class="st">&quot;drop&quot;</span>)</span>
<span id="cb36-376"><a href="#cb36-376" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-377"><a href="#cb36-377" aria-hidden="true" tabindex="-1"></a>g_dup_stages <span class="ot">&lt;-</span> g_stage_trace <span class="sc">+</span></span>
<span id="cb36-378"><a href="#cb36-378" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>percent_duplicates)) <span class="sc">+</span></span>
<span id="cb36-379"><a href="#cb36-379" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">&quot;% Duplicates&quot;</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb36-380"><a href="#cb36-380" aria-hidden="true" tabindex="-1"></a>g_dup_stages</span>
<span id="cb36-381"><a href="#cb36-381" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-382"><a href="#cb36-382" aria-hidden="true" tabindex="-1"></a>g_readlen_stages <span class="ot">&lt;-</span> g_stage_trace <span class="sc">+</span> <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>mean_seq_len)) <span class="sc">+</span></span>
<span id="cb36-383"><a href="#cb36-383" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">&quot;Mean read length (nt)&quot;</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>))</span>
<span id="cb36-384"><a href="#cb36-384" aria-hidden="true" tabindex="-1"></a>g_readlen_stages</span>
<span id="cb36-385"><a href="#cb36-385" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb36-386"><a href="#cb36-386" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-387"><a href="#cb36-387" aria-hidden="true" tabindex="-1"></a>quarto-executable-code-5450563D</span>
<span id="cb36-388"><a href="#cb36-388" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-389"><a href="#cb36-389" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb36-390"><a href="#cb36-390" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: ribo-frac</span></span>
<span id="cb36-391"><a href="#cb36-391" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb36-392"><a href="#cb36-392" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate reads lost during ribodepletion (approximation for % ribosomal reads)</span></span>
<span id="cb36-393"><a href="#cb36-393" aria-hidden="true" tabindex="-1"></a>reads_ribo <span class="ot">&lt;-</span> n_reads_rel <span class="sc">%&gt;%</span> </span>
<span id="cb36-394"><a href="#cb36-394" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(stage <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;dedup&quot;</span>, <span class="st">&quot;ribo_secondary&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb36-395"><a href="#cb36-395" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb36-396"><a href="#cb36-396" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads_ribo=</span><span class="dv">1</span><span class="sc">-</span>n_read_pairs[<span class="dv">2</span>]<span class="sc">/</span>n_read_pairs[<span class="dv">1</span>], <span class="at">.groups =</span> <span class="st">&quot;drop&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-397"><a href="#cb36-397" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by =</span> <span class="st">&#39;sample&#39;</span>)</span>
<span id="cb36-398"><a href="#cb36-398" aria-hidden="true" tabindex="-1"></a>reads_ribo_summ <span class="ot">&lt;-</span> reads_ribo <span class="sc">%&gt;%</span></span>
<span id="cb36-399"><a href="#cb36-399" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb36-400"><a href="#cb36-400" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">min=</span><span class="fu">min</span>(p_reads_ribo), <span class="at">max=</span><span class="fu">max</span>(p_reads_ribo),</span>
<span id="cb36-401"><a href="#cb36-401" aria-hidden="true" tabindex="-1"></a>            <span class="at">mean=</span><span class="fu">mean</span>(p_reads_ribo), <span class="at">.groups =</span> <span class="st">&quot;drop&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-402"><a href="#cb36-402" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by =</span> <span class="st">&#39;sample&#39;</span>)</span>
<span id="cb36-403"><a href="#cb36-403" aria-hidden="true" tabindex="-1"></a>g_reads_ribo <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(reads_ribo, </span>
<span id="cb36-404"><a href="#cb36-404" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">aes</span>(<span class="at">x=</span>library, <span class="at">y=</span>p_reads_ribo)) <span class="sc">+</span></span>
<span id="cb36-405"><a href="#cb36-405" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span> </span>
<span id="cb36-406"><a href="#cb36-406" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">&quot;Approx % ribosomal reads&quot;</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>),</span>
<span id="cb36-407"><a href="#cb36-407" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>)<span class="sc">+</span></span>
<span id="cb36-408"><a href="#cb36-408" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb36-409"><a href="#cb36-409" aria-hidden="true" tabindex="-1"></a>g_reads_ribo</span>
<span id="cb36-410"><a href="#cb36-410" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb36-411"><a href="#cb36-411" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-412"><a href="#cb36-412" aria-hidden="true" tabindex="-1"></a><span class="fu"># Taxonomic composition</span></span>
<span id="cb36-413"><a href="#cb36-413" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-414"><a href="#cb36-414" aria-hidden="true" tabindex="-1"></a><span class="fu">## High-level composition</span></span>
<span id="cb36-415"><a href="#cb36-415" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-416"><a href="#cb36-416" aria-hidden="true" tabindex="-1"></a>To assess the high-level composition of the reads, I ran the ribodepleted files through Kraken2 and summarized the results with Bracken.</span>
<span id="cb36-417"><a href="#cb36-417" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-418"><a href="#cb36-418" aria-hidden="true" tabindex="-1"></a>The groups listed below were created by Will:</span>
<span id="cb36-419"><a href="#cb36-419" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-420"><a href="#cb36-420" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Filtered (removed during cleaning)</span>
<span id="cb36-421"><a href="#cb36-421" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Duplicate (removed during deduplication)</span>
<span id="cb36-422"><a href="#cb36-422" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Ribosomal (removed during ribodepletion)</span>
<span id="cb36-423"><a href="#cb36-423" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Unassigned (non-ribosomal reads that were not assigned to any taxon by Kraken/Bracken)</span>
<span id="cb36-424"><a href="#cb36-424" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Bacterial (non-ribosomal reads assigned to the Bacteria domain by Kraken/Bracken)</span>
<span id="cb36-425"><a href="#cb36-425" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Archaeal (non-ribosomal reads assigned to the Archaea domain by Kraken/Bracken)</span>
<span id="cb36-426"><a href="#cb36-426" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Viral (non-ribosomal reads assigned to the Viruses domain by Kraken/Bracken)</span>
<span id="cb36-427"><a href="#cb36-427" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Human (non-ribosomal reads assigned to the Eukarya domain by Kraken/Bracken)</span>
<span id="cb36-428"><a href="#cb36-428" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-429"><a href="#cb36-429" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-430"><a href="#cb36-430" aria-hidden="true" tabindex="-1"></a>quarto-executable-code-5450563D</span>
<span id="cb36-431"><a href="#cb36-431" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-432"><a href="#cb36-432" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb36-433"><a href="#cb36-433" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: prepare-composition</span></span>
<span id="cb36-434"><a href="#cb36-434" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb36-435"><a href="#cb36-435" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-436"><a href="#cb36-436" aria-hidden="true" tabindex="-1"></a>classifications <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Filtered&quot;</span>, <span class="st">&quot;Duplicate&quot;</span>, <span class="st">&quot;Ribosomal&quot;</span>, <span class="st">&quot;Unassigned&quot;</span>,</span>
<span id="cb36-437"><a href="#cb36-437" aria-hidden="true" tabindex="-1"></a>                     <span class="st">&quot;Bacterial&quot;</span>, <span class="st">&quot;Archaeal&quot;</span>, <span class="st">&quot;Viral&quot;</span>, <span class="st">&quot;Human&quot;</span>)</span>
<span id="cb36-438"><a href="#cb36-438" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-439"><a href="#cb36-439" aria-hidden="true" tabindex="-1"></a><span class="co"># Import composition data</span></span>
<span id="cb36-440"><a href="#cb36-440" aria-hidden="true" tabindex="-1"></a>tax_final_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(results_dir, <span class="st">&quot;taxonomy_final&quot;</span>)</span>
<span id="cb36-441"><a href="#cb36-441" aria-hidden="true" tabindex="-1"></a>comp_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(tax_final_dir, <span class="st">&quot;taxonomic_composition.tsv.gz&quot;</span>)</span>
<span id="cb36-442"><a href="#cb36-442" aria-hidden="true" tabindex="-1"></a>comp <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(comp_path) <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(libraries) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(library <span class="sc">!=</span> <span class="st">&quot;PCL21&quot;</span>)</span>
<span id="cb36-443"><a href="#cb36-443" aria-hidden="true" tabindex="-1"></a>comp_minor <span class="ot">&lt;-</span> comp <span class="sc">%&gt;%</span> </span>
<span id="cb36-444"><a href="#cb36-444" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(classification <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;Archaeal&quot;</span>, <span class="st">&quot;Viral&quot;</span>, <span class="st">&quot;Human&quot;</span>, <span class="st">&quot;Other&quot;</span>))</span>
<span id="cb36-445"><a href="#cb36-445" aria-hidden="true" tabindex="-1"></a>comp_assigned <span class="ot">&lt;-</span> comp <span class="sc">%&gt;%</span></span>
<span id="cb36-446"><a href="#cb36-446" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span> classification <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;Filtered&quot;</span>, <span class="st">&quot;Duplicate&quot;</span>, </span>
<span id="cb36-447"><a href="#cb36-447" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">&quot;Ribosomal&quot;</span>, <span class="st">&quot;Unassigned&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb36-448"><a href="#cb36-448" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb36-449"><a href="#cb36-449" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads =</span> n_reads<span class="sc">/</span><span class="fu">sum</span>(n_reads))</span>
<span id="cb36-450"><a href="#cb36-450" aria-hidden="true" tabindex="-1"></a>comp_assigned_minor <span class="ot">&lt;-</span> comp_assigned <span class="sc">%&gt;%</span> </span>
<span id="cb36-451"><a href="#cb36-451" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(classification <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;Archaeal&quot;</span>, <span class="st">&quot;Viral&quot;</span>, <span class="st">&quot;Human&quot;</span>, <span class="st">&quot;Other&quot;</span>))</span>
<span id="cb36-452"><a href="#cb36-452" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-453"><a href="#cb36-453" aria-hidden="true" tabindex="-1"></a><span class="co"># Summarize composition</span></span>
<span id="cb36-454"><a href="#cb36-454" aria-hidden="true" tabindex="-1"></a>read_comp_summ <span class="ot">&lt;-</span> comp <span class="sc">%&gt;%</span> </span>
<span id="cb36-455"><a href="#cb36-455" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(classification) <span class="sc">%&gt;%</span></span>
<span id="cb36-456"><a href="#cb36-456" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads =</span> <span class="fu">sum</span>(n_reads), <span class="at">.groups =</span> <span class="st">&quot;drop_last&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-457"><a href="#cb36-457" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">n_reads =</span> <span class="fu">replace_na</span>(n_reads,<span class="dv">0</span>),</span>
<span id="cb36-458"><a href="#cb36-458" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads =</span> n_reads<span class="sc">/</span><span class="fu">sum</span>(n_reads),</span>
<span id="cb36-459"><a href="#cb36-459" aria-hidden="true" tabindex="-1"></a>         <span class="at">pc_reads =</span> p_reads<span class="sc">*</span><span class="dv">100</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-460"><a href="#cb36-460" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">classification =</span> <span class="fu">factor</span>(classification, <span class="at">levels=</span>classifications)) <span class="sc">%&gt;%</span></span>
<span id="cb36-461"><a href="#cb36-461" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(classification, n_reads, pc_reads) <span class="sc">%&gt;%</span></span>
<span id="cb36-462"><a href="#cb36-462" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">`</span><span class="at"># of reads</span><span class="st">`</span> <span class="ot">=</span> n_reads, <span class="st">&quot;% of reads&quot;</span> <span class="ot">=</span> pc_reads) <span class="sc">%&gt;%</span></span>
<span id="cb36-463"><a href="#cb36-463" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">% of reads</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sprintf</span>(<span class="st">&quot;%.2f&quot;</span>, <span class="st">`</span><span class="at">% of reads</span><span class="st">`</span>))</span>
<span id="cb36-464"><a href="#cb36-464" aria-hidden="true" tabindex="-1"></a>read_comp_summ</span>
<span id="cb36-465"><a href="#cb36-465" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-466"><a href="#cb36-466" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb36-467"><a href="#cb36-467" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-468"><a href="#cb36-468" aria-hidden="true" tabindex="-1"></a>quarto-executable-code-5450563D</span>
<span id="cb36-469"><a href="#cb36-469" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-470"><a href="#cb36-470" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb36-471"><a href="#cb36-471" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-composition-all</span></span>
<span id="cb36-472"><a href="#cb36-472" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 7</span></span>
<span id="cb36-473"><a href="#cb36-473" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 8</span></span>
<span id="cb36-474"><a href="#cb36-474" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-475"><a href="#cb36-475" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare plotting templates</span></span>
<span id="cb36-476"><a href="#cb36-476" aria-hidden="true" tabindex="-1"></a>g_comp_base <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">x=</span>library, <span class="at">y=</span>p_reads, <span class="at">fill=</span>classification)) <span class="sc">+</span></span>
<span id="cb36-477"><a href="#cb36-477" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">name=</span><span class="st">&quot;Plasma pool&quot;</span>) <span class="sc">+</span></span>
<span id="cb36-478"><a href="#cb36-478" aria-hidden="true" tabindex="-1"></a>  theme_kit <span class="sc">+</span> </span>
<span id="cb36-479"><a href="#cb36-479" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust=</span><span class="dv">0</span>, <span class="at">face=</span><span class="st">&quot;plain&quot;</span>, <span class="at">size=</span><span class="fu">rel</span>(<span class="fl">1.5</span>)))</span>
<span id="cb36-480"><a href="#cb36-480" aria-hidden="true" tabindex="-1"></a>scale_y_pc_reads <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">partial</span>(scale_y_continuous, <span class="at">name =</span> <span class="st">&quot;% Reads&quot;</span>,</span>
<span id="cb36-481"><a href="#cb36-481" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>)</span>
<span id="cb36-482"><a href="#cb36-482" aria-hidden="true" tabindex="-1"></a>geom_comp <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">partial</span>(geom_col, <span class="at">position =</span> <span class="st">&quot;stack&quot;</span>, <span class="at">width =</span> <span class="dv">1</span>)</span>
<span id="cb36-483"><a href="#cb36-483" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-484"><a href="#cb36-484" aria-hidden="true" tabindex="-1"></a><span class="co"># Define a color palette for the classification</span></span>
<span id="cb36-485"><a href="#cb36-485" aria-hidden="true" tabindex="-1"></a>classification_colors <span class="ot">&lt;-</span> <span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">&quot;Accent&quot;</span>)</span>
<span id="cb36-486"><a href="#cb36-486" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(classification_colors) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Filtered&quot;</span>, <span class="st">&quot;Duplicate&quot;</span>, <span class="st">&quot;Ribosomal&quot;</span>, <span class="st">&quot;Unassigned&quot;</span>, <span class="st">&quot;Bacterial&quot;</span>, <span class="st">&quot;Archaeal&quot;</span>, <span class="st">&quot;Viral&quot;</span>, <span class="st">&quot;Human&quot;</span>)</span>
<span id="cb36-487"><a href="#cb36-487" aria-hidden="true" tabindex="-1"></a>scale_fill_classification <span class="ot">&lt;-</span> <span class="cf">function</span>() {</span>
<span id="cb36-488"><a href="#cb36-488" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> classification_colors, <span class="at">name =</span> <span class="st">&quot;Classification&quot;</span>)</span>
<span id="cb36-489"><a href="#cb36-489" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb36-490"><a href="#cb36-490" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-491"><a href="#cb36-491" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot overall composition</span></span>
<span id="cb36-492"><a href="#cb36-492" aria-hidden="true" tabindex="-1"></a>g_comp <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> <span class="fu">geom_comp</span>(<span class="at">data =</span> comp) <span class="sc">+</span></span>
<span id="cb36-493"><a href="#cb36-493" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_pc_reads</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>)) <span class="sc">+</span></span>
<span id="cb36-494"><a href="#cb36-494" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_classification</span>() <span class="sc">+</span> </span>
<span id="cb36-495"><a href="#cb36-495" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Read composition (all reads, all groups)&quot;</span>)</span>
<span id="cb36-496"><a href="#cb36-496" aria-hidden="true" tabindex="-1"></a>g_comp</span>
<span id="cb36-497"><a href="#cb36-497" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-498"><a href="#cb36-498" aria-hidden="true" tabindex="-1"></a>g_comp_assigned <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb36-499"><a href="#cb36-499" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_comp</span>(<span class="at">data =</span> comp_assigned) <span class="sc">+</span></span>
<span id="cb36-500"><a href="#cb36-500" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_pc_reads</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>)) <span class="sc">+</span></span>
<span id="cb36-501"><a href="#cb36-501" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_classification</span>() <span class="sc">+</span> </span>
<span id="cb36-502"><a href="#cb36-502" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Read composition (assigned reads, all groups)&quot;</span>)</span>
<span id="cb36-503"><a href="#cb36-503" aria-hidden="true" tabindex="-1"></a>g_comp_assigned</span>
<span id="cb36-504"><a href="#cb36-504" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-505"><a href="#cb36-505" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb36-506"><a href="#cb36-506" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-507"><a href="#cb36-507" aria-hidden="true" tabindex="-1"></a><span class="fu">## Total viral content</span></span>
<span id="cb36-508"><a href="#cb36-508" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-509"><a href="#cb36-509" aria-hidden="true" tabindex="-1"></a>Total viral fraction average $2.36 \times 10^{-4}$ across samples. As a fraction of assigned (rather than total) reads, this jumped to $3.19 \times 10^{-3}$:</span>
<span id="cb36-510"><a href="#cb36-510" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-511"><a href="#cb36-511" aria-hidden="true" tabindex="-1"></a>quarto-executable-code-5450563D</span>
<span id="cb36-512"><a href="#cb36-512" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-513"><a href="#cb36-513" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb36-514"><a href="#cb36-514" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: p-viral</span></span>
<span id="cb36-515"><a href="#cb36-515" aria-hidden="true" tabindex="-1"></a>p_reads_viral_all <span class="ot">&lt;-</span> comp <span class="sc">%&gt;%</span> <span class="fu">filter</span>(classification <span class="sc">==</span> <span class="st">&quot;Viral&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-516"><a href="#cb36-516" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">read_group =</span> <span class="st">&quot;All reads&quot;</span>)</span>
<span id="cb36-517"><a href="#cb36-517" aria-hidden="true" tabindex="-1"></a>p_reads_viral_assigned <span class="ot">&lt;-</span> comp_assigned <span class="sc">%&gt;%</span> <span class="fu">filter</span>(classification <span class="sc">==</span> <span class="st">&quot;Viral&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-518"><a href="#cb36-518" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">read_group =</span> <span class="st">&quot;Classified reads&quot;</span>)</span>
<span id="cb36-519"><a href="#cb36-519" aria-hidden="true" tabindex="-1"></a>p_reads_viral <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(p_reads_viral_all, p_reads_viral_assigned)</span>
<span id="cb36-520"><a href="#cb36-520" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-521"><a href="#cb36-521" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb36-522"><a href="#cb36-522" aria-hidden="true" tabindex="-1"></a>g_viral <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(p_reads_viral, <span class="fu">aes</span>(<span class="at">x=</span>library, <span class="at">y=</span>p_reads, <span class="at">color =</span> read_group, <span class="at">group =</span> library)) <span class="sc">+</span></span>
<span id="cb36-523"><a href="#cb36-523" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb36-524"><a href="#cb36-524" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">color =</span> <span class="st">&#39;black&#39;</span>) <span class="sc">+</span></span>
<span id="cb36-525"><a href="#cb36-525" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">name=</span><span class="st">&quot;Plasma pool&quot;</span>) <span class="sc">+</span></span>
<span id="cb36-526"><a href="#cb36-526" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>(<span class="at">name=</span><span class="st">&quot;Viral read fraction&quot;</span>, <span class="at">labels =</span> <span class="fu">label_log</span>(<span class="at">digits=</span><span class="dv">2</span>), <span class="at">limits =</span> <span class="fu">c</span>(<span class="fl">1e-6</span>, <span class="fl">1e-2</span>)) <span class="sc">+</span></span>
<span id="cb36-527"><a href="#cb36-527" aria-hidden="true" tabindex="-1"></a>  theme_kit <span class="sc">+</span> </span>
<span id="cb36-528"><a href="#cb36-528" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>()</span>
<span id="cb36-529"><a href="#cb36-529" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-530"><a href="#cb36-530" aria-hidden="true" tabindex="-1"></a>g_viral</span>
<span id="cb36-531"><a href="#cb36-531" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-532"><a href="#cb36-532" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb36-533"><a href="#cb36-533" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-534"><a href="#cb36-534" aria-hidden="true" tabindex="-1"></a><span class="fu">## Taxonomic composition of viruses</span></span>
<span id="cb36-535"><a href="#cb36-535" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-536"><a href="#cb36-536" aria-hidden="true" tabindex="-1"></a>The two dominant viruses we see are Anellovirdae and Phycodnaviridae. The threshold for the label &quot;other&quot; are the set of families that make up less than 5% composition in all samples.</span>
<span id="cb36-537"><a href="#cb36-537" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-538"><a href="#cb36-538" aria-hidden="true" tabindex="-1"></a>quarto-executable-code-5450563D</span>
<span id="cb36-539"><a href="#cb36-539" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-540"><a href="#cb36-540" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb36-541"><a href="#cb36-541" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: extract-viral-taxa</span></span>
<span id="cb36-542"><a href="#cb36-542" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb36-543"><a href="#cb36-543" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-544"><a href="#cb36-544" aria-hidden="true" tabindex="-1"></a><span class="co"># Get viral taxonomy</span></span>
<span id="cb36-545"><a href="#cb36-545" aria-hidden="true" tabindex="-1"></a>viral_taxa_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">&quot;total-virus-db.tsv.gz&quot;</span>)</span>
<span id="cb36-546"><a href="#cb36-546" aria-hidden="true" tabindex="-1"></a>viral_taxa <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(viral_taxa_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb36-547"><a href="#cb36-547" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-548"><a href="#cb36-548" aria-hidden="true" tabindex="-1"></a><span class="co"># Get Kraken reports</span></span>
<span id="cb36-549"><a href="#cb36-549" aria-hidden="true" tabindex="-1"></a>reports_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(tax_final_dir, <span class="st">&quot;kraken_reports.tsv.gz&quot;</span>)</span>
<span id="cb36-550"><a href="#cb36-550" aria-hidden="true" tabindex="-1"></a>reports <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(reports_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-551"><a href="#cb36-551" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">&quot;sample&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(sample)</span>
<span id="cb36-552"><a href="#cb36-552" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-553"><a href="#cb36-553" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter to viral taxa</span></span>
<span id="cb36-554"><a href="#cb36-554" aria-hidden="true" tabindex="-1"></a>kraken_reports_viral <span class="ot">&lt;-</span> <span class="fu">filter</span>(reports, taxid <span class="sc">%in%</span> viral_taxa<span class="sc">$</span>taxid) <span class="sc">%&gt;%</span></span>
<span id="cb36-555"><a href="#cb36-555" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb36-556"><a href="#cb36-556" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_viral =</span> n_reads_clade<span class="sc">/</span>n_reads_clade[<span class="dv">1</span>])</span>
<span id="cb36-557"><a href="#cb36-557" aria-hidden="true" tabindex="-1"></a>kraken_reports_viral_cleaned <span class="ot">&lt;-</span> kraken_reports_viral <span class="sc">%&gt;%</span></span>
<span id="cb36-558"><a href="#cb36-558" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>pc_reads_total, <span class="sc">-</span>n_reads_direct, <span class="sc">-</span><span class="fu">contains</span>(<span class="st">&quot;minimizers&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb36-559"><a href="#cb36-559" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, taxid, p_reads_viral, n_reads_clade, <span class="fu">everything</span>()) <span class="sc">%&gt;%</span> ungroup</span>
<span id="cb36-560"><a href="#cb36-560" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-561"><a href="#cb36-561" aria-hidden="true" tabindex="-1"></a>viral_classes <span class="ot">&lt;-</span> kraken_reports_viral_cleaned <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">==</span> <span class="st">&quot;C&quot;</span>)</span>
<span id="cb36-562"><a href="#cb36-562" aria-hidden="true" tabindex="-1"></a>viral_families <span class="ot">&lt;-</span> kraken_reports_viral_cleaned <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">==</span> <span class="st">&quot;F&quot;</span>)</span>
<span id="cb36-563"><a href="#cb36-563" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-564"><a href="#cb36-564" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb36-565"><a href="#cb36-565" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-566"><a href="#cb36-566" aria-hidden="true" tabindex="-1"></a>quarto-executable-code-5450563D</span>
<span id="cb36-567"><a href="#cb36-567" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-568"><a href="#cb36-568" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb36-569"><a href="#cb36-569" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: viral-family-composition</span></span>
<span id="cb36-570"><a href="#cb36-570" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb36-571"><a href="#cb36-571" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-572"><a href="#cb36-572" aria-hidden="true" tabindex="-1"></a>major_threshold <span class="ot">&lt;-</span> <span class="fl">0.05</span></span>
<span id="cb36-573"><a href="#cb36-573" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-574"><a href="#cb36-574" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify major viral families</span></span>
<span id="cb36-575"><a href="#cb36-575" aria-hidden="true" tabindex="-1"></a>viral_families_major_tab <span class="ot">&lt;-</span> viral_families <span class="sc">%&gt;%</span> </span>
<span id="cb36-576"><a href="#cb36-576" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb36-577"><a href="#cb36-577" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads_viral_max =</span> <span class="fu">max</span>(p_reads_viral), <span class="at">.groups=</span><span class="st">&quot;drop&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-578"><a href="#cb36-578" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_viral_max <span class="sc">&gt;=</span> major_threshold)</span>
<span id="cb36-579"><a href="#cb36-579" aria-hidden="true" tabindex="-1"></a>viral_families_major_list <span class="ot">&lt;-</span> viral_families_major_tab <span class="sc">%&gt;%</span> <span class="fu">pull</span>(name)</span>
<span id="cb36-580"><a href="#cb36-580" aria-hidden="true" tabindex="-1"></a>viral_families_major <span class="ot">&lt;-</span> viral_families <span class="sc">%&gt;%</span> </span>
<span id="cb36-581"><a href="#cb36-581" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(name <span class="sc">%in%</span> viral_families_major_list) <span class="sc">%&gt;%</span></span>
<span id="cb36-582"><a href="#cb36-582" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, taxid, sample, p_reads_viral)</span>
<span id="cb36-583"><a href="#cb36-583" aria-hidden="true" tabindex="-1"></a>viral_families_minor <span class="ot">&lt;-</span> viral_families_major <span class="sc">%&gt;%</span> </span>
<span id="cb36-584"><a href="#cb36-584" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb36-585"><a href="#cb36-585" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads_viral_major =</span> <span class="fu">sum</span>(p_reads_viral), <span class="at">.groups =</span> <span class="st">&quot;drop&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-586"><a href="#cb36-586" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="st">&quot;Other&quot;</span>, <span class="at">taxid=</span><span class="cn">NA</span>, <span class="at">p_reads_viral =</span> <span class="dv">1</span><span class="sc">-</span>p_reads_viral_major) <span class="sc">%&gt;%</span></span>
<span id="cb36-587"><a href="#cb36-587" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, taxid, sample, p_reads_viral)</span>
<span id="cb36-588"><a href="#cb36-588" aria-hidden="true" tabindex="-1"></a>viral_families_display <span class="ot">&lt;-</span> viral_families_major <span class="sc">%&gt;%</span> </span>
<span id="cb36-589"><a href="#cb36-589" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_rows</span>(viral_families_minor) <span class="sc">%&gt;%</span></span>
<span id="cb36-590"><a href="#cb36-590" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_viral)) <span class="sc">%&gt;%</span> </span>
<span id="cb36-591"><a href="#cb36-591" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">factor</span>(name, <span class="at">levels=</span><span class="fu">c</span>(viral_families_major_list, <span class="st">&quot;Other&quot;</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb36-592"><a href="#cb36-592" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_viral, <span class="at">classification=</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb36-593"><a href="#cb36-593" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">&#39;sample&#39;</span>)</span>
<span id="cb36-594"><a href="#cb36-594" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-595"><a href="#cb36-595" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb36-596"><a href="#cb36-596" aria-hidden="true" tabindex="-1"></a>g_families <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb36-597"><a href="#cb36-597" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_comp</span>(<span class="at">data=</span>viral_families_display) <span class="sc">+</span></span>
<span id="cb36-598"><a href="#cb36-598" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">&quot;% Viral Reads&quot;</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb36-599"><a href="#cb36-599" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb36-600"><a href="#cb36-600" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb36-601"><a href="#cb36-601" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">&#39;Accent&#39;</span>)</span>
<span id="cb36-602"><a href="#cb36-602" aria-hidden="true" tabindex="-1"></a>g_families</span>
<span id="cb36-603"><a href="#cb36-603" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb36-604"><a href="#cb36-604" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-605"><a href="#cb36-605" aria-hidden="true" tabindex="-1"></a><span class="fu"># Human-infecting virus reads</span></span>
<span id="cb36-606"><a href="#cb36-606" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-607"><a href="#cb36-607" aria-hidden="true" tabindex="-1"></a><span class="fu">## Overall relative abundance</span></span>
<span id="cb36-608"><a href="#cb36-608" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-609"><a href="#cb36-609" aria-hidden="true" tabindex="-1"></a>I calculated the relative abundance of human-infecting viruses in two ways:</span>
<span id="cb36-610"><a href="#cb36-610" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-611"><a href="#cb36-611" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>First, as the total number of deduplicated human-virus reads in each sample, divided by the number of raw reads (&quot;All reads&quot;). This results in a viral fraction of $1.64 \times 10^{-4}$ across all samples</span>
<span id="cb36-612"><a href="#cb36-612" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-613"><a href="#cb36-613" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Second, as a fraction of preprocessed (cleaned, deduplicated, computationally ribodepleted) reads (&quot;Preprocessed reads&quot;). This results in a viral fraction of $2.42 \times 10^{-4}$ across all samples.</span>
<span id="cb36-614"><a href="#cb36-614" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-615"><a href="#cb36-615" aria-hidden="true" tabindex="-1"></a>quarto-executable-code-5450563D</span>
<span id="cb36-616"><a href="#cb36-616" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-617"><a href="#cb36-617" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb36-618"><a href="#cb36-618" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: prepare-hv</span></span>
<span id="cb36-619"><a href="#cb36-619" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb36-620"><a href="#cb36-620" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-621"><a href="#cb36-621" aria-hidden="true" tabindex="-1"></a><span class="co"># Import and format reads</span></span>
<span id="cb36-622"><a href="#cb36-622" aria-hidden="true" tabindex="-1"></a>hv_reads_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(hv_dir, <span class="st">&quot;hv_hits_putative_collapsed.tsv.gz&quot;</span>)</span>
<span id="cb36-623"><a href="#cb36-623" aria-hidden="true" tabindex="-1"></a>mrg_hv <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(hv_reads_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-624"><a href="#cb36-624" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">&quot;sample&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb36-625"><a href="#cb36-625" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">kraken_label =</span> <span class="fu">ifelse</span>(assigned_hv, <span class="st">&quot;Kraken2 HV assignment&quot;</span>,</span>
<span id="cb36-626"><a href="#cb36-626" aria-hidden="true" tabindex="-1"></a>                               <span class="st">&quot;No Kraken2 assignment&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb36-627"><a href="#cb36-627" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">adj_score_max =</span> <span class="fu">pmax</span>(adj_score_fwd, adj_score_rev),</span>
<span id="cb36-628"><a href="#cb36-628" aria-hidden="true" tabindex="-1"></a>         <span class="at">highscore =</span> adj_score_max <span class="sc">&gt;=</span> <span class="dv">20</span>,</span>
<span id="cb36-629"><a href="#cb36-629" aria-hidden="true" tabindex="-1"></a>         <span class="at">hv_status =</span> assigned_hv <span class="sc">|</span> highscore) <span class="sc">%&gt;%</span></span>
<span id="cb36-630"><a href="#cb36-630" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">taxid_all =</span> taxid, <span class="at">taxid =</span> taxid_best)</span>
<span id="cb36-631"><a href="#cb36-631" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb36-632"><a href="#cb36-632" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-633"><a href="#cb36-633" aria-hidden="true" tabindex="-1"></a>quarto-executable-code-5450563D</span>
<span id="cb36-634"><a href="#cb36-634" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-635"><a href="#cb36-635" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb36-636"><a href="#cb36-636" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: count-hv-reads</span></span>
<span id="cb36-637"><a href="#cb36-637" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb36-638"><a href="#cb36-638" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb36-639"><a href="#cb36-639" aria-hidden="true" tabindex="-1"></a><span class="co"># Get raw read counts</span></span>
<span id="cb36-640"><a href="#cb36-640" aria-hidden="true" tabindex="-1"></a>read_counts_raw <span class="ot">&lt;-</span> <span class="fu">filter</span>(basic_stats_raw) <span class="sc">%&gt;%</span></span>
<span id="cb36-641"><a href="#cb36-641" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, <span class="at">n_reads_raw =</span> n_read_pairs)</span>
<span id="cb36-642"><a href="#cb36-642" aria-hidden="true" tabindex="-1"></a>read_counts_preproc <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">&quot;ribo_initial&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-643"><a href="#cb36-643" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, <span class="at">n_reads_preproc =</span> n_read_pairs)</span>
<span id="cb36-644"><a href="#cb36-644" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-645"><a href="#cb36-645" aria-hidden="true" tabindex="-1"></a><span class="co"># Get HV read counts</span></span>
<span id="cb36-646"><a href="#cb36-646" aria-hidden="true" tabindex="-1"></a>read_counts_hv <span class="ot">&lt;-</span> mrg_hv <span class="sc">%&gt;%</span> <span class="fu">filter</span>(hv_status) <span class="sc">%&gt;%</span> </span>
<span id="cb36-647"><a href="#cb36-647" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb36-648"><a href="#cb36-648" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name=</span><span class="st">&quot;n_reads_hv&quot;</span>)</span>
<span id="cb36-649"><a href="#cb36-649" aria-hidden="true" tabindex="-1"></a>read_counts <span class="ot">&lt;-</span> read_counts_raw <span class="sc">%&gt;%</span></span>
<span id="cb36-650"><a href="#cb36-650" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(read_counts_hv, <span class="at">by=</span><span class="fu">c</span>(<span class="st">&quot;sample&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb36-651"><a href="#cb36-651" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">n_reads_hv =</span> <span class="fu">replace_na</span>(n_reads_hv, <span class="dv">0</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb36-652"><a href="#cb36-652" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(read_counts_preproc, <span class="at">by=</span><span class="fu">c</span>(<span class="st">&quot;sample&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb36-653"><a href="#cb36-653" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="fu">c</span>(<span class="st">&quot;sample&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb36-654"><a href="#cb36-654" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, n_reads_raw, n_reads_preproc, n_reads_hv) <span class="sc">%&gt;%</span></span>
<span id="cb36-655"><a href="#cb36-655" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">n_samples =</span> <span class="dv">1</span>,</span>
<span id="cb36-656"><a href="#cb36-656" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_total =</span> n_reads_hv<span class="sc">/</span>n_reads_raw,</span>
<span id="cb36-657"><a href="#cb36-657" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_preproc =</span> n_reads_hv<span class="sc">/</span>n_reads_preproc)</span>
<span id="cb36-658"><a href="#cb36-658" aria-hidden="true" tabindex="-1"></a>read_counts_long <span class="ot">&lt;-</span> read_counts <span class="sc">%&gt;%</span></span>
<span id="cb36-659"><a href="#cb36-659" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="fu">starts_with</span>(<span class="st">&quot;p_reads&quot;</span>), <span class="at">names_to=</span><span class="st">&quot;read_group&quot;</span>, <span class="at">values_to=</span><span class="st">&quot;p_reads&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-660"><a href="#cb36-660" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">read_group =</span> <span class="fu">ifelse</span>(read_group <span class="sc">==</span> <span class="st">&quot;p_reads_total&quot;</span>, <span class="st">&quot;All reads&quot;</span>, <span class="st">&quot;Preprocessed reads&quot;</span>))</span>
<span id="cb36-661"><a href="#cb36-661" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-662"><a href="#cb36-662" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine for display</span></span>
<span id="cb36-663"><a href="#cb36-663" aria-hidden="true" tabindex="-1"></a>read_counts_agg <span class="ot">&lt;-</span> read_counts <span class="sc">%&gt;%</span></span>
<span id="cb36-664"><a href="#cb36-664" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_total =</span> n_reads_hv<span class="sc">/</span>n_reads_raw,</span>
<span id="cb36-665"><a href="#cb36-665" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_preproc =</span> n_reads_hv<span class="sc">/</span>n_reads_preproc) <span class="sc">%&gt;%</span></span>
<span id="cb36-666"><a href="#cb36-666" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="fu">c</span>(<span class="st">&quot;sample&quot;</span>))</span>
<span id="cb36-667"><a href="#cb36-667" aria-hidden="true" tabindex="-1"></a>read_counts_agg_long <span class="ot">&lt;-</span> read_counts_agg <span class="sc">%&gt;%</span></span>
<span id="cb36-668"><a href="#cb36-668" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="fu">starts_with</span>(<span class="st">&quot;p_reads&quot;</span>), <span class="at">names_to=</span><span class="st">&quot;read_group&quot;</span>, <span class="at">values_to=</span><span class="st">&quot;p_reads&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-669"><a href="#cb36-669" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">read_group =</span> <span class="fu">ifelse</span>(read_group <span class="sc">==</span> <span class="st">&quot;p_reads_total&quot;</span>, <span class="st">&quot;All reads (HV)&quot;</span>, <span class="st">&quot;Preprocessed reads (HV)&quot;</span>)) </span>
<span id="cb36-670"><a href="#cb36-670" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-671"><a href="#cb36-671" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize</span></span>
<span id="cb36-672"><a href="#cb36-672" aria-hidden="true" tabindex="-1"></a>g_read_counts <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(read_counts_agg_long, <span class="fu">aes</span>(<span class="at">x=</span>library, <span class="at">y=</span>p_reads, <span class="at">color =</span> read_group, <span class="at">group =</span> library)) <span class="sc">+</span></span>
<span id="cb36-673"><a href="#cb36-673" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb36-674"><a href="#cb36-674" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">color =</span> <span class="st">&#39;black&#39;</span>) <span class="sc">+</span></span>
<span id="cb36-675"><a href="#cb36-675" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>(<span class="at">name =</span> <span class="st">&quot;Unique human-viral read fraction&quot;</span>, <span class="at">labels =</span> <span class="fu">label_log</span>(<span class="at">digits=</span><span class="dv">2</span>), <span class="at">limits =</span> <span class="fu">c</span>(<span class="fl">1e-6</span>, <span class="fl">1e-2</span>)) <span class="sc">+</span></span>
<span id="cb36-676"><a href="#cb36-676" aria-hidden="true" tabindex="-1"></a>  theme_kit <span class="sc">+</span> </span>
<span id="cb36-677"><a href="#cb36-677" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>()</span>
<span id="cb36-678"><a href="#cb36-678" aria-hidden="true" tabindex="-1"></a>g_read_counts</span>
<span id="cb36-679"><a href="#cb36-679" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-680"><a href="#cb36-680" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb36-681"><a href="#cb36-681" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-682"><a href="#cb36-682" aria-hidden="true" tabindex="-1"></a>quarto-executable-code-5450563D</span>
<span id="cb36-683"><a href="#cb36-683" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-684"><a href="#cb36-684" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb36-685"><a href="#cb36-685" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: compare-virus-to-hv</span></span>
<span id="cb36-686"><a href="#cb36-686" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb36-687"><a href="#cb36-687" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb36-688"><a href="#cb36-688" aria-hidden="true" tabindex="-1"></a>combine_viruses <span class="ot">&lt;-</span> <span class="fu">rbind</span>(p_reads_viral  <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(read_group) <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(<span class="at">mean=</span><span class="fu">mean</span>(p_reads)) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">type =</span> <span class="st">&quot;Viral reads&quot;</span>), read_counts_agg_long  <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(read_group) <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(<span class="at">mean=</span><span class="fu">mean</span>(p_reads))<span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">type =</span> <span class="st">&quot;HV reads&quot;</span>))</span>
<span id="cb36-689"><a href="#cb36-689" aria-hidden="true" tabindex="-1"></a><span class="co"># Format the mean values in scientific notation</span></span>
<span id="cb36-690"><a href="#cb36-690" aria-hidden="true" tabindex="-1"></a>combine_viruses <span class="ot">&lt;-</span> combine_viruses <span class="sc">%&gt;%</span></span>
<span id="cb36-691"><a href="#cb36-691" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mean_formatted =</span> <span class="fu">sprintf</span>(<span class="st">&quot;%.2e&quot;</span>, mean))</span>
<span id="cb36-692"><a href="#cb36-692" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-693"><a href="#cb36-693" aria-hidden="true" tabindex="-1"></a>combine_viruses <span class="sc">%&gt;%</span> <span class="fu">select</span>(type, read_group, <span class="at">mean=</span>mean_formatted)</span>
<span id="cb36-694"><a href="#cb36-694" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-695"><a href="#cb36-695" aria-hidden="true" tabindex="-1"></a><span class="co">#g_read_counts +</span></span>
<span id="cb36-696"><a href="#cb36-696" aria-hidden="true" tabindex="-1"></a><span class="co">#  geom_point(data = p_reads_viral, aes(x = library, y = p_reads, color = read_group), size = 5) +</span></span>
<span id="cb36-697"><a href="#cb36-697" aria-hidden="true" tabindex="-1"></a><span class="co">#  geom_line(data = p_reads_viral, aes(x = library, y = p_reads, group = library), color = &#39;grey&#39;) +</span></span>
<span id="cb36-698"><a href="#cb36-698" aria-hidden="true" tabindex="-1"></a><span class="co">#  scale_color_brewer(palette = &#39;Accent&#39;)</span></span>
<span id="cb36-699"><a href="#cb36-699" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-700"><a href="#cb36-700" aria-hidden="true" tabindex="-1"></a><span class="co">#ggarrange(g_viral, g_read_counts, ncol=2)</span></span>
<span id="cb36-701"><a href="#cb36-701" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb36-702"><a href="#cb36-702" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-703"><a href="#cb36-703" aria-hidden="true" tabindex="-1"></a><span class="fu">## Overall taxonomy and composition</span></span>
<span id="cb36-704"><a href="#cb36-704" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-705"><a href="#cb36-705" aria-hidden="true" tabindex="-1"></a>Composition of HV reads was changed from when looking at all viral reads. The two dominant viruses we see are Anellovirdae and Microviridae (bacteriophage). The threshold for the label &quot;other&quot; are the set of families that make up less than 5% composition in all samples.</span>
<span id="cb36-706"><a href="#cb36-706" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-707"><a href="#cb36-707" aria-hidden="true" tabindex="-1"></a>quarto-executable-code-5450563D</span>
<span id="cb36-708"><a href="#cb36-708" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-709"><a href="#cb36-709" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb36-710"><a href="#cb36-710" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: raise-hv-taxa</span></span>
<span id="cb36-711"><a href="#cb36-711" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb36-712"><a href="#cb36-712" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-713"><a href="#cb36-713" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter samples and add viral taxa information</span></span>
<span id="cb36-714"><a href="#cb36-714" aria-hidden="true" tabindex="-1"></a>samples_keep <span class="ot">&lt;-</span> read_counts <span class="sc">%&gt;%</span> <span class="fu">filter</span>(n_reads_hv <span class="sc">&gt;</span> <span class="dv">5</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(sample)</span>
<span id="cb36-715"><a href="#cb36-715" aria-hidden="true" tabindex="-1"></a>mrg_hv_named <span class="ot">&lt;-</span> mrg_hv <span class="sc">%&gt;%</span> <span class="fu">filter</span>(sample <span class="sc">%in%</span> samples_keep, hv_status) <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(viral_taxa, <span class="at">by=</span><span class="st">&quot;taxid&quot;</span>) </span>
<span id="cb36-716"><a href="#cb36-716" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-717"><a href="#cb36-717" aria-hidden="true" tabindex="-1"></a><span class="co"># Discover viral species &amp; genera for HV reads</span></span>
<span id="cb36-718"><a href="#cb36-718" aria-hidden="true" tabindex="-1"></a>raise_rank <span class="ot">&lt;-</span> <span class="cf">function</span>(read_db, taxid_db, <span class="at">out_rank =</span> <span class="st">&quot;species&quot;</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>){</span>
<span id="cb36-719"><a href="#cb36-719" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get higher ranks than search rank</span></span>
<span id="cb36-720"><a href="#cb36-720" aria-hidden="true" tabindex="-1"></a>  ranks <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;subspecies&quot;</span>, <span class="st">&quot;species&quot;</span>, <span class="st">&quot;subgenus&quot;</span>, <span class="st">&quot;genus&quot;</span>, <span class="st">&quot;subfamily&quot;</span>, <span class="st">&quot;family&quot;</span>, <span class="st">&quot;suborder&quot;</span>, <span class="st">&quot;order&quot;</span>, <span class="st">&quot;class&quot;</span>, <span class="st">&quot;subphylum&quot;</span>, <span class="st">&quot;phylum&quot;</span>, <span class="st">&quot;kingdom&quot;</span>, <span class="st">&quot;superkingdom&quot;</span>)</span>
<span id="cb36-721"><a href="#cb36-721" aria-hidden="true" tabindex="-1"></a>  rank_match <span class="ot">&lt;-</span> <span class="fu">which.max</span>(ranks <span class="sc">==</span> out_rank)</span>
<span id="cb36-722"><a href="#cb36-722" aria-hidden="true" tabindex="-1"></a>  high_ranks <span class="ot">&lt;-</span> ranks[rank_match<span class="sc">:</span><span class="fu">length</span>(ranks)]</span>
<span id="cb36-723"><a href="#cb36-723" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Merge read DB and taxid DB</span></span>
<span id="cb36-724"><a href="#cb36-724" aria-hidden="true" tabindex="-1"></a>  reads <span class="ot">&lt;-</span> read_db <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>parent_taxid, <span class="sc">-</span>rank, <span class="sc">-</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb36-725"><a href="#cb36-725" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(taxid_db, <span class="at">by=</span><span class="st">&quot;taxid&quot;</span>)</span>
<span id="cb36-726"><a href="#cb36-726" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract sequences that are already at appropriate rank</span></span>
<span id="cb36-727"><a href="#cb36-727" aria-hidden="true" tabindex="-1"></a>  reads_rank <span class="ot">&lt;-</span> <span class="fu">filter</span>(reads, rank <span class="sc">==</span> out_rank)</span>
<span id="cb36-728"><a href="#cb36-728" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Drop sequences at a higher rank and return unclassified sequences</span></span>
<span id="cb36-729"><a href="#cb36-729" aria-hidden="true" tabindex="-1"></a>  reads_norank <span class="ot">&lt;-</span> reads <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">!=</span> out_rank, <span class="sc">!</span>rank <span class="sc">%in%</span> high_ranks, <span class="sc">!</span><span class="fu">is.na</span>(taxid))</span>
<span id="cb36-730"><a href="#cb36-730" aria-hidden="true" tabindex="-1"></a>  <span class="cf">while</span>(<span class="fu">nrow</span>(reads_norank) <span class="sc">&gt;</span> <span class="dv">0</span>){ <span class="co"># As long as there are unclassified sequences...</span></span>
<span id="cb36-731"><a href="#cb36-731" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Promote read taxids and re-merge with taxid DB, then re-classify and filter</span></span>
<span id="cb36-732"><a href="#cb36-732" aria-hidden="true" tabindex="-1"></a>    reads_remaining <span class="ot">&lt;-</span> reads_norank <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">taxid =</span> parent_taxid) <span class="sc">%&gt;%</span></span>
<span id="cb36-733"><a href="#cb36-733" aria-hidden="true" tabindex="-1"></a>      <span class="fu">select</span>(<span class="sc">-</span>parent_taxid, <span class="sc">-</span>rank, <span class="sc">-</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb36-734"><a href="#cb36-734" aria-hidden="true" tabindex="-1"></a>      <span class="fu">left_join</span>(taxid_db, <span class="at">by=</span><span class="st">&quot;taxid&quot;</span>)</span>
<span id="cb36-735"><a href="#cb36-735" aria-hidden="true" tabindex="-1"></a>    reads_rank <span class="ot">&lt;-</span> reads_remaining <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">==</span> out_rank) <span class="sc">%&gt;%</span></span>
<span id="cb36-736"><a href="#cb36-736" aria-hidden="true" tabindex="-1"></a>      <span class="fu">bind_rows</span>(reads_rank)</span>
<span id="cb36-737"><a href="#cb36-737" aria-hidden="true" tabindex="-1"></a>    reads_norank <span class="ot">&lt;-</span> reads_remaining <span class="sc">%&gt;%</span></span>
<span id="cb36-738"><a href="#cb36-738" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(rank <span class="sc">!=</span> out_rank, <span class="sc">!</span>rank <span class="sc">%in%</span> high_ranks, <span class="sc">!</span><span class="fu">is.na</span>(taxid))</span>
<span id="cb36-739"><a href="#cb36-739" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb36-740"><a href="#cb36-740" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Finally, extract and append reads that were excluded during the process</span></span>
<span id="cb36-741"><a href="#cb36-741" aria-hidden="true" tabindex="-1"></a>  reads_dropped <span class="ot">&lt;-</span> reads <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span>seq_id <span class="sc">%in%</span> reads_rank<span class="sc">$</span>seq_id)</span>
<span id="cb36-742"><a href="#cb36-742" aria-hidden="true" tabindex="-1"></a>  reads_out <span class="ot">&lt;-</span> reads_rank <span class="sc">%&gt;%</span> <span class="fu">bind_rows</span>(reads_dropped) <span class="sc">%&gt;%</span></span>
<span id="cb36-743"><a href="#cb36-743" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(<span class="sc">-</span>parent_taxid, <span class="sc">-</span>rank, <span class="sc">-</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb36-744"><a href="#cb36-744" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(taxid_db, <span class="at">by=</span><span class="st">&quot;taxid&quot;</span>)</span>
<span id="cb36-745"><a href="#cb36-745" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(reads_out)</span>
<span id="cb36-746"><a href="#cb36-746" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb36-747"><a href="#cb36-747" aria-hidden="true" tabindex="-1"></a>hv_reads_species <span class="ot">&lt;-</span> <span class="fu">raise_rank</span>(mrg_hv_named, viral_taxa, <span class="st">&quot;species&quot;</span>)</span>
<span id="cb36-748"><a href="#cb36-748" aria-hidden="true" tabindex="-1"></a>hv_reads_genus <span class="ot">&lt;-</span> <span class="fu">raise_rank</span>(mrg_hv_named, viral_taxa, <span class="st">&quot;genus&quot;</span>)</span>
<span id="cb36-749"><a href="#cb36-749" aria-hidden="true" tabindex="-1"></a>hv_reads_family <span class="ot">&lt;-</span> <span class="fu">raise_rank</span>(mrg_hv_named, viral_taxa, <span class="st">&quot;family&quot;</span>)</span>
<span id="cb36-750"><a href="#cb36-750" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb36-751"><a href="#cb36-751" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-752"><a href="#cb36-752" aria-hidden="true" tabindex="-1"></a>quarto-executable-code-5450563D</span>
<span id="cb36-753"><a href="#cb36-753" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-754"><a href="#cb36-754" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb36-755"><a href="#cb36-755" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-family</span></span>
<span id="cb36-756"><a href="#cb36-756" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb36-757"><a href="#cb36-757" aria-hidden="true" tabindex="-1"></a>threshold_major_family <span class="ot">&lt;-</span> <span class="fl">0.05</span></span>
<span id="cb36-758"><a href="#cb36-758" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-759"><a href="#cb36-759" aria-hidden="true" tabindex="-1"></a><span class="co"># Count reads for each human-viral family</span></span>
<span id="cb36-760"><a href="#cb36-760" aria-hidden="true" tabindex="-1"></a>hv_family_counts <span class="ot">&lt;-</span> hv_reads_family <span class="sc">%&gt;%</span> </span>
<span id="cb36-761"><a href="#cb36-761" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb36-762"><a href="#cb36-762" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">&quot;n_reads_hv&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-763"><a href="#cb36-763" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb36-764"><a href="#cb36-764" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_hv =</span> n_reads_hv<span class="sc">/</span><span class="fu">sum</span>(n_reads_hv))</span>
<span id="cb36-765"><a href="#cb36-765" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-766"><a href="#cb36-766" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify high-ranking families and group others</span></span>
<span id="cb36-767"><a href="#cb36-767" aria-hidden="true" tabindex="-1"></a>hv_family_major_tab <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span> </span>
<span id="cb36-768"><a href="#cb36-768" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_hv <span class="sc">==</span> <span class="fu">max</span>(p_reads_hv)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-769"><a href="#cb36-769" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_hv)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(p_reads_hv <span class="sc">&gt;</span> threshold_major_family)</span>
<span id="cb36-770"><a href="#cb36-770" aria-hidden="true" tabindex="-1"></a>hv_family_counts_major <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span></span>
<span id="cb36-771"><a href="#cb36-771" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">ifelse</span>(name <span class="sc">%in%</span> hv_family_major_tab<span class="sc">$</span>name, name, <span class="st">&quot;Other&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb36-772"><a href="#cb36-772" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, name_display) <span class="sc">%&gt;%</span></span>
<span id="cb36-773"><a href="#cb36-773" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_hv =</span> <span class="fu">sum</span>(n_reads_hv), <span class="at">p_reads_hv =</span> <span class="fu">sum</span>(p_reads_hv), </span>
<span id="cb36-774"><a href="#cb36-774" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups=</span><span class="st">&quot;drop&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-775"><a href="#cb36-775" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">factor</span>(name_display, </span>
<span id="cb36-776"><a href="#cb36-776" aria-hidden="true" tabindex="-1"></a>                               <span class="at">levels =</span> <span class="fu">c</span>(hv_family_major_tab<span class="sc">$</span>name, <span class="st">&quot;Other&quot;</span>)))</span>
<span id="cb36-777"><a href="#cb36-777" aria-hidden="true" tabindex="-1"></a>hv_family_counts_display <span class="ot">&lt;-</span> hv_family_counts_major <span class="sc">%&gt;%</span></span>
<span id="cb36-778"><a href="#cb36-778" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_hv, <span class="at">classification =</span> name_display) <span class="sc">%&gt;%</span></span>
<span id="cb36-779"><a href="#cb36-779" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by =</span> <span class="st">&#39;sample&#39;</span>)</span>
<span id="cb36-780"><a href="#cb36-780" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-781"><a href="#cb36-781" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb36-782"><a href="#cb36-782" aria-hidden="true" tabindex="-1"></a>g_hv_family <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb36-783"><a href="#cb36-783" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">data=</span>hv_family_counts_display, <span class="at">position =</span> <span class="st">&quot;stack&quot;</span>, <span class="at">width=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb36-784"><a href="#cb36-784" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">&quot;% HV Reads&quot;</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb36-785"><a href="#cb36-785" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb36-786"><a href="#cb36-786" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb36-787"><a href="#cb36-787" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">&#39;Accent&#39;</span>, <span class="at">name =</span> <span class="st">&quot;Viral family&quot;</span>) <span class="sc">+</span></span>
<span id="cb36-788"><a href="#cb36-788" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">&quot;Family composition of human-viral reads&quot;</span>) <span class="sc">+</span></span>
<span id="cb36-789"><a href="#cb36-789" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">4</span>)) <span class="sc">+</span></span>
<span id="cb36-790"><a href="#cb36-790" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">1.4</span>), <span class="at">hjust=</span><span class="dv">0</span>, <span class="at">face=</span><span class="st">&quot;plain&quot;</span>))</span>
<span id="cb36-791"><a href="#cb36-791" aria-hidden="true" tabindex="-1"></a>g_hv_family</span>
<span id="cb36-792"><a href="#cb36-792" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-793"><a href="#cb36-793" aria-hidden="true" tabindex="-1"></a><span class="co"># Get most prominent families for text</span></span>
<span id="cb36-794"><a href="#cb36-794" aria-hidden="true" tabindex="-1"></a>hv_family_collate <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span></span>
<span id="cb36-795"><a href="#cb36-795" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb36-796"><a href="#cb36-796" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_tot =</span> <span class="fu">sum</span>(n_reads_hv),</span>
<span id="cb36-797"><a href="#cb36-797" aria-hidden="true" tabindex="-1"></a>            <span class="at">p_reads_max =</span> <span class="fu">max</span>(p_reads_hv), <span class="at">.groups=</span><span class="st">&quot;drop&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-798"><a href="#cb36-798" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n_reads_tot))</span>
<span id="cb36-799"><a href="#cb36-799" aria-hidden="true" tabindex="-1"></a>hv_family_collate <span class="sc">%&gt;%</span></span>
<span id="cb36-800"><a href="#cb36-800" aria-hidden="true" tabindex="-1"></a> <span class="fu">select</span>(name,taxid, n_reads_tot) <span class="sc">%&gt;%</span></span>
<span id="cb36-801"><a href="#cb36-801" aria-hidden="true" tabindex="-1"></a> <span class="fu">rename</span>(</span>
<span id="cb36-802"><a href="#cb36-802" aria-hidden="true" tabindex="-1"></a>  <span class="st">&#39;family&#39;</span> <span class="ot">=</span> <span class="st">&#39;name&#39;</span>,</span>
<span id="cb36-803"><a href="#cb36-803" aria-hidden="true" tabindex="-1"></a>  <span class="st">&#39;# of total reads&#39;</span> <span class="ot">=</span> <span class="st">&#39;n_reads_tot&#39;</span>,</span>
<span id="cb36-804"><a href="#cb36-804" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb36-805"><a href="#cb36-805" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb36-806"><a href="#cb36-806" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-807"><a href="#cb36-807" aria-hidden="true" tabindex="-1"></a><span class="fu">## Analyzing specific families</span></span>
<span id="cb36-808"><a href="#cb36-808" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-809"><a href="#cb36-809" aria-hidden="true" tabindex="-1"></a>We now investigate the composition of specific families that had more than 5 viral reads. In investigating individual viral families, to avoid distortions from a few rare reads, I restricted myself to samples where that family made up at least 1% of human-viral reads:</span>
<span id="cb36-810"><a href="#cb36-810" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-811"><a href="#cb36-811" aria-hidden="true" tabindex="-1"></a>quarto-executable-code-5450563D</span>
<span id="cb36-812"><a href="#cb36-812" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-813"><a href="#cb36-813" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb36-814"><a href="#cb36-814" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-viral-family-scripts</span></span>
<span id="cb36-815"><a href="#cb36-815" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb36-816"><a href="#cb36-816" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-817"><a href="#cb36-817" aria-hidden="true" tabindex="-1"></a>plot_viral_family_composition <span class="ot">&lt;-</span> <span class="cf">function</span>(taxid_chosen, <span class="at">threshold_major_species =</span> <span class="fl">0.1</span>) {</span>
<span id="cb36-818"><a href="#cb36-818" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get set of chosen family reads</span></span>
<span id="cb36-819"><a href="#cb36-819" aria-hidden="true" tabindex="-1"></a>  family_samples <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span> <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_chosen) <span class="sc">%&gt;%</span></span>
<span id="cb36-820"><a href="#cb36-820" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(p_reads_hv <span class="sc">&gt;=</span> <span class="fl">0.1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-821"><a href="#cb36-821" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pull</span>(sample)</span>
<span id="cb36-822"><a href="#cb36-822" aria-hidden="true" tabindex="-1"></a>  family_ids <span class="ot">&lt;-</span> hv_reads_family <span class="sc">%&gt;%</span> </span>
<span id="cb36-823"><a href="#cb36-823" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_chosen, sample <span class="sc">%in%</span> family_samples) <span class="sc">%&gt;%</span></span>
<span id="cb36-824"><a href="#cb36-824" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pull</span>(seq_id)</span>
<span id="cb36-825"><a href="#cb36-825" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-826"><a href="#cb36-826" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Count reads for each species in the chosen family</span></span>
<span id="cb36-827"><a href="#cb36-827" aria-hidden="true" tabindex="-1"></a>  species_counts <span class="ot">&lt;-</span> hv_reads_species <span class="sc">%&gt;%</span></span>
<span id="cb36-828"><a href="#cb36-828" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(seq_id <span class="sc">%in%</span> family_ids) <span class="sc">%&gt;%</span></span>
<span id="cb36-829"><a href="#cb36-829" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(sample, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb36-830"><a href="#cb36-830" aria-hidden="true" tabindex="-1"></a>    <span class="fu">count</span>(<span class="at">name =</span> <span class="st">&quot;n_reads_hv&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-831"><a href="#cb36-831" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb36-832"><a href="#cb36-832" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">p_reads_family =</span> n_reads_hv<span class="sc">/</span><span class="fu">sum</span>(n_reads_hv))</span>
<span id="cb36-833"><a href="#cb36-833" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-834"><a href="#cb36-834" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Identify high-ranking species and group others</span></span>
<span id="cb36-835"><a href="#cb36-835" aria-hidden="true" tabindex="-1"></a>  species_major_tab <span class="ot">&lt;-</span> species_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span> </span>
<span id="cb36-836"><a href="#cb36-836" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(p_reads_family <span class="sc">==</span> <span class="fu">max</span>(p_reads_family)) <span class="sc">%&gt;%</span> </span>
<span id="cb36-837"><a href="#cb36-837" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-838"><a href="#cb36-838" aria-hidden="true" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_family)) <span class="sc">%&gt;%</span> </span>
<span id="cb36-839"><a href="#cb36-839" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(p_reads_family <span class="sc">&gt;</span> threshold_major_species)</span>
<span id="cb36-840"><a href="#cb36-840" aria-hidden="true" tabindex="-1"></a>  species_counts_major <span class="ot">&lt;-</span> species_counts <span class="sc">%&gt;%</span></span>
<span id="cb36-841"><a href="#cb36-841" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">ifelse</span>(name <span class="sc">%in%</span> species_major_tab<span class="sc">$</span>name, </span>
<span id="cb36-842"><a href="#cb36-842" aria-hidden="true" tabindex="-1"></a>                                 name, <span class="st">&quot;Other&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb36-843"><a href="#cb36-843" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(sample, name_display) <span class="sc">%&gt;%</span></span>
<span id="cb36-844"><a href="#cb36-844" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarize</span>(<span class="at">n_reads_family =</span> <span class="fu">sum</span>(n_reads_hv),</span>
<span id="cb36-845"><a href="#cb36-845" aria-hidden="true" tabindex="-1"></a>              <span class="at">p_reads_family =</span> <span class="fu">sum</span>(p_reads_family), </span>
<span id="cb36-846"><a href="#cb36-846" aria-hidden="true" tabindex="-1"></a>              <span class="at">.groups=</span><span class="st">&quot;drop&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-847"><a href="#cb36-847" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">factor</span>(name_display, </span>
<span id="cb36-848"><a href="#cb36-848" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">levels =</span> <span class="fu">c</span>(species_major_tab<span class="sc">$</span>name, <span class="st">&quot;Other&quot;</span>)))</span>
<span id="cb36-849"><a href="#cb36-849" aria-hidden="true" tabindex="-1"></a>  species_counts_display <span class="ot">&lt;-</span> species_counts_major <span class="sc">%&gt;%</span></span>
<span id="cb36-850"><a href="#cb36-850" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_family, <span class="at">classification =</span> name_display) <span class="sc">%&gt;%</span></span>
<span id="cb36-851"><a href="#cb36-851" aria-hidden="true" tabindex="-1"></a>    <span class="fu">right_join</span>(libraries, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&#39;sample&#39;</span>))</span>
<span id="cb36-852"><a href="#cb36-852" aria-hidden="true" tabindex="-1"></a>    <span class="co">#mutate(p_reads = coalesce(p_reads, 0),</span></span>
<span id="cb36-853"><a href="#cb36-853" aria-hidden="true" tabindex="-1"></a>    <span class="co">#       classification = coalesce(classification, species_major_tab$name[1])) %&gt;%</span></span>
<span id="cb36-854"><a href="#cb36-854" aria-hidden="true" tabindex="-1"></a>    <span class="co">#group_by(location) %&gt;%</span></span>
<span id="cb36-855"><a href="#cb36-855" aria-hidden="true" tabindex="-1"></a>    <span class="co">#mutate(samples_with_reads = sum(p_reads &gt; 0),</span></span>
<span id="cb36-856"><a href="#cb36-856" aria-hidden="true" tabindex="-1"></a>    <span class="co">#  label = paste0(location, &quot; (&quot;, sprintf(&quot;%.1f%%&quot;, 100 * samples_with_reads / total_samples), &quot;)&quot;)) %&gt;%</span></span>
<span id="cb36-857"><a href="#cb36-857" aria-hidden="true" tabindex="-1"></a>    <span class="co">#ungroup()</span></span>
<span id="cb36-858"><a href="#cb36-858" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-859"><a href="#cb36-859" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get family name for the title</span></span>
<span id="cb36-860"><a href="#cb36-860" aria-hidden="true" tabindex="-1"></a>  family_name <span class="ot">&lt;-</span> viral_taxa <span class="sc">%&gt;%</span> </span>
<span id="cb36-861"><a href="#cb36-861" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_chosen) <span class="sc">%&gt;%</span> </span>
<span id="cb36-862"><a href="#cb36-862" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pull</span>(name)</span>
<span id="cb36-863"><a href="#cb36-863" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-864"><a href="#cb36-864" aria-hidden="true" tabindex="-1"></a>  species_count <span class="ot">&lt;-</span> species_counts_display <span class="sc">%&gt;%</span> <span class="fu">select</span>(classification) <span class="sc">%&gt;%</span> <span class="fu">distinct</span>() <span class="sc">%&gt;%</span> <span class="fu">pull</span>() <span class="sc">%&gt;%</span> <span class="fu">length</span>()</span>
<span id="cb36-865"><a href="#cb36-865" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (species_count <span class="sc">&lt;=</span> <span class="dv">8</span>) {</span>
<span id="cb36-866"><a href="#cb36-866" aria-hidden="true" tabindex="-1"></a>    pal <span class="ot">&lt;-</span> <span class="fu">brewer.pal</span>(<span class="at">name =</span> <span class="st">&quot;Accent&quot;</span>, <span class="at">n =</span> species_count)</span>
<span id="cb36-867"><a href="#cb36-867" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb36-868"><a href="#cb36-868" aria-hidden="true" tabindex="-1"></a>    pal <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">brewer.pal</span>(<span class="at">name =</span> <span class="st">&quot;Paired&quot;</span>, <span class="at">n =</span> <span class="dv">12</span>), <span class="fu">brewer.pal</span>(<span class="at">name =</span> <span class="st">&quot;Set3&quot;</span>, <span class="at">n =</span><span class="dv">12</span>), <span class="fu">brewer.pal</span>(<span class="at">name =</span> <span class="st">&quot;Set1&quot;</span>, <span class="at">n =</span> <span class="dv">9</span>))[<span class="dv">1</span><span class="sc">:</span>species_count]</span>
<span id="cb36-869"><a href="#cb36-869" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb36-870"><a href="#cb36-870" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-871"><a href="#cb36-871" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Plot</span></span>
<span id="cb36-872"><a href="#cb36-872" aria-hidden="true" tabindex="-1"></a>  g_species <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb36-873"><a href="#cb36-873" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_col</span>(<span class="at">data=</span>species_counts_display, <span class="at">position =</span> <span class="st">&quot;stack&quot;</span>) <span class="sc">+</span></span>
<span id="cb36-874"><a href="#cb36-874" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="fu">paste0</span>(<span class="st">&quot;% &quot;</span>, family_name, <span class="st">&quot; reads&quot;</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb36-875"><a href="#cb36-875" aria-hidden="true" tabindex="-1"></a>                       <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb36-876"><a href="#cb36-876" aria-hidden="true" tabindex="-1"></a>                       <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb36-877"><a href="#cb36-877" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> pal, <span class="at">name =</span> <span class="st">&quot;Viral species&quot;</span>) <span class="sc">+</span></span>
<span id="cb36-878"><a href="#cb36-878" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title=</span><span class="fu">paste0</span>(<span class="st">&quot;Species composition of &quot;</span>, family_name, <span class="st">&quot; reads&quot;</span>)) <span class="sc">+</span></span>
<span id="cb36-879"><a href="#cb36-879" aria-hidden="true" tabindex="-1"></a>    <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">3</span>)) <span class="sc">+</span></span>
<span id="cb36-880"><a href="#cb36-880" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">1.4</span>), <span class="at">hjust=</span><span class="dv">0</span>, <span class="at">face=</span><span class="st">&quot;plain&quot;</span>))</span>
<span id="cb36-881"><a href="#cb36-881" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(g_species)</span>
<span id="cb36-882"><a href="#cb36-882" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb36-883"><a href="#cb36-883" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-884"><a href="#cb36-884" aria-hidden="true" tabindex="-1"></a>plot_viral_family_histogram <span class="ot">&lt;-</span> <span class="cf">function</span>(taxid_chosen) {</span>
<span id="cb36-885"><a href="#cb36-885" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filter data for the specified virus family</span></span>
<span id="cb36-886"><a href="#cb36-886" aria-hidden="true" tabindex="-1"></a>  family_data <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span></span>
<span id="cb36-887"><a href="#cb36-887" aria-hidden="true" tabindex="-1"></a>    <span class="fu">inner_join</span>(libraries, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&#39;sample&#39;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb36-888"><a href="#cb36-888" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_chosen)</span>
<span id="cb36-889"><a href="#cb36-889" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb36-890"><a href="#cb36-890" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get family name for the title</span></span>
<span id="cb36-891"><a href="#cb36-891" aria-hidden="true" tabindex="-1"></a>  family_name <span class="ot">&lt;-</span> viral_taxa <span class="sc">%&gt;%</span> </span>
<span id="cb36-892"><a href="#cb36-892" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_chosen) <span class="sc">%&gt;%</span> </span>
<span id="cb36-893"><a href="#cb36-893" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pull</span>(name)</span>
<span id="cb36-894"><a href="#cb36-894" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb36-895"><a href="#cb36-895" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create the histogram</span></span>
<span id="cb36-896"><a href="#cb36-896" aria-hidden="true" tabindex="-1"></a>  g_histogram <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(family_data, <span class="fu">aes</span>(<span class="at">x =</span> library, <span class="at">y =</span> n_reads_hv)) <span class="sc">+</span></span>
<span id="cb36-897"><a href="#cb36-897" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">&quot;identity&quot;</span>) <span class="sc">+</span></span>
<span id="cb36-898"><a href="#cb36-898" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label =</span> scales<span class="sc">::</span><span class="fu">comma</span>(n_reads_hv)), <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb36-899"><a href="#cb36-899" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb36-900"><a href="#cb36-900" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb36-901"><a href="#cb36-901" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_discrete</span>(<span class="at">limits =</span> libraries <span class="sc">%&gt;%</span> <span class="fu">pull</span>(library)) <span class="sc">+</span></span>
<span id="cb36-902"><a href="#cb36-902" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(</span>
<span id="cb36-903"><a href="#cb36-903" aria-hidden="true" tabindex="-1"></a>      <span class="at">title =</span> <span class="fu">paste</span>(<span class="st">&quot;Distribution of&quot;</span>, family_name, <span class="st">&quot;Reads by Location&quot;</span>),</span>
<span id="cb36-904"><a href="#cb36-904" aria-hidden="true" tabindex="-1"></a>      <span class="at">x =</span> <span class="st">&quot;Location&quot;</span>,</span>
<span id="cb36-905"><a href="#cb36-905" aria-hidden="true" tabindex="-1"></a>      <span class="at">y =</span> <span class="st">&quot;Number of reads&quot;</span></span>
<span id="cb36-906"><a href="#cb36-906" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb36-907"><a href="#cb36-907" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_continuous</span>(<span class="at">labels =</span> scales<span class="sc">::</span>comma) <span class="sc">+</span></span>
<span id="cb36-908"><a href="#cb36-908" aria-hidden="true" tabindex="-1"></a>    <span class="fu">expand_limits</span>(<span class="at">y =</span> <span class="fu">max</span>(family_data<span class="sc">$</span>n_reads_hv) <span class="sc">*</span> <span class="fl">1.1</span>)<span class="sc">+</span></span>
<span id="cb36-909"><a href="#cb36-909" aria-hidden="true" tabindex="-1"></a>    theme_kit  <span class="co"># Expand y-axis to make room for labels</span></span>
<span id="cb36-910"><a href="#cb36-910" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(g_histogram)</span>
<span id="cb36-911"><a href="#cb36-911" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb36-912"><a href="#cb36-912" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-913"><a href="#cb36-913" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-914"><a href="#cb36-914" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb36-915"><a href="#cb36-915" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-916"><a href="#cb36-916" aria-hidden="true" tabindex="-1"></a><span class="fu">### Anelloviridae (Number of reads: 12,804)</span></span>
<span id="cb36-917"><a href="#cb36-917" aria-hidden="true" tabindex="-1"></a>quarto-executable-code-5450563D</span>
<span id="cb36-918"><a href="#cb36-918" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-919"><a href="#cb36-919" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb36-920"><a href="#cb36-920" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-anelloviridae-histogram</span></span>
<span id="cb36-921"><a href="#cb36-921" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb36-922"><a href="#cb36-922" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 2.5</span></span>
<span id="cb36-923"><a href="#cb36-923" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-924"><a href="#cb36-924" aria-hidden="true" tabindex="-1"></a><span class="co"># Anelloviridae</span></span>
<span id="cb36-925"><a href="#cb36-925" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_histogram</span>(<span class="at">taxid_chosen=</span><span class="dv">687329</span>)</span>
<span id="cb36-926"><a href="#cb36-926" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb36-927"><a href="#cb36-927" aria-hidden="true" tabindex="-1"></a>quarto-executable-code-5450563D</span>
<span id="cb36-928"><a href="#cb36-928" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-929"><a href="#cb36-929" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb36-930"><a href="#cb36-930" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-anelloviridae-composition</span></span>
<span id="cb36-931"><a href="#cb36-931" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb36-932"><a href="#cb36-932" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 7.5</span></span>
<span id="cb36-933"><a href="#cb36-933" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb36-934"><a href="#cb36-934" aria-hidden="true" tabindex="-1"></a><span class="co"># Anelloviridae</span></span>
<span id="cb36-935"><a href="#cb36-935" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_composition</span>(<span class="at">taxid_chosen=</span><span class="dv">687329</span>, <span class="at">threshold_major_species =</span> <span class="fl">0.01</span>)</span>
<span id="cb36-936"><a href="#cb36-936" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb36-937"><a href="#cb36-937" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-938"><a href="#cb36-938" aria-hidden="true" tabindex="-1"></a><span class="fu">### Microviridae (Number of reads: 5,194)</span></span>
<span id="cb36-939"><a href="#cb36-939" aria-hidden="true" tabindex="-1"></a>quarto-executable-code-5450563D</span>
<span id="cb36-940"><a href="#cb36-940" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-941"><a href="#cb36-941" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb36-942"><a href="#cb36-942" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-microviridae-histogram</span></span>
<span id="cb36-943"><a href="#cb36-943" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb36-944"><a href="#cb36-944" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 2.5</span></span>
<span id="cb36-945"><a href="#cb36-945" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-946"><a href="#cb36-946" aria-hidden="true" tabindex="-1"></a><span class="co"># Microviridae</span></span>
<span id="cb36-947"><a href="#cb36-947" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_histogram</span>(<span class="at">taxid_chosen=</span><span class="dv">10841</span>)</span>
<span id="cb36-948"><a href="#cb36-948" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb36-949"><a href="#cb36-949" aria-hidden="true" tabindex="-1"></a>quarto-executable-code-5450563D</span>
<span id="cb36-950"><a href="#cb36-950" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-951"><a href="#cb36-951" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb36-952"><a href="#cb36-952" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-microviridae-composition</span></span>
<span id="cb36-953"><a href="#cb36-953" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb36-954"><a href="#cb36-954" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 7.5</span></span>
<span id="cb36-955"><a href="#cb36-955" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb36-956"><a href="#cb36-956" aria-hidden="true" tabindex="-1"></a><span class="co"># Microviridae</span></span>
<span id="cb36-957"><a href="#cb36-957" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_composition</span>(<span class="at">taxid_chosen=</span><span class="dv">10841</span>, <span class="at">threshold_major_species =</span> <span class="fl">0.01</span>)</span>
<span id="cb36-958"><a href="#cb36-958" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb36-959"><a href="#cb36-959" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-960"><a href="#cb36-960" aria-hidden="true" tabindex="-1"></a><span class="fu">### Sedreoviridae (Number of reads: 278)</span></span>
<span id="cb36-961"><a href="#cb36-961" aria-hidden="true" tabindex="-1"></a>quarto-executable-code-5450563D</span>
<span id="cb36-962"><a href="#cb36-962" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-963"><a href="#cb36-963" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb36-964"><a href="#cb36-964" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-sedreoviridae-histogram</span></span>
<span id="cb36-965"><a href="#cb36-965" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb36-966"><a href="#cb36-966" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 2.5</span></span>
<span id="cb36-967"><a href="#cb36-967" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-968"><a href="#cb36-968" aria-hidden="true" tabindex="-1"></a><span class="co"># Sedreoviridae</span></span>
<span id="cb36-969"><a href="#cb36-969" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_histogram</span>(<span class="at">taxid_chosen=</span><span class="dv">2946186</span>)</span>
<span id="cb36-970"><a href="#cb36-970" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb36-971"><a href="#cb36-971" aria-hidden="true" tabindex="-1"></a>quarto-executable-code-5450563D</span>
<span id="cb36-972"><a href="#cb36-972" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-973"><a href="#cb36-973" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb36-974"><a href="#cb36-974" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-sedreoviridae-composition</span></span>
<span id="cb36-975"><a href="#cb36-975" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb36-976"><a href="#cb36-976" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 7.5</span></span>
<span id="cb36-977"><a href="#cb36-977" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb36-978"><a href="#cb36-978" aria-hidden="true" tabindex="-1"></a><span class="co"># Sedreoviridae</span></span>
<span id="cb36-979"><a href="#cb36-979" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_composition</span>(<span class="at">taxid_chosen=</span><span class="dv">2946186</span>, <span class="at">threshold_major_species =</span> <span class="fl">0.01</span>)</span>
<span id="cb36-980"><a href="#cb36-980" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb36-981"><a href="#cb36-981" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-982"><a href="#cb36-982" aria-hidden="true" tabindex="-1"></a><span class="fu">### Parvoviridae (Number of reads: 266) </span></span>
<span id="cb36-983"><a href="#cb36-983" aria-hidden="true" tabindex="-1"></a>quarto-executable-code-5450563D</span>
<span id="cb36-984"><a href="#cb36-984" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-985"><a href="#cb36-985" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb36-986"><a href="#cb36-986" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-parvoviridae-histogram</span></span>
<span id="cb36-987"><a href="#cb36-987" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb36-988"><a href="#cb36-988" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 2.5</span></span>
<span id="cb36-989"><a href="#cb36-989" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-990"><a href="#cb36-990" aria-hidden="true" tabindex="-1"></a><span class="co"># Parvoviridae</span></span>
<span id="cb36-991"><a href="#cb36-991" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_histogram</span>(<span class="at">taxid_chosen=</span><span class="dv">10780</span>)</span>
<span id="cb36-992"><a href="#cb36-992" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb36-993"><a href="#cb36-993" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-994"><a href="#cb36-994" aria-hidden="true" tabindex="-1"></a>quarto-executable-code-5450563D</span>
<span id="cb36-995"><a href="#cb36-995" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-996"><a href="#cb36-996" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb36-997"><a href="#cb36-997" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-parvoviridae-composition</span></span>
<span id="cb36-998"><a href="#cb36-998" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb36-999"><a href="#cb36-999" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 7.5</span></span>
<span id="cb36-1000"><a href="#cb36-1000" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb36-1001"><a href="#cb36-1001" aria-hidden="true" tabindex="-1"></a><span class="co"># Parvoviridae</span></span>
<span id="cb36-1002"><a href="#cb36-1002" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_composition</span>(<span class="at">taxid_chosen=</span><span class="dv">10780</span>, <span class="at">threshold_major_species =</span> <span class="fl">0.01</span>)</span>
<span id="cb36-1003"><a href="#cb36-1003" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb36-1004"><a href="#cb36-1004" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1005"><a href="#cb36-1005" aria-hidden="true" tabindex="-1"></a><span class="fu">### Adenoviridae (Number of reads: 104) </span></span>
<span id="cb36-1006"><a href="#cb36-1006" aria-hidden="true" tabindex="-1"></a>quarto-executable-code-5450563D</span>
<span id="cb36-1007"><a href="#cb36-1007" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1008"><a href="#cb36-1008" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb36-1009"><a href="#cb36-1009" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-adenoviridae-histogram</span></span>
<span id="cb36-1010"><a href="#cb36-1010" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb36-1011"><a href="#cb36-1011" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 2.5</span></span>
<span id="cb36-1012"><a href="#cb36-1012" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1013"><a href="#cb36-1013" aria-hidden="true" tabindex="-1"></a><span class="co"># Adenoviridae</span></span>
<span id="cb36-1014"><a href="#cb36-1014" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_histogram</span>(<span class="at">taxid_chosen=</span><span class="dv">10508</span>)</span>
<span id="cb36-1015"><a href="#cb36-1015" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb36-1016"><a href="#cb36-1016" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1017"><a href="#cb36-1017" aria-hidden="true" tabindex="-1"></a>None of these reads made up more than 1% of the total reads of any sample.</span>
<span id="cb36-1018"><a href="#cb36-1018" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1019"><a href="#cb36-1019" aria-hidden="true" tabindex="-1"></a><span class="fu">### Polyomaviridae (Number of reads: 95) </span></span>
<span id="cb36-1020"><a href="#cb36-1020" aria-hidden="true" tabindex="-1"></a>quarto-executable-code-5450563D</span>
<span id="cb36-1021"><a href="#cb36-1021" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1022"><a href="#cb36-1022" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb36-1023"><a href="#cb36-1023" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-polyomaviridae-histogram</span></span>
<span id="cb36-1024"><a href="#cb36-1024" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb36-1025"><a href="#cb36-1025" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 2.5</span></span>
<span id="cb36-1026"><a href="#cb36-1026" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1027"><a href="#cb36-1027" aria-hidden="true" tabindex="-1"></a><span class="co"># Polyomaviridae</span></span>
<span id="cb36-1028"><a href="#cb36-1028" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_histogram</span>(<span class="at">taxid_chosen=</span><span class="dv">151341</span>)</span>
<span id="cb36-1029"><a href="#cb36-1029" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb36-1030"><a href="#cb36-1030" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1031"><a href="#cb36-1031" aria-hidden="true" tabindex="-1"></a>None of these reads made up more than 1% of the total reads of any sample.</span>
<span id="cb36-1032"><a href="#cb36-1032" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1033"><a href="#cb36-1033" aria-hidden="true" tabindex="-1"></a><span class="fu">### Papillomaviridae (Number of reads: 16) </span></span>
<span id="cb36-1034"><a href="#cb36-1034" aria-hidden="true" tabindex="-1"></a>quarto-executable-code-5450563D</span>
<span id="cb36-1035"><a href="#cb36-1035" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1036"><a href="#cb36-1036" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb36-1037"><a href="#cb36-1037" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-papillomaviridae-histogram</span></span>
<span id="cb36-1038"><a href="#cb36-1038" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb36-1039"><a href="#cb36-1039" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 2.5</span></span>
<span id="cb36-1040"><a href="#cb36-1040" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1041"><a href="#cb36-1041" aria-hidden="true" tabindex="-1"></a><span class="co"># Polyomaviridae</span></span>
<span id="cb36-1042"><a href="#cb36-1042" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_viral_family_histogram</span>(<span class="at">taxid_chosen=</span><span class="dv">151340</span>)</span>
<span id="cb36-1043"><a href="#cb36-1043" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb36-1044"><a href="#cb36-1044" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1045"><a href="#cb36-1045" aria-hidden="true" tabindex="-1"></a>None of these reads made up more than 1% of the total reads of any sample.</span>
<span id="cb36-1046"><a href="#cb36-1046" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1047"><a href="#cb36-1047" aria-hidden="true" tabindex="-1"></a><span class="fu">## Relative abundance of pathogenic viruses of interest</span></span>
<span id="cb36-1048"><a href="#cb36-1048" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1049"><a href="#cb36-1049" aria-hidden="true" tabindex="-1"></a>quarto-executable-code-5450563D</span>
<span id="cb36-1050"><a href="#cb36-1050" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1051"><a href="#cb36-1051" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb36-1052"><a href="#cb36-1052" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-composition-exclusion</span></span>
<span id="cb36-1053"><a href="#cb36-1053" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb36-1054"><a href="#cb36-1054" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1055"><a href="#cb36-1055" aria-hidden="true" tabindex="-1"></a>total <span class="ot">&lt;-</span> basic_stats_raw <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample, n_read_pairs)</span>
<span id="cb36-1056"><a href="#cb36-1056" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1057"><a href="#cb36-1057" aria-hidden="true" tabindex="-1"></a>hv_family_counts <span class="ot">&lt;-</span> hv_reads_family <span class="sc">%&gt;%</span> </span>
<span id="cb36-1058"><a href="#cb36-1058" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb36-1059"><a href="#cb36-1059" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">&quot;n_reads_hv&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-1060"><a href="#cb36-1060" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(total, <span class="at">by=</span><span class="st">&#39;sample&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-1061"><a href="#cb36-1061" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb36-1062"><a href="#cb36-1062" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">ra_reads_hv =</span> n_reads_hv<span class="sc">/</span>n_read_pairs)</span>
<span id="cb36-1063"><a href="#cb36-1063" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1064"><a href="#cb36-1064" aria-hidden="true" tabindex="-1"></a>hv_family_collate <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span></span>
<span id="cb36-1065"><a href="#cb36-1065" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb36-1066"><a href="#cb36-1066" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_tot =</span> <span class="fu">sum</span>(n_reads_hv),</span>
<span id="cb36-1067"><a href="#cb36-1067" aria-hidden="true" tabindex="-1"></a>            <span class="at">ra_reads_tot =</span> <span class="fu">mean</span>(ra_reads_hv), <span class="at">.groups=</span><span class="st">&quot;drop&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-1068"><a href="#cb36-1068" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n_reads_tot))</span>
<span id="cb36-1069"><a href="#cb36-1069" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1070"><a href="#cb36-1070" aria-hidden="true" tabindex="-1"></a>hv_family_collate <span class="ot">&lt;-</span> hv_family_collate <span class="sc">%&gt;%</span> </span>
<span id="cb36-1071"><a href="#cb36-1071" aria-hidden="true" tabindex="-1"></a><span class="fu">rename</span>(</span>
<span id="cb36-1072"><a href="#cb36-1072" aria-hidden="true" tabindex="-1"></a><span class="st">&#39;family&#39;</span> <span class="ot">=</span> <span class="st">&#39;name&#39;</span>,</span>
<span id="cb36-1073"><a href="#cb36-1073" aria-hidden="true" tabindex="-1"></a><span class="st">&#39;family_n_reads_tot&#39;</span> <span class="ot">=</span> <span class="st">&#39;n_reads_tot&#39;</span>,</span>
<span id="cb36-1074"><a href="#cb36-1074" aria-hidden="true" tabindex="-1"></a><span class="st">&#39;family_ra_reads_tot&#39;</span> <span class="ot">=</span> <span class="st">&#39;ra_reads_tot&#39;</span>,</span>
<span id="cb36-1075"><a href="#cb36-1075" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb36-1076"><a href="#cb36-1076" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1077"><a href="#cb36-1077" aria-hidden="true" tabindex="-1"></a>hv_genus_counts <span class="ot">&lt;-</span> hv_reads_genus <span class="sc">%&gt;%</span> </span>
<span id="cb36-1078"><a href="#cb36-1078" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb36-1079"><a href="#cb36-1079" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">&quot;n_reads_hv&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-1080"><a href="#cb36-1080" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(total, <span class="at">by=</span><span class="st">&#39;sample&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-1081"><a href="#cb36-1081" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb36-1082"><a href="#cb36-1082" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">ra_reads_hv =</span> n_reads_hv<span class="sc">/</span>n_read_pairs)</span>
<span id="cb36-1083"><a href="#cb36-1083" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1084"><a href="#cb36-1084" aria-hidden="true" tabindex="-1"></a>hv_genus_collate <span class="ot">&lt;-</span> hv_genus_counts <span class="sc">%&gt;%</span></span>
<span id="cb36-1085"><a href="#cb36-1085" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb36-1086"><a href="#cb36-1086" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_tot =</span> <span class="fu">sum</span>(n_reads_hv),</span>
<span id="cb36-1087"><a href="#cb36-1087" aria-hidden="true" tabindex="-1"></a>            <span class="at">ra_reads_tot =</span> <span class="fu">mean</span>(ra_reads_hv), <span class="at">.groups=</span><span class="st">&quot;drop&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-1088"><a href="#cb36-1088" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n_reads_tot))</span>
<span id="cb36-1089"><a href="#cb36-1089" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1090"><a href="#cb36-1090" aria-hidden="true" tabindex="-1"></a>hv_genus_collate <span class="ot">&lt;-</span> hv_genus_collate <span class="sc">%&gt;%</span> </span>
<span id="cb36-1091"><a href="#cb36-1091" aria-hidden="true" tabindex="-1"></a><span class="fu">rename</span>(</span>
<span id="cb36-1092"><a href="#cb36-1092" aria-hidden="true" tabindex="-1"></a><span class="st">&#39;genus&#39;</span> <span class="ot">=</span> <span class="st">&#39;name&#39;</span>,</span>
<span id="cb36-1093"><a href="#cb36-1093" aria-hidden="true" tabindex="-1"></a><span class="st">&#39;genus_n_reads_tot&#39;</span> <span class="ot">=</span> <span class="st">&#39;n_reads_tot&#39;</span>,</span>
<span id="cb36-1094"><a href="#cb36-1094" aria-hidden="true" tabindex="-1"></a><span class="st">&#39;genus_ra_reads_tot&#39;</span> <span class="ot">=</span> <span class="st">&#39;ra_reads_tot&#39;</span>,</span>
<span id="cb36-1095"><a href="#cb36-1095" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb36-1096"><a href="#cb36-1096" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1097"><a href="#cb36-1097" aria-hidden="true" tabindex="-1"></a>hv_species_counts <span class="ot">&lt;-</span> hv_reads_species <span class="sc">%&gt;%</span> </span>
<span id="cb36-1098"><a href="#cb36-1098" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb36-1099"><a href="#cb36-1099" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">&quot;n_reads_hv&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-1100"><a href="#cb36-1100" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(total, <span class="at">by=</span><span class="st">&#39;sample&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-1101"><a href="#cb36-1101" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb36-1102"><a href="#cb36-1102" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">ra_reads_hv =</span> n_reads_hv<span class="sc">/</span>n_read_pairs)</span>
<span id="cb36-1103"><a href="#cb36-1103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1104"><a href="#cb36-1104" aria-hidden="true" tabindex="-1"></a>hv_species_collate <span class="ot">&lt;-</span> hv_species_counts <span class="sc">%&gt;%</span></span>
<span id="cb36-1105"><a href="#cb36-1105" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb36-1106"><a href="#cb36-1106" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_tot =</span> <span class="fu">sum</span>(n_reads_hv),</span>
<span id="cb36-1107"><a href="#cb36-1107" aria-hidden="true" tabindex="-1"></a>            <span class="at">ra_reads_tot =</span> <span class="fu">mean</span>(ra_reads_hv), <span class="at">.groups=</span><span class="st">&quot;drop&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-1108"><a href="#cb36-1108" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n_reads_tot))</span>
<span id="cb36-1109"><a href="#cb36-1109" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1110"><a href="#cb36-1110" aria-hidden="true" tabindex="-1"></a>hv_species_collate <span class="ot">&lt;-</span> hv_species_collate <span class="sc">%&gt;%</span> </span>
<span id="cb36-1111"><a href="#cb36-1111" aria-hidden="true" tabindex="-1"></a><span class="fu">rename</span>(</span>
<span id="cb36-1112"><a href="#cb36-1112" aria-hidden="true" tabindex="-1"></a><span class="st">&#39;species&#39;</span> <span class="ot">=</span> <span class="st">&#39;name&#39;</span>,</span>
<span id="cb36-1113"><a href="#cb36-1113" aria-hidden="true" tabindex="-1"></a><span class="st">&#39;species_n_reads_tot&#39;</span> <span class="ot">=</span> <span class="st">&#39;n_reads_tot&#39;</span>,</span>
<span id="cb36-1114"><a href="#cb36-1114" aria-hidden="true" tabindex="-1"></a><span class="st">&#39;species_ra_reads_tot&#39;</span> <span class="ot">=</span> <span class="st">&#39;ra_reads_tot&#39;</span>,</span>
<span id="cb36-1115"><a href="#cb36-1115" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb36-1116"><a href="#cb36-1116" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1117"><a href="#cb36-1117" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb36-1118"><a href="#cb36-1118" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1119"><a href="#cb36-1119" aria-hidden="true" tabindex="-1"></a>quarto-executable-code-5450563D</span>
<span id="cb36-1120"><a href="#cb36-1120" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1121"><a href="#cb36-1121" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb36-1122"><a href="#cb36-1122" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-filter-exclusion</span></span>
<span id="cb36-1123"><a href="#cb36-1123" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb36-1124"><a href="#cb36-1124" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1125"><a href="#cb36-1125" aria-hidden="true" tabindex="-1"></a><span class="co"># Assuming the data is already loaded into tibbles:</span></span>
<span id="cb36-1126"><a href="#cb36-1126" aria-hidden="true" tabindex="-1"></a><span class="co">#hv_family_collate</span></span>
<span id="cb36-1127"><a href="#cb36-1127" aria-hidden="true" tabindex="-1"></a><span class="co">#hv_genus_collate</span></span>
<span id="cb36-1128"><a href="#cb36-1128" aria-hidden="true" tabindex="-1"></a><span class="co">#hv_species_collate</span></span>
<span id="cb36-1129"><a href="#cb36-1129" aria-hidden="true" tabindex="-1"></a><span class="co">#viral_taxa</span></span>
<span id="cb36-1130"><a href="#cb36-1130" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1131"><a href="#cb36-1131" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to find parent taxa recursively</span></span>
<span id="cb36-1132"><a href="#cb36-1132" aria-hidden="true" tabindex="-1"></a>find_parent_taxa <span class="ot">&lt;-</span> <span class="cf">function</span>(taxid, target_ranks, taxa_data) {</span>
<span id="cb36-1133"><a href="#cb36-1133" aria-hidden="true" tabindex="-1"></a> current_taxid <span class="ot">&lt;-</span> taxid</span>
<span id="cb36-1134"><a href="#cb36-1134" aria-hidden="true" tabindex="-1"></a>  result <span class="ot">&lt;-</span> <span class="fu">setNames</span>(<span class="fu">vector</span>(<span class="st">&quot;list&quot;</span>, <span class="fu">length</span>(target_ranks)), target_ranks)</span>
<span id="cb36-1135"><a href="#cb36-1135" aria-hidden="true" tabindex="-1"></a>  <span class="cf">while</span> (<span class="cn">TRUE</span>) {</span>
<span id="cb36-1136"><a href="#cb36-1136" aria-hidden="true" tabindex="-1"></a>    row <span class="ot">&lt;-</span> taxa_data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(taxid <span class="sc">==</span> current_taxid) <span class="sc">%&gt;%</span> <span class="fu">slice</span>(<span class="dv">1</span>)</span>
<span id="cb36-1137"><a href="#cb36-1137" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">nrow</span>(row) <span class="sc">==</span> <span class="dv">0</span>) <span class="cf">break</span>  <span class="co"># If no matching taxid is found, break the loop</span></span>
<span id="cb36-1138"><a href="#cb36-1138" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (row<span class="sc">$</span>rank <span class="sc">%in%</span> target_ranks) {</span>
<span id="cb36-1139"><a href="#cb36-1139" aria-hidden="true" tabindex="-1"></a>      result[[row<span class="sc">$</span>rank]] <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">taxid =</span> row<span class="sc">$</span>taxid, <span class="at">name =</span> row<span class="sc">$</span>name)</span>
<span id="cb36-1140"><a href="#cb36-1140" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span> (<span class="fu">all</span>(<span class="sc">!</span><span class="fu">sapply</span>(result, is.null))) {</span>
<span id="cb36-1141"><a href="#cb36-1141" aria-hidden="true" tabindex="-1"></a>        <span class="cf">break</span>  <span class="co"># If all target ranks are found, break the loop</span></span>
<span id="cb36-1142"><a href="#cb36-1142" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb36-1143"><a href="#cb36-1143" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb36-1144"><a href="#cb36-1144" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (row<span class="sc">$</span>taxid <span class="sc">==</span> row<span class="sc">$</span>parent_taxid) {</span>
<span id="cb36-1145"><a href="#cb36-1145" aria-hidden="true" tabindex="-1"></a>      <span class="cf">break</span>  <span class="co"># Avoid infinite loop at root</span></span>
<span id="cb36-1146"><a href="#cb36-1146" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb36-1147"><a href="#cb36-1147" aria-hidden="true" tabindex="-1"></a>    current_taxid <span class="ot">&lt;-</span> row<span class="sc">$</span>parent_taxid</span>
<span id="cb36-1148"><a href="#cb36-1148" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb36-1149"><a href="#cb36-1149" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Replace NULL with NA for any unfound ranks</span></span>
<span id="cb36-1150"><a href="#cb36-1150" aria-hidden="true" tabindex="-1"></a>  result <span class="ot">&lt;-</span> <span class="fu">lapply</span>(result, <span class="cf">function</span>(x) <span class="cf">if</span> (<span class="fu">is.null</span>(x)) <span class="fu">list</span>(<span class="at">taxid =</span> <span class="cn">NA</span>, <span class="at">name =</span> <span class="cn">NA</span>) <span class="cf">else</span> x)</span>
<span id="cb36-1151"><a href="#cb36-1151" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(result)</span>
<span id="cb36-1152"><a href="#cb36-1152" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb36-1153"><a href="#cb36-1153" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1154"><a href="#cb36-1154" aria-hidden="true" tabindex="-1"></a><span class="co"># Find all relevant parent taxa for each species</span></span>
<span id="cb36-1155"><a href="#cb36-1155" aria-hidden="true" tabindex="-1"></a>result <span class="ot">&lt;-</span> hv_species_collate <span class="sc">%&gt;%</span></span>
<span id="cb36-1156"><a href="#cb36-1156" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb36-1157"><a href="#cb36-1157" aria-hidden="true" tabindex="-1"></a>    <span class="at">parent_info =</span> <span class="fu">map</span>(taxid, <span class="sc">~</span><span class="fu">find_parent_taxa</span>(.x, <span class="fu">c</span>(<span class="st">&quot;species&quot;</span>,<span class="st">&quot;subgenus&quot;</span>, <span class="st">&quot;genus&quot;</span>, <span class="st">&quot;subfamily&quot;</span>, <span class="st">&quot;family&quot;</span>), viral_taxa)),</span>
<span id="cb36-1158"><a href="#cb36-1158" aria-hidden="true" tabindex="-1"></a>    <span class="at">genus_taxid =</span> <span class="fu">map_dbl</span>(parent_info, <span class="sc">~</span>.x<span class="sc">$</span>genus<span class="sc">$</span>taxid),</span>
<span id="cb36-1159"><a href="#cb36-1159" aria-hidden="true" tabindex="-1"></a>    <span class="at">genus =</span> <span class="fu">map_chr</span>(parent_info, <span class="sc">~</span>.x<span class="sc">$</span>genus<span class="sc">$</span>name),</span>
<span id="cb36-1160"><a href="#cb36-1160" aria-hidden="true" tabindex="-1"></a>    <span class="at">family_taxid =</span> <span class="fu">map_dbl</span>(parent_info, <span class="sc">~</span>.x<span class="sc">$</span>family<span class="sc">$</span>taxid),</span>
<span id="cb36-1161"><a href="#cb36-1161" aria-hidden="true" tabindex="-1"></a>    <span class="at">family =</span> <span class="fu">map_chr</span>(parent_info, <span class="sc">~</span>.x<span class="sc">$</span>family<span class="sc">$</span>name)</span>
<span id="cb36-1162"><a href="#cb36-1162" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb36-1163"><a href="#cb36-1163" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>parent_info)</span>
<span id="cb36-1164"><a href="#cb36-1164" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1165"><a href="#cb36-1165" aria-hidden="true" tabindex="-1"></a><span class="co"># Join with genus counts</span></span>
<span id="cb36-1166"><a href="#cb36-1166" aria-hidden="true" tabindex="-1"></a>result <span class="ot">&lt;-</span> result <span class="sc">%&gt;%</span></span>
<span id="cb36-1167"><a href="#cb36-1167" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(</span>
<span id="cb36-1168"><a href="#cb36-1168" aria-hidden="true" tabindex="-1"></a>    hv_genus_collate <span class="sc">%&gt;%</span> <span class="fu">select</span>(taxid, genus_n_reads_tot, genus_ra_reads_tot),</span>
<span id="cb36-1169"><a href="#cb36-1169" aria-hidden="true" tabindex="-1"></a>    <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;genus_taxid&quot;</span> <span class="ot">=</span> <span class="st">&quot;taxid&quot;</span>)</span>
<span id="cb36-1170"><a href="#cb36-1170" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb36-1171"><a href="#cb36-1171" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1172"><a href="#cb36-1172" aria-hidden="true" tabindex="-1"></a><span class="co"># Join with family counts</span></span>
<span id="cb36-1173"><a href="#cb36-1173" aria-hidden="true" tabindex="-1"></a>result <span class="ot">&lt;-</span> result <span class="sc">%&gt;%</span></span>
<span id="cb36-1174"><a href="#cb36-1174" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(</span>
<span id="cb36-1175"><a href="#cb36-1175" aria-hidden="true" tabindex="-1"></a>    hv_family_collate <span class="sc">%&gt;%</span> <span class="fu">select</span>(taxid, family_n_reads_tot, family_ra_reads_tot),</span>
<span id="cb36-1176"><a href="#cb36-1176" aria-hidden="true" tabindex="-1"></a>    <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;family_taxid&quot;</span> <span class="ot">=</span> <span class="st">&quot;taxid&quot;</span>)</span>
<span id="cb36-1177"><a href="#cb36-1177" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb36-1178"><a href="#cb36-1178" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1179"><a href="#cb36-1179" aria-hidden="true" tabindex="-1"></a><span class="co"># Clean up</span></span>
<span id="cb36-1180"><a href="#cb36-1180" aria-hidden="true" tabindex="-1"></a>result <span class="ot">&lt;-</span> result <span class="sc">%&gt;%</span></span>
<span id="cb36-1181"><a href="#cb36-1181" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="fu">ends_with</span>(<span class="st">&quot;_taxid&quot;</span>))</span>
<span id="cb36-1182"><a href="#cb36-1182" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1183"><a href="#cb36-1183" aria-hidden="true" tabindex="-1"></a><span class="co">#result</span></span>
<span id="cb36-1184"><a href="#cb36-1184" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb36-1185"><a href="#cb36-1185" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1186"><a href="#cb36-1186" aria-hidden="true" tabindex="-1"></a>Each dot represents a sample, colored by viral family. The x-axis shows the relative abundance of human-infecting viruses, and the y-axis shows the species.</span>
<span id="cb36-1187"><a href="#cb36-1187" aria-hidden="true" tabindex="-1"></a>quarto-executable-code-5450563D</span>
<span id="cb36-1188"><a href="#cb36-1188" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1189"><a href="#cb36-1189" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb36-1190"><a href="#cb36-1190" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-pathogenic-viruses</span></span>
<span id="cb36-1191"><a href="#cb36-1191" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb36-1192"><a href="#cb36-1192" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 10</span></span>
<span id="cb36-1193"><a href="#cb36-1193" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb36-1194"><a href="#cb36-1194" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1195"><a href="#cb36-1195" aria-hidden="true" tabindex="-1"></a>species_family <span class="ot">&lt;-</span> result <span class="sc">%&gt;%</span> <span class="fu">select</span>(species, family) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="st">&#39;name&#39;</span> <span class="ot">=</span> <span class="st">&#39;species&#39;</span>)</span>
<span id="cb36-1196"><a href="#cb36-1196" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1197"><a href="#cb36-1197" aria-hidden="true" tabindex="-1"></a>play <span class="ot">&lt;-</span> hv_species_counts <span class="sc">%&gt;%</span> </span>
<span id="cb36-1198"><a href="#cb36-1198" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb36-1199"><a href="#cb36-1199" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by =</span> <span class="st">&#39;sample&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-1200"><a href="#cb36-1200" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(species_family, <span class="at">by =</span> <span class="st">&#39;name&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-1201"><a href="#cb36-1201" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">ifelse</span>(name <span class="sc">==</span> <span class="st">&quot;Severe acute respiratory syndrome-related coronavirus&quot;</span>, <span class="st">&quot;SARS-related coronavirus&quot;</span>, name)) <span class="sc">%&gt;%</span></span>
<span id="cb36-1202"><a href="#cb36-1202" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(family <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;Sedoreoviridae&quot;</span>, <span class="st">&quot;Parvoviridae&quot;</span>, <span class="st">&quot;Adenoviridae&quot;</span>, <span class="st">&quot;Papillomaviridae&quot;</span>, <span class="st">&quot;Polyomaviridae&quot;</span>, <span class="st">&quot;Coronaviridae&quot;</span>, <span class="st">&quot;Caliciviridae&quot;</span>, <span class="st">&quot;Picornaviridae&quot;</span>)) </span>
<span id="cb36-1203"><a href="#cb36-1203" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1204"><a href="#cb36-1204" aria-hidden="true" tabindex="-1"></a>adjusted_play <span class="ot">&lt;-</span> play <span class="sc">%&gt;%</span> </span>
<span id="cb36-1205"><a href="#cb36-1205" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span></span>
<span id="cb36-1206"><a href="#cb36-1206" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">virus_prevalence_num =</span> <span class="fu">n_distinct</span>(sample)<span class="sc">/</span><span class="fu">n_distinct</span>(libraries),</span>
<span id="cb36-1207"><a href="#cb36-1207" aria-hidden="true" tabindex="-1"></a>         <span class="at">total_reads_hv =</span> <span class="fu">sum</span>(n_reads_hv)) <span class="sc">%&gt;%</span></span>
<span id="cb36-1208"><a href="#cb36-1208" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb36-1209"><a href="#cb36-1209" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">fct_reorder</span>(name, virus_prevalence_num, <span class="at">.desc=</span><span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb36-1210"><a href="#cb36-1210" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, ra_reads_hv, family, virus_prevalence_num, total_reads_hv, n_reads_hv)</span>
<span id="cb36-1211"><a href="#cb36-1211" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1212"><a href="#cb36-1212" aria-hidden="true" tabindex="-1"></a>pal <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">&#39;Dark2&#39;</span>),<span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">&#39;Accent&#39;</span>))</span>
<span id="cb36-1213"><a href="#cb36-1213" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1214"><a href="#cb36-1214" aria-hidden="true" tabindex="-1"></a><span class="co">#, labels = label_log(digits=2)</span></span>
<span id="cb36-1215"><a href="#cb36-1215" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1216"><a href="#cb36-1216" aria-hidden="true" tabindex="-1"></a>ra_dot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(adjusted_play, <span class="fu">aes</span>(<span class="at">x =</span> ra_reads_hv, <span class="at">y=</span>name)) <span class="sc">+</span></span>
<span id="cb36-1217"><a href="#cb36-1217" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_quasirandom</span>(<span class="at">orientation =</span> <span class="st">&#39;y&#39;</span>, <span class="fu">aes</span>(<span class="at">color =</span> family)) <span class="sc">+</span></span>
<span id="cb36-1218"><a href="#cb36-1218" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> pal) <span class="sc">+</span> </span>
<span id="cb36-1219"><a href="#cb36-1219" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_log10</span>(</span>
<span id="cb36-1220"><a href="#cb36-1220" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;Relative abundance human virus reads&quot;</span>,</span>
<span id="cb36-1221"><a href="#cb36-1221" aria-hidden="true" tabindex="-1"></a>    <span class="at">labels=</span><span class="fu">label_log</span>(<span class="at">digits=</span><span class="dv">3</span>)</span>
<span id="cb36-1222"><a href="#cb36-1222" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb36-1223"><a href="#cb36-1223" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span>  <span class="st">&quot;&quot;</span>,</span>
<span id="cb36-1224"><a href="#cb36-1224" aria-hidden="true" tabindex="-1"></a>       <span class="at">color =</span> <span class="st">&#39;Viral family&#39;</span>) <span class="sc">+</span> </span>
<span id="cb36-1225"><a href="#cb36-1225" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color =</span> <span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">2</span>)) <span class="sc">+</span></span>
<span id="cb36-1226"><a href="#cb36-1226" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_light</span>() <span class="sc">+</span> </span>
<span id="cb36-1227"><a href="#cb36-1227" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(</span>
<span id="cb36-1228"><a href="#cb36-1228" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb36-1229"><a href="#cb36-1229" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>),</span>
<span id="cb36-1230"><a href="#cb36-1230" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.ticks.y =</span> <span class="fu">element_blank</span>(),<span class="at">axis.line =</span> <span class="fu">element_line</span>(<span class="at">colour =</span> <span class="st">&quot;black&quot;</span>),</span>
<span id="cb36-1231"><a href="#cb36-1231" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">14</span>),    </span>
<span id="cb36-1232"><a href="#cb36-1232" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb36-1233"><a href="#cb36-1233" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>, <span class="at">face=</span><span class="st">&quot;bold&quot;</span>),</span>
<span id="cb36-1234"><a href="#cb36-1234" aria-hidden="true" tabindex="-1"></a>    <span class="co">#legend.position = &#39;bottom&#39;,</span></span>
<span id="cb36-1235"><a href="#cb36-1235" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>),  <span class="co"># Move legend to top right</span></span>
<span id="cb36-1236"><a href="#cb36-1236" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.justification =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>),  <span class="co"># Align legend to top right</span></span>
<span id="cb36-1237"><a href="#cb36-1237" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb36-1238"><a href="#cb36-1238" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.border =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb36-1239"><a href="#cb36-1239" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.background =</span> <span class="fu">element_blank</span>())</span>
<span id="cb36-1240"><a href="#cb36-1240" aria-hidden="true" tabindex="-1"></a>ra_dot</span>
<span id="cb36-1241"><a href="#cb36-1241" aria-hidden="true" tabindex="-1"></a><span class="co">#  geom_text(aes(label = total_reads_hv, y = name), </span></span>
<span id="cb36-1242"><a href="#cb36-1242" aria-hidden="true" tabindex="-1"></a><span class="co">#            x = 1e-9, hjust = 0, vjust = 0.5, size = 3, </span></span>
<span id="cb36-1243"><a href="#cb36-1243" aria-hidden="true" tabindex="-1"></a><span class="co">#            check_overlap = TRUE)</span></span>
<span id="cb36-1244"><a href="#cb36-1244" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb36-1245"><a href="#cb36-1245" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1246"><a href="#cb36-1246" aria-hidden="true" tabindex="-1"></a>I can then take all of the viruses that we found and look up what they&#39;re responsible for and whether they&#39;re dangerous.</span>
<span id="cb36-1247"><a href="#cb36-1247" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1248"><a href="#cb36-1248" aria-hidden="true" tabindex="-1"></a>| Virus Name                                            | Common Name                                   | Pathogenic Potential                                                            |</span>
<span id="cb36-1249"><a href="#cb36-1249" aria-hidden="true" tabindex="-1"></a>|-------------------------------------------------------|-----------------------------------------------|---------------------------------------------------------------------------------|</span>
<span id="cb36-1250"><a href="#cb36-1250" aria-hidden="true" tabindex="-1"></a>| Rotavirus A                                           | Rotavirus                                     | High, causes severe diarrhea in children                                        |</span>
<span id="cb36-1251"><a href="#cb36-1251" aria-hidden="true" tabindex="-1"></a>| Enterovirus C                                         | Includes poliovirus and some coxsackieviruses | High, can cause various diseases including polio                                |</span>
<span id="cb36-1252"><a href="#cb36-1252" aria-hidden="true" tabindex="-1"></a>| Severe acute respiratory syndrome-related coronavirus | SARS coronavirus                              | High, causes severe respiratory illness                                         |</span>
<span id="cb36-1253"><a href="#cb36-1253" aria-hidden="true" tabindex="-1"></a>| Sapporo virus                                         | Sapporo virus or Sapovirus                    | Moderate, causes gastroenteritis                                                |</span>
<span id="cb36-1254"><a href="#cb36-1254" aria-hidden="true" tabindex="-1"></a>| Human mastadenovirus F                                | Adenovirus F                                  | Moderate, can cause gastroenteritis                                             |</span>
<span id="cb36-1255"><a href="#cb36-1255" aria-hidden="true" tabindex="-1"></a>| Alphapolyomavirus quintihominis                       | Not widely known                              | Low, but some polyomaviruses can cause disease in immunocompromised individuals |</span>
<span id="cb36-1256"><a href="#cb36-1256" aria-hidden="true" tabindex="-1"></a>| Alphapapillomavirus 4                                 | Human papillomavirus (HPV) type 4             | Low to moderate, can cause warts                                                |</span>
<span id="cb36-1257"><a href="#cb36-1257" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1258"><a href="#cb36-1258" aria-hidden="true" tabindex="-1"></a>None of these viruses are too suprising to see other than the Severe acute respiratory syndrome-related coronavirus, which when BLASTED, comes up as SARS-CoV-2. We believe that this is due to contamination with the COVID-19 pandemic since these samples were taken pre-pandemic.</span>
<span id="cb36-1259"><a href="#cb36-1259" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1260"><a href="#cb36-1260" aria-hidden="true" tabindex="-1"></a><span class="fu">## Relative abundance assuming perfect human read removal</span></span>
<span id="cb36-1261"><a href="#cb36-1261" aria-hidden="true" tabindex="-1"></a>quarto-executable-code-5450563D</span>
<span id="cb36-1262"><a href="#cb36-1262" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1263"><a href="#cb36-1263" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb36-1264"><a href="#cb36-1264" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: perfect-hv-removal</span></span>
<span id="cb36-1265"><a href="#cb36-1265" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb36-1266"><a href="#cb36-1266" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1267"><a href="#cb36-1267" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1268"><a href="#cb36-1268" aria-hidden="true" tabindex="-1"></a>human_reads <span class="ot">&lt;-</span> comp_assigned <span class="sc">%&gt;%</span> <span class="fu">filter</span>(classification <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;Human&quot;</span>)) <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample, <span class="at">human_reads=</span>n_reads)</span>
<span id="cb36-1269"><a href="#cb36-1269" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1270"><a href="#cb36-1270" aria-hidden="true" tabindex="-1"></a>all <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span> </span>
<span id="cb36-1271"><a href="#cb36-1271" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(human_reads, <span class="at">by=</span><span class="st">&#39;sample&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-1272"><a href="#cb36-1272" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb36-1273"><a href="#cb36-1273" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">sum_reads=</span><span class="fu">sum</span>(n_reads_hv), <span class="at">n_read_pairs=</span><span class="fu">first</span>(n_read_pairs) <span class="sc">-</span> <span class="fu">first</span>(human_reads)) <span class="sc">%&gt;%</span></span>
<span id="cb36-1274"><a href="#cb36-1274" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">ra =</span> sum_reads<span class="sc">/</span>n_read_pairs) <span class="sc">%&gt;%</span> </span>
<span id="cb36-1275"><a href="#cb36-1275" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(ra) <span class="sc">%&gt;%</span> <span class="fu">mean</span>()</span>
<span id="cb36-1276"><a href="#cb36-1276" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1277"><a href="#cb36-1277" aria-hidden="true" tabindex="-1"></a>all</span>
<span id="cb36-1278"><a href="#cb36-1278" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb36-1279"><a href="#cb36-1279" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1280"><a href="#cb36-1280" aria-hidden="true" tabindex="-1"></a>Assuming we&#39;re able to perfectly remove all human reads, the average relative abundance of known human infecting virus is $1.87 \times 10^{-4}$.</span>
<span id="cb36-1281"><a href="#cb36-1281" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1282"><a href="#cb36-1282" aria-hidden="true" tabindex="-1"></a><span class="fu"># Conclusion</span></span>
<span id="cb36-1283"><a href="#cb36-1283" aria-hidden="true" tabindex="-1"></a>There were some interesting takeways from this analysis:</span>
<span id="cb36-1284"><a href="#cb36-1284" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1285"><a href="#cb36-1285" aria-hidden="true" tabindex="-1"></a><span class="ss">1. </span>Contamination with SARS-Cov-2 seems to be a common occurence and should be accounted for when interpretting these results.</span>
<span id="cb36-1286"><a href="#cb36-1286" aria-hidden="true" tabindex="-1"></a><span class="ss">2. </span>Gasterointestinal diseases are picked up in plasma.</span>
<span id="cb36-1287"><a href="#cb36-1287" aria-hidden="true" tabindex="-1"></a><span class="ss">3. </span>Sexually transmitted infections are picked up in plasma.</span>
<span id="cb36-1288"><a href="#cb36-1288" aria-hidden="true" tabindex="-1"></a><span class="ss">4. </span>We pick up a decent amount of baceriophage (i.e. Microviridae).</span>
<span id="cb36-1289"><a href="#cb36-1289" aria-hidden="true" tabindex="-1"></a><span class="ss">5. </span>A lot of the Microviridae and Anelloviridae are largely uncharacterized (denoted by the family name followed by the abbreviation &quot;sp.&quot;).</span>
<span id="cb36-1290"><a href="#cb36-1290" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1291"><a href="#cb36-1291" aria-hidden="true" tabindex="-1"></a><span class="fu"># Appendix</span></span>
<span id="cb36-1292"><a href="#cb36-1292" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1293"><a href="#cb36-1293" aria-hidden="true" tabindex="-1"></a><span class="fu">## Human-infecting virus families, genera, and species</span></span>
<span id="cb36-1294"><a href="#cb36-1294" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1295"><a href="#cb36-1295" aria-hidden="true" tabindex="-1"></a>To get a good overview of families, genera, and species, we can look at a Sankey plot where the magnitude of relative abundance, averaged over all samples, is shown in parentheses.</span>
<span id="cb36-1296"><a href="#cb36-1296" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1297"><a href="#cb36-1297" aria-hidden="true" tabindex="-1"></a>quarto-executable-code-5450563D</span>
<span id="cb36-1298"><a href="#cb36-1298" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1299"><a href="#cb36-1299" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb36-1300"><a href="#cb36-1300" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-sankey</span></span>
<span id="cb36-1301"><a href="#cb36-1301" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 22.5</span></span>
<span id="cb36-1302"><a href="#cb36-1302" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb36-1303"><a href="#cb36-1303" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb36-1304"><a href="#cb36-1304" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1305"><a href="#cb36-1305" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to create links</span></span>
<span id="cb36-1306"><a href="#cb36-1306" aria-hidden="true" tabindex="-1"></a>create_links <span class="ot">&lt;-</span> <span class="cf">function</span>(data) {</span>
<span id="cb36-1307"><a href="#cb36-1307" aria-hidden="true" tabindex="-1"></a>  family_to_genus <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb36-1308"><a href="#cb36-1308" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(genus)) <span class="sc">%&gt;%</span></span>
<span id="cb36-1309"><a href="#cb36-1309" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(family, genus) <span class="sc">%&gt;%</span></span>
<span id="cb36-1310"><a href="#cb36-1310" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">value =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">&quot;drop&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-1311"><a href="#cb36-1311" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">source =</span> family, <span class="at">target =</span> genus)</span>
<span id="cb36-1312"><a href="#cb36-1312" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb36-1313"><a href="#cb36-1313" aria-hidden="true" tabindex="-1"></a>  genus_to_species <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb36-1314"><a href="#cb36-1314" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(genus, species) <span class="sc">%&gt;%</span></span>
<span id="cb36-1315"><a href="#cb36-1315" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">value =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">&quot;drop&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-1316"><a href="#cb36-1316" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">source =</span> genus, <span class="at">target =</span> species)</span>
<span id="cb36-1317"><a href="#cb36-1317" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1318"><a href="#cb36-1318" aria-hidden="true" tabindex="-1"></a>  family_to_species <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb36-1319"><a href="#cb36-1319" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="fu">is.na</span>(genus)) <span class="sc">%&gt;%</span></span>
<span id="cb36-1320"><a href="#cb36-1320" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(family, species) <span class="sc">%&gt;%</span></span>
<span id="cb36-1321"><a href="#cb36-1321" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">value =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">&quot;drop&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-1322"><a href="#cb36-1322" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">source =</span> family, <span class="at">target =</span> species)</span>
<span id="cb36-1323"><a href="#cb36-1323" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1324"><a href="#cb36-1324" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_rows</span>(family_to_genus, genus_to_species, family_to_species) <span class="sc">%&gt;%</span></span>
<span id="cb36-1325"><a href="#cb36-1325" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(source))</span>
<span id="cb36-1326"><a href="#cb36-1326" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb36-1327"><a href="#cb36-1327" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1328"><a href="#cb36-1328" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to create nodes</span></span>
<span id="cb36-1329"><a href="#cb36-1329" aria-hidden="true" tabindex="-1"></a>create_nodes <span class="ot">&lt;-</span> <span class="cf">function</span>(links) {</span>
<span id="cb36-1330"><a href="#cb36-1330" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>(</span>
<span id="cb36-1331"><a href="#cb36-1331" aria-hidden="true" tabindex="-1"></a>    <span class="at">name =</span> <span class="fu">c</span>(links<span class="sc">$</span>source, links<span class="sc">$</span>target) <span class="sc">%&gt;%</span> <span class="fu">unique</span>()</span>
<span id="cb36-1332"><a href="#cb36-1332" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb36-1333"><a href="#cb36-1333" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb36-1334"><a href="#cb36-1334" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1335"><a href="#cb36-1335" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to prepare data for Sankey diagram</span></span>
<span id="cb36-1336"><a href="#cb36-1336" aria-hidden="true" tabindex="-1"></a>prepare_sankey_data <span class="ot">&lt;-</span> <span class="cf">function</span>(links, nodes) {</span>
<span id="cb36-1337"><a href="#cb36-1337" aria-hidden="true" tabindex="-1"></a>  links<span class="sc">$</span>IDsource <span class="ot">&lt;-</span> <span class="fu">match</span>(links<span class="sc">$</span>source, nodes<span class="sc">$</span>name) <span class="sc">-</span> <span class="dv">1</span></span>
<span id="cb36-1338"><a href="#cb36-1338" aria-hidden="true" tabindex="-1"></a>  links<span class="sc">$</span>IDtarget <span class="ot">&lt;-</span> <span class="fu">match</span>(links<span class="sc">$</span>target, nodes<span class="sc">$</span>name) <span class="sc">-</span> <span class="dv">1</span></span>
<span id="cb36-1339"><a href="#cb36-1339" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list</span>(<span class="at">links =</span> links, <span class="at">nodes =</span> nodes)</span>
<span id="cb36-1340"><a href="#cb36-1340" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb36-1341"><a href="#cb36-1341" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1342"><a href="#cb36-1342" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to create Sankey plot</span></span>
<span id="cb36-1343"><a href="#cb36-1343" aria-hidden="true" tabindex="-1"></a>create_sankey_plot <span class="ot">&lt;-</span> <span class="cf">function</span>(sankey_data) {</span>
<span id="cb36-1344"><a href="#cb36-1344" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sankeyNetwork</span>(</span>
<span id="cb36-1345"><a href="#cb36-1345" aria-hidden="true" tabindex="-1"></a>    <span class="at">Links =</span> sankey_data<span class="sc">$</span>links, </span>
<span id="cb36-1346"><a href="#cb36-1346" aria-hidden="true" tabindex="-1"></a>    <span class="at">Nodes =</span> sankey_data<span class="sc">$</span>nodes,</span>
<span id="cb36-1347"><a href="#cb36-1347" aria-hidden="true" tabindex="-1"></a>    <span class="at">Source =</span> <span class="st">&quot;IDsource&quot;</span>, </span>
<span id="cb36-1348"><a href="#cb36-1348" aria-hidden="true" tabindex="-1"></a>    <span class="at">Target =</span> <span class="st">&quot;IDtarget&quot;</span>,</span>
<span id="cb36-1349"><a href="#cb36-1349" aria-hidden="true" tabindex="-1"></a>    <span class="at">Value =</span> <span class="st">&quot;value&quot;</span>, </span>
<span id="cb36-1350"><a href="#cb36-1350" aria-hidden="true" tabindex="-1"></a>    <span class="at">NodeID =</span> <span class="st">&quot;name&quot;</span>,</span>
<span id="cb36-1351"><a href="#cb36-1351" aria-hidden="true" tabindex="-1"></a>    <span class="at">sinksRight =</span> <span class="cn">TRUE</span>,</span>
<span id="cb36-1352"><a href="#cb36-1352" aria-hidden="true" tabindex="-1"></a>    <span class="at">nodeWidth =</span> <span class="dv">25</span>,</span>
<span id="cb36-1353"><a href="#cb36-1353" aria-hidden="true" tabindex="-1"></a>    <span class="at">fontSize =</span> <span class="dv">14</span>,</span>
<span id="cb36-1354"><a href="#cb36-1354" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb36-1355"><a href="#cb36-1355" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb36-1356"><a href="#cb36-1356" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1357"><a href="#cb36-1357" aria-hidden="true" tabindex="-1"></a>save_sankey_as_png <span class="ot">&lt;-</span> <span class="cf">function</span>(sankey_plot, <span class="at">width =</span> <span class="dv">1000</span>, <span class="at">height =</span> <span class="dv">800</span>) {</span>
<span id="cb36-1358"><a href="#cb36-1358" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Save the plot as an HTML file</span></span>
<span id="cb36-1359"><a href="#cb36-1359" aria-hidden="true" tabindex="-1"></a>  <span class="fu">saveWidget</span>(sankey_plot, <span class="fu">sprintf</span>(<span class="st">&#39;%s/sankey.html&#39;</span>,data_dir))</span>
<span id="cb36-1360"><a href="#cb36-1360" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb36-1361"><a href="#cb36-1361" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1362"><a href="#cb36-1362" aria-hidden="true" tabindex="-1"></a>format_scientific <span class="ot">&lt;-</span> <span class="cf">function</span>(x, <span class="at">digits=</span><span class="dv">2</span>) {</span>
<span id="cb36-1363"><a href="#cb36-1363" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sapply</span>(x, <span class="cf">function</span>(val) {</span>
<span id="cb36-1364"><a href="#cb36-1364" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">is.na</span>(val) <span class="sc">||</span> <span class="fu">abs</span>(val) <span class="sc">&lt;</span> <span class="fl">1e-15</span>) {</span>
<span id="cb36-1365"><a href="#cb36-1365" aria-hidden="true" tabindex="-1"></a>      <span class="fu">return</span>(<span class="st">&quot;0&quot;</span>)</span>
<span id="cb36-1366"><a href="#cb36-1366" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb36-1367"><a href="#cb36-1367" aria-hidden="true" tabindex="-1"></a>      exponent <span class="ot">&lt;-</span> <span class="fu">floor</span>(<span class="fu">log10</span>(<span class="fu">abs</span>(val)))</span>
<span id="cb36-1368"><a href="#cb36-1368" aria-hidden="true" tabindex="-1"></a>      coef <span class="ot">&lt;-</span> val <span class="sc">/</span> <span class="dv">10</span><span class="sc">^</span>exponent</span>
<span id="cb36-1369"><a href="#cb36-1369" aria-hidden="true" tabindex="-1"></a>      <span class="co">#return(sprintf(&quot;%.1f × 10^%d&quot;, round(coef, digits), exponent))</span></span>
<span id="cb36-1370"><a href="#cb36-1370" aria-hidden="true" tabindex="-1"></a>      <span class="co"># May or may not be smart, just keeping magnitude</span></span>
<span id="cb36-1371"><a href="#cb36-1371" aria-hidden="true" tabindex="-1"></a>      <span class="fu">return</span>(<span class="fu">sprintf</span>(<span class="st">&quot;10^%d&quot;</span>, exponent))</span>
<span id="cb36-1372"><a href="#cb36-1372" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb36-1373"><a href="#cb36-1373" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb36-1374"><a href="#cb36-1374" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb36-1375"><a href="#cb36-1375" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1376"><a href="#cb36-1376" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> result <span class="sc">%&gt;%</span> </span>
<span id="cb36-1377"><a href="#cb36-1377" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">c</span>(genus_n_reads_tot, genus_ra_reads_tot), <span class="sc">~</span><span class="fu">replace_na</span>(., <span class="dv">0</span>)),</span>
<span id="cb36-1378"><a href="#cb36-1378" aria-hidden="true" tabindex="-1"></a>         <span class="at">genus =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(genus), <span class="st">&quot;Unknown Genus&quot;</span>, genus)) <span class="sc">%&gt;%</span></span>
<span id="cb36-1379"><a href="#cb36-1379" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb36-1380"><a href="#cb36-1380" aria-hidden="true" tabindex="-1"></a>  <span class="at">species =</span> <span class="fu">paste0</span>(species, <span class="fu">sprintf</span>(<span class="st">&#39; (%s)&#39;</span>, <span class="fu">format_scientific</span>(species_ra_reads_tot))),</span>
<span id="cb36-1381"><a href="#cb36-1381" aria-hidden="true" tabindex="-1"></a>  <span class="at">genus =</span> <span class="fu">paste0</span>(genus, <span class="fu">sprintf</span>(<span class="st">&#39; (%s)&#39;</span>, <span class="fu">format_scientific</span>(genus_ra_reads_tot))),</span>
<span id="cb36-1382"><a href="#cb36-1382" aria-hidden="true" tabindex="-1"></a>  <span class="at">family =</span> <span class="fu">paste0</span>(family, <span class="fu">sprintf</span>(<span class="st">&#39; (%s)&#39;</span>, <span class="fu">format_scientific</span>(family_ra_reads_tot)))</span>
<span id="cb36-1383"><a href="#cb36-1383" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb36-1384"><a href="#cb36-1384" aria-hidden="true" tabindex="-1"></a>links <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">create_links</span>(data))</span>
<span id="cb36-1385"><a href="#cb36-1385" aria-hidden="true" tabindex="-1"></a>nodes <span class="ot">&lt;-</span> <span class="fu">create_nodes</span>(links)</span>
<span id="cb36-1386"><a href="#cb36-1386" aria-hidden="true" tabindex="-1"></a>sankey_data <span class="ot">&lt;-</span> <span class="fu">prepare_sankey_data</span>(links, nodes)</span>
<span id="cb36-1387"><a href="#cb36-1387" aria-hidden="true" tabindex="-1"></a>sankey <span class="ot">&lt;-</span> <span class="fu">create_sankey_plot</span>(sankey_data)</span>
<span id="cb36-1388"><a href="#cb36-1388" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-1389"><a href="#cb36-1389" aria-hidden="true" tabindex="-1"></a>sankey</span>
<span id="cb36-1390"><a href="#cb36-1390" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span></code></pre></div>
</div>
</section>
</section>

</main> <!-- /main -->
<script id = "quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->

</body>

</html>
